// @react-spectrum/listbox/dist/main.js
"use client";

var $db2f31a656e206dc$exports = require("./ar-AE.main.js");
var $5fc5de12245ee37a$exports = require("./bg-BG.main.js");
var $c6716434befdc578$exports = require("./cs-CZ.main.js");
var $aff0838a0af4442c$exports = require("./da-DK.main.js");
var $e23661cbd2fcf9d9$exports = require("./de-DE.main.js");
var $0665dd51687e6f20$exports = require("./el-GR.main.js");
var $9db92f31bd81178a$exports = require("./en-US.main.js");
var $b85a375140bc142b$exports = require("./es-ES.main.js");
var $9d80b9bee9643813$exports = require("./et-EE.main.js");
var $32fac20f569c89b2$exports = require("./fi-FI.main.js");
var $1ae83646fec0a00f$exports = require("./fr-FR.main.js");
var $1e59e08ce00d3a62$exports = require("./he-IL.main.js");
var $e55a93edef01c480$exports = require("./hr-HR.main.js");
var $c479dd99077bc32f$exports = require("./hu-HU.main.js");
var $93e538a0db81bf51$exports = require("./it-IT.main.js");
var $ba01846c1b5c1a2b$exports = require("./ja-JP.main.js");
var $9efc98866ffb3d03$exports = require("./ko-KR.main.js");
var $72e1b4485a8dff00$exports = require("./lt-LT.main.js");
var $9d82f7842d0956d5$exports = require("./lv-LV.main.js");
var $2eeedcf868421556$exports = require("./nb-NO.main.js");
var $ce5a0f667a7b7bff$exports = require("./nl-NL.main.js");
var $9700a49ad8081602$exports = require("./pl-PL.main.js");
var $65c2d0648e10b51a$exports = require("./pt-BR.main.js");
var $ce0c2fb246a5350d$exports = require("./pt-PT.main.js");
var $8af4a376429f1f9a$exports = require("./ro-RO.main.js");
var $99024a2f9ca1b2c9$exports = require("./ru-RU.main.js");
var $304201c7620897f0$exports = require("./sk-SK.main.js");
var $713bd39bd769c331$exports = require("./sl-SI.main.js");
var $9864766ffc224b5e$exports = require("./sr-SP.main.js");
var $b3456f989b077e88$exports = require("./sv-SE.main.js");
var $5255c7a45247b8dc$exports = require("./tr-TR.main.js");
var $8ee6f6af8918bfbf$exports = require("./uk-UA.main.js");
var $024aed1a793198cd$exports = require("./zh-CN.main.js");
var $be53ff8daaa52db0$exports = require("./zh-TW.main.js");
require("./main.css");
var $3Fwf2$reactstatelycollections = require("@react-stately/collections");
var $3Fwf2$react = require("react");
var $3Fwf2$reactspectrumutils = require("@react-spectrum/utils");
var $3Fwf2$reactstatelylist = require("@react-stately/list");
var $3Fwf2$reactarialistbox = require("@react-aria/listbox");
var $3Fwf2$reactariafocus = require("@react-aria/focus");
var $3Fwf2$reactstatelylayout = require("@react-stately/layout");
var $3Fwf2$reactariautils = require("@react-aria/utils");
var $3Fwf2$reactspectrumprogress = require("@react-spectrum/progress");
var $3Fwf2$reactariai18n = require("@react-aria/i18n");
var $3Fwf2$reactspectrumprovider = require("@react-spectrum/provider");
var $3Fwf2$reactariavirtualizer = require("@react-aria/virtualizer");
var $3Fwf2$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $3Fwf2$reactspectrumlayout = require("@react-spectrum/layout");
var $3Fwf2$reactariainteractions = require("@react-aria/interactions");
var $3Fwf2$reactspectrumtext = require("@react-spectrum/text");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "ListBox", () => $582d86e45be55635$export$41f133550aa26f48);
$parcel$export(module.exports, "useListBoxLayout", () => $60eb4b34c53310cb$export$25768ea656ae32a7);
$parcel$export(module.exports, "ListBoxBase", () => $60eb4b34c53310cb$export$1afdcf349979fb7e);
$parcel$export(module.exports, "Item", () => $3Fwf2$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $3Fwf2$reactstatelycollections.Section);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $3d9fe9b46e2d649a$exports = {};


































$3d9fe9b46e2d649a$exports = {
    "ar-AE": $db2f31a656e206dc$exports,
    "bg-BG": $5fc5de12245ee37a$exports,
    "cs-CZ": $c6716434befdc578$exports,
    "da-DK": $aff0838a0af4442c$exports,
    "de-DE": $e23661cbd2fcf9d9$exports,
    "el-GR": $0665dd51687e6f20$exports,
    "en-US": $9db92f31bd81178a$exports,
    "es-ES": $b85a375140bc142b$exports,
    "et-EE": $9d80b9bee9643813$exports,
    "fi-FI": $32fac20f569c89b2$exports,
    "fr-FR": $1ae83646fec0a00f$exports,
    "he-IL": $1e59e08ce00d3a62$exports,
    "hr-HR": $e55a93edef01c480$exports,
    "hu-HU": $c479dd99077bc32f$exports,
    "it-IT": $93e538a0db81bf51$exports,
    "ja-JP": $ba01846c1b5c1a2b$exports,
    "ko-KR": $9efc98866ffb3d03$exports,
    "lt-LT": $72e1b4485a8dff00$exports,
    "lv-LV": $9d82f7842d0956d5$exports,
    "nb-NO": $2eeedcf868421556$exports,
    "nl-NL": $ce5a0f667a7b7bff$exports,
    "pl-PL": $9700a49ad8081602$exports,
    "pt-BR": $65c2d0648e10b51a$exports,
    "pt-PT": $ce0c2fb246a5350d$exports,
    "ro-RO": $8af4a376429f1f9a$exports,
    "ru-RU": $99024a2f9ca1b2c9$exports,
    "sk-SK": $304201c7620897f0$exports,
    "sl-SI": $713bd39bd769c331$exports,
    "sr-SP": $9864766ffc224b5e$exports,
    "sv-SE": $b3456f989b077e88$exports,
    "tr-TR": $5255c7a45247b8dc$exports,
    "uk-UA": $8ee6f6af8918bfbf$exports,
    "zh-CN": $024aed1a793198cd$exports,
    "zh-TW": $be53ff8daaa52db0$exports
};


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $611e93c7b7c0413c$export$7ff8f37d2d81a48d = (0, ($parcel$interopDefault($3Fwf2$react))).createContext(null);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







var $3deee0fd2b46e64f$exports = {};

$parcel$export($3deee0fd2b46e64f$exports, "checkmark", () => $3deee0fd2b46e64f$export$292abbf31ed842a, (v) => $3deee0fd2b46e64f$export$292abbf31ed842a = v);
$parcel$export($3deee0fd2b46e64f$exports, "chevron", () => $3deee0fd2b46e64f$export$980c1089c0604ea3, (v) => $3deee0fd2b46e64f$export$980c1089c0604ea3 = v);
$parcel$export($3deee0fd2b46e64f$exports, "description", () => $3deee0fd2b46e64f$export$6d59db4903f20f7d, (v) => $3deee0fd2b46e64f$export$6d59db4903f20f7d = v);
$parcel$export($3deee0fd2b46e64f$exports, "end", () => $3deee0fd2b46e64f$export$bd5df0f255a350f8, (v) => $3deee0fd2b46e64f$export$bd5df0f255a350f8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "focus-ring", () => $3deee0fd2b46e64f$export$f39a09f249340e2a, (v) => $3deee0fd2b46e64f$export$f39a09f249340e2a = v);
$parcel$export($3deee0fd2b46e64f$exports, "i18nFontFamily", () => $3deee0fd2b46e64f$export$8c4ee2c50c22c514, (v) => $3deee0fd2b46e64f$export$8c4ee2c50c22c514 = v);
$parcel$export($3deee0fd2b46e64f$exports, "icon", () => $3deee0fd2b46e64f$export$1ca1ec8b29a4ce27, (v) => $3deee0fd2b46e64f$export$1ca1ec8b29a4ce27 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-active", () => $3deee0fd2b46e64f$export$20fd0f7cd4e6112f, (v) => $3deee0fd2b46e64f$export$20fd0f7cd4e6112f = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-disabled", () => $3deee0fd2b46e64f$export$d35bc1e505d1ebbf, (v) => $3deee0fd2b46e64f$export$d35bc1e505d1ebbf = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-expanded", () => $3deee0fd2b46e64f$export$911ebb89bba4b634, (v) => $3deee0fd2b46e64f$export$911ebb89bba4b634 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-focused", () => $3deee0fd2b46e64f$export$e7dc768d35940237, (v) => $3deee0fd2b46e64f$export$e7dc768d35940237 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-highlighted", () => $3deee0fd2b46e64f$export$451c613d0772eb99, (v) => $3deee0fd2b46e64f$export$451c613d0772eb99 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-hovered", () => $3deee0fd2b46e64f$export$b8813cd5d7824ce7, (v) => $3deee0fd2b46e64f$export$b8813cd5d7824ce7 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-open", () => $3deee0fd2b46e64f$export$a9781837241c946d, (v) => $3deee0fd2b46e64f$export$a9781837241c946d = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-selectable", () => $3deee0fd2b46e64f$export$fbd22ba224473193, (v) => $3deee0fd2b46e64f$export$fbd22ba224473193 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-selected", () => $3deee0fd2b46e64f$export$1e0fb04f31d3c22a, (v) => $3deee0fd2b46e64f$export$1e0fb04f31d3c22a = v);
$parcel$export($3deee0fd2b46e64f$exports, "keyboard", () => $3deee0fd2b46e64f$export$cfdd767f5a513509, (v) => $3deee0fd2b46e64f$export$cfdd767f5a513509 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-FocusRing-ring", () => $3deee0fd2b46e64f$export$4109102f950813a6, (v) => $3deee0fd2b46e64f$export$4109102f950813a6 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-FocusRing", () => $3deee0fd2b46e64f$export$24c7f46a6e3605dd, (v) => $3deee0fd2b46e64f$export$24c7f46a6e3605dd = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-FocusRing--quiet", () => $3deee0fd2b46e64f$export$2927016961429360, (v) => $3deee0fd2b46e64f$export$2927016961429360 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Icon", () => $3deee0fd2b46e64f$export$d374b04f30360026, (v) => $3deee0fd2b46e64f$export$d374b04f30360026 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu", () => $3deee0fd2b46e64f$export$fae6af6decd50ac5, (v) => $3deee0fd2b46e64f$export$fae6af6decd50ac5 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-avatar", () => $3deee0fd2b46e64f$export$944daf9b7d423a02, (v) => $3deee0fd2b46e64f$export$944daf9b7d423a02 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-checkmark", () => $3deee0fd2b46e64f$export$8e15a726e4f7e62d, (v) => $3deee0fd2b46e64f$export$8e15a726e4f7e62d = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-chevron", () => $3deee0fd2b46e64f$export$ca45e0db366b8e6c, (v) => $3deee0fd2b46e64f$export$ca45e0db366b8e6c = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-description", () => $3deee0fd2b46e64f$export$36f1275051e39ead, (v) => $3deee0fd2b46e64f$export$36f1275051e39ead = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-divider", () => $3deee0fd2b46e64f$export$e892026f534efed8, (v) => $3deee0fd2b46e64f$export$e892026f534efed8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-end", () => $3deee0fd2b46e64f$export$926a1bb74ed65667, (v) => $3deee0fd2b46e64f$export$926a1bb74ed65667 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-icon", () => $3deee0fd2b46e64f$export$cbbae63ef9e44c7a, (v) => $3deee0fd2b46e64f$export$cbbae63ef9e44c7a = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-item", () => $3deee0fd2b46e64f$export$45faad328e6c532b, (v) => $3deee0fd2b46e64f$export$45faad328e6c532b = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemGrid", () => $3deee0fd2b46e64f$export$ee5f52784f57f9c8, (v) => $3deee0fd2b46e64f$export$ee5f52784f57f9c8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemIcon", () => $3deee0fd2b46e64f$export$f431001e1bb54633, (v) => $3deee0fd2b46e64f$export$f431001e1bb54633 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemLabel", () => $3deee0fd2b46e64f$export$1aa6419c5c650567, (v) => $3deee0fd2b46e64f$export$1aa6419c5c650567 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemLabel--wrapping", () => $3deee0fd2b46e64f$export$465092164a8e1d98, (v) => $3deee0fd2b46e64f$export$465092164a8e1d98 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-keyboard", () => $3deee0fd2b46e64f$export$4f9817a7dff6ba0f, (v) => $3deee0fd2b46e64f$export$4f9817a7dff6ba0f = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-popover", () => $3deee0fd2b46e64f$export$d959f30786125b, (v) => $3deee0fd2b46e64f$export$d959f30786125b = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-section--isFirst", () => $3deee0fd2b46e64f$export$dae331c15950f1e, (v) => $3deee0fd2b46e64f$export$dae331c15950f1e = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-section--isLast", () => $3deee0fd2b46e64f$export$9778864c5b622fbc, (v) => $3deee0fd2b46e64f$export$9778864c5b622fbc = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-section--noHeading", () => $3deee0fd2b46e64f$export$8b61ec49b96ad17e, (v) => $3deee0fd2b46e64f$export$8b61ec49b96ad17e = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-sectionHeading", () => $3deee0fd2b46e64f$export$c20ea4b35020fbdd, (v) => $3deee0fd2b46e64f$export$c20ea4b35020fbdd = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-subdialog", () => $3deee0fd2b46e64f$export$d7e47bfca92b5744, (v) => $3deee0fd2b46e64f$export$d7e47bfca92b5744 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-wrapper", () => $3deee0fd2b46e64f$export$76d36e55740ad742, (v) => $3deee0fd2b46e64f$export$76d36e55740ad742 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-wrapper--isMobile", () => $3deee0fd2b46e64f$export$c873119be41fdb72, (v) => $3deee0fd2b46e64f$export$c873119be41fdb72 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Submenu-heading", () => $3deee0fd2b46e64f$export$c830d2588e34dae1, (v) => $3deee0fd2b46e64f$export$c830d2588e34dae1 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Submenu-headingWrapper", () => $3deee0fd2b46e64f$export$d3fc690f61339a08, (v) => $3deee0fd2b46e64f$export$d3fc690f61339a08 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Submenu-popover", () => $3deee0fd2b46e64f$export$fe967dac6207a986, (v) => $3deee0fd2b46e64f$export$fe967dac6207a986 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Submenu-wrapper", () => $3deee0fd2b46e64f$export$158cbd6607d36c4e, (v) => $3deee0fd2b46e64f$export$158cbd6607d36c4e = v);
$parcel$export($3deee0fd2b46e64f$exports, "text", () => $3deee0fd2b46e64f$export$6f093cfa640b7166, (v) => $3deee0fd2b46e64f$export$6f093cfa640b7166 = v);
var $3deee0fd2b46e64f$export$292abbf31ed842a;
var $3deee0fd2b46e64f$export$980c1089c0604ea3;
var $3deee0fd2b46e64f$export$6d59db4903f20f7d;
var $3deee0fd2b46e64f$export$bd5df0f255a350f8;
var $3deee0fd2b46e64f$export$f39a09f249340e2a;
var $3deee0fd2b46e64f$export$8c4ee2c50c22c514;
var $3deee0fd2b46e64f$export$1ca1ec8b29a4ce27;
var $3deee0fd2b46e64f$export$20fd0f7cd4e6112f;
var $3deee0fd2b46e64f$export$d35bc1e505d1ebbf;
var $3deee0fd2b46e64f$export$911ebb89bba4b634;
var $3deee0fd2b46e64f$export$e7dc768d35940237;
var $3deee0fd2b46e64f$export$451c613d0772eb99;
var $3deee0fd2b46e64f$export$b8813cd5d7824ce7;
var $3deee0fd2b46e64f$export$a9781837241c946d;
var $3deee0fd2b46e64f$export$fbd22ba224473193;
var $3deee0fd2b46e64f$export$1e0fb04f31d3c22a;
var $3deee0fd2b46e64f$export$cfdd767f5a513509;
var $3deee0fd2b46e64f$export$4109102f950813a6;
var $3deee0fd2b46e64f$export$24c7f46a6e3605dd;
var $3deee0fd2b46e64f$export$2927016961429360;
var $3deee0fd2b46e64f$export$d374b04f30360026;
var $3deee0fd2b46e64f$export$fae6af6decd50ac5;
var $3deee0fd2b46e64f$export$944daf9b7d423a02;
var $3deee0fd2b46e64f$export$8e15a726e4f7e62d;
var $3deee0fd2b46e64f$export$ca45e0db366b8e6c;
var $3deee0fd2b46e64f$export$36f1275051e39ead;
var $3deee0fd2b46e64f$export$e892026f534efed8;
var $3deee0fd2b46e64f$export$926a1bb74ed65667;
var $3deee0fd2b46e64f$export$cbbae63ef9e44c7a;
var $3deee0fd2b46e64f$export$45faad328e6c532b;
var $3deee0fd2b46e64f$export$ee5f52784f57f9c8;
var $3deee0fd2b46e64f$export$f431001e1bb54633;
var $3deee0fd2b46e64f$export$1aa6419c5c650567;
var $3deee0fd2b46e64f$export$465092164a8e1d98;
var $3deee0fd2b46e64f$export$4f9817a7dff6ba0f;
var $3deee0fd2b46e64f$export$d959f30786125b;
var $3deee0fd2b46e64f$export$dae331c15950f1e;
var $3deee0fd2b46e64f$export$9778864c5b622fbc;
var $3deee0fd2b46e64f$export$8b61ec49b96ad17e;
var $3deee0fd2b46e64f$export$c20ea4b35020fbdd;
var $3deee0fd2b46e64f$export$d7e47bfca92b5744;
var $3deee0fd2b46e64f$export$76d36e55740ad742;
var $3deee0fd2b46e64f$export$c873119be41fdb72;
var $3deee0fd2b46e64f$export$c830d2588e34dae1;
var $3deee0fd2b46e64f$export$d3fc690f61339a08;
var $3deee0fd2b46e64f$export$fe967dac6207a986;
var $3deee0fd2b46e64f$export$158cbd6607d36c4e;
var $3deee0fd2b46e64f$export$6f093cfa640b7166;
$3deee0fd2b46e64f$export$292abbf31ed842a = `Q7FggG_checkmark`;
$3deee0fd2b46e64f$export$980c1089c0604ea3 = `Q7FggG_chevron`;
$3deee0fd2b46e64f$export$6d59db4903f20f7d = `Q7FggG_description`;
$3deee0fd2b46e64f$export$bd5df0f255a350f8 = `Q7FggG_end`;
$3deee0fd2b46e64f$export$f39a09f249340e2a = `Q7FggG_focus-ring`;
$3deee0fd2b46e64f$export$8c4ee2c50c22c514 = `Q7FggG_i18nFontFamily`;
$3deee0fd2b46e64f$export$1ca1ec8b29a4ce27 = `Q7FggG_icon`;
$3deee0fd2b46e64f$export$20fd0f7cd4e6112f = `Q7FggG_is-active`;
$3deee0fd2b46e64f$export$d35bc1e505d1ebbf = `Q7FggG_is-disabled`;
$3deee0fd2b46e64f$export$911ebb89bba4b634 = `Q7FggG_is-expanded`;
$3deee0fd2b46e64f$export$e7dc768d35940237 = `Q7FggG_is-focused`;
$3deee0fd2b46e64f$export$451c613d0772eb99 = `Q7FggG_is-highlighted`;
$3deee0fd2b46e64f$export$b8813cd5d7824ce7 = `Q7FggG_is-hovered`;
$3deee0fd2b46e64f$export$a9781837241c946d = `Q7FggG_is-open`;
$3deee0fd2b46e64f$export$fbd22ba224473193 = `Q7FggG_is-selectable`;
$3deee0fd2b46e64f$export$1e0fb04f31d3c22a = `Q7FggG_is-selected`;
$3deee0fd2b46e64f$export$cfdd767f5a513509 = `Q7FggG_keyboard`;
$3deee0fd2b46e64f$export$4109102f950813a6 = `Q7FggG_spectrum-FocusRing-ring`;
$3deee0fd2b46e64f$export$24c7f46a6e3605dd = `Q7FggG_spectrum-FocusRing ${$3deee0fd2b46e64f$export$4109102f950813a6}`;
$3deee0fd2b46e64f$export$2927016961429360 = `Q7FggG_spectrum-FocusRing--quiet`;
$3deee0fd2b46e64f$export$d374b04f30360026 = `Q7FggG_spectrum-Icon`;
$3deee0fd2b46e64f$export$fae6af6decd50ac5 = `Q7FggG_spectrum-Menu`;
$3deee0fd2b46e64f$export$944daf9b7d423a02 = `Q7FggG_spectrum-Menu-avatar`;
$3deee0fd2b46e64f$export$8e15a726e4f7e62d = `Q7FggG_spectrum-Menu-checkmark`;
$3deee0fd2b46e64f$export$ca45e0db366b8e6c = `Q7FggG_spectrum-Menu-chevron`;
$3deee0fd2b46e64f$export$36f1275051e39ead = `Q7FggG_spectrum-Menu-description`;
$3deee0fd2b46e64f$export$e892026f534efed8 = `Q7FggG_spectrum-Menu-divider`;
$3deee0fd2b46e64f$export$926a1bb74ed65667 = `Q7FggG_spectrum-Menu-end`;
$3deee0fd2b46e64f$export$cbbae63ef9e44c7a = `Q7FggG_spectrum-Menu-icon`;
$3deee0fd2b46e64f$export$45faad328e6c532b = `Q7FggG_spectrum-Menu-item`;
$3deee0fd2b46e64f$export$ee5f52784f57f9c8 = `Q7FggG_spectrum-Menu-itemGrid`;
$3deee0fd2b46e64f$export$f431001e1bb54633 = `Q7FggG_spectrum-Menu-itemIcon`;
$3deee0fd2b46e64f$export$1aa6419c5c650567 = `Q7FggG_spectrum-Menu-itemLabel`;
$3deee0fd2b46e64f$export$465092164a8e1d98 = `Q7FggG_spectrum-Menu-itemLabel--wrapping`;
$3deee0fd2b46e64f$export$4f9817a7dff6ba0f = `Q7FggG_spectrum-Menu-keyboard`;
$3deee0fd2b46e64f$export$d959f30786125b = `Q7FggG_spectrum-Menu-popover`;
$3deee0fd2b46e64f$export$dae331c15950f1e = `Q7FggG_spectrum-Menu-section--isFirst`;
$3deee0fd2b46e64f$export$9778864c5b622fbc = `Q7FggG_spectrum-Menu-section--isLast`;
$3deee0fd2b46e64f$export$8b61ec49b96ad17e = `Q7FggG_spectrum-Menu-section--noHeading`;
$3deee0fd2b46e64f$export$c20ea4b35020fbdd = `Q7FggG_spectrum-Menu-sectionHeading`;
$3deee0fd2b46e64f$export$d7e47bfca92b5744 = `Q7FggG_spectrum-Menu-subdialog`;
$3deee0fd2b46e64f$export$76d36e55740ad742 = `Q7FggG_spectrum-Menu-wrapper`;
$3deee0fd2b46e64f$export$c873119be41fdb72 = `Q7FggG_spectrum-Menu-wrapper--isMobile`;
$3deee0fd2b46e64f$export$c830d2588e34dae1 = `Q7FggG_spectrum-Submenu-heading`;
$3deee0fd2b46e64f$export$d3fc690f61339a08 = `Q7FggG_spectrum-Submenu-headingWrapper`;
$3deee0fd2b46e64f$export$fe967dac6207a986 = `Q7FggG_spectrum-Submenu-popover`;
$3deee0fd2b46e64f$export$158cbd6607d36c4e = `Q7FggG_spectrum-Submenu-wrapper`;
$3deee0fd2b46e64f$export$6f093cfa640b7166 = `Q7FggG_text`;




function $2205d250ee1cb1ac$export$feb3b6b552c14a12(props) {
    let { item: item, shouldSelectOnPressUp: shouldSelectOnPressUp, shouldFocusOnHover: shouldFocusOnHover, shouldUseVirtualFocus: shouldUseVirtualFocus } = props;
    let { rendered: rendered, key: key } = item;
    let ElementType = item.props.href ? "a" : "div";
    let state = (0, $3Fwf2$react.useContext)((0, $611e93c7b7c0413c$export$7ff8f37d2d81a48d));
    let ref = (0, $3Fwf2$react.useRef)();
    let { optionProps: optionProps, labelProps: labelProps, descriptionProps: descriptionProps, isSelected: isSelected, isDisabled: isDisabled, isFocused: isFocused } = (0, $3Fwf2$reactarialistbox.useOption)({
        "aria-label": item["aria-label"],
        key: key,
        shouldSelectOnPressUp: shouldSelectOnPressUp,
        shouldFocusOnHover: shouldFocusOnHover,
        isVirtualized: true,
        shouldUseVirtualFocus: shouldUseVirtualFocus
    }, state, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $3Fwf2$reactariainteractions.useHover)({
        ...props,
        isDisabled: isDisabled
    });
    let contents = typeof rendered === "string" ? /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactspectrumtext.Text), null, rendered) : rendered;
    let isKeyboardModality = (0, $3Fwf2$reactariainteractions.isFocusVisible)();
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactariafocus.FocusRing), {
        focusRingClass: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement(ElementType, {
        ...(0, $3Fwf2$reactariautils.mergeProps)(optionProps, shouldFocusOnHover ? {} : hoverProps),
        ref: ref,
        className: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-item", {
            // If using virtual focus, apply focused styles to the item when the user is interacting with keyboard modality
            "is-focused": shouldUseVirtualFocus && isFocused && isKeyboardModality,
            "is-disabled": isDisabled,
            "is-selected": isSelected,
            "is-selectable": state.selectionManager.selectionMode !== "none",
            // When shouldFocusOnHover is false, apply hover styles both when hovered with the mouse.
            // Otherwise, apply hover styles when focused using non-keyboard modality.
            "is-hovered": isHovered && !shouldFocusOnHover || isFocused && !isKeyboardModality
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactspectrumlayout.Grid), {
        UNSAFE_className: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-itemGrid")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactspectrumutils.ClearSlots), null, /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactspectrumutils.SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-itemLabel"],
                ...labelProps
            },
            icon: {
                size: "S",
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-icon"]
            },
            avatar: {
                size: "avatar-size-100",
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-avatar"]
            },
            description: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-description"],
                ...descriptionProps
            }
        }
    }, contents, isSelected && /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, ($parcel$interopDefault($3Fwf2$spectrumiconsuiCheckmarkMedium))), {
        slot: "checkmark",
        UNSAFE_className: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-checkmark")
    }))))));
}


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






function $0f5fd8483f48ef09$export$dca12b0bb56e4fc(props) {
    let { children: children, layoutInfo: layoutInfo, headerLayoutInfo: headerLayoutInfo, virtualizer: virtualizer, item: item } = props;
    let { headingProps: headingProps, groupProps: groupProps } = (0, $3Fwf2$reactarialistbox.useListBoxSection)({
        heading: item.rendered,
        "aria-label": item["aria-label"]
    });
    let headerRef = (0, $3Fwf2$react.useRef)();
    (0, $3Fwf2$reactariavirtualizer.useVirtualizerItem)({
        layoutInfo: headerLayoutInfo,
        virtualizer: virtualizer,
        ref: headerRef
    });
    let { direction: direction } = (0, $3Fwf2$reactariai18n.useLocale)();
    let state = (0, $3Fwf2$react.useContext)((0, $611e93c7b7c0413c$export$7ff8f37d2d81a48d));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$react.Fragment), null, /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement("div", {
        role: "presentation",
        ref: headerRef,
        style: (0, $3Fwf2$reactariavirtualizer.layoutInfoToStyle)(headerLayoutInfo, direction)
    }, item.key !== state.collection.getFirstKey() && /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement("div", {
        role: "presentation",
        className: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-divider")
    }), item.rendered && /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement("div", {
        ...headingProps,
        className: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-sectionHeading")
    }, item.rendered)), /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement("div", {
        ...groupProps,
        style: (0, $3Fwf2$reactariavirtualizer.layoutInfoToStyle)(layoutInfo, direction),
        className: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu")
    }, children));
}










function $60eb4b34c53310cb$export$25768ea656ae32a7(state, isLoading) {
    let { scale: scale } = (0, $3Fwf2$reactspectrumprovider.useProvider)();
    let collator = (0, $3Fwf2$reactariai18n.useCollator)({
        usage: "search",
        sensitivity: "base"
    });
    let layout = (0, $3Fwf2$react.useMemo)(()=>new (0, $3Fwf2$reactstatelylayout.ListLayout)({
            estimatedRowHeight: scale === "large" ? 48 : 32,
            estimatedHeadingHeight: scale === "large" ? 33 : 26,
            padding: scale === "large" ? 5 : 4,
            loaderHeight: 40,
            placeholderHeight: scale === "large" ? 48 : 32,
            collator: collator
        }), [
        collator,
        scale
    ]);
    layout.collection = state.collection;
    layout.disabledKeys = state.disabledKeys;
    (0, $3Fwf2$reactariautils.useLayoutEffect)(()=>{
        // Sync loading state into the layout.
        if (layout.isLoading !== isLoading) {
            var _layout_virtualizer;
            layout.isLoading = isLoading;
            (_layout_virtualizer = layout.virtualizer) === null || _layout_virtualizer === void 0 ? void 0 : _layout_virtualizer.relayoutNow();
        }
    }, [
        layout,
        isLoading
    ]);
    return layout;
}
/** @private */ function $60eb4b34c53310cb$var$ListBoxBase(props, ref) {
    let { layout: layout, state: state, shouldSelectOnPressUp: shouldSelectOnPressUp, focusOnPointerEnter: focusOnPointerEnter, shouldUseVirtualFocus: shouldUseVirtualFocus, domProps: domProps = {}, transitionDuration: transitionDuration = 0, onScroll: onScroll } = props;
    let { listBoxProps: listBoxProps } = (0, $3Fwf2$reactarialistbox.useListBox)({
        ...props,
        keyboardDelegate: layout,
        isVirtualized: true
    }, state, ref);
    let { styleProps: styleProps } = (0, $3Fwf2$reactspectrumutils.useStyleProps)(props);
    let stringFormatter = (0, $3Fwf2$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($3d9fe9b46e2d649a$exports))), "@react-spectrum/listbox");
    let renderWrapper = (parent, reusableView, children, renderChildren)=>{
        if (reusableView.viewType === "section") return /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $0f5fd8483f48ef09$export$dca12b0bb56e4fc), {
            key: reusableView.key,
            item: reusableView.content,
            layoutInfo: reusableView.layoutInfo,
            virtualizer: reusableView.virtualizer,
            headerLayoutInfo: children.find((c)=>c.viewType === "header").layoutInfo
        }, renderChildren(children.filter((c)=>c.viewType === "item")));
        return /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactariavirtualizer.VirtualizerItem), {
            key: reusableView.key,
            layoutInfo: reusableView.layoutInfo,
            virtualizer: reusableView.virtualizer,
            parent: parent === null || parent === void 0 ? void 0 : parent.layoutInfo
        }, reusableView.rendered);
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $611e93c7b7c0413c$export$7ff8f37d2d81a48d).Provider, {
        value: state
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactariafocus.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactariavirtualizer.Virtualizer), {
        ...styleProps,
        ...(0, $3Fwf2$reactariautils.mergeProps)(listBoxProps, domProps),
        ref: ref,
        focusedKey: state.selectionManager.focusedKey,
        autoFocus: !!props.autoFocus,
        sizeToFit: "height",
        scrollDirection: "vertical",
        className: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu", styleProps.className),
        layout: layout,
        collection: state.collection,
        renderWrapper: renderWrapper,
        transitionDuration: transitionDuration,
        isLoading: props.isLoading,
        onLoadMore: props.onLoadMore,
        shouldUseVirtualFocus: shouldUseVirtualFocus,
        onScroll: onScroll
    }, (type, item)=>{
        if (type === "item") return /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $2205d250ee1cb1ac$export$feb3b6b552c14a12), {
            item: item,
            shouldSelectOnPressUp: shouldSelectOnPressUp,
            shouldFocusOnHover: focusOnPointerEnter,
            shouldUseVirtualFocus: shouldUseVirtualFocus
        });
        else if (type === "loader") return(// aria-selected isn't needed here since this option is not selectable.
        // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
        /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement("div", {
            role: "option",
            style: {
                display: "flex",
                alignItems: "center",
                justifyContent: "center",
                height: "100%"
            }
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $3Fwf2$reactspectrumprogress.ProgressCircle), {
            isIndeterminate: true,
            size: "S",
            "aria-label": state.collection.size > 0 ? stringFormatter.format("loadingMore") : stringFormatter.format("loading"),
            UNSAFE_className: (0, $3Fwf2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Dropdown-progressCircle")
        })));
        else if (type === "placeholder") {
            let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
            if (emptyState == null) return null;
            return /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement("div", {
                // aria-selected isn't needed here since this option is not selectable.
                // eslint-disable-next-line jsx-a11y/role-has-required-aria-props
                role: "option"
            }, emptyState);
        }
    })));
}
// forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $60eb4b34c53310cb$export$1afdcf349979fb7e = /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).forwardRef($60eb4b34c53310cb$var$ListBoxBase);





function $582d86e45be55635$var$ListBox(props, ref) {
    let state = (0, $3Fwf2$reactstatelylist.useListState)(props);
    let layout = (0, $60eb4b34c53310cb$export$25768ea656ae32a7)(state, props.isLoading);
    let domRef = (0, $3Fwf2$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).createElement((0, $60eb4b34c53310cb$export$1afdcf349979fb7e), {
        ...props,
        ref: domRef,
        state: state,
        layout: layout
    });
}
// forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
/**
 * A list of options that can allow selection of one or more.
 */ const $582d86e45be55635$export$41f133550aa26f48 = /*#__PURE__*/ (0, ($parcel$interopDefault($3Fwf2$react))).forwardRef($582d86e45be55635$var$ListBox);
