// @react-aria/searchfield/dist/main.js
var $bd6662502a469b02$exports = require("./ar-AE.main.js");
var $360cd10b99b1ec90$exports = require("./bg-BG.main.js");
var $ffecf95c9ab0a601$exports = require("./cs-CZ.main.js");
var $a8c1807afa0c6ad0$exports = require("./da-DK.main.js");
var $0e885ea91b94ae2b$exports = require("./de-DE.main.js");
var $a313361aefc92d6a$exports = require("./el-GR.main.js");
var $aca1149af311369c$exports = require("./en-US.main.js");
var $fcf49eac878c2dd4$exports = require("./es-ES.main.js");
var $a4cc5fda0ae4bc39$exports = require("./et-EE.main.js");
var $3490cbf44d66e748$exports = require("./fi-FI.main.js");
var $7139336f2276b57b$exports = require("./fr-FR.main.js");
var $2cfd40eac86ffaf9$exports = require("./he-IL.main.js");
var $ad5cd81fb4b71e02$exports = require("./hr-HR.main.js");
var $efa4f2797ecc2e40$exports = require("./hu-HU.main.js");
var $7d0d440111d7eb6f$exports = require("./it-IT.main.js");
var $15bb9dd1255c87e0$exports = require("./ja-JP.main.js");
var $662038034d042bc2$exports = require("./ko-KR.main.js");
var $469701bdc0ecd008$exports = require("./lt-LT.main.js");
var $bddb04e2df050d62$exports = require("./lv-LV.main.js");
var $a295a578d307d4b5$exports = require("./nb-NO.main.js");
var $e0731003f5c0bbc3$exports = require("./nl-NL.main.js");
var $9acaf16c6e495332$exports = require("./pl-PL.main.js");
var $1087013ecb748501$exports = require("./pt-BR.main.js");
var $71e9e5edc2bbe98a$exports = require("./pt-PT.main.js");
var $a4bd32ef01b4a9e1$exports = require("./ro-RO.main.js");
var $0807874c8146beee$exports = require("./ru-RU.main.js");
var $3ed5a8cd509d389e$exports = require("./sk-SK.main.js");
var $1cc960210efc0d20$exports = require("./sl-SI.main.js");
var $18a26c027bd07b1a$exports = require("./sr-SP.main.js");
var $2ca3577ce1438e9b$exports = require("./sv-SE.main.js");
var $ced44bcac3738390$exports = require("./tr-TR.main.js");
var $abd2f08b465610a9$exports = require("./uk-UA.main.js");
var $becec37c237a6c53$exports = require("./zh-CN.main.js");
var $f7dff162f8d4dac5$exports = require("./zh-TW.main.js");
var $9UOFb$reactariautils = require("@react-aria/utils");
var $9UOFb$reactariai18n = require("@react-aria/i18n");
var $9UOFb$reactariatextfield = require("@react-aria/textfield");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "useSearchField", () => $ccae92a71d96d51a$export$9bb30bbe003b82e0);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
var $8e80a08ea739bb00$exports = {};


































$8e80a08ea739bb00$exports = {
    "ar-AE": $bd6662502a469b02$exports,
    "bg-BG": $360cd10b99b1ec90$exports,
    "cs-CZ": $ffecf95c9ab0a601$exports,
    "da-DK": $a8c1807afa0c6ad0$exports,
    "de-DE": $0e885ea91b94ae2b$exports,
    "el-GR": $a313361aefc92d6a$exports,
    "en-US": $aca1149af311369c$exports,
    "es-ES": $fcf49eac878c2dd4$exports,
    "et-EE": $a4cc5fda0ae4bc39$exports,
    "fi-FI": $3490cbf44d66e748$exports,
    "fr-FR": $7139336f2276b57b$exports,
    "he-IL": $2cfd40eac86ffaf9$exports,
    "hr-HR": $ad5cd81fb4b71e02$exports,
    "hu-HU": $efa4f2797ecc2e40$exports,
    "it-IT": $7d0d440111d7eb6f$exports,
    "ja-JP": $15bb9dd1255c87e0$exports,
    "ko-KR": $662038034d042bc2$exports,
    "lt-LT": $469701bdc0ecd008$exports,
    "lv-LV": $bddb04e2df050d62$exports,
    "nb-NO": $a295a578d307d4b5$exports,
    "nl-NL": $e0731003f5c0bbc3$exports,
    "pl-PL": $9acaf16c6e495332$exports,
    "pt-BR": $1087013ecb748501$exports,
    "pt-PT": $71e9e5edc2bbe98a$exports,
    "ro-RO": $a4bd32ef01b4a9e1$exports,
    "ru-RU": $0807874c8146beee$exports,
    "sk-SK": $3ed5a8cd509d389e$exports,
    "sl-SI": $1cc960210efc0d20$exports,
    "sr-SP": $18a26c027bd07b1a$exports,
    "sv-SE": $2ca3577ce1438e9b$exports,
    "tr-TR": $ced44bcac3738390$exports,
    "uk-UA": $abd2f08b465610a9$exports,
    "zh-CN": $becec37c237a6c53$exports,
    "zh-TW": $f7dff162f8d4dac5$exports
};




function $ccae92a71d96d51a$export$9bb30bbe003b82e0(props, state, inputRef) {
    let stringFormatter = (0, $9UOFb$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($8e80a08ea739bb00$exports))), "@react-aria/searchfield");
    let { isDisabled: isDisabled, isReadOnly: isReadOnly, onSubmit: onSubmit = ()=>{}, onClear: onClear, type: type = "search" } = props;
    let onKeyDown = (e)=>{
        const key = e.key;
        if (key === "Enter") e.preventDefault();
        if (isDisabled || isReadOnly) return;
        if (key === "Enter") onSubmit(state.value);
        if (key === "Escape") {
            if (state.value === "") e.continuePropagation();
            else {
                state.setValue("");
                if (onClear) onClear();
            }
        }
    };
    let onClearButtonClick = ()=>{
        state.setValue("");
        if (onClear) onClear();
    };
    let onPressStart = ()=>{
        var // this is in PressStart for mobile so that touching the clear button doesn't remove focus from
        // the input and close the keyboard
        _inputRef_current;
        (_inputRef_current = inputRef.current) === null || _inputRef_current === void 0 ? void 0 : _inputRef_current.focus();
    };
    let { labelProps: labelProps, inputProps: inputProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, ...validation } = (0, $9UOFb$reactariatextfield.useTextField)({
        ...props,
        value: state.value,
        onChange: state.setValue,
        onKeyDown: !isReadOnly ? (0, $9UOFb$reactariautils.chain)(onKeyDown, props.onKeyDown) : props.onKeyDown,
        type: type
    }, inputRef);
    return {
        labelProps: labelProps,
        inputProps: {
            ...inputProps,
            // already handled by useSearchFieldState
            defaultValue: undefined
        },
        clearButtonProps: {
            "aria-label": stringFormatter.format("Clear search"),
            excludeFromTabOrder: true,
            // @ts-ignore
            preventFocusOnPress: true,
            isDisabled: isDisabled || isReadOnly,
            onPress: onClearButtonClick,
            onPressStart: onPressStart
        },
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        ...validation
    };
}
