// @react-spectrum/picker/dist/main.js
"use client";

var $f7a89742a1d9c438$exports = require("./ar-AE.main.js");
var $146876e2177a34aa$exports = require("./bg-BG.main.js");
var $7349bb6d8a31ab4d$exports = require("./cs-CZ.main.js");
var $22b13068b0faa1ee$exports = require("./da-DK.main.js");
var $f6f852b8e013557d$exports = require("./de-DE.main.js");
var $637897118d3f68d8$exports = require("./el-GR.main.js");
var $700f837bfbf941fe$exports = require("./en-US.main.js");
var $e282d05496cfad1f$exports = require("./es-ES.main.js");
var $48a0d21d9506984b$exports = require("./et-EE.main.js");
var $3fb5f1756acecef2$exports = require("./fi-FI.main.js");
var $0290f6379e910cd9$exports = require("./fr-FR.main.js");
var $3d4dfe62507241ac$exports = require("./he-IL.main.js");
var $d2de2dae9493507c$exports = require("./hr-HR.main.js");
var $3e1bdddd8bf290da$exports = require("./hu-HU.main.js");
var $97be3e7a38e5c495$exports = require("./it-IT.main.js");
var $78bf12aa1b42968d$exports = require("./ja-JP.main.js");
var $670fecc37f4b8108$exports = require("./ko-KR.main.js");
var $5e5d0618bf4f3327$exports = require("./lt-LT.main.js");
var $0d3d0fdc7f4f8388$exports = require("./lv-LV.main.js");
var $9f49317ff63b766b$exports = require("./nb-NO.main.js");
var $57fc5bc347087de7$exports = require("./nl-NL.main.js");
var $d3f8820431992168$exports = require("./pl-PL.main.js");
var $c6002ee953000561$exports = require("./pt-BR.main.js");
var $743a55d845ecf1ea$exports = require("./pt-PT.main.js");
var $86b5655730fca6eb$exports = require("./ro-RO.main.js");
var $a6a9d47612edd183$exports = require("./ru-RU.main.js");
var $18de310aa582d102$exports = require("./sk-SK.main.js");
var $d3a70391c7504501$exports = require("./sl-SI.main.js");
var $58bfffdba11d41e8$exports = require("./sr-SP.main.js");
var $3ddb7e24330a8914$exports = require("./sv-SE.main.js");
var $a18856d5cf06f4a4$exports = require("./tr-TR.main.js");
var $a5401074209ec434$exports = require("./uk-UA.main.js");
var $3ff296eeb695953b$exports = require("./zh-CN.main.js");
var $d34d51bd3b86cf19$exports = require("./zh-TW.main.js");
require("./main.css");
var $4FjNO$reactstatelycollections = require("@react-stately/collections");
var $4FjNO$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $4FjNO$spectrumiconsuiChevronDownMedium = require("@spectrum-icons/ui/ChevronDownMedium");
var $4FjNO$reactspectrumutils = require("@react-spectrum/utils");
var $4FjNO$reactspectrumlabel = require("@react-spectrum/label");
var $4FjNO$reactspectrumbutton = require("@react-spectrum/button");
var $4FjNO$reactariaselect = require("@react-aria/select");
var $4FjNO$reactspectrumlistbox = require("@react-spectrum/listbox");
var $4FjNO$reactariautils = require("@react-aria/utils");
var $4FjNO$reactspectrumoverlays = require("@react-spectrum/overlays");
var $4FjNO$reactariainteractions = require("@react-aria/interactions");
var $4FjNO$reactspectrumprogress = require("@react-spectrum/progress");
var $4FjNO$react = require("react");
var $4FjNO$reactspectrumtext = require("@react-spectrum/text");
var $4FjNO$reactspectrumform = require("@react-spectrum/form");
var $4FjNO$reactariai18n = require("@react-aria/i18n");
var $4FjNO$reactspectrumprovider = require("@react-spectrum/provider");
var $4FjNO$reactstatelyselect = require("@react-stately/select");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Picker", () => $902e54cfaa4270a9$export$ba25329847403e11);
$parcel$export(module.exports, "Item", () => $4FjNO$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $4FjNO$reactstatelycollections.Section);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





var $b95bf71d4a256ee8$exports = {};


































$b95bf71d4a256ee8$exports = {
    "ar-AE": $f7a89742a1d9c438$exports,
    "bg-BG": $146876e2177a34aa$exports,
    "cs-CZ": $7349bb6d8a31ab4d$exports,
    "da-DK": $22b13068b0faa1ee$exports,
    "de-DE": $f6f852b8e013557d$exports,
    "el-GR": $637897118d3f68d8$exports,
    "en-US": $700f837bfbf941fe$exports,
    "es-ES": $e282d05496cfad1f$exports,
    "et-EE": $48a0d21d9506984b$exports,
    "fi-FI": $3fb5f1756acecef2$exports,
    "fr-FR": $0290f6379e910cd9$exports,
    "he-IL": $3d4dfe62507241ac$exports,
    "hr-HR": $d2de2dae9493507c$exports,
    "hu-HU": $3e1bdddd8bf290da$exports,
    "it-IT": $97be3e7a38e5c495$exports,
    "ja-JP": $78bf12aa1b42968d$exports,
    "ko-KR": $670fecc37f4b8108$exports,
    "lt-LT": $5e5d0618bf4f3327$exports,
    "lv-LV": $0d3d0fdc7f4f8388$exports,
    "nb-NO": $9f49317ff63b766b$exports,
    "nl-NL": $57fc5bc347087de7$exports,
    "pl-PL": $d3f8820431992168$exports,
    "pt-BR": $c6002ee953000561$exports,
    "pt-PT": $743a55d845ecf1ea$exports,
    "ro-RO": $86b5655730fca6eb$exports,
    "ru-RU": $a6a9d47612edd183$exports,
    "sk-SK": $18de310aa582d102$exports,
    "sl-SI": $d3a70391c7504501$exports,
    "sr-SP": $58bfffdba11d41e8$exports,
    "sv-SE": $3ddb7e24330a8914$exports,
    "tr-TR": $a18856d5cf06f4a4$exports,
    "uk-UA": $a5401074209ec434$exports,
    "zh-CN": $3ff296eeb695953b$exports,
    "zh-TW": $d34d51bd3b86cf19$exports
};








var $b39b306bfc9a5551$exports = {};

$parcel$export($b39b306bfc9a5551$exports, "focus-ring", () => $b39b306bfc9a5551$export$f39a09f249340e2a, (v) => $b39b306bfc9a5551$export$f39a09f249340e2a = v);
$parcel$export($b39b306bfc9a5551$exports, "i18nFontFamily", () => $b39b306bfc9a5551$export$8c4ee2c50c22c514, (v) => $b39b306bfc9a5551$export$8c4ee2c50c22c514 = v);
$parcel$export($b39b306bfc9a5551$exports, "is-disabled", () => $b39b306bfc9a5551$export$d35bc1e505d1ebbf, (v) => $b39b306bfc9a5551$export$d35bc1e505d1ebbf = v);
$parcel$export($b39b306bfc9a5551$exports, "is-hovered", () => $b39b306bfc9a5551$export$b8813cd5d7824ce7, (v) => $b39b306bfc9a5551$export$b8813cd5d7824ce7 = v);
$parcel$export($b39b306bfc9a5551$exports, "is-invalid", () => $b39b306bfc9a5551$export$14a30de2866685fa, (v) => $b39b306bfc9a5551$export$14a30de2866685fa = v);
$parcel$export($b39b306bfc9a5551$exports, "is-placeholder", () => $b39b306bfc9a5551$export$e5b2f5233e4e5194, (v) => $b39b306bfc9a5551$export$e5b2f5233e4e5194 = v);
$parcel$export($b39b306bfc9a5551$exports, "is-selected", () => $b39b306bfc9a5551$export$1e0fb04f31d3c22a, (v) => $b39b306bfc9a5551$export$1e0fb04f31d3c22a = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown", () => $b39b306bfc9a5551$export$f33c6833ddc773bd, (v) => $b39b306bfc9a5551$export$f33c6833ddc773bd = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown--quiet", () => $b39b306bfc9a5551$export$b2306417a70c7a62, (v) => $b39b306bfc9a5551$export$b2306417a70c7a62 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-avatar", () => $b39b306bfc9a5551$export$f8e5a201376aedb0, (v) => $b39b306bfc9a5551$export$f8e5a201376aedb0 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-chevron", () => $b39b306bfc9a5551$export$abbb0cf96dbbb8a0, (v) => $b39b306bfc9a5551$export$abbb0cf96dbbb8a0 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-fieldWrapper--positionSide", () => $b39b306bfc9a5551$export$c0b4e9596538c7b1, (v) => $b39b306bfc9a5551$export$c0b4e9596538c7b1 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-fieldWrapper--quiet", () => $b39b306bfc9a5551$export$66ddc2410b0e9eba, (v) => $b39b306bfc9a5551$export$66ddc2410b0e9eba = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-invalidIcon", () => $b39b306bfc9a5551$export$a32fdb5a063ea755, (v) => $b39b306bfc9a5551$export$a32fdb5a063ea755 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-label", () => $b39b306bfc9a5551$export$1235a00795b885e7, (v) => $b39b306bfc9a5551$export$1235a00795b885e7 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-popover--quiet", () => $b39b306bfc9a5551$export$cbeaf986973c0633, (v) => $b39b306bfc9a5551$export$cbeaf986973c0633 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-progressCircle", () => $b39b306bfc9a5551$export$18281c34cabbd13f, (v) => $b39b306bfc9a5551$export$18281c34cabbd13f = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Dropdown-trigger", () => $b39b306bfc9a5551$export$37a6d27a7cb47ce5, (v) => $b39b306bfc9a5551$export$37a6d27a7cb47ce5 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Field", () => $b39b306bfc9a5551$export$f6d480ae1e56eba0, (v) => $b39b306bfc9a5551$export$f6d480ae1e56eba0 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-FocusRing-ring", () => $b39b306bfc9a5551$export$4109102f950813a6, (v) => $b39b306bfc9a5551$export$4109102f950813a6 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-FocusRing", () => $b39b306bfc9a5551$export$24c7f46a6e3605dd, (v) => $b39b306bfc9a5551$export$24c7f46a6e3605dd = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-FocusRing--quiet", () => $b39b306bfc9a5551$export$2927016961429360, (v) => $b39b306bfc9a5551$export$2927016961429360 = v);
$parcel$export($b39b306bfc9a5551$exports, "spectrum-Icon", () => $b39b306bfc9a5551$export$d374b04f30360026, (v) => $b39b306bfc9a5551$export$d374b04f30360026 = v);
var $b39b306bfc9a5551$export$f39a09f249340e2a;
var $b39b306bfc9a5551$export$8c4ee2c50c22c514;
var $b39b306bfc9a5551$export$d35bc1e505d1ebbf;
var $b39b306bfc9a5551$export$b8813cd5d7824ce7;
var $b39b306bfc9a5551$export$14a30de2866685fa;
var $b39b306bfc9a5551$export$e5b2f5233e4e5194;
var $b39b306bfc9a5551$export$1e0fb04f31d3c22a;
var $b39b306bfc9a5551$export$f33c6833ddc773bd;
var $b39b306bfc9a5551$export$b2306417a70c7a62;
var $b39b306bfc9a5551$export$f8e5a201376aedb0;
var $b39b306bfc9a5551$export$abbb0cf96dbbb8a0;
var $b39b306bfc9a5551$export$c0b4e9596538c7b1;
var $b39b306bfc9a5551$export$66ddc2410b0e9eba;
var $b39b306bfc9a5551$export$a32fdb5a063ea755;
var $b39b306bfc9a5551$export$1235a00795b885e7;
var $b39b306bfc9a5551$export$cbeaf986973c0633;
var $b39b306bfc9a5551$export$18281c34cabbd13f;
var $b39b306bfc9a5551$export$37a6d27a7cb47ce5;
var $b39b306bfc9a5551$export$f6d480ae1e56eba0;
var $b39b306bfc9a5551$export$4109102f950813a6;
var $b39b306bfc9a5551$export$24c7f46a6e3605dd;
var $b39b306bfc9a5551$export$2927016961429360;
var $b39b306bfc9a5551$export$d374b04f30360026;
$b39b306bfc9a5551$export$f39a09f249340e2a = `PBsjDW_focus-ring`;
$b39b306bfc9a5551$export$8c4ee2c50c22c514 = `PBsjDW_i18nFontFamily`;
$b39b306bfc9a5551$export$d35bc1e505d1ebbf = `PBsjDW_is-disabled`;
$b39b306bfc9a5551$export$b8813cd5d7824ce7 = `PBsjDW_is-hovered`;
$b39b306bfc9a5551$export$14a30de2866685fa = `PBsjDW_is-invalid`;
$b39b306bfc9a5551$export$e5b2f5233e4e5194 = `PBsjDW_is-placeholder`;
$b39b306bfc9a5551$export$1e0fb04f31d3c22a = `PBsjDW_is-selected`;
$b39b306bfc9a5551$export$f33c6833ddc773bd = `PBsjDW_spectrum-Dropdown`;
$b39b306bfc9a5551$export$b2306417a70c7a62 = `PBsjDW_spectrum-Dropdown--quiet`;
$b39b306bfc9a5551$export$f8e5a201376aedb0 = `PBsjDW_spectrum-Dropdown-avatar`;
$b39b306bfc9a5551$export$abbb0cf96dbbb8a0 = `PBsjDW_spectrum-Dropdown-chevron`;
$b39b306bfc9a5551$export$c0b4e9596538c7b1 = `PBsjDW_spectrum-Dropdown-fieldWrapper--positionSide`;
$b39b306bfc9a5551$export$66ddc2410b0e9eba = `PBsjDW_spectrum-Dropdown-fieldWrapper--quiet`;
$b39b306bfc9a5551$export$a32fdb5a063ea755 = `PBsjDW_spectrum-Dropdown-invalidIcon`;
$b39b306bfc9a5551$export$1235a00795b885e7 = `PBsjDW_spectrum-Dropdown-label`;
$b39b306bfc9a5551$export$cbeaf986973c0633 = `PBsjDW_spectrum-Dropdown-popover--quiet`;
$b39b306bfc9a5551$export$18281c34cabbd13f = `PBsjDW_spectrum-Dropdown-progressCircle`;
$b39b306bfc9a5551$export$37a6d27a7cb47ce5 = `PBsjDW_spectrum-Dropdown-trigger`;
$b39b306bfc9a5551$export$f6d480ae1e56eba0 = `PBsjDW_spectrum-Field`;
$b39b306bfc9a5551$export$4109102f950813a6 = `PBsjDW_spectrum-FocusRing-ring`;
$b39b306bfc9a5551$export$24c7f46a6e3605dd = `PBsjDW_spectrum-FocusRing ${$b39b306bfc9a5551$export$4109102f950813a6}`;
$b39b306bfc9a5551$export$2927016961429360 = `PBsjDW_spectrum-FocusRing--quiet`;
$b39b306bfc9a5551$export$d374b04f30360026 = `PBsjDW_spectrum-Icon`;







function $902e54cfaa4270a9$var$Picker(props, ref) {
    props = (0, $4FjNO$reactspectrumutils.useSlotProps)(props, "picker");
    props = (0, $4FjNO$reactspectrumprovider.useProviderProps)(props);
    props = (0, $4FjNO$reactspectrumform.useFormProps)(props);
    let stringFormatter = (0, $4FjNO$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($b95bf71d4a256ee8$exports))), "@react-spectrum/picker");
    let { autoComplete: autoComplete, isDisabled: isDisabled, direction: direction = "bottom", align: align = "start", shouldFlip: shouldFlip = true, placeholder: placeholder = stringFormatter.format("placeholder"), isQuiet: isQuiet, label: label, labelPosition: labelPosition = "top", menuWidth: menuWidth, name: name, autoFocus: autoFocus } = props;
    let state = (0, $4FjNO$reactstatelyselect.useSelectState)(props);
    let domRef = (0, $4FjNO$reactspectrumutils.useDOMRef)(ref);
    let popoverRef = (0, $4FjNO$react.useRef)();
    let triggerRef = (0, $4FjNO$react.useRef)();
    let unwrappedTriggerRef = (0, $4FjNO$reactspectrumutils.useUnwrapDOMRef)(triggerRef);
    let listboxRef = (0, $4FjNO$react.useRef)();
    let isLoadingInitial = props.isLoading && state.collection.size === 0;
    let isLoadingMore = props.isLoading && state.collection.size > 0;
    let progressCircleId = (0, $4FjNO$reactariautils.useId)();
    // We create the listbox layout in Picker and pass it to ListBoxBase below
    // so that the layout information can be cached even while the listbox is not mounted.
    // We also use the layout as the keyboard delegate for type to select.
    let layout = (0, $4FjNO$reactspectrumlistbox.useListBoxLayout)(state, isLoadingMore);
    let { labelProps: labelProps, triggerProps: triggerProps, valueProps: valueProps, menuProps: menuProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $4FjNO$reactariaselect.useSelect)({
        ...props,
        "aria-describedby": isLoadingInitial ? progressCircleId : undefined,
        keyboardDelegate: layout
    }, state, unwrappedTriggerRef);
    let isMobile = (0, $4FjNO$reactspectrumutils.useIsMobileDevice)();
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $4FjNO$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    // On small screen devices, the listbox is rendered in a tray, otherwise a popover.
    let listbox = /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactspectrumlistbox.ListBoxBase), {
        ...menuProps,
        ref: listboxRef,
        disallowEmptySelection: true,
        autoFocus: state.focusStrategy || true,
        shouldSelectOnPressUp: true,
        focusOnPointerEnter: true,
        layout: layout,
        state: state,
        width: isMobile ? "100%" : undefined,
        // Set max height: inherit so Tray scrolling works
        UNSAFE_style: {
            maxHeight: "inherit"
        },
        isLoading: props.isLoading,
        onLoadMore: props.onLoadMore
    });
    // Measure the width of the button to inform the width of the menu (below).
    let [buttonWidth, setButtonWidth] = (0, $4FjNO$react.useState)(null);
    let { scale: scale } = (0, $4FjNO$reactspectrumprovider.useProvider)();
    let onResize = (0, $4FjNO$react.useCallback)(()=>{
        if (!isMobile && unwrappedTriggerRef.current) {
            let width = unwrappedTriggerRef.current.offsetWidth;
            setButtonWidth(width);
        }
    }, [
        unwrappedTriggerRef,
        setButtonWidth,
        isMobile
    ]);
    (0, $4FjNO$reactariautils.useResizeObserver)({
        ref: unwrappedTriggerRef,
        onResize: onResize
    });
    (0, $4FjNO$reactariautils.useLayoutEffect)(onResize, [
        scale,
        state.selectedKey,
        onResize
    ]);
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactspectrumoverlays.Tray), {
        state: state
    }, listbox);
    else {
        // If quiet, use the default width, otherwise match the width of the button. This can be overridden by the menuWidth prop.
        // Always have a minimum width of the button width. When quiet, there is an extra offset to add.
        // Not using style props for this because they don't support `calc`.
        let width = isQuiet ? null : buttonWidth;
        let style = {
            width: menuWidth ? (0, $4FjNO$reactspectrumutils.dimensionValue)(menuWidth) : width,
            minWidth: isQuiet ? `calc(${buttonWidth}px + calc(2 * var(--spectrum-dropdown-quiet-offset)))` : buttonWidth
        };
        overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactspectrumoverlays.Popover), {
            UNSAFE_style: style,
            UNSAFE_className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Dropdown-popover", {
                "spectrum-Dropdown-popover--quiet": isQuiet
            }),
            ref: popoverRef,
            placement: `${direction} ${align}`,
            shouldFlip: shouldFlip,
            hideArrow: true,
            state: state,
            triggerRef: unwrappedTriggerRef,
            scrollRef: listboxRef
        }, listbox);
    }
    let contents = state.selectedItem ? state.selectedItem.rendered : placeholder;
    if (typeof contents === "string") contents = /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactspectrumtext.Text), null, contents);
    let picker = /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement("div", {
        className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Dropdown", {
            "is-invalid": isInvalid && !isDisabled,
            "is-disabled": isDisabled,
            "spectrum-Dropdown--quiet": isQuiet
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactariaselect.HiddenSelect), {
        autoComplete: autoComplete,
        isDisabled: isDisabled,
        state: state,
        triggerRef: unwrappedTriggerRef,
        label: label,
        name: name
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactariainteractions.PressResponder), (0, $4FjNO$reactariautils.mergeProps)(hoverProps, triggerProps), /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactspectrumbutton.FieldButton), {
        ref: triggerRef,
        isActive: state.isOpen,
        isQuiet: isQuiet,
        isDisabled: isDisabled,
        isInvalid: isInvalid,
        autoFocus: autoFocus,
        UNSAFE_className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Dropdown-trigger", {
            "is-hovered": isHovered
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactspectrumutils.SlotProvider), {
        slots: {
            icon: {
                UNSAFE_className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Icon"),
                size: "S"
            },
            avatar: {
                UNSAFE_className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Dropdown-avatar"),
                size: "avatar-size-100"
            },
            text: {
                ...valueProps,
                UNSAFE_className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Dropdown-label", {
                    "is-placeholder": !state.selectedItem
                })
            },
            description: {
                isHidden: true
            }
        }
    }, contents), isLoadingInitial && /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactspectrumprogress.ProgressCircle), {
        id: progressCircleId,
        isIndeterminate: true,
        size: "S",
        "aria-label": stringFormatter.format("loading"),
        UNSAFE_className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Dropdown-progressCircle")
    }), isInvalid && !isLoadingInitial && !isDisabled && /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, ($parcel$interopDefault($4FjNO$spectrumiconsuiAlertMedium))), {
        UNSAFE_className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Dropdown-invalidIcon")
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, ($parcel$interopDefault($4FjNO$spectrumiconsuiChevronDownMedium))), {
        UNSAFE_className: (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Dropdown-chevron")
    }))), state.collection.size === 0 ? null : overlay);
    let wrapperClassName = label ? (0, $4FjNO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b39b306bfc9a5551$exports))), "spectrum-Field", {
        "spectrum-Dropdown-fieldWrapper--quiet": isQuiet,
        "spectrum-Dropdown-fieldWrapper--positionSide": labelPosition === "side"
    }) : "";
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).createElement((0, $4FjNO$reactspectrumlabel.Field), {
        ...props,
        ref: domRef,
        wrapperClassName: wrapperClassName,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        showErrorIcon: false,
        includeNecessityIndicatorInAccessibilityName: true,
        elementType: "span"
    }, picker);
}
/**
 * Pickers allow users to choose a single option from a collapsible list of options when space is limited.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $902e54cfaa4270a9$export$ba25329847403e11 = /*#__PURE__*/ (0, ($parcel$interopDefault($4FjNO$react))).forwardRef($902e54cfaa4270a9$var$Picker);
