// @react-spectrum/meter/dist/main.js
"use client";

require("./main.css");
var $6eMp1$reactspectrumutils = require("@react-spectrum/utils");
var $6eMp1$reactspectrumprogress = require("@react-spectrum/progress");
var $6eMp1$react = require("react");
var $6eMp1$reactariameter = require("@react-aria/meter");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Meter", () => $fc7fd823390a6f61$export$62e3ae2a4090b879);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $9f78a6c009f32d0a$exports = {};

$parcel$export($9f78a6c009f32d0a$exports, "focus-ring", () => $9f78a6c009f32d0a$export$f39a09f249340e2a, (v) => $9f78a6c009f32d0a$export$f39a09f249340e2a = v);
$parcel$export($9f78a6c009f32d0a$exports, "i18nFontFamily", () => $9f78a6c009f32d0a$export$8c4ee2c50c22c514, (v) => $9f78a6c009f32d0a$export$8c4ee2c50c22c514 = v);
$parcel$export($9f78a6c009f32d0a$exports, "indeterminate-loop-ltr", () => $9f78a6c009f32d0a$export$a889038a25113dd4, (v) => $9f78a6c009f32d0a$export$a889038a25113dd4 = v);
$parcel$export($9f78a6c009f32d0a$exports, "indeterminate-loop-rtl", () => $9f78a6c009f32d0a$export$aeff243cf3356f00, (v) => $9f78a6c009f32d0a$export$aeff243cf3356f00 = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-critical", () => $9f78a6c009f32d0a$export$a1f608a9810da50f, (v) => $9f78a6c009f32d0a$export$a1f608a9810da50f = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-positive", () => $9f78a6c009f32d0a$export$aea9fd919467f03a, (v) => $9f78a6c009f32d0a$export$aea9fd919467f03a = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-warning", () => $9f78a6c009f32d0a$export$d4ebda6218901908, (v) => $9f78a6c009f32d0a$export$d4ebda6218901908 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader", () => $9f78a6c009f32d0a$export$162ba336fb0c239f, (v) => $9f78a6c009f32d0a$export$162ba336fb0c239f = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--indeterminate", () => $9f78a6c009f32d0a$export$1249dfbb98223794, (v) => $9f78a6c009f32d0a$export$1249dfbb98223794 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--overBackground", () => $9f78a6c009f32d0a$export$55612d0bd9b5d41, (v) => $9f78a6c009f32d0a$export$55612d0bd9b5d41 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--sideLabel", () => $9f78a6c009f32d0a$export$99af8802cc2a10dc, (v) => $9f78a6c009f32d0a$export$99af8802cc2a10dc = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--small", () => $9f78a6c009f32d0a$export$b945305960ad74a7, (v) => $9f78a6c009f32d0a$export$b945305960ad74a7 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--staticBlack", () => $9f78a6c009f32d0a$export$f45257a01e7aabeb, (v) => $9f78a6c009f32d0a$export$f45257a01e7aabeb = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--staticWhite", () => $9f78a6c009f32d0a$export$734e9273ae1f542e, (v) => $9f78a6c009f32d0a$export$734e9273ae1f542e = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-fill", () => $9f78a6c009f32d0a$export$84c9ee7bc724b71b, (v) => $9f78a6c009f32d0a$export$84c9ee7bc724b71b = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-label", () => $9f78a6c009f32d0a$export$5379ba198163be86, (v) => $9f78a6c009f32d0a$export$5379ba198163be86 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-percentage", () => $9f78a6c009f32d0a$export$ecc9c825de9d1a18, (v) => $9f78a6c009f32d0a$export$ecc9c825de9d1a18 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-track", () => $9f78a6c009f32d0a$export$3e2e13fb57b5c74b, (v) => $9f78a6c009f32d0a$export$3e2e13fb57b5c74b = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-FocusRing-ring", () => $9f78a6c009f32d0a$export$4109102f950813a6, (v) => $9f78a6c009f32d0a$export$4109102f950813a6 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-FocusRing", () => $9f78a6c009f32d0a$export$24c7f46a6e3605dd, (v) => $9f78a6c009f32d0a$export$24c7f46a6e3605dd = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-FocusRing--quiet", () => $9f78a6c009f32d0a$export$2927016961429360, (v) => $9f78a6c009f32d0a$export$2927016961429360 = v);
var $9f78a6c009f32d0a$export$f39a09f249340e2a;
var $9f78a6c009f32d0a$export$8c4ee2c50c22c514;
var $9f78a6c009f32d0a$export$a889038a25113dd4;
var $9f78a6c009f32d0a$export$aeff243cf3356f00;
var $9f78a6c009f32d0a$export$a1f608a9810da50f;
var $9f78a6c009f32d0a$export$aea9fd919467f03a;
var $9f78a6c009f32d0a$export$d4ebda6218901908;
var $9f78a6c009f32d0a$export$162ba336fb0c239f;
var $9f78a6c009f32d0a$export$1249dfbb98223794;
var $9f78a6c009f32d0a$export$55612d0bd9b5d41;
var $9f78a6c009f32d0a$export$99af8802cc2a10dc;
var $9f78a6c009f32d0a$export$b945305960ad74a7;
var $9f78a6c009f32d0a$export$f45257a01e7aabeb;
var $9f78a6c009f32d0a$export$734e9273ae1f542e;
var $9f78a6c009f32d0a$export$84c9ee7bc724b71b;
var $9f78a6c009f32d0a$export$5379ba198163be86;
var $9f78a6c009f32d0a$export$ecc9c825de9d1a18;
var $9f78a6c009f32d0a$export$3e2e13fb57b5c74b;
var $9f78a6c009f32d0a$export$4109102f950813a6;
var $9f78a6c009f32d0a$export$24c7f46a6e3605dd;
var $9f78a6c009f32d0a$export$2927016961429360;
$9f78a6c009f32d0a$export$f39a09f249340e2a = `uFrYBa_focus-ring`;
$9f78a6c009f32d0a$export$8c4ee2c50c22c514 = `uFrYBa_i18nFontFamily`;
$9f78a6c009f32d0a$export$a889038a25113dd4 = `uFrYBa_indeterminate-loop-ltr`;
$9f78a6c009f32d0a$export$a889038a25113dd4;
$9f78a6c009f32d0a$export$aeff243cf3356f00 = `uFrYBa_indeterminate-loop-rtl`;
$9f78a6c009f32d0a$export$aeff243cf3356f00;
$9f78a6c009f32d0a$export$a1f608a9810da50f = `uFrYBa_is-critical`;
$9f78a6c009f32d0a$export$aea9fd919467f03a = `uFrYBa_is-positive`;
$9f78a6c009f32d0a$export$d4ebda6218901908 = `uFrYBa_is-warning`;
$9f78a6c009f32d0a$export$162ba336fb0c239f = `uFrYBa_spectrum-BarLoader`;
$9f78a6c009f32d0a$export$1249dfbb98223794 = `uFrYBa_spectrum-BarLoader--indeterminate`;
$9f78a6c009f32d0a$export$55612d0bd9b5d41 = `uFrYBa_spectrum-BarLoader--overBackground`;
$9f78a6c009f32d0a$export$99af8802cc2a10dc = `uFrYBa_spectrum-BarLoader--sideLabel`;
$9f78a6c009f32d0a$export$b945305960ad74a7 = `uFrYBa_spectrum-BarLoader--small`;
$9f78a6c009f32d0a$export$f45257a01e7aabeb = `uFrYBa_spectrum-BarLoader--staticBlack`;
$9f78a6c009f32d0a$export$734e9273ae1f542e = `uFrYBa_spectrum-BarLoader--staticWhite`;
$9f78a6c009f32d0a$export$84c9ee7bc724b71b = `uFrYBa_spectrum-BarLoader-fill`;
$9f78a6c009f32d0a$export$5379ba198163be86 = `uFrYBa_spectrum-BarLoader-label`;
$9f78a6c009f32d0a$export$ecc9c825de9d1a18 = `uFrYBa_spectrum-BarLoader-percentage`;
$9f78a6c009f32d0a$export$3e2e13fb57b5c74b = `uFrYBa_spectrum-BarLoader-track`;
$9f78a6c009f32d0a$export$4109102f950813a6 = `uFrYBa_spectrum-FocusRing-ring`;
$9f78a6c009f32d0a$export$24c7f46a6e3605dd = `uFrYBa_spectrum-FocusRing ${$9f78a6c009f32d0a$export$4109102f950813a6}`;
$9f78a6c009f32d0a$export$2927016961429360 = `uFrYBa_spectrum-FocusRing--quiet`;



function $fc7fd823390a6f61$var$Meter(props, ref) {
    let { variant: variant, ...otherProps } = props;
    const { meterProps: meterProps, labelProps: labelProps } = (0, $6eMp1$reactariameter.useMeter)(props);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6eMp1$react))).createElement((0, $6eMp1$reactspectrumprogress.ProgressBarBase), {
        ...otherProps,
        ref: ref,
        barProps: meterProps,
        labelProps: labelProps,
        barClassName: (0, $6eMp1$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($9f78a6c009f32d0a$exports))), {
            "is-positive": variant === "positive",
            "is-warning": variant === "warning",
            "is-critical": variant === "critical"
        })
    });
}
/**
 * Meters are visual representations of a quantity or an achievement.
 * Their progress is determined by user actions, rather than system actions.
 */ let $fc7fd823390a6f61$export$62e3ae2a4090b879 = /*#__PURE__*/ (0, ($parcel$interopDefault($6eMp1$react))).forwardRef($fc7fd823390a6f61$var$Meter);
