// @react-spectrum/meter/dist/module.js
"use client";

import "./main.css";
import {classNames as $dMbnw$classNames} from "@react-spectrum/utils";
import {ProgressBarBase as $dMbnw$ProgressBarBase} from "@react-spectrum/progress";
import $dMbnw$react from "react";
import {useMeter as $dMbnw$useMeter} from "@react-aria/meter";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $a43f840f49ffbe32$exports = {};

$parcel$export($a43f840f49ffbe32$exports, "focus-ring", () => $a43f840f49ffbe32$export$f39a09f249340e2a, (v) => $a43f840f49ffbe32$export$f39a09f249340e2a = v);
$parcel$export($a43f840f49ffbe32$exports, "i18nFontFamily", () => $a43f840f49ffbe32$export$8c4ee2c50c22c514, (v) => $a43f840f49ffbe32$export$8c4ee2c50c22c514 = v);
$parcel$export($a43f840f49ffbe32$exports, "indeterminate-loop-ltr", () => $a43f840f49ffbe32$export$a889038a25113dd4, (v) => $a43f840f49ffbe32$export$a889038a25113dd4 = v);
$parcel$export($a43f840f49ffbe32$exports, "indeterminate-loop-rtl", () => $a43f840f49ffbe32$export$aeff243cf3356f00, (v) => $a43f840f49ffbe32$export$aeff243cf3356f00 = v);
$parcel$export($a43f840f49ffbe32$exports, "is-critical", () => $a43f840f49ffbe32$export$a1f608a9810da50f, (v) => $a43f840f49ffbe32$export$a1f608a9810da50f = v);
$parcel$export($a43f840f49ffbe32$exports, "is-positive", () => $a43f840f49ffbe32$export$aea9fd919467f03a, (v) => $a43f840f49ffbe32$export$aea9fd919467f03a = v);
$parcel$export($a43f840f49ffbe32$exports, "is-warning", () => $a43f840f49ffbe32$export$d4ebda6218901908, (v) => $a43f840f49ffbe32$export$d4ebda6218901908 = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader", () => $a43f840f49ffbe32$export$162ba336fb0c239f, (v) => $a43f840f49ffbe32$export$162ba336fb0c239f = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader--indeterminate", () => $a43f840f49ffbe32$export$1249dfbb98223794, (v) => $a43f840f49ffbe32$export$1249dfbb98223794 = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader--overBackground", () => $a43f840f49ffbe32$export$55612d0bd9b5d41, (v) => $a43f840f49ffbe32$export$55612d0bd9b5d41 = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader--sideLabel", () => $a43f840f49ffbe32$export$99af8802cc2a10dc, (v) => $a43f840f49ffbe32$export$99af8802cc2a10dc = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader--small", () => $a43f840f49ffbe32$export$b945305960ad74a7, (v) => $a43f840f49ffbe32$export$b945305960ad74a7 = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader--staticBlack", () => $a43f840f49ffbe32$export$f45257a01e7aabeb, (v) => $a43f840f49ffbe32$export$f45257a01e7aabeb = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader--staticWhite", () => $a43f840f49ffbe32$export$734e9273ae1f542e, (v) => $a43f840f49ffbe32$export$734e9273ae1f542e = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader-fill", () => $a43f840f49ffbe32$export$84c9ee7bc724b71b, (v) => $a43f840f49ffbe32$export$84c9ee7bc724b71b = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader-label", () => $a43f840f49ffbe32$export$5379ba198163be86, (v) => $a43f840f49ffbe32$export$5379ba198163be86 = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader-percentage", () => $a43f840f49ffbe32$export$ecc9c825de9d1a18, (v) => $a43f840f49ffbe32$export$ecc9c825de9d1a18 = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-BarLoader-track", () => $a43f840f49ffbe32$export$3e2e13fb57b5c74b, (v) => $a43f840f49ffbe32$export$3e2e13fb57b5c74b = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-FocusRing-ring", () => $a43f840f49ffbe32$export$4109102f950813a6, (v) => $a43f840f49ffbe32$export$4109102f950813a6 = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-FocusRing", () => $a43f840f49ffbe32$export$24c7f46a6e3605dd, (v) => $a43f840f49ffbe32$export$24c7f46a6e3605dd = v);
$parcel$export($a43f840f49ffbe32$exports, "spectrum-FocusRing--quiet", () => $a43f840f49ffbe32$export$2927016961429360, (v) => $a43f840f49ffbe32$export$2927016961429360 = v);
var $a43f840f49ffbe32$export$f39a09f249340e2a;
var $a43f840f49ffbe32$export$8c4ee2c50c22c514;
var $a43f840f49ffbe32$export$a889038a25113dd4;
var $a43f840f49ffbe32$export$aeff243cf3356f00;
var $a43f840f49ffbe32$export$a1f608a9810da50f;
var $a43f840f49ffbe32$export$aea9fd919467f03a;
var $a43f840f49ffbe32$export$d4ebda6218901908;
var $a43f840f49ffbe32$export$162ba336fb0c239f;
var $a43f840f49ffbe32$export$1249dfbb98223794;
var $a43f840f49ffbe32$export$55612d0bd9b5d41;
var $a43f840f49ffbe32$export$99af8802cc2a10dc;
var $a43f840f49ffbe32$export$b945305960ad74a7;
var $a43f840f49ffbe32$export$f45257a01e7aabeb;
var $a43f840f49ffbe32$export$734e9273ae1f542e;
var $a43f840f49ffbe32$export$84c9ee7bc724b71b;
var $a43f840f49ffbe32$export$5379ba198163be86;
var $a43f840f49ffbe32$export$ecc9c825de9d1a18;
var $a43f840f49ffbe32$export$3e2e13fb57b5c74b;
var $a43f840f49ffbe32$export$4109102f950813a6;
var $a43f840f49ffbe32$export$24c7f46a6e3605dd;
var $a43f840f49ffbe32$export$2927016961429360;
$a43f840f49ffbe32$export$f39a09f249340e2a = `uFrYBa_focus-ring`;
$a43f840f49ffbe32$export$8c4ee2c50c22c514 = `uFrYBa_i18nFontFamily`;
$a43f840f49ffbe32$export$a889038a25113dd4 = `uFrYBa_indeterminate-loop-ltr`;
$a43f840f49ffbe32$export$a889038a25113dd4;
$a43f840f49ffbe32$export$aeff243cf3356f00 = `uFrYBa_indeterminate-loop-rtl`;
$a43f840f49ffbe32$export$aeff243cf3356f00;
$a43f840f49ffbe32$export$a1f608a9810da50f = `uFrYBa_is-critical`;
$a43f840f49ffbe32$export$aea9fd919467f03a = `uFrYBa_is-positive`;
$a43f840f49ffbe32$export$d4ebda6218901908 = `uFrYBa_is-warning`;
$a43f840f49ffbe32$export$162ba336fb0c239f = `uFrYBa_spectrum-BarLoader`;
$a43f840f49ffbe32$export$1249dfbb98223794 = `uFrYBa_spectrum-BarLoader--indeterminate`;
$a43f840f49ffbe32$export$55612d0bd9b5d41 = `uFrYBa_spectrum-BarLoader--overBackground`;
$a43f840f49ffbe32$export$99af8802cc2a10dc = `uFrYBa_spectrum-BarLoader--sideLabel`;
$a43f840f49ffbe32$export$b945305960ad74a7 = `uFrYBa_spectrum-BarLoader--small`;
$a43f840f49ffbe32$export$f45257a01e7aabeb = `uFrYBa_spectrum-BarLoader--staticBlack`;
$a43f840f49ffbe32$export$734e9273ae1f542e = `uFrYBa_spectrum-BarLoader--staticWhite`;
$a43f840f49ffbe32$export$84c9ee7bc724b71b = `uFrYBa_spectrum-BarLoader-fill`;
$a43f840f49ffbe32$export$5379ba198163be86 = `uFrYBa_spectrum-BarLoader-label`;
$a43f840f49ffbe32$export$ecc9c825de9d1a18 = `uFrYBa_spectrum-BarLoader-percentage`;
$a43f840f49ffbe32$export$3e2e13fb57b5c74b = `uFrYBa_spectrum-BarLoader-track`;
$a43f840f49ffbe32$export$4109102f950813a6 = `uFrYBa_spectrum-FocusRing-ring`;
$a43f840f49ffbe32$export$24c7f46a6e3605dd = `uFrYBa_spectrum-FocusRing ${$a43f840f49ffbe32$export$4109102f950813a6}`;
$a43f840f49ffbe32$export$2927016961429360 = `uFrYBa_spectrum-FocusRing--quiet`;



function $559585cb28581b56$var$Meter(props, ref) {
    let { variant: variant, ...otherProps } = props;
    const { meterProps: meterProps, labelProps: labelProps } = (0, $dMbnw$useMeter)(props);
    return /*#__PURE__*/ (0, $dMbnw$react).createElement((0, $dMbnw$ProgressBarBase), {
        ...otherProps,
        ref: ref,
        barProps: meterProps,
        labelProps: labelProps,
        barClassName: (0, $dMbnw$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a43f840f49ffbe32$exports))), {
            "is-positive": variant === "positive",
            "is-warning": variant === "warning",
            "is-critical": variant === "critical"
        })
    });
}
/**
 * Meters are visual representations of a quantity or an achievement.
 * Their progress is determined by user actions, rather than system actions.
 */ let $559585cb28581b56$export$62e3ae2a4090b879 = /*#__PURE__*/ (0, $dMbnw$react).forwardRef($559585cb28581b56$var$Meter);




export {$559585cb28581b56$export$62e3ae2a4090b879 as Meter};
