// @react-spectrum/well/dist/import.mjs
"use client";

import "./main.css";
import {useDOMRef as $2uC0Z$useDOMRef, useStyleProps as $2uC0Z$useStyleProps, classNames as $2uC0Z$classNames} from "@react-spectrum/utils";
import {filterDOMProps as $2uC0Z$filterDOMProps} from "@react-aria/utils";
import $2uC0Z$react, {forwardRef as $2uC0Z$forwardRef} from "react";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $f53ecf77408821c3$exports = {};

$parcel$export($f53ecf77408821c3$exports, "focus-ring", () => $f53ecf77408821c3$export$f39a09f249340e2a, (v) => $f53ecf77408821c3$export$f39a09f249340e2a = v);
$parcel$export($f53ecf77408821c3$exports, "i18nFontFamily", () => $f53ecf77408821c3$export$8c4ee2c50c22c514, (v) => $f53ecf77408821c3$export$8c4ee2c50c22c514 = v);
$parcel$export($f53ecf77408821c3$exports, "spectrum-FocusRing-ring", () => $f53ecf77408821c3$export$4109102f950813a6, (v) => $f53ecf77408821c3$export$4109102f950813a6 = v);
$parcel$export($f53ecf77408821c3$exports, "spectrum-FocusRing", () => $f53ecf77408821c3$export$24c7f46a6e3605dd, (v) => $f53ecf77408821c3$export$24c7f46a6e3605dd = v);
$parcel$export($f53ecf77408821c3$exports, "spectrum-FocusRing--quiet", () => $f53ecf77408821c3$export$2927016961429360, (v) => $f53ecf77408821c3$export$2927016961429360 = v);
$parcel$export($f53ecf77408821c3$exports, "spectrum-Well", () => $f53ecf77408821c3$export$98e76d3873e01aa9, (v) => $f53ecf77408821c3$export$98e76d3873e01aa9 = v);
var $f53ecf77408821c3$export$f39a09f249340e2a;
var $f53ecf77408821c3$export$8c4ee2c50c22c514;
var $f53ecf77408821c3$export$4109102f950813a6;
var $f53ecf77408821c3$export$24c7f46a6e3605dd;
var $f53ecf77408821c3$export$2927016961429360;
var $f53ecf77408821c3$export$98e76d3873e01aa9;
$f53ecf77408821c3$export$f39a09f249340e2a = `SWvLfG_focus-ring`;
$f53ecf77408821c3$export$8c4ee2c50c22c514 = `SWvLfG_i18nFontFamily`;
$f53ecf77408821c3$export$4109102f950813a6 = `SWvLfG_spectrum-FocusRing-ring`;
$f53ecf77408821c3$export$24c7f46a6e3605dd = `SWvLfG_spectrum-FocusRing ${$f53ecf77408821c3$export$4109102f950813a6}`;
$f53ecf77408821c3$export$2927016961429360 = `SWvLfG_spectrum-FocusRing--quiet`;
$f53ecf77408821c3$export$98e76d3873e01aa9 = `SWvLfG_spectrum-Well`;


function $126c538221395de1$var$Well(props, ref) {
    let { children: children, role: role, ...otherProps } = props;
    let domRef = (0, $2uC0Z$useDOMRef)(ref);
    let { styleProps: styleProps } = (0, $2uC0Z$useStyleProps)(otherProps);
    if (!role && (props["aria-label"] || props["aria-labelledby"])) console.warn("A labelled Well must have a role.");
    return /*#__PURE__*/ (0, $2uC0Z$react).createElement("div", {
        ...(0, $2uC0Z$filterDOMProps)(otherProps, {
            labelable: !!role
        }),
        ...styleProps,
        role: role,
        ref: domRef,
        className: (0, $2uC0Z$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f53ecf77408821c3$exports))), "spectrum-Well", styleProps.className)
    }, children);
}
/**
 * A Well is a content container that displays non-editable content separate from other content on the screen.
 * Often this is used to display preformatted text, such as code/markup examples on a documentation page.
 */ const $126c538221395de1$export$d89c679ee2f73242 = /*#__PURE__*/ (0, $2uC0Z$forwardRef)($126c538221395de1$var$Well);




export {$126c538221395de1$export$d89c679ee2f73242 as Well};
