// @react-spectrum/well/dist/main.js
"use client";

require("./main.css");
var $f5FMV$reactspectrumutils = require("@react-spectrum/utils");
var $f5FMV$reactariautils = require("@react-aria/utils");
var $f5FMV$react = require("react");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Well", () => $fb0fd853f763d005$export$d89c679ee2f73242);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $b516e4acb09598d3$exports = {};

$parcel$export($b516e4acb09598d3$exports, "focus-ring", () => $b516e4acb09598d3$export$f39a09f249340e2a, (v) => $b516e4acb09598d3$export$f39a09f249340e2a = v);
$parcel$export($b516e4acb09598d3$exports, "i18nFontFamily", () => $b516e4acb09598d3$export$8c4ee2c50c22c514, (v) => $b516e4acb09598d3$export$8c4ee2c50c22c514 = v);
$parcel$export($b516e4acb09598d3$exports, "spectrum-FocusRing-ring", () => $b516e4acb09598d3$export$4109102f950813a6, (v) => $b516e4acb09598d3$export$4109102f950813a6 = v);
$parcel$export($b516e4acb09598d3$exports, "spectrum-FocusRing", () => $b516e4acb09598d3$export$24c7f46a6e3605dd, (v) => $b516e4acb09598d3$export$24c7f46a6e3605dd = v);
$parcel$export($b516e4acb09598d3$exports, "spectrum-FocusRing--quiet", () => $b516e4acb09598d3$export$2927016961429360, (v) => $b516e4acb09598d3$export$2927016961429360 = v);
$parcel$export($b516e4acb09598d3$exports, "spectrum-Well", () => $b516e4acb09598d3$export$98e76d3873e01aa9, (v) => $b516e4acb09598d3$export$98e76d3873e01aa9 = v);
var $b516e4acb09598d3$export$f39a09f249340e2a;
var $b516e4acb09598d3$export$8c4ee2c50c22c514;
var $b516e4acb09598d3$export$4109102f950813a6;
var $b516e4acb09598d3$export$24c7f46a6e3605dd;
var $b516e4acb09598d3$export$2927016961429360;
var $b516e4acb09598d3$export$98e76d3873e01aa9;
$b516e4acb09598d3$export$f39a09f249340e2a = `SWvLfG_focus-ring`;
$b516e4acb09598d3$export$8c4ee2c50c22c514 = `SWvLfG_i18nFontFamily`;
$b516e4acb09598d3$export$4109102f950813a6 = `SWvLfG_spectrum-FocusRing-ring`;
$b516e4acb09598d3$export$24c7f46a6e3605dd = `SWvLfG_spectrum-FocusRing ${$b516e4acb09598d3$export$4109102f950813a6}`;
$b516e4acb09598d3$export$2927016961429360 = `SWvLfG_spectrum-FocusRing--quiet`;
$b516e4acb09598d3$export$98e76d3873e01aa9 = `SWvLfG_spectrum-Well`;


function $fb0fd853f763d005$var$Well(props, ref) {
    let { children: children, role: role, ...otherProps } = props;
    let domRef = (0, $f5FMV$reactspectrumutils.useDOMRef)(ref);
    let { styleProps: styleProps } = (0, $f5FMV$reactspectrumutils.useStyleProps)(otherProps);
    if (!role && (props["aria-label"] || props["aria-labelledby"])) console.warn("A labelled Well must have a role.");
    return /*#__PURE__*/ (0, ($parcel$interopDefault($f5FMV$react))).createElement("div", {
        ...(0, $f5FMV$reactariautils.filterDOMProps)(otherProps, {
            labelable: !!role
        }),
        ...styleProps,
        role: role,
        ref: domRef,
        className: (0, $f5FMV$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($b516e4acb09598d3$exports))), "spectrum-Well", styleProps.className)
    }, children);
}
/**
 * A Well is a content container that displays non-editable content separate from other content on the screen.
 * Often this is used to display preformatted text, such as code/markup examples on a documentation page.
 */ const $fb0fd853f763d005$export$d89c679ee2f73242 = /*#__PURE__*/ (0, $f5FMV$react.forwardRef)($fb0fd853f763d005$var$Well);
