// @react-spectrum/overlays/dist/main.js
"use client";

require("./main.css");
var $eLPAa$reactspectrumprovider = require("@react-spectrum/provider");
var $eLPAa$react = require("react");
var $eLPAa$reactariaoverlays = require("@react-aria/overlays");
var $eLPAa$reacttransitiongroup = require("react-transition-group");
var $eLPAa$reactspectrumutils = require("@react-spectrum/utils");
var $eLPAa$reactariainteractions = require("@react-aria/interactions");
var $eLPAa$reactariautils = require("@react-aria/utils");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Overlay", () => $2aebdc186fd41e87$export$c6fdb837b070b4ff);
$parcel$export(module.exports, "Popover", () => $f02bba55a62f20dd$export$5b6b19405a83ff9d);
$parcel$export(module.exports, "Modal", () => $fc75a6aa7a2b905b$export$2b77a92f1a5ad772);
$parcel$export(module.exports, "Tray", () => $6b386deda21be662$export$4589ed81930b555c);
$parcel$export(module.exports, "OpenTransition", () => $13f51cdc44d228b9$export$b847a40ee92eff38);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

const $13f51cdc44d228b9$var$OPEN_STATES = {
    entering: false,
    entered: true
};
function $13f51cdc44d228b9$export$b847a40ee92eff38(props) {
    var child;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $eLPAa$reacttransitiongroup.Transition), {
        timeout: {
            enter: 0,
            exit: 350
        },
        ...props
    }, (state)=>(0, ($parcel$interopDefault($eLPAa$react))).Children.map(props.children, (child)=>child && /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).cloneElement(child, {
                isOpen: !!$13f51cdc44d228b9$var$OPEN_STATES[state]
            })));
}





function $2aebdc186fd41e87$var$Overlay(props, ref) {
    let { children: children, isOpen: isOpen, disableFocusManagement: disableFocusManagement, container: container, onEnter: onEnter, onEntering: onEntering, onEntered: onEntered, onExit: onExit, onExiting: onExiting, onExited: onExited, nodeRef: nodeRef } = props;
    let [exited, setExited] = (0, $eLPAa$react.useState)(!isOpen);
    let handleEntered = (0, $eLPAa$react.useCallback)(()=>{
        setExited(false);
        if (onEntered) onEntered();
    }, [
        onEntered
    ]);
    let handleExited = (0, $eLPAa$react.useCallback)(()=>{
        setExited(true);
        if (onExited) onExited();
    }, [
        onExited
    ]);
    // Don't un-render the overlay while it's transitioning out.
    let mountOverlay = isOpen || !exited;
    if (!mountOverlay) // Don't bother showing anything if we don't have to.
    return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $eLPAa$reactariaoverlays.Overlay), {
        portalContainer: container,
        disableFocusManagement: disableFocusManagement,
        isExiting: !isOpen
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $eLPAa$reactspectrumprovider.Provider), {
        ref: ref,
        UNSAFE_style: {
            background: "transparent",
            isolation: "isolate"
        },
        isDisabled: false
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $13f51cdc44d228b9$export$b847a40ee92eff38), {
        in: isOpen,
        appear: true,
        onExit: onExit,
        onExiting: onExiting,
        onExited: handleExited,
        onEnter: onEnter,
        onEntering: onEntering,
        onEntered: handleEntered,
        nodeRef: nodeRef
    }, children)));
}
let $2aebdc186fd41e87$export$c6fdb837b070b4ff = /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).forwardRef($2aebdc186fd41e87$var$Overlay);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $e9a663ccb19ed1b0$exports = {};

$parcel$export($e9a663ccb19ed1b0$exports, "react-spectrum-Modal", () => $e9a663ccb19ed1b0$export$f8edeb62d7fed8c1, (v) => $e9a663ccb19ed1b0$export$f8edeb62d7fed8c1 = v);
$parcel$export($e9a663ccb19ed1b0$exports, "react-spectrum-Modal-wrapper", () => $e9a663ccb19ed1b0$export$e088717c213c31d7, (v) => $e9a663ccb19ed1b0$export$e088717c213c31d7 = v);
$parcel$export($e9a663ccb19ed1b0$exports, "react-spectrum-Popover", () => $e9a663ccb19ed1b0$export$884f6b5065bdc41a, (v) => $e9a663ccb19ed1b0$export$884f6b5065bdc41a = v);
$parcel$export($e9a663ccb19ed1b0$exports, "react-spectrum-Tray", () => $e9a663ccb19ed1b0$export$69dafa24343974dd, (v) => $e9a663ccb19ed1b0$export$69dafa24343974dd = v);
$parcel$export($e9a663ccb19ed1b0$exports, "spectrum-Dialog-content", () => $e9a663ccb19ed1b0$export$cb8eccd0f3639238, (v) => $e9a663ccb19ed1b0$export$cb8eccd0f3639238 = v);
$parcel$export($e9a663ccb19ed1b0$exports, "spectrum-Modal", () => $e9a663ccb19ed1b0$export$4c0b1e44c3834c85, (v) => $e9a663ccb19ed1b0$export$4c0b1e44c3834c85 = v);
$parcel$export($e9a663ccb19ed1b0$exports, "spectrum-Modal-wrapper", () => $e9a663ccb19ed1b0$export$35bea187ff802494, (v) => $e9a663ccb19ed1b0$export$35bea187ff802494 = v);
$parcel$export($e9a663ccb19ed1b0$exports, "spectrum-Popover", () => $e9a663ccb19ed1b0$export$a62dc1b6ab23a7bb, (v) => $e9a663ccb19ed1b0$export$a62dc1b6ab23a7bb = v);
$parcel$export($e9a663ccb19ed1b0$exports, "spectrum-Tray", () => $e9a663ccb19ed1b0$export$ce0704cd084c4f0d, (v) => $e9a663ccb19ed1b0$export$ce0704cd084c4f0d = v);
var $e9a663ccb19ed1b0$export$f8edeb62d7fed8c1;
var $e9a663ccb19ed1b0$export$e088717c213c31d7;
var $e9a663ccb19ed1b0$export$884f6b5065bdc41a;
var $e9a663ccb19ed1b0$export$69dafa24343974dd;
var $e9a663ccb19ed1b0$export$cb8eccd0f3639238;
var $e9a663ccb19ed1b0$export$4c0b1e44c3834c85;
var $e9a663ccb19ed1b0$export$35bea187ff802494;
var $e9a663ccb19ed1b0$export$a62dc1b6ab23a7bb;
var $e9a663ccb19ed1b0$export$ce0704cd084c4f0d;
$e9a663ccb19ed1b0$export$f8edeb62d7fed8c1 = `OhnpTq_react-spectrum-Modal`;
$e9a663ccb19ed1b0$export$e088717c213c31d7 = `OhnpTq_react-spectrum-Modal-wrapper`;
$e9a663ccb19ed1b0$export$884f6b5065bdc41a = `OhnpTq_react-spectrum-Popover`;
$e9a663ccb19ed1b0$export$69dafa24343974dd = `OhnpTq_react-spectrum-Tray`;
$e9a663ccb19ed1b0$export$cb8eccd0f3639238 = `OhnpTq_spectrum-Dialog-content`;
$e9a663ccb19ed1b0$export$4c0b1e44c3834c85 = `OhnpTq_spectrum-Modal`;
$e9a663ccb19ed1b0$export$35bea187ff802494 = `OhnpTq_spectrum-Modal-wrapper`;
$e9a663ccb19ed1b0$export$a62dc1b6ab23a7bb = `OhnpTq_spectrum-Popover`;
$e9a663ccb19ed1b0$export$ce0704cd084c4f0d = `OhnpTq_spectrum-Tray`;



var $16e1dca4664d0b1c$exports = {};

$parcel$export($16e1dca4664d0b1c$exports, "focus-ring", () => $16e1dca4664d0b1c$export$f39a09f249340e2a, (v) => $16e1dca4664d0b1c$export$f39a09f249340e2a = v);
$parcel$export($16e1dca4664d0b1c$exports, "i18nFontFamily", () => $16e1dca4664d0b1c$export$8c4ee2c50c22c514, (v) => $16e1dca4664d0b1c$export$8c4ee2c50c22c514 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-overlay--open", () => $16e1dca4664d0b1c$export$a397eb58038aa381, (v) => $16e1dca4664d0b1c$export$a397eb58038aa381 = v);
$parcel$export($16e1dca4664d0b1c$exports, "is-open", () => $16e1dca4664d0b1c$export$a9781837241c946d, (v) => $16e1dca4664d0b1c$export$a9781837241c946d = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-overlay--bottom--open", () => $16e1dca4664d0b1c$export$a96755f239de61ee, (v) => $16e1dca4664d0b1c$export$a96755f239de61ee = v);
$parcel$export($16e1dca4664d0b1c$exports, "is-open--bottom", () => $16e1dca4664d0b1c$export$1f160596ffe52357, (v) => $16e1dca4664d0b1c$export$1f160596ffe52357 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-overlay--left--open", () => $16e1dca4664d0b1c$export$a85c0c6b6cf8598c, (v) => $16e1dca4664d0b1c$export$a85c0c6b6cf8598c = v);
$parcel$export($16e1dca4664d0b1c$exports, "is-open--left", () => $16e1dca4664d0b1c$export$d6106cb3acaf56ae, (v) => $16e1dca4664d0b1c$export$d6106cb3acaf56ae = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-overlay--right--open", () => $16e1dca4664d0b1c$export$85aaa87e65e040bf, (v) => $16e1dca4664d0b1c$export$85aaa87e65e040bf = v);
$parcel$export($16e1dca4664d0b1c$exports, "is-open--right", () => $16e1dca4664d0b1c$export$8935d072812c3169, (v) => $16e1dca4664d0b1c$export$8935d072812c3169 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-overlay--top--open", () => $16e1dca4664d0b1c$export$b9768db7ffc1bc2, (v) => $16e1dca4664d0b1c$export$b9768db7ffc1bc2 = v);
$parcel$export($16e1dca4664d0b1c$exports, "is-open--top", () => $16e1dca4664d0b1c$export$cd4bf1e07ae86214, (v) => $16e1dca4664d0b1c$export$cd4bf1e07ae86214 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-FocusRing-ring", () => $16e1dca4664d0b1c$export$4109102f950813a6, (v) => $16e1dca4664d0b1c$export$4109102f950813a6 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-FocusRing", () => $16e1dca4664d0b1c$export$24c7f46a6e3605dd, (v) => $16e1dca4664d0b1c$export$24c7f46a6e3605dd = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-FocusRing--quiet", () => $16e1dca4664d0b1c$export$2927016961429360, (v) => $16e1dca4664d0b1c$export$2927016961429360 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-overlay", () => $16e1dca4664d0b1c$export$39eed38e2296a1d6, (v) => $16e1dca4664d0b1c$export$39eed38e2296a1d6 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover", () => $16e1dca4664d0b1c$export$a62dc1b6ab23a7bb, (v) => $16e1dca4664d0b1c$export$a62dc1b6ab23a7bb = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover--bottom", () => $16e1dca4664d0b1c$export$6a19b19b14780d4a, (v) => $16e1dca4664d0b1c$export$6a19b19b14780d4a = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover--dialog", () => $16e1dca4664d0b1c$export$cf634262e726cd19, (v) => $16e1dca4664d0b1c$export$cf634262e726cd19 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover--left", () => $16e1dca4664d0b1c$export$79cc55906ac5b00, (v) => $16e1dca4664d0b1c$export$79cc55906ac5b00 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover--right", () => $16e1dca4664d0b1c$export$5fc8b429396c9f58, (v) => $16e1dca4664d0b1c$export$5fc8b429396c9f58 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover--top", () => $16e1dca4664d0b1c$export$f30becfb1df0cae4, (v) => $16e1dca4664d0b1c$export$f30becfb1df0cae4 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover--withTip", () => $16e1dca4664d0b1c$export$d8b16c5377728262, (v) => $16e1dca4664d0b1c$export$d8b16c5377728262 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover-tip", () => $16e1dca4664d0b1c$export$3100ec4dc1668df1, (v) => $16e1dca4664d0b1c$export$3100ec4dc1668df1 = v);
$parcel$export($16e1dca4664d0b1c$exports, "spectrum-Popover-tip-triangle", () => $16e1dca4664d0b1c$export$ed733fa6cf19ab80, (v) => $16e1dca4664d0b1c$export$ed733fa6cf19ab80 = v);
var $16e1dca4664d0b1c$export$f39a09f249340e2a;
var $16e1dca4664d0b1c$export$8c4ee2c50c22c514;
var $16e1dca4664d0b1c$export$a397eb58038aa381;
var $16e1dca4664d0b1c$export$a9781837241c946d;
var $16e1dca4664d0b1c$export$a96755f239de61ee;
var $16e1dca4664d0b1c$export$1f160596ffe52357;
var $16e1dca4664d0b1c$export$a85c0c6b6cf8598c;
var $16e1dca4664d0b1c$export$d6106cb3acaf56ae;
var $16e1dca4664d0b1c$export$85aaa87e65e040bf;
var $16e1dca4664d0b1c$export$8935d072812c3169;
var $16e1dca4664d0b1c$export$b9768db7ffc1bc2;
var $16e1dca4664d0b1c$export$cd4bf1e07ae86214;
var $16e1dca4664d0b1c$export$4109102f950813a6;
var $16e1dca4664d0b1c$export$24c7f46a6e3605dd;
var $16e1dca4664d0b1c$export$2927016961429360;
var $16e1dca4664d0b1c$export$39eed38e2296a1d6;
var $16e1dca4664d0b1c$export$a62dc1b6ab23a7bb;
var $16e1dca4664d0b1c$export$6a19b19b14780d4a;
var $16e1dca4664d0b1c$export$cf634262e726cd19;
var $16e1dca4664d0b1c$export$79cc55906ac5b00;
var $16e1dca4664d0b1c$export$5fc8b429396c9f58;
var $16e1dca4664d0b1c$export$f30becfb1df0cae4;
var $16e1dca4664d0b1c$export$d8b16c5377728262;
var $16e1dca4664d0b1c$export$3100ec4dc1668df1;
var $16e1dca4664d0b1c$export$ed733fa6cf19ab80;
$16e1dca4664d0b1c$export$f39a09f249340e2a = `cH0MeW_focus-ring`;
$16e1dca4664d0b1c$export$8c4ee2c50c22c514 = `cH0MeW_i18nFontFamily`;
$16e1dca4664d0b1c$export$a397eb58038aa381 = `cH0MeW_spectrum-overlay--open`;
$16e1dca4664d0b1c$export$a9781837241c946d = `cH0MeW_is-open ${$16e1dca4664d0b1c$export$a397eb58038aa381}`;
$16e1dca4664d0b1c$export$a96755f239de61ee = `cH0MeW_spectrum-overlay--bottom--open`;
$16e1dca4664d0b1c$export$1f160596ffe52357 = `cH0MeW_is-open--bottom ${$16e1dca4664d0b1c$export$a96755f239de61ee}`;
$16e1dca4664d0b1c$export$a85c0c6b6cf8598c = `cH0MeW_spectrum-overlay--left--open`;
$16e1dca4664d0b1c$export$d6106cb3acaf56ae = `cH0MeW_is-open--left ${$16e1dca4664d0b1c$export$a85c0c6b6cf8598c}`;
$16e1dca4664d0b1c$export$85aaa87e65e040bf = `cH0MeW_spectrum-overlay--right--open`;
$16e1dca4664d0b1c$export$8935d072812c3169 = `cH0MeW_is-open--right ${$16e1dca4664d0b1c$export$85aaa87e65e040bf}`;
$16e1dca4664d0b1c$export$b9768db7ffc1bc2 = `cH0MeW_spectrum-overlay--top--open`;
$16e1dca4664d0b1c$export$cd4bf1e07ae86214 = `cH0MeW_is-open--top ${$16e1dca4664d0b1c$export$b9768db7ffc1bc2}`;
$16e1dca4664d0b1c$export$4109102f950813a6 = `cH0MeW_spectrum-FocusRing-ring`;
$16e1dca4664d0b1c$export$24c7f46a6e3605dd = `cH0MeW_spectrum-FocusRing ${$16e1dca4664d0b1c$export$4109102f950813a6}`;
$16e1dca4664d0b1c$export$2927016961429360 = `cH0MeW_spectrum-FocusRing--quiet`;
$16e1dca4664d0b1c$export$39eed38e2296a1d6 = `cH0MeW_spectrum-overlay`;
$16e1dca4664d0b1c$export$a62dc1b6ab23a7bb = `cH0MeW_spectrum-Popover ${$16e1dca4664d0b1c$export$39eed38e2296a1d6}`;
$16e1dca4664d0b1c$export$6a19b19b14780d4a = `cH0MeW_spectrum-Popover--bottom`;
$16e1dca4664d0b1c$export$cf634262e726cd19 = `cH0MeW_spectrum-Popover--dialog`;
$16e1dca4664d0b1c$export$79cc55906ac5b00 = `cH0MeW_spectrum-Popover--left`;
$16e1dca4664d0b1c$export$5fc8b429396c9f58 = `cH0MeW_spectrum-Popover--right`;
$16e1dca4664d0b1c$export$f30becfb1df0cae4 = `cH0MeW_spectrum-Popover--top`;
$16e1dca4664d0b1c$export$d8b16c5377728262 = `cH0MeW_spectrum-Popover--withTip`;
$16e1dca4664d0b1c$export$3100ec4dc1668df1 = `cH0MeW_spectrum-Popover-tip`;
$16e1dca4664d0b1c$export$ed733fa6cf19ab80 = `cH0MeW_spectrum-Popover-tip-triangle`;


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

var $42908e91b4c5f428$exports = {};

$parcel$export($42908e91b4c5f428$exports, "focus-ring", () => $42908e91b4c5f428$export$f39a09f249340e2a, (v) => $42908e91b4c5f428$export$f39a09f249340e2a = v);
$parcel$export($42908e91b4c5f428$exports, "i18nFontFamily", () => $42908e91b4c5f428$export$8c4ee2c50c22c514, (v) => $42908e91b4c5f428$export$8c4ee2c50c22c514 = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-overlay--open", () => $42908e91b4c5f428$export$a397eb58038aa381, (v) => $42908e91b4c5f428$export$a397eb58038aa381 = v);
$parcel$export($42908e91b4c5f428$exports, "is-open", () => $42908e91b4c5f428$export$a9781837241c946d, (v) => $42908e91b4c5f428$export$a9781837241c946d = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-FocusRing-ring", () => $42908e91b4c5f428$export$4109102f950813a6, (v) => $42908e91b4c5f428$export$4109102f950813a6 = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-FocusRing", () => $42908e91b4c5f428$export$24c7f46a6e3605dd, (v) => $42908e91b4c5f428$export$24c7f46a6e3605dd = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-FocusRing--quiet", () => $42908e91b4c5f428$export$2927016961429360, (v) => $42908e91b4c5f428$export$2927016961429360 = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-overlay", () => $42908e91b4c5f428$export$39eed38e2296a1d6, (v) => $42908e91b4c5f428$export$39eed38e2296a1d6 = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-Underlay", () => $42908e91b4c5f428$export$74470528f463af97, (v) => $42908e91b4c5f428$export$74470528f463af97 = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-Underlay--transparent", () => $42908e91b4c5f428$export$8eb3860104039b44, (v) => $42908e91b4c5f428$export$8eb3860104039b44 = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-overlay--bottom--open", () => $42908e91b4c5f428$export$a96755f239de61ee, (v) => $42908e91b4c5f428$export$a96755f239de61ee = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-overlay--left--open", () => $42908e91b4c5f428$export$a85c0c6b6cf8598c, (v) => $42908e91b4c5f428$export$a85c0c6b6cf8598c = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-overlay--right--open", () => $42908e91b4c5f428$export$85aaa87e65e040bf, (v) => $42908e91b4c5f428$export$85aaa87e65e040bf = v);
$parcel$export($42908e91b4c5f428$exports, "spectrum-overlay--top--open", () => $42908e91b4c5f428$export$b9768db7ffc1bc2, (v) => $42908e91b4c5f428$export$b9768db7ffc1bc2 = v);
var $42908e91b4c5f428$export$f39a09f249340e2a;
var $42908e91b4c5f428$export$8c4ee2c50c22c514;
var $42908e91b4c5f428$export$a397eb58038aa381;
var $42908e91b4c5f428$export$a9781837241c946d;
var $42908e91b4c5f428$export$4109102f950813a6;
var $42908e91b4c5f428$export$24c7f46a6e3605dd;
var $42908e91b4c5f428$export$2927016961429360;
var $42908e91b4c5f428$export$39eed38e2296a1d6;
var $42908e91b4c5f428$export$74470528f463af97;
var $42908e91b4c5f428$export$8eb3860104039b44;
var $42908e91b4c5f428$export$a96755f239de61ee;
var $42908e91b4c5f428$export$a85c0c6b6cf8598c;
var $42908e91b4c5f428$export$85aaa87e65e040bf;
var $42908e91b4c5f428$export$b9768db7ffc1bc2;
$42908e91b4c5f428$export$f39a09f249340e2a = `F-oYUa_focus-ring`;
$42908e91b4c5f428$export$8c4ee2c50c22c514 = `F-oYUa_i18nFontFamily`;
$42908e91b4c5f428$export$a397eb58038aa381 = `F-oYUa_spectrum-overlay--open`;
$42908e91b4c5f428$export$a9781837241c946d = `F-oYUa_is-open ${$42908e91b4c5f428$export$a397eb58038aa381}`;
$42908e91b4c5f428$export$4109102f950813a6 = `F-oYUa_spectrum-FocusRing-ring`;
$42908e91b4c5f428$export$24c7f46a6e3605dd = `F-oYUa_spectrum-FocusRing ${$42908e91b4c5f428$export$4109102f950813a6}`;
$42908e91b4c5f428$export$2927016961429360 = `F-oYUa_spectrum-FocusRing--quiet`;
$42908e91b4c5f428$export$39eed38e2296a1d6 = `F-oYUa_spectrum-overlay`;
$42908e91b4c5f428$export$74470528f463af97 = `F-oYUa_spectrum-Underlay ${$42908e91b4c5f428$export$39eed38e2296a1d6}`;
$42908e91b4c5f428$export$8eb3860104039b44 = `F-oYUa_spectrum-Underlay--transparent`;
$42908e91b4c5f428$export$a96755f239de61ee = `F-oYUa_spectrum-overlay--bottom--open`;
$42908e91b4c5f428$export$a85c0c6b6cf8598c = `F-oYUa_spectrum-overlay--left--open`;
$42908e91b4c5f428$export$85aaa87e65e040bf = `F-oYUa_spectrum-overlay--right--open`;
$42908e91b4c5f428$export$b9768db7ffc1bc2 = `F-oYUa_spectrum-overlay--top--open`;


function $b900e75089bdd9cd$export$f360afc887607b02({ isOpen: isOpen, isTransparent: isTransparent, ...otherProps }) {
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        "data-testid": "underlay",
        ...otherProps,
        className: (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($42908e91b4c5f428$exports))), "spectrum-Underlay", {
            "is-open": isOpen,
            "spectrum-Underlay--transparent": isTransparent
        })
    });
}


/**
 * Arrow placement can be done pointing right or down because those paths start at 0, x or y. Because the
 * other two don't, they start at a fractional pixel value, it introduces rounding differences between browsers and
 * between display types (retina with subpixels vs not retina). By flipping them with CSS we can ensure that
 * the path always starts at 0 so that it perfectly overlaps the popover's border.
 * See bottom of file for more explanation.
 */ let $f02bba55a62f20dd$var$arrowPlacement = {
    left: "right",
    right: "right",
    top: "bottom",
    bottom: "bottom"
};
function $f02bba55a62f20dd$var$Popover(props, ref) {
    let { children: children, state: state, ...otherProps } = props;
    let domRef = (0, $eLPAa$reactspectrumutils.useDOMRef)(ref);
    let wrapperRef = (0, $eLPAa$react.useRef)(null);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $2aebdc186fd41e87$export$c6fdb837b070b4ff), {
        ...otherProps,
        isOpen: state.isOpen,
        nodeRef: wrapperRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement($f02bba55a62f20dd$var$PopoverWrapper, {
        ref: domRef,
        ...props,
        wrapperRef: wrapperRef
    }, children));
}
const $f02bba55a62f20dd$var$PopoverWrapper = /*#__PURE__*/ (0, $eLPAa$react.forwardRef)((props, ref)=>{
    let { children: children, isOpen: isOpen, hideArrow: hideArrow, isNonModal: isNonModal, enableBothDismissButtons: enableBothDismissButtons, state: state, wrapperRef: wrapperRef, onDismissButtonPress: onDismissButtonPress = ()=>state.close() } = props;
    let { styleProps: styleProps } = (0, $eLPAa$reactspectrumutils.useStyleProps)(props);
    let { size: size, borderWidth: borderWidth, arrowRef: arrowRef } = $f02bba55a62f20dd$var$useArrowSize();
    const borderRadius = $f02bba55a62f20dd$var$usePopoverBorderRadius(ref);
    let borderDiagonal = borderWidth * Math.SQRT2;
    let primary = size + borderDiagonal;
    let secondary = primary * 2;
    let { popoverProps: popoverProps, arrowProps: arrowProps, underlayProps: underlayProps, placement: placement } = (0, $eLPAa$reactariaoverlays.usePopover)({
        ...props,
        popoverRef: ref,
        maxHeight: null,
        arrowSize: hideArrow ? 0 : secondary,
        arrowBoundaryOffset: borderRadius
    }, state);
    let { focusWithinProps: focusWithinProps } = (0, $eLPAa$reactariainteractions.useFocusWithin)(props);
    let onPointerDown = ()=>{
        state.close();
    };
    // Attach Transition's nodeRef to outermost wrapper for node.reflow: https://github.com/reactjs/react-transition-group/blob/c89f807067b32eea6f68fd6c622190d88ced82e2/src/Transition.js#L231
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        ref: wrapperRef
    }, !isNonModal && /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $b900e75089bdd9cd$export$f360afc887607b02), {
        isTransparent: true,
        ...(0, $eLPAa$reactariautils.mergeProps)(underlayProps, {
            onPointerDown: onPointerDown
        }),
        isOpen: isOpen
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        ...styleProps,
        ...(0, $eLPAa$reactariautils.mergeProps)(popoverProps, focusWithinProps),
        style: {
            ...styleProps.style,
            ...popoverProps.style
        },
        ref: ref,
        className: (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($16e1dca4664d0b1c$exports))), "spectrum-Popover", `spectrum-Popover--${placement}`, {
            "spectrum-Popover--withTip": !hideArrow,
            "is-open": isOpen,
            [`is-open--${placement}`]: isOpen
        }, (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e9a663ccb19ed1b0$exports))), "spectrum-Popover", "react-spectrum-Popover"), styleProps.className),
        role: "presentation",
        "data-testid": "popover"
    }, (!isNonModal || enableBothDismissButtons) && /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $eLPAa$reactariaoverlays.DismissButton), {
        onDismiss: onDismissButtonPress
    }), children, hideArrow ? null : /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement($f02bba55a62f20dd$var$Arrow, {
        arrowProps: arrowProps,
        isLandscape: $f02bba55a62f20dd$var$arrowPlacement[placement] === "bottom",
        arrowRef: arrowRef,
        primary: primary,
        secondary: secondary,
        borderDiagonal: borderDiagonal
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $eLPAa$reactariaoverlays.DismissButton), {
        onDismiss: onDismissButtonPress
    })));
});
function $f02bba55a62f20dd$var$usePopoverBorderRadius(popoverRef) {
    let [borderRadius, setBorderRadius] = (0, $eLPAa$react.useState)(0);
    (0, $eLPAa$reactariautils.useLayoutEffect)(()=>{
        if (popoverRef.current) {
            let spectrumBorderRadius = window.getComputedStyle(popoverRef.current).borderRadius;
            if (spectrumBorderRadius !== "") setBorderRadius(parseInt(spectrumBorderRadius, 10));
        }
    }, [
        popoverRef
    ]);
    return borderRadius;
}
function $f02bba55a62f20dd$var$useArrowSize() {
    let [size, setSize] = (0, $eLPAa$react.useState)(20);
    let [borderWidth, setBorderWidth] = (0, $eLPAa$react.useState)(1);
    let arrowRef = (0, $eLPAa$react.useRef)(null);
    // get the css value for the tip size and divide it by 2 for this arrow implementation
    (0, $eLPAa$reactariautils.useLayoutEffect)(()=>{
        if (arrowRef.current) {
            let spectrumTipWidth = window.getComputedStyle(arrowRef.current).getPropertyValue("--spectrum-popover-tip-size");
            if (spectrumTipWidth !== "") setSize(parseInt(spectrumTipWidth, 10) / 2);
            let spectrumBorderWidth = window.getComputedStyle(arrowRef.current).getPropertyValue("--spectrum-popover-tip-borderWidth");
            if (spectrumBorderWidth !== "") setBorderWidth(parseInt(spectrumBorderWidth, 10));
        }
    }, []);
    return {
        size: size,
        borderWidth: borderWidth,
        arrowRef: arrowRef
    };
}
function $f02bba55a62f20dd$var$Arrow(props) {
    let { primary: primary, secondary: secondary, isLandscape: isLandscape, arrowProps: arrowProps, borderDiagonal: borderDiagonal, arrowRef: arrowRef } = props;
    let halfBorderDiagonal = borderDiagonal / 2;
    let primaryStart = 0;
    let primaryEnd = primary - halfBorderDiagonal;
    let secondaryStart = halfBorderDiagonal;
    let secondaryMiddle = secondary / 2;
    let secondaryEnd = secondary - halfBorderDiagonal;
    let pathData = isLandscape ? [
        "M",
        secondaryStart,
        primaryStart,
        "L",
        secondaryMiddle,
        primaryEnd,
        "L",
        secondaryEnd,
        primaryStart
    ] : [
        "M",
        primaryStart,
        secondaryStart,
        "L",
        primaryEnd,
        secondaryMiddle,
        "L",
        primaryStart,
        secondaryEnd
    ];
    /* use ceil because the svg needs to always accommodate the path inside it */ return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("svg", {
        xmlns: "http://www.w3.org/svg/2000",
        width: Math.ceil(isLandscape ? secondary : primary),
        height: Math.ceil(isLandscape ? primary : secondary),
        className: (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($16e1dca4664d0b1c$exports))), "spectrum-Popover-tip"),
        ref: arrowRef,
        ...arrowProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("path", {
        className: (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($16e1dca4664d0b1c$exports))), "spectrum-Popover-tip-triangle"),
        d: pathData.join(" ")
    }));
}
let $f02bba55a62f20dd$export$5b6b19405a83ff9d = /*#__PURE__*/ (0, $eLPAa$react.forwardRef)($f02bba55a62f20dd$var$Popover);
 /**
 * More explanation on popover tips.
 * - I tried changing the calculation of the popover placement in an effort to get it squarely onto the pixel grid.
 * This did not work because the problem was in the svg partial pixel end of the path in the popover right and popover bottom.
 * - I tried creating an extra 'bandaid' path that matched the background color and would overlap the popover border.
 * This didn't work because the border on the svg triangle didn't extend all the way to match nicely with the popover border.
 * - I tried getting the client bounding box and setting the svg to that partial pixel value
 * This didn't work because again the issue was inside the svg
 * - I didn't try drawing the svg backwards
 * This could still be tried
 * - I tried changing the calculation of the popover placement AND the svg height/width so that they were all rounded
 * This seems to have done the trick.
 */ 

/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

var $86571b803f942ff3$exports = {};

$parcel$export($86571b803f942ff3$exports, "focus-ring", () => $86571b803f942ff3$export$f39a09f249340e2a, (v) => $86571b803f942ff3$export$f39a09f249340e2a = v);
$parcel$export($86571b803f942ff3$exports, "i18nFontFamily", () => $86571b803f942ff3$export$8c4ee2c50c22c514, (v) => $86571b803f942ff3$export$8c4ee2c50c22c514 = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-overlay--open", () => $86571b803f942ff3$export$a397eb58038aa381, (v) => $86571b803f942ff3$export$a397eb58038aa381 = v);
$parcel$export($86571b803f942ff3$exports, "is-open", () => $86571b803f942ff3$export$a9781837241c946d, (v) => $86571b803f942ff3$export$a9781837241c946d = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-FocusRing-ring", () => $86571b803f942ff3$export$4109102f950813a6, (v) => $86571b803f942ff3$export$4109102f950813a6 = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-FocusRing", () => $86571b803f942ff3$export$24c7f46a6e3605dd, (v) => $86571b803f942ff3$export$24c7f46a6e3605dd = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-FocusRing--quiet", () => $86571b803f942ff3$export$2927016961429360, (v) => $86571b803f942ff3$export$2927016961429360 = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-overlay", () => $86571b803f942ff3$export$39eed38e2296a1d6, (v) => $86571b803f942ff3$export$39eed38e2296a1d6 = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-Modal", () => $86571b803f942ff3$export$4c0b1e44c3834c85, (v) => $86571b803f942ff3$export$4c0b1e44c3834c85 = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-Modal--fullscreen", () => $86571b803f942ff3$export$a4423fa77727431a, (v) => $86571b803f942ff3$export$a4423fa77727431a = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-Modal--fullscreenTakeover", () => $86571b803f942ff3$export$175dd699720b5fb7, (v) => $86571b803f942ff3$export$175dd699720b5fb7 = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-Modal--responsive", () => $86571b803f942ff3$export$fd1e42aa8ea02a1, (v) => $86571b803f942ff3$export$fd1e42aa8ea02a1 = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-Modal-wrapper", () => $86571b803f942ff3$export$35bea187ff802494, (v) => $86571b803f942ff3$export$35bea187ff802494 = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-overlay--bottom--open", () => $86571b803f942ff3$export$a96755f239de61ee, (v) => $86571b803f942ff3$export$a96755f239de61ee = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-overlay--left--open", () => $86571b803f942ff3$export$a85c0c6b6cf8598c, (v) => $86571b803f942ff3$export$a85c0c6b6cf8598c = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-overlay--right--open", () => $86571b803f942ff3$export$85aaa87e65e040bf, (v) => $86571b803f942ff3$export$85aaa87e65e040bf = v);
$parcel$export($86571b803f942ff3$exports, "spectrum-overlay--top--open", () => $86571b803f942ff3$export$b9768db7ffc1bc2, (v) => $86571b803f942ff3$export$b9768db7ffc1bc2 = v);
var $86571b803f942ff3$export$f39a09f249340e2a;
var $86571b803f942ff3$export$8c4ee2c50c22c514;
var $86571b803f942ff3$export$a397eb58038aa381;
var $86571b803f942ff3$export$a9781837241c946d;
var $86571b803f942ff3$export$4109102f950813a6;
var $86571b803f942ff3$export$24c7f46a6e3605dd;
var $86571b803f942ff3$export$2927016961429360;
var $86571b803f942ff3$export$39eed38e2296a1d6;
var $86571b803f942ff3$export$4c0b1e44c3834c85;
var $86571b803f942ff3$export$a4423fa77727431a;
var $86571b803f942ff3$export$175dd699720b5fb7;
var $86571b803f942ff3$export$fd1e42aa8ea02a1;
var $86571b803f942ff3$export$35bea187ff802494;
var $86571b803f942ff3$export$a96755f239de61ee;
var $86571b803f942ff3$export$a85c0c6b6cf8598c;
var $86571b803f942ff3$export$85aaa87e65e040bf;
var $86571b803f942ff3$export$b9768db7ffc1bc2;
$86571b803f942ff3$export$f39a09f249340e2a = `_0YML2q_focus-ring`;
$86571b803f942ff3$export$8c4ee2c50c22c514 = `_0YML2q_i18nFontFamily`;
$86571b803f942ff3$export$a397eb58038aa381 = `_0YML2q_spectrum-overlay--open`;
$86571b803f942ff3$export$a9781837241c946d = `_0YML2q_is-open ${$86571b803f942ff3$export$a397eb58038aa381}`;
$86571b803f942ff3$export$4109102f950813a6 = `_0YML2q_spectrum-FocusRing-ring`;
$86571b803f942ff3$export$24c7f46a6e3605dd = `_0YML2q_spectrum-FocusRing ${$86571b803f942ff3$export$4109102f950813a6}`;
$86571b803f942ff3$export$2927016961429360 = `_0YML2q_spectrum-FocusRing--quiet`;
$86571b803f942ff3$export$39eed38e2296a1d6 = `_0YML2q_spectrum-overlay`;
$86571b803f942ff3$export$4c0b1e44c3834c85 = `_0YML2q_spectrum-Modal ${$86571b803f942ff3$export$39eed38e2296a1d6}`;
$86571b803f942ff3$export$a4423fa77727431a = `_0YML2q_spectrum-Modal--fullscreen`;
$86571b803f942ff3$export$175dd699720b5fb7 = `_0YML2q_spectrum-Modal--fullscreenTakeover`;
$86571b803f942ff3$export$fd1e42aa8ea02a1 = `_0YML2q_spectrum-Modal--responsive`;
$86571b803f942ff3$export$35bea187ff802494 = `_0YML2q_spectrum-Modal-wrapper`;
$86571b803f942ff3$export$a96755f239de61ee = `_0YML2q_spectrum-overlay--bottom--open`;
$86571b803f942ff3$export$a85c0c6b6cf8598c = `_0YML2q_spectrum-overlay--left--open`;
$86571b803f942ff3$export$85aaa87e65e040bf = `_0YML2q_spectrum-overlay--right--open`;
$86571b803f942ff3$export$b9768db7ffc1bc2 = `_0YML2q_spectrum-overlay--top--open`;







function $fc75a6aa7a2b905b$var$Modal(props, ref) {
    let { children: children, state: state, ...otherProps } = props;
    let domRef = (0, $eLPAa$reactspectrumutils.useDOMRef)(ref);
    let wrapperRef = (0, $eLPAa$react.useRef)(null);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $2aebdc186fd41e87$export$c6fdb837b070b4ff), {
        ...otherProps,
        isOpen: state.isOpen,
        nodeRef: wrapperRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement($fc75a6aa7a2b905b$var$ModalWrapper, {
        ...props,
        wrapperRef: wrapperRef,
        ref: domRef
    }, children));
}
let $fc75a6aa7a2b905b$var$typeMap = {
    fullscreen: "fullscreen",
    fullscreenTakeover: "fullscreenTakeover"
};
let $fc75a6aa7a2b905b$var$ModalWrapper = /*#__PURE__*/ (0, $eLPAa$react.forwardRef)(function(props, ref) {
    let { type: type, children: children, state: state, isOpen: isOpen, wrapperRef: wrapperRef } = props;
    let typeVariant = $fc75a6aa7a2b905b$var$typeMap[type];
    let { styleProps: styleProps } = (0, $eLPAa$reactspectrumutils.useStyleProps)(props);
    let { modalProps: modalProps, underlayProps: underlayProps } = (0, $eLPAa$reactariaoverlays.useModalOverlay)(props, state, ref);
    let wrapperClassName = (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($86571b803f942ff3$exports))), "spectrum-Modal-wrapper", (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e9a663ccb19ed1b0$exports))), "spectrum-Modal-wrapper", "react-spectrum-Modal-wrapper"));
    let modalClassName = (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($86571b803f942ff3$exports))), "spectrum-Modal", {
        "is-open": isOpen
    }, (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e9a663ccb19ed1b0$exports))), "spectrum-Modal", "react-spectrum-Modal"), {
        [`spectrum-Modal--${typeVariant}`]: typeVariant
    }, styleProps.className);
    let viewport = (0, $eLPAa$reactariautils.useViewportSize)();
    let style = {
        "--spectrum-visual-viewport-height": viewport.height + "px"
    };
    // Attach Transition's nodeRef to outer most wrapper for node.reflow: https://github.com/reactjs/react-transition-group/blob/c89f807067b32eea6f68fd6c622190d88ced82e2/src/Transition.js#L231
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        ref: wrapperRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $b900e75089bdd9cd$export$f360afc887607b02), {
        ...underlayProps,
        isOpen: isOpen
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        className: wrapperClassName,
        style: style
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        ...styleProps,
        ...modalProps,
        ref: ref,
        className: modalClassName,
        "data-testid": "modal"
    }, children)));
});
let $fc75a6aa7a2b905b$export$2b77a92f1a5ad772 = /*#__PURE__*/ (0, $eLPAa$react.forwardRef)($fc75a6aa7a2b905b$var$Modal);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $3ba16c5c57f2a636$exports = {};

$parcel$export($3ba16c5c57f2a636$exports, "focus-ring", () => $3ba16c5c57f2a636$export$f39a09f249340e2a, (v) => $3ba16c5c57f2a636$export$f39a09f249340e2a = v);
$parcel$export($3ba16c5c57f2a636$exports, "i18nFontFamily", () => $3ba16c5c57f2a636$export$8c4ee2c50c22c514, (v) => $3ba16c5c57f2a636$export$8c4ee2c50c22c514 = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-overlay--open", () => $3ba16c5c57f2a636$export$a397eb58038aa381, (v) => $3ba16c5c57f2a636$export$a397eb58038aa381 = v);
$parcel$export($3ba16c5c57f2a636$exports, "is-open", () => $3ba16c5c57f2a636$export$a9781837241c946d, (v) => $3ba16c5c57f2a636$export$a9781837241c946d = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-FocusRing-ring", () => $3ba16c5c57f2a636$export$4109102f950813a6, (v) => $3ba16c5c57f2a636$export$4109102f950813a6 = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-FocusRing", () => $3ba16c5c57f2a636$export$24c7f46a6e3605dd, (v) => $3ba16c5c57f2a636$export$24c7f46a6e3605dd = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-FocusRing--quiet", () => $3ba16c5c57f2a636$export$2927016961429360, (v) => $3ba16c5c57f2a636$export$2927016961429360 = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-overlay", () => $3ba16c5c57f2a636$export$39eed38e2296a1d6, (v) => $3ba16c5c57f2a636$export$39eed38e2296a1d6 = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-Tray", () => $3ba16c5c57f2a636$export$ce0704cd084c4f0d, (v) => $3ba16c5c57f2a636$export$ce0704cd084c4f0d = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-Tray--fixedHeight", () => $3ba16c5c57f2a636$export$79bfb05e59a300b, (v) => $3ba16c5c57f2a636$export$79bfb05e59a300b = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-Tray-wrapper", () => $3ba16c5c57f2a636$export$641b9b70c23dce66, (v) => $3ba16c5c57f2a636$export$641b9b70c23dce66 = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-overlay--bottom--open", () => $3ba16c5c57f2a636$export$a96755f239de61ee, (v) => $3ba16c5c57f2a636$export$a96755f239de61ee = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-overlay--left--open", () => $3ba16c5c57f2a636$export$a85c0c6b6cf8598c, (v) => $3ba16c5c57f2a636$export$a85c0c6b6cf8598c = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-overlay--right--open", () => $3ba16c5c57f2a636$export$85aaa87e65e040bf, (v) => $3ba16c5c57f2a636$export$85aaa87e65e040bf = v);
$parcel$export($3ba16c5c57f2a636$exports, "spectrum-overlay--top--open", () => $3ba16c5c57f2a636$export$b9768db7ffc1bc2, (v) => $3ba16c5c57f2a636$export$b9768db7ffc1bc2 = v);
var $3ba16c5c57f2a636$export$f39a09f249340e2a;
var $3ba16c5c57f2a636$export$8c4ee2c50c22c514;
var $3ba16c5c57f2a636$export$a397eb58038aa381;
var $3ba16c5c57f2a636$export$a9781837241c946d;
var $3ba16c5c57f2a636$export$4109102f950813a6;
var $3ba16c5c57f2a636$export$24c7f46a6e3605dd;
var $3ba16c5c57f2a636$export$2927016961429360;
var $3ba16c5c57f2a636$export$39eed38e2296a1d6;
var $3ba16c5c57f2a636$export$ce0704cd084c4f0d;
var $3ba16c5c57f2a636$export$79bfb05e59a300b;
var $3ba16c5c57f2a636$export$641b9b70c23dce66;
var $3ba16c5c57f2a636$export$a96755f239de61ee;
var $3ba16c5c57f2a636$export$a85c0c6b6cf8598c;
var $3ba16c5c57f2a636$export$85aaa87e65e040bf;
var $3ba16c5c57f2a636$export$b9768db7ffc1bc2;
$3ba16c5c57f2a636$export$f39a09f249340e2a = `UuxJvG_focus-ring`;
$3ba16c5c57f2a636$export$8c4ee2c50c22c514 = `UuxJvG_i18nFontFamily`;
$3ba16c5c57f2a636$export$a397eb58038aa381 = `UuxJvG_spectrum-overlay--open`;
$3ba16c5c57f2a636$export$a9781837241c946d = `UuxJvG_is-open ${$3ba16c5c57f2a636$export$a397eb58038aa381}`;
$3ba16c5c57f2a636$export$4109102f950813a6 = `UuxJvG_spectrum-FocusRing-ring`;
$3ba16c5c57f2a636$export$24c7f46a6e3605dd = `UuxJvG_spectrum-FocusRing ${$3ba16c5c57f2a636$export$4109102f950813a6}`;
$3ba16c5c57f2a636$export$2927016961429360 = `UuxJvG_spectrum-FocusRing--quiet`;
$3ba16c5c57f2a636$export$39eed38e2296a1d6 = `UuxJvG_spectrum-overlay`;
$3ba16c5c57f2a636$export$ce0704cd084c4f0d = `UuxJvG_spectrum-Tray ${$3ba16c5c57f2a636$export$39eed38e2296a1d6}`;
$3ba16c5c57f2a636$export$79bfb05e59a300b = `UuxJvG_spectrum-Tray--fixedHeight`;
$3ba16c5c57f2a636$export$641b9b70c23dce66 = `UuxJvG_spectrum-Tray-wrapper`;
$3ba16c5c57f2a636$export$a96755f239de61ee = `UuxJvG_spectrum-overlay--bottom--open`;
$3ba16c5c57f2a636$export$a85c0c6b6cf8598c = `UuxJvG_spectrum-overlay--left--open`;
$3ba16c5c57f2a636$export$85aaa87e65e040bf = `UuxJvG_spectrum-overlay--right--open`;
$3ba16c5c57f2a636$export$b9768db7ffc1bc2 = `UuxJvG_spectrum-overlay--top--open`;




function $6b386deda21be662$var$Tray(props, ref) {
    let { children: children, state: state, ...otherProps } = props;
    let domRef = (0, $eLPAa$reactspectrumutils.useDOMRef)(ref);
    let wrapperRef = (0, $eLPAa$react.useRef)(null);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $2aebdc186fd41e87$export$c6fdb837b070b4ff), {
        ...otherProps,
        isOpen: state.isOpen,
        nodeRef: wrapperRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement($6b386deda21be662$var$TrayWrapper, {
        ...props,
        wrapperRef: wrapperRef,
        ref: domRef
    }, children));
}
let $6b386deda21be662$var$TrayWrapper = /*#__PURE__*/ (0, $eLPAa$react.forwardRef)(function(props, ref) {
    let { children: children, isOpen: isOpen, isFixedHeight: isFixedHeight, state: state, wrapperRef: wrapperRef } = props;
    let { styleProps: styleProps } = (0, $eLPAa$reactspectrumutils.useStyleProps)(props);
    let { modalProps: modalProps, underlayProps: underlayProps } = (0, $eLPAa$reactariaoverlays.useModalOverlay)({
        ...props,
        isDismissable: true
    }, state, ref);
    // We need to measure the window's height in JS rather than using percentages in CSS
    // so that contents (e.g. menu) can inherit the max-height properly. Using percentages
    // does not work properly because there is nothing to base the percentage on.
    // We cannot use vh units because mobile browsers adjust the window height dynamically
    // when the address bar/bottom toolbars show and hide on scroll and vh units are fixed.
    // Also, the visual viewport is smaller than the layout viewport when the virtual keyboard
    // is up, so use the VisualViewport API to ensure the tray is displayed above the keyboard.
    let viewport = (0, $eLPAa$reactariautils.useViewportSize)();
    let wrapperStyle = {
        "--spectrum-visual-viewport-height": viewport.height + "px"
    };
    let wrapperClassName = (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3ba16c5c57f2a636$exports))), "spectrum-Tray-wrapper");
    let className = (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3ba16c5c57f2a636$exports))), "spectrum-Tray", {
        "is-open": isOpen,
        "spectrum-Tray--fixedHeight": isFixedHeight
    }, (0, $eLPAa$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e9a663ccb19ed1b0$exports))), "spectrum-Tray", "react-spectrum-Tray"), styleProps.className);
    // Attach Transition's nodeRef to outer most wrapper for node.reflow: https://github.com/reactjs/react-transition-group/blob/c89f807067b32eea6f68fd6c622190d88ced82e2/src/Transition.js#L231
    return /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        ref: wrapperRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $b900e75089bdd9cd$export$f360afc887607b02), {
        ...underlayProps,
        isOpen: isOpen
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        className: wrapperClassName,
        style: wrapperStyle
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement("div", {
        ...styleProps,
        ...modalProps,
        className: className,
        ref: ref,
        "data-testid": "tray"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $eLPAa$reactariaoverlays.DismissButton), {
        onDismiss: state.close
    }), children, /*#__PURE__*/ (0, ($parcel$interopDefault($eLPAa$react))).createElement((0, $eLPAa$reactariaoverlays.DismissButton), {
        onDismiss: state.close
    }))));
});
let $6b386deda21be662$export$4589ed81930b555c = /*#__PURE__*/ (0, $eLPAa$react.forwardRef)($6b386deda21be662$var$Tray);
