// @react-spectrum/image/dist/module.js
"use client";

import "./main.css";
import {useSlotProps as $9XFMV$useSlotProps, useStyleProps as $9XFMV$useStyleProps, useDOMRef as $9XFMV$useDOMRef, classNames as $9XFMV$classNames} from "@react-spectrum/utils";
import {filterDOMProps as $9XFMV$filterDOMProps} from "@react-aria/utils";
import $9XFMV$react from "react";
import {useProviderProps as $9XFMV$useProviderProps} from "@react-spectrum/provider";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $cf2f2ea6e1c0495a$exports = {};

$parcel$export($cf2f2ea6e1c0495a$exports, "spectrum-Image-img", () => $cf2f2ea6e1c0495a$export$3238177972e11200, (v) => $cf2f2ea6e1c0495a$export$3238177972e11200 = v);
var $cf2f2ea6e1c0495a$export$3238177972e11200;
$cf2f2ea6e1c0495a$export$3238177972e11200 = `Gv9sRq_spectrum-Image-img`;



// incomplete component for show right now
function $77e0ac4d4aa3e5ef$var$Image(props, ref) {
    /* Slots should be able to pass an alt for default behavior, but in Images, the child may know better. */ let userProvidedAlt = props.alt;
    props = (0, $9XFMV$useSlotProps)(props, "image");
    props = (0, $9XFMV$useProviderProps)(props);
    let { objectFit: objectFit, src: src, alt: alt, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $9XFMV$useStyleProps)(otherProps);
    let domRef = (0, $9XFMV$useDOMRef)(ref);
    if (alt == null) console.warn('The `alt` prop was not provided to an image. Add `alt` text for screen readers, or set `alt=""` prop to indicate that the image is decorative or redundant with displayed text and should not be announced by screen readers.');
    return /*#__PURE__*/ (0, $9XFMV$react).createElement("div", {
        ...(0, $9XFMV$filterDOMProps)(props),
        ...styleProps,
        className: (0, $9XFMV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($cf2f2ea6e1c0495a$exports))), styleProps.className),
        style: {
            ...styleProps.style,
            overflow: "hidden"
        },
        ref: domRef
    }, /*#__PURE__*/ (0, $9XFMV$react).createElement("img", {
        src: src,
        alt: userProvidedAlt || alt,
        style: {
            objectFit: objectFit
        },
        className: (0, $9XFMV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($cf2f2ea6e1c0495a$exports))), "spectrum-Image-img")
    }));
}
/**
 * Image is used to insert and display an image within a component.
 */ const $77e0ac4d4aa3e5ef$export$3e431a229df88919 = /*#__PURE__*/ (0, $9XFMV$react).forwardRef($77e0ac4d4aa3e5ef$var$Image);




export {$77e0ac4d4aa3e5ef$export$3e431a229df88919 as Image};
