// @aemforms/af-react-components/lib/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerticalTab = exports.HorizontalTab = exports.VerticalFlex = exports.HorizontalFlex = exports.CheckboxGroup = exports.DateField = exports.NumberField = exports.Repeater = exports.FileUploadComponent = exports.PlainText = exports.ComboBox = exports.RadioGroup = exports.Checkbox = exports.Panel = exports.Button = exports.TextField = exports.mappings = void 0;
/**
 * Copyright 2022 Adobe, Inc.
 *
 * Your access and use of this software is governed by the Adobe Customer Feedback Program Terms and Conditions or other Beta License Agreement signed by your employer and Adobe, Inc.. This software is NOT open source and may not be used without one of the foregoing licenses. Even with a foregoing license, your access and use of this file is limited to the earlier of (a) 180 days, (b) general availability of the product(s) which utilize this software (i.e. AEM Forms), (c) January 1, 2023, (d) Adobe providing notice to you that you may no longer use the software or that your beta trial has otherwise ended.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL ADOBE NOR ITS THIRD PARTY PROVIDERS AND PARTNERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 * @license Adobe Customer Feedback Program Terms and Conditions
 */
const mappings_1 = __importDefault(require("./utils/mappings"));
exports.mappings = mappings_1.default;
const TextField_1 = __importDefault(require("./components/TextField"));
exports.TextField = TextField_1.default;
const Button_1 = __importDefault(require("./components/Button"));
exports.Button = Button_1.default;
const Panel_1 = __importDefault(require("./components/Panel"));
exports.Panel = Panel_1.default;
const Checkbox_1 = __importDefault(require("./components/Checkbox"));
exports.Checkbox = Checkbox_1.default;
const RadioButtonGroup_1 = __importDefault(require("./components/RadioButtonGroup"));
exports.RadioGroup = RadioButtonGroup_1.default;
const DropDownList_1 = __importDefault(require("./components/DropDownList"));
exports.ComboBox = DropDownList_1.default;
const PlainText_1 = __importDefault(require("./components/PlainText"));
exports.PlainText = PlainText_1.default;
const FileUpload_1 = __importDefault(require("./components/FileUpload"));
exports.FileUploadComponent = FileUpload_1.default;
const Repeater_1 = __importDefault(require("./components/Repeater"));
exports.Repeater = Repeater_1.default;
const NumberField_1 = __importDefault(require("./components/NumberField"));
exports.NumberField = NumberField_1.default;
const Date_1 = __importDefault(require("./components/Date"));
exports.DateField = Date_1.default;
const CheckboxGroup_1 = __importDefault(require("./components/CheckboxGroup"));
exports.CheckboxGroup = CheckboxGroup_1.default;
const HorizontalFlex_1 = __importDefault(require("./components/flex/HorizontalFlex"));
exports.HorizontalFlex = HorizontalFlex_1.default;
const VerticalFlex_1 = __importDefault(require("./components/flex/VerticalFlex"));
exports.VerticalFlex = VerticalFlex_1.default;
const HorizontalTabs_1 = __importDefault(require("./components/tabs/HorizontalTabs"));
exports.HorizontalTab = HorizontalTabs_1.default;
const VerticalTabs_1 = __importDefault(require("./components/tabs/VerticalTabs"));
exports.VerticalTab = VerticalTabs_1.default;
__exportStar(require("./utils/SpectrumMappers"), exports);
