// @adobe/exc-app/tests/appapi.test.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
/*************************************************************************
 * Copyright 2020 Adobe
 * All Rights Reserved.
 *
 * NOTICE: Adobe permits you to use, modify, and distribute this file in
 * accordance with the terms of the Adobe license agreement accompanying
 * it. If you have received this file from a source other than Adobe,
 * then your use, modification, or distribution of it requires the prior
 * written permission of Adobe.
 **************************************************************************/
const appapi_1 = __importStar(require("../appapi"));
const Global_1 = __importDefault(require("../src/Global"));
describe('appapi', () => {
    beforeEach(() => {
        Global_1.default['exc-module-runtime'] = {
            appApi: () => {
                return {
                    get: (type) => {
                        if (type === 'analytics') {
                            return {
                                adobeBrandType: 'application',
                                appId: 'analytics',
                                description: 'Collect, organize, analyze and report on everything your customers do',
                                enabled: false,
                                functionalIconReact: 'DataAnalytics',
                                functionalIconUrl: 'https://cdn.experience-stage.adobe.net/assets/BrandIcons.33cbb3cf.svg#DataAnalytics',
                                href: 'https://analytics-discovery-qe.adobe.net/1.0/loginRedirect?allow_project_redirect=1&IMS=1',
                                iconUrl: '',
                                learnMoreLink: 'http://www.adobe.com/marketing-cloud/web-analytics.html',
                                longName: 'Adobe Analytics',
                                name: 'Analytics',
                                settingsAppId: 'analytics',
                                visible: false
                            };
                        }
                        if (type === 'target') {
                            return {
                                adobeBrandType: 'application',
                                appId: 'target',
                                description: 'Use experimentation, testing and machine learning automation to deliver the best content to each visitor',
                                enabled: false,
                                functionalIconReact: 'ContentProfile',
                                functionalIconUrl: 'https://cdn.experience-stage.adobe.net/assets/BrandIcons.33cbb3cf.svg#ContentProfile',
                                href: 'https://experience-stage.adobe.com/#/target',
                                iconUrl: '',
                                learnMoreLink: 'https://www.adobe.com/marketing/target.html',
                                longName: 'Adobe Target',
                                name: 'Target',
                                settingsAppId: '',
                                visible: false
                            };
                        }
                        return {};
                    }
                };
            }
        };
    });
    test('get', () => __awaiter(void 0, void 0, void 0, function* () {
        const analytics = yield appapi_1.default.get(appapi_1.AppIds.ANALYTICS);
        const target = yield appapi_1.default.get(appapi_1.AppIds.TARGET);
        expect(analytics.appId).toBe('analytics');
        expect(analytics.enabled).toBe(false);
        expect(analytics.functionalIconUrl).toBe('https://cdn.experience-stage.adobe.net/assets/BrandIcons.33cbb3cf.svg#DataAnalytics');
        expect(analytics.href).toBe('https://analytics-discovery-qe.adobe.net/1.0/loginRedirect?allow_project_redirect=1&IMS=1');
        expect(target.appId).toBe('target');
        expect(target.enabled).toBe(false);
        expect(target.functionalIconUrl).toBe('https://cdn.experience-stage.adobe.net/assets/BrandIcons.33cbb3cf.svg#ContentProfile');
        expect(target.href).toBe('https://experience-stage.adobe.com/#/target');
    }));
});
