// @adobe/exc-app/tests/nps.test.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
/*************************************************************************
 * Copyright 2020 Adobe
 * All Rights Reserved.
 *
 * NOTICE: Adobe permits you to use, modify, and distribute this file in
 * accordance with the terms of the Adobe license agreement accompanying
 * it. If you have received this file from a source other than Adobe,
 * then your use, modification, or distribution of it requires the prior
 * written permission of Adobe.
 **************************************************************************/
const Global_1 = __importDefault(require("../src/Global"));
const nps_1 = __importDefault(require("../nps"));
describe('nps/index.js', () => {
    beforeEach(() => {
        Global_1.default['exc-module-runtime'] = {
            nps: {
                config: {
                    enabled: true,
                    sampling: 100,
                    showToAdobeUsers: true
                },
                triggerNps: () => { }
            }
        };
    });
    it('correctly sets up the connection with module-runtime for triggerNps function', () => {
        const triggerNpsSpy = jest.spyOn(Global_1.default['exc-module-runtime'].nps, 'triggerNps');
        nps_1.default.triggerNps();
        expect(triggerNpsSpy).toBeCalledTimes(1);
    });
    it('correctly sets up the connection with module-runtime for config object', () => {
        nps_1.default.config = { enabled: true, sampling: 55 };
        expect(Global_1.default['exc-module-runtime'].nps.config).toStrictEqual({ enabled: true, sampling: 55 });
    });
});
