// @react-spectrum/dialog/dist/main.js
"use client";

var $a55551a02f9dcb5c$exports = require("./ar-AE.main.js");
var $d9c7f58ad2bbfa4c$exports = require("./bg-BG.main.js");
var $cc8c40dafb8d03b0$exports = require("./cs-CZ.main.js");
var $d970ec0fa7e711e7$exports = require("./da-DK.main.js");
var $c619cc654d2ca674$exports = require("./de-DE.main.js");
var $2d258a772c4fa392$exports = require("./el-GR.main.js");
var $88f649eb626e5bda$exports = require("./en-US.main.js");
var $4cd7244dbd86e2ba$exports = require("./es-ES.main.js");
var $86a88b5d274957dd$exports = require("./et-EE.main.js");
var $6b6909c1e7df2772$exports = require("./fi-FI.main.js");
var $c05d64ee0d510606$exports = require("./fr-FR.main.js");
var $b54ffde1ebba1e69$exports = require("./he-IL.main.js");
var $6d5732627fd95466$exports = require("./hr-HR.main.js");
var $525291eaaaf36ab1$exports = require("./hu-HU.main.js");
var $6dc44201330bcb0a$exports = require("./it-IT.main.js");
var $38fba7087c48d89b$exports = require("./ja-JP.main.js");
var $0d321e557ee0ff2b$exports = require("./ko-KR.main.js");
var $81ef7c03f6531c38$exports = require("./lt-LT.main.js");
var $dab9caf08f2773b2$exports = require("./lv-LV.main.js");
var $8af697b2f49b994d$exports = require("./nb-NO.main.js");
var $94b03262a47d5c5b$exports = require("./nl-NL.main.js");
var $9531f14fa20d8709$exports = require("./pl-PL.main.js");
var $d9e8b49831ccbf2a$exports = require("./pt-BR.main.js");
var $6a11f56de734f9d5$exports = require("./pt-PT.main.js");
var $e6f020c13f71513f$exports = require("./ro-RO.main.js");
var $3268a43e923b3cbf$exports = require("./ru-RU.main.js");
var $ff45268c9e1a78be$exports = require("./sk-SK.main.js");
var $f2cb6f88707095ee$exports = require("./sl-SI.main.js");
var $ece68e97a81612d8$exports = require("./sr-SP.main.js");
var $cce4ae8855cd8246$exports = require("./sv-SE.main.js");
var $763283ab7e680652$exports = require("./tr-TR.main.js");
var $ec9a4b7a09770ea4$exports = require("./uk-UA.main.js");
var $56f81c88b16f2b96$exports = require("./zh-CN.main.js");
var $6fb708b0a8fb135e$exports = require("./zh-TW.main.js");
require("./main.css");
var $iaoFN$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $iaoFN$reactspectrumbutton = require("@react-spectrum/button");
var $iaoFN$reactspectrumbuttongroup = require("@react-spectrum/buttongroup");
var $iaoFN$reactariautils = require("@react-aria/utils");
var $iaoFN$reactspectrumutils = require("@react-spectrum/utils");
var $iaoFN$reactspectrumview = require("@react-spectrum/view");
var $iaoFN$reactspectrumdivider = require("@react-spectrum/divider");
var $iaoFN$reactspectrumtext = require("@react-spectrum/text");
var $iaoFN$react = require("react");
var $iaoFN$reactariai18n = require("@react-aria/i18n");
var $iaoFN$spectrumiconsuiCrossLarge = require("@spectrum-icons/ui/CrossLarge");
var $iaoFN$reactspectrumlayout = require("@react-spectrum/layout");
var $iaoFN$reactariadialog = require("@react-aria/dialog");
var $iaoFN$reactspectrumoverlays = require("@react-spectrum/overlays");
var $iaoFN$reactstatelyoverlays = require("@react-stately/overlays");
var $iaoFN$reactariainteractions = require("@react-aria/interactions");
var $iaoFN$reactariaoverlays = require("@react-aria/overlays");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "AlertDialog", () => $66f8653ff750aa27$export$de466dd8317b0b75);
$parcel$export(module.exports, "Dialog", () => $a57fc2323bf98a33$export$3ddf2d174ce01153);
$parcel$export(module.exports, "DialogTrigger", () => $46e54a192dc519c9$export$2e1e1122cf0cba88);
$parcel$export(module.exports, "DialogContainer", () => $1b7050825e0eb5b6$export$547754aada6e339f);
$parcel$export(module.exports, "useDialogContainer", () => $1f05ce3abe26c4cc$export$a2f2d8fa6720dab1);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $0eb909ad2388d989$export$8b93a07348a7730c = (0, ($parcel$interopDefault($iaoFN$react))).createContext(null);



var $7827d180f5c8b41b$exports = {};


































$7827d180f5c8b41b$exports = {
    "ar-AE": $a55551a02f9dcb5c$exports,
    "bg-BG": $d9c7f58ad2bbfa4c$exports,
    "cs-CZ": $cc8c40dafb8d03b0$exports,
    "da-DK": $d970ec0fa7e711e7$exports,
    "de-DE": $c619cc654d2ca674$exports,
    "el-GR": $2d258a772c4fa392$exports,
    "en-US": $88f649eb626e5bda$exports,
    "es-ES": $4cd7244dbd86e2ba$exports,
    "et-EE": $86a88b5d274957dd$exports,
    "fi-FI": $6b6909c1e7df2772$exports,
    "fr-FR": $c05d64ee0d510606$exports,
    "he-IL": $b54ffde1ebba1e69$exports,
    "hr-HR": $6d5732627fd95466$exports,
    "hu-HU": $525291eaaaf36ab1$exports,
    "it-IT": $6dc44201330bcb0a$exports,
    "ja-JP": $38fba7087c48d89b$exports,
    "ko-KR": $0d321e557ee0ff2b$exports,
    "lt-LT": $81ef7c03f6531c38$exports,
    "lv-LV": $dab9caf08f2773b2$exports,
    "nb-NO": $8af697b2f49b994d$exports,
    "nl-NL": $94b03262a47d5c5b$exports,
    "pl-PL": $9531f14fa20d8709$exports,
    "pt-BR": $d9e8b49831ccbf2a$exports,
    "pt-PT": $6a11f56de734f9d5$exports,
    "ro-RO": $e6f020c13f71513f$exports,
    "ru-RU": $3268a43e923b3cbf$exports,
    "sk-SK": $ff45268c9e1a78be$exports,
    "sl-SI": $f2cb6f88707095ee$exports,
    "sr-SP": $ece68e97a81612d8$exports,
    "sv-SE": $cce4ae8855cd8246$exports,
    "tr-TR": $763283ab7e680652$exports,
    "uk-UA": $ec9a4b7a09770ea4$exports,
    "zh-CN": $56f81c88b16f2b96$exports,
    "zh-TW": $6fb708b0a8fb135e$exports
};




var $a74c0984b1adb651$exports = {};

$parcel$export($a74c0984b1adb651$exports, "buttonGroup", () => $a74c0984b1adb651$export$9c686a975407e95c, (v) => $a74c0984b1adb651$export$9c686a975407e95c = v);
$parcel$export($a74c0984b1adb651$exports, "buttonGroup-end", () => $a74c0984b1adb651$export$92daf1b0113da0, (v) => $a74c0984b1adb651$export$92daf1b0113da0 = v);
$parcel$export($a74c0984b1adb651$exports, "closeButton", () => $a74c0984b1adb651$export$efe229eab43a9869, (v) => $a74c0984b1adb651$export$efe229eab43a9869 = v);
$parcel$export($a74c0984b1adb651$exports, "content", () => $a74c0984b1adb651$export$a7db06668cad9adb, (v) => $a74c0984b1adb651$export$a7db06668cad9adb = v);
$parcel$export($a74c0984b1adb651$exports, "divider", () => $a74c0984b1adb651$export$e8c9778bf415dd08, (v) => $a74c0984b1adb651$export$e8c9778bf415dd08 = v);
$parcel$export($a74c0984b1adb651$exports, "focus-ring", () => $a74c0984b1adb651$export$f39a09f249340e2a, (v) => $a74c0984b1adb651$export$f39a09f249340e2a = v);
$parcel$export($a74c0984b1adb651$exports, "footer", () => $a74c0984b1adb651$export$adb608be33961c98, (v) => $a74c0984b1adb651$export$adb608be33961c98 = v);
$parcel$export($a74c0984b1adb651$exports, "footer-start", () => $a74c0984b1adb651$export$ba5cbcdd7db6a805, (v) => $a74c0984b1adb651$export$ba5cbcdd7db6a805 = v);
$parcel$export($a74c0984b1adb651$exports, "header", () => $a74c0984b1adb651$export$38e42c68cf43b5d4, (v) => $a74c0984b1adb651$export$38e42c68cf43b5d4 = v);
$parcel$export($a74c0984b1adb651$exports, "header-end", () => $a74c0984b1adb651$export$9ff84a8ef5e6eb7e, (v) => $a74c0984b1adb651$export$9ff84a8ef5e6eb7e = v);
$parcel$export($a74c0984b1adb651$exports, "header-start", () => $a74c0984b1adb651$export$6e2679145d8d7578, (v) => $a74c0984b1adb651$export$6e2679145d8d7578 = v);
$parcel$export($a74c0984b1adb651$exports, "heading", () => $a74c0984b1adb651$export$46e4a324ac90507f, (v) => $a74c0984b1adb651$export$46e4a324ac90507f = v);
$parcel$export($a74c0984b1adb651$exports, "heading-start", () => $a74c0984b1adb651$export$9f99175ea85abb45, (v) => $a74c0984b1adb651$export$9f99175ea85abb45 = v);
$parcel$export($a74c0984b1adb651$exports, "hero", () => $a74c0984b1adb651$export$124049a9e3d0d4c3, (v) => $a74c0984b1adb651$export$124049a9e3d0d4c3 = v);
$parcel$export($a74c0984b1adb651$exports, "i18nFontFamily", () => $a74c0984b1adb651$export$8c4ee2c50c22c514, (v) => $a74c0984b1adb651$export$8c4ee2c50c22c514 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Button", () => $a74c0984b1adb651$export$1db4cca5b4ade39a, (v) => $a74c0984b1adb651$export$1db4cca5b4ade39a = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog", () => $a74c0984b1adb651$export$4681f3bcac39b2b1, (v) => $a74c0984b1adb651$export$4681f3bcac39b2b1 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--dismissable", () => $a74c0984b1adb651$export$9461b285d631d575, (v) => $a74c0984b1adb651$export$9461b285d631d575 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--error", () => $a74c0984b1adb651$export$6b5beff56972a3b, (v) => $a74c0984b1adb651$export$6b5beff56972a3b = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--fullscreen", () => $a74c0984b1adb651$export$118506da5d3ee7c4, (v) => $a74c0984b1adb651$export$118506da5d3ee7c4 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--fullscreenTakeover", () => $a74c0984b1adb651$export$5c061b2d3664fda9, (v) => $a74c0984b1adb651$export$5c061b2d3664fda9 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--large", () => $a74c0984b1adb651$export$b73c0c2fca84be2b, (v) => $a74c0984b1adb651$export$b73c0c2fca84be2b = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--medium", () => $a74c0984b1adb651$export$b9b8ef443d0a67ca, (v) => $a74c0984b1adb651$export$b9b8ef443d0a67ca = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--noDivider", () => $a74c0984b1adb651$export$22425527a8943917, (v) => $a74c0984b1adb651$export$22425527a8943917 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--small", () => $a74c0984b1adb651$export$5f2d9cc47409bd63, (v) => $a74c0984b1adb651$export$5f2d9cc47409bd63 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog--warning", () => $a74c0984b1adb651$export$4e66396adf61534f, (v) => $a74c0984b1adb651$export$4e66396adf61534f = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-buttonGroup", () => $a74c0984b1adb651$export$570ae852861555b7, (v) => $a74c0984b1adb651$export$570ae852861555b7 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-buttonGroup--noFooter", () => $a74c0984b1adb651$export$39dd75369bf2e830, (v) => $a74c0984b1adb651$export$39dd75369bf2e830 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-closeButton", () => $a74c0984b1adb651$export$a8494287d763a75d, (v) => $a74c0984b1adb651$export$a8494287d763a75d = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-content", () => $a74c0984b1adb651$export$cb8eccd0f3639238, (v) => $a74c0984b1adb651$export$cb8eccd0f3639238 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-divider", () => $a74c0984b1adb651$export$dd7633d642d0bc1c, (v) => $a74c0984b1adb651$export$dd7633d642d0bc1c = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-footer", () => $a74c0984b1adb651$export$3ecb5025a06fe08f, (v) => $a74c0984b1adb651$export$3ecb5025a06fe08f = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-grid", () => $a74c0984b1adb651$export$c03cd5e23af67ff5, (v) => $a74c0984b1adb651$export$c03cd5e23af67ff5 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-header", () => $a74c0984b1adb651$export$1e158eaa6363e131, (v) => $a74c0984b1adb651$export$1e158eaa6363e131 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-header--noTypeIcon", () => $a74c0984b1adb651$export$141d8fd0950fea8b, (v) => $a74c0984b1adb651$export$141d8fd0950fea8b = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-heading", () => $a74c0984b1adb651$export$94335d73b7b209ec, (v) => $a74c0984b1adb651$export$94335d73b7b209ec = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-heading--noHeader", () => $a74c0984b1adb651$export$ade8307239736e89, (v) => $a74c0984b1adb651$export$ade8307239736e89 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-heading--noTypeIcon", () => $a74c0984b1adb651$export$c53ac004df29b46, (v) => $a74c0984b1adb651$export$c53ac004df29b46 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-hero", () => $a74c0984b1adb651$export$d405fb0a0231aa1a, (v) => $a74c0984b1adb651$export$d405fb0a0231aa1a = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-Dialog-typeIcon", () => $a74c0984b1adb651$export$5beee9480fc6ff07, (v) => $a74c0984b1adb651$export$5beee9480fc6ff07 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-FocusRing-ring", () => $a74c0984b1adb651$export$4109102f950813a6, (v) => $a74c0984b1adb651$export$4109102f950813a6 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-FocusRing", () => $a74c0984b1adb651$export$24c7f46a6e3605dd, (v) => $a74c0984b1adb651$export$24c7f46a6e3605dd = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-FocusRing--quiet", () => $a74c0984b1adb651$export$2927016961429360, (v) => $a74c0984b1adb651$export$2927016961429360 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-overlay", () => $a74c0984b1adb651$export$39eed38e2296a1d6, (v) => $a74c0984b1adb651$export$39eed38e2296a1d6 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-overlay--bottom--open", () => $a74c0984b1adb651$export$a96755f239de61ee, (v) => $a74c0984b1adb651$export$a96755f239de61ee = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-overlay--left--open", () => $a74c0984b1adb651$export$a85c0c6b6cf8598c, (v) => $a74c0984b1adb651$export$a85c0c6b6cf8598c = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-overlay--open", () => $a74c0984b1adb651$export$a397eb58038aa381, (v) => $a74c0984b1adb651$export$a397eb58038aa381 = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-overlay--right--open", () => $a74c0984b1adb651$export$85aaa87e65e040bf, (v) => $a74c0984b1adb651$export$85aaa87e65e040bf = v);
$parcel$export($a74c0984b1adb651$exports, "spectrum-overlay--top--open", () => $a74c0984b1adb651$export$b9768db7ffc1bc2, (v) => $a74c0984b1adb651$export$b9768db7ffc1bc2 = v);
$parcel$export($a74c0984b1adb651$exports, "typeIcon", () => $a74c0984b1adb651$export$7c523679e5521442, (v) => $a74c0984b1adb651$export$7c523679e5521442 = v);
$parcel$export($a74c0984b1adb651$exports, "typeIcon-end", () => $a74c0984b1adb651$export$b050c62b00f90727, (v) => $a74c0984b1adb651$export$b050c62b00f90727 = v);
var $a74c0984b1adb651$export$9c686a975407e95c;
var $a74c0984b1adb651$export$92daf1b0113da0;
var $a74c0984b1adb651$export$efe229eab43a9869;
var $a74c0984b1adb651$export$a7db06668cad9adb;
var $a74c0984b1adb651$export$e8c9778bf415dd08;
var $a74c0984b1adb651$export$f39a09f249340e2a;
var $a74c0984b1adb651$export$adb608be33961c98;
var $a74c0984b1adb651$export$ba5cbcdd7db6a805;
var $a74c0984b1adb651$export$38e42c68cf43b5d4;
var $a74c0984b1adb651$export$9ff84a8ef5e6eb7e;
var $a74c0984b1adb651$export$6e2679145d8d7578;
var $a74c0984b1adb651$export$46e4a324ac90507f;
var $a74c0984b1adb651$export$9f99175ea85abb45;
var $a74c0984b1adb651$export$124049a9e3d0d4c3;
var $a74c0984b1adb651$export$8c4ee2c50c22c514;
var $a74c0984b1adb651$export$1db4cca5b4ade39a;
var $a74c0984b1adb651$export$4681f3bcac39b2b1;
var $a74c0984b1adb651$export$9461b285d631d575;
var $a74c0984b1adb651$export$6b5beff56972a3b;
var $a74c0984b1adb651$export$118506da5d3ee7c4;
var $a74c0984b1adb651$export$5c061b2d3664fda9;
var $a74c0984b1adb651$export$b73c0c2fca84be2b;
var $a74c0984b1adb651$export$b9b8ef443d0a67ca;
var $a74c0984b1adb651$export$22425527a8943917;
var $a74c0984b1adb651$export$5f2d9cc47409bd63;
var $a74c0984b1adb651$export$4e66396adf61534f;
var $a74c0984b1adb651$export$570ae852861555b7;
var $a74c0984b1adb651$export$39dd75369bf2e830;
var $a74c0984b1adb651$export$a8494287d763a75d;
var $a74c0984b1adb651$export$cb8eccd0f3639238;
var $a74c0984b1adb651$export$dd7633d642d0bc1c;
var $a74c0984b1adb651$export$3ecb5025a06fe08f;
var $a74c0984b1adb651$export$c03cd5e23af67ff5;
var $a74c0984b1adb651$export$1e158eaa6363e131;
var $a74c0984b1adb651$export$141d8fd0950fea8b;
var $a74c0984b1adb651$export$94335d73b7b209ec;
var $a74c0984b1adb651$export$ade8307239736e89;
var $a74c0984b1adb651$export$c53ac004df29b46;
var $a74c0984b1adb651$export$d405fb0a0231aa1a;
var $a74c0984b1adb651$export$5beee9480fc6ff07;
var $a74c0984b1adb651$export$4109102f950813a6;
var $a74c0984b1adb651$export$24c7f46a6e3605dd;
var $a74c0984b1adb651$export$2927016961429360;
var $a74c0984b1adb651$export$39eed38e2296a1d6;
var $a74c0984b1adb651$export$a96755f239de61ee;
var $a74c0984b1adb651$export$a85c0c6b6cf8598c;
var $a74c0984b1adb651$export$a397eb58038aa381;
var $a74c0984b1adb651$export$85aaa87e65e040bf;
var $a74c0984b1adb651$export$b9768db7ffc1bc2;
var $a74c0984b1adb651$export$7c523679e5521442;
var $a74c0984b1adb651$export$b050c62b00f90727;
$a74c0984b1adb651$export$9c686a975407e95c = `h_OVWW_buttonGroup`;
$a74c0984b1adb651$export$92daf1b0113da0 = `h_OVWW_buttonGroup-end`;
$a74c0984b1adb651$export$efe229eab43a9869 = `h_OVWW_closeButton`;
$a74c0984b1adb651$export$a7db06668cad9adb = `h_OVWW_content`;
$a74c0984b1adb651$export$e8c9778bf415dd08 = `h_OVWW_divider`;
$a74c0984b1adb651$export$f39a09f249340e2a = `h_OVWW_focus-ring`;
$a74c0984b1adb651$export$adb608be33961c98 = `h_OVWW_footer`;
$a74c0984b1adb651$export$ba5cbcdd7db6a805 = `h_OVWW_footer-start`;
$a74c0984b1adb651$export$38e42c68cf43b5d4 = `h_OVWW_header`;
$a74c0984b1adb651$export$9ff84a8ef5e6eb7e = `h_OVWW_header-end`;
$a74c0984b1adb651$export$6e2679145d8d7578 = `h_OVWW_header-start`;
$a74c0984b1adb651$export$46e4a324ac90507f = `h_OVWW_heading`;
$a74c0984b1adb651$export$9f99175ea85abb45 = `h_OVWW_heading-start`;
$a74c0984b1adb651$export$124049a9e3d0d4c3 = `h_OVWW_hero`;
$a74c0984b1adb651$export$8c4ee2c50c22c514 = `h_OVWW_i18nFontFamily`;
$a74c0984b1adb651$export$1db4cca5b4ade39a = `h_OVWW_spectrum-Button`;
$a74c0984b1adb651$export$4681f3bcac39b2b1 = `h_OVWW_spectrum-Dialog`;
$a74c0984b1adb651$export$9461b285d631d575 = `h_OVWW_spectrum-Dialog--dismissable`;
$a74c0984b1adb651$export$6b5beff56972a3b = `h_OVWW_spectrum-Dialog--error`;
$a74c0984b1adb651$export$118506da5d3ee7c4 = `h_OVWW_spectrum-Dialog--fullscreen`;
$a74c0984b1adb651$export$5c061b2d3664fda9 = `h_OVWW_spectrum-Dialog--fullscreenTakeover`;
$a74c0984b1adb651$export$b73c0c2fca84be2b = `h_OVWW_spectrum-Dialog--large`;
$a74c0984b1adb651$export$b9b8ef443d0a67ca = `h_OVWW_spectrum-Dialog--medium`;
$a74c0984b1adb651$export$22425527a8943917 = `h_OVWW_spectrum-Dialog--noDivider`;
$a74c0984b1adb651$export$5f2d9cc47409bd63 = `h_OVWW_spectrum-Dialog--small`;
$a74c0984b1adb651$export$4e66396adf61534f = `h_OVWW_spectrum-Dialog--warning`;
$a74c0984b1adb651$export$570ae852861555b7 = `h_OVWW_spectrum-Dialog-buttonGroup`;
$a74c0984b1adb651$export$39dd75369bf2e830 = `h_OVWW_spectrum-Dialog-buttonGroup--noFooter`;
$a74c0984b1adb651$export$a8494287d763a75d = `h_OVWW_spectrum-Dialog-closeButton`;
$a74c0984b1adb651$export$cb8eccd0f3639238 = `h_OVWW_spectrum-Dialog-content`;
$a74c0984b1adb651$export$dd7633d642d0bc1c = `h_OVWW_spectrum-Dialog-divider`;
$a74c0984b1adb651$export$3ecb5025a06fe08f = `h_OVWW_spectrum-Dialog-footer`;
$a74c0984b1adb651$export$c03cd5e23af67ff5 = `h_OVWW_spectrum-Dialog-grid`;
$a74c0984b1adb651$export$1e158eaa6363e131 = `h_OVWW_spectrum-Dialog-header`;
$a74c0984b1adb651$export$141d8fd0950fea8b = `h_OVWW_spectrum-Dialog-header--noTypeIcon`;
$a74c0984b1adb651$export$94335d73b7b209ec = `h_OVWW_spectrum-Dialog-heading`;
$a74c0984b1adb651$export$ade8307239736e89 = `h_OVWW_spectrum-Dialog-heading--noHeader`;
$a74c0984b1adb651$export$c53ac004df29b46 = `h_OVWW_spectrum-Dialog-heading--noTypeIcon`;
$a74c0984b1adb651$export$d405fb0a0231aa1a = `h_OVWW_spectrum-Dialog-hero`;
$a74c0984b1adb651$export$5beee9480fc6ff07 = `h_OVWW_spectrum-Dialog-typeIcon`;
$a74c0984b1adb651$export$4109102f950813a6 = `h_OVWW_spectrum-FocusRing-ring`;
$a74c0984b1adb651$export$24c7f46a6e3605dd = `h_OVWW_spectrum-FocusRing ${$a74c0984b1adb651$export$4109102f950813a6}`;
$a74c0984b1adb651$export$2927016961429360 = `h_OVWW_spectrum-FocusRing--quiet`;
$a74c0984b1adb651$export$39eed38e2296a1d6 = `h_OVWW_spectrum-overlay`;
$a74c0984b1adb651$export$a96755f239de61ee = `h_OVWW_spectrum-overlay--bottom--open`;
$a74c0984b1adb651$export$a85c0c6b6cf8598c = `h_OVWW_spectrum-overlay--left--open`;
$a74c0984b1adb651$export$a397eb58038aa381 = `h_OVWW_spectrum-overlay--open`;
$a74c0984b1adb651$export$85aaa87e65e040bf = `h_OVWW_spectrum-overlay--right--open`;
$a74c0984b1adb651$export$b9768db7ffc1bc2 = `h_OVWW_spectrum-overlay--top--open`;
$a74c0984b1adb651$export$7c523679e5521442 = `h_OVWW_typeIcon`;
$a74c0984b1adb651$export$b050c62b00f90727 = `h_OVWW_typeIcon-end`;




let $a57fc2323bf98a33$var$sizeMap = {
    S: "small",
    M: "medium",
    L: "large",
    fullscreen: "fullscreen",
    fullscreenTakeover: "fullscreenTakeover"
};
function $a57fc2323bf98a33$var$Dialog(props, ref) {
    props = (0, $iaoFN$reactspectrumutils.useSlotProps)(props, "dialog");
    let { type: type = "modal", ...contextProps } = (0, $iaoFN$react.useContext)((0, $0eb909ad2388d989$export$8b93a07348a7730c)) || {};
    let { children: children, isDismissable: isDismissable = contextProps.isDismissable, onDismiss: onDismiss = contextProps.onClose, size: size, ...otherProps } = props;
    let stringFormatter = (0, $iaoFN$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($7827d180f5c8b41b$exports))), "@react-spectrum/dialog");
    let { styleProps: styleProps } = (0, $iaoFN$reactspectrumutils.useStyleProps)(otherProps);
    size = type === "popover" ? size || "S" : size || "L";
    let domRef = (0, $iaoFN$reactspectrumutils.useDOMRef)(ref);
    let gridRef = (0, $iaoFN$react.useRef)(null);
    let sizeVariant = $a57fc2323bf98a33$var$sizeMap[type] || $a57fc2323bf98a33$var$sizeMap[size];
    let { dialogProps: dialogProps, titleProps: titleProps } = (0, $iaoFN$reactariadialog.useDialog)((0, $iaoFN$reactariautils.mergeProps)(contextProps, props), domRef);
    let hasHeader = (0, $iaoFN$reactspectrumutils.useHasChild)(`.${(0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-header"]}`, (0, $iaoFN$reactspectrumutils.unwrapDOMRef)(gridRef));
    let hasHeading = (0, $iaoFN$reactspectrumutils.useHasChild)(`.${(0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-heading"]}`, (0, $iaoFN$reactspectrumutils.unwrapDOMRef)(gridRef));
    let hasFooter = (0, $iaoFN$reactspectrumutils.useHasChild)(`.${(0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-footer"]}`, (0, $iaoFN$reactspectrumutils.unwrapDOMRef)(gridRef));
    let hasTypeIcon = (0, $iaoFN$reactspectrumutils.useHasChild)(`.${(0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-typeIcon"]}`, (0, $iaoFN$reactspectrumutils.unwrapDOMRef)(gridRef));
    let slots = (0, $iaoFN$react.useMemo)(()=>({
            hero: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-hero"]
            },
            heading: {
                UNSAFE_className: (0, $iaoFN$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))), "spectrum-Dialog-heading", {
                    "spectrum-Dialog-heading--noHeader": !hasHeader,
                    "spectrum-Dialog-heading--noTypeIcon": !hasTypeIcon
                }),
                level: 2,
                ...titleProps
            },
            header: {
                UNSAFE_className: (0, $iaoFN$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))), "spectrum-Dialog-header", {
                    "spectrum-Dialog-header--noHeading": !hasHeading,
                    "spectrum-Dialog-header--noTypeIcon": !hasTypeIcon
                })
            },
            typeIcon: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-typeIcon"]
            },
            divider: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-divider"],
                size: "M"
            },
            content: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-content"]
            },
            footer: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-footer"]
            },
            buttonGroup: {
                UNSAFE_className: (0, $iaoFN$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))), "spectrum-Dialog-buttonGroup", {
                    "spectrum-Dialog-buttonGroup--noFooter": !hasFooter
                }),
                align: "end"
            }
        }), [
        hasFooter,
        hasHeader,
        titleProps
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement("section", {
        ...styleProps,
        ...dialogProps,
        className: (0, $iaoFN$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))), "spectrum-Dialog", {
            [`spectrum-Dialog--${sizeVariant}`]: sizeVariant,
            "spectrum-Dialog--dismissable": isDismissable
        }, styleProps.className),
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumlayout.Grid), {
        ref: gridRef,
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-grid"]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumutils.SlotProvider), {
        slots: slots
    }, children), isDismissable && /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumbutton.ActionButton), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports)))["spectrum-Dialog-closeButton"],
        isQuiet: true,
        "aria-label": stringFormatter.format("dismiss"),
        onPress: onDismiss
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, ($parcel$interopDefault($iaoFN$spectrumiconsuiCrossLarge))), null))));
}
/**
 * Dialogs are windows containing contextual information, tasks, or workflows that appear over the user interface.
 * Depending on the kind of Dialog, further interactions may be blocked until the Dialog is acknowledged.
 */ let $a57fc2323bf98a33$export$3ddf2d174ce01153 = /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).forwardRef($a57fc2323bf98a33$var$Dialog);









/**
 * AlertDialogs are a specific type of Dialog. They display important information that users need to acknowledge.
 */ function $66f8653ff750aa27$var$AlertDialog(props, ref) {
    let { onClose: onClose = ()=>{} } = (0, $iaoFN$react.useContext)((0, $0eb909ad2388d989$export$8b93a07348a7730c)) || {};
    let { variant: variant, children: children, primaryActionLabel: primaryActionLabel, secondaryActionLabel: secondaryActionLabel, cancelLabel: cancelLabel, autoFocusButton: autoFocusButton, title: title, isPrimaryActionDisabled: isPrimaryActionDisabled, isSecondaryActionDisabled: isSecondaryActionDisabled, onCancel: onCancel = ()=>{}, onPrimaryAction: onPrimaryAction = ()=>{}, onSecondaryAction: onSecondaryAction = ()=>{}, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $iaoFN$reactspectrumutils.useStyleProps)(otherProps);
    let stringFormatter = (0, $iaoFN$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($7827d180f5c8b41b$exports))), "@react-spectrum/dialog");
    let confirmVariant = "primary";
    if (variant) {
        if (variant === "confirmation") confirmVariant = "cta";
        else if (variant === "destructive") confirmVariant = "negative";
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $a57fc2323bf98a33$export$3ddf2d174ce01153), {
        UNSAFE_style: styleProps.style,
        UNSAFE_className: (0, $iaoFN$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a74c0984b1adb651$exports))), {
            [`spectrum-Dialog--${variant}`]: variant
        }, styleProps.className),
        isHidden: styleProps.hidden,
        size: "M",
        role: "alertdialog",
        ref: ref
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumtext.Heading), null, title), (variant === "error" || variant === "warning") && /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, ($parcel$interopDefault($iaoFN$spectrumiconsuiAlertMedium))), {
        slot: "typeIcon",
        "aria-label": stringFormatter.format("alert")
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumdivider.Divider), null), /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumview.Content), null, children), /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumbuttongroup.ButtonGroup), {
        align: "end"
    }, cancelLabel && /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumbutton.Button), {
        variant: "secondary",
        onPress: ()=>(0, $iaoFN$reactariautils.chain)(onClose(), onCancel()),
        autoFocus: autoFocusButton === "cancel"
    }, cancelLabel), secondaryActionLabel && /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumbutton.Button), {
        variant: "secondary",
        onPress: ()=>(0, $iaoFN$reactariautils.chain)(onClose(), onSecondaryAction()),
        isDisabled: isSecondaryActionDisabled,
        autoFocus: autoFocusButton === "secondary"
    }, secondaryActionLabel), /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumbutton.Button), {
        variant: confirmVariant,
        onPress: ()=>(0, $iaoFN$reactariautils.chain)(onClose(), onPrimaryAction()),
        isDisabled: isPrimaryActionDisabled,
        autoFocus: autoFocusButton === "primary"
    }, primaryActionLabel)));
}
/**
 * AlertDialogs are a specific type of Dialog. They display important information that users need to acknowledge.
 */ let $66f8653ff750aa27$export$de466dd8317b0b75 = /*#__PURE__*/ (0, $iaoFN$react.forwardRef)($66f8653ff750aa27$var$AlertDialog);



/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






function $46e54a192dc519c9$var$DialogTrigger(props) {
    let { children: children, type: type = "modal", mobileType: mobileType = type === "popover" ? "modal" : type, hideArrow: hideArrow, targetRef: targetRef, isDismissable: isDismissable, isKeyboardDismissDisabled: isKeyboardDismissDisabled, ...positionProps } = props;
    if (!Array.isArray(children) || children.length > 2) throw new Error("DialogTrigger must have exactly 2 children");
    // if a function is passed as the second child, it won't appear in toArray
    let [trigger, content] = children;
    // On small devices, show a modal or tray instead of a popover.
    let isMobile = (0, $iaoFN$reactspectrumutils.useIsMobileDevice)();
    if (isMobile) {
        // handle cases where desktop popovers need a close button for the mobile modal view
        if (type !== "modal" && mobileType === "modal") isDismissable = true;
        type = mobileType;
    }
    let state = (0, $iaoFN$reactstatelyoverlays.useOverlayTriggerState)(props);
    let wasOpen = (0, $iaoFN$react.useRef)(false);
    (0, $iaoFN$react.useEffect)(()=>{
        wasOpen.current = state.isOpen;
    }, [
        state.isOpen
    ]);
    let isExiting = (0, $iaoFN$react.useRef)(false);
    let onExiting = ()=>isExiting.current = true;
    let onExited = ()=>isExiting.current = false;
    // eslint-disable-next-line arrow-body-style
    (0, $iaoFN$react.useEffect)(()=>{
        return ()=>{
            if ((wasOpen.current || isExiting.current) && type !== "popover" && type !== "tray") console.warn("A DialogTrigger unmounted while open. This is likely due to being placed within a trigger that unmounts or inside a conditional. Consider using a DialogContainer instead.");
        };
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    if (type === "popover") return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement($46e54a192dc519c9$var$PopoverTrigger, {
        ...positionProps,
        state: state,
        targetRef: targetRef,
        trigger: trigger,
        content: content,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled,
        hideArrow: hideArrow
    });
    let renderOverlay = ()=>{
        switch(type){
            case "fullscreen":
            case "fullscreenTakeover":
            case "modal":
                return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumoverlays.Modal), {
                    state: state,
                    isDismissable: type === "modal" ? isDismissable : false,
                    type: type,
                    isKeyboardDismissDisabled: isKeyboardDismissDisabled,
                    onExiting: onExiting,
                    onExited: onExited
                }, typeof content === "function" ? content(state.close) : content);
            case "tray":
                return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumoverlays.Tray), {
                    state: state,
                    isKeyboardDismissDisabled: isKeyboardDismissDisabled
                }, typeof content === "function" ? content(state.close) : content);
        }
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement($46e54a192dc519c9$var$DialogTriggerBase, {
        type: type,
        state: state,
        isDismissable: isDismissable,
        trigger: trigger,
        overlay: renderOverlay()
    });
}
// Support DialogTrigger inside components using CollectionBuilder.
$46e54a192dc519c9$var$DialogTrigger.getCollectionNode = function*(props) {
    // @ts-ignore - seems like types are wrong. Function children work fine.
    let [trigger] = (0, ($parcel$interopDefault($iaoFN$react))).Children.toArray(props.children);
    let [, content] = props.children;
    yield {
        element: trigger,
        wrapper: (element)=>/*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement($46e54a192dc519c9$var$DialogTrigger, {
                key: element.key,
                ...props
            }, element, content)
    };
};
/**
 * DialogTrigger serves as a wrapper around a Dialog and its associated trigger, linking the Dialog's
 * open state with the trigger's press state. Additionally, it allows you to customize the type and
 * positioning of the Dialog.
 */ // We don't want getCollectionNode to show up in the type definition
let $46e54a192dc519c9$export$2e1e1122cf0cba88 = $46e54a192dc519c9$var$DialogTrigger;
function $46e54a192dc519c9$var$PopoverTrigger({ state: state, targetRef: targetRef, trigger: trigger, content: content, hideArrow: hideArrow, ...props }) {
    let triggerRef = (0, $iaoFN$react.useRef)(null);
    let { triggerProps: triggerProps, overlayProps: overlayProps } = (0, $iaoFN$reactariaoverlays.useOverlayTrigger)({
        type: "dialog"
    }, state, triggerRef);
    let triggerPropsWithRef = {
        ...triggerProps,
        ref: targetRef ? undefined : triggerRef
    };
    let overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumoverlays.Popover), {
        ...props,
        hideArrow: hideArrow,
        triggerRef: targetRef || triggerRef,
        state: state
    }, typeof content === "function" ? content(state.close) : content);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement($46e54a192dc519c9$var$DialogTriggerBase, {
        type: "popover",
        state: state,
        triggerProps: triggerPropsWithRef,
        dialogProps: overlayProps,
        trigger: trigger,
        overlay: overlay
    });
}
function $46e54a192dc519c9$var$DialogTriggerBase({ type: type, state: state, isDismissable: isDismissable, dialogProps: dialogProps = {}, triggerProps: triggerProps = {}, overlay: overlay, trigger: trigger }) {
    let context = {
        type: type,
        onClose: state.close,
        isDismissable: isDismissable,
        ...dialogProps
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$react.Fragment), null, /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactariainteractions.PressResponder), {
        ...triggerProps,
        onPress: state.toggle,
        isPressed: state.isOpen && type !== "modal" && type !== "fullscreen" && type !== "fullscreenTakeover"
    }, trigger), /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $0eb909ad2388d989$export$8b93a07348a7730c).Provider, {
        value: context
    }, overlay));
}


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



function $1b7050825e0eb5b6$export$547754aada6e339f(props) {
    let { children: children, type: type = "modal", onDismiss: onDismiss, isDismissable: isDismissable, isKeyboardDismissDisabled: isKeyboardDismissDisabled } = props;
    let childArray = (0, ($parcel$interopDefault($iaoFN$react))).Children.toArray(children);
    if (childArray.length > 1) throw new Error("Only a single child can be passed to DialogContainer.");
    let [lastChild, setLastChild] = (0, $iaoFN$react.useState)(null);
    // React.Children.toArray mutates the children, and we need them to be stable
    // between renders so that the lastChild comparison works.
    let child = undefined;
    if (Array.isArray(children)) child = children.find((0, ($parcel$interopDefault($iaoFN$react))).isValidElement);
    else if (/*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).isValidElement(children)) child = children;
    if (child && child !== lastChild) setLastChild(child);
    let context = {
        type: type,
        onClose: onDismiss,
        isDismissable: isDismissable
    };
    let state = (0, $iaoFN$reactstatelyoverlays.useOverlayTriggerState)({
        isOpen: !!child,
        onOpenChange: (isOpen)=>{
            if (!isOpen) onDismiss();
        }
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $iaoFN$reactspectrumoverlays.Modal), {
        state: state,
        type: type,
        isDismissable: isDismissable,
        isKeyboardDismissDisabled: isKeyboardDismissDisabled
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($iaoFN$react))).createElement((0, $0eb909ad2388d989$export$8b93a07348a7730c).Provider, {
        value: context
    }, lastChild));
}


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

function $1f05ce3abe26c4cc$export$a2f2d8fa6720dab1() {
    let context = (0, $iaoFN$react.useContext)((0, $0eb909ad2388d989$export$8b93a07348a7730c));
    if (!context) throw new Error("Cannot call useDialogContext outside a <DialogTrigger> or <DialogContainer>.");
    return {
        type: context.type,
        dismiss () {
            context === null || context === void 0 ? void 0 : context.onClose();
        }
    };
}
