// @react-spectrum/menu/dist/main.js
"use client";

require("./main.css");
var $517603d8c7b8e5e5$exports = require("./ar-AE.main.js");
var $da4736ecb2b2fc3d$exports = require("./bg-BG.main.js");
var $474902e64d15239d$exports = require("./cs-CZ.main.js");
var $ab827474128c0a44$exports = require("./da-DK.main.js");
var $6decd364a8a60c6f$exports = require("./de-DE.main.js");
var $a79ffc3f42b40ea0$exports = require("./el-GR.main.js");
var $2b0d2ea968c3f427$exports = require("./en-US.main.js");
var $3655ccfea2cb795f$exports = require("./es-ES.main.js");
var $fdd5fdd03c65691f$exports = require("./et-EE.main.js");
var $5b53a5d82dacf660$exports = require("./fi-FI.main.js");
var $53cf7303c3810c0d$exports = require("./fr-FR.main.js");
var $1e6d66e0b4792817$exports = require("./he-IL.main.js");
var $eb204bb933ebd45f$exports = require("./hr-HR.main.js");
var $3b770de73ab860bc$exports = require("./hu-HU.main.js");
var $010ff727ed3102ae$exports = require("./it-IT.main.js");
var $dead61005b56a753$exports = require("./ja-JP.main.js");
var $059a8cc6da15c9c5$exports = require("./ko-KR.main.js");
var $115228ca3cb07b54$exports = require("./lt-LT.main.js");
var $1b70a2b594f0946f$exports = require("./lv-LV.main.js");
var $10afb7333b6cd21a$exports = require("./nb-NO.main.js");
var $6fb3df0b34cc117d$exports = require("./nl-NL.main.js");
var $f6cf802e78a19c98$exports = require("./pl-PL.main.js");
var $7401b34fc83fec2d$exports = require("./pt-BR.main.js");
var $812f012e4938f38d$exports = require("./pt-PT.main.js");
var $8c983e7e2aa14f7a$exports = require("./ro-RO.main.js");
var $5eddd4454bdacc3e$exports = require("./ru-RU.main.js");
var $3dc143fbc7df8c0b$exports = require("./sk-SK.main.js");
var $089b9b9eacba8462$exports = require("./sl-SI.main.js");
var $ae817dc85563e7c9$exports = require("./sr-SP.main.js");
var $39f858f9383858ce$exports = require("./sv-SE.main.js");
var $fb520ca0abee09d8$exports = require("./tr-TR.main.js");
var $7950b4249316e2c0$exports = require("./uk-UA.main.js");
var $86c8e6b13fa9c9ad$exports = require("./zh-CN.main.js");
var $f3cd1d6d75907b41$exports = require("./zh-TW.main.js");
var $3ThwF$reactstatelycollections = require("@react-stately/collections");
var $3ThwF$reactspectrumutils = require("@react-spectrum/utils");
var $3ThwF$reactspectrumoverlays = require("@react-spectrum/overlays");
var $3ThwF$reactariainteractions = require("@react-aria/interactions");
var $3ThwF$react = require("react");
var $3ThwF$reactariamenu = require("@react-aria/menu");
var $3ThwF$reactstatelymenu = require("@react-stately/menu");
var $3ThwF$reactariautils = require("@react-aria/utils");
var $3ThwF$reactdom = require("react-dom");
var $3ThwF$reactariai18n = require("@react-aria/i18n");
var $3ThwF$reactspectrumbutton = require("@react-spectrum/button");
var $3ThwF$spectrumiconsuiArrowDownSmall = require("@spectrum-icons/ui/ArrowDownSmall");
var $3ThwF$reactariafocus = require("@react-aria/focus");
var $3ThwF$reactstatelytree = require("@react-stately/tree");
var $3ThwF$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $3ThwF$spectrumiconsworkflowChevronLeft = require("@spectrum-icons/workflow/ChevronLeft");
var $3ThwF$spectrumiconsworkflowChevronRight = require("@spectrum-icons/workflow/ChevronRight");
var $3ThwF$reactspectrumlayout = require("@react-spectrum/layout");
var $3ThwF$spectrumiconsworkflowInfoOutline = require("@spectrum-icons/workflow/InfoOutline");
var $3ThwF$reactspectrumtext = require("@react-spectrum/text");
var $3ThwF$reactariaseparator = require("@react-aria/separator");
var $3ThwF$spectrumiconsworkflowMore = require("@spectrum-icons/workflow/More");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "MenuTrigger", () => $e56311f7204399c7$export$27d2ad3c5815583e);
$parcel$export(module.exports, "SubmenuTrigger", () => $86ed235c4695ff3f$export$ecabc99eeffab7ca);
$parcel$export(module.exports, "Menu", () => $63a7dff9cbe2d046$export$d9b273488cd8ce6f);
$parcel$export(module.exports, "ActionMenu", () => $53a2f2636c97a561$export$ed57a210b9d13bb6);
$parcel$export(module.exports, "ContextualHelpTrigger", () => $db380858d398b658$export$5413b169fff83e61);
$parcel$export(module.exports, "Item", () => $3ThwF$reactstatelycollections.Item);
$parcel$export(module.exports, "Section", () => $3ThwF$reactstatelycollections.Section);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $cac834c4bc0a51d3$export$c7e742effb1c51e2 = (0, ($parcel$interopDefault($3ThwF$react))).createContext({});
function $cac834c4bc0a51d3$export$21c7ab35b39f78ec() {
    return (0, $3ThwF$react.useContext)($cac834c4bc0a51d3$export$c7e742effb1c51e2);
}
const $cac834c4bc0a51d3$export$8d97fe02339fc0e3 = (0, ($parcel$interopDefault($3ThwF$react))).createContext(undefined);
function $cac834c4bc0a51d3$export$dc2161044aa0b36d() {
    return (0, $3ThwF$react.useContext)($cac834c4bc0a51d3$export$8d97fe02339fc0e3);
}
const $cac834c4bc0a51d3$export$24aad8519b95b41b = (0, ($parcel$interopDefault($3ThwF$react))).createContext(undefined);
function $cac834c4bc0a51d3$export$efa3856fc0e85e7f() {
    return (0, $3ThwF$react.useContext)($cac834c4bc0a51d3$export$24aad8519b95b41b);
}





var $3deee0fd2b46e64f$exports = {};

$parcel$export($3deee0fd2b46e64f$exports, "checkmark", () => $3deee0fd2b46e64f$export$292abbf31ed842a, (v) => $3deee0fd2b46e64f$export$292abbf31ed842a = v);
$parcel$export($3deee0fd2b46e64f$exports, "chevron", () => $3deee0fd2b46e64f$export$980c1089c0604ea3, (v) => $3deee0fd2b46e64f$export$980c1089c0604ea3 = v);
$parcel$export($3deee0fd2b46e64f$exports, "description", () => $3deee0fd2b46e64f$export$6d59db4903f20f7d, (v) => $3deee0fd2b46e64f$export$6d59db4903f20f7d = v);
$parcel$export($3deee0fd2b46e64f$exports, "end", () => $3deee0fd2b46e64f$export$bd5df0f255a350f8, (v) => $3deee0fd2b46e64f$export$bd5df0f255a350f8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "focus-ring", () => $3deee0fd2b46e64f$export$f39a09f249340e2a, (v) => $3deee0fd2b46e64f$export$f39a09f249340e2a = v);
$parcel$export($3deee0fd2b46e64f$exports, "i18nFontFamily", () => $3deee0fd2b46e64f$export$8c4ee2c50c22c514, (v) => $3deee0fd2b46e64f$export$8c4ee2c50c22c514 = v);
$parcel$export($3deee0fd2b46e64f$exports, "icon", () => $3deee0fd2b46e64f$export$1ca1ec8b29a4ce27, (v) => $3deee0fd2b46e64f$export$1ca1ec8b29a4ce27 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-active", () => $3deee0fd2b46e64f$export$20fd0f7cd4e6112f, (v) => $3deee0fd2b46e64f$export$20fd0f7cd4e6112f = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-disabled", () => $3deee0fd2b46e64f$export$d35bc1e505d1ebbf, (v) => $3deee0fd2b46e64f$export$d35bc1e505d1ebbf = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-expanded", () => $3deee0fd2b46e64f$export$911ebb89bba4b634, (v) => $3deee0fd2b46e64f$export$911ebb89bba4b634 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-focused", () => $3deee0fd2b46e64f$export$e7dc768d35940237, (v) => $3deee0fd2b46e64f$export$e7dc768d35940237 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-highlighted", () => $3deee0fd2b46e64f$export$451c613d0772eb99, (v) => $3deee0fd2b46e64f$export$451c613d0772eb99 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-hovered", () => $3deee0fd2b46e64f$export$b8813cd5d7824ce7, (v) => $3deee0fd2b46e64f$export$b8813cd5d7824ce7 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-open", () => $3deee0fd2b46e64f$export$a9781837241c946d, (v) => $3deee0fd2b46e64f$export$a9781837241c946d = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-selectable", () => $3deee0fd2b46e64f$export$fbd22ba224473193, (v) => $3deee0fd2b46e64f$export$fbd22ba224473193 = v);
$parcel$export($3deee0fd2b46e64f$exports, "is-selected", () => $3deee0fd2b46e64f$export$1e0fb04f31d3c22a, (v) => $3deee0fd2b46e64f$export$1e0fb04f31d3c22a = v);
$parcel$export($3deee0fd2b46e64f$exports, "keyboard", () => $3deee0fd2b46e64f$export$cfdd767f5a513509, (v) => $3deee0fd2b46e64f$export$cfdd767f5a513509 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-FocusRing-ring", () => $3deee0fd2b46e64f$export$4109102f950813a6, (v) => $3deee0fd2b46e64f$export$4109102f950813a6 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-FocusRing", () => $3deee0fd2b46e64f$export$24c7f46a6e3605dd, (v) => $3deee0fd2b46e64f$export$24c7f46a6e3605dd = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-FocusRing--quiet", () => $3deee0fd2b46e64f$export$2927016961429360, (v) => $3deee0fd2b46e64f$export$2927016961429360 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Icon", () => $3deee0fd2b46e64f$export$d374b04f30360026, (v) => $3deee0fd2b46e64f$export$d374b04f30360026 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu", () => $3deee0fd2b46e64f$export$fae6af6decd50ac5, (v) => $3deee0fd2b46e64f$export$fae6af6decd50ac5 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-avatar", () => $3deee0fd2b46e64f$export$944daf9b7d423a02, (v) => $3deee0fd2b46e64f$export$944daf9b7d423a02 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-checkmark", () => $3deee0fd2b46e64f$export$8e15a726e4f7e62d, (v) => $3deee0fd2b46e64f$export$8e15a726e4f7e62d = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-chevron", () => $3deee0fd2b46e64f$export$ca45e0db366b8e6c, (v) => $3deee0fd2b46e64f$export$ca45e0db366b8e6c = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-description", () => $3deee0fd2b46e64f$export$36f1275051e39ead, (v) => $3deee0fd2b46e64f$export$36f1275051e39ead = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-divider", () => $3deee0fd2b46e64f$export$e892026f534efed8, (v) => $3deee0fd2b46e64f$export$e892026f534efed8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-end", () => $3deee0fd2b46e64f$export$926a1bb74ed65667, (v) => $3deee0fd2b46e64f$export$926a1bb74ed65667 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-icon", () => $3deee0fd2b46e64f$export$cbbae63ef9e44c7a, (v) => $3deee0fd2b46e64f$export$cbbae63ef9e44c7a = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-item", () => $3deee0fd2b46e64f$export$45faad328e6c532b, (v) => $3deee0fd2b46e64f$export$45faad328e6c532b = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemGrid", () => $3deee0fd2b46e64f$export$ee5f52784f57f9c8, (v) => $3deee0fd2b46e64f$export$ee5f52784f57f9c8 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemIcon", () => $3deee0fd2b46e64f$export$f431001e1bb54633, (v) => $3deee0fd2b46e64f$export$f431001e1bb54633 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemLabel", () => $3deee0fd2b46e64f$export$1aa6419c5c650567, (v) => $3deee0fd2b46e64f$export$1aa6419c5c650567 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-itemLabel--wrapping", () => $3deee0fd2b46e64f$export$465092164a8e1d98, (v) => $3deee0fd2b46e64f$export$465092164a8e1d98 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-keyboard", () => $3deee0fd2b46e64f$export$4f9817a7dff6ba0f, (v) => $3deee0fd2b46e64f$export$4f9817a7dff6ba0f = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-popover", () => $3deee0fd2b46e64f$export$d959f30786125b, (v) => $3deee0fd2b46e64f$export$d959f30786125b = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-section--isFirst", () => $3deee0fd2b46e64f$export$dae331c15950f1e, (v) => $3deee0fd2b46e64f$export$dae331c15950f1e = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-section--isLast", () => $3deee0fd2b46e64f$export$9778864c5b622fbc, (v) => $3deee0fd2b46e64f$export$9778864c5b622fbc = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-section--noHeading", () => $3deee0fd2b46e64f$export$8b61ec49b96ad17e, (v) => $3deee0fd2b46e64f$export$8b61ec49b96ad17e = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-sectionHeading", () => $3deee0fd2b46e64f$export$c20ea4b35020fbdd, (v) => $3deee0fd2b46e64f$export$c20ea4b35020fbdd = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-subdialog", () => $3deee0fd2b46e64f$export$d7e47bfca92b5744, (v) => $3deee0fd2b46e64f$export$d7e47bfca92b5744 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-wrapper", () => $3deee0fd2b46e64f$export$76d36e55740ad742, (v) => $3deee0fd2b46e64f$export$76d36e55740ad742 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Menu-wrapper--isMobile", () => $3deee0fd2b46e64f$export$c873119be41fdb72, (v) => $3deee0fd2b46e64f$export$c873119be41fdb72 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Submenu-heading", () => $3deee0fd2b46e64f$export$c830d2588e34dae1, (v) => $3deee0fd2b46e64f$export$c830d2588e34dae1 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Submenu-headingWrapper", () => $3deee0fd2b46e64f$export$d3fc690f61339a08, (v) => $3deee0fd2b46e64f$export$d3fc690f61339a08 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Submenu-popover", () => $3deee0fd2b46e64f$export$fe967dac6207a986, (v) => $3deee0fd2b46e64f$export$fe967dac6207a986 = v);
$parcel$export($3deee0fd2b46e64f$exports, "spectrum-Submenu-wrapper", () => $3deee0fd2b46e64f$export$158cbd6607d36c4e, (v) => $3deee0fd2b46e64f$export$158cbd6607d36c4e = v);
$parcel$export($3deee0fd2b46e64f$exports, "text", () => $3deee0fd2b46e64f$export$6f093cfa640b7166, (v) => $3deee0fd2b46e64f$export$6f093cfa640b7166 = v);
var $3deee0fd2b46e64f$export$292abbf31ed842a;
var $3deee0fd2b46e64f$export$980c1089c0604ea3;
var $3deee0fd2b46e64f$export$6d59db4903f20f7d;
var $3deee0fd2b46e64f$export$bd5df0f255a350f8;
var $3deee0fd2b46e64f$export$f39a09f249340e2a;
var $3deee0fd2b46e64f$export$8c4ee2c50c22c514;
var $3deee0fd2b46e64f$export$1ca1ec8b29a4ce27;
var $3deee0fd2b46e64f$export$20fd0f7cd4e6112f;
var $3deee0fd2b46e64f$export$d35bc1e505d1ebbf;
var $3deee0fd2b46e64f$export$911ebb89bba4b634;
var $3deee0fd2b46e64f$export$e7dc768d35940237;
var $3deee0fd2b46e64f$export$451c613d0772eb99;
var $3deee0fd2b46e64f$export$b8813cd5d7824ce7;
var $3deee0fd2b46e64f$export$a9781837241c946d;
var $3deee0fd2b46e64f$export$fbd22ba224473193;
var $3deee0fd2b46e64f$export$1e0fb04f31d3c22a;
var $3deee0fd2b46e64f$export$cfdd767f5a513509;
var $3deee0fd2b46e64f$export$4109102f950813a6;
var $3deee0fd2b46e64f$export$24c7f46a6e3605dd;
var $3deee0fd2b46e64f$export$2927016961429360;
var $3deee0fd2b46e64f$export$d374b04f30360026;
var $3deee0fd2b46e64f$export$fae6af6decd50ac5;
var $3deee0fd2b46e64f$export$944daf9b7d423a02;
var $3deee0fd2b46e64f$export$8e15a726e4f7e62d;
var $3deee0fd2b46e64f$export$ca45e0db366b8e6c;
var $3deee0fd2b46e64f$export$36f1275051e39ead;
var $3deee0fd2b46e64f$export$e892026f534efed8;
var $3deee0fd2b46e64f$export$926a1bb74ed65667;
var $3deee0fd2b46e64f$export$cbbae63ef9e44c7a;
var $3deee0fd2b46e64f$export$45faad328e6c532b;
var $3deee0fd2b46e64f$export$ee5f52784f57f9c8;
var $3deee0fd2b46e64f$export$f431001e1bb54633;
var $3deee0fd2b46e64f$export$1aa6419c5c650567;
var $3deee0fd2b46e64f$export$465092164a8e1d98;
var $3deee0fd2b46e64f$export$4f9817a7dff6ba0f;
var $3deee0fd2b46e64f$export$d959f30786125b;
var $3deee0fd2b46e64f$export$dae331c15950f1e;
var $3deee0fd2b46e64f$export$9778864c5b622fbc;
var $3deee0fd2b46e64f$export$8b61ec49b96ad17e;
var $3deee0fd2b46e64f$export$c20ea4b35020fbdd;
var $3deee0fd2b46e64f$export$d7e47bfca92b5744;
var $3deee0fd2b46e64f$export$76d36e55740ad742;
var $3deee0fd2b46e64f$export$c873119be41fdb72;
var $3deee0fd2b46e64f$export$c830d2588e34dae1;
var $3deee0fd2b46e64f$export$d3fc690f61339a08;
var $3deee0fd2b46e64f$export$fe967dac6207a986;
var $3deee0fd2b46e64f$export$158cbd6607d36c4e;
var $3deee0fd2b46e64f$export$6f093cfa640b7166;
$3deee0fd2b46e64f$export$292abbf31ed842a = `Q7FggG_checkmark`;
$3deee0fd2b46e64f$export$980c1089c0604ea3 = `Q7FggG_chevron`;
$3deee0fd2b46e64f$export$6d59db4903f20f7d = `Q7FggG_description`;
$3deee0fd2b46e64f$export$bd5df0f255a350f8 = `Q7FggG_end`;
$3deee0fd2b46e64f$export$f39a09f249340e2a = `Q7FggG_focus-ring`;
$3deee0fd2b46e64f$export$8c4ee2c50c22c514 = `Q7FggG_i18nFontFamily`;
$3deee0fd2b46e64f$export$1ca1ec8b29a4ce27 = `Q7FggG_icon`;
$3deee0fd2b46e64f$export$20fd0f7cd4e6112f = `Q7FggG_is-active`;
$3deee0fd2b46e64f$export$d35bc1e505d1ebbf = `Q7FggG_is-disabled`;
$3deee0fd2b46e64f$export$911ebb89bba4b634 = `Q7FggG_is-expanded`;
$3deee0fd2b46e64f$export$e7dc768d35940237 = `Q7FggG_is-focused`;
$3deee0fd2b46e64f$export$451c613d0772eb99 = `Q7FggG_is-highlighted`;
$3deee0fd2b46e64f$export$b8813cd5d7824ce7 = `Q7FggG_is-hovered`;
$3deee0fd2b46e64f$export$a9781837241c946d = `Q7FggG_is-open`;
$3deee0fd2b46e64f$export$fbd22ba224473193 = `Q7FggG_is-selectable`;
$3deee0fd2b46e64f$export$1e0fb04f31d3c22a = `Q7FggG_is-selected`;
$3deee0fd2b46e64f$export$cfdd767f5a513509 = `Q7FggG_keyboard`;
$3deee0fd2b46e64f$export$4109102f950813a6 = `Q7FggG_spectrum-FocusRing-ring`;
$3deee0fd2b46e64f$export$24c7f46a6e3605dd = `Q7FggG_spectrum-FocusRing ${$3deee0fd2b46e64f$export$4109102f950813a6}`;
$3deee0fd2b46e64f$export$2927016961429360 = `Q7FggG_spectrum-FocusRing--quiet`;
$3deee0fd2b46e64f$export$d374b04f30360026 = `Q7FggG_spectrum-Icon`;
$3deee0fd2b46e64f$export$fae6af6decd50ac5 = `Q7FggG_spectrum-Menu`;
$3deee0fd2b46e64f$export$944daf9b7d423a02 = `Q7FggG_spectrum-Menu-avatar`;
$3deee0fd2b46e64f$export$8e15a726e4f7e62d = `Q7FggG_spectrum-Menu-checkmark`;
$3deee0fd2b46e64f$export$ca45e0db366b8e6c = `Q7FggG_spectrum-Menu-chevron`;
$3deee0fd2b46e64f$export$36f1275051e39ead = `Q7FggG_spectrum-Menu-description`;
$3deee0fd2b46e64f$export$e892026f534efed8 = `Q7FggG_spectrum-Menu-divider`;
$3deee0fd2b46e64f$export$926a1bb74ed65667 = `Q7FggG_spectrum-Menu-end`;
$3deee0fd2b46e64f$export$cbbae63ef9e44c7a = `Q7FggG_spectrum-Menu-icon`;
$3deee0fd2b46e64f$export$45faad328e6c532b = `Q7FggG_spectrum-Menu-item`;
$3deee0fd2b46e64f$export$ee5f52784f57f9c8 = `Q7FggG_spectrum-Menu-itemGrid`;
$3deee0fd2b46e64f$export$f431001e1bb54633 = `Q7FggG_spectrum-Menu-itemIcon`;
$3deee0fd2b46e64f$export$1aa6419c5c650567 = `Q7FggG_spectrum-Menu-itemLabel`;
$3deee0fd2b46e64f$export$465092164a8e1d98 = `Q7FggG_spectrum-Menu-itemLabel--wrapping`;
$3deee0fd2b46e64f$export$4f9817a7dff6ba0f = `Q7FggG_spectrum-Menu-keyboard`;
$3deee0fd2b46e64f$export$d959f30786125b = `Q7FggG_spectrum-Menu-popover`;
$3deee0fd2b46e64f$export$dae331c15950f1e = `Q7FggG_spectrum-Menu-section--isFirst`;
$3deee0fd2b46e64f$export$9778864c5b622fbc = `Q7FggG_spectrum-Menu-section--isLast`;
$3deee0fd2b46e64f$export$8b61ec49b96ad17e = `Q7FggG_spectrum-Menu-section--noHeading`;
$3deee0fd2b46e64f$export$c20ea4b35020fbdd = `Q7FggG_spectrum-Menu-sectionHeading`;
$3deee0fd2b46e64f$export$d7e47bfca92b5744 = `Q7FggG_spectrum-Menu-subdialog`;
$3deee0fd2b46e64f$export$76d36e55740ad742 = `Q7FggG_spectrum-Menu-wrapper`;
$3deee0fd2b46e64f$export$c873119be41fdb72 = `Q7FggG_spectrum-Menu-wrapper--isMobile`;
$3deee0fd2b46e64f$export$c830d2588e34dae1 = `Q7FggG_spectrum-Submenu-heading`;
$3deee0fd2b46e64f$export$d3fc690f61339a08 = `Q7FggG_spectrum-Submenu-headingWrapper`;
$3deee0fd2b46e64f$export$fe967dac6207a986 = `Q7FggG_spectrum-Submenu-popover`;
$3deee0fd2b46e64f$export$158cbd6607d36c4e = `Q7FggG_spectrum-Submenu-wrapper`;
$3deee0fd2b46e64f$export$6f093cfa640b7166 = `Q7FggG_text`;




function $e56311f7204399c7$var$MenuTrigger(props, ref) {
    let triggerRef = (0, $3ThwF$react.useRef)();
    let domRef = (0, $3ThwF$reactspectrumutils.useDOMRef)(ref);
    let menuTriggerRef = domRef || triggerRef;
    let menuRef = (0, $3ThwF$react.useRef)();
    let { children: children, align: align = "start", shouldFlip: shouldFlip = true, direction: direction = "bottom", closeOnSelect: closeOnSelect, trigger: trigger = "press" } = props;
    let [menuTrigger, menu] = (0, ($parcel$interopDefault($3ThwF$react))).Children.toArray(children);
    let state = (0, $3ThwF$reactstatelymenu.useMenuTriggerState)(props);
    let { menuTriggerProps: menuTriggerProps, menuProps: menuProps } = (0, $3ThwF$reactariamenu.useMenuTrigger)({
        trigger: trigger
    }, state, menuTriggerRef);
    let initialPlacement;
    switch(direction){
        case "left":
        case "right":
        case "start":
        case "end":
            initialPlacement = `${direction} ${align === "end" ? "bottom" : "top"}`;
            break;
        case "bottom":
        case "top":
        default:
            initialPlacement = `${direction} ${align}`;
    }
    let isMobile = (0, $3ThwF$reactspectrumutils.useIsMobileDevice)();
    let menuContext = {
        ...menuProps,
        ref: menuRef,
        onClose: state.close,
        closeOnSelect: closeOnSelect,
        autoFocus: state.focusStrategy || true,
        UNSAFE_style: isMobile ? {
            width: "100%",
            maxHeight: "inherit"
        } : undefined,
        UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), {
            "spectrum-Menu-popover": !isMobile
        }),
        state: state
    };
    // On small screen devices, the menu is rendered in a tray, otherwise a popover.
    let overlay;
    if (isMobile) overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumoverlays.Tray), {
        state: state,
        isFixedHeight: true
    }, menu);
    else overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumoverlays.Popover), {
        UNSAFE_style: {
            clipPath: "unset",
            overflow: "visible",
            filter: "unset",
            borderWidth: "0px"
        },
        state: state,
        triggerRef: menuTriggerRef,
        scrollRef: menuRef,
        placement: initialPlacement,
        hideArrow: true,
        shouldFlip: shouldFlip
    }, menu);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$react.Fragment), null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumutils.SlotProvider), {
        slots: {
            actionButton: {
                holdAffordance: trigger === "longPress"
            }
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariainteractions.PressResponder), {
        ...menuTriggerProps,
        ref: menuTriggerRef,
        isPressed: state.isOpen
    }, menuTrigger)), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $cac834c4bc0a51d3$export$c7e742effb1c51e2).Provider, {
        value: menuContext
    }, overlay));
}
/**
 * The MenuTrigger serves as a wrapper around a Menu and its associated trigger,
 * linking the Menu's open state with the trigger's press state.
 */ let $e56311f7204399c7$export$27d2ad3c5815583e = /*#__PURE__*/ (0, $3ThwF$react.forwardRef)($e56311f7204399c7$var$MenuTrigger);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 









function $86ed235c4695ff3f$var$SubmenuTrigger(props) {
    let triggerRef = (0, $3ThwF$react.useRef)();
    let { children: children, targetKey: targetKey } = props;
    let [menuTrigger, menu] = (0, ($parcel$interopDefault($3ThwF$react))).Children.toArray(children);
    let { popoverContainerRef: popoverContainerRef, trayContainerRef: trayContainerRef, menu: parentMenuRef, submenu: menuRef, rootMenuTriggerState: rootMenuTriggerState, state: state } = (0, $cac834c4bc0a51d3$export$efa3856fc0e85e7f)();
    let triggerNode = state.collection.getItem(targetKey);
    let submenuTriggerState = (0, $3ThwF$reactstatelymenu.UNSTABLE_useSubmenuTriggerState)({
        triggerKey: targetKey
    }, rootMenuTriggerState);
    let { submenuTriggerProps: submenuTriggerProps, submenuProps: submenuProps, popoverProps: popoverProps } = (0, $3ThwF$reactariamenu.UNSTABLE_useSubmenuTrigger)({
        node: triggerNode,
        parentMenuRef: parentMenuRef,
        submenuRef: menuRef
    }, submenuTriggerState, triggerRef);
    let isMobile = (0, $3ThwF$reactspectrumutils.useIsMobileDevice)();
    let onBackButtonPress = ()=>{
        submenuTriggerState.close();
        if (parentMenuRef.current && !parentMenuRef.current.contains(document.activeElement)) parentMenuRef.current.focus();
    };
    let { direction: direction } = (0, $3ThwF$reactariai18n.useLocale)();
    let mobileSubmenuKeyDown = (e)=>{
        switch(e.key){
            case "ArrowLeft":
                if (direction === "ltr") triggerRef.current.focus();
                break;
            case "ArrowRight":
                if (direction === "rtl") triggerRef.current.focus();
                break;
        }
    };
    let overlay;
    let [offset, setOffset] = (0, $3ThwF$react.useState)(0);
    (0, $3ThwF$reactariautils.useLayoutEffect)(()=>{
        if (parentMenuRef.current) {
            var _window_getComputedStyle, _window;
            let offset = (_window = window) === null || _window === void 0 ? void 0 : (_window_getComputedStyle = _window.getComputedStyle(parentMenuRef === null || parentMenuRef === void 0 ? void 0 : parentMenuRef.current)) === null || _window_getComputedStyle === void 0 ? void 0 : _window_getComputedStyle.getPropertyValue("--spectrum-submenu-offset-distance");
            if (offset !== "") setOffset(-1 * parseInt(offset, 10));
        }
    }, [
        parentMenuRef
    ]);
    if (isMobile) {
        var _submenuProps;
        delete submenuTriggerProps.onBlur;
        delete submenuTriggerProps.onHoverChange;
        var _autoFocus;
        (_autoFocus = (_submenuProps = submenuProps).autoFocus) !== null && _autoFocus !== void 0 ? _autoFocus : _submenuProps.autoFocus = true;
        if (trayContainerRef.current && submenuTriggerState.isOpen) overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$reactdom))).createPortal(menu, trayContainerRef.current);
    } else {
        let onDismissButtonPress = ()=>{
            var _parentMenuRef_current;
            submenuTriggerState.close();
            (_parentMenuRef_current = parentMenuRef.current) === null || _parentMenuRef_current === void 0 ? void 0 : _parentMenuRef_current.focus();
        };
        overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumoverlays.Popover), {
            ...popoverProps,
            onDismissButtonPress: onDismissButtonPress,
            UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Submenu-popover"),
            container: popoverContainerRef.current,
            containerPadding: 0,
            crossOffset: offset,
            offset: offset,
            enableBothDismissButtons: true,
            UNSAFE_style: {
                clipPath: "unset",
                overflow: "visible",
                borderWidth: "0px"
            },
            state: submenuTriggerState,
            triggerRef: triggerRef,
            scrollRef: menuRef,
            placement: "end top",
            hideArrow: true
        }, menu);
    }
    let menuContext = {
        ...(0, $3ThwF$reactariautils.mergeProps)(submenuProps, {
            ref: menuRef,
            UNSAFE_style: isMobile ? {
                width: "100%",
                maxHeight: "inherit"
            } : undefined,
            UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), {
                "spectrum-Menu-popover": !isMobile
            }),
            ...isMobile && {
                onBackButtonPress: onBackButtonPress,
                onKeyDown: mobileSubmenuKeyDown
            }
        })
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $cac834c4bc0a51d3$export$8d97fe02339fc0e3).Provider, {
        value: {
            triggerRef: triggerRef,
            ...submenuTriggerProps
        }
    }, menuTrigger), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $cac834c4bc0a51d3$export$c7e742effb1c51e2).Provider, {
        value: menuContext
    }, overlay));
}
$86ed235c4695ff3f$var$SubmenuTrigger.getCollectionNode = function*(props) {
    let childArray = [];
    (0, ($parcel$interopDefault($3ThwF$react))).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).isValidElement(child)) childArray.push(child);
    });
    let [trigger] = childArray;
    let [, content] = props.children;
    yield {
        element: /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).cloneElement(trigger, {
            ...trigger.props,
            hasChildItems: true,
            isTrigger: true
        }),
        wrapper: (element)=>/*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement($86ed235c4695ff3f$var$SubmenuTrigger, {
                key: element.key,
                targetKey: element.key,
                ...props
            }, element, content)
    };
};
let $86ed235c4695ff3f$export$ecabc99eeffab7ca = $86ed235c4695ff3f$var$SubmenuTrigger;


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $2f9ceea24fba5443$exports = {};


































$2f9ceea24fba5443$exports = {
    "ar-AE": $517603d8c7b8e5e5$exports,
    "bg-BG": $da4736ecb2b2fc3d$exports,
    "cs-CZ": $474902e64d15239d$exports,
    "da-DK": $ab827474128c0a44$exports,
    "de-DE": $6decd364a8a60c6f$exports,
    "el-GR": $a79ffc3f42b40ea0$exports,
    "en-US": $2b0d2ea968c3f427$exports,
    "es-ES": $3655ccfea2cb795f$exports,
    "et-EE": $fdd5fdd03c65691f$exports,
    "fi-FI": $5b53a5d82dacf660$exports,
    "fr-FR": $53cf7303c3810c0d$exports,
    "he-IL": $1e6d66e0b4792817$exports,
    "hr-HR": $eb204bb933ebd45f$exports,
    "hu-HU": $3b770de73ab860bc$exports,
    "it-IT": $010ff727ed3102ae$exports,
    "ja-JP": $dead61005b56a753$exports,
    "ko-KR": $059a8cc6da15c9c5$exports,
    "lt-LT": $115228ca3cb07b54$exports,
    "lv-LV": $1b70a2b594f0946f$exports,
    "nb-NO": $10afb7333b6cd21a$exports,
    "nl-NL": $6fb3df0b34cc117d$exports,
    "pl-PL": $f6cf802e78a19c98$exports,
    "pt-BR": $7401b34fc83fec2d$exports,
    "pt-PT": $812f012e4938f38d$exports,
    "ro-RO": $8c983e7e2aa14f7a$exports,
    "ru-RU": $5eddd4454bdacc3e$exports,
    "sk-SK": $3dc143fbc7df8c0b$exports,
    "sl-SI": $089b9b9eacba8462$exports,
    "sr-SP": $ae817dc85563e7c9$exports,
    "sv-SE": $39f858f9383858ce$exports,
    "tr-TR": $fb520ca0abee09d8$exports,
    "uk-UA": $7950b4249316e2c0$exports,
    "zh-CN": $86c8e6b13fa9c9ad$exports,
    "zh-TW": $f3cd1d6d75907b41$exports
};



/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 














function $1d718563708db5b5$export$2ce376c2cc3355c8(props) {
    let { item: item, state: state, isVirtualized: isVirtualized, onAction: onAction } = props;
    let { closeOnSelect: closeOnSelect } = (0, $cac834c4bc0a51d3$export$21c7ab35b39f78ec)();
    let { rendered: rendered, key: key } = item;
    let stringFormatter = (0, $3ThwF$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($2f9ceea24fba5443$exports))), "@react-spectrum/menu");
    let { direction: direction } = (0, $3ThwF$reactariai18n.useLocale)();
    let submenuTriggerContext = (0, $cac834c4bc0a51d3$export$dc2161044aa0b36d)();
    let { triggerRef: triggerRef, ...submenuTriggerProps } = submenuTriggerContext || {};
    let isSubmenuTrigger = !!submenuTriggerContext;
    let isUnavailable;
    let ElementType = item.props.href ? "a" : "div";
    if (isSubmenuTrigger) isUnavailable = submenuTriggerContext.isUnavailable;
    let isDisabled = state.disabledKeys.has(key);
    let isSelectable = !isSubmenuTrigger && state.selectionManager.selectionMode !== "none";
    let isSelected = isSelectable && state.selectionManager.isSelected(key);
    let itemref = (0, $3ThwF$react.useRef)(null);
    let ref = (0, $3ThwF$reactariautils.useObjectRef)((0, $3ThwF$react.useMemo)(()=>(0, $3ThwF$reactariautils.mergeRefs)(itemref, triggerRef), [
        itemref,
        triggerRef
    ]));
    let { menuItemProps: menuItemProps, labelProps: labelProps, descriptionProps: descriptionProps, keyboardShortcutProps: keyboardShortcutProps } = (0, $3ThwF$reactariamenu.useMenuItem)({
        isSelected: isSelected,
        isDisabled: isDisabled,
        "aria-label": item["aria-label"],
        key: key,
        closeOnSelect: closeOnSelect,
        isVirtualized: isVirtualized,
        onAction: onAction,
        ...submenuTriggerProps
    }, state, ref);
    let endId = (0, $3ThwF$reactariautils.useSlotId)();
    let endProps = {};
    if (endId) {
        endProps.id = endId;
        menuItemProps["aria-describedby"] = [
            menuItemProps["aria-describedby"],
            endId
        ].filter(Boolean).join(" ");
    }
    let contents = typeof rendered === "string" ? /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumtext.Text), null, rendered) : rendered;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariafocus.FocusRing), {
        focusRingClass: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement(ElementType, {
        ...menuItemProps,
        ref: ref,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-item", {
            "is-disabled": isDisabled,
            "is-selected": isSelected,
            "is-selectable": isSelectable,
            "is-open": submenuTriggerProps.isOpen
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumlayout.Grid), {
        UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-itemGrid")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumutils.ClearSlots), null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumutils.SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-itemLabel"],
                ...labelProps
            },
            end: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-end"],
                ...endProps
            },
            icon: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-icon"],
                size: "S"
            },
            description: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-description"],
                ...descriptionProps
            },
            keyboard: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-keyboard"],
                ...keyboardShortcutProps
            },
            chevron: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports)))["spectrum-Menu-chevron"],
                size: "S"
            }
        }
    }, contents, isSelected && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsuiCheckmarkMedium))), {
        slot: "checkmark",
        UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-checkmark")
    }), isUnavailable && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsworkflowInfoOutline))), {
        slot: "end",
        size: "XS",
        alignSelf: "center",
        "aria-label": stringFormatter.format("unavailable")
    }), isUnavailable == null && isSubmenuTrigger && (direction === "rtl" ? /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsworkflowChevronLeft))), {
        slot: "chevron"
    }) : /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsworkflowChevronRight))), {
        slot: "chevron"
    })))))));
}


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






function $5793ff412227d2fc$export$4b1545b4f2016d26(props) {
    var _filter_at;
    let { item: item, state: state, onAction: onAction } = props;
    let { itemProps: itemProps, headingProps: headingProps, groupProps: groupProps } = (0, $3ThwF$reactariamenu.useMenuSection)({
        heading: item.rendered,
        "aria-label": item["aria-label"]
    });
    let { separatorProps: separatorProps } = (0, $3ThwF$reactariaseparator.useSeparator)({
        elementType: "div"
    });
    let firstSectionKey = state.collection.getFirstKey();
    let lastSectionKey = (_filter_at = [
        ...state.collection
    ].filter((node)=>node.type === "section").at(-1)) === null || _filter_at === void 0 ? void 0 : _filter_at.key;
    let sectionIsFirst = firstSectionKey === item.key && state.collection.getFirstKey() === firstSectionKey;
    let sectionIsLast = lastSectionKey === item.key && state.collection.getItem(state.collection.getLastKey()).parentKey === lastSectionKey;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$react.Fragment), null, item.key !== state.collection.getFirstKey() && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        ...separatorProps,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-divider")
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", itemProps, item.rendered && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("span", {
        ...headingProps,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-sectionHeading")
    }, item.rendered), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        ...groupProps,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu", {
            "spectrum-Menu-section--noHeading": item.rendered == null,
            "spectrum-Menu-section--isFirst": sectionIsFirst,
            "spectrum-Menu-section--isLast": sectionIsLast
        })
    }, [
        ...(0, $3ThwF$reactstatelycollections.getChildNodes)(item, state.collection)
    ].map((node)=>{
        let item = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $1d718563708db5b5$export$2ce376c2cc3355c8), {
            key: node.key,
            item: node,
            state: state,
            onAction: onAction
        });
        if (node.wrapper) item = node.wrapper(item);
        return item;
    }))));
}








function $63a7dff9cbe2d046$var$Menu(props, ref) {
    let isSubmenu = true;
    let contextProps = (0, $3ThwF$react.useContext)((0, $cac834c4bc0a51d3$export$c7e742effb1c51e2));
    let parentMenuContext = (0, $cac834c4bc0a51d3$export$efa3856fc0e85e7f)();
    let { rootMenuTriggerState: rootMenuTriggerState, state: parentMenuTreeState } = parentMenuContext || {
        rootMenuTriggerState: contextProps.state
    };
    if (!parentMenuContext) isSubmenu = false;
    let completeProps = {
        ...(0, $3ThwF$reactariautils.mergeProps)(contextProps, props)
    };
    let domRef = (0, $3ThwF$reactspectrumutils.useDOMRef)(ref);
    let popoverContainerRef = (0, $3ThwF$react.useRef)(null);
    let trayContainerRef = (0, $3ThwF$react.useRef)(null);
    let state = (0, $3ThwF$reactstatelytree.useTreeState)(completeProps);
    let submenuRef = (0, $3ThwF$react.useRef)(null);
    let { menuProps: menuProps } = (0, $3ThwF$reactariamenu.useMenu)(completeProps, state, domRef);
    let { styleProps: styleProps } = (0, $3ThwF$reactspectrumutils.useStyleProps)(completeProps);
    (0, $3ThwF$reactariautils.useSyncRef)(contextProps, domRef);
    let [leftOffset, setLeftOffset] = (0, $3ThwF$react.useState)({
        left: 0
    });
    (0, $3ThwF$react.useEffect)(()=>{
        if (popoverContainerRef.current) {
            let { left: left } = popoverContainerRef.current.getBoundingClientRect();
            setLeftOffset({
                left: -1 * left
            });
        }
    }, []);
    var _contextProps_submenuLevel;
    let menuLevel = (_contextProps_submenuLevel = contextProps.submenuLevel) !== null && _contextProps_submenuLevel !== void 0 ? _contextProps_submenuLevel : -1;
    let hasOpenSubmenu = state.collection.getItem(rootMenuTriggerState === null || rootMenuTriggerState === void 0 ? void 0 : rootMenuTriggerState.UNSTABLE_expandedKeysStack[menuLevel + 1]) != null;
    // TODO: add slide transition
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $cac834c4bc0a51d3$export$24aad8519b95b41b).Provider, {
        value: {
            popoverContainerRef: popoverContainerRef,
            trayContainerRef: trayContainerRef,
            menu: domRef,
            submenu: submenuRef,
            rootMenuTriggerState: rootMenuTriggerState,
            state: state
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        style: {
            height: hasOpenSubmenu ? "100%" : undefined
        },
        ref: trayContainerRef
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariafocus.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement($63a7dff9cbe2d046$export$3dfe97b5c32d8d8c, {
        onBackButtonPress: contextProps.onBackButtonPress,
        hasOpenSubmenu: hasOpenSubmenu,
        isSubmenu: isSubmenu,
        parentMenuTreeState: parentMenuTreeState,
        rootMenuTriggerState: rootMenuTriggerState
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        ...menuProps,
        style: (0, $3ThwF$reactariautils.mergeProps)(styleProps.style, menuProps.style),
        ref: domRef,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu", styleProps.className)
    }, [
        ...state.collection
    ].map((item)=>{
        if (item.type === "section") return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $5793ff412227d2fc$export$4b1545b4f2016d26), {
            key: item.key,
            item: item,
            state: state,
            onAction: completeProps.onAction
        });
        let menuItem = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $1d718563708db5b5$export$2ce376c2cc3355c8), {
            key: item.key,
            item: item,
            state: state,
            onAction: completeProps.onAction
        });
        if (item.wrapper) menuItem = item.wrapper(menuItem);
        return menuItem;
    }))), (rootMenuTriggerState === null || rootMenuTriggerState === void 0 ? void 0 : rootMenuTriggerState.isOpen) && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        ref: popoverContainerRef,
        style: {
            width: "100vw",
            position: "absolute",
            top: -5,
            ...leftOffset
        }
    })));
}
function $63a7dff9cbe2d046$export$3dfe97b5c32d8d8c(props) {
    var _parentMenuTreeState_collection_getItem;
    let { children: children, isSubmenu: isSubmenu, hasOpenSubmenu: hasOpenSubmenu, parentMenuTreeState: parentMenuTreeState, rootMenuTriggerState: rootMenuTriggerState, onBackButtonPress: onBackButtonPress, wrapperKeyDown: wrapperKeyDown } = props;
    let stringFormatter = (0, $3ThwF$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($2f9ceea24fba5443$exports))), "@react-spectrum/menu");
    let backButtonText = parentMenuTreeState === null || parentMenuTreeState === void 0 ? void 0 : (_parentMenuTreeState_collection_getItem = parentMenuTreeState.collection.getItem(rootMenuTriggerState === null || rootMenuTriggerState === void 0 ? void 0 : rootMenuTriggerState.UNSTABLE_expandedKeysStack.slice(-1)[0])) === null || _parentMenuTreeState_collection_getItem === void 0 ? void 0 : _parentMenuTreeState_collection_getItem.textValue;
    let backButtonLabel = stringFormatter.format("backButton", {
        prevMenuButton: backButtonText
    });
    let headingId = (0, $3ThwF$reactariautils.useSlotId)();
    let isMobile = (0, $3ThwF$reactspectrumutils.useIsMobileDevice)();
    let { direction: direction } = (0, $3ThwF$reactariai18n.useLocale)();
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        role: headingId ? "dialog" : undefined,
        "aria-labelledby": headingId,
        "aria-hidden": isMobile && hasOpenSubmenu,
        "data-testid": "menu-wrapper",
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Menu-wrapper", {
            "spectrum-Menu-wrapper--isMobile": isMobile,
            "is-expanded": hasOpenSubmenu
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        role: "presentation",
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Submenu-wrapper"),
        onKeyDown: wrapperKeyDown
    }, isMobile && isSubmenu && !hasOpenSubmenu && /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("div", {
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Submenu-headingWrapper")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumbutton.ActionButton), {
        "aria-label": backButtonLabel,
        isQuiet: true,
        onPress: onBackButtonPress
    }, direction === "rtl" ? /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsuiArrowDownSmall))), {
        UNSAFE_style: {
            rotate: "270deg"
        }
    }) : /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsuiArrowDownSmall))), {
        UNSAFE_style: {
            rotate: "90deg"
        }
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement("h1", {
        id: headingId,
        className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Submenu-heading")
    }, backButtonText)), children)));
}
/**
 * Menus display a list of actions or options that a user can choose.
 */ // forwardRef doesn't support generic parameters, so cast the result to the correct type
// https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref
const $63a7dff9cbe2d046$export$d9b273488cd8ce6f = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).forwardRef($63a7dff9cbe2d046$var$Menu);


/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 








function $53a2f2636c97a561$var$ActionMenu(props, ref) {
    props = (0, $3ThwF$reactspectrumutils.useSlotProps)(props, "actionMenu");
    let stringFormatter = (0, $3ThwF$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($2f9ceea24fba5443$exports))), "@react-spectrum/menu");
    let buttonProps = (0, $3ThwF$reactariautils.filterDOMProps)(props, {
        labelable: true
    });
    if (buttonProps["aria-label"] === undefined) buttonProps["aria-label"] = stringFormatter.format("moreActions");
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $e56311f7204399c7$export$27d2ad3c5815583e), {
        isOpen: props.isOpen,
        defaultOpen: props.defaultOpen,
        onOpenChange: props.onOpenChange,
        align: props.align,
        direction: props.direction,
        shouldFlip: props.shouldFlip
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumbutton.ActionButton), {
        ref: ref,
        ...props,
        ...buttonProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$spectrumiconsworkflowMore))), null)), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $63a7dff9cbe2d046$export$d9b273488cd8ce6f), {
        children: props.children,
        items: props.items,
        disabledKeys: props.disabledKeys,
        onAction: props.onAction
    }));
}
/**
 * ActionMenu combines an ActionButton with a Menu for simple "more actions" use cases.
 */ const $53a2f2636c97a561$export$ed57a210b9d13bb6 = /*#__PURE__*/ (0, $3ThwF$react.forwardRef)($53a2f2636c97a561$var$ActionMenu);


/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $a10e565ec42bcf96$exports = {};

$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-button", () => $a10e565ec42bcf96$export$20d9583c9f1529c1, (v) => $a10e565ec42bcf96$export$20d9583c9f1529c1 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-content", () => $a10e565ec42bcf96$export$a1f805890cf58c4, (v) => $a10e565ec42bcf96$export$a1f805890cf58c4 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-dialog", () => $a10e565ec42bcf96$export$ab2d873674d20d28, (v) => $a10e565ec42bcf96$export$ab2d873674d20d28 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-dialog--isMobile", () => $a10e565ec42bcf96$export$a438776606a9a3cc, (v) => $a10e565ec42bcf96$export$a438776606a9a3cc = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-footer", () => $a10e565ec42bcf96$export$d901bc3fa82b87a6, (v) => $a10e565ec42bcf96$export$d901bc3fa82b87a6 = v);
var $a10e565ec42bcf96$export$20d9583c9f1529c1;
var $a10e565ec42bcf96$export$a1f805890cf58c4;
var $a10e565ec42bcf96$export$ab2d873674d20d28;
var $a10e565ec42bcf96$export$a438776606a9a3cc;
var $a10e565ec42bcf96$export$d901bc3fa82b87a6;
$a10e565ec42bcf96$export$20d9583c9f1529c1 = `L6gUaa_react-spectrum-ContextualHelp-button`;
$a10e565ec42bcf96$export$a1f805890cf58c4 = `L6gUaa_react-spectrum-ContextualHelp-content`;
$a10e565ec42bcf96$export$ab2d873674d20d28 = `L6gUaa_react-spectrum-ContextualHelp-dialog`;
$a10e565ec42bcf96$export$a438776606a9a3cc = `L6gUaa_react-spectrum-ContextualHelp-dialog--isMobile`;
$a10e565ec42bcf96$export$d901bc3fa82b87a6 = `L6gUaa_react-spectrum-ContextualHelp-footer`;











function $db380858d398b658$var$ContextualHelpTrigger(props) {
    let { isUnavailable: isUnavailable = false, targetKey: targetKey } = props;
    let triggerRef = (0, $3ThwF$react.useRef)(null);
    let popoverRef = (0, $3ThwF$react.useRef)(null);
    let { popoverContainerRef: popoverContainerRef, trayContainerRef: trayContainerRef, rootMenuTriggerState: rootMenuTriggerState, menu: parentMenuRef, state: state } = (0, $cac834c4bc0a51d3$export$efa3856fc0e85e7f)();
    let triggerNode = state.collection.getItem(targetKey);
    let submenuTriggerState = (0, $3ThwF$reactstatelymenu.UNSTABLE_useSubmenuTriggerState)({
        triggerKey: targetKey
    }, {
        ...rootMenuTriggerState,
        ...state
    });
    let submenuRef = (0, $3ThwF$reactspectrumutils.unwrapDOMRef)(popoverRef);
    let { submenuTriggerProps: submenuTriggerProps, popoverProps: popoverProps } = (0, $3ThwF$reactariamenu.UNSTABLE_useSubmenuTrigger)({
        node: triggerNode,
        parentMenuRef: parentMenuRef,
        submenuRef: submenuRef,
        type: "dialog",
        isDisabled: !isUnavailable
    }, submenuTriggerState, triggerRef);
    let isMobile = (0, $3ThwF$reactspectrumutils.useIsMobileDevice)();
    let slots = {};
    if (isUnavailable) slots = {
        dialog: {
            UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports))), "react-spectrum-ContextualHelp-dialog", {
                "react-spectrum-ContextualHelp-dialog--isMobile": isMobile
            }, (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), {
                "spectrum-Menu-subdialog": !isMobile
            }))
        },
        content: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports)))["react-spectrum-ContextualHelp-content"]
        },
        footer: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports)))["react-spectrum-ContextualHelp-footer"]
        }
    };
    let [trigger] = (0, ($parcel$interopDefault($3ThwF$react))).Children.toArray(props.children);
    let [, content] = props.children;
    let onBlurWithin = (e)=>{
        var _popoverRef_current_UNSAFE_getDOMNode, _popoverRef_current;
        if (e.relatedTarget && popoverRef.current && !(popoverRef === null || popoverRef === void 0 ? void 0 : (_popoverRef_current = popoverRef.current) === null || _popoverRef_current === void 0 ? void 0 : (_popoverRef_current_UNSAFE_getDOMNode = _popoverRef_current.UNSAFE_getDOMNode()) === null || _popoverRef_current_UNSAFE_getDOMNode === void 0 ? void 0 : _popoverRef_current_UNSAFE_getDOMNode.contains(e.relatedTarget)) && !(e.relatedTarget === triggerRef.current && (0, $3ThwF$reactariainteractions.getInteractionModality)() === "pointer")) {
            if (submenuTriggerState.isOpen) submenuTriggerState.close();
        }
    };
    let overlay;
    let tray;
    let onBackButtonPress = ()=>{
        submenuTriggerState.close();
        if (parentMenuRef.current && !parentMenuRef.current.contains(document.activeElement)) parentMenuRef.current.focus();
    };
    let [offset, setOffset] = (0, $3ThwF$react.useState)(0);
    (0, $3ThwF$reactariautils.useLayoutEffect)(()=>{
        if (parentMenuRef.current) {
            var _window_getComputedStyle, _window;
            let offset = (_window = window) === null || _window === void 0 ? void 0 : (_window_getComputedStyle = _window.getComputedStyle(parentMenuRef === null || parentMenuRef === void 0 ? void 0 : parentMenuRef.current)) === null || _window_getComputedStyle === void 0 ? void 0 : _window_getComputedStyle.getPropertyValue("--spectrum-submenu-offset-distance");
            if (offset !== "") setOffset(-1 * parseInt(offset, 10));
        }
    }, [
        parentMenuRef
    ]);
    if (isMobile) {
        delete submenuTriggerProps.onBlur;
        delete submenuTriggerProps.onHoverChange;
        if (trayContainerRef.current && submenuTriggerState.isOpen) {
            let subDialogKeyDown = (e)=>{
                switch(e.key){
                    case "Escape":
                        e.stopPropagation();
                        onBackButtonPress();
                        break;
                }
            };
            tray = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $63a7dff9cbe2d046$export$3dfe97b5c32d8d8c), {
                isSubmenu: true,
                parentMenuTreeState: state,
                rootMenuTriggerState: rootMenuTriggerState,
                wrapperKeyDown: subDialogKeyDown,
                onBackButtonPress: onBackButtonPress
            }, content);
            overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$reactdom))).createPortal(tray, trayContainerRef.current);
        }
    } else {
        let onDismissButtonPress = ()=>{
            var _parentMenuRef_current;
            submenuTriggerState.close();
            (_parentMenuRef_current = parentMenuRef.current) === null || _parentMenuRef_current === void 0 ? void 0 : _parentMenuRef_current.focus();
        };
        overlay = /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumoverlays.Popover), {
            ...popoverProps,
            UNSAFE_style: {
                clipPath: "unset",
                overflow: "visible",
                filter: "unset",
                borderWidth: "0px"
            },
            UNSAFE_className: (0, $3ThwF$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3deee0fd2b46e64f$exports))), "spectrum-Submenu-popover"),
            onDismissButtonPress: onDismissButtonPress,
            onBlurWithin: onBlurWithin,
            container: popoverContainerRef.current,
            state: submenuTriggerState,
            ref: popoverRef,
            triggerRef: triggerRef,
            placement: "end top",
            containerPadding: 0,
            crossOffset: offset,
            offset: offset,
            hideArrow: true,
            enableBothDismissButtons: true
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactariafocus.FocusScope), {
            restoreFocus: true
        }, content));
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, ($parcel$interopDefault($3ThwF$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $cac834c4bc0a51d3$export$8d97fe02339fc0e3).Provider, {
        value: {
            isUnavailable: isUnavailable,
            triggerRef: triggerRef,
            ...submenuTriggerProps
        }
    }, trigger), /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement((0, $3ThwF$reactspectrumutils.SlotProvider), {
        slots: slots
    }, submenuTriggerState.isOpen && overlay));
}
$db380858d398b658$var$ContextualHelpTrigger.getCollectionNode = function* getCollectionNode(props) {
    let childArray = [];
    (0, ($parcel$interopDefault($3ThwF$react))).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).isValidElement(child)) childArray.push(child);
    });
    let [trigger] = childArray;
    let [, content] = props.children;
    yield {
        element: /*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).cloneElement(trigger, {
            ...trigger.props,
            hasChildItems: true,
            isTrigger: true
        }),
        wrapper: (element)=>/*#__PURE__*/ (0, ($parcel$interopDefault($3ThwF$react))).createElement($db380858d398b658$var$ContextualHelpTrigger, {
                key: element.key,
                targetKey: element.key,
                ...props
            }, element, content)
    };
};
let $db380858d398b658$export$5413b169fff83e61 = $db380858d398b658$var$ContextualHelpTrigger;
