// @adobe/exc-app/component.js
"use strict";
/*************************************************************************
 * Copyright 2022 Adobe
 * All Rights Reserved.
 *
 * NOTICE: Adobe permits you to use, modify, and distribute this file in
 * accordance with the terms of the Adobe license agreement accompanying
 * it. If you have received this file from a source other than Adobe,
 * then your use, modification, or distribution of it requires the prior
 * written permission of Adobe.
 **************************************************************************/
Object.defineProperty(exports, "__esModule", { value: true });
exports.COMPONENT = void 0;
/**
 * APIs that let solutions manage the opening and closing of the component frame modal.
 *
 * ***Import:***
 *
 * ```typescript
 * import component from '@adobe/exc-app/component';
 * ```
 *
 * ***Default export:***
 *
 * [ComponentApi](../interfaces/component.componentapi.md#interface-componentapi)
 *
 * ***Usage:***
 *
 * ```typescript
 * import component from '@adobe/exc-app/component';
 *
 * component.open(COMPONENT.EMAIL_DESIGNER).then((data: Record<string, any>) => {
 *   // Handle the data
 * });
 * component.close({some: 'data'});
 * ```
 *
 * Can optionally open the component to a non-default route by including
 * `componentPath` with the bootstrap data argument in the open function. All
 * data passed in this argument will be provided to the component in the READY
 * event in the `componentBootstrapData` property.
 *
 * ```typescript
 * const bootstrapData = {
 *   componentPath: '/some/path/in/component',
 *   propertyForComponent: 'value'
 * };
 * component.open(COMPONENT.EMAIL_DESIGNER, bootstrapData).then((data: Record<string, any>) => {
 *   // Handle the data
 * });
 * ```
 * @packageDocumentation
 * @module component
 */
const Global_1 = require("./src/Global");
var COMPONENT;
(function (COMPONENT) {
    COMPONENT["EMAIL_DESIGNER"] = "cjm-email-designer";
    COMPONENT["ACC_EMAIL_DESIGNER"] = "creative-designer";
})(COMPONENT = exports.COMPONENT || (exports.COMPONENT = {}));
const component = {
    close: (data) => {
        return (0, Global_1.getImpl)('component')().close(data);
    },
    isComponent: () => {
        return (0, Global_1.getImpl)('component')().isComponent();
    },
    open: (comp, data) => {
        return (0, Global_1.getImpl)('component')().open(comp, data);
    }
};
exports.default = component;
