// @adobe/exc-app/page.js
"use strict";
/*************************************************************************
 * Copyright 2020 Adobe
 * All Rights Reserved.
 *
 * NOTICE: Adobe permits you to use, modify, and distribute this file in
 * accordance with the terms of the Adobe license agreement accompanying
 * it. If you have received this file from a source other than Adobe,
 * then your use, modification, or distribution of it requires the prior
 * written permission of Adobe.
 **************************************************************************/
Object.defineProperty(exports, "__esModule", { value: true });
exports.THUNDERBIRD = exports.SRC_DOC = exports.RELEASE_TYPE = exports.ObservableType = void 0;
/**
 * APIs that let solutions interact with the main page and personalize it, e.g. setting the title,
 * favicon, refreshing the solution iframe, etc.
 *
 * ***Import:***
 *
 * ```typescript
 * import page from '@adobe/exc-app/page';
 * ```
 *
 * ***Default export:***
 *
 * [PageApi](../interfaces/page.pageapi.md#interface-pageapi)
 *
 * ***Usage:***
 *
 * ```typescript
 * import page from '@adobe/exc-app/page';
 *
 * page.title = 'Experience Cloud';
 *
 * // Show spinner while performing an async operation
 * page.spinner = true;
 * try {
 *   await performOperation();
 * } finally {
 *   page.spinner = false;
 * }
 *
 * // Generate a shell URL that directly opens the specified solution URL
 * const shellUrl = page.generateShellUrl('/relative/path');
 *
 * // Navigate to another solution
 * page.shellRedirect('/target');
 * ```
 * @packageDocumentation
 * @module page
 */
const Global_1 = require("./src/Global");
var ObservableType;
(function (ObservableType) {
    ObservableType["MODAL"] = "MODAL";
    ObservableType["POPOVER"] = "POPOVER";
})(ObservableType = exports.ObservableType || (exports.ObservableType = {}));
/**
 * Solution's release type in the product lifecycle
 */
var RELEASE_TYPE;
(function (RELEASE_TYPE) {
    RELEASE_TYPE["POC"] = "poc";
    RELEASE_TYPE["DEV"] = "dev";
    RELEASE_TYPE["ALPHA"] = "alpha";
    RELEASE_TYPE["BETA"] = "beta";
    RELEASE_TYPE["GA"] = "ga";
})(RELEASE_TYPE = exports.RELEASE_TYPE || (exports.RELEASE_TYPE = {}));
/**
 * @deprecated Solution's list of multiple subdomains
 */
var SRC_DOC;
(function (SRC_DOC) {
    SRC_DOC[SRC_DOC["DISABLED"] = 0] = "DISABLED";
    SRC_DOC[SRC_DOC["HTML"] = 1] = "HTML";
    SRC_DOC[SRC_DOC["MANIFEST"] = 2] = "MANIFEST"; // Manifest mode will be added as phase 2, not yet supported.
})(SRC_DOC = exports.SRC_DOC || (exports.SRC_DOC = {}));
var THUNDERBIRD;
(function (THUNDERBIRD) {
    THUNDERBIRD[THUNDERBIRD["OFF"] = 0] = "OFF";
    THUNDERBIRD[THUNDERBIRD["SRC_DOC"] = 1] = "SRC_DOC";
    THUNDERBIRD[THUNDERBIRD["SERVICE_WORKER"] = 2] = "SERVICE_WORKER";
})(THUNDERBIRD = exports.THUNDERBIRD || (exports.THUNDERBIRD = {}));
const page = (0, Global_1.connect)('page', [
    ['afterPrintHandler'],
    ['appContainer'],
    ['beforePrintHandler'],
    ['blockNavigation', true],
    ['clipboardWrite', true],
    ['done', true],
    ['generateShellUrl', true],
    ['getModalQuerySelectors'],
    ['getObservableQuerySelectors'],
    ['favicon'],
    ['iframeReload', true],
    ['modal'],
    ['modalAutoDetect'],
    ['notFound', true],
    ['openInNewTab', true],
    ['preventDefaultCombos'],
    ['print', true],
    ['shellRedirect', true],
    ['setModalQuerySelectors'],
    ['setObservableQuerySelectors'],
    ['spinner'],
    ['title'],
    ['toggleAutoDetect'],
    ['unloadPromptMessage']
]);
exports.default = page;
