// @react-spectrum/list/dist/main.js
"use client";

require("./main.css");
var $1beeff0a5a0ccb2a$exports = require("./ar-AE.main.js");
var $8d53c5c3a365f51f$exports = require("./bg-BG.main.js");
var $99a0008efa578202$exports = require("./cs-CZ.main.js");
var $fed6a92f3a9a90c2$exports = require("./da-DK.main.js");
var $ad6943b7faf38b41$exports = require("./de-DE.main.js");
var $ed6da119df8f733b$exports = require("./el-GR.main.js");
var $a922ddd791b2701d$exports = require("./en-US.main.js");
var $1616ac893d9df40d$exports = require("./es-ES.main.js");
var $71139980c308673a$exports = require("./et-EE.main.js");
var $df400f4d8990deb0$exports = require("./fi-FI.main.js");
var $4fedac3a4a024339$exports = require("./fr-FR.main.js");
var $0c1298be372fa672$exports = require("./he-IL.main.js");
var $aac34baf9a6ccbce$exports = require("./hr-HR.main.js");
var $6d93c8a72fbf4a90$exports = require("./hu-HU.main.js");
var $69346d8975e9f344$exports = require("./it-IT.main.js");
var $b263dd2c132d2852$exports = require("./ja-JP.main.js");
var $b79146096cdb30c7$exports = require("./ko-KR.main.js");
var $adc50e375b062a9e$exports = require("./lt-LT.main.js");
var $20ba6eae003d4c10$exports = require("./lv-LV.main.js");
var $6a03ad2ccd528f82$exports = require("./nb-NO.main.js");
var $6bdc84276c605774$exports = require("./nl-NL.main.js");
var $41e9b40a686907b7$exports = require("./pl-PL.main.js");
var $332f285483e392b9$exports = require("./pt-BR.main.js");
var $21a7a272ae3b9fd4$exports = require("./pt-PT.main.js");
var $7f773822d0ecfb7d$exports = require("./ro-RO.main.js");
var $7d8f25b69820e6ce$exports = require("./ru-RU.main.js");
var $ef1cbac310feac6e$exports = require("./sk-SK.main.js");
var $a37847ff0890b3f8$exports = require("./sl-SI.main.js");
var $f95d6d9be7c5418d$exports = require("./sr-SP.main.js");
var $eb2bbd94105316d8$exports = require("./sv-SE.main.js");
var $afc69bf18d2f3b1a$exports = require("./tr-TR.main.js");
var $eda3e7326f5e4dcb$exports = require("./uk-UA.main.js");
var $6d455ae2db499a05$exports = require("./zh-CN.main.js");
var $0fa2d17f8a898b48$exports = require("./zh-TW.main.js");
var $k9Du2$reactstatelycollections = require("@react-stately/collections");
var $k9Du2$reactariagridlist = require("@react-aria/gridlist");
var $k9Du2$reactspectrumutils = require("@react-spectrum/utils");
var $k9Du2$reactariautils = require("@react-aria/utils");
var $k9Du2$reactariafocus = require("@react-aria/focus");
var $k9Du2$reactstatelylayout = require("@react-stately/layout");
var $k9Du2$reactstatelylist = require("@react-stately/list");
var $k9Du2$reactspectrumprogress = require("@react-spectrum/progress");
var $k9Du2$react = require("react");
var $k9Du2$reactariai18n = require("@react-aria/i18n");
var $k9Du2$reactspectrumprovider = require("@react-spectrum/provider");
var $k9Du2$reactariavirtualizer = require("@react-aria/virtualizer");
var $k9Du2$reactariavisuallyhidden = require("@react-aria/visually-hidden");
var $k9Du2$reactspectrumcheckbox = require("@react-spectrum/checkbox");
var $k9Du2$spectrumiconsuiChevronLeftMedium = require("@spectrum-icons/ui/ChevronLeftMedium");
var $k9Du2$spectrumiconsuiChevronRightMedium = require("@spectrum-icons/ui/ChevronRightMedium");
var $k9Du2$reacttransitiongroup = require("react-transition-group");
var $k9Du2$reactspectrumlayout = require("@react-spectrum/layout");
var $k9Du2$reactariainteractions = require("@react-aria/interactions");
var $k9Du2$spectrumiconsuiListGripper = require("@spectrum-icons/ui/ListGripper");
var $k9Du2$reactspectrumtext = require("@react-spectrum/text");
var $k9Du2$reactariabutton = require("@react-aria/button");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "ListView", () => $60684b441be8e84c$export$84d0dd190d551cd1);
$parcel$export(module.exports, "Item", () => $k9Du2$reactstatelycollections.Item);
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $279507a6f189e509$exports = {};

$parcel$export($279507a6f189e509$exports, "actionmenu", () => $279507a6f189e509$export$bd7f6383fb9d6b48, (v) => $279507a6f189e509$export$bd7f6383fb9d6b48 = v);
$parcel$export($279507a6f189e509$exports, "actions", () => $279507a6f189e509$export$e324594224ef24da, (v) => $279507a6f189e509$export$e324594224ef24da = v);
$parcel$export($279507a6f189e509$exports, "badge", () => $279507a6f189e509$export$4aebd143f31d0c0d, (v) => $279507a6f189e509$export$4aebd143f31d0c0d = v);
$parcel$export($279507a6f189e509$exports, "checkbox", () => $279507a6f189e509$export$b07e517d22efa1ca, (v) => $279507a6f189e509$export$b07e517d22efa1ca = v);
$parcel$export($279507a6f189e509$exports, "chevron", () => $279507a6f189e509$export$980c1089c0604ea3, (v) => $279507a6f189e509$export$980c1089c0604ea3 = v);
$parcel$export($279507a6f189e509$exports, "content", () => $279507a6f189e509$export$a7db06668cad9adb, (v) => $279507a6f189e509$export$a7db06668cad9adb = v);
$parcel$export($279507a6f189e509$exports, "description", () => $279507a6f189e509$export$6d59db4903f20f7d, (v) => $279507a6f189e509$export$6d59db4903f20f7d = v);
$parcel$export($279507a6f189e509$exports, "draghandle", () => $279507a6f189e509$export$7b34ab0dc3a6dc3b, (v) => $279507a6f189e509$export$7b34ab0dc3a6dc3b = v);
$parcel$export($279507a6f189e509$exports, "focus-ring", () => $279507a6f189e509$export$f39a09f249340e2a, (v) => $279507a6f189e509$export$f39a09f249340e2a = v);
$parcel$export($279507a6f189e509$exports, "has-checkbox", () => $279507a6f189e509$export$27f4d14f7fc5b94, (v) => $279507a6f189e509$export$27f4d14f7fc5b94 = v);
$parcel$export($279507a6f189e509$exports, "is-active", () => $279507a6f189e509$export$20fd0f7cd4e6112f, (v) => $279507a6f189e509$export$20fd0f7cd4e6112f = v);
$parcel$export($279507a6f189e509$exports, "is-disabled", () => $279507a6f189e509$export$d35bc1e505d1ebbf, (v) => $279507a6f189e509$export$d35bc1e505d1ebbf = v);
$parcel$export($279507a6f189e509$exports, "is-focused", () => $279507a6f189e509$export$e7dc768d35940237, (v) => $279507a6f189e509$export$e7dc768d35940237 = v);
$parcel$export($279507a6f189e509$exports, "is-hovered", () => $279507a6f189e509$export$b8813cd5d7824ce7, (v) => $279507a6f189e509$export$b8813cd5d7824ce7 = v);
$parcel$export($279507a6f189e509$exports, "is-next-selected", () => $279507a6f189e509$export$53aa811f86efbfad, (v) => $279507a6f189e509$export$53aa811f86efbfad = v);
$parcel$export($279507a6f189e509$exports, "is-selected", () => $279507a6f189e509$export$1e0fb04f31d3c22a, (v) => $279507a6f189e509$export$1e0fb04f31d3c22a = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView", () => $279507a6f189e509$export$4ad4709e9aee8e02, (v) => $279507a6f189e509$export$4ad4709e9aee8e02 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--compact", () => $279507a6f189e509$export$1f55b27adf96a61e, (v) => $279507a6f189e509$export$1f55b27adf96a61e = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--draggable", () => $279507a6f189e509$export$a2d35348c0cc9bca, (v) => $279507a6f189e509$export$a2d35348c0cc9bca = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--dropTarget", () => $279507a6f189e509$export$d9ba3f873d0eaec6, (v) => $279507a6f189e509$export$d9ba3f873d0eaec6 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--emphasized", () => $279507a6f189e509$export$eb0d5352b2a55473, (v) => $279507a6f189e509$export$eb0d5352b2a55473 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--hasAnyChildren", () => $279507a6f189e509$export$7150e72e440612bf, (v) => $279507a6f189e509$export$7150e72e440612bf = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--hasDescription", () => $279507a6f189e509$export$7c9efe74c4b77579, (v) => $279507a6f189e509$export$7c9efe74c4b77579 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--isHorizontalScrollbarVisible", () => $279507a6f189e509$export$d9b13621bc5dd4d1, (v) => $279507a6f189e509$export$d9b13621bc5dd4d1 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--isVerticalScrollbarVisible", () => $279507a6f189e509$export$1bf50375883926, (v) => $279507a6f189e509$export$1bf50375883926 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--loadingMore", () => $279507a6f189e509$export$5aa06adeaa92dbca, (v) => $279507a6f189e509$export$5aa06adeaa92dbca = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--quiet", () => $279507a6f189e509$export$b01cb9680b0b8d22, (v) => $279507a6f189e509$export$b01cb9680b0b8d22 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--spacious", () => $279507a6f189e509$export$cbd30c7809acdd81, (v) => $279507a6f189e509$export$cbd30c7809acdd81 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView--wrap", () => $279507a6f189e509$export$68b45d42c77fe1cf, (v) => $279507a6f189e509$export$68b45d42c77fe1cf = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView-centeredWrapper", () => $279507a6f189e509$export$1b66ee5cd8394caf, (v) => $279507a6f189e509$export$1b66ee5cd8394caf = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView-centeredWrapper--loadingMore", () => $279507a6f189e509$export$6b3b7792abd9e5f4, (v) => $279507a6f189e509$export$6b3b7792abd9e5f4 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListView-row", () => $279507a6f189e509$export$f80410436c6352c8, (v) => $279507a6f189e509$export$f80410436c6352c8 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewInsertionIndicator", () => $279507a6f189e509$export$2a6a235e0baff403, (v) => $279507a6f189e509$export$2a6a235e0baff403 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewInsertionIndicator--dropTarget", () => $279507a6f189e509$export$927a243805a411ca, (v) => $279507a6f189e509$export$927a243805a411ca = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem", () => $279507a6f189e509$export$ce8c93b50eec2d4f, (v) => $279507a6f189e509$export$ce8c93b50eec2d4f = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem--dropTarget", () => $279507a6f189e509$export$8afa006321ee3a8f, (v) => $279507a6f189e509$export$8afa006321ee3a8f = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem--firstRow", () => $279507a6f189e509$export$60162f05499b7560, (v) => $279507a6f189e509$export$60162f05499b7560 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem--isFlushBottom", () => $279507a6f189e509$export$f9ed481cac4d935a, (v) => $279507a6f189e509$export$f9ed481cac4d935a = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem--lastRow", () => $279507a6f189e509$export$659c081622dc13b0, (v) => $279507a6f189e509$export$659c081622dc13b0 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-actionmenu", () => $279507a6f189e509$export$58dfda3903289699, (v) => $279507a6f189e509$export$58dfda3903289699 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-actions", () => $279507a6f189e509$export$26480389595d4515, (v) => $279507a6f189e509$export$26480389595d4515 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-badge", () => $279507a6f189e509$export$6498e436231eea33, (v) => $279507a6f189e509$export$6498e436231eea33 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox", () => $279507a6f189e509$export$f979f76ca45242f2, (v) => $279507a6f189e509$export$f979f76ca45242f2 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox--enter", () => $279507a6f189e509$export$9dc11c0e0825a249, (v) => $279507a6f189e509$export$9dc11c0e0825a249 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox--enterActive", () => $279507a6f189e509$export$f89865e352230a35, (v) => $279507a6f189e509$export$f89865e352230a35 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox--exit", () => $279507a6f189e509$export$867c05a16e5d2a7c, (v) => $279507a6f189e509$export$867c05a16e5d2a7c = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkbox--exitActive", () => $279507a6f189e509$export$61b66671605f1b1a, (v) => $279507a6f189e509$export$61b66671605f1b1a = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-checkboxWrapper", () => $279507a6f189e509$export$41a33ae9818c51be, (v) => $279507a6f189e509$export$41a33ae9818c51be = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-content", () => $279507a6f189e509$export$dc2b0b08d47555, (v) => $279507a6f189e509$export$dc2b0b08d47555 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-description", () => $279507a6f189e509$export$5d2adfeb248e654a, (v) => $279507a6f189e509$export$5d2adfeb248e654a = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-dragPreview", () => $279507a6f189e509$export$84286d36e5c00c55, (v) => $279507a6f189e509$export$84286d36e5c00c55 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-dragPreview--compact", () => $279507a6f189e509$export$fcabaf52adc14efe, (v) => $279507a6f189e509$export$fcabaf52adc14efe = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-dragPreview--multiple", () => $279507a6f189e509$export$2fb3c7c2d5c0d720, (v) => $279507a6f189e509$export$2fb3c7c2d5c0d720 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-dragPreview--spacious", () => $279507a6f189e509$export$256b97e812db1642, (v) => $279507a6f189e509$export$256b97e812db1642 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-draghandle-button", () => $279507a6f189e509$export$4784f0c2ec8bbf7b, (v) => $279507a6f189e509$export$4784f0c2ec8bbf7b = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-draghandle-container", () => $279507a6f189e509$export$f385e7456ddfee84, (v) => $279507a6f189e509$export$f385e7456ddfee84 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-grid", () => $279507a6f189e509$export$aaaf83100348d389, (v) => $279507a6f189e509$export$aaaf83100348d389 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-parentIndicator", () => $279507a6f189e509$export$39b42a8a24290c17, (v) => $279507a6f189e509$export$39b42a8a24290c17 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-parentIndicator--hasChildItems", () => $279507a6f189e509$export$c29dcfa80dab8c88, (v) => $279507a6f189e509$export$c29dcfa80dab8c88 = v);
$parcel$export($279507a6f189e509$exports, "react-spectrum-ListViewItem-thumbnail", () => $279507a6f189e509$export$38c6ebb7f58dedaa, (v) => $279507a6f189e509$export$38c6ebb7f58dedaa = v);
$parcel$export($279507a6f189e509$exports, "round-bottoms", () => $279507a6f189e509$export$8a26470723c97997, (v) => $279507a6f189e509$export$8a26470723c97997 = v);
$parcel$export($279507a6f189e509$exports, "round-tops", () => $279507a6f189e509$export$3b4577d06eb82ec0, (v) => $279507a6f189e509$export$3b4577d06eb82ec0 = v);
$parcel$export($279507a6f189e509$exports, "thumbnail", () => $279507a6f189e509$export$e77609e6ef08541e, (v) => $279507a6f189e509$export$e77609e6ef08541e = v);
var $279507a6f189e509$export$bd7f6383fb9d6b48;
var $279507a6f189e509$export$e324594224ef24da;
var $279507a6f189e509$export$4aebd143f31d0c0d;
var $279507a6f189e509$export$b07e517d22efa1ca;
var $279507a6f189e509$export$980c1089c0604ea3;
var $279507a6f189e509$export$a7db06668cad9adb;
var $279507a6f189e509$export$6d59db4903f20f7d;
var $279507a6f189e509$export$7b34ab0dc3a6dc3b;
var $279507a6f189e509$export$f39a09f249340e2a;
var $279507a6f189e509$export$27f4d14f7fc5b94;
var $279507a6f189e509$export$20fd0f7cd4e6112f;
var $279507a6f189e509$export$d35bc1e505d1ebbf;
var $279507a6f189e509$export$e7dc768d35940237;
var $279507a6f189e509$export$b8813cd5d7824ce7;
var $279507a6f189e509$export$53aa811f86efbfad;
var $279507a6f189e509$export$1e0fb04f31d3c22a;
var $279507a6f189e509$export$4ad4709e9aee8e02;
var $279507a6f189e509$export$1f55b27adf96a61e;
var $279507a6f189e509$export$a2d35348c0cc9bca;
var $279507a6f189e509$export$d9ba3f873d0eaec6;
var $279507a6f189e509$export$eb0d5352b2a55473;
var $279507a6f189e509$export$7150e72e440612bf;
var $279507a6f189e509$export$7c9efe74c4b77579;
var $279507a6f189e509$export$d9b13621bc5dd4d1;
var $279507a6f189e509$export$1bf50375883926;
var $279507a6f189e509$export$5aa06adeaa92dbca;
var $279507a6f189e509$export$b01cb9680b0b8d22;
var $279507a6f189e509$export$cbd30c7809acdd81;
var $279507a6f189e509$export$68b45d42c77fe1cf;
var $279507a6f189e509$export$1b66ee5cd8394caf;
var $279507a6f189e509$export$6b3b7792abd9e5f4;
var $279507a6f189e509$export$f80410436c6352c8;
var $279507a6f189e509$export$2a6a235e0baff403;
var $279507a6f189e509$export$927a243805a411ca;
var $279507a6f189e509$export$ce8c93b50eec2d4f;
var $279507a6f189e509$export$8afa006321ee3a8f;
var $279507a6f189e509$export$60162f05499b7560;
var $279507a6f189e509$export$f9ed481cac4d935a;
var $279507a6f189e509$export$659c081622dc13b0;
var $279507a6f189e509$export$58dfda3903289699;
var $279507a6f189e509$export$26480389595d4515;
var $279507a6f189e509$export$6498e436231eea33;
var $279507a6f189e509$export$f979f76ca45242f2;
var $279507a6f189e509$export$9dc11c0e0825a249;
var $279507a6f189e509$export$f89865e352230a35;
var $279507a6f189e509$export$867c05a16e5d2a7c;
var $279507a6f189e509$export$61b66671605f1b1a;
var $279507a6f189e509$export$41a33ae9818c51be;
var $279507a6f189e509$export$dc2b0b08d47555;
var $279507a6f189e509$export$5d2adfeb248e654a;
var $279507a6f189e509$export$84286d36e5c00c55;
var $279507a6f189e509$export$fcabaf52adc14efe;
var $279507a6f189e509$export$2fb3c7c2d5c0d720;
var $279507a6f189e509$export$256b97e812db1642;
var $279507a6f189e509$export$4784f0c2ec8bbf7b;
var $279507a6f189e509$export$f385e7456ddfee84;
var $279507a6f189e509$export$aaaf83100348d389;
var $279507a6f189e509$export$39b42a8a24290c17;
var $279507a6f189e509$export$c29dcfa80dab8c88;
var $279507a6f189e509$export$38c6ebb7f58dedaa;
var $279507a6f189e509$export$8a26470723c97997;
var $279507a6f189e509$export$3b4577d06eb82ec0;
var $279507a6f189e509$export$e77609e6ef08541e;
$279507a6f189e509$export$bd7f6383fb9d6b48 = `C64cMW_actionmenu`;
$279507a6f189e509$export$e324594224ef24da = `C64cMW_actions`;
$279507a6f189e509$export$4aebd143f31d0c0d = `C64cMW_badge`;
$279507a6f189e509$export$b07e517d22efa1ca = `C64cMW_checkbox`;
$279507a6f189e509$export$980c1089c0604ea3 = `C64cMW_chevron`;
$279507a6f189e509$export$a7db06668cad9adb = `C64cMW_content`;
$279507a6f189e509$export$6d59db4903f20f7d = `C64cMW_description`;
$279507a6f189e509$export$7b34ab0dc3a6dc3b = `C64cMW_draghandle`;
$279507a6f189e509$export$f39a09f249340e2a = `C64cMW_focus-ring`;
$279507a6f189e509$export$27f4d14f7fc5b94 = `C64cMW_has-checkbox`;
$279507a6f189e509$export$20fd0f7cd4e6112f = `C64cMW_is-active`;
$279507a6f189e509$export$d35bc1e505d1ebbf = `C64cMW_is-disabled`;
$279507a6f189e509$export$e7dc768d35940237 = `C64cMW_is-focused`;
$279507a6f189e509$export$b8813cd5d7824ce7 = `C64cMW_is-hovered`;
$279507a6f189e509$export$53aa811f86efbfad = `C64cMW_is-next-selected`;
$279507a6f189e509$export$1e0fb04f31d3c22a = `C64cMW_is-selected`;
$279507a6f189e509$export$4ad4709e9aee8e02 = `C64cMW_react-spectrum-ListView`;
$279507a6f189e509$export$1f55b27adf96a61e = `C64cMW_react-spectrum-ListView--compact`;
$279507a6f189e509$export$a2d35348c0cc9bca = `C64cMW_react-spectrum-ListView--draggable`;
$279507a6f189e509$export$d9ba3f873d0eaec6 = `C64cMW_react-spectrum-ListView--dropTarget`;
$279507a6f189e509$export$eb0d5352b2a55473 = `C64cMW_react-spectrum-ListView--emphasized`;
$279507a6f189e509$export$7150e72e440612bf = `C64cMW_react-spectrum-ListView--hasAnyChildren`;
$279507a6f189e509$export$7c9efe74c4b77579 = `C64cMW_react-spectrum-ListView--hasDescription`;
$279507a6f189e509$export$d9b13621bc5dd4d1 = `C64cMW_react-spectrum-ListView--isHorizontalScrollbarVisible`;
$279507a6f189e509$export$1bf50375883926 = `C64cMW_react-spectrum-ListView--isVerticalScrollbarVisible`;
$279507a6f189e509$export$5aa06adeaa92dbca = `C64cMW_react-spectrum-ListView--loadingMore`;
$279507a6f189e509$export$b01cb9680b0b8d22 = `C64cMW_react-spectrum-ListView--quiet`;
$279507a6f189e509$export$cbd30c7809acdd81 = `C64cMW_react-spectrum-ListView--spacious`;
$279507a6f189e509$export$68b45d42c77fe1cf = `C64cMW_react-spectrum-ListView--wrap`;
$279507a6f189e509$export$1b66ee5cd8394caf = `C64cMW_react-spectrum-ListView-centeredWrapper`;
$279507a6f189e509$export$6b3b7792abd9e5f4 = `C64cMW_react-spectrum-ListView-centeredWrapper--loadingMore`;
$279507a6f189e509$export$f80410436c6352c8 = `C64cMW_react-spectrum-ListView-row`;
$279507a6f189e509$export$2a6a235e0baff403 = `C64cMW_react-spectrum-ListViewInsertionIndicator`;
$279507a6f189e509$export$927a243805a411ca = `C64cMW_react-spectrum-ListViewInsertionIndicator--dropTarget`;
$279507a6f189e509$export$ce8c93b50eec2d4f = `C64cMW_react-spectrum-ListViewItem`;
$279507a6f189e509$export$8afa006321ee3a8f = `C64cMW_react-spectrum-ListViewItem--dropTarget`;
$279507a6f189e509$export$60162f05499b7560 = `C64cMW_react-spectrum-ListViewItem--firstRow`;
$279507a6f189e509$export$f9ed481cac4d935a = `C64cMW_react-spectrum-ListViewItem--isFlushBottom`;
$279507a6f189e509$export$659c081622dc13b0 = `C64cMW_react-spectrum-ListViewItem--lastRow`;
$279507a6f189e509$export$58dfda3903289699 = `C64cMW_react-spectrum-ListViewItem-actionmenu`;
$279507a6f189e509$export$26480389595d4515 = `C64cMW_react-spectrum-ListViewItem-actions`;
$279507a6f189e509$export$6498e436231eea33 = `C64cMW_react-spectrum-ListViewItem-badge`;
$279507a6f189e509$export$f979f76ca45242f2 = `C64cMW_react-spectrum-ListViewItem-checkbox`;
$279507a6f189e509$export$9dc11c0e0825a249 = `C64cMW_react-spectrum-ListViewItem-checkbox--enter`;
$279507a6f189e509$export$f89865e352230a35 = `C64cMW_react-spectrum-ListViewItem-checkbox--enterActive`;
$279507a6f189e509$export$867c05a16e5d2a7c = `C64cMW_react-spectrum-ListViewItem-checkbox--exit`;
$279507a6f189e509$export$61b66671605f1b1a = `C64cMW_react-spectrum-ListViewItem-checkbox--exitActive`;
$279507a6f189e509$export$41a33ae9818c51be = `C64cMW_react-spectrum-ListViewItem-checkboxWrapper`;
$279507a6f189e509$export$dc2b0b08d47555 = `C64cMW_react-spectrum-ListViewItem-content`;
$279507a6f189e509$export$5d2adfeb248e654a = `C64cMW_react-spectrum-ListViewItem-description`;
$279507a6f189e509$export$84286d36e5c00c55 = `C64cMW_react-spectrum-ListViewItem-dragPreview`;
$279507a6f189e509$export$fcabaf52adc14efe = `C64cMW_react-spectrum-ListViewItem-dragPreview--compact`;
$279507a6f189e509$export$2fb3c7c2d5c0d720 = `C64cMW_react-spectrum-ListViewItem-dragPreview--multiple`;
$279507a6f189e509$export$256b97e812db1642 = `C64cMW_react-spectrum-ListViewItem-dragPreview--spacious`;
$279507a6f189e509$export$4784f0c2ec8bbf7b = `C64cMW_react-spectrum-ListViewItem-draghandle-button`;
$279507a6f189e509$export$f385e7456ddfee84 = `C64cMW_react-spectrum-ListViewItem-draghandle-container`;
$279507a6f189e509$export$aaaf83100348d389 = `C64cMW_react-spectrum-ListViewItem-grid`;
$279507a6f189e509$export$39b42a8a24290c17 = `C64cMW_react-spectrum-ListViewItem-parentIndicator`;
$279507a6f189e509$export$c29dcfa80dab8c88 = `C64cMW_react-spectrum-ListViewItem-parentIndicator--hasChildItems`;
$279507a6f189e509$export$38c6ebb7f58dedaa = `C64cMW_react-spectrum-ListViewItem-thumbnail`;
$279507a6f189e509$export$8a26470723c97997 = `C64cMW_round-bottoms`;
$279507a6f189e509$export$3b4577d06eb82ec0 = `C64cMW_round-tops`;
$279507a6f189e509$export$e77609e6ef08541e = `C64cMW_thumbnail`;





function $fc02fd49ab58c72e$export$2e2bcd8739ae039(props) {
    let { dropState: dropState, dragAndDropHooks: dragAndDropHooks } = (0, $k9Du2$react.useContext)((0, $60684b441be8e84c$export$870039b0abfe3de0));
    const { target: target, isPresentationOnly: isPresentationOnly } = props;
    let ref = (0, $k9Du2$react.useRef)();
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator(props, dropState, ref);
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $k9Du2$reactariavisuallyhidden.useVisuallyHidden)();
    let isDropTarget = dropState.isDropTarget(target);
    if (!isDropTarget && dropIndicatorProps["aria-hidden"]) return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps["aria-hidden"]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        role: "gridcell",
        "aria-selected": "false",
        className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListViewInsertionIndicator", {
            "react-spectrum-ListViewInsertionIndicator--dropTarget": isDropTarget
        })
    }, !isPresentationOnly && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        ...visuallyHiddenProps,
        role: "button",
        ...dropIndicatorProps,
        ref: ref
    })));
}


var $2ebea2d8da6d3b78$exports = {};


































$2ebea2d8da6d3b78$exports = {
    "ar-AE": $1beeff0a5a0ccb2a$exports,
    "bg-BG": $8d53c5c3a365f51f$exports,
    "cs-CZ": $99a0008efa578202$exports,
    "da-DK": $fed6a92f3a9a90c2$exports,
    "de-DE": $ad6943b7faf38b41$exports,
    "el-GR": $ed6da119df8f733b$exports,
    "en-US": $a922ddd791b2701d$exports,
    "es-ES": $1616ac893d9df40d$exports,
    "et-EE": $71139980c308673a$exports,
    "fi-FI": $df400f4d8990deb0$exports,
    "fr-FR": $4fedac3a4a024339$exports,
    "he-IL": $0c1298be372fa672$exports,
    "hr-HR": $aac34baf9a6ccbce$exports,
    "hu-HU": $6d93c8a72fbf4a90$exports,
    "it-IT": $69346d8975e9f344$exports,
    "ja-JP": $b263dd2c132d2852$exports,
    "ko-KR": $b79146096cdb30c7$exports,
    "lt-LT": $adc50e375b062a9e$exports,
    "lv-LV": $20ba6eae003d4c10$exports,
    "nb-NO": $6a03ad2ccd528f82$exports,
    "nl-NL": $6bdc84276c605774$exports,
    "pl-PL": $41e9b40a686907b7$exports,
    "pt-BR": $332f285483e392b9$exports,
    "pt-PT": $21a7a272ae3b9fd4$exports,
    "ro-RO": $7f773822d0ecfb7d$exports,
    "ru-RU": $7d8f25b69820e6ce$exports,
    "sk-SK": $ef1cbac310feac6e$exports,
    "sl-SI": $a37847ff0890b3f8$exports,
    "sr-SP": $f95d6d9be7c5418d$exports,
    "sv-SE": $eb2bbd94105316d8$exports,
    "tr-TR": $afc69bf18d2f3b1a$exports,
    "uk-UA": $eda3e7326f5e4dcb$exports,
    "zh-CN": $6d455ae2db499a05$exports,
    "zh-TW": $0fa2d17f8a898b48$exports
};





/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


















function $9eae7a1cb1535a6b$export$c6bde0c04b033c0e(props) {
    var _layout_getContentSize, _layout_virtualizer;
    let { item: item, isEmphasized: isEmphasized } = props;
    let { state: state, dragState: dragState, dropState: dropState, isListDraggable: isListDraggable, isListDroppable: isListDroppable, layout: layout, dragAndDropHooks: dragAndDropHooks, loadingState: loadingState } = (0, $k9Du2$react.useContext)((0, $60684b441be8e84c$export$870039b0abfe3de0));
    let { direction: direction } = (0, $k9Du2$reactariai18n.useLocale)();
    let rowRef = (0, $k9Du2$react.useRef)();
    let checkboxWrapperRef = (0, $k9Du2$react.useRef)();
    let { isFocusVisible: isFocusVisibleWithin, focusProps: focusWithinProps } = (0, $k9Du2$reactariafocus.useFocusRing)({
        within: true
    });
    let { isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $k9Du2$reactariafocus.useFocusRing)();
    let { rowProps: rowProps, gridCellProps: gridCellProps, isPressed: isPressed, descriptionProps: descriptionProps, isSelected: isSelected, isDisabled: isDisabled, allowsSelection: allowsSelection, hasAction: hasAction } = (0, $k9Du2$reactariagridlist.useGridListItem)({
        node: item,
        isVirtualized: true,
        shouldSelectOnPressUp: isListDraggable
    }, state, rowRef);
    let isDroppable = isListDroppable && !isDisabled;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $k9Du2$reactariainteractions.useHover)({
        isDisabled: !allowsSelection && !hasAction
    });
    let { checkboxProps: checkboxProps } = (0, $k9Du2$reactariagridlist.useGridListSelectionCheckbox)({
        key: item.key
    }, state);
    let hasDescription = (0, $k9Du2$reactspectrumutils.useHasChild)(`.${(0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-description"]}`, rowRef);
    let draggableItem;
    if (isListDraggable) {
        // eslint-disable-next-line react-hooks/rules-of-hooks
        draggableItem = dragAndDropHooks.useDraggableItem({
            key: item.key,
            hasDragButton: true
        }, dragState);
        if (isDisabled) draggableItem = null;
    }
    let droppableItem;
    let isDropTarget;
    let dropIndicator;
    let dropIndicatorRef = (0, $k9Du2$react.useRef)();
    if (isListDroppable) {
        let target = {
            type: "item",
            key: item.key,
            dropPosition: "on"
        };
        isDropTarget = dropState.isDropTarget(target);
        // eslint-disable-next-line react-hooks/rules-of-hooks
        dropIndicator = dragAndDropHooks.useDropIndicator({
            target: target
        }, dropState, dropIndicatorRef);
    }
    let dragButtonRef = (0, ($parcel$interopDefault($k9Du2$react))).useRef();
    let { buttonProps: buttonProps } = (0, $k9Du2$reactariabutton.useButton)({
        ...draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragButtonProps,
        elementType: "div"
    }, dragButtonRef);
    let chevron = direction === "ltr" ? /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, ($parcel$interopDefault($k9Du2$spectrumiconsuiChevronRightMedium))), {
        "aria-hidden": "true",
        UNSAFE_className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListViewItem-parentIndicator", {
            "react-spectrum-ListViewItem-parentIndicator--hasChildItems": item.props.hasChildItems,
            "is-disabled": !hasAction
        })
    }) : /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, ($parcel$interopDefault($k9Du2$spectrumiconsuiChevronLeftMedium))), {
        "aria-hidden": "true",
        UNSAFE_className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListViewItem-parentIndicator", {
            "react-spectrum-ListViewItem-parentIndicator--hasChildItems": item.props.hasChildItems,
            "is-disabled": !hasAction
        })
    });
    let showCheckbox = state.selectionManager.selectionMode !== "none" && state.selectionManager.selectionBehavior === "toggle";
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $k9Du2$reactariavisuallyhidden.useVisuallyHidden)();
    let dropProps = isDroppable ? droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps : {
        "aria-hidden": droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps["aria-hidden"]
    };
    const mergedProps = (0, $k9Du2$reactariautils.mergeProps)(rowProps, draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragProps, dropProps, hoverProps, focusWithinProps, focusProps, // Remove tab index from list row if performing a screenreader drag. This prevents TalkBack from focusing the row,
    // allowing for single swipe navigation between row drop indicator
    (dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.isVirtualDragging()) && {
        tabIndex: null
    });
    let isFirstRow = item.prevKey == null;
    let isLastRow = item.nextKey == null;
    // Figure out if the ListView content is equal or greater in height to the container. If so, we'll need to round the bottom
    // border corners of the last row when selected and we can get rid of the bottom border if it isn't selected to avoid border overlap
    // with bottom border
    let isFlushWithContainerBottom = false;
    if (isLastRow && loadingState !== "loadingMore") {
        if (((_layout_getContentSize = layout.getContentSize()) === null || _layout_getContentSize === void 0 ? void 0 : _layout_getContentSize.height) >= ((_layout_virtualizer = layout.virtualizer) === null || _layout_virtualizer === void 0 ? void 0 : _layout_virtualizer.getVisibleRect().height)) isFlushWithContainerBottom = true;
    }
    // previous item isn't selected
    // and the previous item isn't focused or, if it is focused, then if focus globally isn't visible or just focus isn't in the listview
    let roundTops = !state.selectionManager.isSelected(item.prevKey) && (state.selectionManager.focusedKey !== item.prevKey || !((0, $k9Du2$reactariainteractions.isFocusVisible)() && state.selectionManager.isFocused));
    let roundBottoms = !state.selectionManager.isSelected(item.nextKey) && (state.selectionManager.focusedKey !== item.nextKey || !((0, $k9Du2$reactariainteractions.isFocusVisible)() && state.selectionManager.isFocused));
    let content = typeof item.rendered === "string" ? /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumtext.Text), null, item.rendered) : item.rendered;
    if (isDisabled) content = /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumprovider.Provider), {
        isDisabled: true
    }, content);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        ...mergedProps,
        className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListView-row", {
            "focus-ring": isFocusVisible,
            "round-tops": roundTops || isHovered && !isSelected && state.selectionManager.focusedKey !== item.key,
            "round-bottoms": roundBottoms || isHovered && !isSelected && state.selectionManager.focusedKey !== item.key
        }),
        ref: rowRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        // TODO: refactor the css here now that we are focusing the row?
        className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListViewItem", {
            "is-active": isPressed,
            "is-focused": isFocusVisibleWithin,
            "focus-ring": isFocusVisible,
            "is-hovered": isHovered,
            "is-selected": isSelected,
            "is-disabled": isDisabled,
            "is-prev-selected": state.selectionManager.isSelected(item.prevKey),
            "is-next-selected": state.selectionManager.isSelected(item.nextKey),
            "react-spectrum-ListViewItem--highlightSelection": state.selectionManager.selectionBehavior === "replace" && (isSelected || state.selectionManager.isSelected(item.nextKey)),
            "react-spectrum-ListViewItem--dropTarget": !!isDropTarget,
            "react-spectrum-ListViewItem--firstRow": isFirstRow,
            "react-spectrum-ListViewItem--lastRow": isLastRow,
            "react-spectrum-ListViewItem--isFlushBottom": isFlushWithContainerBottom,
            "react-spectrum-ListViewItem--hasDescription": hasDescription
        }),
        ...gridCellProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumlayout.Grid), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-grid"]
    }, isListDraggable && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-draghandle-container"]
    }, !isDisabled && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactariafocus.FocusRing), {
        focusRingClass: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        ...buttonProps,
        className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListViewItem-draghandle-button"),
        style: !isFocusVisibleWithin ? {
            ...visuallyHiddenProps.style
        } : {},
        ref: dragButtonRef,
        draggable: "true"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, ($parcel$interopDefault($k9Du2$spectrumiconsuiListGripper))), null)))), isListDroppable && !(dropIndicator === null || dropIndicator === void 0 ? void 0 : dropIndicator.isHidden) && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicator === null || dropIndicator === void 0 ? void 0 : dropIndicator.dropIndicatorProps,
        ref: dropIndicatorRef
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reacttransitiongroup.CSSTransition), {
        in: showCheckbox,
        unmountOnExit: true,
        classNames: {
            enter: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-checkbox--enter"],
            enterActive: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-checkbox--enterActive"],
            exit: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-checkbox--exit"],
            exitActive: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-checkbox--exitActive"]
        },
        timeout: 160,
        nodeRef: checkboxWrapperRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        ref: checkboxWrapperRef,
        className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-checkboxWrapper"]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumcheckbox.Checkbox), {
        ...checkboxProps,
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-checkbox"],
        isEmphasized: isEmphasized
    }))), /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumutils.SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-content"]
            },
            description: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-description"],
                ...descriptionProps
            },
            illustration: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-thumbnail"]
            },
            image: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-thumbnail"]
            },
            actionButton: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-actions"],
                isQuiet: true
            },
            actionGroup: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-actions"],
                isQuiet: true,
                density: "compact"
            },
            actionMenu: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-actionmenu"],
                isQuiet: true
            }
        }
    }, content, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumutils.ClearSlots), null, chevron)))));
}







function $c59d3898f402f50b$export$2e2bcd8739ae039() {
    let { dropState: dropState, dragAndDropHooks: dragAndDropHooks } = (0, $k9Du2$react.useContext)((0, $60684b441be8e84c$export$870039b0abfe3de0));
    let ref = (0, $k9Du2$react.useRef)();
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator({
        target: {
            type: "root"
        }
    }, dropState, ref);
    let isDropTarget = dropState.isDropTarget({
        type: "root"
    });
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $k9Du2$reactariavisuallyhidden.useVisuallyHidden)();
    if (!isDropTarget && dropIndicatorProps["aria-hidden"]) return null;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps["aria-hidden"]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        role: "gridcell",
        "aria-selected": "false"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicatorProps,
        ref: ref
    })));
}


/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $3d665b19f7865ff9$export$905ab40ac2179daa(props) {
    let { item: item, itemCount: itemCount, itemHeight: itemHeight, density: density } = props;
    let isDraggingMultiple = itemCount > 1;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        style: {
            height: itemHeight
        },
        className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListViewItem", "react-spectrum-ListViewItem-dragPreview", {
            "react-spectrum-ListViewItem-dragPreview--multiple": isDraggingMultiple
        }, `react-spectrum-ListViewItem-dragPreview--${density}`)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumlayout.Grid), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-grid"]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumutils.SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-content"]
            },
            description: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-description"]
            },
            illustration: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-thumbnail"]
            },
            image: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-thumbnail"]
            },
            actionButton: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-actions"],
                isQuiet: true
            },
            actionGroup: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-actions"],
                isQuiet: true,
                density: "compact"
            },
            actionMenu: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports)))["react-spectrum-ListViewItem-actionmenu"],
                isQuiet: true
            }
        }
    }, typeof item.rendered === "string" ? /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumtext.Text), null, item.rendered) : item.rendered, isDraggingMultiple && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListViewItem-badge")
    }, itemCount))));
}





const $60684b441be8e84c$export$870039b0abfe3de0 = /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createContext(null);
const $60684b441be8e84c$var$ROW_HEIGHTS = {
    compact: {
        medium: 32,
        large: 40
    },
    regular: {
        medium: 40,
        large: 50
    },
    spacious: {
        medium: 48,
        large: 60
    }
};
function $60684b441be8e84c$var$useListLayout(state, density, overflowMode) {
    let { scale: scale } = (0, $k9Du2$reactspectrumprovider.useProvider)();
    let collator = (0, $k9Du2$reactariai18n.useCollator)({
        usage: "search",
        sensitivity: "base"
    });
    let isEmpty = state.collection.size === 0;
    let layout = (0, $k9Du2$react.useMemo)(()=>new (0, $k9Du2$reactstatelylayout.ListLayout)({
            estimatedRowHeight: $60684b441be8e84c$var$ROW_HEIGHTS[density][scale],
            padding: 0,
            collator: collator,
            loaderHeight: isEmpty ? null : $60684b441be8e84c$var$ROW_HEIGHTS[density][scale]
        }), [
        collator,
        scale,
        density,
        isEmpty,
        overflowMode
    ]);
    layout.collection = state.collection;
    layout.disabledKeys = state.disabledKeys;
    return layout;
}
function $60684b441be8e84c$var$ListView(props, ref) {
    var _dropState_target;
    let { density: density = "regular", loadingState: loadingState, onLoadMore: onLoadMore, isQuiet: isQuiet, overflowMode: overflowMode = "truncate", onAction: onAction, dragAndDropHooks: dragAndDropHooks, ...otherProps } = props;
    let isListDraggable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDraggableCollectionState);
    let isListDroppable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDroppableCollectionState);
    let dragHooksProvided = (0, $k9Du2$react.useRef)(isListDraggable);
    let dropHooksProvided = (0, $k9Du2$react.useRef)(isListDroppable);
    (0, $k9Du2$react.useEffect)(()=>{
        if (dragHooksProvided.current !== isListDraggable) console.warn("Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.");
        if (dropHooksProvided.current !== isListDroppable) console.warn("Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.");
    }, [
        isListDraggable,
        isListDroppable
    ]);
    let domRef = (0, $k9Du2$reactspectrumutils.useDOMRef)(ref);
    let state = (0, $k9Du2$reactstatelylist.useListState)({
        ...props,
        selectionBehavior: props.selectionStyle === "highlight" ? "replace" : "toggle"
    });
    let { collection: collection, selectionManager: selectionManager } = state;
    let stringFormatter = (0, $k9Du2$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($2ebea2d8da6d3b78$exports))), "@react-spectrum/list");
    let isLoading = loadingState === "loading" || loadingState === "loadingMore";
    let { styleProps: styleProps } = (0, $k9Du2$reactspectrumutils.useStyleProps)(props);
    let dragState;
    let preview = (0, $k9Du2$react.useRef)(null);
    if (isListDraggable) {
        dragState = dragAndDropHooks.useDraggableCollectionState({
            collection: collection,
            selectionManager: selectionManager,
            preview: preview
        });
        dragAndDropHooks.useDraggableCollection({}, dragState, domRef);
    }
    let layout = $60684b441be8e84c$var$useListLayout(state, props.density || "regular", overflowMode);
    // !!0 is false, so we can cast size or undefined and they'll be falsy
    layout.allowDisabledKeyFocus = state.selectionManager.disabledBehavior === "selection" || !!(dragState === null || dragState === void 0 ? void 0 : dragState.draggingKeys.size);
    let DragPreview = dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.DragPreview;
    let dropState;
    let droppableCollection;
    let isRootDropTarget;
    if (isListDroppable) {
        dropState = dragAndDropHooks.useDroppableCollectionState({
            collection: collection,
            selectionManager: selectionManager
        });
        droppableCollection = dragAndDropHooks.useDroppableCollection({
            keyboardDelegate: layout,
            dropTargetDelegate: layout
        }, dropState, domRef);
        isRootDropTarget = dropState.isDropTarget({
            type: "root"
        });
    }
    let { gridProps: gridProps } = (0, $k9Du2$reactariagridlist.useGridList)({
        ...props,
        isVirtualized: true,
        keyboardDelegate: layout,
        onAction: onAction
    }, state, domRef);
    // Sync loading state into the layout.
    layout.isLoading = isLoading;
    let focusedKey = selectionManager.focusedKey;
    if ((dropState === null || dropState === void 0 ? void 0 : (_dropState_target = dropState.target) === null || _dropState_target === void 0 ? void 0 : _dropState_target.type) === "item") focusedKey = dropState.target.key;
    // wait for layout to get accurate measurements
    let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = (0, $k9Du2$react.useState)(false);
    let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = (0, $k9Du2$react.useState)(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    (0, $k9Du2$reactariautils.useLayoutEffect)(()=>{
        if (domRef.current) {
            // 2 is the width of the border which is not part of the box size
            setVerticalScollbarVisible(domRef.current.clientWidth + 2 < domRef.current.offsetWidth);
            setHorizontalScollbarVisible(domRef.current.clientHeight + 2 < domRef.current.offsetHeight);
        }
    });
    let hasAnyChildren = (0, $k9Du2$react.useMemo)(()=>[
            ...collection
        ].some((item)=>item.hasChildNodes), [
        collection
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement($60684b441be8e84c$export$870039b0abfe3de0.Provider, {
        value: {
            state: state,
            dragState: dragState,
            dropState: dropState,
            dragAndDropHooks: dragAndDropHooks,
            onAction: onAction,
            isListDraggable: isListDraggable,
            isListDroppable: isListDroppable,
            layout: layout,
            loadingState: loadingState
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactariafocus.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactariafocus.FocusRing), {
        focusRingClass: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactariavirtualizer.Virtualizer), {
        ...(0, $k9Du2$reactariautils.mergeProps)(isListDroppable && (droppableCollection === null || droppableCollection === void 0 ? void 0 : droppableCollection.collectionProps), gridProps),
        ...(0, $k9Du2$reactariautils.filterDOMProps)(otherProps),
        ...gridProps,
        ...styleProps,
        isLoading: isLoading,
        onLoadMore: onLoadMore,
        ref: domRef,
        focusedKey: focusedKey,
        scrollDirection: "vertical",
        className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListView", `react-spectrum-ListView--${density}`, "react-spectrum-ListView--emphasized", {
            "react-spectrum-ListView--quiet": isQuiet,
            "react-spectrum-ListView--loadingMore": loadingState === "loadingMore",
            "react-spectrum-ListView--draggable": !!isListDraggable,
            "react-spectrum-ListView--dropTarget": !!isRootDropTarget,
            "react-spectrum-ListView--isVerticalScrollbarVisible": isVerticalScrollbarVisible,
            "react-spectrum-ListView--isHorizontalScrollbarVisible": isHorizontalScrollbarVisible,
            "react-spectrum-ListView--hasAnyChildren": hasAnyChildren,
            "react-spectrum-ListView--wrap": overflowMode === "wrap"
        }, styleProps.className),
        layout: layout,
        collection: collection,
        transitionDuration: isLoading ? 160 : 220
    }, (type, item)=>{
        if (type === "item") return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, ($parcel$interopDefault($k9Du2$react))).Fragment, null, isListDroppable && collection.getKeyBefore(item.key) == null && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $c59d3898f402f50b$export$2e2bcd8739ae039), {
            key: "root"
        }), isListDroppable && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $fc02fd49ab58c72e$export$2e2bcd8739ae039), {
            key: `${item.key}-before`,
            target: {
                key: item.key,
                type: "item",
                dropPosition: "before"
            }
        }), /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $9eae7a1cb1535a6b$export$c6bde0c04b033c0e), {
            item: item,
            isEmphasized: true,
            hasActions: !!onAction
        }), isListDroppable && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $fc02fd49ab58c72e$export$2e2bcd8739ae039), {
            key: `${item.key}-after`,
            target: {
                key: item.key,
                type: "item",
                dropPosition: "after"
            },
            isPresentationOnly: collection.getKeyAfter(item.key) != null
        }));
        else if (type === "loader") return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement($60684b441be8e84c$var$CenteredWrapper, null, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $k9Du2$reactspectrumprogress.ProgressCircle), {
            isIndeterminate: true,
            "aria-label": collection.size > 0 ? stringFormatter.format("loadingMore") : stringFormatter.format("loading")
        }));
        else if (type === "placeholder") {
            let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
            if (emptyState == null) return null;
            return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement($60684b441be8e84c$var$CenteredWrapper, null, emptyState);
        }
    }))), DragPreview && isListDraggable && /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement(DragPreview, {
        ref: preview
    }, ()=>{
        if (dragAndDropHooks.renderPreview) return dragAndDropHooks.renderPreview(dragState.draggingKeys, dragState.draggedKey);
        let item = state.collection.getItem(dragState.draggedKey);
        let itemCount = dragState.draggingKeys.size;
        let itemHeight = layout.getLayoutInfo(dragState.draggedKey).rect.height;
        return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement((0, $3d665b19f7865ff9$export$905ab40ac2179daa), {
            item: item,
            itemCount: itemCount,
            itemHeight: itemHeight,
            density: density
        });
    }));
}
function $60684b441be8e84c$var$CenteredWrapper({ children: children }) {
    let { state: state } = (0, $k9Du2$react.useContext)($60684b441be8e84c$export$870039b0abfe3de0);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.size + 1,
        className: (0, $k9Du2$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($279507a6f189e509$exports))), "react-spectrum-ListView-centeredWrapper", {
            "react-spectrum-ListView-centeredWrapper--loadingMore": state.collection.size > 0
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).createElement("div", {
        role: "gridcell"
    }, children));
}
/**
 * A ListView displays a list of interactive items, and allows a user to navigate, select, or perform an action.
 */ const $60684b441be8e84c$export$84d0dd190d551cd1 = /*#__PURE__*/ (0, ($parcel$interopDefault($k9Du2$react))).forwardRef($60684b441be8e84c$var$ListView);
