// @react-spectrum/list/dist/module.js
"use client";

import "./main.css";
import $a8be062dce739a7d$exports from "./ar-AE.module.js";
import $ebee5026543c5714$exports from "./bg-BG.module.js";
import $ec2dceffa5b2d037$exports from "./cs-CZ.module.js";
import $e49254d6635b1a35$exports from "./da-DK.module.js";
import $c25de0af306f5a70$exports from "./de-DE.module.js";
import $1aedf9be304cf865$exports from "./el-GR.module.js";
import $653a020875a8fea3$exports from "./en-US.module.js";
import $24fba9a73a2aba1c$exports from "./es-ES.module.js";
import $0994847e6d27862d$exports from "./et-EE.module.js";
import $084b0bc3d6e3d2a1$exports from "./fi-FI.module.js";
import $3c567f46aec1205a$exports from "./fr-FR.module.js";
import $e5a01f2966bf0b9a$exports from "./he-IL.module.js";
import $223cc8f6bb1d23dd$exports from "./hr-HR.module.js";
import $c577fab129d558ed$exports from "./hu-HU.module.js";
import $a6e5541030242f67$exports from "./it-IT.module.js";
import $cce2c1ef24294875$exports from "./ja-JP.module.js";
import $01de2837b3b6f15a$exports from "./ko-KR.module.js";
import $5394997707852da6$exports from "./lt-LT.module.js";
import $3f13373950080912$exports from "./lv-LV.module.js";
import $041a8a2796a43a59$exports from "./nb-NO.module.js";
import $ef957358eac33027$exports from "./nl-NL.module.js";
import $09a0ddc6d157c4fe$exports from "./pl-PL.module.js";
import $5e4abb947f3a742c$exports from "./pt-BR.module.js";
import $18faaf4c3266f94f$exports from "./pt-PT.module.js";
import $b3819b882aaf2b05$exports from "./ro-RO.module.js";
import $bbb9489f20c7c6c0$exports from "./ru-RU.module.js";
import $12e7bbe01da0d5ad$exports from "./sk-SK.module.js";
import $5582a397c204a97c$exports from "./sl-SI.module.js";
import $aa6549babfa00736$exports from "./sr-SP.module.js";
import $13248984d46c1e4f$exports from "./sv-SE.module.js";
import $7af10700b945dced$exports from "./tr-TR.module.js";
import $00a02749c6fa842b$exports from "./uk-UA.module.js";
import $e3b2b7128a600af6$exports from "./zh-CN.module.js";
import $8751d9847e270a98$exports from "./zh-TW.module.js";
import {Item as $75e4a7b01832b12d$re_export$Item} from "@react-stately/collections";
import {useGridList as $fWI2a$useGridList, useGridListItem as $fWI2a$useGridListItem, useGridListSelectionCheckbox as $fWI2a$useGridListSelectionCheckbox} from "@react-aria/gridlist";
import {useDOMRef as $fWI2a$useDOMRef, useStyleProps as $fWI2a$useStyleProps, classNames as $fWI2a$classNames, useHasChild as $fWI2a$useHasChild, SlotProvider as $fWI2a$SlotProvider, ClearSlots as $fWI2a$ClearSlots} from "@react-spectrum/utils";
import {useLayoutEffect as $fWI2a$useLayoutEffect, mergeProps as $fWI2a$mergeProps, filterDOMProps as $fWI2a$filterDOMProps} from "@react-aria/utils";
import {FocusScope as $fWI2a$FocusScope, FocusRing as $fWI2a$FocusRing, useFocusRing as $fWI2a$useFocusRing} from "@react-aria/focus";
import {ListLayout as $fWI2a$ListLayout} from "@react-stately/layout";
import {useListState as $fWI2a$useListState} from "@react-stately/list";
import {ProgressCircle as $fWI2a$ProgressCircle} from "@react-spectrum/progress";
import $fWI2a$react, {useMemo as $fWI2a$useMemo, useRef as $fWI2a$useRef, useEffect as $fWI2a$useEffect, useState as $fWI2a$useState, useContext as $fWI2a$useContext} from "react";
import {useCollator as $fWI2a$useCollator, useLocalizedStringFormatter as $fWI2a$useLocalizedStringFormatter, useLocale as $fWI2a$useLocale} from "@react-aria/i18n";
import {useProvider as $fWI2a$useProvider, Provider as $fWI2a$Provider} from "@react-spectrum/provider";
import {Virtualizer as $fWI2a$Virtualizer} from "@react-aria/virtualizer";
import {useVisuallyHidden as $fWI2a$useVisuallyHidden} from "@react-aria/visually-hidden";
import {Checkbox as $fWI2a$Checkbox} from "@react-spectrum/checkbox";
import $fWI2a$spectrumiconsuiChevronLeftMedium from "@spectrum-icons/ui/ChevronLeftMedium";
import $fWI2a$spectrumiconsuiChevronRightMedium from "@spectrum-icons/ui/ChevronRightMedium";
import {CSSTransition as $fWI2a$CSSTransition} from "react-transition-group";
import {Grid as $fWI2a$Grid} from "@react-spectrum/layout";
import {useHover as $fWI2a$useHover, isFocusVisible as $fWI2a$isFocusVisible} from "@react-aria/interactions";
import $fWI2a$spectrumiconsuiListGripper from "@spectrum-icons/ui/ListGripper";
import {Text as $fWI2a$Text} from "@react-spectrum/text";
import {useButton as $fWI2a$useButton} from "@react-aria/button";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $c8bc378300208597$exports = {};

$parcel$export($c8bc378300208597$exports, "actionmenu", () => $c8bc378300208597$export$bd7f6383fb9d6b48, (v) => $c8bc378300208597$export$bd7f6383fb9d6b48 = v);
$parcel$export($c8bc378300208597$exports, "actions", () => $c8bc378300208597$export$e324594224ef24da, (v) => $c8bc378300208597$export$e324594224ef24da = v);
$parcel$export($c8bc378300208597$exports, "badge", () => $c8bc378300208597$export$4aebd143f31d0c0d, (v) => $c8bc378300208597$export$4aebd143f31d0c0d = v);
$parcel$export($c8bc378300208597$exports, "checkbox", () => $c8bc378300208597$export$b07e517d22efa1ca, (v) => $c8bc378300208597$export$b07e517d22efa1ca = v);
$parcel$export($c8bc378300208597$exports, "chevron", () => $c8bc378300208597$export$980c1089c0604ea3, (v) => $c8bc378300208597$export$980c1089c0604ea3 = v);
$parcel$export($c8bc378300208597$exports, "content", () => $c8bc378300208597$export$a7db06668cad9adb, (v) => $c8bc378300208597$export$a7db06668cad9adb = v);
$parcel$export($c8bc378300208597$exports, "description", () => $c8bc378300208597$export$6d59db4903f20f7d, (v) => $c8bc378300208597$export$6d59db4903f20f7d = v);
$parcel$export($c8bc378300208597$exports, "draghandle", () => $c8bc378300208597$export$7b34ab0dc3a6dc3b, (v) => $c8bc378300208597$export$7b34ab0dc3a6dc3b = v);
$parcel$export($c8bc378300208597$exports, "focus-ring", () => $c8bc378300208597$export$f39a09f249340e2a, (v) => $c8bc378300208597$export$f39a09f249340e2a = v);
$parcel$export($c8bc378300208597$exports, "has-checkbox", () => $c8bc378300208597$export$27f4d14f7fc5b94, (v) => $c8bc378300208597$export$27f4d14f7fc5b94 = v);
$parcel$export($c8bc378300208597$exports, "is-active", () => $c8bc378300208597$export$20fd0f7cd4e6112f, (v) => $c8bc378300208597$export$20fd0f7cd4e6112f = v);
$parcel$export($c8bc378300208597$exports, "is-disabled", () => $c8bc378300208597$export$d35bc1e505d1ebbf, (v) => $c8bc378300208597$export$d35bc1e505d1ebbf = v);
$parcel$export($c8bc378300208597$exports, "is-focused", () => $c8bc378300208597$export$e7dc768d35940237, (v) => $c8bc378300208597$export$e7dc768d35940237 = v);
$parcel$export($c8bc378300208597$exports, "is-hovered", () => $c8bc378300208597$export$b8813cd5d7824ce7, (v) => $c8bc378300208597$export$b8813cd5d7824ce7 = v);
$parcel$export($c8bc378300208597$exports, "is-next-selected", () => $c8bc378300208597$export$53aa811f86efbfad, (v) => $c8bc378300208597$export$53aa811f86efbfad = v);
$parcel$export($c8bc378300208597$exports, "is-selected", () => $c8bc378300208597$export$1e0fb04f31d3c22a, (v) => $c8bc378300208597$export$1e0fb04f31d3c22a = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView", () => $c8bc378300208597$export$4ad4709e9aee8e02, (v) => $c8bc378300208597$export$4ad4709e9aee8e02 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--compact", () => $c8bc378300208597$export$1f55b27adf96a61e, (v) => $c8bc378300208597$export$1f55b27adf96a61e = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--draggable", () => $c8bc378300208597$export$a2d35348c0cc9bca, (v) => $c8bc378300208597$export$a2d35348c0cc9bca = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--dropTarget", () => $c8bc378300208597$export$d9ba3f873d0eaec6, (v) => $c8bc378300208597$export$d9ba3f873d0eaec6 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--emphasized", () => $c8bc378300208597$export$eb0d5352b2a55473, (v) => $c8bc378300208597$export$eb0d5352b2a55473 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--hasAnyChildren", () => $c8bc378300208597$export$7150e72e440612bf, (v) => $c8bc378300208597$export$7150e72e440612bf = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--hasDescription", () => $c8bc378300208597$export$7c9efe74c4b77579, (v) => $c8bc378300208597$export$7c9efe74c4b77579 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--isHorizontalScrollbarVisible", () => $c8bc378300208597$export$d9b13621bc5dd4d1, (v) => $c8bc378300208597$export$d9b13621bc5dd4d1 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--isVerticalScrollbarVisible", () => $c8bc378300208597$export$1bf50375883926, (v) => $c8bc378300208597$export$1bf50375883926 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--loadingMore", () => $c8bc378300208597$export$5aa06adeaa92dbca, (v) => $c8bc378300208597$export$5aa06adeaa92dbca = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--quiet", () => $c8bc378300208597$export$b01cb9680b0b8d22, (v) => $c8bc378300208597$export$b01cb9680b0b8d22 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--spacious", () => $c8bc378300208597$export$cbd30c7809acdd81, (v) => $c8bc378300208597$export$cbd30c7809acdd81 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView--wrap", () => $c8bc378300208597$export$68b45d42c77fe1cf, (v) => $c8bc378300208597$export$68b45d42c77fe1cf = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView-centeredWrapper", () => $c8bc378300208597$export$1b66ee5cd8394caf, (v) => $c8bc378300208597$export$1b66ee5cd8394caf = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView-centeredWrapper--loadingMore", () => $c8bc378300208597$export$6b3b7792abd9e5f4, (v) => $c8bc378300208597$export$6b3b7792abd9e5f4 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListView-row", () => $c8bc378300208597$export$f80410436c6352c8, (v) => $c8bc378300208597$export$f80410436c6352c8 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewInsertionIndicator", () => $c8bc378300208597$export$2a6a235e0baff403, (v) => $c8bc378300208597$export$2a6a235e0baff403 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewInsertionIndicator--dropTarget", () => $c8bc378300208597$export$927a243805a411ca, (v) => $c8bc378300208597$export$927a243805a411ca = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem", () => $c8bc378300208597$export$ce8c93b50eec2d4f, (v) => $c8bc378300208597$export$ce8c93b50eec2d4f = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem--dropTarget", () => $c8bc378300208597$export$8afa006321ee3a8f, (v) => $c8bc378300208597$export$8afa006321ee3a8f = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem--firstRow", () => $c8bc378300208597$export$60162f05499b7560, (v) => $c8bc378300208597$export$60162f05499b7560 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem--isFlushBottom", () => $c8bc378300208597$export$f9ed481cac4d935a, (v) => $c8bc378300208597$export$f9ed481cac4d935a = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem--lastRow", () => $c8bc378300208597$export$659c081622dc13b0, (v) => $c8bc378300208597$export$659c081622dc13b0 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-actionmenu", () => $c8bc378300208597$export$58dfda3903289699, (v) => $c8bc378300208597$export$58dfda3903289699 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-actions", () => $c8bc378300208597$export$26480389595d4515, (v) => $c8bc378300208597$export$26480389595d4515 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-badge", () => $c8bc378300208597$export$6498e436231eea33, (v) => $c8bc378300208597$export$6498e436231eea33 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-checkbox", () => $c8bc378300208597$export$f979f76ca45242f2, (v) => $c8bc378300208597$export$f979f76ca45242f2 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-checkbox--enter", () => $c8bc378300208597$export$9dc11c0e0825a249, (v) => $c8bc378300208597$export$9dc11c0e0825a249 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-checkbox--enterActive", () => $c8bc378300208597$export$f89865e352230a35, (v) => $c8bc378300208597$export$f89865e352230a35 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-checkbox--exit", () => $c8bc378300208597$export$867c05a16e5d2a7c, (v) => $c8bc378300208597$export$867c05a16e5d2a7c = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-checkbox--exitActive", () => $c8bc378300208597$export$61b66671605f1b1a, (v) => $c8bc378300208597$export$61b66671605f1b1a = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-checkboxWrapper", () => $c8bc378300208597$export$41a33ae9818c51be, (v) => $c8bc378300208597$export$41a33ae9818c51be = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-content", () => $c8bc378300208597$export$dc2b0b08d47555, (v) => $c8bc378300208597$export$dc2b0b08d47555 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-description", () => $c8bc378300208597$export$5d2adfeb248e654a, (v) => $c8bc378300208597$export$5d2adfeb248e654a = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-dragPreview", () => $c8bc378300208597$export$84286d36e5c00c55, (v) => $c8bc378300208597$export$84286d36e5c00c55 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-dragPreview--compact", () => $c8bc378300208597$export$fcabaf52adc14efe, (v) => $c8bc378300208597$export$fcabaf52adc14efe = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-dragPreview--multiple", () => $c8bc378300208597$export$2fb3c7c2d5c0d720, (v) => $c8bc378300208597$export$2fb3c7c2d5c0d720 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-dragPreview--spacious", () => $c8bc378300208597$export$256b97e812db1642, (v) => $c8bc378300208597$export$256b97e812db1642 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-draghandle-button", () => $c8bc378300208597$export$4784f0c2ec8bbf7b, (v) => $c8bc378300208597$export$4784f0c2ec8bbf7b = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-draghandle-container", () => $c8bc378300208597$export$f385e7456ddfee84, (v) => $c8bc378300208597$export$f385e7456ddfee84 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-grid", () => $c8bc378300208597$export$aaaf83100348d389, (v) => $c8bc378300208597$export$aaaf83100348d389 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-parentIndicator", () => $c8bc378300208597$export$39b42a8a24290c17, (v) => $c8bc378300208597$export$39b42a8a24290c17 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-parentIndicator--hasChildItems", () => $c8bc378300208597$export$c29dcfa80dab8c88, (v) => $c8bc378300208597$export$c29dcfa80dab8c88 = v);
$parcel$export($c8bc378300208597$exports, "react-spectrum-ListViewItem-thumbnail", () => $c8bc378300208597$export$38c6ebb7f58dedaa, (v) => $c8bc378300208597$export$38c6ebb7f58dedaa = v);
$parcel$export($c8bc378300208597$exports, "round-bottoms", () => $c8bc378300208597$export$8a26470723c97997, (v) => $c8bc378300208597$export$8a26470723c97997 = v);
$parcel$export($c8bc378300208597$exports, "round-tops", () => $c8bc378300208597$export$3b4577d06eb82ec0, (v) => $c8bc378300208597$export$3b4577d06eb82ec0 = v);
$parcel$export($c8bc378300208597$exports, "thumbnail", () => $c8bc378300208597$export$e77609e6ef08541e, (v) => $c8bc378300208597$export$e77609e6ef08541e = v);
var $c8bc378300208597$export$bd7f6383fb9d6b48;
var $c8bc378300208597$export$e324594224ef24da;
var $c8bc378300208597$export$4aebd143f31d0c0d;
var $c8bc378300208597$export$b07e517d22efa1ca;
var $c8bc378300208597$export$980c1089c0604ea3;
var $c8bc378300208597$export$a7db06668cad9adb;
var $c8bc378300208597$export$6d59db4903f20f7d;
var $c8bc378300208597$export$7b34ab0dc3a6dc3b;
var $c8bc378300208597$export$f39a09f249340e2a;
var $c8bc378300208597$export$27f4d14f7fc5b94;
var $c8bc378300208597$export$20fd0f7cd4e6112f;
var $c8bc378300208597$export$d35bc1e505d1ebbf;
var $c8bc378300208597$export$e7dc768d35940237;
var $c8bc378300208597$export$b8813cd5d7824ce7;
var $c8bc378300208597$export$53aa811f86efbfad;
var $c8bc378300208597$export$1e0fb04f31d3c22a;
var $c8bc378300208597$export$4ad4709e9aee8e02;
var $c8bc378300208597$export$1f55b27adf96a61e;
var $c8bc378300208597$export$a2d35348c0cc9bca;
var $c8bc378300208597$export$d9ba3f873d0eaec6;
var $c8bc378300208597$export$eb0d5352b2a55473;
var $c8bc378300208597$export$7150e72e440612bf;
var $c8bc378300208597$export$7c9efe74c4b77579;
var $c8bc378300208597$export$d9b13621bc5dd4d1;
var $c8bc378300208597$export$1bf50375883926;
var $c8bc378300208597$export$5aa06adeaa92dbca;
var $c8bc378300208597$export$b01cb9680b0b8d22;
var $c8bc378300208597$export$cbd30c7809acdd81;
var $c8bc378300208597$export$68b45d42c77fe1cf;
var $c8bc378300208597$export$1b66ee5cd8394caf;
var $c8bc378300208597$export$6b3b7792abd9e5f4;
var $c8bc378300208597$export$f80410436c6352c8;
var $c8bc378300208597$export$2a6a235e0baff403;
var $c8bc378300208597$export$927a243805a411ca;
var $c8bc378300208597$export$ce8c93b50eec2d4f;
var $c8bc378300208597$export$8afa006321ee3a8f;
var $c8bc378300208597$export$60162f05499b7560;
var $c8bc378300208597$export$f9ed481cac4d935a;
var $c8bc378300208597$export$659c081622dc13b0;
var $c8bc378300208597$export$58dfda3903289699;
var $c8bc378300208597$export$26480389595d4515;
var $c8bc378300208597$export$6498e436231eea33;
var $c8bc378300208597$export$f979f76ca45242f2;
var $c8bc378300208597$export$9dc11c0e0825a249;
var $c8bc378300208597$export$f89865e352230a35;
var $c8bc378300208597$export$867c05a16e5d2a7c;
var $c8bc378300208597$export$61b66671605f1b1a;
var $c8bc378300208597$export$41a33ae9818c51be;
var $c8bc378300208597$export$dc2b0b08d47555;
var $c8bc378300208597$export$5d2adfeb248e654a;
var $c8bc378300208597$export$84286d36e5c00c55;
var $c8bc378300208597$export$fcabaf52adc14efe;
var $c8bc378300208597$export$2fb3c7c2d5c0d720;
var $c8bc378300208597$export$256b97e812db1642;
var $c8bc378300208597$export$4784f0c2ec8bbf7b;
var $c8bc378300208597$export$f385e7456ddfee84;
var $c8bc378300208597$export$aaaf83100348d389;
var $c8bc378300208597$export$39b42a8a24290c17;
var $c8bc378300208597$export$c29dcfa80dab8c88;
var $c8bc378300208597$export$38c6ebb7f58dedaa;
var $c8bc378300208597$export$8a26470723c97997;
var $c8bc378300208597$export$3b4577d06eb82ec0;
var $c8bc378300208597$export$e77609e6ef08541e;
$c8bc378300208597$export$bd7f6383fb9d6b48 = `C64cMW_actionmenu`;
$c8bc378300208597$export$e324594224ef24da = `C64cMW_actions`;
$c8bc378300208597$export$4aebd143f31d0c0d = `C64cMW_badge`;
$c8bc378300208597$export$b07e517d22efa1ca = `C64cMW_checkbox`;
$c8bc378300208597$export$980c1089c0604ea3 = `C64cMW_chevron`;
$c8bc378300208597$export$a7db06668cad9adb = `C64cMW_content`;
$c8bc378300208597$export$6d59db4903f20f7d = `C64cMW_description`;
$c8bc378300208597$export$7b34ab0dc3a6dc3b = `C64cMW_draghandle`;
$c8bc378300208597$export$f39a09f249340e2a = `C64cMW_focus-ring`;
$c8bc378300208597$export$27f4d14f7fc5b94 = `C64cMW_has-checkbox`;
$c8bc378300208597$export$20fd0f7cd4e6112f = `C64cMW_is-active`;
$c8bc378300208597$export$d35bc1e505d1ebbf = `C64cMW_is-disabled`;
$c8bc378300208597$export$e7dc768d35940237 = `C64cMW_is-focused`;
$c8bc378300208597$export$b8813cd5d7824ce7 = `C64cMW_is-hovered`;
$c8bc378300208597$export$53aa811f86efbfad = `C64cMW_is-next-selected`;
$c8bc378300208597$export$1e0fb04f31d3c22a = `C64cMW_is-selected`;
$c8bc378300208597$export$4ad4709e9aee8e02 = `C64cMW_react-spectrum-ListView`;
$c8bc378300208597$export$1f55b27adf96a61e = `C64cMW_react-spectrum-ListView--compact`;
$c8bc378300208597$export$a2d35348c0cc9bca = `C64cMW_react-spectrum-ListView--draggable`;
$c8bc378300208597$export$d9ba3f873d0eaec6 = `C64cMW_react-spectrum-ListView--dropTarget`;
$c8bc378300208597$export$eb0d5352b2a55473 = `C64cMW_react-spectrum-ListView--emphasized`;
$c8bc378300208597$export$7150e72e440612bf = `C64cMW_react-spectrum-ListView--hasAnyChildren`;
$c8bc378300208597$export$7c9efe74c4b77579 = `C64cMW_react-spectrum-ListView--hasDescription`;
$c8bc378300208597$export$d9b13621bc5dd4d1 = `C64cMW_react-spectrum-ListView--isHorizontalScrollbarVisible`;
$c8bc378300208597$export$1bf50375883926 = `C64cMW_react-spectrum-ListView--isVerticalScrollbarVisible`;
$c8bc378300208597$export$5aa06adeaa92dbca = `C64cMW_react-spectrum-ListView--loadingMore`;
$c8bc378300208597$export$b01cb9680b0b8d22 = `C64cMW_react-spectrum-ListView--quiet`;
$c8bc378300208597$export$cbd30c7809acdd81 = `C64cMW_react-spectrum-ListView--spacious`;
$c8bc378300208597$export$68b45d42c77fe1cf = `C64cMW_react-spectrum-ListView--wrap`;
$c8bc378300208597$export$1b66ee5cd8394caf = `C64cMW_react-spectrum-ListView-centeredWrapper`;
$c8bc378300208597$export$6b3b7792abd9e5f4 = `C64cMW_react-spectrum-ListView-centeredWrapper--loadingMore`;
$c8bc378300208597$export$f80410436c6352c8 = `C64cMW_react-spectrum-ListView-row`;
$c8bc378300208597$export$2a6a235e0baff403 = `C64cMW_react-spectrum-ListViewInsertionIndicator`;
$c8bc378300208597$export$927a243805a411ca = `C64cMW_react-spectrum-ListViewInsertionIndicator--dropTarget`;
$c8bc378300208597$export$ce8c93b50eec2d4f = `C64cMW_react-spectrum-ListViewItem`;
$c8bc378300208597$export$8afa006321ee3a8f = `C64cMW_react-spectrum-ListViewItem--dropTarget`;
$c8bc378300208597$export$60162f05499b7560 = `C64cMW_react-spectrum-ListViewItem--firstRow`;
$c8bc378300208597$export$f9ed481cac4d935a = `C64cMW_react-spectrum-ListViewItem--isFlushBottom`;
$c8bc378300208597$export$659c081622dc13b0 = `C64cMW_react-spectrum-ListViewItem--lastRow`;
$c8bc378300208597$export$58dfda3903289699 = `C64cMW_react-spectrum-ListViewItem-actionmenu`;
$c8bc378300208597$export$26480389595d4515 = `C64cMW_react-spectrum-ListViewItem-actions`;
$c8bc378300208597$export$6498e436231eea33 = `C64cMW_react-spectrum-ListViewItem-badge`;
$c8bc378300208597$export$f979f76ca45242f2 = `C64cMW_react-spectrum-ListViewItem-checkbox`;
$c8bc378300208597$export$9dc11c0e0825a249 = `C64cMW_react-spectrum-ListViewItem-checkbox--enter`;
$c8bc378300208597$export$f89865e352230a35 = `C64cMW_react-spectrum-ListViewItem-checkbox--enterActive`;
$c8bc378300208597$export$867c05a16e5d2a7c = `C64cMW_react-spectrum-ListViewItem-checkbox--exit`;
$c8bc378300208597$export$61b66671605f1b1a = `C64cMW_react-spectrum-ListViewItem-checkbox--exitActive`;
$c8bc378300208597$export$41a33ae9818c51be = `C64cMW_react-spectrum-ListViewItem-checkboxWrapper`;
$c8bc378300208597$export$dc2b0b08d47555 = `C64cMW_react-spectrum-ListViewItem-content`;
$c8bc378300208597$export$5d2adfeb248e654a = `C64cMW_react-spectrum-ListViewItem-description`;
$c8bc378300208597$export$84286d36e5c00c55 = `C64cMW_react-spectrum-ListViewItem-dragPreview`;
$c8bc378300208597$export$fcabaf52adc14efe = `C64cMW_react-spectrum-ListViewItem-dragPreview--compact`;
$c8bc378300208597$export$2fb3c7c2d5c0d720 = `C64cMW_react-spectrum-ListViewItem-dragPreview--multiple`;
$c8bc378300208597$export$256b97e812db1642 = `C64cMW_react-spectrum-ListViewItem-dragPreview--spacious`;
$c8bc378300208597$export$4784f0c2ec8bbf7b = `C64cMW_react-spectrum-ListViewItem-draghandle-button`;
$c8bc378300208597$export$f385e7456ddfee84 = `C64cMW_react-spectrum-ListViewItem-draghandle-container`;
$c8bc378300208597$export$aaaf83100348d389 = `C64cMW_react-spectrum-ListViewItem-grid`;
$c8bc378300208597$export$39b42a8a24290c17 = `C64cMW_react-spectrum-ListViewItem-parentIndicator`;
$c8bc378300208597$export$c29dcfa80dab8c88 = `C64cMW_react-spectrum-ListViewItem-parentIndicator--hasChildItems`;
$c8bc378300208597$export$38c6ebb7f58dedaa = `C64cMW_react-spectrum-ListViewItem-thumbnail`;
$c8bc378300208597$export$8a26470723c97997 = `C64cMW_round-bottoms`;
$c8bc378300208597$export$3b4577d06eb82ec0 = `C64cMW_round-tops`;
$c8bc378300208597$export$e77609e6ef08541e = `C64cMW_thumbnail`;





function $0a834ddbc989a3e3$export$2e2bcd8739ae039(props) {
    let { dropState: dropState, dragAndDropHooks: dragAndDropHooks } = (0, $fWI2a$useContext)((0, $f85fb77f9d4cbc6c$export$870039b0abfe3de0));
    const { target: target, isPresentationOnly: isPresentationOnly } = props;
    let ref = (0, $fWI2a$useRef)();
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator(props, dropState, ref);
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $fWI2a$useVisuallyHidden)();
    let isDropTarget = dropState.isDropTarget(target);
    if (!isDropTarget && dropIndicatorProps["aria-hidden"]) return null;
    return /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps["aria-hidden"]
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        role: "gridcell",
        "aria-selected": "false",
        className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListViewInsertionIndicator", {
            "react-spectrum-ListViewInsertionIndicator--dropTarget": isDropTarget
        })
    }, !isPresentationOnly && /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        ...visuallyHiddenProps,
        role: "button",
        ...dropIndicatorProps,
        ref: ref
    })));
}


var $ae327a38dafd8ae2$exports = {};


































$ae327a38dafd8ae2$exports = {
    "ar-AE": $a8be062dce739a7d$exports,
    "bg-BG": $ebee5026543c5714$exports,
    "cs-CZ": $ec2dceffa5b2d037$exports,
    "da-DK": $e49254d6635b1a35$exports,
    "de-DE": $c25de0af306f5a70$exports,
    "el-GR": $1aedf9be304cf865$exports,
    "en-US": $653a020875a8fea3$exports,
    "es-ES": $24fba9a73a2aba1c$exports,
    "et-EE": $0994847e6d27862d$exports,
    "fi-FI": $084b0bc3d6e3d2a1$exports,
    "fr-FR": $3c567f46aec1205a$exports,
    "he-IL": $e5a01f2966bf0b9a$exports,
    "hr-HR": $223cc8f6bb1d23dd$exports,
    "hu-HU": $c577fab129d558ed$exports,
    "it-IT": $a6e5541030242f67$exports,
    "ja-JP": $cce2c1ef24294875$exports,
    "ko-KR": $01de2837b3b6f15a$exports,
    "lt-LT": $5394997707852da6$exports,
    "lv-LV": $3f13373950080912$exports,
    "nb-NO": $041a8a2796a43a59$exports,
    "nl-NL": $ef957358eac33027$exports,
    "pl-PL": $09a0ddc6d157c4fe$exports,
    "pt-BR": $5e4abb947f3a742c$exports,
    "pt-PT": $18faaf4c3266f94f$exports,
    "ro-RO": $b3819b882aaf2b05$exports,
    "ru-RU": $bbb9489f20c7c6c0$exports,
    "sk-SK": $12e7bbe01da0d5ad$exports,
    "sl-SI": $5582a397c204a97c$exports,
    "sr-SP": $aa6549babfa00736$exports,
    "sv-SE": $13248984d46c1e4f$exports,
    "tr-TR": $7af10700b945dced$exports,
    "uk-UA": $00a02749c6fa842b$exports,
    "zh-CN": $e3b2b7128a600af6$exports,
    "zh-TW": $8751d9847e270a98$exports
};





/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


















function $d7c07ca2efc5ba02$export$c6bde0c04b033c0e(props) {
    var _layout_getContentSize, _layout_virtualizer;
    let { item: item, isEmphasized: isEmphasized } = props;
    let { state: state, dragState: dragState, dropState: dropState, isListDraggable: isListDraggable, isListDroppable: isListDroppable, layout: layout, dragAndDropHooks: dragAndDropHooks, loadingState: loadingState } = (0, $fWI2a$useContext)((0, $f85fb77f9d4cbc6c$export$870039b0abfe3de0));
    let { direction: direction } = (0, $fWI2a$useLocale)();
    let rowRef = (0, $fWI2a$useRef)();
    let checkboxWrapperRef = (0, $fWI2a$useRef)();
    let { isFocusVisible: isFocusVisibleWithin, focusProps: focusWithinProps } = (0, $fWI2a$useFocusRing)({
        within: true
    });
    let { isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $fWI2a$useFocusRing)();
    let { rowProps: rowProps, gridCellProps: gridCellProps, isPressed: isPressed, descriptionProps: descriptionProps, isSelected: isSelected, isDisabled: isDisabled, allowsSelection: allowsSelection, hasAction: hasAction } = (0, $fWI2a$useGridListItem)({
        node: item,
        isVirtualized: true,
        shouldSelectOnPressUp: isListDraggable
    }, state, rowRef);
    let isDroppable = isListDroppable && !isDisabled;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $fWI2a$useHover)({
        isDisabled: !allowsSelection && !hasAction
    });
    let { checkboxProps: checkboxProps } = (0, $fWI2a$useGridListSelectionCheckbox)({
        key: item.key
    }, state);
    let hasDescription = (0, $fWI2a$useHasChild)(`.${(0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-description"]}`, rowRef);
    let draggableItem;
    if (isListDraggable) {
        // eslint-disable-next-line react-hooks/rules-of-hooks
        draggableItem = dragAndDropHooks.useDraggableItem({
            key: item.key,
            hasDragButton: true
        }, dragState);
        if (isDisabled) draggableItem = null;
    }
    let droppableItem;
    let isDropTarget;
    let dropIndicator;
    let dropIndicatorRef = (0, $fWI2a$useRef)();
    if (isListDroppable) {
        let target = {
            type: "item",
            key: item.key,
            dropPosition: "on"
        };
        isDropTarget = dropState.isDropTarget(target);
        // eslint-disable-next-line react-hooks/rules-of-hooks
        dropIndicator = dragAndDropHooks.useDropIndicator({
            target: target
        }, dropState, dropIndicatorRef);
    }
    let dragButtonRef = (0, $fWI2a$react).useRef();
    let { buttonProps: buttonProps } = (0, $fWI2a$useButton)({
        ...draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragButtonProps,
        elementType: "div"
    }, dragButtonRef);
    let chevron = direction === "ltr" ? /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$spectrumiconsuiChevronRightMedium), {
        "aria-hidden": "true",
        UNSAFE_className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListViewItem-parentIndicator", {
            "react-spectrum-ListViewItem-parentIndicator--hasChildItems": item.props.hasChildItems,
            "is-disabled": !hasAction
        })
    }) : /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$spectrumiconsuiChevronLeftMedium), {
        "aria-hidden": "true",
        UNSAFE_className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListViewItem-parentIndicator", {
            "react-spectrum-ListViewItem-parentIndicator--hasChildItems": item.props.hasChildItems,
            "is-disabled": !hasAction
        })
    });
    let showCheckbox = state.selectionManager.selectionMode !== "none" && state.selectionManager.selectionBehavior === "toggle";
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $fWI2a$useVisuallyHidden)();
    let dropProps = isDroppable ? droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps : {
        "aria-hidden": droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps["aria-hidden"]
    };
    const mergedProps = (0, $fWI2a$mergeProps)(rowProps, draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragProps, dropProps, hoverProps, focusWithinProps, focusProps, // Remove tab index from list row if performing a screenreader drag. This prevents TalkBack from focusing the row,
    // allowing for single swipe navigation between row drop indicator
    (dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.isVirtualDragging()) && {
        tabIndex: null
    });
    let isFirstRow = item.prevKey == null;
    let isLastRow = item.nextKey == null;
    // Figure out if the ListView content is equal or greater in height to the container. If so, we'll need to round the bottom
    // border corners of the last row when selected and we can get rid of the bottom border if it isn't selected to avoid border overlap
    // with bottom border
    let isFlushWithContainerBottom = false;
    if (isLastRow && loadingState !== "loadingMore") {
        if (((_layout_getContentSize = layout.getContentSize()) === null || _layout_getContentSize === void 0 ? void 0 : _layout_getContentSize.height) >= ((_layout_virtualizer = layout.virtualizer) === null || _layout_virtualizer === void 0 ? void 0 : _layout_virtualizer.getVisibleRect().height)) isFlushWithContainerBottom = true;
    }
    // previous item isn't selected
    // and the previous item isn't focused or, if it is focused, then if focus globally isn't visible or just focus isn't in the listview
    let roundTops = !state.selectionManager.isSelected(item.prevKey) && (state.selectionManager.focusedKey !== item.prevKey || !((0, $fWI2a$isFocusVisible)() && state.selectionManager.isFocused));
    let roundBottoms = !state.selectionManager.isSelected(item.nextKey) && (state.selectionManager.focusedKey !== item.nextKey || !((0, $fWI2a$isFocusVisible)() && state.selectionManager.isFocused));
    let content = typeof item.rendered === "string" ? /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$Text), null, item.rendered) : item.rendered;
    if (isDisabled) content = /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$Provider), {
        isDisabled: true
    }, content);
    return /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        ...mergedProps,
        className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListView-row", {
            "focus-ring": isFocusVisible,
            "round-tops": roundTops || isHovered && !isSelected && state.selectionManager.focusedKey !== item.key,
            "round-bottoms": roundBottoms || isHovered && !isSelected && state.selectionManager.focusedKey !== item.key
        }),
        ref: rowRef
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        // TODO: refactor the css here now that we are focusing the row?
        className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListViewItem", {
            "is-active": isPressed,
            "is-focused": isFocusVisibleWithin,
            "focus-ring": isFocusVisible,
            "is-hovered": isHovered,
            "is-selected": isSelected,
            "is-disabled": isDisabled,
            "is-prev-selected": state.selectionManager.isSelected(item.prevKey),
            "is-next-selected": state.selectionManager.isSelected(item.nextKey),
            "react-spectrum-ListViewItem--highlightSelection": state.selectionManager.selectionBehavior === "replace" && (isSelected || state.selectionManager.isSelected(item.nextKey)),
            "react-spectrum-ListViewItem--dropTarget": !!isDropTarget,
            "react-spectrum-ListViewItem--firstRow": isFirstRow,
            "react-spectrum-ListViewItem--lastRow": isLastRow,
            "react-spectrum-ListViewItem--isFlushBottom": isFlushWithContainerBottom,
            "react-spectrum-ListViewItem--hasDescription": hasDescription
        }),
        ...gridCellProps
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$Grid), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-grid"]
    }, isListDraggable && /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-draghandle-container"]
    }, !isDisabled && /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$FocusRing), {
        focusRingClass: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        ...buttonProps,
        className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListViewItem-draghandle-button"),
        style: !isFocusVisibleWithin ? {
            ...visuallyHiddenProps.style
        } : {},
        ref: dragButtonRef,
        draggable: "true"
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$spectrumiconsuiListGripper), null)))), isListDroppable && !(dropIndicator === null || dropIndicator === void 0 ? void 0 : dropIndicator.isHidden) && /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicator === null || dropIndicator === void 0 ? void 0 : dropIndicator.dropIndicatorProps,
        ref: dropIndicatorRef
    }), /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$CSSTransition), {
        in: showCheckbox,
        unmountOnExit: true,
        classNames: {
            enter: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-checkbox--enter"],
            enterActive: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-checkbox--enterActive"],
            exit: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-checkbox--exit"],
            exitActive: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-checkbox--exitActive"]
        },
        timeout: 160,
        nodeRef: checkboxWrapperRef
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        ref: checkboxWrapperRef,
        className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-checkboxWrapper"]
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$Checkbox), {
        ...checkboxProps,
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-checkbox"],
        isEmphasized: isEmphasized
    }))), /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-content"]
            },
            description: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-description"],
                ...descriptionProps
            },
            illustration: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-thumbnail"]
            },
            image: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-thumbnail"]
            },
            actionButton: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-actions"],
                isQuiet: true
            },
            actionGroup: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-actions"],
                isQuiet: true,
                density: "compact"
            },
            actionMenu: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-actionmenu"],
                isQuiet: true
            }
        }
    }, content, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$ClearSlots), null, chevron)))));
}







function $41a60729487a82d7$export$2e2bcd8739ae039() {
    let { dropState: dropState, dragAndDropHooks: dragAndDropHooks } = (0, $fWI2a$useContext)((0, $f85fb77f9d4cbc6c$export$870039b0abfe3de0));
    let ref = (0, $fWI2a$useRef)();
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator({
        target: {
            type: "root"
        }
    }, dropState, ref);
    let isDropTarget = dropState.isDropTarget({
        type: "root"
    });
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $fWI2a$useVisuallyHidden)();
    if (!isDropTarget && dropIndicatorProps["aria-hidden"]) return null;
    return /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps["aria-hidden"]
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        role: "gridcell",
        "aria-selected": "false"
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicatorProps,
        ref: ref
    })));
}


/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $cd61e55c47e3c0f5$export$905ab40ac2179daa(props) {
    let { item: item, itemCount: itemCount, itemHeight: itemHeight, density: density } = props;
    let isDraggingMultiple = itemCount > 1;
    return /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        style: {
            height: itemHeight
        },
        className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListViewItem", "react-spectrum-ListViewItem-dragPreview", {
            "react-spectrum-ListViewItem-dragPreview--multiple": isDraggingMultiple
        }, `react-spectrum-ListViewItem-dragPreview--${density}`)
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$Grid), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-grid"]
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$SlotProvider), {
        slots: {
            text: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-content"]
            },
            description: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-description"]
            },
            illustration: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-thumbnail"]
            },
            image: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-thumbnail"]
            },
            actionButton: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-actions"],
                isQuiet: true
            },
            actionGroup: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-actions"],
                isQuiet: true,
                density: "compact"
            },
            actionMenu: {
                UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports)))["react-spectrum-ListViewItem-actionmenu"],
                isQuiet: true
            }
        }
    }, typeof item.rendered === "string" ? /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$Text), null, item.rendered) : item.rendered, isDraggingMultiple && /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListViewItem-badge")
    }, itemCount))));
}





const $f85fb77f9d4cbc6c$export$870039b0abfe3de0 = /*#__PURE__*/ (0, $fWI2a$react).createContext(null);
const $f85fb77f9d4cbc6c$var$ROW_HEIGHTS = {
    compact: {
        medium: 32,
        large: 40
    },
    regular: {
        medium: 40,
        large: 50
    },
    spacious: {
        medium: 48,
        large: 60
    }
};
function $f85fb77f9d4cbc6c$var$useListLayout(state, density, overflowMode) {
    let { scale: scale } = (0, $fWI2a$useProvider)();
    let collator = (0, $fWI2a$useCollator)({
        usage: "search",
        sensitivity: "base"
    });
    let isEmpty = state.collection.size === 0;
    let layout = (0, $fWI2a$useMemo)(()=>new (0, $fWI2a$ListLayout)({
            estimatedRowHeight: $f85fb77f9d4cbc6c$var$ROW_HEIGHTS[density][scale],
            padding: 0,
            collator: collator,
            loaderHeight: isEmpty ? null : $f85fb77f9d4cbc6c$var$ROW_HEIGHTS[density][scale]
        }), [
        collator,
        scale,
        density,
        isEmpty,
        overflowMode
    ]);
    layout.collection = state.collection;
    layout.disabledKeys = state.disabledKeys;
    return layout;
}
function $f85fb77f9d4cbc6c$var$ListView(props, ref) {
    var _dropState_target;
    let { density: density = "regular", loadingState: loadingState, onLoadMore: onLoadMore, isQuiet: isQuiet, overflowMode: overflowMode = "truncate", onAction: onAction, dragAndDropHooks: dragAndDropHooks, ...otherProps } = props;
    let isListDraggable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDraggableCollectionState);
    let isListDroppable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDroppableCollectionState);
    let dragHooksProvided = (0, $fWI2a$useRef)(isListDraggable);
    let dropHooksProvided = (0, $fWI2a$useRef)(isListDroppable);
    (0, $fWI2a$useEffect)(()=>{
        if (dragHooksProvided.current !== isListDraggable) console.warn("Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.");
        if (dropHooksProvided.current !== isListDroppable) console.warn("Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.");
    }, [
        isListDraggable,
        isListDroppable
    ]);
    let domRef = (0, $fWI2a$useDOMRef)(ref);
    let state = (0, $fWI2a$useListState)({
        ...props,
        selectionBehavior: props.selectionStyle === "highlight" ? "replace" : "toggle"
    });
    let { collection: collection, selectionManager: selectionManager } = state;
    let stringFormatter = (0, $fWI2a$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($ae327a38dafd8ae2$exports))), "@react-spectrum/list");
    let isLoading = loadingState === "loading" || loadingState === "loadingMore";
    let { styleProps: styleProps } = (0, $fWI2a$useStyleProps)(props);
    let dragState;
    let preview = (0, $fWI2a$useRef)(null);
    if (isListDraggable) {
        dragState = dragAndDropHooks.useDraggableCollectionState({
            collection: collection,
            selectionManager: selectionManager,
            preview: preview
        });
        dragAndDropHooks.useDraggableCollection({}, dragState, domRef);
    }
    let layout = $f85fb77f9d4cbc6c$var$useListLayout(state, props.density || "regular", overflowMode);
    // !!0 is false, so we can cast size or undefined and they'll be falsy
    layout.allowDisabledKeyFocus = state.selectionManager.disabledBehavior === "selection" || !!(dragState === null || dragState === void 0 ? void 0 : dragState.draggingKeys.size);
    let DragPreview = dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.DragPreview;
    let dropState;
    let droppableCollection;
    let isRootDropTarget;
    if (isListDroppable) {
        dropState = dragAndDropHooks.useDroppableCollectionState({
            collection: collection,
            selectionManager: selectionManager
        });
        droppableCollection = dragAndDropHooks.useDroppableCollection({
            keyboardDelegate: layout,
            dropTargetDelegate: layout
        }, dropState, domRef);
        isRootDropTarget = dropState.isDropTarget({
            type: "root"
        });
    }
    let { gridProps: gridProps } = (0, $fWI2a$useGridList)({
        ...props,
        isVirtualized: true,
        keyboardDelegate: layout,
        onAction: onAction
    }, state, domRef);
    // Sync loading state into the layout.
    layout.isLoading = isLoading;
    let focusedKey = selectionManager.focusedKey;
    if ((dropState === null || dropState === void 0 ? void 0 : (_dropState_target = dropState.target) === null || _dropState_target === void 0 ? void 0 : _dropState_target.type) === "item") focusedKey = dropState.target.key;
    // wait for layout to get accurate measurements
    let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = (0, $fWI2a$useState)(false);
    let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = (0, $fWI2a$useState)(false);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    (0, $fWI2a$useLayoutEffect)(()=>{
        if (domRef.current) {
            // 2 is the width of the border which is not part of the box size
            setVerticalScollbarVisible(domRef.current.clientWidth + 2 < domRef.current.offsetWidth);
            setHorizontalScollbarVisible(domRef.current.clientHeight + 2 < domRef.current.offsetHeight);
        }
    });
    let hasAnyChildren = (0, $fWI2a$useMemo)(()=>[
            ...collection
        ].some((item)=>item.hasChildNodes), [
        collection
    ]);
    return /*#__PURE__*/ (0, $fWI2a$react).createElement($f85fb77f9d4cbc6c$export$870039b0abfe3de0.Provider, {
        value: {
            state: state,
            dragState: dragState,
            dropState: dropState,
            dragAndDropHooks: dragAndDropHooks,
            onAction: onAction,
            isListDraggable: isListDraggable,
            isListDroppable: isListDroppable,
            layout: layout,
            loadingState: loadingState
        }
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$FocusScope), null, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$FocusRing), {
        focusRingClass: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$Virtualizer), {
        ...(0, $fWI2a$mergeProps)(isListDroppable && (droppableCollection === null || droppableCollection === void 0 ? void 0 : droppableCollection.collectionProps), gridProps),
        ...(0, $fWI2a$filterDOMProps)(otherProps),
        ...gridProps,
        ...styleProps,
        isLoading: isLoading,
        onLoadMore: onLoadMore,
        ref: domRef,
        focusedKey: focusedKey,
        scrollDirection: "vertical",
        className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListView", `react-spectrum-ListView--${density}`, "react-spectrum-ListView--emphasized", {
            "react-spectrum-ListView--quiet": isQuiet,
            "react-spectrum-ListView--loadingMore": loadingState === "loadingMore",
            "react-spectrum-ListView--draggable": !!isListDraggable,
            "react-spectrum-ListView--dropTarget": !!isRootDropTarget,
            "react-spectrum-ListView--isVerticalScrollbarVisible": isVerticalScrollbarVisible,
            "react-spectrum-ListView--isHorizontalScrollbarVisible": isHorizontalScrollbarVisible,
            "react-spectrum-ListView--hasAnyChildren": hasAnyChildren,
            "react-spectrum-ListView--wrap": overflowMode === "wrap"
        }, styleProps.className),
        layout: layout,
        collection: collection,
        transitionDuration: isLoading ? 160 : 220
    }, (type, item)=>{
        if (type === "item") return /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$react).Fragment, null, isListDroppable && collection.getKeyBefore(item.key) == null && /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $41a60729487a82d7$export$2e2bcd8739ae039), {
            key: "root"
        }), isListDroppable && /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $0a834ddbc989a3e3$export$2e2bcd8739ae039), {
            key: `${item.key}-before`,
            target: {
                key: item.key,
                type: "item",
                dropPosition: "before"
            }
        }), /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $d7c07ca2efc5ba02$export$c6bde0c04b033c0e), {
            item: item,
            isEmphasized: true,
            hasActions: !!onAction
        }), isListDroppable && /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $0a834ddbc989a3e3$export$2e2bcd8739ae039), {
            key: `${item.key}-after`,
            target: {
                key: item.key,
                type: "item",
                dropPosition: "after"
            },
            isPresentationOnly: collection.getKeyAfter(item.key) != null
        }));
        else if (type === "loader") return /*#__PURE__*/ (0, $fWI2a$react).createElement($f85fb77f9d4cbc6c$var$CenteredWrapper, null, /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $fWI2a$ProgressCircle), {
            isIndeterminate: true,
            "aria-label": collection.size > 0 ? stringFormatter.format("loadingMore") : stringFormatter.format("loading")
        }));
        else if (type === "placeholder") {
            let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
            if (emptyState == null) return null;
            return /*#__PURE__*/ (0, $fWI2a$react).createElement($f85fb77f9d4cbc6c$var$CenteredWrapper, null, emptyState);
        }
    }))), DragPreview && isListDraggable && /*#__PURE__*/ (0, $fWI2a$react).createElement(DragPreview, {
        ref: preview
    }, ()=>{
        if (dragAndDropHooks.renderPreview) return dragAndDropHooks.renderPreview(dragState.draggingKeys, dragState.draggedKey);
        let item = state.collection.getItem(dragState.draggedKey);
        let itemCount = dragState.draggingKeys.size;
        let itemHeight = layout.getLayoutInfo(dragState.draggedKey).rect.height;
        return /*#__PURE__*/ (0, $fWI2a$react).createElement((0, $cd61e55c47e3c0f5$export$905ab40ac2179daa), {
            item: item,
            itemCount: itemCount,
            itemHeight: itemHeight,
            density: density
        });
    }));
}
function $f85fb77f9d4cbc6c$var$CenteredWrapper({ children: children }) {
    let { state: state } = (0, $fWI2a$useContext)($f85fb77f9d4cbc6c$export$870039b0abfe3de0);
    return /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        role: "row",
        "aria-rowindex": state.collection.size + 1,
        className: (0, $fWI2a$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c8bc378300208597$exports))), "react-spectrum-ListView-centeredWrapper", {
            "react-spectrum-ListView-centeredWrapper--loadingMore": state.collection.size > 0
        })
    }, /*#__PURE__*/ (0, $fWI2a$react).createElement("div", {
        role: "gridcell"
    }, children));
}
/**
 * A ListView displays a list of interactive items, and allows a user to navigate, select, or perform an action.
 */ const $f85fb77f9d4cbc6c$export$84d0dd190d551cd1 = /*#__PURE__*/ (0, $fWI2a$react).forwardRef($f85fb77f9d4cbc6c$var$ListView);





export {$f85fb77f9d4cbc6c$export$84d0dd190d551cd1 as ListView, $75e4a7b01832b12d$re_export$Item as Item};
