// @aemforms/af-react-components/lib/utils/SpectrumMappers.js
"use strict";
/*
 * Copyright 2022 Adobe, Inc.
 *
 * Your access and use of this software is governed by the Adobe Customer Feedback Program Terms and Conditions or other Beta License Agreement signed by your employer and Adobe, Inc.. This software is NOT open source and may not be used without one of the foregoing licenses. Even with a foregoing license, your access and use of this file is limited to the earlier of (a) 180 days, (b) general availability of the product(s) which utilize this software (i.e. AEM Forms), (c) January 1, 2023, (d) Adobe providing notice to you that you may no longer use the software or that your beta trial has otherwise ended.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL ADOBE NOR ITS THIRD PARTY PROVIDERS AND PARTNERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.withErrorMessage = exports.inputTypeConvertor = exports.enumConvertor = exports.enumToChildConvertor = exports.stringConstraintConvertor = exports.fieldConvertor = exports.constraintConvertor = exports.baseConvertor = exports.richTextString = exports.combineConvertors = void 0;
const af_core_1 = require("@aemforms/af-core");
const react_1 = __importDefault(require("react"));
const sanitize_html_1 = __importDefault(require("sanitize-html"));
const af_react_renderer_1 = require("@aemforms/af-react-renderer");
require("../styles.css");
const clsx_1 = __importDefault(require("clsx"));
const FormUtils_1 = require("@aemforms/af-core/lib/utils/FormUtils");
const DEFAULT_ERROR_MESSAGE = 'There is an error in the field';
const combineConvertors = function (...convertors) {
    const newConvertor = (a, b, f) => {
        return convertors.reduce(function (newVal, curr) {
            return Object.assign(Object.assign({}, newVal), curr(a, b, f));
        }, {});
    };
    return newConvertor;
};
exports.combineConvertors = combineConvertors;
const richTextString = (stringMsg = '') => {
    const htmlProp = { __html: (0, sanitize_html_1.default)(stringMsg) };
    return (react_1.default.createElement("div", { dangerouslySetInnerHTML: htmlProp }));
};
exports.richTextString = richTextString;
const baseConvertor = (a, b, f) => {
    var _a, _b, _c, _d, _e, _f, _g;
    let localizedDescription = f('description');
    const layout = ((_a = a === null || a === void 0 ? void 0 : a.properties) === null || _a === void 0 ? void 0 : _a['afs:layout']) || {};
    return Object.assign(Object.assign({}, layout), { isHidden: a.visible === false, name: a.name, isDisabled: a.enabled === false, label: ((_b = a.label) === null || _b === void 0 ? void 0 : _b.visible) === false ? '' : (((_c = a.label) === null || _c === void 0 ? void 0 : _c.richText) === true ? (0, exports.richTextString)((_d = a.label) === null || _d === void 0 ? void 0 : _d.value) : (_e = a.label) === null || _e === void 0 ? void 0 : _e.value), description: (localizedDescription && localizedDescription.length > 0) ? (0, exports.richTextString)(localizedDescription) : null, 'aria-label': ((_f = a.label) === null || _f === void 0 ? void 0 : _f.visible) === false ? (_g = a.label) === null || _g === void 0 ? void 0 : _g.value : undefined });
};
exports.baseConvertor = baseConvertor;
const constraintConvertor = (a) => {
    // if there are no constraints and type is string, valid would anyways always be true, hence validationState would be optional
    const optionalValidation = !(0, af_core_1.checkIfConstraintsArePresent)(a) && a.type === 'string';
    return Object.assign(Object.assign({}, (a.required && {
        isRequired: true,
        necessityIndicator: 'icon'
    })), { validationState: a.valid === false ? 'invalid' : ((a.valid === undefined || (0, FormUtils_1.isEmpty)(a.value) || optionalValidation) ? undefined : 'valid') });
};
exports.constraintConvertor = constraintConvertor;
const fieldConvertor = (a, b, f) => {
    const i18n = (0, af_react_renderer_1.useFormIntl)();
    const formatedMessage = i18n.formatMessage({ id: 'defaultErrorMessage', defaultMessage: DEFAULT_ERROR_MESSAGE });
    const errorMessage = a.errorMessage === '' && a.valid === false ? formatedMessage : a.errorMessage;
    // if there are no constraints and type is string, valid would anyways always be true, hence validationState would be optional
    const optionalValidation = !(0, af_core_1.checkIfConstraintsArePresent)(a) && a.type === 'string';
    return {
        placeholder: f('placeholder'),
        value: a.value == null ? '' : a.value,
        validationState: a.valid === false ? 'invalid' : ((a.valid === undefined || (0, FormUtils_1.isEmpty)(a.value) || optionalValidation) ? undefined : 'valid'),
        onChange: b.dispatchChange,
        onBlur: b.dispatchBlur,
        isReadOnly: a.readOnly === true,
        errorMessage
    };
};
exports.fieldConvertor = fieldConvertor;
const stringConstraintConvertor = (a) => {
    return {
        minLength: a.minLength,
        maxLength: a.maxLength,
        pattern: a.pattern
    };
};
exports.stringConstraintConvertor = stringConstraintConvertor;
const enumToChildConvertor = (Component) => {
    return (0, exports.enumConvertor)('children', (text, value) => {
        return react_1.default.createElement(Component, { key: value, value: value }, text + '');
    });
};
exports.enumToChildConvertor = enumToChildConvertor;
const enumConvertor = (propertyName, callback) => (a, b, f) => {
    const options = a.enum || [];
    const localizedOptions = f('enum');
    const localizedOptionsName = f('enumNames');
    const radio = (option, i) => {
        const value = option;
        const text = (localizedOptionsName && i < localizedOptionsName.length) ? localizedOptionsName[i] : localizedOptions[i];
        return callback(text, value);
    };
    return {
        [propertyName]: options.map(radio)
    };
};
exports.enumConvertor = enumConvertor;
const inputTypeConvertor = (a) => {
    return Object.assign({}, (a.type && {
        type: a.type
    }));
};
exports.inputTypeConvertor = inputTypeConvertor;
const withErrorMessage = (Component) => (props) => {
    const invalid = props.validationState === 'invalid';
    const helpText = invalid ? props.errorMessage || '' : props.description;
    const hasHelpText = (typeof helpText === 'string' && helpText.length > 0) || helpText != null;
    return (react_1.default.createElement("div", { className: (0, clsx_1.default)('formField', invalid && 'formField--invalid') },
        react_1.default.createElement(Component, Object.assign({}, props)),
        hasHelpText ? react_1.default.createElement("div", { className: 'formField__helpText' }, helpText) : null));
};
exports.withErrorMessage = withErrorMessage;
