// @react-spectrum/illustratedmessage/dist/import.mjs
"use client";

import "./main.css";
import {useSlotProps as $RWtz0$useSlotProps, useStyleProps as $RWtz0$useStyleProps, classNames as $RWtz0$classNames, ClearSlots as $RWtz0$ClearSlots, SlotProvider as $RWtz0$SlotProvider} from "@react-spectrum/utils";
import {filterDOMProps as $RWtz0$filterDOMProps} from "@react-aria/utils";
import {Flex as $RWtz0$Flex} from "@react-spectrum/layout";
import $RWtz0$react, {forwardRef as $RWtz0$forwardRef} from "react";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $4c3e55dfb549f378$exports = {};

$parcel$export($4c3e55dfb549f378$exports, "spectrum-IllustratedMessage", () => $4c3e55dfb549f378$export$e22c6753c239ec31, (v) => $4c3e55dfb549f378$export$e22c6753c239ec31 = v);
$parcel$export($4c3e55dfb549f378$exports, "spectrum-IllustratedMessage--cta", () => $4c3e55dfb549f378$export$29744d89c9a76d50, (v) => $4c3e55dfb549f378$export$29744d89c9a76d50 = v);
$parcel$export($4c3e55dfb549f378$exports, "spectrum-IllustratedMessage-description", () => $4c3e55dfb549f378$export$cf9d959111789fe7, (v) => $4c3e55dfb549f378$export$cf9d959111789fe7 = v);
$parcel$export($4c3e55dfb549f378$exports, "spectrum-IllustratedMessage-heading", () => $4c3e55dfb549f378$export$5df23faa5ed344c4, (v) => $4c3e55dfb549f378$export$5df23faa5ed344c4 = v);
var $4c3e55dfb549f378$export$e22c6753c239ec31;
var $4c3e55dfb549f378$export$29744d89c9a76d50;
var $4c3e55dfb549f378$export$cf9d959111789fe7;
var $4c3e55dfb549f378$export$5df23faa5ed344c4;
$4c3e55dfb549f378$export$e22c6753c239ec31 = `uhJ7Ra_spectrum-IllustratedMessage`;
$4c3e55dfb549f378$export$29744d89c9a76d50 = `uhJ7Ra_spectrum-IllustratedMessage--cta`;
$4c3e55dfb549f378$export$cf9d959111789fe7 = `uhJ7Ra_spectrum-IllustratedMessage-description`;
$4c3e55dfb549f378$export$5df23faa5ed344c4 = `uhJ7Ra_spectrum-IllustratedMessage-heading`;


function $ccaf5c86c16f58c0$var$IllustratedMessage(props, ref) {
    props = (0, $RWtz0$useSlotProps)(props, "illustration");
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $RWtz0$useStyleProps)(otherProps);
    let headingClassName = (0, $RWtz0$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($4c3e55dfb549f378$exports))), "spectrum-IllustratedMessage-heading");
    let contentClassName = (0, $RWtz0$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($4c3e55dfb549f378$exports))), "spectrum-IllustratedMessage-description");
    let slots = {
        heading: {
            UNSAFE_className: headingClassName
        },
        content: {
            UNSAFE_className: contentClassName
        }
    };
    return /*#__PURE__*/ (0, $RWtz0$react).createElement((0, $RWtz0$Flex), {
        ...(0, $RWtz0$filterDOMProps)(otherProps),
        UNSAFE_style: styleProps.style,
        isHidden: styleProps.hidden,
        UNSAFE_className: (0, $RWtz0$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($4c3e55dfb549f378$exports))), "spectrum-IllustratedMessage", styleProps.className),
        ref: ref
    }, /*#__PURE__*/ (0, $RWtz0$react).createElement((0, $RWtz0$ClearSlots), null, /*#__PURE__*/ (0, $RWtz0$react).createElement((0, $RWtz0$SlotProvider), {
        slots: slots
    }, children)));
}
/**
 * An IllustratedMessage displays an illustration and a message, usually
 * for an empty state or an error page.
 */ let $ccaf5c86c16f58c0$export$406dbc84c317ece0 = /*#__PURE__*/ (0, $RWtz0$forwardRef)($ccaf5c86c16f58c0$var$IllustratedMessage);




export {$ccaf5c86c16f58c0$export$406dbc84c317ece0 as IllustratedMessage};
