// @react-spectrum/illustratedmessage/dist/main.js
"use client";

require("./main.css");
var $6dAMz$reactspectrumutils = require("@react-spectrum/utils");
var $6dAMz$reactariautils = require("@react-aria/utils");
var $6dAMz$reactspectrumlayout = require("@react-spectrum/layout");
var $6dAMz$react = require("react");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "IllustratedMessage", () => $90e6e9abe7fc12ca$export$406dbc84c317ece0);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $e7750c05f28ceb4f$exports = {};

$parcel$export($e7750c05f28ceb4f$exports, "spectrum-IllustratedMessage", () => $e7750c05f28ceb4f$export$e22c6753c239ec31, (v) => $e7750c05f28ceb4f$export$e22c6753c239ec31 = v);
$parcel$export($e7750c05f28ceb4f$exports, "spectrum-IllustratedMessage--cta", () => $e7750c05f28ceb4f$export$29744d89c9a76d50, (v) => $e7750c05f28ceb4f$export$29744d89c9a76d50 = v);
$parcel$export($e7750c05f28ceb4f$exports, "spectrum-IllustratedMessage-description", () => $e7750c05f28ceb4f$export$cf9d959111789fe7, (v) => $e7750c05f28ceb4f$export$cf9d959111789fe7 = v);
$parcel$export($e7750c05f28ceb4f$exports, "spectrum-IllustratedMessage-heading", () => $e7750c05f28ceb4f$export$5df23faa5ed344c4, (v) => $e7750c05f28ceb4f$export$5df23faa5ed344c4 = v);
var $e7750c05f28ceb4f$export$e22c6753c239ec31;
var $e7750c05f28ceb4f$export$29744d89c9a76d50;
var $e7750c05f28ceb4f$export$cf9d959111789fe7;
var $e7750c05f28ceb4f$export$5df23faa5ed344c4;
$e7750c05f28ceb4f$export$e22c6753c239ec31 = `uhJ7Ra_spectrum-IllustratedMessage`;
$e7750c05f28ceb4f$export$29744d89c9a76d50 = `uhJ7Ra_spectrum-IllustratedMessage--cta`;
$e7750c05f28ceb4f$export$cf9d959111789fe7 = `uhJ7Ra_spectrum-IllustratedMessage-description`;
$e7750c05f28ceb4f$export$5df23faa5ed344c4 = `uhJ7Ra_spectrum-IllustratedMessage-heading`;


function $90e6e9abe7fc12ca$var$IllustratedMessage(props, ref) {
    props = (0, $6dAMz$reactspectrumutils.useSlotProps)(props, "illustration");
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $6dAMz$reactspectrumutils.useStyleProps)(otherProps);
    let headingClassName = (0, $6dAMz$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e7750c05f28ceb4f$exports))), "spectrum-IllustratedMessage-heading");
    let contentClassName = (0, $6dAMz$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e7750c05f28ceb4f$exports))), "spectrum-IllustratedMessage-description");
    let slots = {
        heading: {
            UNSAFE_className: headingClassName
        },
        content: {
            UNSAFE_className: contentClassName
        }
    };
    return /*#__PURE__*/ (0, ($parcel$interopDefault($6dAMz$react))).createElement((0, $6dAMz$reactspectrumlayout.Flex), {
        ...(0, $6dAMz$reactariautils.filterDOMProps)(otherProps),
        UNSAFE_style: styleProps.style,
        isHidden: styleProps.hidden,
        UNSAFE_className: (0, $6dAMz$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($e7750c05f28ceb4f$exports))), "spectrum-IllustratedMessage", styleProps.className),
        ref: ref
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($6dAMz$react))).createElement((0, $6dAMz$reactspectrumutils.ClearSlots), null, /*#__PURE__*/ (0, ($parcel$interopDefault($6dAMz$react))).createElement((0, $6dAMz$reactspectrumutils.SlotProvider), {
        slots: slots
    }, children)));
}
/**
 * An IllustratedMessage displays an illustration and a message, usually
 * for an empty state or an error page.
 */ let $90e6e9abe7fc12ca$export$406dbc84c317ece0 = /*#__PURE__*/ (0, $6dAMz$react.forwardRef)($90e6e9abe7fc12ca$var$IllustratedMessage);
