// @react-spectrum/slider/dist/module.js
"use client";

import "./main.css";
import $e8390eb96e247aac$exports from "./ar-AE.module.js";
import $50d639be1d52f61f$exports from "./bg-BG.module.js";
import $f8b1a75b07b20cfe$exports from "./cs-CZ.module.js";
import $5ece97d9dd84c4a6$exports from "./da-DK.module.js";
import $8c2d8eab91e8fc21$exports from "./de-DE.module.js";
import $c99d1f332dc5c52f$exports from "./el-GR.module.js";
import $6ff942e88faf5460$exports from "./en-US.module.js";
import $9bbe7d2a4b627f76$exports from "./es-ES.module.js";
import $f95dc783f4af1258$exports from "./et-EE.module.js";
import $042c971985d4b96a$exports from "./fi-FI.module.js";
import $eb1655a686c703e5$exports from "./fr-FR.module.js";
import $4d5c142cc5703c47$exports from "./he-IL.module.js";
import $04a7af02228fd881$exports from "./hr-HR.module.js";
import $4f76f1f805f21373$exports from "./hu-HU.module.js";
import $50e77a8887ec2ed1$exports from "./it-IT.module.js";
import $b6c57fa0cd7a21d2$exports from "./ja-JP.module.js";
import $da3e708015f0739a$exports from "./ko-KR.module.js";
import $38ae0524671e48fd$exports from "./lt-LT.module.js";
import $38b2122ea1a115cf$exports from "./lv-LV.module.js";
import $1abc56118d256aa1$exports from "./nb-NO.module.js";
import $f3a9d466bd3005e4$exports from "./nl-NL.module.js";
import $f8eb710b9573f592$exports from "./pl-PL.module.js";
import $f80715c2c0b326b4$exports from "./pt-BR.module.js";
import $54be727db6762488$exports from "./pt-PT.module.js";
import $0e518af4f08c31a6$exports from "./ro-RO.module.js";
import $df57f3dfdac40a9a$exports from "./ru-RU.module.js";
import $0b73a22f92f43e6f$exports from "./sk-SK.module.js";
import $5d3aeb5a18a0b757$exports from "./sl-SI.module.js";
import $530d6fbe3f80ba06$exports from "./sr-SP.module.js";
import $c88fa8b24ae60741$exports from "./sv-SE.module.js";
import $3d93d7022ddd1ceb$exports from "./tr-TR.module.js";
import $3108f34a285e911e$exports from "./uk-UA.module.js";
import $404def46e9659a20$exports from "./zh-CN.module.js";
import $e54d1a8757bfdcd2$exports from "./zh-TW.module.js";
import {clamp as $9lJNF$clamp, mergeProps as $9lJNF$mergeProps} from "@react-aria/utils";
import {classNames as $9lJNF$classNames, useStyleProps as $9lJNF$useStyleProps, useFocusableRef as $9lJNF$useFocusableRef, SlotProvider as $9lJNF$SlotProvider} from "@react-spectrum/utils";
import $9lJNF$react, {useRef as $9lJNF$useRef} from "react";
import {useLocale as $9lJNF$useLocale, useNumberFormatter as $9lJNF$useNumberFormatter, useLocalizedStringFormatter as $9lJNF$useLocalizedStringFormatter} from "@react-aria/i18n";
import {useSliderState as $9lJNF$useSliderState} from "@react-stately/slider";
import {useProviderProps as $9lJNF$useProviderProps} from "@react-spectrum/provider";
import {useSlider as $9lJNF$useSlider, useSliderThumb as $9lJNF$useSliderThumb} from "@react-aria/slider";
import {FocusRing as $9lJNF$FocusRing} from "@react-aria/focus";
import {useHover as $9lJNF$useHover} from "@react-aria/interactions";
import {VisuallyHidden as $9lJNF$VisuallyHidden} from "@react-aria/visually-hidden";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $bfbf422e6c44be4b$exports = {};

$parcel$export($bfbf422e6c44be4b$exports, "contextualHelp", () => $bfbf422e6c44be4b$export$4ba534aba1836d, (v) => $bfbf422e6c44be4b$export$4ba534aba1836d = v);
$parcel$export($bfbf422e6c44be4b$exports, "focus-ring", () => $bfbf422e6c44be4b$export$f39a09f249340e2a, (v) => $bfbf422e6c44be4b$export$f39a09f249340e2a = v);
$parcel$export($bfbf422e6c44be4b$exports, "i18nFontFamily", () => $bfbf422e6c44be4b$export$8c4ee2c50c22c514, (v) => $bfbf422e6c44be4b$export$8c4ee2c50c22c514 = v);
$parcel$export($bfbf422e6c44be4b$exports, "is-disabled", () => $bfbf422e6c44be4b$export$d35bc1e505d1ebbf, (v) => $bfbf422e6c44be4b$export$d35bc1e505d1ebbf = v);
$parcel$export($bfbf422e6c44be4b$exports, "is-dragged", () => $bfbf422e6c44be4b$export$8778c911bed6c759, (v) => $bfbf422e6c44be4b$export$8778c911bed6c759 = v);
$parcel$export($bfbf422e6c44be4b$exports, "is-focused", () => $bfbf422e6c44be4b$export$e7dc768d35940237, (v) => $bfbf422e6c44be4b$export$e7dc768d35940237 = v);
$parcel$export($bfbf422e6c44be4b$exports, "is-hovered", () => $bfbf422e6c44be4b$export$b8813cd5d7824ce7, (v) => $bfbf422e6c44be4b$export$b8813cd5d7824ce7 = v);
$parcel$export($bfbf422e6c44be4b$exports, "is-tophandle", () => $bfbf422e6c44be4b$export$e7c8b64dd65a8bd6, (v) => $bfbf422e6c44be4b$export$e7c8b64dd65a8bd6 = v);
$parcel$export($bfbf422e6c44be4b$exports, "label", () => $bfbf422e6c44be4b$export$1237798dc640739a, (v) => $bfbf422e6c44be4b$export$1237798dc640739a = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider", () => $bfbf422e6c44be4b$export$fc844a18ff837543, (v) => $bfbf422e6c44be4b$export$fc844a18ff837543 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Dial", () => $bfbf422e6c44be4b$export$db619d28010f1dd7, (v) => $bfbf422e6c44be4b$export$db619d28010f1dd7 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Dial--small", () => $bfbf422e6c44be4b$export$40db493959f853b2, (v) => $bfbf422e6c44be4b$export$40db493959f853b2 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-controls", () => $bfbf422e6c44be4b$export$eed070f7d409a4eb, (v) => $bfbf422e6c44be4b$export$eed070f7d409a4eb = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Dial-controls", () => $bfbf422e6c44be4b$export$9a4ea6c1b342a483, (v) => $bfbf422e6c44be4b$export$9a4ea6c1b342a483 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-handle", () => $bfbf422e6c44be4b$export$4e56e324ec89fe53, (v) => $bfbf422e6c44be4b$export$4e56e324ec89fe53 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Dial-handle", () => $bfbf422e6c44be4b$export$41c59246a9248fa9, (v) => $bfbf422e6c44be4b$export$41c59246a9248fa9 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-input", () => $bfbf422e6c44be4b$export$fdecbf00e674772b, (v) => $bfbf422e6c44be4b$export$fdecbf00e674772b = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Dial-input", () => $bfbf422e6c44be4b$export$39043e8612543a9e, (v) => $bfbf422e6c44be4b$export$39043e8612543a9e = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Dial-label", () => $bfbf422e6c44be4b$export$d93200a6e46e9ce7, (v) => $bfbf422e6c44be4b$export$d93200a6e46e9ce7 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-labelContainer", () => $bfbf422e6c44be4b$export$6a95ef18f88e804b, (v) => $bfbf422e6c44be4b$export$6a95ef18f88e804b = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Dial-labelContainer", () => $bfbf422e6c44be4b$export$f64ec32f85626909, (v) => $bfbf422e6c44be4b$export$f64ec32f85626909 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Dial-value", () => $bfbf422e6c44be4b$export$be548803c10dddf5, (v) => $bfbf422e6c44be4b$export$be548803c10dddf5 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-FocusRing-ring", () => $bfbf422e6c44be4b$export$4109102f950813a6, (v) => $bfbf422e6c44be4b$export$4109102f950813a6 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-FocusRing", () => $bfbf422e6c44be4b$export$24c7f46a6e3605dd, (v) => $bfbf422e6c44be4b$export$24c7f46a6e3605dd = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-FocusRing--quiet", () => $bfbf422e6c44be4b$export$2927016961429360, (v) => $bfbf422e6c44be4b$export$2927016961429360 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider--color", () => $bfbf422e6c44be4b$export$c0271c7b757e2c6d, (v) => $bfbf422e6c44be4b$export$c0271c7b757e2c6d = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider--filled", () => $bfbf422e6c44be4b$export$743466ce5e272176, (v) => $bfbf422e6c44be4b$export$743466ce5e272176 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider--label-side", () => $bfbf422e6c44be4b$export$fc5a803c4caf36a, (v) => $bfbf422e6c44be4b$export$fc5a803c4caf36a = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider--positionSide", () => $bfbf422e6c44be4b$export$7e6c269c6bbb2ffa, (v) => $bfbf422e6c44be4b$export$7e6c269c6bbb2ffa = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider--positionTop", () => $bfbf422e6c44be4b$export$3d0239f4af6225c0, (v) => $bfbf422e6c44be4b$export$3d0239f4af6225c0 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider--ramp", () => $bfbf422e6c44be4b$export$87c47a8a21ea9aa, (v) => $bfbf422e6c44be4b$export$87c47a8a21ea9aa = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider--range", () => $bfbf422e6c44be4b$export$91bbf9ca430efa38, (v) => $bfbf422e6c44be4b$export$91bbf9ca430efa38 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-buffer", () => $bfbf422e6c44be4b$export$8ac5a980ecf86d70, (v) => $bfbf422e6c44be4b$export$8ac5a980ecf86d70 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-contextualHelp", () => $bfbf422e6c44be4b$export$21be524027245261, (v) => $bfbf422e6c44be4b$export$21be524027245261 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-fill", () => $bfbf422e6c44be4b$export$79e384500d3b9c9d, (v) => $bfbf422e6c44be4b$export$79e384500d3b9c9d = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-fill--right", () => $bfbf422e6c44be4b$export$4ad5f741edcaf768, (v) => $bfbf422e6c44be4b$export$4ad5f741edcaf768 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-label", () => $bfbf422e6c44be4b$export$87d97f82d3be9540, (v) => $bfbf422e6c44be4b$export$87d97f82d3be9540 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-ramp", () => $bfbf422e6c44be4b$export$74af90f7b59a4a0b, (v) => $bfbf422e6c44be4b$export$74af90f7b59a4a0b = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-tick", () => $bfbf422e6c44be4b$export$daf7ddf4a129876d, (v) => $bfbf422e6c44be4b$export$daf7ddf4a129876d = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-tickLabel", () => $bfbf422e6c44be4b$export$e045256579534563, (v) => $bfbf422e6c44be4b$export$e045256579534563 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-ticks", () => $bfbf422e6c44be4b$export$8015a78273599416, (v) => $bfbf422e6c44be4b$export$8015a78273599416 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-track", () => $bfbf422e6c44be4b$export$16e664b3091a0b47, (v) => $bfbf422e6c44be4b$export$16e664b3091a0b47 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-value", () => $bfbf422e6c44be4b$export$d1c0699c6d985532, (v) => $bfbf422e6c44be4b$export$d1c0699c6d985532 = v);
$parcel$export($bfbf422e6c44be4b$exports, "spectrum-Slider-valueLabelContainer", () => $bfbf422e6c44be4b$export$2183ef1701a88013, (v) => $bfbf422e6c44be4b$export$2183ef1701a88013 = v);
$parcel$export($bfbf422e6c44be4b$exports, "u-isGrabbing", () => $bfbf422e6c44be4b$export$af9e6e484daea602, (v) => $bfbf422e6c44be4b$export$af9e6e484daea602 = v);
$parcel$export($bfbf422e6c44be4b$exports, "value", () => $bfbf422e6c44be4b$export$2ab9a8f9f1186f14, (v) => $bfbf422e6c44be4b$export$2ab9a8f9f1186f14 = v);
var $bfbf422e6c44be4b$export$4ba534aba1836d;
var $bfbf422e6c44be4b$export$f39a09f249340e2a;
var $bfbf422e6c44be4b$export$8c4ee2c50c22c514;
var $bfbf422e6c44be4b$export$d35bc1e505d1ebbf;
var $bfbf422e6c44be4b$export$8778c911bed6c759;
var $bfbf422e6c44be4b$export$e7dc768d35940237;
var $bfbf422e6c44be4b$export$b8813cd5d7824ce7;
var $bfbf422e6c44be4b$export$e7c8b64dd65a8bd6;
var $bfbf422e6c44be4b$export$1237798dc640739a;
var $bfbf422e6c44be4b$export$fc844a18ff837543;
var $bfbf422e6c44be4b$export$db619d28010f1dd7;
var $bfbf422e6c44be4b$export$40db493959f853b2;
var $bfbf422e6c44be4b$export$eed070f7d409a4eb;
var $bfbf422e6c44be4b$export$9a4ea6c1b342a483;
var $bfbf422e6c44be4b$export$4e56e324ec89fe53;
var $bfbf422e6c44be4b$export$41c59246a9248fa9;
var $bfbf422e6c44be4b$export$fdecbf00e674772b;
var $bfbf422e6c44be4b$export$39043e8612543a9e;
var $bfbf422e6c44be4b$export$d93200a6e46e9ce7;
var $bfbf422e6c44be4b$export$6a95ef18f88e804b;
var $bfbf422e6c44be4b$export$f64ec32f85626909;
var $bfbf422e6c44be4b$export$be548803c10dddf5;
var $bfbf422e6c44be4b$export$4109102f950813a6;
var $bfbf422e6c44be4b$export$24c7f46a6e3605dd;
var $bfbf422e6c44be4b$export$2927016961429360;
var $bfbf422e6c44be4b$export$c0271c7b757e2c6d;
var $bfbf422e6c44be4b$export$743466ce5e272176;
var $bfbf422e6c44be4b$export$fc5a803c4caf36a;
var $bfbf422e6c44be4b$export$7e6c269c6bbb2ffa;
var $bfbf422e6c44be4b$export$3d0239f4af6225c0;
var $bfbf422e6c44be4b$export$87c47a8a21ea9aa;
var $bfbf422e6c44be4b$export$91bbf9ca430efa38;
var $bfbf422e6c44be4b$export$8ac5a980ecf86d70;
var $bfbf422e6c44be4b$export$21be524027245261;
var $bfbf422e6c44be4b$export$79e384500d3b9c9d;
var $bfbf422e6c44be4b$export$4ad5f741edcaf768;
var $bfbf422e6c44be4b$export$87d97f82d3be9540;
var $bfbf422e6c44be4b$export$74af90f7b59a4a0b;
var $bfbf422e6c44be4b$export$daf7ddf4a129876d;
var $bfbf422e6c44be4b$export$e045256579534563;
var $bfbf422e6c44be4b$export$8015a78273599416;
var $bfbf422e6c44be4b$export$16e664b3091a0b47;
var $bfbf422e6c44be4b$export$d1c0699c6d985532;
var $bfbf422e6c44be4b$export$2183ef1701a88013;
var $bfbf422e6c44be4b$export$af9e6e484daea602;
var $bfbf422e6c44be4b$export$2ab9a8f9f1186f14;
$bfbf422e6c44be4b$export$4ba534aba1836d = `BTW8mG_contextualHelp`;
$bfbf422e6c44be4b$export$f39a09f249340e2a = `BTW8mG_focus-ring`;
$bfbf422e6c44be4b$export$8c4ee2c50c22c514 = `BTW8mG_i18nFontFamily`;
$bfbf422e6c44be4b$export$d35bc1e505d1ebbf = `BTW8mG_is-disabled`;
$bfbf422e6c44be4b$export$8778c911bed6c759 = `BTW8mG_is-dragged`;
$bfbf422e6c44be4b$export$e7dc768d35940237 = `BTW8mG_is-focused`;
$bfbf422e6c44be4b$export$b8813cd5d7824ce7 = `BTW8mG_is-hovered`;
$bfbf422e6c44be4b$export$e7c8b64dd65a8bd6 = `BTW8mG_is-tophandle`;
$bfbf422e6c44be4b$export$1237798dc640739a = `BTW8mG_label`;
$bfbf422e6c44be4b$export$fc844a18ff837543 = `BTW8mG_spectrum-Slider`;
$bfbf422e6c44be4b$export$db619d28010f1dd7 = `BTW8mG_spectrum-Dial ${$bfbf422e6c44be4b$export$fc844a18ff837543}`;
$bfbf422e6c44be4b$export$40db493959f853b2 = `BTW8mG_spectrum-Dial--small`;
$bfbf422e6c44be4b$export$eed070f7d409a4eb = `BTW8mG_spectrum-Slider-controls`;
$bfbf422e6c44be4b$export$9a4ea6c1b342a483 = `BTW8mG_spectrum-Dial-controls ${$bfbf422e6c44be4b$export$eed070f7d409a4eb}`;
$bfbf422e6c44be4b$export$4e56e324ec89fe53 = `BTW8mG_spectrum-Slider-handle`;
$bfbf422e6c44be4b$export$41c59246a9248fa9 = `BTW8mG_spectrum-Dial-handle ${$bfbf422e6c44be4b$export$4e56e324ec89fe53}`;
$bfbf422e6c44be4b$export$fdecbf00e674772b = `BTW8mG_spectrum-Slider-input`;
$bfbf422e6c44be4b$export$39043e8612543a9e = `BTW8mG_spectrum-Dial-input ${$bfbf422e6c44be4b$export$fdecbf00e674772b}`;
$bfbf422e6c44be4b$export$d93200a6e46e9ce7 = `BTW8mG_spectrum-Dial-label`;
$bfbf422e6c44be4b$export$6a95ef18f88e804b = `BTW8mG_spectrum-Slider-labelContainer`;
$bfbf422e6c44be4b$export$f64ec32f85626909 = `BTW8mG_spectrum-Dial-labelContainer ${$bfbf422e6c44be4b$export$6a95ef18f88e804b}`;
$bfbf422e6c44be4b$export$be548803c10dddf5 = `BTW8mG_spectrum-Dial-value`;
$bfbf422e6c44be4b$export$4109102f950813a6 = `BTW8mG_spectrum-FocusRing-ring`;
$bfbf422e6c44be4b$export$24c7f46a6e3605dd = `BTW8mG_spectrum-FocusRing ${$bfbf422e6c44be4b$export$4109102f950813a6}`;
$bfbf422e6c44be4b$export$2927016961429360 = `BTW8mG_spectrum-FocusRing--quiet`;
$bfbf422e6c44be4b$export$c0271c7b757e2c6d = `BTW8mG_spectrum-Slider--color`;
$bfbf422e6c44be4b$export$743466ce5e272176 = `BTW8mG_spectrum-Slider--filled`;
$bfbf422e6c44be4b$export$fc5a803c4caf36a = `BTW8mG_spectrum-Slider--label-side`;
$bfbf422e6c44be4b$export$7e6c269c6bbb2ffa = `BTW8mG_spectrum-Slider--positionSide`;
$bfbf422e6c44be4b$export$3d0239f4af6225c0 = `BTW8mG_spectrum-Slider--positionTop`;
$bfbf422e6c44be4b$export$87c47a8a21ea9aa = `BTW8mG_spectrum-Slider--ramp`;
$bfbf422e6c44be4b$export$91bbf9ca430efa38 = `BTW8mG_spectrum-Slider--range`;
$bfbf422e6c44be4b$export$8ac5a980ecf86d70 = `BTW8mG_spectrum-Slider-buffer`;
$bfbf422e6c44be4b$export$21be524027245261 = `BTW8mG_spectrum-Slider-contextualHelp`;
$bfbf422e6c44be4b$export$79e384500d3b9c9d = `BTW8mG_spectrum-Slider-fill`;
$bfbf422e6c44be4b$export$4ad5f741edcaf768 = `BTW8mG_spectrum-Slider-fill--right`;
$bfbf422e6c44be4b$export$87d97f82d3be9540 = `BTW8mG_spectrum-Slider-label`;
$bfbf422e6c44be4b$export$74af90f7b59a4a0b = `BTW8mG_spectrum-Slider-ramp`;
$bfbf422e6c44be4b$export$daf7ddf4a129876d = `BTW8mG_spectrum-Slider-tick`;
$bfbf422e6c44be4b$export$e045256579534563 = `BTW8mG_spectrum-Slider-tickLabel`;
$bfbf422e6c44be4b$export$8015a78273599416 = `BTW8mG_spectrum-Slider-ticks`;
$bfbf422e6c44be4b$export$16e664b3091a0b47 = `BTW8mG_spectrum-Slider-track`;
$bfbf422e6c44be4b$export$d1c0699c6d985532 = `BTW8mG_spectrum-Slider-value`;
$bfbf422e6c44be4b$export$2183ef1701a88013 = `BTW8mG_spectrum-Slider-valueLabelContainer`;
$bfbf422e6c44be4b$export$af9e6e484daea602 = `BTW8mG_u-isGrabbing`;
$bfbf422e6c44be4b$export$2ab9a8f9f1186f14 = `BTW8mG_value`;





function $39572711324a3b8e$var$SliderBase(props, ref) {
    props = (0, $9lJNF$useProviderProps)(props);
    let { isDisabled: isDisabled, children: children, classes: classes, style: style, labelPosition: labelPosition = "top", getValueLabel: getValueLabel, showValueLabel: showValueLabel = !!props.label, formatOptions: formatOptions, minValue: minValue = 0, maxValue: maxValue = 100, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $9lJNF$useStyleProps)(otherProps);
    // `Math.abs(Math.sign(a) - Math.sign(b)) === 2` is true if the values have a different sign.
    let alwaysDisplaySign = Math.abs(Math.sign(minValue) - Math.sign(maxValue)) === 2;
    if (alwaysDisplaySign) {
        if (formatOptions != null) {
            if (!("signDisplay" in formatOptions)) formatOptions = {
                ...formatOptions,
                // @ts-ignore
                signDisplay: "exceptZero"
            };
        } else // @ts-ignore
        formatOptions = {
            signDisplay: "exceptZero"
        };
    }
    const formatter = (0, $9lJNF$useNumberFormatter)(formatOptions);
    const state = (0, $9lJNF$useSliderState)({
        ...props,
        numberFormatter: formatter,
        minValue: minValue,
        maxValue: maxValue
    });
    let trackRef = (0, $9lJNF$useRef)();
    let { groupProps: groupProps, trackProps: trackProps, labelProps: labelProps, outputProps: outputProps } = (0, $9lJNF$useSlider)(props, state, trackRef);
    let inputRef = (0, $9lJNF$useRef)();
    let domRef = (0, $9lJNF$useFocusableRef)(ref, inputRef);
    let displayValue = "";
    let maxLabelLength = undefined;
    if (typeof getValueLabel === "function") {
        displayValue = getValueLabel(state.values);
        switch(state.values.length){
            case 1:
                maxLabelLength = Math.max(getValueLabel([
                    minValue
                ]).length, getValueLabel([
                    maxValue
                ]).length);
                break;
            case 2:
                // Try all possible combinations of min and max values.
                maxLabelLength = Math.max(getValueLabel([
                    minValue,
                    minValue
                ]).length, getValueLabel([
                    minValue,
                    maxValue
                ]).length, getValueLabel([
                    maxValue,
                    minValue
                ]).length, getValueLabel([
                    maxValue,
                    maxValue
                ]).length);
                break;
            default:
                throw new Error("Only sliders with 1 or 2 handles are supported!");
        }
    } else {
        maxLabelLength = Math.max([
            ...formatter.format(minValue)
        ].length, [
            ...formatter.format(maxValue)
        ].length);
        switch(state.values.length){
            case 1:
                displayValue = state.getThumbValueLabel(0);
                break;
            case 2:
                // This should really use the NumberFormat#formatRange proposal...
                // https://github.com/tc39/ecma402/issues/393
                // https://github.com/tc39/proposal-intl-numberformat-v3#formatrange-ecma-402-393
                displayValue = `${state.getThumbValueLabel(0)} \u{2013} ${state.getThumbValueLabel(1)}`;
                maxLabelLength = 3 + 2 * Math.max(maxLabelLength, [
                    ...formatter.format(minValue)
                ].length, [
                    ...formatter.format(maxValue)
                ].length);
                break;
            default:
                throw new Error("Only sliders with 1 or 2 handles are supported!");
        }
    }
    let labelNode = /*#__PURE__*/ (0, $9lJNF$react).createElement("label", {
        className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-label"),
        ...labelProps
    }, props.label);
    let valueNode = /*#__PURE__*/ (0, $9lJNF$react).createElement("output", {
        ...outputProps,
        className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-value"),
        style: maxLabelLength && {
            width: `${maxLabelLength}ch`,
            minWidth: `${maxLabelLength}ch`
        }
    }, displayValue);
    return /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
        ref: domRef,
        className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider", {
            "spectrum-Slider--positionTop": labelPosition === "top",
            "spectrum-Slider--positionSide": labelPosition === "side",
            "is-disabled": isDisabled
        }, classes, styleProps.className),
        style: {
            ...style,
            ...styleProps.style
        },
        ...groupProps
    }, props.label && /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
        className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-labelContainer"),
        role: "presentation"
    }, props.label && labelNode, props.contextualHelp && /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $9lJNF$SlotProvider), {
        slots: {
            actionButton: {
                UNSAFE_className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-contextualHelp")
            }
        }
    }, props.contextualHelp), labelPosition === "top" && showValueLabel && valueNode), /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
        className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-controls"),
        ref: trackRef,
        ...trackProps,
        role: "presentation"
    }, children({
        trackRef: trackRef,
        inputRef: inputRef,
        state: state
    })), labelPosition === "side" && /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
        className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-valueLabelContainer"),
        role: "presentation"
    }, showValueLabel && valueNode));
}
const $39572711324a3b8e$export$9418495bb635ebde = /*#__PURE__*/ (0, $9lJNF$react).forwardRef($39572711324a3b8e$var$SliderBase);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







function $9058ca69d66a724f$export$2c1b491743890dec(props) {
    let { inputRef: inputRef, state: state } = props;
    let backupRef = (0, $9lJNF$useRef)();
    inputRef = inputRef || backupRef;
    let { thumbProps: thumbProps, inputProps: inputProps, isDragging: isDragging, isFocused: isFocused } = (0, $9lJNF$useSliderThumb)({
        ...props,
        inputRef: inputRef
    }, state);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $9lJNF$useHover)({});
    return /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $9lJNF$FocusRing), {
        within: true,
        focusRingClass: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "is-focused")
    }, /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
        className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-handle", {
            "is-hovered": isHovered,
            "is-dragged": isDragging,
            "is-tophandle": isFocused
        }),
        ...(0, $9lJNF$mergeProps)(thumbProps, hoverProps),
        role: "presentation"
    }, /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $9lJNF$VisuallyHidden), null, /*#__PURE__*/ (0, $9lJNF$react).createElement("input", {
        className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-input"),
        ref: inputRef,
        ...inputProps
    }))));
}




function $a1251704aa53e1a5$var$Slider(props, ref) {
    let { onChange: onChange, onChangeEnd: onChangeEnd, value: value, defaultValue: defaultValue, isFilled: isFilled, fillOffset: fillOffset, trackGradient: trackGradient, getValueLabel: getValueLabel, ...otherProps } = props;
    let baseProps = {
        ...otherProps,
        // Normalize `value: number[]` to `value: number`
        value: value != null ? [
            value
        ] : undefined,
        defaultValue: defaultValue != null ? [
            defaultValue
        ] : undefined,
        onChange: (v)=>{
            onChange === null || onChange === void 0 ? void 0 : onChange(v[0]);
        },
        onChangeEnd: (v)=>{
            onChangeEnd === null || onChangeEnd === void 0 ? void 0 : onChangeEnd(v[0]);
        },
        getValueLabel: getValueLabel ? ([v])=>getValueLabel(v) : undefined
    };
    let { direction: direction } = (0, $9lJNF$useLocale)();
    return /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $39572711324a3b8e$export$9418495bb635ebde), {
        ...baseProps,
        ref: ref,
        classes: {
            "spectrum-Slider--filled": isFilled && fillOffset == null
        },
        style: // @ts-ignore
        {
            "--spectrum-slider-track-gradient": trackGradient && `linear-gradient(to ${direction === "ltr" ? "right" : "left"}, ${trackGradient.join(", ")})`
        }
    }, ({ trackRef: trackRef, inputRef: inputRef, state: state })=>{
        fillOffset = fillOffset != null ? (0, $9lJNF$clamp)(fillOffset, state.getThumbMinValue(0), state.getThumbMaxValue(0)) : fillOffset;
        let cssDirection = direction === "rtl" ? "right" : "left";
        let lowerTrack = /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
            className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-track"),
            style: {
                width: `${state.getThumbPercent(0) * 100}%`,
                // TODO not sure if it has advantages, but this could also be implemented as CSS calc():
                // .track::before {
                //    background-size: calc((1/ (var(--width)/100)) * 100%);
                //    width: calc(var(--width) * 1%)M
                // }
                // @ts-ignore
                "--spectrum-track-background-size": `${1 / state.getThumbPercent(0) * 100}%`,
                "--spectrum-track-background-position": direction === "ltr" ? "0" : "100%"
            }
        });
        let upperTrack = /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
            className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-track"),
            style: {
                width: `${(1 - state.getThumbPercent(0)) * 100}%`,
                // @ts-ignore
                "--spectrum-track-background-size": `${1 / (1 - state.getThumbPercent(0)) * 100}%`,
                "--spectrum-track-background-position": direction === "ltr" ? "100%" : "0"
            }
        });
        let filledTrack = null;
        if (isFilled && fillOffset != null) {
            let width = state.getThumbPercent(0) - state.getValuePercent(fillOffset);
            let isRightOfOffset = width > 0;
            let offset = isRightOfOffset ? state.getValuePercent(fillOffset) : state.getThumbPercent(0);
            filledTrack = /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
                className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-fill", {
                    "spectrum-Slider-fill--right": isRightOfOffset
                }),
                style: {
                    [cssDirection]: `${offset * 100}%`,
                    width: `${Math.abs(width) * 100}%`
                }
            });
        }
        return /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $9lJNF$react).Fragment, null, lowerTrack, /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $9058ca69d66a724f$export$2c1b491743890dec), {
            index: 0,
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            inputRef: inputRef,
            state: state,
            name: props.name
        }), filledTrack, upperTrack);
    });
}
/**
 * Sliders allow users to quickly select a value within a range. They should be used when the upper and lower bounds to the range are invariable.
 */ const $a1251704aa53e1a5$export$472062a354075cee = /*#__PURE__*/ (0, $9lJNF$react).forwardRef($a1251704aa53e1a5$var$Slider);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
var $c45b24830a4cc91c$exports = {};


































$c45b24830a4cc91c$exports = {
    "ar-AE": $e8390eb96e247aac$exports,
    "bg-BG": $50d639be1d52f61f$exports,
    "cs-CZ": $f8b1a75b07b20cfe$exports,
    "da-DK": $5ece97d9dd84c4a6$exports,
    "de-DE": $8c2d8eab91e8fc21$exports,
    "el-GR": $c99d1f332dc5c52f$exports,
    "en-US": $6ff942e88faf5460$exports,
    "es-ES": $9bbe7d2a4b627f76$exports,
    "et-EE": $f95dc783f4af1258$exports,
    "fi-FI": $042c971985d4b96a$exports,
    "fr-FR": $eb1655a686c703e5$exports,
    "he-IL": $4d5c142cc5703c47$exports,
    "hr-HR": $04a7af02228fd881$exports,
    "hu-HU": $4f76f1f805f21373$exports,
    "it-IT": $50e77a8887ec2ed1$exports,
    "ja-JP": $b6c57fa0cd7a21d2$exports,
    "ko-KR": $da3e708015f0739a$exports,
    "lt-LT": $38ae0524671e48fd$exports,
    "lv-LV": $38b2122ea1a115cf$exports,
    "nb-NO": $1abc56118d256aa1$exports,
    "nl-NL": $f3a9d466bd3005e4$exports,
    "pl-PL": $f8eb710b9573f592$exports,
    "pt-BR": $f80715c2c0b326b4$exports,
    "pt-PT": $54be727db6762488$exports,
    "ro-RO": $0e518af4f08c31a6$exports,
    "ru-RU": $df57f3dfdac40a9a$exports,
    "sk-SK": $0b73a22f92f43e6f$exports,
    "sl-SI": $5d3aeb5a18a0b757$exports,
    "sr-SP": $530d6fbe3f80ba06$exports,
    "sv-SE": $c88fa8b24ae60741$exports,
    "tr-TR": $3d93d7022ddd1ceb$exports,
    "uk-UA": $3108f34a285e911e$exports,
    "zh-CN": $404def46e9659a20$exports,
    "zh-TW": $e54d1a8757bfdcd2$exports
};







function $d42a197b273dc031$var$RangeSlider(props, ref) {
    let { onChange: onChange, onChangeEnd: onChangeEnd, value: value, defaultValue: defaultValue, getValueLabel: getValueLabel, ...otherProps } = props;
    var _props_minValue, _props_maxValue;
    let baseProps = {
        ...otherProps,
        value: value != null ? [
            value.start,
            value.end
        ] : undefined,
        defaultValue: defaultValue != null ? [
            defaultValue.start,
            defaultValue.end
        ] : [
            (_props_minValue = props.minValue) !== null && _props_minValue !== void 0 ? _props_minValue : 0,
            (_props_maxValue = props.maxValue) !== null && _props_maxValue !== void 0 ? _props_maxValue : 100
        ],
        onChange (v) {
            onChange === null || onChange === void 0 ? void 0 : onChange({
                start: v[0],
                end: v[1]
            });
        },
        onChangeEnd (v) {
            onChangeEnd === null || onChangeEnd === void 0 ? void 0 : onChangeEnd({
                start: v[0],
                end: v[1]
            });
        },
        getValueLabel: getValueLabel ? ([start, end])=>getValueLabel({
                start: start,
                end: end
            }) : undefined
    };
    let stringFormatter = (0, $9lJNF$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($c45b24830a4cc91c$exports))), "@react-spectrum/slider");
    let { direction: direction } = (0, $9lJNF$useLocale)();
    return /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $39572711324a3b8e$export$9418495bb635ebde), {
        ...baseProps,
        classes: "spectrum-Slider--range",
        ref: ref
    }, ({ trackRef: trackRef, inputRef: inputRef, state: state })=>{
        let cssDirection = direction === "rtl" ? "right" : "left";
        return /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $9lJNF$react).Fragment, null, /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
            className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-track"),
            style: {
                width: `${state.getThumbPercent(0) * 100}%`
            }
        }), /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $9058ca69d66a724f$export$2c1b491743890dec), {
            index: 0,
            "aria-label": stringFormatter.format("minimum"),
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            inputRef: inputRef,
            state: state,
            name: props.startName
        }), /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
            className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-track"),
            style: {
                [cssDirection]: `${state.getThumbPercent(0) * 100}%`,
                width: `${Math.abs(state.getThumbPercent(0) - state.getThumbPercent(1)) * 100}%`
            }
        }), /*#__PURE__*/ (0, $9lJNF$react).createElement((0, $9058ca69d66a724f$export$2c1b491743890dec), {
            index: 1,
            "aria-label": stringFormatter.format("maximum"),
            isDisabled: props.isDisabled,
            trackRef: trackRef,
            state: state,
            name: props.endName
        }), /*#__PURE__*/ (0, $9lJNF$react).createElement("div", {
            className: (0, $9lJNF$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bfbf422e6c44be4b$exports))), "spectrum-Slider-track"),
            style: {
                width: `${(1 - state.getThumbPercent(1)) * 100}%`
            }
        }));
    });
}
/**
 * RangeSliders allow users to quickly select a subset range. They should be used when the upper and lower bounds to the range are invariable.
 */ const $d42a197b273dc031$export$826424dabc3dd705 = /*#__PURE__*/ (0, $9lJNF$react).forwardRef($d42a197b273dc031$var$RangeSlider);




export {$a1251704aa53e1a5$export$472062a354075cee as Slider, $d42a197b273dc031$export$826424dabc3dd705 as RangeSlider};
