// @react-spectrum/searchfield/dist/import.mjs
"use client";

import "./main.css";
import {useSlotProps as $hlecP$useSlotProps, classNames as $hlecP$classNames} from "@react-spectrum/utils";
import {ClearButton as $hlecP$ClearButton} from "@react-spectrum/button";
import $hlecP$spectrumiconsuiMagnifier from "@spectrum-icons/ui/Magnifier";
import $hlecP$react, {useRef as $hlecP$useRef, forwardRef as $hlecP$forwardRef} from "react";
import {TextFieldBase as $hlecP$TextFieldBase} from "@react-spectrum/textfield";
import {useFormProps as $hlecP$useFormProps} from "@react-spectrum/form";
import {useProviderProps as $hlecP$useProviderProps} from "@react-spectrum/provider";
import {useSearchField as $hlecP$useSearchField} from "@react-aria/searchfield";
import {useSearchFieldState as $hlecP$useSearchFieldState} from "@react-stately/searchfield";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $5589e2270a5c2f51$exports = {};

$parcel$export($5589e2270a5c2f51$exports, "focus-ring", () => $5589e2270a5c2f51$export$f39a09f249340e2a, (v) => $5589e2270a5c2f51$export$f39a09f249340e2a = v);
$parcel$export($5589e2270a5c2f51$exports, "i18nFontFamily", () => $5589e2270a5c2f51$export$8c4ee2c50c22c514, (v) => $5589e2270a5c2f51$export$8c4ee2c50c22c514 = v);
$parcel$export($5589e2270a5c2f51$exports, "is-quiet", () => $5589e2270a5c2f51$export$6d5c5433f0da4241, (v) => $5589e2270a5c2f51$export$6d5c5433f0da4241 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-ClearButton", () => $5589e2270a5c2f51$export$b752ce409e5660c, (v) => $5589e2270a5c2f51$export$b752ce409e5660c = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-FocusRing-ring", () => $5589e2270a5c2f51$export$4109102f950813a6, (v) => $5589e2270a5c2f51$export$4109102f950813a6 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-FocusRing", () => $5589e2270a5c2f51$export$24c7f46a6e3605dd, (v) => $5589e2270a5c2f51$export$24c7f46a6e3605dd = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-FocusRing--quiet", () => $5589e2270a5c2f51$export$2927016961429360, (v) => $5589e2270a5c2f51$export$2927016961429360 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search", () => $5589e2270a5c2f51$export$713911d754e1e4ae, (v) => $5589e2270a5c2f51$export$713911d754e1e4ae = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search--invalid", () => $5589e2270a5c2f51$export$77a6151a0b46a6a3, (v) => $5589e2270a5c2f51$export$77a6151a0b46a6a3 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search--loadable", () => $5589e2270a5c2f51$export$9c74d64afae3b2d4, (v) => $5589e2270a5c2f51$export$9c74d64afae3b2d4 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search--valid", () => $5589e2270a5c2f51$export$aa81e3ab40f37567, (v) => $5589e2270a5c2f51$export$aa81e3ab40f37567 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search-circleLoader", () => $5589e2270a5c2f51$export$70ae4353602451dc, (v) => $5589e2270a5c2f51$export$70ae4353602451dc = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search-input", () => $5589e2270a5c2f51$export$50e02603f2d97ae1, (v) => $5589e2270a5c2f51$export$50e02603f2d97ae1 = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Search-validationIcon", () => $5589e2270a5c2f51$export$938f0981825eae6e, (v) => $5589e2270a5c2f51$export$938f0981825eae6e = v);
$parcel$export($5589e2270a5c2f51$exports, "spectrum-Textfield", () => $5589e2270a5c2f51$export$2c8a10299d8b3418, (v) => $5589e2270a5c2f51$export$2c8a10299d8b3418 = v);
var $5589e2270a5c2f51$export$f39a09f249340e2a;
var $5589e2270a5c2f51$export$8c4ee2c50c22c514;
var $5589e2270a5c2f51$export$6d5c5433f0da4241;
var $5589e2270a5c2f51$export$b752ce409e5660c;
var $5589e2270a5c2f51$export$4109102f950813a6;
var $5589e2270a5c2f51$export$24c7f46a6e3605dd;
var $5589e2270a5c2f51$export$2927016961429360;
var $5589e2270a5c2f51$export$713911d754e1e4ae;
var $5589e2270a5c2f51$export$77a6151a0b46a6a3;
var $5589e2270a5c2f51$export$9c74d64afae3b2d4;
var $5589e2270a5c2f51$export$aa81e3ab40f37567;
var $5589e2270a5c2f51$export$70ae4353602451dc;
var $5589e2270a5c2f51$export$50e02603f2d97ae1;
var $5589e2270a5c2f51$export$938f0981825eae6e;
var $5589e2270a5c2f51$export$2c8a10299d8b3418;
$5589e2270a5c2f51$export$f39a09f249340e2a = `SOkmSG_focus-ring`;
$5589e2270a5c2f51$export$8c4ee2c50c22c514 = `SOkmSG_i18nFontFamily`;
$5589e2270a5c2f51$export$6d5c5433f0da4241 = `SOkmSG_is-quiet`;
$5589e2270a5c2f51$export$b752ce409e5660c = `SOkmSG_spectrum-ClearButton`;
$5589e2270a5c2f51$export$4109102f950813a6 = `SOkmSG_spectrum-FocusRing-ring`;
$5589e2270a5c2f51$export$24c7f46a6e3605dd = `SOkmSG_spectrum-FocusRing ${$5589e2270a5c2f51$export$4109102f950813a6}`;
$5589e2270a5c2f51$export$2927016961429360 = `SOkmSG_spectrum-FocusRing--quiet`;
$5589e2270a5c2f51$export$713911d754e1e4ae = `SOkmSG_spectrum-Search`;
$5589e2270a5c2f51$export$77a6151a0b46a6a3 = `SOkmSG_spectrum-Search--invalid`;
$5589e2270a5c2f51$export$9c74d64afae3b2d4 = `SOkmSG_spectrum-Search--loadable`;
$5589e2270a5c2f51$export$aa81e3ab40f37567 = `SOkmSG_spectrum-Search--valid`;
$5589e2270a5c2f51$export$70ae4353602451dc = `SOkmSG_spectrum-Search-circleLoader`;
$5589e2270a5c2f51$export$50e02603f2d97ae1 = `SOkmSG_spectrum-Search-input`;
$5589e2270a5c2f51$export$938f0981825eae6e = `SOkmSG_spectrum-Search-validationIcon`;
$5589e2270a5c2f51$export$2c8a10299d8b3418 = `SOkmSG_spectrum-Textfield`;







function $a6a4ace9217ce6df$var$SearchField(props, ref) {
    props = (0, $hlecP$useSlotProps)(props, "searchfield");
    props = (0, $hlecP$useProviderProps)(props);
    props = (0, $hlecP$useFormProps)(props);
    let defaultIcon = /*#__PURE__*/ (0, $hlecP$react).createElement((0, $hlecP$spectrumiconsuiMagnifier), {
        "data-testid": "searchicon"
    });
    let { icon: icon = defaultIcon, isDisabled: isDisabled, UNSAFE_className: UNSAFE_className, placeholder: placeholder, ...otherProps } = props;
    if (placeholder) console.warn("Placeholders are deprecated due to accessibility issues. Please use help text instead. See the docs for details: https://react-spectrum.adobe.com/react-spectrum/SearchField.html#help-text");
    let state = (0, $hlecP$useSearchFieldState)(props);
    let inputRef = (0, $hlecP$useRef)(null);
    let { clearButtonProps: clearButtonProps, ...result } = (0, $hlecP$useSearchField)(props, state, inputRef);
    let clearButton = /*#__PURE__*/ (0, $hlecP$react).createElement((0, $hlecP$ClearButton), {
        ...clearButtonProps,
        preventFocus: true,
        UNSAFE_className: (0, $hlecP$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($5589e2270a5c2f51$exports))), "spectrum-ClearButton"),
        isDisabled: isDisabled
    });
    let validationState = props.validationState || (result.isInvalid ? "invalid" : undefined);
    return /*#__PURE__*/ (0, $hlecP$react).createElement((0, $hlecP$TextFieldBase), {
        ...otherProps,
        ...result,
        validationState: validationState,
        UNSAFE_className: (0, $hlecP$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($5589e2270a5c2f51$exports))), "spectrum-Search", "spectrum-Textfield", {
            "is-disabled": isDisabled,
            "is-quiet": props.isQuiet,
            "spectrum-Search--invalid": validationState === "invalid" && !isDisabled,
            "spectrum-Search--valid": validationState === "valid" && !isDisabled
        }, UNSAFE_className),
        inputClassName: (0, $hlecP$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($5589e2270a5c2f51$exports))), "spectrum-Search-input"),
        ref: ref,
        inputRef: inputRef,
        isDisabled: isDisabled,
        icon: icon,
        wrapperChildren: state.value !== "" && !props.isReadOnly ? clearButton : undefined
    });
}
/**
 * A SearchField is a text field designed for searches.
 */ let $a6a4ace9217ce6df$export$b94867ecbd698f21 = /*#__PURE__*/ (0, $hlecP$forwardRef)($a6a4ace9217ce6df$var$SearchField);




export {$a6a4ace9217ce6df$export$b94867ecbd698f21 as SearchField};
