// @react-spectrum/searchfield/dist/main.js
"use client";

require("./main.css");
var $kYnKt$reactspectrumutils = require("@react-spectrum/utils");
var $kYnKt$reactspectrumbutton = require("@react-spectrum/button");
var $kYnKt$spectrumiconsuiMagnifier = require("@spectrum-icons/ui/Magnifier");
var $kYnKt$react = require("react");
var $kYnKt$reactspectrumtextfield = require("@react-spectrum/textfield");
var $kYnKt$reactspectrumform = require("@react-spectrum/form");
var $kYnKt$reactspectrumprovider = require("@react-spectrum/provider");
var $kYnKt$reactariasearchfield = require("@react-aria/searchfield");
var $kYnKt$reactstatelysearchfield = require("@react-stately/searchfield");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "SearchField", () => $111fe330b4c1f3ac$export$b94867ecbd698f21);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $1e139f06a0a4b696$exports = {};

$parcel$export($1e139f06a0a4b696$exports, "focus-ring", () => $1e139f06a0a4b696$export$f39a09f249340e2a, (v) => $1e139f06a0a4b696$export$f39a09f249340e2a = v);
$parcel$export($1e139f06a0a4b696$exports, "i18nFontFamily", () => $1e139f06a0a4b696$export$8c4ee2c50c22c514, (v) => $1e139f06a0a4b696$export$8c4ee2c50c22c514 = v);
$parcel$export($1e139f06a0a4b696$exports, "is-quiet", () => $1e139f06a0a4b696$export$6d5c5433f0da4241, (v) => $1e139f06a0a4b696$export$6d5c5433f0da4241 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-ClearButton", () => $1e139f06a0a4b696$export$b752ce409e5660c, (v) => $1e139f06a0a4b696$export$b752ce409e5660c = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-FocusRing-ring", () => $1e139f06a0a4b696$export$4109102f950813a6, (v) => $1e139f06a0a4b696$export$4109102f950813a6 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-FocusRing", () => $1e139f06a0a4b696$export$24c7f46a6e3605dd, (v) => $1e139f06a0a4b696$export$24c7f46a6e3605dd = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-FocusRing--quiet", () => $1e139f06a0a4b696$export$2927016961429360, (v) => $1e139f06a0a4b696$export$2927016961429360 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search", () => $1e139f06a0a4b696$export$713911d754e1e4ae, (v) => $1e139f06a0a4b696$export$713911d754e1e4ae = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search--invalid", () => $1e139f06a0a4b696$export$77a6151a0b46a6a3, (v) => $1e139f06a0a4b696$export$77a6151a0b46a6a3 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search--loadable", () => $1e139f06a0a4b696$export$9c74d64afae3b2d4, (v) => $1e139f06a0a4b696$export$9c74d64afae3b2d4 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search--valid", () => $1e139f06a0a4b696$export$aa81e3ab40f37567, (v) => $1e139f06a0a4b696$export$aa81e3ab40f37567 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search-circleLoader", () => $1e139f06a0a4b696$export$70ae4353602451dc, (v) => $1e139f06a0a4b696$export$70ae4353602451dc = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search-input", () => $1e139f06a0a4b696$export$50e02603f2d97ae1, (v) => $1e139f06a0a4b696$export$50e02603f2d97ae1 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search-validationIcon", () => $1e139f06a0a4b696$export$938f0981825eae6e, (v) => $1e139f06a0a4b696$export$938f0981825eae6e = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Textfield", () => $1e139f06a0a4b696$export$2c8a10299d8b3418, (v) => $1e139f06a0a4b696$export$2c8a10299d8b3418 = v);
var $1e139f06a0a4b696$export$f39a09f249340e2a;
var $1e139f06a0a4b696$export$8c4ee2c50c22c514;
var $1e139f06a0a4b696$export$6d5c5433f0da4241;
var $1e139f06a0a4b696$export$b752ce409e5660c;
var $1e139f06a0a4b696$export$4109102f950813a6;
var $1e139f06a0a4b696$export$24c7f46a6e3605dd;
var $1e139f06a0a4b696$export$2927016961429360;
var $1e139f06a0a4b696$export$713911d754e1e4ae;
var $1e139f06a0a4b696$export$77a6151a0b46a6a3;
var $1e139f06a0a4b696$export$9c74d64afae3b2d4;
var $1e139f06a0a4b696$export$aa81e3ab40f37567;
var $1e139f06a0a4b696$export$70ae4353602451dc;
var $1e139f06a0a4b696$export$50e02603f2d97ae1;
var $1e139f06a0a4b696$export$938f0981825eae6e;
var $1e139f06a0a4b696$export$2c8a10299d8b3418;
$1e139f06a0a4b696$export$f39a09f249340e2a = `SOkmSG_focus-ring`;
$1e139f06a0a4b696$export$8c4ee2c50c22c514 = `SOkmSG_i18nFontFamily`;
$1e139f06a0a4b696$export$6d5c5433f0da4241 = `SOkmSG_is-quiet`;
$1e139f06a0a4b696$export$b752ce409e5660c = `SOkmSG_spectrum-ClearButton`;
$1e139f06a0a4b696$export$4109102f950813a6 = `SOkmSG_spectrum-FocusRing-ring`;
$1e139f06a0a4b696$export$24c7f46a6e3605dd = `SOkmSG_spectrum-FocusRing ${$1e139f06a0a4b696$export$4109102f950813a6}`;
$1e139f06a0a4b696$export$2927016961429360 = `SOkmSG_spectrum-FocusRing--quiet`;
$1e139f06a0a4b696$export$713911d754e1e4ae = `SOkmSG_spectrum-Search`;
$1e139f06a0a4b696$export$77a6151a0b46a6a3 = `SOkmSG_spectrum-Search--invalid`;
$1e139f06a0a4b696$export$9c74d64afae3b2d4 = `SOkmSG_spectrum-Search--loadable`;
$1e139f06a0a4b696$export$aa81e3ab40f37567 = `SOkmSG_spectrum-Search--valid`;
$1e139f06a0a4b696$export$70ae4353602451dc = `SOkmSG_spectrum-Search-circleLoader`;
$1e139f06a0a4b696$export$50e02603f2d97ae1 = `SOkmSG_spectrum-Search-input`;
$1e139f06a0a4b696$export$938f0981825eae6e = `SOkmSG_spectrum-Search-validationIcon`;
$1e139f06a0a4b696$export$2c8a10299d8b3418 = `SOkmSG_spectrum-Textfield`;







function $111fe330b4c1f3ac$var$SearchField(props, ref) {
    props = (0, $kYnKt$reactspectrumutils.useSlotProps)(props, "searchfield");
    props = (0, $kYnKt$reactspectrumprovider.useProviderProps)(props);
    props = (0, $kYnKt$reactspectrumform.useFormProps)(props);
    let defaultIcon = /*#__PURE__*/ (0, ($parcel$interopDefault($kYnKt$react))).createElement((0, ($parcel$interopDefault($kYnKt$spectrumiconsuiMagnifier))), {
        "data-testid": "searchicon"
    });
    let { icon: icon = defaultIcon, isDisabled: isDisabled, UNSAFE_className: UNSAFE_className, placeholder: placeholder, ...otherProps } = props;
    if (placeholder) console.warn("Placeholders are deprecated due to accessibility issues. Please use help text instead. See the docs for details: https://react-spectrum.adobe.com/react-spectrum/SearchField.html#help-text");
    let state = (0, $kYnKt$reactstatelysearchfield.useSearchFieldState)(props);
    let inputRef = (0, $kYnKt$react.useRef)(null);
    let { clearButtonProps: clearButtonProps, ...result } = (0, $kYnKt$reactariasearchfield.useSearchField)(props, state, inputRef);
    let clearButton = /*#__PURE__*/ (0, ($parcel$interopDefault($kYnKt$react))).createElement((0, $kYnKt$reactspectrumbutton.ClearButton), {
        ...clearButtonProps,
        preventFocus: true,
        UNSAFE_className: (0, $kYnKt$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($1e139f06a0a4b696$exports))), "spectrum-ClearButton"),
        isDisabled: isDisabled
    });
    let validationState = props.validationState || (result.isInvalid ? "invalid" : undefined);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kYnKt$react))).createElement((0, $kYnKt$reactspectrumtextfield.TextFieldBase), {
        ...otherProps,
        ...result,
        validationState: validationState,
        UNSAFE_className: (0, $kYnKt$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($1e139f06a0a4b696$exports))), "spectrum-Search", "spectrum-Textfield", {
            "is-disabled": isDisabled,
            "is-quiet": props.isQuiet,
            "spectrum-Search--invalid": validationState === "invalid" && !isDisabled,
            "spectrum-Search--valid": validationState === "valid" && !isDisabled
        }, UNSAFE_className),
        inputClassName: (0, $kYnKt$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($1e139f06a0a4b696$exports))), "spectrum-Search-input"),
        ref: ref,
        inputRef: inputRef,
        isDisabled: isDisabled,
        icon: icon,
        wrapperChildren: state.value !== "" && !props.isReadOnly ? clearButton : undefined
    });
}
/**
 * A SearchField is a text field designed for searches.
 */ let $111fe330b4c1f3ac$export$b94867ecbd698f21 = /*#__PURE__*/ (0, $kYnKt$react.forwardRef)($111fe330b4c1f3ac$var$SearchField);
