// @react-spectrum/tag/dist/import.mjs
"use client";

import $4f97095b36611c9c$exports from "./ar-AE.mjs";
import $d1606ced019b0a9e$exports from "./bg-BG.mjs";
import $01e300f7d0078144$exports from "./cs-CZ.mjs";
import $f3862a5e5adf53a4$exports from "./da-DK.mjs";
import $74c0b6e11258b497$exports from "./de-DE.mjs";
import $54fc4e3090259431$exports from "./el-GR.mjs";
import $edf02907a7caaf96$exports from "./en-US.mjs";
import $b15ad3cf4b213b8e$exports from "./es-ES.mjs";
import $36cfda096c3db653$exports from "./et-EE.mjs";
import $1a42e74299cab64d$exports from "./fi-FI.mjs";
import $dd8d3f41a5bfc373$exports from "./fr-FR.mjs";
import $35d68da324c3ed3e$exports from "./he-IL.mjs";
import $92746ce1284cd29f$exports from "./hr-HR.mjs";
import $570956f4f59f34b9$exports from "./hu-HU.mjs";
import $f680d16d6f0d92d8$exports from "./it-IT.mjs";
import $d28c56f33dc08452$exports from "./ja-JP.mjs";
import $fc6f973338bd2d9c$exports from "./ko-KR.mjs";
import $8664f3f6a18cfaa6$exports from "./lt-LT.mjs";
import $bd6e0fb18b482534$exports from "./lv-LV.mjs";
import $3df66303c61fb25b$exports from "./nb-NO.mjs";
import $0f533b61cdf70176$exports from "./nl-NL.mjs";
import $03a6cd8f61b18787$exports from "./pl-PL.mjs";
import $c726a9d931f60c66$exports from "./pt-BR.mjs";
import $ae9982f17f7235eb$exports from "./pt-PT.mjs";
import $c756723b814c63fa$exports from "./ro-RO.mjs";
import $1601284ef6ec0730$exports from "./ru-RU.mjs";
import $0388aac8e3bc5b59$exports from "./sk-SK.mjs";
import $247d9c4a71da6b08$exports from "./sl-SI.mjs";
import $91424dbafdc02f68$exports from "./sr-SP.mjs";
import $23c23096b6c1b0fa$exports from "./sv-SE.mjs";
import $d82875ae28963e58$exports from "./tr-TR.mjs";
import $1dd153688f5574fc$exports from "./uk-UA.mjs";
import $6427307c9596258c$exports from "./zh-CN.mjs";
import $18b85da2aafaaf3e$exports from "./zh-TW.mjs";
import "./main.css";
import {Item as $c14698d57a91452e$re_export$Item} from "@react-stately/collections";
import {ActionButton as $fUaQ2$ActionButton, ClearButton as $fUaQ2$ClearButton} from "@react-spectrum/button";
import {useTagGroup as $fUaQ2$useTagGroup, useTag as $fUaQ2$useTag} from "@react-aria/tag";
import {useDOMRef as $fUaQ2$useDOMRef, classNames as $fUaQ2$classNames, useStyleProps as $fUaQ2$useStyleProps, SlotProvider as $fUaQ2$SlotProvider, ClearSlots as $fUaQ2$ClearSlots} from "@react-spectrum/utils";
import {Field as $fUaQ2$Field} from "@react-spectrum/label";
import {FocusScope as $fUaQ2$FocusScope, FocusRing as $fUaQ2$FocusRing, useFocusRing as $fUaQ2$useFocusRing} from "@react-aria/focus";
import {useListState as $fUaQ2$useListState, ListCollection as $fUaQ2$ListCollection} from "@react-stately/list";
import {ListKeyboardDelegate as $fUaQ2$ListKeyboardDelegate} from "@react-aria/selection";
import {useProviderProps as $fUaQ2$useProviderProps, useProvider as $fUaQ2$useProvider, Provider as $fUaQ2$Provider} from "@react-spectrum/provider";
import $fUaQ2$react, {useRef as $fUaQ2$useRef, useState as $fUaQ2$useState, useMemo as $fUaQ2$useMemo, useCallback as $fUaQ2$useCallback, useEffect as $fUaQ2$useEffect} from "react";
import {useFormProps as $fUaQ2$useFormProps} from "@react-spectrum/form";
import {useValueEffect as $fUaQ2$useValueEffect, useId as $fUaQ2$useId, useResizeObserver as $fUaQ2$useResizeObserver, useLayoutEffect as $fUaQ2$useLayoutEffect, mergeProps as $fUaQ2$mergeProps} from "@react-aria/utils";
import {useLocale as $fUaQ2$useLocale, useLocalizedStringFormatter as $fUaQ2$useLocalizedStringFormatter} from "@react-aria/i18n";
import {Text as $fUaQ2$Text} from "@react-spectrum/text";
import {useHover as $fUaQ2$useHover} from "@react-aria/interactions";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $fd5ee5726b29a200$exports = {};


































$fd5ee5726b29a200$exports = {
    "ar-AE": $4f97095b36611c9c$exports,
    "bg-BG": $d1606ced019b0a9e$exports,
    "cs-CZ": $01e300f7d0078144$exports,
    "da-DK": $f3862a5e5adf53a4$exports,
    "de-DE": $74c0b6e11258b497$exports,
    "el-GR": $54fc4e3090259431$exports,
    "en-US": $edf02907a7caaf96$exports,
    "es-ES": $b15ad3cf4b213b8e$exports,
    "et-EE": $36cfda096c3db653$exports,
    "fi-FI": $1a42e74299cab64d$exports,
    "fr-FR": $dd8d3f41a5bfc373$exports,
    "he-IL": $35d68da324c3ed3e$exports,
    "hr-HR": $92746ce1284cd29f$exports,
    "hu-HU": $570956f4f59f34b9$exports,
    "it-IT": $f680d16d6f0d92d8$exports,
    "ja-JP": $d28c56f33dc08452$exports,
    "ko-KR": $fc6f973338bd2d9c$exports,
    "lt-LT": $8664f3f6a18cfaa6$exports,
    "lv-LV": $bd6e0fb18b482534$exports,
    "nb-NO": $3df66303c61fb25b$exports,
    "nl-NL": $0f533b61cdf70176$exports,
    "pl-PL": $03a6cd8f61b18787$exports,
    "pt-BR": $c726a9d931f60c66$exports,
    "pt-PT": $ae9982f17f7235eb$exports,
    "ro-RO": $c756723b814c63fa$exports,
    "ru-RU": $1601284ef6ec0730$exports,
    "sk-SK": $0388aac8e3bc5b59$exports,
    "sl-SI": $247d9c4a71da6b08$exports,
    "sr-SP": $91424dbafdc02f68$exports,
    "sv-SE": $23c23096b6c1b0fa$exports,
    "tr-TR": $d82875ae28963e58$exports,
    "uk-UA": $1dd153688f5574fc$exports,
    "zh-CN": $6427307c9596258c$exports,
    "zh-TW": $18b85da2aafaaf3e$exports
};






var $c033a7c057e84ef1$exports = {};

$parcel$export($c033a7c057e84ef1$exports, "action", () => $c033a7c057e84ef1$export$31eb53bd5b582a97, (v) => $c033a7c057e84ef1$export$31eb53bd5b582a97 = v);
$parcel$export($c033a7c057e84ef1$exports, "content", () => $c033a7c057e84ef1$export$a7db06668cad9adb, (v) => $c033a7c057e84ef1$export$a7db06668cad9adb = v);
$parcel$export($c033a7c057e84ef1$exports, "decoration", () => $c033a7c057e84ef1$export$3454f895b9105e06, (v) => $c033a7c057e84ef1$export$3454f895b9105e06 = v);
$parcel$export($c033a7c057e84ef1$exports, "focus-ring", () => $c033a7c057e84ef1$export$f39a09f249340e2a, (v) => $c033a7c057e84ef1$export$f39a09f249340e2a = v);
$parcel$export($c033a7c057e84ef1$exports, "i18nFontFamily", () => $c033a7c057e84ef1$export$8c4ee2c50c22c514, (v) => $c033a7c057e84ef1$export$8c4ee2c50c22c514 = v);
$parcel$export($c033a7c057e84ef1$exports, "is-disabled", () => $c033a7c057e84ef1$export$d35bc1e505d1ebbf, (v) => $c033a7c057e84ef1$export$d35bc1e505d1ebbf = v);
$parcel$export($c033a7c057e84ef1$exports, "is-focused", () => $c033a7c057e84ef1$export$e7dc768d35940237, (v) => $c033a7c057e84ef1$export$e7dc768d35940237 = v);
$parcel$export($c033a7c057e84ef1$exports, "is-hovered", () => $c033a7c057e84ef1$export$b8813cd5d7824ce7, (v) => $c033a7c057e84ef1$export$b8813cd5d7824ce7 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-FocusRing-ring", () => $c033a7c057e84ef1$export$4109102f950813a6, (v) => $c033a7c057e84ef1$export$4109102f950813a6 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-FocusRing", () => $c033a7c057e84ef1$export$24c7f46a6e3605dd, (v) => $c033a7c057e84ef1$export$24c7f46a6e3605dd = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-FocusRing--quiet", () => $c033a7c057e84ef1$export$2927016961429360, (v) => $c033a7c057e84ef1$export$2927016961429360 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag", () => $c033a7c057e84ef1$export$6174bd1d20b4c022, (v) => $c033a7c057e84ef1$export$6174bd1d20b4c022 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag--removable", () => $c033a7c057e84ef1$export$87165794583be111, (v) => $c033a7c057e84ef1$export$87165794583be111 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-avatar", () => $c033a7c057e84ef1$export$7b57f5221a1b5306, (v) => $c033a7c057e84ef1$export$7b57f5221a1b5306 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-cell", () => $c033a7c057e84ef1$export$9c4fd4d7faada408, (v) => $c033a7c057e84ef1$export$9c4fd4d7faada408 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-content", () => $c033a7c057e84ef1$export$d281c967f4ee071e, (v) => $c033a7c057e84ef1$export$d281c967f4ee071e = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-icon", () => $c033a7c057e84ef1$export$21f5254139e14871, (v) => $c033a7c057e84ef1$export$21f5254139e14871 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tag-removeButton", () => $c033a7c057e84ef1$export$23a76b996f0a6c1, (v) => $c033a7c057e84ef1$export$23a76b996f0a6c1 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags", () => $c033a7c057e84ef1$export$eb7e80013a1212da, (v) => $c033a7c057e84ef1$export$eb7e80013a1212da = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-actionButton", () => $c033a7c057e84ef1$export$e6882f8db018e3bd, (v) => $c033a7c057e84ef1$export$e6882f8db018e3bd = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-actions", () => $c033a7c057e84ef1$export$8afaa02110915bbb, (v) => $c033a7c057e84ef1$export$8afaa02110915bbb = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-container", () => $c033a7c057e84ef1$export$3300169c5d27fe96, (v) => $c033a7c057e84ef1$export$3300169c5d27fe96 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-container--empty", () => $c033a7c057e84ef1$export$a5e59d283417e02b, (v) => $c033a7c057e84ef1$export$a5e59d283417e02b = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-empty-state", () => $c033a7c057e84ef1$export$cd48f925394bb16, (v) => $c033a7c057e84ef1$export$cd48f925394bb16 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-fieldWrapper", () => $c033a7c057e84ef1$export$c9b7ca327d69b2b5, (v) => $c033a7c057e84ef1$export$c9b7ca327d69b2b5 = v);
$parcel$export($c033a7c057e84ef1$exports, "spectrum-Tags-fieldWrapper--positionSide", () => $c033a7c057e84ef1$export$a7c11d4acc244e09, (v) => $c033a7c057e84ef1$export$a7c11d4acc244e09 = v);
var $c033a7c057e84ef1$export$31eb53bd5b582a97;
var $c033a7c057e84ef1$export$a7db06668cad9adb;
var $c033a7c057e84ef1$export$3454f895b9105e06;
var $c033a7c057e84ef1$export$f39a09f249340e2a;
var $c033a7c057e84ef1$export$8c4ee2c50c22c514;
var $c033a7c057e84ef1$export$d35bc1e505d1ebbf;
var $c033a7c057e84ef1$export$e7dc768d35940237;
var $c033a7c057e84ef1$export$b8813cd5d7824ce7;
var $c033a7c057e84ef1$export$4109102f950813a6;
var $c033a7c057e84ef1$export$24c7f46a6e3605dd;
var $c033a7c057e84ef1$export$2927016961429360;
var $c033a7c057e84ef1$export$6174bd1d20b4c022;
var $c033a7c057e84ef1$export$87165794583be111;
var $c033a7c057e84ef1$export$7b57f5221a1b5306;
var $c033a7c057e84ef1$export$9c4fd4d7faada408;
var $c033a7c057e84ef1$export$d281c967f4ee071e;
var $c033a7c057e84ef1$export$21f5254139e14871;
var $c033a7c057e84ef1$export$23a76b996f0a6c1;
var $c033a7c057e84ef1$export$eb7e80013a1212da;
var $c033a7c057e84ef1$export$e6882f8db018e3bd;
var $c033a7c057e84ef1$export$8afaa02110915bbb;
var $c033a7c057e84ef1$export$3300169c5d27fe96;
var $c033a7c057e84ef1$export$a5e59d283417e02b;
var $c033a7c057e84ef1$export$cd48f925394bb16;
var $c033a7c057e84ef1$export$c9b7ca327d69b2b5;
var $c033a7c057e84ef1$export$a7c11d4acc244e09;
$c033a7c057e84ef1$export$31eb53bd5b582a97 = `fx8d1q_action`;
$c033a7c057e84ef1$export$a7db06668cad9adb = `fx8d1q_content`;
$c033a7c057e84ef1$export$3454f895b9105e06 = `fx8d1q_decoration`;
$c033a7c057e84ef1$export$f39a09f249340e2a = `fx8d1q_focus-ring`;
$c033a7c057e84ef1$export$8c4ee2c50c22c514 = `fx8d1q_i18nFontFamily`;
$c033a7c057e84ef1$export$d35bc1e505d1ebbf = `fx8d1q_is-disabled`;
$c033a7c057e84ef1$export$e7dc768d35940237 = `fx8d1q_is-focused`;
$c033a7c057e84ef1$export$b8813cd5d7824ce7 = `fx8d1q_is-hovered`;
$c033a7c057e84ef1$export$4109102f950813a6 = `fx8d1q_spectrum-FocusRing-ring`;
$c033a7c057e84ef1$export$24c7f46a6e3605dd = `fx8d1q_spectrum-FocusRing ${$c033a7c057e84ef1$export$4109102f950813a6}`;
$c033a7c057e84ef1$export$2927016961429360 = `fx8d1q_spectrum-FocusRing--quiet`;
$c033a7c057e84ef1$export$6174bd1d20b4c022 = `fx8d1q_spectrum-Tag ${$c033a7c057e84ef1$export$24c7f46a6e3605dd}`;
$c033a7c057e84ef1$export$87165794583be111 = `fx8d1q_spectrum-Tag--removable`;
$c033a7c057e84ef1$export$7b57f5221a1b5306 = `fx8d1q_spectrum-Tag-avatar`;
$c033a7c057e84ef1$export$9c4fd4d7faada408 = `fx8d1q_spectrum-Tag-cell`;
$c033a7c057e84ef1$export$d281c967f4ee071e = `fx8d1q_spectrum-Tag-content`;
$c033a7c057e84ef1$export$21f5254139e14871 = `fx8d1q_spectrum-Tag-icon`;
$c033a7c057e84ef1$export$23a76b996f0a6c1 = `fx8d1q_spectrum-Tag-removeButton`;
$c033a7c057e84ef1$export$eb7e80013a1212da = `fx8d1q_spectrum-Tags`;
$c033a7c057e84ef1$export$e6882f8db018e3bd = `fx8d1q_spectrum-Tags-actionButton`;
$c033a7c057e84ef1$export$8afaa02110915bbb = `fx8d1q_spectrum-Tags-actions`;
$c033a7c057e84ef1$export$3300169c5d27fe96 = `fx8d1q_spectrum-Tags-container`;
$c033a7c057e84ef1$export$a5e59d283417e02b = `fx8d1q_spectrum-Tags-container--empty`;
$c033a7c057e84ef1$export$cd48f925394bb16 = `fx8d1q_spectrum-Tags-empty-state`;
$c033a7c057e84ef1$export$c9b7ca327d69b2b5 = `fx8d1q_spectrum-Tags-fieldWrapper`;
$c033a7c057e84ef1$export$a7c11d4acc244e09 = `fx8d1q_spectrum-Tags-fieldWrapper--positionSide`;


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 








function $b5f5ee35d3b6b95c$export$3288d34c523a1192(props) {
    const { item: item, state: state, ...otherProps } = props;
    // @ts-ignore
    let { styleProps: styleProps } = (0, $fUaQ2$useStyleProps)(otherProps);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $fUaQ2$useHover)({});
    let { isFocused: isFocused, isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $fUaQ2$useFocusRing)({
        within: true
    });
    let ref = (0, $fUaQ2$useRef)(null);
    let { removeButtonProps: removeButtonProps, gridCellProps: gridCellProps, rowProps: rowProps, allowsRemoving: allowsRemoving } = (0, $fUaQ2$useTag)({
        ...props,
        item: item
    }, state, ref);
    return /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        ...(0, $fUaQ2$mergeProps)(rowProps, hoverProps, focusProps),
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag", {
            "focus-ring": isFocusVisible,
            "is-focused": isFocused,
            "is-hovered": isHovered,
            "spectrum-Tag--removable": allowsRemoving
        }, styleProps.className),
        ref: ref
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-cell"),
        ...gridCellProps
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$SlotProvider), {
        slots: {
            icon: {
                UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-icon"),
                size: "XS"
            },
            text: {
                UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-content")
            },
            avatar: {
                UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-avatar"),
                size: "avatar-size-50"
            }
        }
    }, typeof item.rendered === "string" ? /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$Text), null, item.rendered) : item.rendered, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$ClearSlots), null, allowsRemoving && /*#__PURE__*/ (0, $fUaQ2$react).createElement($b5f5ee35d3b6b95c$var$TagRemoveButton, {
        item: item,
        ...removeButtonProps,
        UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tag-removeButton")
    })))));
}
function $b5f5ee35d3b6b95c$var$TagRemoveButton(props) {
    let { styleProps: styleProps } = (0, $fUaQ2$useStyleProps)(props);
    return /*#__PURE__*/ (0, $fUaQ2$react).createElement("span", styleProps, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$ClearButton), props));
}





const $880d112c7a588209$var$TAG_STYLES = {
    medium: {
        height: 24,
        margin: 4
    },
    large: {
        height: 30,
        margin: 5
    }
};
function $880d112c7a588209$var$TagGroup(props, ref) {
    props = (0, $fUaQ2$useProviderProps)(props);
    props = (0, $fUaQ2$useFormProps)(props);
    let { maxRows: maxRows, children: children, actionLabel: actionLabel, onAction: onAction, labelPosition: labelPosition, renderEmptyState: renderEmptyState = ()=>stringFormatter.format("noTags") } = props;
    let domRef = (0, $fUaQ2$useDOMRef)(ref);
    let containerRef = (0, $fUaQ2$useRef)(null);
    let tagsRef = (0, $fUaQ2$useRef)(null);
    let { direction: direction } = (0, $fUaQ2$useLocale)();
    let { scale: scale } = (0, $fUaQ2$useProvider)();
    let stringFormatter = (0, $fUaQ2$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($fd5ee5726b29a200$exports))), "@react-spectrum/tag");
    let [isCollapsed, setIsCollapsed] = (0, $fUaQ2$useState)(maxRows != null);
    let state = (0, $fUaQ2$useListState)(props);
    let [tagState, setTagState] = (0, $fUaQ2$useValueEffect)({
        visibleTagCount: state.collection.size,
        showCollapseButton: false
    });
    let keyboardDelegate = (0, $fUaQ2$useMemo)(()=>{
        let collection = isCollapsed ? new (0, $fUaQ2$ListCollection)([
            ...state.collection
        ].slice(0, tagState.visibleTagCount)) : new (0, $fUaQ2$ListCollection)([
            ...state.collection
        ]);
        return new (0, $fUaQ2$ListKeyboardDelegate)({
            collection: collection,
            ref: domRef,
            direction: direction,
            orientation: "horizontal"
        });
    }, [
        direction,
        isCollapsed,
        state.collection,
        tagState.visibleTagCount,
        domRef
    ]);
    // Remove onAction from props so it doesn't make it into useGridList.
    delete props.onAction;
    let { gridProps: gridProps, labelProps: labelProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps } = (0, $fUaQ2$useTagGroup)({
        ...props,
        keyboardDelegate: keyboardDelegate
    }, state, tagsRef);
    let actionsId = (0, $fUaQ2$useId)();
    let actionsRef = (0, $fUaQ2$useRef)(null);
    let updateVisibleTagCount = (0, $fUaQ2$useCallback)(()=>{
        if (maxRows && maxRows > 0) {
            let computeVisibleTagCount = ()=>{
                // Refs can be null at runtime.
                let currContainerRef = containerRef.current;
                let currTagsRef = tagsRef.current;
                let currActionsRef = actionsRef.current;
                if (!currContainerRef || !currTagsRef || !currActionsRef || state.collection.size === 0) return {
                    visibleTagCount: 0,
                    showCollapseButton: false
                };
                // Count rows and show tags until we hit the maxRows.
                let tags = [
                    ...currTagsRef.children
                ];
                let currY = -Infinity;
                let rowCount = 0;
                let index = 0;
                let tagWidths = [];
                for (let tag of tags){
                    let { width: width, y: y } = tag.getBoundingClientRect();
                    if (y !== currY) {
                        currY = y;
                        rowCount++;
                    }
                    if (maxRows && rowCount > maxRows) break;
                    tagWidths.push(width);
                    index++;
                }
                // Remove tags until there is space for the collapse button and action button (if present) on the last row.
                let buttons = [
                    ...currActionsRef.children
                ];
                if (maxRows && buttons.length > 0 && rowCount >= maxRows) {
                    var _tags_;
                    let buttonsWidth = buttons.reduce((acc, curr)=>acc += curr.getBoundingClientRect().width, 0);
                    buttonsWidth += $880d112c7a588209$var$TAG_STYLES[scale].margin * 2 * buttons.length;
                    let end = direction === "ltr" ? "right" : "left";
                    let containerEnd = currContainerRef.parentElement.getBoundingClientRect()[end];
                    let lastTagEnd = (_tags_ = tags[index - 1]) === null || _tags_ === void 0 ? void 0 : _tags_.getBoundingClientRect()[end];
                    lastTagEnd += $880d112c7a588209$var$TAG_STYLES[scale].margin;
                    let availableWidth = containerEnd - lastTagEnd;
                    while(availableWidth < buttonsWidth && index > 0){
                        availableWidth += tagWidths.pop();
                        index--;
                    }
                }
                return {
                    visibleTagCount: Math.max(index, 1),
                    showCollapseButton: index < state.collection.size
                };
            };
            setTagState(function*() {
                // Update to show all items.
                yield {
                    visibleTagCount: state.collection.size,
                    showCollapseButton: true
                };
                // Measure, and update to show the items until maxRows is reached.
                yield computeVisibleTagCount();
            });
        }
    }, [
        maxRows,
        setTagState,
        direction,
        scale,
        state.collection.size
    ]);
    (0, $fUaQ2$useResizeObserver)({
        ref: containerRef,
        onResize: updateVisibleTagCount
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
    (0, $fUaQ2$useLayoutEffect)(updateVisibleTagCount, [
        children
    ]);
    (0, $fUaQ2$useEffect)(()=>{
        var // Recalculate visible tags when fonts are loaded.
        _document_fonts;
        (_document_fonts = document.fonts) === null || _document_fonts === void 0 ? void 0 : _document_fonts.ready.then(()=>updateVisibleTagCount());
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    let visibleTags = (0, $fUaQ2$useMemo)(()=>[
            ...state.collection
        ].slice(0, isCollapsed ? tagState.visibleTagCount : state.collection.size), [
        isCollapsed,
        state.collection,
        tagState.visibleTagCount
    ]);
    let handlePressCollapse = ()=>{
        // Prevents button from losing focus if focusedKey got collapsed.
        state.selectionManager.setFocusedKey(null);
        setIsCollapsed((prevCollapsed)=>!prevCollapsed);
    };
    let showActions = tagState.showCollapseButton || actionLabel && onAction;
    let isEmpty = state.collection.size === 0;
    let containerStyle = (0, $fUaQ2$useMemo)(()=>{
        if (maxRows == null || !isCollapsed || isEmpty) return undefined;
        let maxHeight = ($880d112c7a588209$var$TAG_STYLES[scale].height + $880d112c7a588209$var$TAG_STYLES[scale].margin * 2) * maxRows;
        return {
            maxHeight: maxHeight,
            overflow: "hidden"
        };
    }, [
        isCollapsed,
        maxRows,
        isEmpty,
        scale
    ]);
    return /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$FocusScope), null, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$Field), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: true,
        ref: domRef,
        elementType: "span",
        wrapperClassName: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-fieldWrapper", {
            "spectrum-Tags-fieldWrapper--positionSide": labelPosition === "side"
        })
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        ref: containerRef,
        style: containerStyle,
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-container", {
            "spectrum-Tags-container--empty": isEmpty
        })
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$FocusRing), {
        focusRingClass: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        ref: tagsRef,
        ...gridProps,
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags")
    }, visibleTags.map((item)=>/*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $b5f5ee35d3b6b95c$export$3288d34c523a1192), {
            ...item.props,
            key: item.key,
            item: item,
            state: state
        }, item.rendered)), isEmpty && /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-empty-state")
    }, renderEmptyState()))), showActions && !isEmpty && /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$Provider), {
        isDisabled: false
    }, /*#__PURE__*/ (0, $fUaQ2$react).createElement("div", {
        role: "group",
        ref: actionsRef,
        id: actionsId,
        "aria-label": stringFormatter.format("actions"),
        "aria-labelledby": `${gridProps.id} ${actionsId}`,
        className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-actions")
    }, tagState.showCollapseButton && /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$ActionButton), {
        isQuiet: true,
        onPress: handlePressCollapse,
        UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-actionButton")
    }, isCollapsed ? stringFormatter.format("showAllButtonLabel", {
        tagCount: state.collection.size
    }) : stringFormatter.format("hideButtonLabel")), actionLabel && onAction && /*#__PURE__*/ (0, $fUaQ2$react).createElement((0, $fUaQ2$ActionButton), {
        isQuiet: true,
        onPress: ()=>onAction === null || onAction === void 0 ? void 0 : onAction(),
        UNSAFE_className: (0, $fUaQ2$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c033a7c057e84ef1$exports))), "spectrum-Tags-actionButton")
    }, actionLabel))))));
}
/** Tags allow users to categorize content. They can represent keywords or people, and are grouped to describe an item or a search request. */ const $880d112c7a588209$export$67ea30858aaf75e3 = /*#__PURE__*/ (0, $fUaQ2$react).forwardRef($880d112c7a588209$var$TagGroup);





export {$880d112c7a588209$export$67ea30858aaf75e3 as TagGroup, $c14698d57a91452e$re_export$Item as Item};
