// @react-spectrum/tag/dist/main.js
"use client";

var $e4833c69a6a8a469$exports = require("./ar-AE.main.js");
var $9b536cac3e0b637e$exports = require("./bg-BG.main.js");
var $06ebf46bd7a639dd$exports = require("./cs-CZ.main.js");
var $d0c8889aa143e531$exports = require("./da-DK.main.js");
var $09f15015489dc476$exports = require("./de-DE.main.js");
var $ec81f332ee1bb67a$exports = require("./el-GR.main.js");
var $a5d4273001869f16$exports = require("./en-US.main.js");
var $355c19ee0d058106$exports = require("./es-ES.main.js");
var $94d247b24eb969ef$exports = require("./et-EE.main.js");
var $fe04eda9b3e92668$exports = require("./fi-FI.main.js");
var $8c0ca11f9a10e544$exports = require("./fr-FR.main.js");
var $5cf885b76293f7ea$exports = require("./he-IL.main.js");
var $26efb27e85fe3886$exports = require("./hr-HR.main.js");
var $5afbffe322be4f78$exports = require("./hu-HU.main.js");
var $e2e8477c50691c5a$exports = require("./it-IT.main.js");
var $aa322e153c7fc6cf$exports = require("./ja-JP.main.js");
var $ef5171b4e7fef8b6$exports = require("./ko-KR.main.js");
var $11cf3f21588c59d5$exports = require("./lt-LT.main.js");
var $7798af2d0360d513$exports = require("./lv-LV.main.js");
var $36fbb928ae891fff$exports = require("./nb-NO.main.js");
var $64f658e8416b1edc$exports = require("./nl-NL.main.js");
var $d83dc3596c86b9b1$exports = require("./pl-PL.main.js");
var $f05393b8526809a5$exports = require("./pt-BR.main.js");
var $d7b3c6e34da453ec$exports = require("./pt-PT.main.js");
var $671c8a587799a566$exports = require("./ro-RO.main.js");
var $f7d5f2f6333669ae$exports = require("./ru-RU.main.js");
var $6872c5a06d3d5bd6$exports = require("./sk-SK.main.js");
var $8d44aaec5d910562$exports = require("./sl-SI.main.js");
var $48a1f90aac47b13f$exports = require("./sr-SP.main.js");
var $e67f8d08ca5a16ce$exports = require("./sv-SE.main.js");
var $dfe98dff52c184ea$exports = require("./tr-TR.main.js");
var $ba516e0f6c423a4a$exports = require("./uk-UA.main.js");
var $7c2f3cb525451680$exports = require("./zh-CN.main.js");
var $00b7e0167640331e$exports = require("./zh-TW.main.js");
require("./main.css");
var $dTV3V$reactstatelycollections = require("@react-stately/collections");
var $dTV3V$reactspectrumbutton = require("@react-spectrum/button");
var $dTV3V$reactariatag = require("@react-aria/tag");
var $dTV3V$reactspectrumutils = require("@react-spectrum/utils");
var $dTV3V$reactspectrumlabel = require("@react-spectrum/label");
var $dTV3V$reactariafocus = require("@react-aria/focus");
var $dTV3V$reactstatelylist = require("@react-stately/list");
var $dTV3V$reactariaselection = require("@react-aria/selection");
var $dTV3V$reactspectrumprovider = require("@react-spectrum/provider");
var $dTV3V$react = require("react");
var $dTV3V$reactspectrumform = require("@react-spectrum/form");
var $dTV3V$reactariautils = require("@react-aria/utils");
var $dTV3V$reactariai18n = require("@react-aria/i18n");
var $dTV3V$reactspectrumtext = require("@react-spectrum/text");
var $dTV3V$reactariainteractions = require("@react-aria/interactions");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "TagGroup", () => $e7b72d8874e98cd4$export$67ea30858aaf75e3);
$parcel$export(module.exports, "Item", () => $dTV3V$reactstatelycollections.Item);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $a85da3418799ba7d$exports = {};


































$a85da3418799ba7d$exports = {
    "ar-AE": $e4833c69a6a8a469$exports,
    "bg-BG": $9b536cac3e0b637e$exports,
    "cs-CZ": $06ebf46bd7a639dd$exports,
    "da-DK": $d0c8889aa143e531$exports,
    "de-DE": $09f15015489dc476$exports,
    "el-GR": $ec81f332ee1bb67a$exports,
    "en-US": $a5d4273001869f16$exports,
    "es-ES": $355c19ee0d058106$exports,
    "et-EE": $94d247b24eb969ef$exports,
    "fi-FI": $fe04eda9b3e92668$exports,
    "fr-FR": $8c0ca11f9a10e544$exports,
    "he-IL": $5cf885b76293f7ea$exports,
    "hr-HR": $26efb27e85fe3886$exports,
    "hu-HU": $5afbffe322be4f78$exports,
    "it-IT": $e2e8477c50691c5a$exports,
    "ja-JP": $aa322e153c7fc6cf$exports,
    "ko-KR": $ef5171b4e7fef8b6$exports,
    "lt-LT": $11cf3f21588c59d5$exports,
    "lv-LV": $7798af2d0360d513$exports,
    "nb-NO": $36fbb928ae891fff$exports,
    "nl-NL": $64f658e8416b1edc$exports,
    "pl-PL": $d83dc3596c86b9b1$exports,
    "pt-BR": $f05393b8526809a5$exports,
    "pt-PT": $d7b3c6e34da453ec$exports,
    "ro-RO": $671c8a587799a566$exports,
    "ru-RU": $f7d5f2f6333669ae$exports,
    "sk-SK": $6872c5a06d3d5bd6$exports,
    "sl-SI": $8d44aaec5d910562$exports,
    "sr-SP": $48a1f90aac47b13f$exports,
    "sv-SE": $e67f8d08ca5a16ce$exports,
    "tr-TR": $dfe98dff52c184ea$exports,
    "uk-UA": $ba516e0f6c423a4a$exports,
    "zh-CN": $7c2f3cb525451680$exports,
    "zh-TW": $00b7e0167640331e$exports
};






var $ed889f8f1f091fc8$exports = {};

$parcel$export($ed889f8f1f091fc8$exports, "action", () => $ed889f8f1f091fc8$export$31eb53bd5b582a97, (v) => $ed889f8f1f091fc8$export$31eb53bd5b582a97 = v);
$parcel$export($ed889f8f1f091fc8$exports, "content", () => $ed889f8f1f091fc8$export$a7db06668cad9adb, (v) => $ed889f8f1f091fc8$export$a7db06668cad9adb = v);
$parcel$export($ed889f8f1f091fc8$exports, "decoration", () => $ed889f8f1f091fc8$export$3454f895b9105e06, (v) => $ed889f8f1f091fc8$export$3454f895b9105e06 = v);
$parcel$export($ed889f8f1f091fc8$exports, "focus-ring", () => $ed889f8f1f091fc8$export$f39a09f249340e2a, (v) => $ed889f8f1f091fc8$export$f39a09f249340e2a = v);
$parcel$export($ed889f8f1f091fc8$exports, "i18nFontFamily", () => $ed889f8f1f091fc8$export$8c4ee2c50c22c514, (v) => $ed889f8f1f091fc8$export$8c4ee2c50c22c514 = v);
$parcel$export($ed889f8f1f091fc8$exports, "is-disabled", () => $ed889f8f1f091fc8$export$d35bc1e505d1ebbf, (v) => $ed889f8f1f091fc8$export$d35bc1e505d1ebbf = v);
$parcel$export($ed889f8f1f091fc8$exports, "is-focused", () => $ed889f8f1f091fc8$export$e7dc768d35940237, (v) => $ed889f8f1f091fc8$export$e7dc768d35940237 = v);
$parcel$export($ed889f8f1f091fc8$exports, "is-hovered", () => $ed889f8f1f091fc8$export$b8813cd5d7824ce7, (v) => $ed889f8f1f091fc8$export$b8813cd5d7824ce7 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-FocusRing-ring", () => $ed889f8f1f091fc8$export$4109102f950813a6, (v) => $ed889f8f1f091fc8$export$4109102f950813a6 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-FocusRing", () => $ed889f8f1f091fc8$export$24c7f46a6e3605dd, (v) => $ed889f8f1f091fc8$export$24c7f46a6e3605dd = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-FocusRing--quiet", () => $ed889f8f1f091fc8$export$2927016961429360, (v) => $ed889f8f1f091fc8$export$2927016961429360 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tag", () => $ed889f8f1f091fc8$export$6174bd1d20b4c022, (v) => $ed889f8f1f091fc8$export$6174bd1d20b4c022 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tag--removable", () => $ed889f8f1f091fc8$export$87165794583be111, (v) => $ed889f8f1f091fc8$export$87165794583be111 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tag-avatar", () => $ed889f8f1f091fc8$export$7b57f5221a1b5306, (v) => $ed889f8f1f091fc8$export$7b57f5221a1b5306 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tag-cell", () => $ed889f8f1f091fc8$export$9c4fd4d7faada408, (v) => $ed889f8f1f091fc8$export$9c4fd4d7faada408 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tag-content", () => $ed889f8f1f091fc8$export$d281c967f4ee071e, (v) => $ed889f8f1f091fc8$export$d281c967f4ee071e = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tag-icon", () => $ed889f8f1f091fc8$export$21f5254139e14871, (v) => $ed889f8f1f091fc8$export$21f5254139e14871 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tag-removeButton", () => $ed889f8f1f091fc8$export$23a76b996f0a6c1, (v) => $ed889f8f1f091fc8$export$23a76b996f0a6c1 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tags", () => $ed889f8f1f091fc8$export$eb7e80013a1212da, (v) => $ed889f8f1f091fc8$export$eb7e80013a1212da = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tags-actionButton", () => $ed889f8f1f091fc8$export$e6882f8db018e3bd, (v) => $ed889f8f1f091fc8$export$e6882f8db018e3bd = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tags-actions", () => $ed889f8f1f091fc8$export$8afaa02110915bbb, (v) => $ed889f8f1f091fc8$export$8afaa02110915bbb = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tags-container", () => $ed889f8f1f091fc8$export$3300169c5d27fe96, (v) => $ed889f8f1f091fc8$export$3300169c5d27fe96 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tags-container--empty", () => $ed889f8f1f091fc8$export$a5e59d283417e02b, (v) => $ed889f8f1f091fc8$export$a5e59d283417e02b = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tags-empty-state", () => $ed889f8f1f091fc8$export$cd48f925394bb16, (v) => $ed889f8f1f091fc8$export$cd48f925394bb16 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tags-fieldWrapper", () => $ed889f8f1f091fc8$export$c9b7ca327d69b2b5, (v) => $ed889f8f1f091fc8$export$c9b7ca327d69b2b5 = v);
$parcel$export($ed889f8f1f091fc8$exports, "spectrum-Tags-fieldWrapper--positionSide", () => $ed889f8f1f091fc8$export$a7c11d4acc244e09, (v) => $ed889f8f1f091fc8$export$a7c11d4acc244e09 = v);
var $ed889f8f1f091fc8$export$31eb53bd5b582a97;
var $ed889f8f1f091fc8$export$a7db06668cad9adb;
var $ed889f8f1f091fc8$export$3454f895b9105e06;
var $ed889f8f1f091fc8$export$f39a09f249340e2a;
var $ed889f8f1f091fc8$export$8c4ee2c50c22c514;
var $ed889f8f1f091fc8$export$d35bc1e505d1ebbf;
var $ed889f8f1f091fc8$export$e7dc768d35940237;
var $ed889f8f1f091fc8$export$b8813cd5d7824ce7;
var $ed889f8f1f091fc8$export$4109102f950813a6;
var $ed889f8f1f091fc8$export$24c7f46a6e3605dd;
var $ed889f8f1f091fc8$export$2927016961429360;
var $ed889f8f1f091fc8$export$6174bd1d20b4c022;
var $ed889f8f1f091fc8$export$87165794583be111;
var $ed889f8f1f091fc8$export$7b57f5221a1b5306;
var $ed889f8f1f091fc8$export$9c4fd4d7faada408;
var $ed889f8f1f091fc8$export$d281c967f4ee071e;
var $ed889f8f1f091fc8$export$21f5254139e14871;
var $ed889f8f1f091fc8$export$23a76b996f0a6c1;
var $ed889f8f1f091fc8$export$eb7e80013a1212da;
var $ed889f8f1f091fc8$export$e6882f8db018e3bd;
var $ed889f8f1f091fc8$export$8afaa02110915bbb;
var $ed889f8f1f091fc8$export$3300169c5d27fe96;
var $ed889f8f1f091fc8$export$a5e59d283417e02b;
var $ed889f8f1f091fc8$export$cd48f925394bb16;
var $ed889f8f1f091fc8$export$c9b7ca327d69b2b5;
var $ed889f8f1f091fc8$export$a7c11d4acc244e09;
$ed889f8f1f091fc8$export$31eb53bd5b582a97 = `fx8d1q_action`;
$ed889f8f1f091fc8$export$a7db06668cad9adb = `fx8d1q_content`;
$ed889f8f1f091fc8$export$3454f895b9105e06 = `fx8d1q_decoration`;
$ed889f8f1f091fc8$export$f39a09f249340e2a = `fx8d1q_focus-ring`;
$ed889f8f1f091fc8$export$8c4ee2c50c22c514 = `fx8d1q_i18nFontFamily`;
$ed889f8f1f091fc8$export$d35bc1e505d1ebbf = `fx8d1q_is-disabled`;
$ed889f8f1f091fc8$export$e7dc768d35940237 = `fx8d1q_is-focused`;
$ed889f8f1f091fc8$export$b8813cd5d7824ce7 = `fx8d1q_is-hovered`;
$ed889f8f1f091fc8$export$4109102f950813a6 = `fx8d1q_spectrum-FocusRing-ring`;
$ed889f8f1f091fc8$export$24c7f46a6e3605dd = `fx8d1q_spectrum-FocusRing ${$ed889f8f1f091fc8$export$4109102f950813a6}`;
$ed889f8f1f091fc8$export$2927016961429360 = `fx8d1q_spectrum-FocusRing--quiet`;
$ed889f8f1f091fc8$export$6174bd1d20b4c022 = `fx8d1q_spectrum-Tag ${$ed889f8f1f091fc8$export$24c7f46a6e3605dd}`;
$ed889f8f1f091fc8$export$87165794583be111 = `fx8d1q_spectrum-Tag--removable`;
$ed889f8f1f091fc8$export$7b57f5221a1b5306 = `fx8d1q_spectrum-Tag-avatar`;
$ed889f8f1f091fc8$export$9c4fd4d7faada408 = `fx8d1q_spectrum-Tag-cell`;
$ed889f8f1f091fc8$export$d281c967f4ee071e = `fx8d1q_spectrum-Tag-content`;
$ed889f8f1f091fc8$export$21f5254139e14871 = `fx8d1q_spectrum-Tag-icon`;
$ed889f8f1f091fc8$export$23a76b996f0a6c1 = `fx8d1q_spectrum-Tag-removeButton`;
$ed889f8f1f091fc8$export$eb7e80013a1212da = `fx8d1q_spectrum-Tags`;
$ed889f8f1f091fc8$export$e6882f8db018e3bd = `fx8d1q_spectrum-Tags-actionButton`;
$ed889f8f1f091fc8$export$8afaa02110915bbb = `fx8d1q_spectrum-Tags-actions`;
$ed889f8f1f091fc8$export$3300169c5d27fe96 = `fx8d1q_spectrum-Tags-container`;
$ed889f8f1f091fc8$export$a5e59d283417e02b = `fx8d1q_spectrum-Tags-container--empty`;
$ed889f8f1f091fc8$export$cd48f925394bb16 = `fx8d1q_spectrum-Tags-empty-state`;
$ed889f8f1f091fc8$export$c9b7ca327d69b2b5 = `fx8d1q_spectrum-Tags-fieldWrapper`;
$ed889f8f1f091fc8$export$a7c11d4acc244e09 = `fx8d1q_spectrum-Tags-fieldWrapper--positionSide`;


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 








function $dd2c5d00108c1840$export$3288d34c523a1192(props) {
    const { item: item, state: state, ...otherProps } = props;
    // @ts-ignore
    let { styleProps: styleProps } = (0, $dTV3V$reactspectrumutils.useStyleProps)(otherProps);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $dTV3V$reactariainteractions.useHover)({});
    let { isFocused: isFocused, isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $dTV3V$reactariafocus.useFocusRing)({
        within: true
    });
    let ref = (0, $dTV3V$react.useRef)(null);
    let { removeButtonProps: removeButtonProps, gridCellProps: gridCellProps, rowProps: rowProps, allowsRemoving: allowsRemoving } = (0, $dTV3V$reactariatag.useTag)({
        ...props,
        item: item
    }, state, ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement("div", {
        ...(0, $dTV3V$reactariautils.mergeProps)(rowProps, hoverProps, focusProps),
        className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tag", {
            "focus-ring": isFocusVisible,
            "is-focused": isFocused,
            "is-hovered": isHovered,
            "spectrum-Tag--removable": allowsRemoving
        }, styleProps.className),
        ref: ref
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement("div", {
        className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tag-cell"),
        ...gridCellProps
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactspectrumutils.SlotProvider), {
        slots: {
            icon: {
                UNSAFE_className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tag-icon"),
                size: "XS"
            },
            text: {
                UNSAFE_className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tag-content")
            },
            avatar: {
                UNSAFE_className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tag-avatar"),
                size: "avatar-size-50"
            }
        }
    }, typeof item.rendered === "string" ? /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactspectrumtext.Text), null, item.rendered) : item.rendered, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactspectrumutils.ClearSlots), null, allowsRemoving && /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement($dd2c5d00108c1840$var$TagRemoveButton, {
        item: item,
        ...removeButtonProps,
        UNSAFE_className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tag-removeButton")
    })))));
}
function $dd2c5d00108c1840$var$TagRemoveButton(props) {
    let { styleProps: styleProps } = (0, $dTV3V$reactspectrumutils.useStyleProps)(props);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement("span", styleProps, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactspectrumbutton.ClearButton), props));
}





const $e7b72d8874e98cd4$var$TAG_STYLES = {
    medium: {
        height: 24,
        margin: 4
    },
    large: {
        height: 30,
        margin: 5
    }
};
function $e7b72d8874e98cd4$var$TagGroup(props, ref) {
    props = (0, $dTV3V$reactspectrumprovider.useProviderProps)(props);
    props = (0, $dTV3V$reactspectrumform.useFormProps)(props);
    let { maxRows: maxRows, children: children, actionLabel: actionLabel, onAction: onAction, labelPosition: labelPosition, renderEmptyState: renderEmptyState = ()=>stringFormatter.format("noTags") } = props;
    let domRef = (0, $dTV3V$reactspectrumutils.useDOMRef)(ref);
    let containerRef = (0, $dTV3V$react.useRef)(null);
    let tagsRef = (0, $dTV3V$react.useRef)(null);
    let { direction: direction } = (0, $dTV3V$reactariai18n.useLocale)();
    let { scale: scale } = (0, $dTV3V$reactspectrumprovider.useProvider)();
    let stringFormatter = (0, $dTV3V$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($a85da3418799ba7d$exports))), "@react-spectrum/tag");
    let [isCollapsed, setIsCollapsed] = (0, $dTV3V$react.useState)(maxRows != null);
    let state = (0, $dTV3V$reactstatelylist.useListState)(props);
    let [tagState, setTagState] = (0, $dTV3V$reactariautils.useValueEffect)({
        visibleTagCount: state.collection.size,
        showCollapseButton: false
    });
    let keyboardDelegate = (0, $dTV3V$react.useMemo)(()=>{
        let collection = isCollapsed ? new (0, $dTV3V$reactstatelylist.ListCollection)([
            ...state.collection
        ].slice(0, tagState.visibleTagCount)) : new (0, $dTV3V$reactstatelylist.ListCollection)([
            ...state.collection
        ]);
        return new (0, $dTV3V$reactariaselection.ListKeyboardDelegate)({
            collection: collection,
            ref: domRef,
            direction: direction,
            orientation: "horizontal"
        });
    }, [
        direction,
        isCollapsed,
        state.collection,
        tagState.visibleTagCount,
        domRef
    ]);
    // Remove onAction from props so it doesn't make it into useGridList.
    delete props.onAction;
    let { gridProps: gridProps, labelProps: labelProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps } = (0, $dTV3V$reactariatag.useTagGroup)({
        ...props,
        keyboardDelegate: keyboardDelegate
    }, state, tagsRef);
    let actionsId = (0, $dTV3V$reactariautils.useId)();
    let actionsRef = (0, $dTV3V$react.useRef)(null);
    let updateVisibleTagCount = (0, $dTV3V$react.useCallback)(()=>{
        if (maxRows && maxRows > 0) {
            let computeVisibleTagCount = ()=>{
                // Refs can be null at runtime.
                let currContainerRef = containerRef.current;
                let currTagsRef = tagsRef.current;
                let currActionsRef = actionsRef.current;
                if (!currContainerRef || !currTagsRef || !currActionsRef || state.collection.size === 0) return {
                    visibleTagCount: 0,
                    showCollapseButton: false
                };
                // Count rows and show tags until we hit the maxRows.
                let tags = [
                    ...currTagsRef.children
                ];
                let currY = -Infinity;
                let rowCount = 0;
                let index = 0;
                let tagWidths = [];
                for (let tag of tags){
                    let { width: width, y: y } = tag.getBoundingClientRect();
                    if (y !== currY) {
                        currY = y;
                        rowCount++;
                    }
                    if (maxRows && rowCount > maxRows) break;
                    tagWidths.push(width);
                    index++;
                }
                // Remove tags until there is space for the collapse button and action button (if present) on the last row.
                let buttons = [
                    ...currActionsRef.children
                ];
                if (maxRows && buttons.length > 0 && rowCount >= maxRows) {
                    var _tags_;
                    let buttonsWidth = buttons.reduce((acc, curr)=>acc += curr.getBoundingClientRect().width, 0);
                    buttonsWidth += $e7b72d8874e98cd4$var$TAG_STYLES[scale].margin * 2 * buttons.length;
                    let end = direction === "ltr" ? "right" : "left";
                    let containerEnd = currContainerRef.parentElement.getBoundingClientRect()[end];
                    let lastTagEnd = (_tags_ = tags[index - 1]) === null || _tags_ === void 0 ? void 0 : _tags_.getBoundingClientRect()[end];
                    lastTagEnd += $e7b72d8874e98cd4$var$TAG_STYLES[scale].margin;
                    let availableWidth = containerEnd - lastTagEnd;
                    while(availableWidth < buttonsWidth && index > 0){
                        availableWidth += tagWidths.pop();
                        index--;
                    }
                }
                return {
                    visibleTagCount: Math.max(index, 1),
                    showCollapseButton: index < state.collection.size
                };
            };
            setTagState(function*() {
                // Update to show all items.
                yield {
                    visibleTagCount: state.collection.size,
                    showCollapseButton: true
                };
                // Measure, and update to show the items until maxRows is reached.
                yield computeVisibleTagCount();
            });
        }
    }, [
        maxRows,
        setTagState,
        direction,
        scale,
        state.collection.size
    ]);
    (0, $dTV3V$reactariautils.useResizeObserver)({
        ref: containerRef,
        onResize: updateVisibleTagCount
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
    (0, $dTV3V$reactariautils.useLayoutEffect)(updateVisibleTagCount, [
        children
    ]);
    (0, $dTV3V$react.useEffect)(()=>{
        var // Recalculate visible tags when fonts are loaded.
        _document_fonts;
        (_document_fonts = document.fonts) === null || _document_fonts === void 0 ? void 0 : _document_fonts.ready.then(()=>updateVisibleTagCount());
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
    let visibleTags = (0, $dTV3V$react.useMemo)(()=>[
            ...state.collection
        ].slice(0, isCollapsed ? tagState.visibleTagCount : state.collection.size), [
        isCollapsed,
        state.collection,
        tagState.visibleTagCount
    ]);
    let handlePressCollapse = ()=>{
        // Prevents button from losing focus if focusedKey got collapsed.
        state.selectionManager.setFocusedKey(null);
        setIsCollapsed((prevCollapsed)=>!prevCollapsed);
    };
    let showActions = tagState.showCollapseButton || actionLabel && onAction;
    let isEmpty = state.collection.size === 0;
    let containerStyle = (0, $dTV3V$react.useMemo)(()=>{
        if (maxRows == null || !isCollapsed || isEmpty) return undefined;
        let maxHeight = ($e7b72d8874e98cd4$var$TAG_STYLES[scale].height + $e7b72d8874e98cd4$var$TAG_STYLES[scale].margin * 2) * maxRows;
        return {
            maxHeight: maxHeight,
            overflow: "hidden"
        };
    }, [
        isCollapsed,
        maxRows,
        isEmpty,
        scale
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactariafocus.FocusScope), null, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactspectrumlabel.Field), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        showErrorIcon: true,
        ref: domRef,
        elementType: "span",
        wrapperClassName: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tags-fieldWrapper", {
            "spectrum-Tags-fieldWrapper--positionSide": labelPosition === "side"
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement("div", {
        ref: containerRef,
        style: containerStyle,
        className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tags-container", {
            "spectrum-Tags-container--empty": isEmpty
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactariafocus.FocusRing), {
        focusRingClass: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement("div", {
        ref: tagsRef,
        ...gridProps,
        className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tags")
    }, visibleTags.map((item)=>/*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dd2c5d00108c1840$export$3288d34c523a1192), {
            ...item.props,
            key: item.key,
            item: item,
            state: state
        }, item.rendered)), isEmpty && /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement("div", {
        className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tags-empty-state")
    }, renderEmptyState()))), showActions && !isEmpty && /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactspectrumprovider.Provider), {
        isDisabled: false
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement("div", {
        role: "group",
        ref: actionsRef,
        id: actionsId,
        "aria-label": stringFormatter.format("actions"),
        "aria-labelledby": `${gridProps.id} ${actionsId}`,
        className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tags-actions")
    }, tagState.showCollapseButton && /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactspectrumbutton.ActionButton), {
        isQuiet: true,
        onPress: handlePressCollapse,
        UNSAFE_className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tags-actionButton")
    }, isCollapsed ? stringFormatter.format("showAllButtonLabel", {
        tagCount: state.collection.size
    }) : stringFormatter.format("hideButtonLabel")), actionLabel && onAction && /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).createElement((0, $dTV3V$reactspectrumbutton.ActionButton), {
        isQuiet: true,
        onPress: ()=>onAction === null || onAction === void 0 ? void 0 : onAction(),
        UNSAFE_className: (0, $dTV3V$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ed889f8f1f091fc8$exports))), "spectrum-Tags-actionButton")
    }, actionLabel))))));
}
/** Tags allow users to categorize content. They can represent keywords or people, and are grouped to describe an item or a search request. */ const $e7b72d8874e98cd4$export$67ea30858aaf75e3 = /*#__PURE__*/ (0, ($parcel$interopDefault($dTV3V$react))).forwardRef($e7b72d8874e98cd4$var$TagGroup);
