// @react-spectrum/text/dist/main.js
"use client";

var $kLXM6$reactariautils = require("@react-aria/utils");
var $kLXM6$react = require("react");
var $kLXM6$reactspectrumutils = require("@react-spectrum/utils");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Text", () => $1efae25dd196d679$export$5f1af8db9871e1d6);
$parcel$export(module.exports, "Heading", () => $c7c750c966e80a46$export$a8a3e93435678ff9);
$parcel$export(module.exports, "Keyboard", () => $961f7ef9e414f5f0$export$16e4d70cc375e707);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $1efae25dd196d679$var$Text(props, ref) {
    props = (0, $kLXM6$reactspectrumutils.useSlotProps)(props, "text");
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $kLXM6$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $kLXM6$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kLXM6$react))).createElement("span", {
        ...(0, $kLXM6$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, children);
}
/**
 * Text represents text with no specific semantic meaning.
 */ const $1efae25dd196d679$export$5f1af8db9871e1d6 = /*#__PURE__*/ (0, $kLXM6$react.forwardRef)($1efae25dd196d679$var$Text);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $c7c750c966e80a46$var$Heading(props, ref) {
    props = (0, $kLXM6$reactspectrumutils.useSlotProps)(props, "heading");
    let { children: children, level: level = 3, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $kLXM6$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $kLXM6$reactspectrumutils.useDOMRef)(ref);
    let HeadingTag = `h${level}`;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kLXM6$react))).createElement(HeadingTag, {
        ...(0, $kLXM6$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, children);
}
/**
 * Heading is used to create various levels of typographic hierarchies.
 */ const $c7c750c966e80a46$export$a8a3e93435678ff9 = /*#__PURE__*/ (0, $kLXM6$react.forwardRef)($c7c750c966e80a46$var$Heading);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $961f7ef9e414f5f0$var$Keyboard(props, ref) {
    props = (0, $kLXM6$reactspectrumutils.useSlotProps)(props, "keyboard");
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $kLXM6$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $kLXM6$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kLXM6$react))).createElement("kbd", {
        ...(0, $kLXM6$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, children);
}
/**
 * Keyboard represents text that specifies a keyboard command.
 */ const $961f7ef9e414f5f0$export$16e4d70cc375e707 = /*#__PURE__*/ (0, $kLXM6$react.forwardRef)($961f7ef9e414f5f0$var$Keyboard);
