// @react-spectrum/actionbar/dist/import.mjs
"use client";

import $42f1aaf0cd835ede$exports from "./ar-AE.mjs";
import $22c931b8c5fa9db9$exports from "./bg-BG.mjs";
import $07dc24deaaf64ab0$exports from "./cs-CZ.mjs";
import $44fd3a75baaf9937$exports from "./da-DK.mjs";
import $3281990be7bd78f6$exports from "./de-DE.mjs";
import $242cbd951f3bcfeb$exports from "./el-GR.mjs";
import $117deeeb64547dc7$exports from "./en-US.mjs";
import $5b8605d86a9af2ee$exports from "./es-ES.mjs";
import $5530e5c9f2623025$exports from "./et-EE.mjs";
import $7a405d1dfc563131$exports from "./fi-FI.mjs";
import $e0c3da022513542b$exports from "./fr-FR.mjs";
import $796ff5aed019aee1$exports from "./he-IL.mjs";
import $e1066348ea80fd36$exports from "./hr-HR.mjs";
import $939ac78e6c54d005$exports from "./hu-HU.mjs";
import $087af352bd8a67a5$exports from "./it-IT.mjs";
import $ef25b75f84bf9640$exports from "./ja-JP.mjs";
import $a9fcaa28ab16ac58$exports from "./ko-KR.mjs";
import $7103396bd37d192c$exports from "./lt-LT.mjs";
import $da39587a937fb6fd$exports from "./lv-LV.mjs";
import $fef3c3414d151198$exports from "./nb-NO.mjs";
import $3cfd427955fee95c$exports from "./nl-NL.mjs";
import $ecd102793531599a$exports from "./pl-PL.mjs";
import $70b0a95af4186e29$exports from "./pt-BR.mjs";
import $4b15e180aed5c52d$exports from "./pt-PT.mjs";
import $d0ef0a1cef9c2d6c$exports from "./ro-RO.mjs";
import $58fc1ff371f0c7fd$exports from "./ru-RU.mjs";
import $b5f65f4e7988d1dd$exports from "./sk-SK.mjs";
import $5409ad409aeaf678$exports from "./sl-SI.mjs";
import $43feb38b3d4bbf1f$exports from "./sr-SP.mjs";
import $cbe41ff823ebe108$exports from "./sv-SE.mjs";
import $57b680843fcbe09e$exports from "./tr-TR.mjs";
import $1e4c96bc308e0da2$exports from "./uk-UA.mjs";
import $18faa55a67dca8f3$exports from "./zh-CN.mjs";
import $804c64c525579787$exports from "./zh-TW.mjs";
import "./main.css";
import {Item as $da70baa779e17e7a$re_export$Item} from "@react-stately/collections";
import {ActionButton as $N9qJl$ActionButton} from "@react-spectrum/button";
import {ActionGroup as $N9qJl$ActionGroup} from "@react-spectrum/actiongroup";
import {announce as $N9qJl$announce} from "@react-aria/live-announcer";
import {useDOMRef as $N9qJl$useDOMRef, useStyleProps as $N9qJl$useStyleProps, classNames as $N9qJl$classNames} from "@react-spectrum/utils";
import $N9qJl$spectrumiconsuiCrossLarge from "@spectrum-icons/ui/CrossLarge";
import {filterDOMProps as $N9qJl$filterDOMProps} from "@react-aria/utils";
import {FocusScope as $N9qJl$FocusScope} from "@react-aria/focus";
import {OpenTransition as $N9qJl$OpenTransition} from "@react-spectrum/overlays";
import $N9qJl$react, {useState as $N9qJl$useState, useRef as $N9qJl$useRef, useEffect as $N9qJl$useEffect} from "react";
import {Text as $N9qJl$Text} from "@react-spectrum/text";
import {useKeyboard as $N9qJl$useKeyboard} from "@react-aria/interactions";
import {useLocalizedStringFormatter as $N9qJl$useLocalizedStringFormatter} from "@react-aria/i18n";
import {useProviderProps as $N9qJl$useProviderProps} from "@react-spectrum/provider";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






var $29b198d669109c8b$exports = {};


































$29b198d669109c8b$exports = {
    "ar-AE": $42f1aaf0cd835ede$exports,
    "bg-BG": $22c931b8c5fa9db9$exports,
    "cs-CZ": $07dc24deaaf64ab0$exports,
    "da-DK": $44fd3a75baaf9937$exports,
    "de-DE": $3281990be7bd78f6$exports,
    "el-GR": $242cbd951f3bcfeb$exports,
    "en-US": $117deeeb64547dc7$exports,
    "es-ES": $5b8605d86a9af2ee$exports,
    "et-EE": $5530e5c9f2623025$exports,
    "fi-FI": $7a405d1dfc563131$exports,
    "fr-FR": $e0c3da022513542b$exports,
    "he-IL": $796ff5aed019aee1$exports,
    "hr-HR": $e1066348ea80fd36$exports,
    "hu-HU": $939ac78e6c54d005$exports,
    "it-IT": $087af352bd8a67a5$exports,
    "ja-JP": $ef25b75f84bf9640$exports,
    "ko-KR": $a9fcaa28ab16ac58$exports,
    "lt-LT": $7103396bd37d192c$exports,
    "lv-LV": $da39587a937fb6fd$exports,
    "nb-NO": $fef3c3414d151198$exports,
    "nl-NL": $3cfd427955fee95c$exports,
    "pl-PL": $ecd102793531599a$exports,
    "pt-BR": $70b0a95af4186e29$exports,
    "pt-PT": $4b15e180aed5c52d$exports,
    "ro-RO": $d0ef0a1cef9c2d6c$exports,
    "ru-RU": $58fc1ff371f0c7fd$exports,
    "sk-SK": $b5f65f4e7988d1dd$exports,
    "sl-SI": $5409ad409aeaf678$exports,
    "sr-SP": $43feb38b3d4bbf1f$exports,
    "sv-SE": $cbe41ff823ebe108$exports,
    "tr-TR": $57b680843fcbe09e$exports,
    "uk-UA": $1e4c96bc308e0da2$exports,
    "zh-CN": $18faa55a67dca8f3$exports,
    "zh-TW": $804c64c525579787$exports
};




var $7bd9eeb2cd70bd0c$exports = {};

$parcel$export($7bd9eeb2cd70bd0c$exports, "ActionBarContainer", () => $7bd9eeb2cd70bd0c$export$ac2eb07f267e434c, (v) => $7bd9eeb2cd70bd0c$export$ac2eb07f267e434c = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "actiongroup", () => $7bd9eeb2cd70bd0c$export$7c12d760fcaf48e4, (v) => $7bd9eeb2cd70bd0c$export$7c12d760fcaf48e4 = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "clear", () => $7bd9eeb2cd70bd0c$export$42ffd38884aecdac, (v) => $7bd9eeb2cd70bd0c$export$42ffd38884aecdac = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "is-open", () => $7bd9eeb2cd70bd0c$export$a9781837241c946d, (v) => $7bd9eeb2cd70bd0c$export$a9781837241c946d = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar", () => $7bd9eeb2cd70bd0c$export$71a4c2be37bc4ee5, (v) => $7bd9eeb2cd70bd0c$export$71a4c2be37bc4ee5 = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar--emphasized", () => $7bd9eeb2cd70bd0c$export$3377f5e3cddd68a7, (v) => $7bd9eeb2cd70bd0c$export$3377f5e3cddd68a7 = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar-actionGroup", () => $7bd9eeb2cd70bd0c$export$adc2706661946993, (v) => $7bd9eeb2cd70bd0c$export$adc2706661946993 = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar-bar", () => $7bd9eeb2cd70bd0c$export$ea5a1c79ca7998e9, (v) => $7bd9eeb2cd70bd0c$export$ea5a1c79ca7998e9 = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "react-spectrum-ActionBar-selectedCount", () => $7bd9eeb2cd70bd0c$export$8b9fb52122670f9e, (v) => $7bd9eeb2cd70bd0c$export$8b9fb52122670f9e = v);
$parcel$export($7bd9eeb2cd70bd0c$exports, "selected", () => $7bd9eeb2cd70bd0c$export$d90250155de6d7e7, (v) => $7bd9eeb2cd70bd0c$export$d90250155de6d7e7 = v);
var $7bd9eeb2cd70bd0c$export$ac2eb07f267e434c;
var $7bd9eeb2cd70bd0c$export$7c12d760fcaf48e4;
var $7bd9eeb2cd70bd0c$export$42ffd38884aecdac;
var $7bd9eeb2cd70bd0c$export$a9781837241c946d;
var $7bd9eeb2cd70bd0c$export$71a4c2be37bc4ee5;
var $7bd9eeb2cd70bd0c$export$3377f5e3cddd68a7;
var $7bd9eeb2cd70bd0c$export$adc2706661946993;
var $7bd9eeb2cd70bd0c$export$ea5a1c79ca7998e9;
var $7bd9eeb2cd70bd0c$export$8b9fb52122670f9e;
var $7bd9eeb2cd70bd0c$export$d90250155de6d7e7;
$7bd9eeb2cd70bd0c$export$ac2eb07f267e434c = `R4yOqG_ActionBarContainer`;
$7bd9eeb2cd70bd0c$export$7c12d760fcaf48e4 = `R4yOqG_actiongroup`;
$7bd9eeb2cd70bd0c$export$42ffd38884aecdac = `R4yOqG_clear`;
$7bd9eeb2cd70bd0c$export$a9781837241c946d = `R4yOqG_is-open`;
$7bd9eeb2cd70bd0c$export$71a4c2be37bc4ee5 = `R4yOqG_react-spectrum-ActionBar`;
$7bd9eeb2cd70bd0c$export$3377f5e3cddd68a7 = `R4yOqG_react-spectrum-ActionBar--emphasized`;
$7bd9eeb2cd70bd0c$export$adc2706661946993 = `R4yOqG_react-spectrum-ActionBar-actionGroup`;
$7bd9eeb2cd70bd0c$export$ea5a1c79ca7998e9 = `R4yOqG_react-spectrum-ActionBar-bar`;
$7bd9eeb2cd70bd0c$export$8b9fb52122670f9e = `R4yOqG_react-spectrum-ActionBar-selectedCount`;
$7bd9eeb2cd70bd0c$export$d90250155de6d7e7 = `R4yOqG_selected`;






function $a4e665fbd087d71e$var$ActionBar(props, ref) {
    let isOpen = props.selectedItemCount !== 0;
    let domRef = (0, $N9qJl$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$OpenTransition), {
        nodeRef: domRef,
        in: isOpen,
        mountOnEnter: true,
        unmountOnExit: true
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement($a4e665fbd087d71e$var$ActionBarInnerWithRef, {
        ...props,
        ref: domRef
    }));
}
function $a4e665fbd087d71e$var$ActionBarInner(props, ref) {
    props = (0, $N9qJl$useProviderProps)(props);
    let { children: children, isEmphasized: isEmphasized, onAction: onAction, onClearSelection: onClearSelection, selectedItemCount: selectedItemCount, isOpen: isOpen, items: items } = props;
    let { styleProps: styleProps } = (0, $N9qJl$useStyleProps)(props);
    let stringFormatter = (0, $N9qJl$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($29b198d669109c8b$exports))), "@react-spectrum/actionbar");
    // Store the last count greater than zero in a ref so that we can retain it while rendering the fade-out animation.
    let [lastCount, setLastCount] = (0, $N9qJl$useState)(selectedItemCount);
    if ((selectedItemCount === "all" || selectedItemCount > 0) && selectedItemCount !== lastCount) setLastCount(selectedItemCount);
    let { keyboardProps: keyboardProps } = (0, $N9qJl$useKeyboard)({
        onKeyDown (e) {
            if (e.key === "Escape") {
                e.preventDefault();
                onClearSelection();
            }
        }
    });
    // Announce "actions available" on mount.
    let isInitial = (0, $N9qJl$useRef)(true);
    (0, $N9qJl$useEffect)(()=>{
        if (isInitial.current) {
            isInitial.current = false;
            (0, $N9qJl$announce)(stringFormatter.format("actionsAvailable"));
        }
    }, [
        stringFormatter
    ]);
    return /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$FocusScope), {
        restoreFocus: true
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement("div", {
        ...(0, $N9qJl$filterDOMProps)(props),
        ...styleProps,
        ...keyboardProps,
        ref: ref,
        className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "react-spectrum-ActionBar", {
            "react-spectrum-ActionBar--emphasized": isEmphasized,
            "is-open": isOpen
        }, styleProps.className)
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement("div", {
        className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "react-spectrum-ActionBar-bar")
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$ActionGroup), {
        items: items,
        "aria-label": stringFormatter.format("actions"),
        isQuiet: true,
        staticColor: isEmphasized ? "white" : undefined,
        overflowMode: "collapse",
        buttonLabelBehavior: "collapse",
        onAction: onAction,
        UNSAFE_className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "react-spectrum-ActionBar-actionGroup")
    }, children), /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$ActionButton), {
        gridArea: (0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))).clear,
        "aria-label": stringFormatter.format("clearSelection"),
        onPress: ()=>onClearSelection(),
        isQuiet: true,
        staticColor: isEmphasized ? "white" : undefined
    }, /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$spectrumiconsuiCrossLarge), null)), /*#__PURE__*/ (0, $N9qJl$react).createElement((0, $N9qJl$Text), {
        UNSAFE_className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "react-spectrum-ActionBar-selectedCount")
    }, lastCount === "all" ? stringFormatter.format("selectedAll") : stringFormatter.format("selected", {
        count: lastCount
    })))));
}
const $a4e665fbd087d71e$var$ActionBarInnerWithRef = /*#__PURE__*/ (0, $N9qJl$react).forwardRef($a4e665fbd087d71e$var$ActionBarInner);
/**
 * Action bars are used for single and bulk selection patterns when a user needs to perform actions on one or more items at the same time.
 */ const $a4e665fbd087d71e$export$e213cebad6250b4a = /*#__PURE__*/ (0, $N9qJl$react).forwardRef($a4e665fbd087d71e$var$ActionBar);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $2f6c8b20aa071c6d$var$ActionBarContainer(props, ref) {
    // Grabs specific props from the closest Provider (see https://react-spectrum.adobe.com/react-spectrum/Provider.html#property-groups). Remove if your component doesn't support any of the listed props.
    props = (0, $N9qJl$useProviderProps)(props);
    let { children: children } = props;
    let { styleProps: styleProps } = (0, $N9qJl$useStyleProps)(props);
    let domRef = (0, $N9qJl$useDOMRef)(ref);
    return /*#__PURE__*/ (0, $N9qJl$react).createElement("div", {
        ...(0, $N9qJl$filterDOMProps)(props),
        ...styleProps,
        ref: domRef,
        className: (0, $N9qJl$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7bd9eeb2cd70bd0c$exports))), "ActionBarContainer", styleProps.className)
    }, children);
}
/**
 * ActionBarContainer wraps around an ActionBar and a component that supports selection. It handles
 * the ActionBar's position with respect to its linked component.
 */ const $2f6c8b20aa071c6d$export$ac2eb07f267e434c = /*#__PURE__*/ (0, $N9qJl$react).forwardRef($2f6c8b20aa071c6d$var$ActionBarContainer);





export {$a4e665fbd087d71e$export$e213cebad6250b4a as ActionBar, $2f6c8b20aa071c6d$export$ac2eb07f267e434c as ActionBarContainer, $da70baa779e17e7a$re_export$Item as Item};
