// @react-spectrum/actionbar/dist/main.js
"use client";

var $a42f3d97a30aa6c9$exports = require("./ar-AE.main.js");
var $527a50f3b3c911b7$exports = require("./bg-BG.main.js");
var $cd90476fdc303242$exports = require("./cs-CZ.main.js");
var $081b7298af5f5b69$exports = require("./da-DK.main.js");
var $6ef7dffbce87e74f$exports = require("./de-DE.main.js");
var $43c62672ca45763f$exports = require("./el-GR.main.js");
var $4a93342b52e9ddcf$exports = require("./en-US.main.js");
var $d1fe0a28d16ba803$exports = require("./es-ES.main.js");
var $3f607057b4497412$exports = require("./et-EE.main.js");
var $dffc096b505cd1a5$exports = require("./fi-FI.main.js");
var $fced00dcdde4e781$exports = require("./fr-FR.main.js");
var $e568cd7cb63d7ce3$exports = require("./he-IL.main.js");
var $50951315acf49d05$exports = require("./hr-HR.main.js");
var $fd06dc9cc87d0258$exports = require("./hu-HU.main.js");
var $9bd9ce3c11c5d11e$exports = require("./it-IT.main.js");
var $9f76d796bf707b10$exports = require("./ja-JP.main.js");
var $94efe7bbbb291ef4$exports = require("./ko-KR.main.js");
var $92a06c1a52267a65$exports = require("./lt-LT.main.js");
var $670c23292e727d95$exports = require("./lv-LV.main.js");
var $2636117ad082d1ea$exports = require("./nb-NO.main.js");
var $e4ccf12b3bee45db$exports = require("./nl-NL.main.js");
var $b259f128f72a6ac6$exports = require("./pl-PL.main.js");
var $edd692d35256da76$exports = require("./pt-BR.main.js");
var $0af50a83741a7f54$exports = require("./pt-PT.main.js");
var $1b60e4cd5d1e5b6b$exports = require("./ro-RO.main.js");
var $e9ca6eba44abffce$exports = require("./ru-RU.main.js");
var $508bfd1ac48eb61a$exports = require("./sk-SK.main.js");
var $d785f6c5ba148ce4$exports = require("./sl-SI.main.js");
var $dfc7dc8ff40cea70$exports = require("./sr-SP.main.js");
var $bea42116379eb09f$exports = require("./sv-SE.main.js");
var $dc03da5d819982a5$exports = require("./tr-TR.main.js");
var $c93c67a8485a02db$exports = require("./uk-UA.main.js");
var $dd336a8fdfa9aa7b$exports = require("./zh-CN.main.js");
var $30d1241741295d61$exports = require("./zh-TW.main.js");
require("./main.css");
var $grxqE$reactstatelycollections = require("@react-stately/collections");
var $grxqE$reactspectrumbutton = require("@react-spectrum/button");
var $grxqE$reactspectrumactiongroup = require("@react-spectrum/actiongroup");
var $grxqE$reactarialiveannouncer = require("@react-aria/live-announcer");
var $grxqE$reactspectrumutils = require("@react-spectrum/utils");
var $grxqE$spectrumiconsuiCrossLarge = require("@spectrum-icons/ui/CrossLarge");
var $grxqE$reactariautils = require("@react-aria/utils");
var $grxqE$reactariafocus = require("@react-aria/focus");
var $grxqE$reactspectrumoverlays = require("@react-spectrum/overlays");
var $grxqE$react = require("react");
var $grxqE$reactspectrumtext = require("@react-spectrum/text");
var $grxqE$reactariainteractions = require("@react-aria/interactions");
var $grxqE$reactariai18n = require("@react-aria/i18n");
var $grxqE$reactspectrumprovider = require("@react-spectrum/provider");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "ActionBar", () => $7b5467653feb1153$export$e213cebad6250b4a);
$parcel$export(module.exports, "ActionBarContainer", () => $961b22d2b8bd2da1$export$ac2eb07f267e434c);
$parcel$export(module.exports, "Item", () => $grxqE$reactstatelycollections.Item);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






var $b1ba52ca91e22687$exports = {};


































$b1ba52ca91e22687$exports = {
    "ar-AE": $a42f3d97a30aa6c9$exports,
    "bg-BG": $527a50f3b3c911b7$exports,
    "cs-CZ": $cd90476fdc303242$exports,
    "da-DK": $081b7298af5f5b69$exports,
    "de-DE": $6ef7dffbce87e74f$exports,
    "el-GR": $43c62672ca45763f$exports,
    "en-US": $4a93342b52e9ddcf$exports,
    "es-ES": $d1fe0a28d16ba803$exports,
    "et-EE": $3f607057b4497412$exports,
    "fi-FI": $dffc096b505cd1a5$exports,
    "fr-FR": $fced00dcdde4e781$exports,
    "he-IL": $e568cd7cb63d7ce3$exports,
    "hr-HR": $50951315acf49d05$exports,
    "hu-HU": $fd06dc9cc87d0258$exports,
    "it-IT": $9bd9ce3c11c5d11e$exports,
    "ja-JP": $9f76d796bf707b10$exports,
    "ko-KR": $94efe7bbbb291ef4$exports,
    "lt-LT": $92a06c1a52267a65$exports,
    "lv-LV": $670c23292e727d95$exports,
    "nb-NO": $2636117ad082d1ea$exports,
    "nl-NL": $e4ccf12b3bee45db$exports,
    "pl-PL": $b259f128f72a6ac6$exports,
    "pt-BR": $edd692d35256da76$exports,
    "pt-PT": $0af50a83741a7f54$exports,
    "ro-RO": $1b60e4cd5d1e5b6b$exports,
    "ru-RU": $e9ca6eba44abffce$exports,
    "sk-SK": $508bfd1ac48eb61a$exports,
    "sl-SI": $d785f6c5ba148ce4$exports,
    "sr-SP": $dfc7dc8ff40cea70$exports,
    "sv-SE": $bea42116379eb09f$exports,
    "tr-TR": $dc03da5d819982a5$exports,
    "uk-UA": $c93c67a8485a02db$exports,
    "zh-CN": $dd336a8fdfa9aa7b$exports,
    "zh-TW": $30d1241741295d61$exports
};




var $ff24c7bbe0c168db$exports = {};

$parcel$export($ff24c7bbe0c168db$exports, "ActionBarContainer", () => $ff24c7bbe0c168db$export$ac2eb07f267e434c, (v) => $ff24c7bbe0c168db$export$ac2eb07f267e434c = v);
$parcel$export($ff24c7bbe0c168db$exports, "actiongroup", () => $ff24c7bbe0c168db$export$7c12d760fcaf48e4, (v) => $ff24c7bbe0c168db$export$7c12d760fcaf48e4 = v);
$parcel$export($ff24c7bbe0c168db$exports, "clear", () => $ff24c7bbe0c168db$export$42ffd38884aecdac, (v) => $ff24c7bbe0c168db$export$42ffd38884aecdac = v);
$parcel$export($ff24c7bbe0c168db$exports, "is-open", () => $ff24c7bbe0c168db$export$a9781837241c946d, (v) => $ff24c7bbe0c168db$export$a9781837241c946d = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar", () => $ff24c7bbe0c168db$export$71a4c2be37bc4ee5, (v) => $ff24c7bbe0c168db$export$71a4c2be37bc4ee5 = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar--emphasized", () => $ff24c7bbe0c168db$export$3377f5e3cddd68a7, (v) => $ff24c7bbe0c168db$export$3377f5e3cddd68a7 = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar-actionGroup", () => $ff24c7bbe0c168db$export$adc2706661946993, (v) => $ff24c7bbe0c168db$export$adc2706661946993 = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar-bar", () => $ff24c7bbe0c168db$export$ea5a1c79ca7998e9, (v) => $ff24c7bbe0c168db$export$ea5a1c79ca7998e9 = v);
$parcel$export($ff24c7bbe0c168db$exports, "react-spectrum-ActionBar-selectedCount", () => $ff24c7bbe0c168db$export$8b9fb52122670f9e, (v) => $ff24c7bbe0c168db$export$8b9fb52122670f9e = v);
$parcel$export($ff24c7bbe0c168db$exports, "selected", () => $ff24c7bbe0c168db$export$d90250155de6d7e7, (v) => $ff24c7bbe0c168db$export$d90250155de6d7e7 = v);
var $ff24c7bbe0c168db$export$ac2eb07f267e434c;
var $ff24c7bbe0c168db$export$7c12d760fcaf48e4;
var $ff24c7bbe0c168db$export$42ffd38884aecdac;
var $ff24c7bbe0c168db$export$a9781837241c946d;
var $ff24c7bbe0c168db$export$71a4c2be37bc4ee5;
var $ff24c7bbe0c168db$export$3377f5e3cddd68a7;
var $ff24c7bbe0c168db$export$adc2706661946993;
var $ff24c7bbe0c168db$export$ea5a1c79ca7998e9;
var $ff24c7bbe0c168db$export$8b9fb52122670f9e;
var $ff24c7bbe0c168db$export$d90250155de6d7e7;
$ff24c7bbe0c168db$export$ac2eb07f267e434c = `R4yOqG_ActionBarContainer`;
$ff24c7bbe0c168db$export$7c12d760fcaf48e4 = `R4yOqG_actiongroup`;
$ff24c7bbe0c168db$export$42ffd38884aecdac = `R4yOqG_clear`;
$ff24c7bbe0c168db$export$a9781837241c946d = `R4yOqG_is-open`;
$ff24c7bbe0c168db$export$71a4c2be37bc4ee5 = `R4yOqG_react-spectrum-ActionBar`;
$ff24c7bbe0c168db$export$3377f5e3cddd68a7 = `R4yOqG_react-spectrum-ActionBar--emphasized`;
$ff24c7bbe0c168db$export$adc2706661946993 = `R4yOqG_react-spectrum-ActionBar-actionGroup`;
$ff24c7bbe0c168db$export$ea5a1c79ca7998e9 = `R4yOqG_react-spectrum-ActionBar-bar`;
$ff24c7bbe0c168db$export$8b9fb52122670f9e = `R4yOqG_react-spectrum-ActionBar-selectedCount`;
$ff24c7bbe0c168db$export$d90250155de6d7e7 = `R4yOqG_selected`;






function $7b5467653feb1153$var$ActionBar(props, ref) {
    let isOpen = props.selectedItemCount !== 0;
    let domRef = (0, $grxqE$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactspectrumoverlays.OpenTransition), {
        nodeRef: domRef,
        in: isOpen,
        mountOnEnter: true,
        unmountOnExit: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement($7b5467653feb1153$var$ActionBarInnerWithRef, {
        ...props,
        ref: domRef
    }));
}
function $7b5467653feb1153$var$ActionBarInner(props, ref) {
    props = (0, $grxqE$reactspectrumprovider.useProviderProps)(props);
    let { children: children, isEmphasized: isEmphasized, onAction: onAction, onClearSelection: onClearSelection, selectedItemCount: selectedItemCount, isOpen: isOpen, items: items } = props;
    let { styleProps: styleProps } = (0, $grxqE$reactspectrumutils.useStyleProps)(props);
    let stringFormatter = (0, $grxqE$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($b1ba52ca91e22687$exports))), "@react-spectrum/actionbar");
    // Store the last count greater than zero in a ref so that we can retain it while rendering the fade-out animation.
    let [lastCount, setLastCount] = (0, $grxqE$react.useState)(selectedItemCount);
    if ((selectedItemCount === "all" || selectedItemCount > 0) && selectedItemCount !== lastCount) setLastCount(selectedItemCount);
    let { keyboardProps: keyboardProps } = (0, $grxqE$reactariainteractions.useKeyboard)({
        onKeyDown (e) {
            if (e.key === "Escape") {
                e.preventDefault();
                onClearSelection();
            }
        }
    });
    // Announce "actions available" on mount.
    let isInitial = (0, $grxqE$react.useRef)(true);
    (0, $grxqE$react.useEffect)(()=>{
        if (isInitial.current) {
            isInitial.current = false;
            (0, $grxqE$reactarialiveannouncer.announce)(stringFormatter.format("actionsAvailable"));
        }
    }, [
        stringFormatter
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactariafocus.FocusScope), {
        restoreFocus: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement("div", {
        ...(0, $grxqE$reactariautils.filterDOMProps)(props),
        ...styleProps,
        ...keyboardProps,
        ref: ref,
        className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "react-spectrum-ActionBar", {
            "react-spectrum-ActionBar--emphasized": isEmphasized,
            "is-open": isOpen
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement("div", {
        className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "react-spectrum-ActionBar-bar")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactspectrumactiongroup.ActionGroup), {
        items: items,
        "aria-label": stringFormatter.format("actions"),
        isQuiet: true,
        staticColor: isEmphasized ? "white" : undefined,
        overflowMode: "collapse",
        buttonLabelBehavior: "collapse",
        onAction: onAction,
        UNSAFE_className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "react-spectrum-ActionBar-actionGroup")
    }, children), /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactspectrumbutton.ActionButton), {
        gridArea: (0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))).clear,
        "aria-label": stringFormatter.format("clearSelection"),
        onPress: ()=>onClearSelection(),
        isQuiet: true,
        staticColor: isEmphasized ? "white" : undefined
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, ($parcel$interopDefault($grxqE$spectrumiconsuiCrossLarge))), null)), /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement((0, $grxqE$reactspectrumtext.Text), {
        UNSAFE_className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "react-spectrum-ActionBar-selectedCount")
    }, lastCount === "all" ? stringFormatter.format("selectedAll") : stringFormatter.format("selected", {
        count: lastCount
    })))));
}
const $7b5467653feb1153$var$ActionBarInnerWithRef = /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).forwardRef($7b5467653feb1153$var$ActionBarInner);
/**
 * Action bars are used for single and bulk selection patterns when a user needs to perform actions on one or more items at the same time.
 */ const $7b5467653feb1153$export$e213cebad6250b4a = /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).forwardRef($7b5467653feb1153$var$ActionBar);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $961b22d2b8bd2da1$var$ActionBarContainer(props, ref) {
    // Grabs specific props from the closest Provider (see https://react-spectrum.adobe.com/react-spectrum/Provider.html#property-groups). Remove if your component doesn't support any of the listed props.
    props = (0, $grxqE$reactspectrumprovider.useProviderProps)(props);
    let { children: children } = props;
    let { styleProps: styleProps } = (0, $grxqE$reactspectrumutils.useStyleProps)(props);
    let domRef = (0, $grxqE$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).createElement("div", {
        ...(0, $grxqE$reactariautils.filterDOMProps)(props),
        ...styleProps,
        ref: domRef,
        className: (0, $grxqE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ff24c7bbe0c168db$exports))), "ActionBarContainer", styleProps.className)
    }, children);
}
/**
 * ActionBarContainer wraps around an ActionBar and a component that supports selection. It handles
 * the ActionBar's position with respect to its linked component.
 */ const $961b22d2b8bd2da1$export$ac2eb07f267e434c = /*#__PURE__*/ (0, ($parcel$interopDefault($grxqE$react))).forwardRef($961b22d2b8bd2da1$var$ActionBarContainer);
