// @react-spectrum/breadcrumbs/dist/import.mjs
"use client";

import "./main.css";
import {Item as $37781313afe0968e$re_export$Item} from "@react-stately/collections";
import {ActionButton as $koRP3$ActionButton} from "@react-spectrum/button";
import {useDOMRef as $koRP3$useDOMRef, useStyleProps as $koRP3$useStyleProps, classNames as $koRP3$classNames} from "@react-spectrum/utils";
import $koRP3$spectrumiconsuiFolderBreadcrumb from "@spectrum-icons/ui/FolderBreadcrumb";
import {MenuTrigger as $koRP3$MenuTrigger, Menu as $koRP3$Menu} from "@react-spectrum/menu";
import $koRP3$react, {useRef as $koRP3$useRef, useCallback as $koRP3$useCallback, Fragment as $koRP3$Fragment} from "react";
import {useBreadcrumbs as $koRP3$useBreadcrumbs, useBreadcrumbItem as $koRP3$useBreadcrumbItem} from "@react-aria/breadcrumbs";
import {useValueEffect as $koRP3$useValueEffect, useResizeObserver as $koRP3$useResizeObserver, useLayoutEffect as $koRP3$useLayoutEffect, mergeProps as $koRP3$mergeProps} from "@react-aria/utils";
import {useProviderProps as $koRP3$useProviderProps} from "@react-spectrum/provider";
import $koRP3$spectrumiconsuiChevronRightSmall from "@spectrum-icons/ui/ChevronRightSmall";
import {FocusRing as $koRP3$FocusRing} from "@react-aria/focus";
import {useHover as $koRP3$useHover} from "@react-aria/interactions";
import {useLocale as $koRP3$useLocale} from "@react-aria/i18n";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $48e85557f3729e3d$exports = {};

$parcel$export($48e85557f3729e3d$exports, "focus-ring", () => $48e85557f3729e3d$export$f39a09f249340e2a, (v) => $48e85557f3729e3d$export$f39a09f249340e2a = v);
$parcel$export($48e85557f3729e3d$exports, "i18nFontFamily", () => $48e85557f3729e3d$export$8c4ee2c50c22c514, (v) => $48e85557f3729e3d$export$8c4ee2c50c22c514 = v);
$parcel$export($48e85557f3729e3d$exports, "is-disabled", () => $48e85557f3729e3d$export$d35bc1e505d1ebbf, (v) => $48e85557f3729e3d$export$d35bc1e505d1ebbf = v);
$parcel$export($48e85557f3729e3d$exports, "is-dragged", () => $48e85557f3729e3d$export$8778c911bed6c759, (v) => $48e85557f3729e3d$export$8778c911bed6c759 = v);
$parcel$export($48e85557f3729e3d$exports, "is-hovered", () => $48e85557f3729e3d$export$b8813cd5d7824ce7, (v) => $48e85557f3729e3d$export$b8813cd5d7824ce7 = v);
$parcel$export($48e85557f3729e3d$exports, "is-reversed", () => $48e85557f3729e3d$export$a4561bd665abb5db, (v) => $48e85557f3729e3d$export$a4561bd665abb5db = v);
$parcel$export($48e85557f3729e3d$exports, "is-selected", () => $48e85557f3729e3d$export$1e0fb04f31d3c22a, (v) => $48e85557f3729e3d$export$1e0fb04f31d3c22a = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumb", () => $48e85557f3729e3d$export$45a5316056cfac3c, (v) => $48e85557f3729e3d$export$45a5316056cfac3c = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumbs", () => $48e85557f3729e3d$export$8c6320d2db82081f, (v) => $48e85557f3729e3d$export$8c6320d2db82081f = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumbs--medium", () => $48e85557f3729e3d$export$19f957201d941895, (v) => $48e85557f3729e3d$export$19f957201d941895 = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumbs--multiline", () => $48e85557f3729e3d$export$713961e8e432825b, (v) => $48e85557f3729e3d$export$713961e8e432825b = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumbs--small", () => $48e85557f3729e3d$export$e01337e7944312f6, (v) => $48e85557f3729e3d$export$e01337e7944312f6 = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumbs-actionButton", () => $48e85557f3729e3d$export$9753a085493f5648, (v) => $48e85557f3729e3d$export$9753a085493f5648 = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumbs-item", () => $48e85557f3729e3d$export$492a5983c5fbf603, (v) => $48e85557f3729e3d$export$492a5983c5fbf603 = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumbs-itemLink", () => $48e85557f3729e3d$export$5af6a1bcd207d9f6, (v) => $48e85557f3729e3d$export$5af6a1bcd207d9f6 = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Breadcrumbs-itemSeparator", () => $48e85557f3729e3d$export$f72910e36cd91d1e, (v) => $48e85557f3729e3d$export$f72910e36cd91d1e = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-FocusRing-ring", () => $48e85557f3729e3d$export$4109102f950813a6, (v) => $48e85557f3729e3d$export$4109102f950813a6 = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-FocusRing", () => $48e85557f3729e3d$export$24c7f46a6e3605dd, (v) => $48e85557f3729e3d$export$24c7f46a6e3605dd = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-FocusRing--quiet", () => $48e85557f3729e3d$export$2927016961429360, (v) => $48e85557f3729e3d$export$2927016961429360 = v);
$parcel$export($48e85557f3729e3d$exports, "spectrum-Heading--pageTitle", () => $48e85557f3729e3d$export$63e9e087a73f808b, (v) => $48e85557f3729e3d$export$63e9e087a73f808b = v);
var $48e85557f3729e3d$export$f39a09f249340e2a;
var $48e85557f3729e3d$export$8c4ee2c50c22c514;
var $48e85557f3729e3d$export$d35bc1e505d1ebbf;
var $48e85557f3729e3d$export$8778c911bed6c759;
var $48e85557f3729e3d$export$b8813cd5d7824ce7;
var $48e85557f3729e3d$export$a4561bd665abb5db;
var $48e85557f3729e3d$export$1e0fb04f31d3c22a;
var $48e85557f3729e3d$export$45a5316056cfac3c;
var $48e85557f3729e3d$export$8c6320d2db82081f;
var $48e85557f3729e3d$export$19f957201d941895;
var $48e85557f3729e3d$export$713961e8e432825b;
var $48e85557f3729e3d$export$e01337e7944312f6;
var $48e85557f3729e3d$export$9753a085493f5648;
var $48e85557f3729e3d$export$492a5983c5fbf603;
var $48e85557f3729e3d$export$5af6a1bcd207d9f6;
var $48e85557f3729e3d$export$f72910e36cd91d1e;
var $48e85557f3729e3d$export$4109102f950813a6;
var $48e85557f3729e3d$export$24c7f46a6e3605dd;
var $48e85557f3729e3d$export$2927016961429360;
var $48e85557f3729e3d$export$63e9e087a73f808b;
$48e85557f3729e3d$export$f39a09f249340e2a = `_1gDPVa_focus-ring`;
$48e85557f3729e3d$export$8c4ee2c50c22c514 = `_1gDPVa_i18nFontFamily`;
$48e85557f3729e3d$export$d35bc1e505d1ebbf = `_1gDPVa_is-disabled`;
$48e85557f3729e3d$export$8778c911bed6c759 = `_1gDPVa_is-dragged`;
$48e85557f3729e3d$export$b8813cd5d7824ce7 = `_1gDPVa_is-hovered`;
$48e85557f3729e3d$export$a4561bd665abb5db = `_1gDPVa_is-reversed`;
$48e85557f3729e3d$export$1e0fb04f31d3c22a = `_1gDPVa_is-selected`;
$48e85557f3729e3d$export$45a5316056cfac3c = `_1gDPVa_spectrum-Breadcrumb`;
$48e85557f3729e3d$export$8c6320d2db82081f = `_1gDPVa_spectrum-Breadcrumbs`;
$48e85557f3729e3d$export$19f957201d941895 = `_1gDPVa_spectrum-Breadcrumbs--medium`;
$48e85557f3729e3d$export$713961e8e432825b = `_1gDPVa_spectrum-Breadcrumbs--multiline`;
$48e85557f3729e3d$export$e01337e7944312f6 = `_1gDPVa_spectrum-Breadcrumbs--small`;
$48e85557f3729e3d$export$9753a085493f5648 = `_1gDPVa_spectrum-Breadcrumbs-actionButton`;
$48e85557f3729e3d$export$492a5983c5fbf603 = `_1gDPVa_spectrum-Breadcrumbs-item`;
$48e85557f3729e3d$export$5af6a1bcd207d9f6 = `_1gDPVa_spectrum-Breadcrumbs-itemLink`;
$48e85557f3729e3d$export$f72910e36cd91d1e = `_1gDPVa_spectrum-Breadcrumbs-itemSeparator`;
$48e85557f3729e3d$export$4109102f950813a6 = `_1gDPVa_spectrum-FocusRing-ring`;
$48e85557f3729e3d$export$24c7f46a6e3605dd = `_1gDPVa_spectrum-FocusRing ${$48e85557f3729e3d$export$4109102f950813a6}`;
$48e85557f3729e3d$export$2927016961429360 = `_1gDPVa_spectrum-FocusRing--quiet`;
$48e85557f3729e3d$export$63e9e087a73f808b = `_1gDPVa_spectrum-Heading--pageTitle`;





function $2bfa2c84c665c457$export$c13f210c706eb549(props) {
    let { children: children, isCurrent: isCurrent, isDisabled: isDisabled, isMenu: isMenu } = props;
    let { direction: direction } = (0, $koRP3$useLocale)();
    let ref = (0, $koRP3$useRef)(null);
    let ElementType = props.href ? "a" : "span";
    let { itemProps: itemProps } = (0, $koRP3$useBreadcrumbItem)({
        ...props,
        elementType: ElementType
    }, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $koRP3$useHover)(props);
    // If this item contains a menu button, then it shouldn't be a link.
    if (isMenu) itemProps = {};
    return /*#__PURE__*/ (0, $koRP3$react).createElement((0, $koRP3$Fragment), null, /*#__PURE__*/ (0, $koRP3$react).createElement((0, $koRP3$FocusRing), {
        focusRingClass: (0, $koRP3$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($48e85557f3729e3d$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $koRP3$react).createElement(ElementType, {
        ...(0, $koRP3$mergeProps)(itemProps, hoverProps),
        ref: ref,
        className: (0, $koRP3$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($48e85557f3729e3d$exports))), {
            "spectrum-Breadcrumbs-itemLink": !isMenu,
            "is-disabled": !isCurrent && isDisabled,
            "is-hovered": isHovered
        })
    }, children)), /*#__PURE__*/ (0, $koRP3$react).createElement((0, $koRP3$spectrumiconsuiChevronRightSmall), {
        UNSAFE_className: (0, $koRP3$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($48e85557f3729e3d$exports))), "spectrum-Breadcrumbs-itemSeparator", {
            "is-reversed": direction === "rtl"
        })
    }));
}










const $2adaf67aabd3300b$var$MIN_VISIBLE_ITEMS = 1;
const $2adaf67aabd3300b$var$MAX_VISIBLE_ITEMS = 4;
function $2adaf67aabd3300b$var$Breadcrumbs(props, ref) {
    props = (0, $koRP3$useProviderProps)(props);
    let { size: size = "L", isMultiline: isMultiline, children: children, showRoot: showRoot, isDisabled: isDisabled, onAction: onAction, autoFocusCurrent: autoFocusCurrent, ...otherProps } = props;
    // Not using React.Children.toArray because it mutates the key prop.
    let childArray = [];
    (0, $koRP3$react).Children.forEach(children, (child, index)=>{
        if (/*#__PURE__*/ (0, $koRP3$react).isValidElement(child)) {
            if (child.key == null) child = /*#__PURE__*/ (0, $koRP3$react).cloneElement(child, {
                key: index
            });
            childArray.push(child);
        }
    });
    let domRef = (0, $koRP3$useDOMRef)(ref);
    let listRef = (0, $koRP3$useRef)(null);
    let [visibleItems, setVisibleItems] = (0, $koRP3$useValueEffect)(childArray.length);
    let { navProps: navProps } = (0, $koRP3$useBreadcrumbs)(props);
    let { styleProps: styleProps } = (0, $koRP3$useStyleProps)(otherProps);
    let updateOverflow = (0, $koRP3$useCallback)(()=>{
        let computeVisibleItems = (visibleItems)=>{
            // Refs can be null at runtime.
            let currListRef = listRef.current;
            if (!currListRef) return visibleItems;
            let listItems = Array.from(currListRef.children);
            if (listItems.length <= 0) return visibleItems;
            let containerWidth = currListRef.offsetWidth;
            let isShowingMenu = childArray.length > visibleItems;
            let calculatedWidth = 0;
            let newVisibleItems = 0;
            let maxVisibleItems = $2adaf67aabd3300b$var$MAX_VISIBLE_ITEMS;
            if (showRoot) {
                calculatedWidth += listItems.shift().offsetWidth;
                newVisibleItems++;
            }
            if (isShowingMenu) {
                calculatedWidth += listItems.shift().offsetWidth;
                maxVisibleItems--;
            }
            if (showRoot && calculatedWidth >= containerWidth) newVisibleItems--;
            // TODO: what if multiline and only one breadcrumb??
            if (isMultiline) {
                listItems.pop();
                newVisibleItems++;
            } else if (listItems.length > 0) {
                // Ensure the last breadcrumb isn't truncated when we measure it.
                let last = listItems.pop();
                last.style.overflow = "visible";
                calculatedWidth += last.offsetWidth;
                if (calculatedWidth < containerWidth) newVisibleItems++;
                last.style.overflow = "";
            }
            for (let breadcrumb of listItems.reverse()){
                calculatedWidth += breadcrumb.offsetWidth;
                if (calculatedWidth < containerWidth) newVisibleItems++;
            }
            return Math.max($2adaf67aabd3300b$var$MIN_VISIBLE_ITEMS, Math.min(maxVisibleItems, newVisibleItems));
        };
        setVisibleItems(function*() {
            // Update to show all items.
            yield childArray.length;
            // Measure, and update to show the items that fit.
            let newVisibleItems = computeVisibleItems(childArray.length);
            yield newVisibleItems;
            // If the number of items is less than the number of children,
            // then update again to ensure that the menu fits.
            if (newVisibleItems < childArray.length && newVisibleItems > 1) yield computeVisibleItems(newVisibleItems);
        });
    }, [
        childArray.length,
        setVisibleItems,
        showRoot,
        isMultiline
    ]);
    (0, $koRP3$useResizeObserver)({
        ref: domRef,
        onResize: updateOverflow
    });
    let lastChildren = (0, $koRP3$useRef)(null);
    (0, $koRP3$useLayoutEffect)(()=>{
        if (children !== lastChildren.current) {
            lastChildren.current = children;
            updateOverflow();
        }
    });
    let contents = childArray;
    if (childArray.length > visibleItems) {
        let selectedItem = childArray[childArray.length - 1];
        var _selectedItem_key;
        let selectedKey = (_selectedItem_key = selectedItem.key) !== null && _selectedItem_key !== void 0 ? _selectedItem_key : childArray.length - 1;
        let onMenuAction = (key)=>{
            // Don't fire onAction when clicking on the last item
            if (key !== selectedKey && onAction) onAction(key);
        };
        let menuItem = /*#__PURE__*/ (0, $koRP3$react).createElement((0, $2bfa2c84c665c457$export$c13f210c706eb549), {
            key: "menu",
            isMenu: true
        }, /*#__PURE__*/ (0, $koRP3$react).createElement((0, $koRP3$MenuTrigger), null, /*#__PURE__*/ (0, $koRP3$react).createElement((0, $koRP3$ActionButton), {
            UNSAFE_className: (0, $koRP3$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($48e85557f3729e3d$exports))), "spectrum-Breadcrumbs-actionButton"),
            "aria-label": "\u2026",
            isQuiet: true,
            isDisabled: isDisabled
        }, /*#__PURE__*/ (0, $koRP3$react).createElement((0, $koRP3$spectrumiconsuiFolderBreadcrumb), null)), /*#__PURE__*/ (0, $koRP3$react).createElement((0, $koRP3$Menu), {
            selectionMode: "single",
            selectedKeys: [
                selectedKey
            ],
            onAction: onMenuAction
        }, childArray)));
        contents = [
            menuItem
        ];
        let breadcrumbs = [
            ...childArray
        ];
        let endItems = visibleItems;
        if (showRoot && visibleItems > 1) {
            let rootItem = breadcrumbs.shift();
            if (rootItem) contents.unshift(rootItem);
            endItems--;
        }
        contents.push(...breadcrumbs.slice(-endItems));
    }
    let lastIndex = contents.length - 1;
    let breadcrumbItems = contents.map((child, index)=>{
        let isCurrent = index === lastIndex;
        var _child_key;
        let key = (_child_key = child.key) !== null && _child_key !== void 0 ? _child_key : index;
        let onPress = ()=>{
            if (onAction) onAction(key);
        };
        return /*#__PURE__*/ (0, $koRP3$react).createElement("li", {
            key: index,
            className: (0, $koRP3$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($48e85557f3729e3d$exports))), "spectrum-Breadcrumbs-item")
        }, /*#__PURE__*/ (0, $koRP3$react).createElement((0, $2bfa2c84c665c457$export$c13f210c706eb549), {
            ...child.props,
            key: key,
            isCurrent: isCurrent,
            isDisabled: isDisabled,
            onPress: onPress,
            autoFocus: isCurrent && autoFocusCurrent
        }, child.props.children));
    });
    return /*#__PURE__*/ (0, $koRP3$react).createElement("nav", {
        ...styleProps,
        ...navProps,
        ref: domRef
    }, /*#__PURE__*/ (0, $koRP3$react).createElement("ul", {
        ref: listRef,
        className: (0, $koRP3$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($48e85557f3729e3d$exports))), "spectrum-Breadcrumbs", {
            "spectrum-Breadcrumbs--small": size === "S",
            "spectrum-Breadcrumbs--medium": size === "M",
            "spectrum-Breadcrumbs--multiline": isMultiline,
            "spectrum-Breadcrumbs--showRoot": showRoot,
            "is-disabled": isDisabled
        }, styleProps.className)
    }, breadcrumbItems));
}
/**
 * Breadcrumbs show hierarchy and navigational context for a user’s location within an application.
 */ let $2adaf67aabd3300b$export$2dc68d50d56fbbd = /*#__PURE__*/ (0, $koRP3$react).forwardRef($2adaf67aabd3300b$var$Breadcrumbs);





export {$2adaf67aabd3300b$export$2dc68d50d56fbbd as Breadcrumbs, $37781313afe0968e$re_export$Item as Item};
