// @react-spectrum/breadcrumbs/dist/main.js
"use client";

require("./main.css");
var $8bWUw$reactstatelycollections = require("@react-stately/collections");
var $8bWUw$reactspectrumbutton = require("@react-spectrum/button");
var $8bWUw$reactspectrumutils = require("@react-spectrum/utils");
var $8bWUw$spectrumiconsuiFolderBreadcrumb = require("@spectrum-icons/ui/FolderBreadcrumb");
var $8bWUw$reactspectrummenu = require("@react-spectrum/menu");
var $8bWUw$react = require("react");
var $8bWUw$reactariabreadcrumbs = require("@react-aria/breadcrumbs");
var $8bWUw$reactariautils = require("@react-aria/utils");
var $8bWUw$reactspectrumprovider = require("@react-spectrum/provider");
var $8bWUw$spectrumiconsuiChevronRightSmall = require("@spectrum-icons/ui/ChevronRightSmall");
var $8bWUw$reactariafocus = require("@react-aria/focus");
var $8bWUw$reactariainteractions = require("@react-aria/interactions");
var $8bWUw$reactariai18n = require("@react-aria/i18n");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Breadcrumbs", () => $299bf776d9162a05$export$2dc68d50d56fbbd);
$parcel$export(module.exports, "Item", () => $8bWUw$reactstatelycollections.Item);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $3292ef9f328419fc$exports = {};

$parcel$export($3292ef9f328419fc$exports, "focus-ring", () => $3292ef9f328419fc$export$f39a09f249340e2a, (v) => $3292ef9f328419fc$export$f39a09f249340e2a = v);
$parcel$export($3292ef9f328419fc$exports, "i18nFontFamily", () => $3292ef9f328419fc$export$8c4ee2c50c22c514, (v) => $3292ef9f328419fc$export$8c4ee2c50c22c514 = v);
$parcel$export($3292ef9f328419fc$exports, "is-disabled", () => $3292ef9f328419fc$export$d35bc1e505d1ebbf, (v) => $3292ef9f328419fc$export$d35bc1e505d1ebbf = v);
$parcel$export($3292ef9f328419fc$exports, "is-dragged", () => $3292ef9f328419fc$export$8778c911bed6c759, (v) => $3292ef9f328419fc$export$8778c911bed6c759 = v);
$parcel$export($3292ef9f328419fc$exports, "is-hovered", () => $3292ef9f328419fc$export$b8813cd5d7824ce7, (v) => $3292ef9f328419fc$export$b8813cd5d7824ce7 = v);
$parcel$export($3292ef9f328419fc$exports, "is-reversed", () => $3292ef9f328419fc$export$a4561bd665abb5db, (v) => $3292ef9f328419fc$export$a4561bd665abb5db = v);
$parcel$export($3292ef9f328419fc$exports, "is-selected", () => $3292ef9f328419fc$export$1e0fb04f31d3c22a, (v) => $3292ef9f328419fc$export$1e0fb04f31d3c22a = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumb", () => $3292ef9f328419fc$export$45a5316056cfac3c, (v) => $3292ef9f328419fc$export$45a5316056cfac3c = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumbs", () => $3292ef9f328419fc$export$8c6320d2db82081f, (v) => $3292ef9f328419fc$export$8c6320d2db82081f = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumbs--medium", () => $3292ef9f328419fc$export$19f957201d941895, (v) => $3292ef9f328419fc$export$19f957201d941895 = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumbs--multiline", () => $3292ef9f328419fc$export$713961e8e432825b, (v) => $3292ef9f328419fc$export$713961e8e432825b = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumbs--small", () => $3292ef9f328419fc$export$e01337e7944312f6, (v) => $3292ef9f328419fc$export$e01337e7944312f6 = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumbs-actionButton", () => $3292ef9f328419fc$export$9753a085493f5648, (v) => $3292ef9f328419fc$export$9753a085493f5648 = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumbs-item", () => $3292ef9f328419fc$export$492a5983c5fbf603, (v) => $3292ef9f328419fc$export$492a5983c5fbf603 = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumbs-itemLink", () => $3292ef9f328419fc$export$5af6a1bcd207d9f6, (v) => $3292ef9f328419fc$export$5af6a1bcd207d9f6 = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Breadcrumbs-itemSeparator", () => $3292ef9f328419fc$export$f72910e36cd91d1e, (v) => $3292ef9f328419fc$export$f72910e36cd91d1e = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-FocusRing-ring", () => $3292ef9f328419fc$export$4109102f950813a6, (v) => $3292ef9f328419fc$export$4109102f950813a6 = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-FocusRing", () => $3292ef9f328419fc$export$24c7f46a6e3605dd, (v) => $3292ef9f328419fc$export$24c7f46a6e3605dd = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-FocusRing--quiet", () => $3292ef9f328419fc$export$2927016961429360, (v) => $3292ef9f328419fc$export$2927016961429360 = v);
$parcel$export($3292ef9f328419fc$exports, "spectrum-Heading--pageTitle", () => $3292ef9f328419fc$export$63e9e087a73f808b, (v) => $3292ef9f328419fc$export$63e9e087a73f808b = v);
var $3292ef9f328419fc$export$f39a09f249340e2a;
var $3292ef9f328419fc$export$8c4ee2c50c22c514;
var $3292ef9f328419fc$export$d35bc1e505d1ebbf;
var $3292ef9f328419fc$export$8778c911bed6c759;
var $3292ef9f328419fc$export$b8813cd5d7824ce7;
var $3292ef9f328419fc$export$a4561bd665abb5db;
var $3292ef9f328419fc$export$1e0fb04f31d3c22a;
var $3292ef9f328419fc$export$45a5316056cfac3c;
var $3292ef9f328419fc$export$8c6320d2db82081f;
var $3292ef9f328419fc$export$19f957201d941895;
var $3292ef9f328419fc$export$713961e8e432825b;
var $3292ef9f328419fc$export$e01337e7944312f6;
var $3292ef9f328419fc$export$9753a085493f5648;
var $3292ef9f328419fc$export$492a5983c5fbf603;
var $3292ef9f328419fc$export$5af6a1bcd207d9f6;
var $3292ef9f328419fc$export$f72910e36cd91d1e;
var $3292ef9f328419fc$export$4109102f950813a6;
var $3292ef9f328419fc$export$24c7f46a6e3605dd;
var $3292ef9f328419fc$export$2927016961429360;
var $3292ef9f328419fc$export$63e9e087a73f808b;
$3292ef9f328419fc$export$f39a09f249340e2a = `_1gDPVa_focus-ring`;
$3292ef9f328419fc$export$8c4ee2c50c22c514 = `_1gDPVa_i18nFontFamily`;
$3292ef9f328419fc$export$d35bc1e505d1ebbf = `_1gDPVa_is-disabled`;
$3292ef9f328419fc$export$8778c911bed6c759 = `_1gDPVa_is-dragged`;
$3292ef9f328419fc$export$b8813cd5d7824ce7 = `_1gDPVa_is-hovered`;
$3292ef9f328419fc$export$a4561bd665abb5db = `_1gDPVa_is-reversed`;
$3292ef9f328419fc$export$1e0fb04f31d3c22a = `_1gDPVa_is-selected`;
$3292ef9f328419fc$export$45a5316056cfac3c = `_1gDPVa_spectrum-Breadcrumb`;
$3292ef9f328419fc$export$8c6320d2db82081f = `_1gDPVa_spectrum-Breadcrumbs`;
$3292ef9f328419fc$export$19f957201d941895 = `_1gDPVa_spectrum-Breadcrumbs--medium`;
$3292ef9f328419fc$export$713961e8e432825b = `_1gDPVa_spectrum-Breadcrumbs--multiline`;
$3292ef9f328419fc$export$e01337e7944312f6 = `_1gDPVa_spectrum-Breadcrumbs--small`;
$3292ef9f328419fc$export$9753a085493f5648 = `_1gDPVa_spectrum-Breadcrumbs-actionButton`;
$3292ef9f328419fc$export$492a5983c5fbf603 = `_1gDPVa_spectrum-Breadcrumbs-item`;
$3292ef9f328419fc$export$5af6a1bcd207d9f6 = `_1gDPVa_spectrum-Breadcrumbs-itemLink`;
$3292ef9f328419fc$export$f72910e36cd91d1e = `_1gDPVa_spectrum-Breadcrumbs-itemSeparator`;
$3292ef9f328419fc$export$4109102f950813a6 = `_1gDPVa_spectrum-FocusRing-ring`;
$3292ef9f328419fc$export$24c7f46a6e3605dd = `_1gDPVa_spectrum-FocusRing ${$3292ef9f328419fc$export$4109102f950813a6}`;
$3292ef9f328419fc$export$2927016961429360 = `_1gDPVa_spectrum-FocusRing--quiet`;
$3292ef9f328419fc$export$63e9e087a73f808b = `_1gDPVa_spectrum-Heading--pageTitle`;





function $d90ab02c74e9be0c$export$c13f210c706eb549(props) {
    let { children: children, isCurrent: isCurrent, isDisabled: isDisabled, isMenu: isMenu } = props;
    let { direction: direction } = (0, $8bWUw$reactariai18n.useLocale)();
    let ref = (0, $8bWUw$react.useRef)(null);
    let ElementType = props.href ? "a" : "span";
    let { itemProps: itemProps } = (0, $8bWUw$reactariabreadcrumbs.useBreadcrumbItem)({
        ...props,
        elementType: ElementType
    }, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $8bWUw$reactariainteractions.useHover)(props);
    // If this item contains a menu button, then it shouldn't be a link.
    if (isMenu) itemProps = {};
    return /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, $8bWUw$react.Fragment), null, /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, $8bWUw$reactariafocus.FocusRing), {
        focusRingClass: (0, $8bWUw$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3292ef9f328419fc$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement(ElementType, {
        ...(0, $8bWUw$reactariautils.mergeProps)(itemProps, hoverProps),
        ref: ref,
        className: (0, $8bWUw$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3292ef9f328419fc$exports))), {
            "spectrum-Breadcrumbs-itemLink": !isMenu,
            "is-disabled": !isCurrent && isDisabled,
            "is-hovered": isHovered
        })
    }, children)), /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, ($parcel$interopDefault($8bWUw$spectrumiconsuiChevronRightSmall))), {
        UNSAFE_className: (0, $8bWUw$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3292ef9f328419fc$exports))), "spectrum-Breadcrumbs-itemSeparator", {
            "is-reversed": direction === "rtl"
        })
    }));
}










const $299bf776d9162a05$var$MIN_VISIBLE_ITEMS = 1;
const $299bf776d9162a05$var$MAX_VISIBLE_ITEMS = 4;
function $299bf776d9162a05$var$Breadcrumbs(props, ref) {
    props = (0, $8bWUw$reactspectrumprovider.useProviderProps)(props);
    let { size: size = "L", isMultiline: isMultiline, children: children, showRoot: showRoot, isDisabled: isDisabled, onAction: onAction, autoFocusCurrent: autoFocusCurrent, ...otherProps } = props;
    // Not using React.Children.toArray because it mutates the key prop.
    let childArray = [];
    (0, ($parcel$interopDefault($8bWUw$react))).Children.forEach(children, (child, index)=>{
        if (/*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).isValidElement(child)) {
            if (child.key == null) child = /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).cloneElement(child, {
                key: index
            });
            childArray.push(child);
        }
    });
    let domRef = (0, $8bWUw$reactspectrumutils.useDOMRef)(ref);
    let listRef = (0, $8bWUw$react.useRef)(null);
    let [visibleItems, setVisibleItems] = (0, $8bWUw$reactariautils.useValueEffect)(childArray.length);
    let { navProps: navProps } = (0, $8bWUw$reactariabreadcrumbs.useBreadcrumbs)(props);
    let { styleProps: styleProps } = (0, $8bWUw$reactspectrumutils.useStyleProps)(otherProps);
    let updateOverflow = (0, $8bWUw$react.useCallback)(()=>{
        let computeVisibleItems = (visibleItems)=>{
            // Refs can be null at runtime.
            let currListRef = listRef.current;
            if (!currListRef) return visibleItems;
            let listItems = Array.from(currListRef.children);
            if (listItems.length <= 0) return visibleItems;
            let containerWidth = currListRef.offsetWidth;
            let isShowingMenu = childArray.length > visibleItems;
            let calculatedWidth = 0;
            let newVisibleItems = 0;
            let maxVisibleItems = $299bf776d9162a05$var$MAX_VISIBLE_ITEMS;
            if (showRoot) {
                calculatedWidth += listItems.shift().offsetWidth;
                newVisibleItems++;
            }
            if (isShowingMenu) {
                calculatedWidth += listItems.shift().offsetWidth;
                maxVisibleItems--;
            }
            if (showRoot && calculatedWidth >= containerWidth) newVisibleItems--;
            // TODO: what if multiline and only one breadcrumb??
            if (isMultiline) {
                listItems.pop();
                newVisibleItems++;
            } else if (listItems.length > 0) {
                // Ensure the last breadcrumb isn't truncated when we measure it.
                let last = listItems.pop();
                last.style.overflow = "visible";
                calculatedWidth += last.offsetWidth;
                if (calculatedWidth < containerWidth) newVisibleItems++;
                last.style.overflow = "";
            }
            for (let breadcrumb of listItems.reverse()){
                calculatedWidth += breadcrumb.offsetWidth;
                if (calculatedWidth < containerWidth) newVisibleItems++;
            }
            return Math.max($299bf776d9162a05$var$MIN_VISIBLE_ITEMS, Math.min(maxVisibleItems, newVisibleItems));
        };
        setVisibleItems(function*() {
            // Update to show all items.
            yield childArray.length;
            // Measure, and update to show the items that fit.
            let newVisibleItems = computeVisibleItems(childArray.length);
            yield newVisibleItems;
            // If the number of items is less than the number of children,
            // then update again to ensure that the menu fits.
            if (newVisibleItems < childArray.length && newVisibleItems > 1) yield computeVisibleItems(newVisibleItems);
        });
    }, [
        childArray.length,
        setVisibleItems,
        showRoot,
        isMultiline
    ]);
    (0, $8bWUw$reactariautils.useResizeObserver)({
        ref: domRef,
        onResize: updateOverflow
    });
    let lastChildren = (0, $8bWUw$react.useRef)(null);
    (0, $8bWUw$reactariautils.useLayoutEffect)(()=>{
        if (children !== lastChildren.current) {
            lastChildren.current = children;
            updateOverflow();
        }
    });
    let contents = childArray;
    if (childArray.length > visibleItems) {
        let selectedItem = childArray[childArray.length - 1];
        var _selectedItem_key;
        let selectedKey = (_selectedItem_key = selectedItem.key) !== null && _selectedItem_key !== void 0 ? _selectedItem_key : childArray.length - 1;
        let onMenuAction = (key)=>{
            // Don't fire onAction when clicking on the last item
            if (key !== selectedKey && onAction) onAction(key);
        };
        let menuItem = /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, $d90ab02c74e9be0c$export$c13f210c706eb549), {
            key: "menu",
            isMenu: true
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, $8bWUw$reactspectrummenu.MenuTrigger), null, /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, $8bWUw$reactspectrumbutton.ActionButton), {
            UNSAFE_className: (0, $8bWUw$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3292ef9f328419fc$exports))), "spectrum-Breadcrumbs-actionButton"),
            "aria-label": "\u2026",
            isQuiet: true,
            isDisabled: isDisabled
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, ($parcel$interopDefault($8bWUw$spectrumiconsuiFolderBreadcrumb))), null)), /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, $8bWUw$reactspectrummenu.Menu), {
            selectionMode: "single",
            selectedKeys: [
                selectedKey
            ],
            onAction: onMenuAction
        }, childArray)));
        contents = [
            menuItem
        ];
        let breadcrumbs = [
            ...childArray
        ];
        let endItems = visibleItems;
        if (showRoot && visibleItems > 1) {
            let rootItem = breadcrumbs.shift();
            if (rootItem) contents.unshift(rootItem);
            endItems--;
        }
        contents.push(...breadcrumbs.slice(-endItems));
    }
    let lastIndex = contents.length - 1;
    let breadcrumbItems = contents.map((child, index)=>{
        let isCurrent = index === lastIndex;
        var _child_key;
        let key = (_child_key = child.key) !== null && _child_key !== void 0 ? _child_key : index;
        let onPress = ()=>{
            if (onAction) onAction(key);
        };
        return /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement("li", {
            key: index,
            className: (0, $8bWUw$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3292ef9f328419fc$exports))), "spectrum-Breadcrumbs-item")
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement((0, $d90ab02c74e9be0c$export$c13f210c706eb549), {
            ...child.props,
            key: key,
            isCurrent: isCurrent,
            isDisabled: isDisabled,
            onPress: onPress,
            autoFocus: isCurrent && autoFocusCurrent
        }, child.props.children));
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement("nav", {
        ...styleProps,
        ...navProps,
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).createElement("ul", {
        ref: listRef,
        className: (0, $8bWUw$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3292ef9f328419fc$exports))), "spectrum-Breadcrumbs", {
            "spectrum-Breadcrumbs--small": size === "S",
            "spectrum-Breadcrumbs--medium": size === "M",
            "spectrum-Breadcrumbs--multiline": isMultiline,
            "spectrum-Breadcrumbs--showRoot": showRoot,
            "is-disabled": isDisabled
        }, styleProps.className)
    }, breadcrumbItems));
}
/**
 * Breadcrumbs show hierarchy and navigational context for a user’s location within an application.
 */ let $299bf776d9162a05$export$2dc68d50d56fbbd = /*#__PURE__*/ (0, ($parcel$interopDefault($8bWUw$react))).forwardRef($299bf776d9162a05$var$Breadcrumbs);
