// @react-spectrum/switch/dist/main.js
"use client";

require("./main.css");
var $3fKzX$reactspectrumutils = require("@react-spectrum/utils");
var $3fKzX$reactariafocus = require("@react-aria/focus");
var $3fKzX$react = require("react");
var $3fKzX$reactariainteractions = require("@react-aria/interactions");
var $3fKzX$reactspectrumprovider = require("@react-spectrum/provider");
var $3fKzX$reactariaswitch = require("@react-aria/switch");
var $3fKzX$reactstatelytoggle = require("@react-stately/toggle");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Switch", () => $e216104aee091306$export$b5d5cf8927ab7262);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $faeb9d1b92049ba7$exports = {};

$parcel$export($faeb9d1b92049ba7$exports, "focus-ring", () => $faeb9d1b92049ba7$export$f39a09f249340e2a, (v) => $faeb9d1b92049ba7$export$f39a09f249340e2a = v);
$parcel$export($faeb9d1b92049ba7$exports, "i18nFontFamily", () => $faeb9d1b92049ba7$export$8c4ee2c50c22c514, (v) => $faeb9d1b92049ba7$export$8c4ee2c50c22c514 = v);
$parcel$export($faeb9d1b92049ba7$exports, "is-hovered", () => $faeb9d1b92049ba7$export$b8813cd5d7824ce7, (v) => $faeb9d1b92049ba7$export$b8813cd5d7824ce7 = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-FocusRing-ring", () => $faeb9d1b92049ba7$export$4109102f950813a6, (v) => $faeb9d1b92049ba7$export$4109102f950813a6 = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-FocusRing", () => $faeb9d1b92049ba7$export$24c7f46a6e3605dd, (v) => $faeb9d1b92049ba7$export$24c7f46a6e3605dd = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-FocusRing--quiet", () => $faeb9d1b92049ba7$export$2927016961429360, (v) => $faeb9d1b92049ba7$export$2927016961429360 = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-ToggleSwitch", () => $faeb9d1b92049ba7$export$89a9162656c0c38c, (v) => $faeb9d1b92049ba7$export$89a9162656c0c38c = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-ToggleSwitch--ab", () => $faeb9d1b92049ba7$export$dc5a33fddabf5882, (v) => $faeb9d1b92049ba7$export$dc5a33fddabf5882 = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-ToggleSwitch--quiet", () => $faeb9d1b92049ba7$export$ce1b1dcfc10f3df4, (v) => $faeb9d1b92049ba7$export$ce1b1dcfc10f3df4 = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-ToggleSwitch-input", () => $faeb9d1b92049ba7$export$38df7ea4ffef819a, (v) => $faeb9d1b92049ba7$export$38df7ea4ffef819a = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-ToggleSwitch-label", () => $faeb9d1b92049ba7$export$8f1a69021a3897b2, (v) => $faeb9d1b92049ba7$export$8f1a69021a3897b2 = v);
$parcel$export($faeb9d1b92049ba7$exports, "spectrum-ToggleSwitch-switch", () => $faeb9d1b92049ba7$export$b82d06d1b520a22f, (v) => $faeb9d1b92049ba7$export$b82d06d1b520a22f = v);
var $faeb9d1b92049ba7$export$f39a09f249340e2a;
var $faeb9d1b92049ba7$export$8c4ee2c50c22c514;
var $faeb9d1b92049ba7$export$b8813cd5d7824ce7;
var $faeb9d1b92049ba7$export$4109102f950813a6;
var $faeb9d1b92049ba7$export$24c7f46a6e3605dd;
var $faeb9d1b92049ba7$export$2927016961429360;
var $faeb9d1b92049ba7$export$89a9162656c0c38c;
var $faeb9d1b92049ba7$export$dc5a33fddabf5882;
var $faeb9d1b92049ba7$export$ce1b1dcfc10f3df4;
var $faeb9d1b92049ba7$export$38df7ea4ffef819a;
var $faeb9d1b92049ba7$export$8f1a69021a3897b2;
var $faeb9d1b92049ba7$export$b82d06d1b520a22f;
$faeb9d1b92049ba7$export$f39a09f249340e2a = `XHynUq_focus-ring`;
$faeb9d1b92049ba7$export$8c4ee2c50c22c514 = `XHynUq_i18nFontFamily`;
$faeb9d1b92049ba7$export$b8813cd5d7824ce7 = `XHynUq_is-hovered`;
$faeb9d1b92049ba7$export$4109102f950813a6 = `XHynUq_spectrum-FocusRing-ring`;
$faeb9d1b92049ba7$export$24c7f46a6e3605dd = `XHynUq_spectrum-FocusRing ${$faeb9d1b92049ba7$export$4109102f950813a6}`;
$faeb9d1b92049ba7$export$2927016961429360 = `XHynUq_spectrum-FocusRing--quiet`;
$faeb9d1b92049ba7$export$89a9162656c0c38c = `XHynUq_spectrum-ToggleSwitch`;
$faeb9d1b92049ba7$export$dc5a33fddabf5882 = `XHynUq_spectrum-ToggleSwitch--ab`;
$faeb9d1b92049ba7$export$ce1b1dcfc10f3df4 = `XHynUq_spectrum-ToggleSwitch--quiet`;
$faeb9d1b92049ba7$export$38df7ea4ffef819a = `XHynUq_spectrum-ToggleSwitch-input`;
$faeb9d1b92049ba7$export$8f1a69021a3897b2 = `XHynUq_spectrum-ToggleSwitch-label`;
$faeb9d1b92049ba7$export$b82d06d1b520a22f = `XHynUq_spectrum-ToggleSwitch-switch`;






function $e216104aee091306$var$Switch(props, ref) {
    props = (0, $3fKzX$reactspectrumprovider.useProviderProps)(props);
    let { isEmphasized: isEmphasized = false, isDisabled: isDisabled = false, autoFocus: autoFocus, children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $3fKzX$reactspectrumutils.useStyleProps)(otherProps);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $3fKzX$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let inputRef = (0, $3fKzX$react.useRef)(null);
    let domRef = (0, $3fKzX$reactspectrumutils.useFocusableRef)(ref, inputRef);
    let state = (0, $3fKzX$reactstatelytoggle.useToggleState)(props);
    let { inputProps: inputProps } = (0, $3fKzX$reactariaswitch.useSwitch)(props, state, inputRef);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3fKzX$react))).createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: (0, $3fKzX$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($faeb9d1b92049ba7$exports))), "spectrum-ToggleSwitch", {
            "spectrum-ToggleSwitch--quiet": !isEmphasized,
            "is-disabled": isDisabled,
            "is-hovered": isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3fKzX$react))).createElement((0, $3fKzX$reactariafocus.FocusRing), {
        focusRingClass: (0, $3fKzX$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($faeb9d1b92049ba7$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3fKzX$react))).createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: (0, $3fKzX$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($faeb9d1b92049ba7$exports))), "spectrum-ToggleSwitch-input")
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($3fKzX$react))).createElement("span", {
        className: (0, $3fKzX$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($faeb9d1b92049ba7$exports))), "spectrum-ToggleSwitch-switch")
    }), children && /*#__PURE__*/ (0, ($parcel$interopDefault($3fKzX$react))).createElement("span", {
        className: (0, $3fKzX$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($faeb9d1b92049ba7$exports))), "spectrum-ToggleSwitch-label")
    }, children));
}
/**
 * Switches allow users to turn an individual option on or off.
 * They are usually used to activate or deactivate a specific setting.
 */ const $e216104aee091306$export$b5d5cf8927ab7262 = /*#__PURE__*/ (0, $3fKzX$react.forwardRef)($e216104aee091306$var$Switch);
