// @react-spectrum/numberfield/dist/main.js
"use client";

require("./main.css");
var $i8sTE$reactspectrumutils = require("@react-spectrum/utils");
var $i8sTE$reactspectrumlabel = require("@react-spectrum/label");
var $i8sTE$reactariafocus = require("@react-aria/focus");
var $i8sTE$reactariautils = require("@react-aria/utils");
var $i8sTE$reactstatelynumberfield = require("@react-stately/numberfield");
var $i8sTE$react = require("react");
var $i8sTE$reactspectrumtextfield = require("@react-spectrum/textfield");
var $i8sTE$reactspectrumform = require("@react-spectrum/form");
var $i8sTE$reactariainteractions = require("@react-aria/interactions");
var $i8sTE$reactariai18n = require("@react-aria/i18n");
var $i8sTE$reactarianumberfield = require("@react-aria/numberfield");
var $i8sTE$reactspectrumprovider = require("@react-spectrum/provider");
var $i8sTE$spectrumiconsworkflowAdd = require("@spectrum-icons/workflow/Add");
var $i8sTE$spectrumiconsuiChevronDownSmall = require("@spectrum-icons/ui/ChevronDownSmall");
var $i8sTE$spectrumiconsuiChevronUpSmall = require("@spectrum-icons/ui/ChevronUpSmall");
var $i8sTE$spectrumiconsworkflowRemove = require("@spectrum-icons/workflow/Remove");
var $i8sTE$reactariabutton = require("@react-aria/button");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "NumberField", () => $9b3df9c012d62e52$export$63c5fa0b2fdccd2e);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







var $3649bcbba015c4d7$exports = {};

$parcel$export($3649bcbba015c4d7$exports, "decrement", () => $3649bcbba015c4d7$export$8f85cc9753f280aa, (v) => $3649bcbba015c4d7$export$8f85cc9753f280aa = v);
$parcel$export($3649bcbba015c4d7$exports, "field", () => $3649bcbba015c4d7$export$e0f35d825088c098, (v) => $3649bcbba015c4d7$export$e0f35d825088c098 = v);
$parcel$export($3649bcbba015c4d7$exports, "focus-ring", () => $3649bcbba015c4d7$export$f39a09f249340e2a, (v) => $3649bcbba015c4d7$export$f39a09f249340e2a = v);
$parcel$export($3649bcbba015c4d7$exports, "i18nFontFamily", () => $3649bcbba015c4d7$export$8c4ee2c50c22c514, (v) => $3649bcbba015c4d7$export$8c4ee2c50c22c514 = v);
$parcel$export($3649bcbba015c4d7$exports, "increment", () => $3649bcbba015c4d7$export$a2647aa13413c947, (v) => $3649bcbba015c4d7$export$a2647aa13413c947 = v);
$parcel$export($3649bcbba015c4d7$exports, "is-active", () => $3649bcbba015c4d7$export$20fd0f7cd4e6112f, (v) => $3649bcbba015c4d7$export$20fd0f7cd4e6112f = v);
$parcel$export($3649bcbba015c4d7$exports, "is-disabled", () => $3649bcbba015c4d7$export$d35bc1e505d1ebbf, (v) => $3649bcbba015c4d7$export$d35bc1e505d1ebbf = v);
$parcel$export($3649bcbba015c4d7$exports, "is-focused", () => $3649bcbba015c4d7$export$e7dc768d35940237, (v) => $3649bcbba015c4d7$export$e7dc768d35940237 = v);
$parcel$export($3649bcbba015c4d7$exports, "is-hovered", () => $3649bcbba015c4d7$export$b8813cd5d7824ce7, (v) => $3649bcbba015c4d7$export$b8813cd5d7824ce7 = v);
$parcel$export($3649bcbba015c4d7$exports, "is-invalid", () => $3649bcbba015c4d7$export$14a30de2866685fa, (v) => $3649bcbba015c4d7$export$14a30de2866685fa = v);
$parcel$export($3649bcbba015c4d7$exports, "is-open", () => $3649bcbba015c4d7$export$a9781837241c946d, (v) => $3649bcbba015c4d7$export$a9781837241c946d = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-BaseButton", () => $3649bcbba015c4d7$export$f7b415378ab4f839, (v) => $3649bcbba015c4d7$export$f7b415378ab4f839 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-FocusRing-ring", () => $3649bcbba015c4d7$export$4109102f950813a6, (v) => $3649bcbba015c4d7$export$4109102f950813a6 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-FocusRing", () => $3649bcbba015c4d7$export$24c7f46a6e3605dd, (v) => $3649bcbba015c4d7$export$24c7f46a6e3605dd = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-ActionButton", () => $3649bcbba015c4d7$export$53da69f51b770d3, (v) => $3649bcbba015c4d7$export$53da69f51b770d3 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-ActionButton--quiet", () => $3649bcbba015c4d7$export$40e1f009544bacf3, (v) => $3649bcbba015c4d7$export$40e1f009544bacf3 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-ActionButton-hold", () => $3649bcbba015c4d7$export$4a4f01c443eff992, (v) => $3649bcbba015c4d7$export$4a4f01c443eff992 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-ActionButton-label", () => $3649bcbba015c4d7$export$ebb4635c66bacdb9, (v) => $3649bcbba015c4d7$export$ebb4635c66bacdb9 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Button", () => $3649bcbba015c4d7$export$1db4cca5b4ade39a, (v) => $3649bcbba015c4d7$export$1db4cca5b4ade39a = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Button--iconOnly", () => $3649bcbba015c4d7$export$c1d306ec0fae3838, (v) => $3649bcbba015c4d7$export$c1d306ec0fae3838 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Button--pending", () => $3649bcbba015c4d7$export$54592d0984f245d5, (v) => $3649bcbba015c4d7$export$54592d0984f245d5 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Button-circleLoader", () => $3649bcbba015c4d7$export$76d4531f0342bbe0, (v) => $3649bcbba015c4d7$export$76d4531f0342bbe0 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Button-label", () => $3649bcbba015c4d7$export$c081ba562e204942, (v) => $3649bcbba015c4d7$export$c081ba562e204942 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-ClearButton", () => $3649bcbba015c4d7$export$b752ce409e5660c, (v) => $3649bcbba015c4d7$export$b752ce409e5660c = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-ClearButton--small", () => $3649bcbba015c4d7$export$49280ffa5236ba8b, (v) => $3649bcbba015c4d7$export$49280ffa5236ba8b = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-FieldButton", () => $3649bcbba015c4d7$export$a024c99ecf6b5741, (v) => $3649bcbba015c4d7$export$a024c99ecf6b5741 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-FocusRing--quiet", () => $3649bcbba015c4d7$export$2927016961429360, (v) => $3649bcbba015c4d7$export$2927016961429360 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-FieldButton--quiet", () => $3649bcbba015c4d7$export$5ca7b274e4fff9fc, (v) => $3649bcbba015c4d7$export$5ca7b274e4fff9fc = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Icon", () => $3649bcbba015c4d7$export$d374b04f30360026, (v) => $3649bcbba015c4d7$export$d374b04f30360026 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-LogicButton", () => $3649bcbba015c4d7$export$36ae67a131730482, (v) => $3649bcbba015c4d7$export$36ae67a131730482 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper", () => $3649bcbba015c4d7$export$f49c123746e1fb8, (v) => $3649bcbba015c4d7$export$f49c123746e1fb8 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper--isMobile", () => $3649bcbba015c4d7$export$75e6f9c4d1d95421, (v) => $3649bcbba015c4d7$export$75e6f9c4d1d95421 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper--isQuiet", () => $3649bcbba015c4d7$export$70ed0eb2ba7512f8, (v) => $3649bcbba015c4d7$export$70ed0eb2ba7512f8 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper--readonly", () => $3649bcbba015c4d7$export$ba8faf273ff0a970, (v) => $3649bcbba015c4d7$export$ba8faf273ff0a970 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper--showStepper", () => $3649bcbba015c4d7$export$a4e366275982a12f, (v) => $3649bcbba015c4d7$export$a4e366275982a12f = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-button", () => $3649bcbba015c4d7$export$4e11596482f95f63, (v) => $3649bcbba015c4d7$export$4e11596482f95f63 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-button--stepDown", () => $3649bcbba015c4d7$export$e4ad52f6ba4e317b, (v) => $3649bcbba015c4d7$export$e4ad52f6ba4e317b = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-button--stepUp", () => $3649bcbba015c4d7$export$6c76d609d604c6e5, (v) => $3649bcbba015c4d7$export$6c76d609d604c6e5 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-button-icon", () => $3649bcbba015c4d7$export$7d7cb0c4055e98ab, (v) => $3649bcbba015c4d7$export$7d7cb0c4055e98ab = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-container", () => $3649bcbba015c4d7$export$f49a6eb8460c3f93, (v) => $3649bcbba015c4d7$export$f49a6eb8460c3f93 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-container--isMobile", () => $3649bcbba015c4d7$export$3486d98f7bb904b1, (v) => $3649bcbba015c4d7$export$3486d98f7bb904b1 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-field", () => $3649bcbba015c4d7$export$faac6573b15f75d6, (v) => $3649bcbba015c4d7$export$faac6573b15f75d6 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-icon", () => $3649bcbba015c4d7$export$639c56ba6082b615, (v) => $3649bcbba015c4d7$export$639c56ba6082b615 = v);
$parcel$export($3649bcbba015c4d7$exports, "spectrum-Stepper-input", () => $3649bcbba015c4d7$export$8273fb931edd57d4, (v) => $3649bcbba015c4d7$export$8273fb931edd57d4 = v);
var $3649bcbba015c4d7$export$8f85cc9753f280aa;
var $3649bcbba015c4d7$export$e0f35d825088c098;
var $3649bcbba015c4d7$export$f39a09f249340e2a;
var $3649bcbba015c4d7$export$8c4ee2c50c22c514;
var $3649bcbba015c4d7$export$a2647aa13413c947;
var $3649bcbba015c4d7$export$20fd0f7cd4e6112f;
var $3649bcbba015c4d7$export$d35bc1e505d1ebbf;
var $3649bcbba015c4d7$export$e7dc768d35940237;
var $3649bcbba015c4d7$export$b8813cd5d7824ce7;
var $3649bcbba015c4d7$export$14a30de2866685fa;
var $3649bcbba015c4d7$export$a9781837241c946d;
var $3649bcbba015c4d7$export$f7b415378ab4f839;
var $3649bcbba015c4d7$export$4109102f950813a6;
var $3649bcbba015c4d7$export$24c7f46a6e3605dd;
var $3649bcbba015c4d7$export$53da69f51b770d3;
var $3649bcbba015c4d7$export$40e1f009544bacf3;
var $3649bcbba015c4d7$export$4a4f01c443eff992;
var $3649bcbba015c4d7$export$ebb4635c66bacdb9;
var $3649bcbba015c4d7$export$1db4cca5b4ade39a;
var $3649bcbba015c4d7$export$c1d306ec0fae3838;
var $3649bcbba015c4d7$export$54592d0984f245d5;
var $3649bcbba015c4d7$export$76d4531f0342bbe0;
var $3649bcbba015c4d7$export$c081ba562e204942;
var $3649bcbba015c4d7$export$b752ce409e5660c;
var $3649bcbba015c4d7$export$49280ffa5236ba8b;
var $3649bcbba015c4d7$export$a024c99ecf6b5741;
var $3649bcbba015c4d7$export$2927016961429360;
var $3649bcbba015c4d7$export$5ca7b274e4fff9fc;
var $3649bcbba015c4d7$export$d374b04f30360026;
var $3649bcbba015c4d7$export$36ae67a131730482;
var $3649bcbba015c4d7$export$f49c123746e1fb8;
var $3649bcbba015c4d7$export$75e6f9c4d1d95421;
var $3649bcbba015c4d7$export$70ed0eb2ba7512f8;
var $3649bcbba015c4d7$export$ba8faf273ff0a970;
var $3649bcbba015c4d7$export$a4e366275982a12f;
var $3649bcbba015c4d7$export$4e11596482f95f63;
var $3649bcbba015c4d7$export$e4ad52f6ba4e317b;
var $3649bcbba015c4d7$export$6c76d609d604c6e5;
var $3649bcbba015c4d7$export$7d7cb0c4055e98ab;
var $3649bcbba015c4d7$export$f49a6eb8460c3f93;
var $3649bcbba015c4d7$export$3486d98f7bb904b1;
var $3649bcbba015c4d7$export$faac6573b15f75d6;
var $3649bcbba015c4d7$export$639c56ba6082b615;
var $3649bcbba015c4d7$export$8273fb931edd57d4;
$3649bcbba015c4d7$export$8f85cc9753f280aa = `Ug6T-W_decrement`;
$3649bcbba015c4d7$export$e0f35d825088c098 = `Ug6T-W_field`;
$3649bcbba015c4d7$export$f39a09f249340e2a = `Ug6T-W_focus-ring`;
$3649bcbba015c4d7$export$8c4ee2c50c22c514 = `Ug6T-W_i18nFontFamily`;
$3649bcbba015c4d7$export$a2647aa13413c947 = `Ug6T-W_increment`;
$3649bcbba015c4d7$export$20fd0f7cd4e6112f = `Ug6T-W_is-active`;
$3649bcbba015c4d7$export$d35bc1e505d1ebbf = `Ug6T-W_is-disabled`;
$3649bcbba015c4d7$export$e7dc768d35940237 = `Ug6T-W_is-focused`;
$3649bcbba015c4d7$export$b8813cd5d7824ce7 = `Ug6T-W_is-hovered`;
$3649bcbba015c4d7$export$14a30de2866685fa = `Ug6T-W_is-invalid`;
$3649bcbba015c4d7$export$a9781837241c946d = `Ug6T-W_is-open`;
$3649bcbba015c4d7$export$f7b415378ab4f839 = `Ug6T-W_spectrum-BaseButton ${$3649bcbba015c4d7$export$8c4ee2c50c22c514}`;
$3649bcbba015c4d7$export$4109102f950813a6 = `Ug6T-W_spectrum-FocusRing-ring`;
$3649bcbba015c4d7$export$24c7f46a6e3605dd = `Ug6T-W_spectrum-FocusRing ${$3649bcbba015c4d7$export$4109102f950813a6}`;
$3649bcbba015c4d7$export$53da69f51b770d3 = `Ug6T-W_spectrum-ActionButton ${$3649bcbba015c4d7$export$f7b415378ab4f839} ${$3649bcbba015c4d7$export$24c7f46a6e3605dd}`;
$3649bcbba015c4d7$export$40e1f009544bacf3 = `Ug6T-W_spectrum-ActionButton--quiet`;
$3649bcbba015c4d7$export$4a4f01c443eff992 = `Ug6T-W_spectrum-ActionButton-hold`;
$3649bcbba015c4d7$export$ebb4635c66bacdb9 = `Ug6T-W_spectrum-ActionButton-label`;
$3649bcbba015c4d7$export$1db4cca5b4ade39a = `Ug6T-W_spectrum-Button ${$3649bcbba015c4d7$export$f7b415378ab4f839} ${$3649bcbba015c4d7$export$24c7f46a6e3605dd}`;
$3649bcbba015c4d7$export$c1d306ec0fae3838 = `Ug6T-W_spectrum-Button--iconOnly`;
$3649bcbba015c4d7$export$54592d0984f245d5 = `Ug6T-W_spectrum-Button--pending`;
$3649bcbba015c4d7$export$76d4531f0342bbe0 = `Ug6T-W_spectrum-Button-circleLoader`;
$3649bcbba015c4d7$export$c081ba562e204942 = `Ug6T-W_spectrum-Button-label`;
$3649bcbba015c4d7$export$b752ce409e5660c = `Ug6T-W_spectrum-ClearButton ${$3649bcbba015c4d7$export$f7b415378ab4f839} ${$3649bcbba015c4d7$export$24c7f46a6e3605dd}`;
$3649bcbba015c4d7$export$49280ffa5236ba8b = `Ug6T-W_spectrum-ClearButton--small`;
$3649bcbba015c4d7$export$a024c99ecf6b5741 = `Ug6T-W_spectrum-FieldButton ${$3649bcbba015c4d7$export$f7b415378ab4f839} ${$3649bcbba015c4d7$export$24c7f46a6e3605dd}`;
$3649bcbba015c4d7$export$2927016961429360 = `Ug6T-W_spectrum-FocusRing--quiet`;
$3649bcbba015c4d7$export$5ca7b274e4fff9fc = `Ug6T-W_spectrum-FieldButton--quiet ${$3649bcbba015c4d7$export$2927016961429360}`;
$3649bcbba015c4d7$export$d374b04f30360026 = `Ug6T-W_spectrum-Icon`;
$3649bcbba015c4d7$export$36ae67a131730482 = `Ug6T-W_spectrum-LogicButton ${$3649bcbba015c4d7$export$f7b415378ab4f839} ${$3649bcbba015c4d7$export$24c7f46a6e3605dd}`;
$3649bcbba015c4d7$export$f49c123746e1fb8 = `Ug6T-W_spectrum-Stepper ${$3649bcbba015c4d7$export$24c7f46a6e3605dd}`;
$3649bcbba015c4d7$export$75e6f9c4d1d95421 = `Ug6T-W_spectrum-Stepper--isMobile`;
$3649bcbba015c4d7$export$70ed0eb2ba7512f8 = `Ug6T-W_spectrum-Stepper--isQuiet ${$3649bcbba015c4d7$export$2927016961429360}`;
$3649bcbba015c4d7$export$ba8faf273ff0a970 = `Ug6T-W_spectrum-Stepper--readonly`;
$3649bcbba015c4d7$export$a4e366275982a12f = `Ug6T-W_spectrum-Stepper--showStepper`;
$3649bcbba015c4d7$export$4e11596482f95f63 = `Ug6T-W_spectrum-Stepper-button ${$3649bcbba015c4d7$export$f7b415378ab4f839}`;
$3649bcbba015c4d7$export$e4ad52f6ba4e317b = `Ug6T-W_spectrum-Stepper-button--stepDown`;
$3649bcbba015c4d7$export$6c76d609d604c6e5 = `Ug6T-W_spectrum-Stepper-button--stepUp`;
$3649bcbba015c4d7$export$7d7cb0c4055e98ab = `Ug6T-W_spectrum-Stepper-button-icon`;
$3649bcbba015c4d7$export$f49a6eb8460c3f93 = `Ug6T-W_spectrum-Stepper-container`;
$3649bcbba015c4d7$export$3486d98f7bb904b1 = `Ug6T-W_spectrum-Stepper-container--isMobile`;
$3649bcbba015c4d7$export$faac6573b15f75d6 = `Ug6T-W_spectrum-Stepper-field`;
$3649bcbba015c4d7$export$639c56ba6082b615 = `Ug6T-W_spectrum-Stepper-icon`;
$3649bcbba015c4d7$export$8273fb931edd57d4 = `Ug6T-W_spectrum-Stepper-input`;





function $2f3cded204daae76$var$StepButton(props, ref) {
    props = (0, $i8sTE$reactspectrumprovider.useProviderProps)(props);
    let { scale: scale } = (0, $i8sTE$reactspectrumprovider.useProvider)();
    let { direction: direction, isDisabled: isDisabled, isQuiet: isQuiet } = props;
    /**
   * Must use div for now because Safari pointer event bugs on disabled form elements.
   * Link https://bugs.webkit.org/show_bug.cgi?id=219188.
   */ let { buttonProps: buttonProps, isPressed: isPressed } = (0, $i8sTE$reactariabutton.useButton)({
        ...props,
        elementType: "div"
    }, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $i8sTE$reactariainteractions.useHover)(props);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, $i8sTE$reactariafocus.FocusRing), {
        focusRingClass: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement("div", {
        className: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-button", {
            "spectrum-Stepper-button--stepUp": direction === "up",
            "spectrum-Stepper-button--stepDown": direction === "down",
            "spectrum-Stepper-button--isQuiet": isQuiet,
            "is-hovered": isHovered,
            "is-active": isPressed,
            "is-disabled": isDisabled
        }),
        ...(0, $i8sTE$reactariautils.mergeProps)(hoverProps, buttonProps),
        ref: ref
    }, direction === "up" && scale === "large" && /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, ($parcel$interopDefault($i8sTE$spectrumiconsworkflowAdd))), {
        UNSAFE_className: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-button-icon", "spectrum-Stepper-stepUpIcon"),
        size: "S"
    }), direction === "up" && scale === "medium" && /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, ($parcel$interopDefault($i8sTE$spectrumiconsuiChevronUpSmall))), {
        UNSAFE_className: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-button-icon", "spectrum-Stepper-stepUpIcon")
    }), direction === "down" && scale === "large" && /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, ($parcel$interopDefault($i8sTE$spectrumiconsworkflowRemove))), {
        UNSAFE_className: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-button-icon", "spectrum-Stepper-stepDownIcon"),
        size: "S"
    }), direction === "down" && scale === "medium" && /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, ($parcel$interopDefault($i8sTE$spectrumiconsuiChevronDownSmall))), {
        UNSAFE_className: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-button-icon", "spectrum-Stepper-stepDownIcon")
    })));
}
/**
 * Buttons for NumberField.
 */ let $2f3cded204daae76$export$b2f6b60c1d32d6aa = /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).forwardRef($2f3cded204daae76$var$StepButton);









function $9b3df9c012d62e52$var$NumberField(props, ref) {
    props = (0, $i8sTE$reactspectrumprovider.useProviderProps)(props);
    props = (0, $i8sTE$reactspectrumform.useFormProps)(props);
    let provider = (0, $i8sTE$reactspectrumprovider.useProvider)();
    let { isQuiet: isQuiet, isReadOnly: isReadOnly, isDisabled: isDisabled, hideStepper: hideStepper } = props;
    let { styleProps: style } = (0, $i8sTE$reactspectrumutils.useStyleProps)(props);
    let { locale: locale } = (0, $i8sTE$reactariai18n.useLocale)();
    let state = (0, $i8sTE$reactstatelynumberfield.useNumberFieldState)({
        ...props,
        locale: locale
    });
    let inputRef = (0, $i8sTE$react.useRef)();
    let domRef = (0, $i8sTE$reactspectrumutils.useFocusableRef)(ref, inputRef);
    let { groupProps: groupProps, labelProps: labelProps, inputProps: inputProps, incrementButtonProps: incrementButtonProps, decrementButtonProps: decrementButtonProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $i8sTE$reactarianumberfield.useNumberField)(props, state, inputRef);
    let isMobile = provider.scale === "large";
    let showStepper = !hideStepper;
    let { isHovered: isHovered, hoverProps: hoverProps } = (0, $i8sTE$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let validationState = props.validationState || (isInvalid ? "invalid" : null);
    let className = (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper", {
        "spectrum-Stepper--isQuiet": isQuiet,
        "is-disabled": isDisabled,
        "spectrum-Stepper--readonly": isReadOnly,
        "is-invalid": validationState === "invalid" && !isDisabled,
        "spectrum-Stepper--showStepper": showStepper,
        "spectrum-Stepper--isMobile": isMobile,
        "is-hovered": isHovered,
        // because FocusRing won't pass along the className from Field, we have to handle that ourselves
        [style.className]: !props.label && style.className
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, $i8sTE$reactspectrumlabel.Field), {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        labelProps: labelProps,
        ref: domRef,
        wrapperClassName: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-container", {
            "spectrum-Stepper-container--isMobile": isMobile
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement($9b3df9c012d62e52$var$NumberFieldInput, {
        ...props,
        groupProps: (0, $i8sTE$reactariautils.mergeProps)(groupProps, hoverProps),
        inputProps: inputProps,
        inputRef: inputRef,
        incrementProps: incrementButtonProps,
        decrementProps: decrementButtonProps,
        className: className,
        style: style,
        state: state,
        validationState: validationState
    }));
}
const $9b3df9c012d62e52$var$NumberFieldInput = /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).forwardRef(function NumberFieldInput(props, ref) {
    let { groupProps: groupProps, inputProps: inputProps, inputRef: inputRef, incrementProps: incrementProps, decrementProps: decrementProps, className: className, style: style, autoFocus: autoFocus, isQuiet: isQuiet, isDisabled: isDisabled, hideStepper: hideStepper, validationState: validationState, name: name, state: state } = props;
    let showStepper = !hideStepper;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, $i8sTE$reactariafocus.FocusRing), {
        within: true,
        isTextInput: true,
        focusClass: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "is-focused"),
        focusRingClass: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement("div", {
        ...groupProps,
        ref: ref,
        style: style,
        className: className
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, $i8sTE$reactspectrumtextfield.TextFieldBase), {
        UNSAFE_className: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-field"),
        inputClassName: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-input"),
        validationIconClassName: (0, $i8sTE$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($3649bcbba015c4d7$exports))), "spectrum-Stepper-icon"),
        isQuiet: isQuiet,
        inputRef: inputRef,
        validationState: validationState,
        inputProps: inputProps,
        isDisabled: isDisabled,
        disableFocusRing: true
    }), showStepper && /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, ($parcel$interopDefault($i8sTE$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, $2f3cded204daae76$export$b2f6b60c1d32d6aa), {
        direction: "up",
        isQuiet: isQuiet,
        ...incrementProps
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement((0, $2f3cded204daae76$export$b2f6b60c1d32d6aa), {
        direction: "down",
        isQuiet: isQuiet,
        ...decrementProps
    })), name && /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).createElement("input", {
        type: "hidden",
        name: name,
        value: isNaN(state.numberValue) ? "" : state.numberValue
    })));
});
/**
 * NumberFields allow users to enter a number, and increment or decrement the value using stepper buttons.
 */ let $9b3df9c012d62e52$export$63c5fa0b2fdccd2e = /*#__PURE__*/ (0, ($parcel$interopDefault($i8sTE$react))).forwardRef($9b3df9c012d62e52$var$NumberField);
