// @react-spectrum/numberfield/dist/module.js
"use client";

import "./main.css";
import {useStyleProps as $k6K00$useStyleProps, useFocusableRef as $k6K00$useFocusableRef, classNames as $k6K00$classNames} from "@react-spectrum/utils";
import {Field as $k6K00$Field} from "@react-spectrum/label";
import {FocusRing as $k6K00$FocusRing} from "@react-aria/focus";
import {mergeProps as $k6K00$mergeProps} from "@react-aria/utils";
import {useNumberFieldState as $k6K00$useNumberFieldState} from "@react-stately/numberfield";
import $k6K00$react, {useRef as $k6K00$useRef} from "react";
import {TextFieldBase as $k6K00$TextFieldBase} from "@react-spectrum/textfield";
import {useFormProps as $k6K00$useFormProps} from "@react-spectrum/form";
import {useHover as $k6K00$useHover} from "@react-aria/interactions";
import {useLocale as $k6K00$useLocale} from "@react-aria/i18n";
import {useNumberField as $k6K00$useNumberField} from "@react-aria/numberfield";
import {useProviderProps as $k6K00$useProviderProps, useProvider as $k6K00$useProvider} from "@react-spectrum/provider";
import $k6K00$spectrumiconsworkflowAdd from "@spectrum-icons/workflow/Add";
import $k6K00$spectrumiconsuiChevronDownSmall from "@spectrum-icons/ui/ChevronDownSmall";
import $k6K00$spectrumiconsuiChevronUpSmall from "@spectrum-icons/ui/ChevronUpSmall";
import $k6K00$spectrumiconsworkflowRemove from "@spectrum-icons/workflow/Remove";
import {useButton as $k6K00$useButton} from "@react-aria/button";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







var $ba58a329f3e1a776$exports = {};

$parcel$export($ba58a329f3e1a776$exports, "decrement", () => $ba58a329f3e1a776$export$8f85cc9753f280aa, (v) => $ba58a329f3e1a776$export$8f85cc9753f280aa = v);
$parcel$export($ba58a329f3e1a776$exports, "field", () => $ba58a329f3e1a776$export$e0f35d825088c098, (v) => $ba58a329f3e1a776$export$e0f35d825088c098 = v);
$parcel$export($ba58a329f3e1a776$exports, "focus-ring", () => $ba58a329f3e1a776$export$f39a09f249340e2a, (v) => $ba58a329f3e1a776$export$f39a09f249340e2a = v);
$parcel$export($ba58a329f3e1a776$exports, "i18nFontFamily", () => $ba58a329f3e1a776$export$8c4ee2c50c22c514, (v) => $ba58a329f3e1a776$export$8c4ee2c50c22c514 = v);
$parcel$export($ba58a329f3e1a776$exports, "increment", () => $ba58a329f3e1a776$export$a2647aa13413c947, (v) => $ba58a329f3e1a776$export$a2647aa13413c947 = v);
$parcel$export($ba58a329f3e1a776$exports, "is-active", () => $ba58a329f3e1a776$export$20fd0f7cd4e6112f, (v) => $ba58a329f3e1a776$export$20fd0f7cd4e6112f = v);
$parcel$export($ba58a329f3e1a776$exports, "is-disabled", () => $ba58a329f3e1a776$export$d35bc1e505d1ebbf, (v) => $ba58a329f3e1a776$export$d35bc1e505d1ebbf = v);
$parcel$export($ba58a329f3e1a776$exports, "is-focused", () => $ba58a329f3e1a776$export$e7dc768d35940237, (v) => $ba58a329f3e1a776$export$e7dc768d35940237 = v);
$parcel$export($ba58a329f3e1a776$exports, "is-hovered", () => $ba58a329f3e1a776$export$b8813cd5d7824ce7, (v) => $ba58a329f3e1a776$export$b8813cd5d7824ce7 = v);
$parcel$export($ba58a329f3e1a776$exports, "is-invalid", () => $ba58a329f3e1a776$export$14a30de2866685fa, (v) => $ba58a329f3e1a776$export$14a30de2866685fa = v);
$parcel$export($ba58a329f3e1a776$exports, "is-open", () => $ba58a329f3e1a776$export$a9781837241c946d, (v) => $ba58a329f3e1a776$export$a9781837241c946d = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-BaseButton", () => $ba58a329f3e1a776$export$f7b415378ab4f839, (v) => $ba58a329f3e1a776$export$f7b415378ab4f839 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-FocusRing-ring", () => $ba58a329f3e1a776$export$4109102f950813a6, (v) => $ba58a329f3e1a776$export$4109102f950813a6 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-FocusRing", () => $ba58a329f3e1a776$export$24c7f46a6e3605dd, (v) => $ba58a329f3e1a776$export$24c7f46a6e3605dd = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-ActionButton", () => $ba58a329f3e1a776$export$53da69f51b770d3, (v) => $ba58a329f3e1a776$export$53da69f51b770d3 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-ActionButton--quiet", () => $ba58a329f3e1a776$export$40e1f009544bacf3, (v) => $ba58a329f3e1a776$export$40e1f009544bacf3 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-ActionButton-hold", () => $ba58a329f3e1a776$export$4a4f01c443eff992, (v) => $ba58a329f3e1a776$export$4a4f01c443eff992 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-ActionButton-label", () => $ba58a329f3e1a776$export$ebb4635c66bacdb9, (v) => $ba58a329f3e1a776$export$ebb4635c66bacdb9 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Button", () => $ba58a329f3e1a776$export$1db4cca5b4ade39a, (v) => $ba58a329f3e1a776$export$1db4cca5b4ade39a = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Button--iconOnly", () => $ba58a329f3e1a776$export$c1d306ec0fae3838, (v) => $ba58a329f3e1a776$export$c1d306ec0fae3838 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Button--pending", () => $ba58a329f3e1a776$export$54592d0984f245d5, (v) => $ba58a329f3e1a776$export$54592d0984f245d5 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Button-circleLoader", () => $ba58a329f3e1a776$export$76d4531f0342bbe0, (v) => $ba58a329f3e1a776$export$76d4531f0342bbe0 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Button-label", () => $ba58a329f3e1a776$export$c081ba562e204942, (v) => $ba58a329f3e1a776$export$c081ba562e204942 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-ClearButton", () => $ba58a329f3e1a776$export$b752ce409e5660c, (v) => $ba58a329f3e1a776$export$b752ce409e5660c = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-ClearButton--small", () => $ba58a329f3e1a776$export$49280ffa5236ba8b, (v) => $ba58a329f3e1a776$export$49280ffa5236ba8b = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-FieldButton", () => $ba58a329f3e1a776$export$a024c99ecf6b5741, (v) => $ba58a329f3e1a776$export$a024c99ecf6b5741 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-FocusRing--quiet", () => $ba58a329f3e1a776$export$2927016961429360, (v) => $ba58a329f3e1a776$export$2927016961429360 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-FieldButton--quiet", () => $ba58a329f3e1a776$export$5ca7b274e4fff9fc, (v) => $ba58a329f3e1a776$export$5ca7b274e4fff9fc = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Icon", () => $ba58a329f3e1a776$export$d374b04f30360026, (v) => $ba58a329f3e1a776$export$d374b04f30360026 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-LogicButton", () => $ba58a329f3e1a776$export$36ae67a131730482, (v) => $ba58a329f3e1a776$export$36ae67a131730482 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper", () => $ba58a329f3e1a776$export$f49c123746e1fb8, (v) => $ba58a329f3e1a776$export$f49c123746e1fb8 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper--isMobile", () => $ba58a329f3e1a776$export$75e6f9c4d1d95421, (v) => $ba58a329f3e1a776$export$75e6f9c4d1d95421 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper--isQuiet", () => $ba58a329f3e1a776$export$70ed0eb2ba7512f8, (v) => $ba58a329f3e1a776$export$70ed0eb2ba7512f8 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper--readonly", () => $ba58a329f3e1a776$export$ba8faf273ff0a970, (v) => $ba58a329f3e1a776$export$ba8faf273ff0a970 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper--showStepper", () => $ba58a329f3e1a776$export$a4e366275982a12f, (v) => $ba58a329f3e1a776$export$a4e366275982a12f = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-button", () => $ba58a329f3e1a776$export$4e11596482f95f63, (v) => $ba58a329f3e1a776$export$4e11596482f95f63 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-button--stepDown", () => $ba58a329f3e1a776$export$e4ad52f6ba4e317b, (v) => $ba58a329f3e1a776$export$e4ad52f6ba4e317b = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-button--stepUp", () => $ba58a329f3e1a776$export$6c76d609d604c6e5, (v) => $ba58a329f3e1a776$export$6c76d609d604c6e5 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-button-icon", () => $ba58a329f3e1a776$export$7d7cb0c4055e98ab, (v) => $ba58a329f3e1a776$export$7d7cb0c4055e98ab = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-container", () => $ba58a329f3e1a776$export$f49a6eb8460c3f93, (v) => $ba58a329f3e1a776$export$f49a6eb8460c3f93 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-container--isMobile", () => $ba58a329f3e1a776$export$3486d98f7bb904b1, (v) => $ba58a329f3e1a776$export$3486d98f7bb904b1 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-field", () => $ba58a329f3e1a776$export$faac6573b15f75d6, (v) => $ba58a329f3e1a776$export$faac6573b15f75d6 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-icon", () => $ba58a329f3e1a776$export$639c56ba6082b615, (v) => $ba58a329f3e1a776$export$639c56ba6082b615 = v);
$parcel$export($ba58a329f3e1a776$exports, "spectrum-Stepper-input", () => $ba58a329f3e1a776$export$8273fb931edd57d4, (v) => $ba58a329f3e1a776$export$8273fb931edd57d4 = v);
var $ba58a329f3e1a776$export$8f85cc9753f280aa;
var $ba58a329f3e1a776$export$e0f35d825088c098;
var $ba58a329f3e1a776$export$f39a09f249340e2a;
var $ba58a329f3e1a776$export$8c4ee2c50c22c514;
var $ba58a329f3e1a776$export$a2647aa13413c947;
var $ba58a329f3e1a776$export$20fd0f7cd4e6112f;
var $ba58a329f3e1a776$export$d35bc1e505d1ebbf;
var $ba58a329f3e1a776$export$e7dc768d35940237;
var $ba58a329f3e1a776$export$b8813cd5d7824ce7;
var $ba58a329f3e1a776$export$14a30de2866685fa;
var $ba58a329f3e1a776$export$a9781837241c946d;
var $ba58a329f3e1a776$export$f7b415378ab4f839;
var $ba58a329f3e1a776$export$4109102f950813a6;
var $ba58a329f3e1a776$export$24c7f46a6e3605dd;
var $ba58a329f3e1a776$export$53da69f51b770d3;
var $ba58a329f3e1a776$export$40e1f009544bacf3;
var $ba58a329f3e1a776$export$4a4f01c443eff992;
var $ba58a329f3e1a776$export$ebb4635c66bacdb9;
var $ba58a329f3e1a776$export$1db4cca5b4ade39a;
var $ba58a329f3e1a776$export$c1d306ec0fae3838;
var $ba58a329f3e1a776$export$54592d0984f245d5;
var $ba58a329f3e1a776$export$76d4531f0342bbe0;
var $ba58a329f3e1a776$export$c081ba562e204942;
var $ba58a329f3e1a776$export$b752ce409e5660c;
var $ba58a329f3e1a776$export$49280ffa5236ba8b;
var $ba58a329f3e1a776$export$a024c99ecf6b5741;
var $ba58a329f3e1a776$export$2927016961429360;
var $ba58a329f3e1a776$export$5ca7b274e4fff9fc;
var $ba58a329f3e1a776$export$d374b04f30360026;
var $ba58a329f3e1a776$export$36ae67a131730482;
var $ba58a329f3e1a776$export$f49c123746e1fb8;
var $ba58a329f3e1a776$export$75e6f9c4d1d95421;
var $ba58a329f3e1a776$export$70ed0eb2ba7512f8;
var $ba58a329f3e1a776$export$ba8faf273ff0a970;
var $ba58a329f3e1a776$export$a4e366275982a12f;
var $ba58a329f3e1a776$export$4e11596482f95f63;
var $ba58a329f3e1a776$export$e4ad52f6ba4e317b;
var $ba58a329f3e1a776$export$6c76d609d604c6e5;
var $ba58a329f3e1a776$export$7d7cb0c4055e98ab;
var $ba58a329f3e1a776$export$f49a6eb8460c3f93;
var $ba58a329f3e1a776$export$3486d98f7bb904b1;
var $ba58a329f3e1a776$export$faac6573b15f75d6;
var $ba58a329f3e1a776$export$639c56ba6082b615;
var $ba58a329f3e1a776$export$8273fb931edd57d4;
$ba58a329f3e1a776$export$8f85cc9753f280aa = `Ug6T-W_decrement`;
$ba58a329f3e1a776$export$e0f35d825088c098 = `Ug6T-W_field`;
$ba58a329f3e1a776$export$f39a09f249340e2a = `Ug6T-W_focus-ring`;
$ba58a329f3e1a776$export$8c4ee2c50c22c514 = `Ug6T-W_i18nFontFamily`;
$ba58a329f3e1a776$export$a2647aa13413c947 = `Ug6T-W_increment`;
$ba58a329f3e1a776$export$20fd0f7cd4e6112f = `Ug6T-W_is-active`;
$ba58a329f3e1a776$export$d35bc1e505d1ebbf = `Ug6T-W_is-disabled`;
$ba58a329f3e1a776$export$e7dc768d35940237 = `Ug6T-W_is-focused`;
$ba58a329f3e1a776$export$b8813cd5d7824ce7 = `Ug6T-W_is-hovered`;
$ba58a329f3e1a776$export$14a30de2866685fa = `Ug6T-W_is-invalid`;
$ba58a329f3e1a776$export$a9781837241c946d = `Ug6T-W_is-open`;
$ba58a329f3e1a776$export$f7b415378ab4f839 = `Ug6T-W_spectrum-BaseButton ${$ba58a329f3e1a776$export$8c4ee2c50c22c514}`;
$ba58a329f3e1a776$export$4109102f950813a6 = `Ug6T-W_spectrum-FocusRing-ring`;
$ba58a329f3e1a776$export$24c7f46a6e3605dd = `Ug6T-W_spectrum-FocusRing ${$ba58a329f3e1a776$export$4109102f950813a6}`;
$ba58a329f3e1a776$export$53da69f51b770d3 = `Ug6T-W_spectrum-ActionButton ${$ba58a329f3e1a776$export$f7b415378ab4f839} ${$ba58a329f3e1a776$export$24c7f46a6e3605dd}`;
$ba58a329f3e1a776$export$40e1f009544bacf3 = `Ug6T-W_spectrum-ActionButton--quiet`;
$ba58a329f3e1a776$export$4a4f01c443eff992 = `Ug6T-W_spectrum-ActionButton-hold`;
$ba58a329f3e1a776$export$ebb4635c66bacdb9 = `Ug6T-W_spectrum-ActionButton-label`;
$ba58a329f3e1a776$export$1db4cca5b4ade39a = `Ug6T-W_spectrum-Button ${$ba58a329f3e1a776$export$f7b415378ab4f839} ${$ba58a329f3e1a776$export$24c7f46a6e3605dd}`;
$ba58a329f3e1a776$export$c1d306ec0fae3838 = `Ug6T-W_spectrum-Button--iconOnly`;
$ba58a329f3e1a776$export$54592d0984f245d5 = `Ug6T-W_spectrum-Button--pending`;
$ba58a329f3e1a776$export$76d4531f0342bbe0 = `Ug6T-W_spectrum-Button-circleLoader`;
$ba58a329f3e1a776$export$c081ba562e204942 = `Ug6T-W_spectrum-Button-label`;
$ba58a329f3e1a776$export$b752ce409e5660c = `Ug6T-W_spectrum-ClearButton ${$ba58a329f3e1a776$export$f7b415378ab4f839} ${$ba58a329f3e1a776$export$24c7f46a6e3605dd}`;
$ba58a329f3e1a776$export$49280ffa5236ba8b = `Ug6T-W_spectrum-ClearButton--small`;
$ba58a329f3e1a776$export$a024c99ecf6b5741 = `Ug6T-W_spectrum-FieldButton ${$ba58a329f3e1a776$export$f7b415378ab4f839} ${$ba58a329f3e1a776$export$24c7f46a6e3605dd}`;
$ba58a329f3e1a776$export$2927016961429360 = `Ug6T-W_spectrum-FocusRing--quiet`;
$ba58a329f3e1a776$export$5ca7b274e4fff9fc = `Ug6T-W_spectrum-FieldButton--quiet ${$ba58a329f3e1a776$export$2927016961429360}`;
$ba58a329f3e1a776$export$d374b04f30360026 = `Ug6T-W_spectrum-Icon`;
$ba58a329f3e1a776$export$36ae67a131730482 = `Ug6T-W_spectrum-LogicButton ${$ba58a329f3e1a776$export$f7b415378ab4f839} ${$ba58a329f3e1a776$export$24c7f46a6e3605dd}`;
$ba58a329f3e1a776$export$f49c123746e1fb8 = `Ug6T-W_spectrum-Stepper ${$ba58a329f3e1a776$export$24c7f46a6e3605dd}`;
$ba58a329f3e1a776$export$75e6f9c4d1d95421 = `Ug6T-W_spectrum-Stepper--isMobile`;
$ba58a329f3e1a776$export$70ed0eb2ba7512f8 = `Ug6T-W_spectrum-Stepper--isQuiet ${$ba58a329f3e1a776$export$2927016961429360}`;
$ba58a329f3e1a776$export$ba8faf273ff0a970 = `Ug6T-W_spectrum-Stepper--readonly`;
$ba58a329f3e1a776$export$a4e366275982a12f = `Ug6T-W_spectrum-Stepper--showStepper`;
$ba58a329f3e1a776$export$4e11596482f95f63 = `Ug6T-W_spectrum-Stepper-button ${$ba58a329f3e1a776$export$f7b415378ab4f839}`;
$ba58a329f3e1a776$export$e4ad52f6ba4e317b = `Ug6T-W_spectrum-Stepper-button--stepDown`;
$ba58a329f3e1a776$export$6c76d609d604c6e5 = `Ug6T-W_spectrum-Stepper-button--stepUp`;
$ba58a329f3e1a776$export$7d7cb0c4055e98ab = `Ug6T-W_spectrum-Stepper-button-icon`;
$ba58a329f3e1a776$export$f49a6eb8460c3f93 = `Ug6T-W_spectrum-Stepper-container`;
$ba58a329f3e1a776$export$3486d98f7bb904b1 = `Ug6T-W_spectrum-Stepper-container--isMobile`;
$ba58a329f3e1a776$export$faac6573b15f75d6 = `Ug6T-W_spectrum-Stepper-field`;
$ba58a329f3e1a776$export$639c56ba6082b615 = `Ug6T-W_spectrum-Stepper-icon`;
$ba58a329f3e1a776$export$8273fb931edd57d4 = `Ug6T-W_spectrum-Stepper-input`;





function $7f00b388e0d3b2f6$var$StepButton(props, ref) {
    props = (0, $k6K00$useProviderProps)(props);
    let { scale: scale } = (0, $k6K00$useProvider)();
    let { direction: direction, isDisabled: isDisabled, isQuiet: isQuiet } = props;
    /**
   * Must use div for now because Safari pointer event bugs on disabled form elements.
   * Link https://bugs.webkit.org/show_bug.cgi?id=219188.
   */ let { buttonProps: buttonProps, isPressed: isPressed } = (0, $k6K00$useButton)({
        ...props,
        elementType: "div"
    }, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $k6K00$useHover)(props);
    return /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$FocusRing), {
        focusRingClass: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $k6K00$react).createElement("div", {
        className: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-button", {
            "spectrum-Stepper-button--stepUp": direction === "up",
            "spectrum-Stepper-button--stepDown": direction === "down",
            "spectrum-Stepper-button--isQuiet": isQuiet,
            "is-hovered": isHovered,
            "is-active": isPressed,
            "is-disabled": isDisabled
        }),
        ...(0, $k6K00$mergeProps)(hoverProps, buttonProps),
        ref: ref
    }, direction === "up" && scale === "large" && /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$spectrumiconsworkflowAdd), {
        UNSAFE_className: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-button-icon", "spectrum-Stepper-stepUpIcon"),
        size: "S"
    }), direction === "up" && scale === "medium" && /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$spectrumiconsuiChevronUpSmall), {
        UNSAFE_className: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-button-icon", "spectrum-Stepper-stepUpIcon")
    }), direction === "down" && scale === "large" && /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$spectrumiconsworkflowRemove), {
        UNSAFE_className: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-button-icon", "spectrum-Stepper-stepDownIcon"),
        size: "S"
    }), direction === "down" && scale === "medium" && /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$spectrumiconsuiChevronDownSmall), {
        UNSAFE_className: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-button-icon", "spectrum-Stepper-stepDownIcon")
    })));
}
/**
 * Buttons for NumberField.
 */ let $7f00b388e0d3b2f6$export$b2f6b60c1d32d6aa = /*#__PURE__*/ (0, $k6K00$react).forwardRef($7f00b388e0d3b2f6$var$StepButton);









function $48b879bdd771eb85$var$NumberField(props, ref) {
    props = (0, $k6K00$useProviderProps)(props);
    props = (0, $k6K00$useFormProps)(props);
    let provider = (0, $k6K00$useProvider)();
    let { isQuiet: isQuiet, isReadOnly: isReadOnly, isDisabled: isDisabled, hideStepper: hideStepper } = props;
    let { styleProps: style } = (0, $k6K00$useStyleProps)(props);
    let { locale: locale } = (0, $k6K00$useLocale)();
    let state = (0, $k6K00$useNumberFieldState)({
        ...props,
        locale: locale
    });
    let inputRef = (0, $k6K00$useRef)();
    let domRef = (0, $k6K00$useFocusableRef)(ref, inputRef);
    let { groupProps: groupProps, labelProps: labelProps, inputProps: inputProps, incrementButtonProps: incrementButtonProps, decrementButtonProps: decrementButtonProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = (0, $k6K00$useNumberField)(props, state, inputRef);
    let isMobile = provider.scale === "large";
    let showStepper = !hideStepper;
    let { isHovered: isHovered, hoverProps: hoverProps } = (0, $k6K00$useHover)({
        isDisabled: isDisabled
    });
    let validationState = props.validationState || (isInvalid ? "invalid" : null);
    let className = (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper", {
        "spectrum-Stepper--isQuiet": isQuiet,
        "is-disabled": isDisabled,
        "spectrum-Stepper--readonly": isReadOnly,
        "is-invalid": validationState === "invalid" && !isDisabled,
        "spectrum-Stepper--showStepper": showStepper,
        "spectrum-Stepper--isMobile": isMobile,
        "is-hovered": isHovered,
        // because FocusRing won't pass along the className from Field, we have to handle that ourselves
        [style.className]: !props.label && style.className
    });
    return /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$Field), {
        ...props,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails,
        labelProps: labelProps,
        ref: domRef,
        wrapperClassName: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-container", {
            "spectrum-Stepper-container--isMobile": isMobile
        })
    }, /*#__PURE__*/ (0, $k6K00$react).createElement($48b879bdd771eb85$var$NumberFieldInput, {
        ...props,
        groupProps: (0, $k6K00$mergeProps)(groupProps, hoverProps),
        inputProps: inputProps,
        inputRef: inputRef,
        incrementProps: incrementButtonProps,
        decrementProps: decrementButtonProps,
        className: className,
        style: style,
        state: state,
        validationState: validationState
    }));
}
const $48b879bdd771eb85$var$NumberFieldInput = /*#__PURE__*/ (0, $k6K00$react).forwardRef(function NumberFieldInput(props, ref) {
    let { groupProps: groupProps, inputProps: inputProps, inputRef: inputRef, incrementProps: incrementProps, decrementProps: decrementProps, className: className, style: style, autoFocus: autoFocus, isQuiet: isQuiet, isDisabled: isDisabled, hideStepper: hideStepper, validationState: validationState, name: name, state: state } = props;
    let showStepper = !hideStepper;
    return /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$FocusRing), {
        within: true,
        isTextInput: true,
        focusClass: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "is-focused"),
        focusRingClass: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, $k6K00$react).createElement("div", {
        ...groupProps,
        ref: ref,
        style: style,
        className: className
    }, /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$TextFieldBase), {
        UNSAFE_className: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-field"),
        inputClassName: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-input"),
        validationIconClassName: (0, $k6K00$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba58a329f3e1a776$exports))), "spectrum-Stepper-icon"),
        isQuiet: isQuiet,
        inputRef: inputRef,
        validationState: validationState,
        inputProps: inputProps,
        isDisabled: isDisabled,
        disableFocusRing: true
    }), showStepper && /*#__PURE__*/ (0, $k6K00$react).createElement((0, $k6K00$react).Fragment, null, /*#__PURE__*/ (0, $k6K00$react).createElement((0, $7f00b388e0d3b2f6$export$b2f6b60c1d32d6aa), {
        direction: "up",
        isQuiet: isQuiet,
        ...incrementProps
    }), /*#__PURE__*/ (0, $k6K00$react).createElement((0, $7f00b388e0d3b2f6$export$b2f6b60c1d32d6aa), {
        direction: "down",
        isQuiet: isQuiet,
        ...decrementProps
    })), name && /*#__PURE__*/ (0, $k6K00$react).createElement("input", {
        type: "hidden",
        name: name,
        value: isNaN(state.numberValue) ? "" : state.numberValue
    })));
});
/**
 * NumberFields allow users to enter a number, and increment or decrement the value using stepper buttons.
 */ let $48b879bdd771eb85$export$63c5fa0b2fdccd2e = /*#__PURE__*/ (0, $k6K00$react).forwardRef($48b879bdd771eb85$var$NumberField);




export {$48b879bdd771eb85$export$63c5fa0b2fdccd2e as NumberField};
