// @react-spectrum/provider/dist/main.js
"use client";

require("./main.css");
var $kmRpa$reactspectrumutils = require("@react-spectrum/utils");
var $kmRpa$clsx = require("clsx");
var $kmRpa$reactariautils = require("@react-aria/utils");
var $kmRpa$reactariai18n = require("@react-aria/i18n");
var $kmRpa$reactariaoverlays = require("@react-aria/overlays");
var $kmRpa$react = require("react");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Provider", () => $96cabdf5b0aed55b$export$2881499e37b75b9a);
$parcel$export(module.exports, "useProvider", () => $96cabdf5b0aed55b$export$693cdb10cec23617);
$parcel$export(module.exports, "useProviderProps", () => $96cabdf5b0aed55b$export$521c373ccc32c300);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $a5b4f4e252b5a021$export$841858b892ce1f4c = (0, ($parcel$interopDefault($kmRpa$react))).createContext(null);
$a5b4f4e252b5a021$export$841858b892ce1f4c.displayName = "ProviderContext";






var $22365f0f5f2dcdd5$exports = {};

$parcel$export($22365f0f5f2dcdd5$exports, "focus-ring", () => $22365f0f5f2dcdd5$export$f39a09f249340e2a, (v) => $22365f0f5f2dcdd5$export$f39a09f249340e2a = v);
$parcel$export($22365f0f5f2dcdd5$exports, "i18nFontFamily", () => $22365f0f5f2dcdd5$export$8c4ee2c50c22c514, (v) => $22365f0f5f2dcdd5$export$8c4ee2c50c22c514 = v);
$parcel$export($22365f0f5f2dcdd5$exports, "spectrum", () => $22365f0f5f2dcdd5$export$3311ab3a441bc141, (v) => $22365f0f5f2dcdd5$export$3311ab3a441bc141 = v);
$parcel$export($22365f0f5f2dcdd5$exports, "spectrum-FocusRing-ring", () => $22365f0f5f2dcdd5$export$4109102f950813a6, (v) => $22365f0f5f2dcdd5$export$4109102f950813a6 = v);
$parcel$export($22365f0f5f2dcdd5$exports, "spectrum-FocusRing", () => $22365f0f5f2dcdd5$export$24c7f46a6e3605dd, (v) => $22365f0f5f2dcdd5$export$24c7f46a6e3605dd = v);
$parcel$export($22365f0f5f2dcdd5$exports, "spectrum-FocusRing--quiet", () => $22365f0f5f2dcdd5$export$2927016961429360, (v) => $22365f0f5f2dcdd5$export$2927016961429360 = v);
var $22365f0f5f2dcdd5$export$f39a09f249340e2a;
var $22365f0f5f2dcdd5$export$8c4ee2c50c22c514;
var $22365f0f5f2dcdd5$export$3311ab3a441bc141;
var $22365f0f5f2dcdd5$export$4109102f950813a6;
var $22365f0f5f2dcdd5$export$24c7f46a6e3605dd;
var $22365f0f5f2dcdd5$export$2927016961429360;
$22365f0f5f2dcdd5$export$f39a09f249340e2a = `JuTe6q_focus-ring`;
$22365f0f5f2dcdd5$export$8c4ee2c50c22c514 = `JuTe6q_i18nFontFamily`;
$22365f0f5f2dcdd5$export$3311ab3a441bc141 = `JuTe6q_spectrum`;
$22365f0f5f2dcdd5$export$4109102f950813a6 = `JuTe6q_spectrum-FocusRing-ring`;
$22365f0f5f2dcdd5$export$24c7f46a6e3605dd = `JuTe6q_spectrum-FocusRing ${$22365f0f5f2dcdd5$export$4109102f950813a6}`;
$22365f0f5f2dcdd5$export$2927016961429360 = `JuTe6q_spectrum-FocusRing--quiet`;


var $19fbfce217006b12$exports = {};

$parcel$export($19fbfce217006b12$exports, "focus-ring", () => $19fbfce217006b12$export$f39a09f249340e2a, (v) => $19fbfce217006b12$export$f39a09f249340e2a = v);
$parcel$export($19fbfce217006b12$exports, "i18nFontFamily", () => $19fbfce217006b12$export$8c4ee2c50c22c514, (v) => $19fbfce217006b12$export$8c4ee2c50c22c514 = v);
$parcel$export($19fbfce217006b12$exports, "spectrum", () => $19fbfce217006b12$export$3311ab3a441bc141, (v) => $19fbfce217006b12$export$3311ab3a441bc141 = v);
$parcel$export($19fbfce217006b12$exports, "spectrum-Body", () => $19fbfce217006b12$export$a30bf7810c8453d1, (v) => $19fbfce217006b12$export$a30bf7810c8453d1 = v);
$parcel$export($19fbfce217006b12$exports, "spectrum-Body--italic", () => $19fbfce217006b12$export$7b2dac1166f6ec4c, (v) => $19fbfce217006b12$export$7b2dac1166f6ec4c = v);
$parcel$export($19fbfce217006b12$exports, "spectrum-FocusRing-ring", () => $19fbfce217006b12$export$4109102f950813a6, (v) => $19fbfce217006b12$export$4109102f950813a6 = v);
$parcel$export($19fbfce217006b12$exports, "spectrum-FocusRing", () => $19fbfce217006b12$export$24c7f46a6e3605dd, (v) => $19fbfce217006b12$export$24c7f46a6e3605dd = v);
$parcel$export($19fbfce217006b12$exports, "spectrum-FocusRing--quiet", () => $19fbfce217006b12$export$2927016961429360, (v) => $19fbfce217006b12$export$2927016961429360 = v);
var $19fbfce217006b12$export$f39a09f249340e2a;
var $19fbfce217006b12$export$8c4ee2c50c22c514;
var $19fbfce217006b12$export$3311ab3a441bc141;
var $19fbfce217006b12$export$a30bf7810c8453d1;
var $19fbfce217006b12$export$7b2dac1166f6ec4c;
var $19fbfce217006b12$export$4109102f950813a6;
var $19fbfce217006b12$export$24c7f46a6e3605dd;
var $19fbfce217006b12$export$2927016961429360;
$19fbfce217006b12$export$f39a09f249340e2a = `_5QszkG_focus-ring`;
$19fbfce217006b12$export$8c4ee2c50c22c514 = `_5QszkG_i18nFontFamily`;
$19fbfce217006b12$export$3311ab3a441bc141 = `_5QszkG_spectrum ${$19fbfce217006b12$export$8c4ee2c50c22c514}`;
$19fbfce217006b12$export$a30bf7810c8453d1 = `_5QszkG_spectrum-Body`;
$19fbfce217006b12$export$7b2dac1166f6ec4c = `_5QszkG_spectrum-Body--italic`;
$19fbfce217006b12$export$4109102f950813a6 = `_5QszkG_spectrum-FocusRing-ring`;
$19fbfce217006b12$export$24c7f46a6e3605dd = `_5QszkG_spectrum-FocusRing ${$19fbfce217006b12$export$4109102f950813a6}`;
$19fbfce217006b12$export$2927016961429360 = `_5QszkG_spectrum-FocusRing--quiet`;


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
function $a6e2b5f4621e4690$export$6343629ee1b29116(theme, defaultColorScheme) {
    let matchesDark = (0, $kmRpa$reactspectrumutils.useMediaQuery)("(prefers-color-scheme: dark)");
    let matchesLight = (0, $kmRpa$reactspectrumutils.useMediaQuery)("(prefers-color-scheme: light)");
    // importance OS > default > omitted
    if (theme.dark && matchesDark) return "dark";
    if (theme.light && matchesLight) return "light";
    if (theme.dark && defaultColorScheme === "dark") return "dark";
    if (theme.light && defaultColorScheme === "light") return "light";
    if (!theme.dark) return "light";
    if (!theme.light) return "dark";
    return "light";
}
function $a6e2b5f4621e4690$export$a8d2043b2d807f4d(theme) {
    let matchesFine = (0, $kmRpa$reactspectrumutils.useMediaQuery)("(any-pointer: fine)");
    if (matchesFine && theme.medium) return "medium";
    if (theme.large) return "large";
    return "medium";
}


var $4db360e38265184f$exports = {};
$4db360e38265184f$exports = JSON.parse('{"name":"@react-spectrum/provider","version":"3.9.2","description":"Spectrum UI components in React","license":"Apache-2.0","main":"dist/main.js","module":"dist/module.js","exports":{"types":"./dist/types.d.ts","import":"./dist/import.mjs","require":"./dist/main.js"},"types":"dist/types.d.ts","source":"src/index.ts","files":["dist","src"],"sideEffects":["*.css"],"targets":{"main":{"includeNodeModules":["@adobe/spectrum-css-temp"]},"module":{"includeNodeModules":["@adobe/spectrum-css-temp"]}},"repository":{"type":"git","url":"https://github.com/adobe/react-spectrum"},"dependencies":{"@react-aria/i18n":"^3.9.0","@react-aria/overlays":"^3.19.0","@react-aria/utils":"^3.22.0","@react-spectrum/utils":"^3.11.2","@react-types/provider":"^3.7.1","@react-types/shared":"^3.22.0","@swc/helpers":"^0.5.0","clsx":"^2.0.0"},"devDependencies":{"@adobe/spectrum-css-temp":"3.0.0-alpha.1"},"peerDependencies":{"react":"^16.8.0 || ^17.0.0-rc.1 || ^18.0.0","react-dom":"^16.8.0 || ^17.0.0-rc.1 || ^18.0.0"},"publishConfig":{"access":"public"}}');


const $96cabdf5b0aed55b$var$DEFAULT_BREAKPOINTS = {
    S: 640,
    M: 768,
    L: 1024,
    XL: 1280,
    XXL: 1536
};
function $96cabdf5b0aed55b$var$Provider(props, ref) {
    let prevContext = $96cabdf5b0aed55b$export$693cdb10cec23617();
    let prevColorScheme = prevContext && prevContext.colorScheme;
    let prevBreakpoints = prevContext && prevContext.breakpoints;
    let { theme: theme = prevContext && prevContext.theme, defaultColorScheme: defaultColorScheme } = props;
    if (!theme) throw new Error("theme not found, the parent provider must have a theme provided");
    // Hooks must always be called.
    let autoColorScheme = (0, $a6e2b5f4621e4690$export$6343629ee1b29116)(theme, defaultColorScheme);
    let autoScale = (0, $a6e2b5f4621e4690$export$a8d2043b2d807f4d)(theme);
    let { locale: prevLocale } = (0, $kmRpa$reactariai18n.useLocale)();
    // if the new theme doesn't support the prevColorScheme, we must resort to the auto
    let usePrevColorScheme = !!theme[prevColorScheme];
    // importance of color scheme props > parent > auto:(OS > default > omitted)
    let { colorScheme: colorScheme = usePrevColorScheme ? prevColorScheme : autoColorScheme, scale: scale = prevContext ? prevContext.scale : autoScale, locale: locale = prevContext ? prevLocale : null, breakpoints: breakpoints = prevContext ? prevBreakpoints : $96cabdf5b0aed55b$var$DEFAULT_BREAKPOINTS, children: children, isQuiet: isQuiet, isEmphasized: isEmphasized, isDisabled: isDisabled, isRequired: isRequired, isReadOnly: isReadOnly, validationState: validationState, router: router, ...otherProps } = props;
    // select only the props with values so undefined props don't overwrite prevContext values
    let currentProps = {
        version: $4db360e38265184f$exports.version,
        theme: theme,
        breakpoints: breakpoints,
        colorScheme: colorScheme,
        scale: scale,
        isQuiet: isQuiet,
        isEmphasized: isEmphasized,
        isDisabled: isDisabled,
        isRequired: isRequired,
        isReadOnly: isReadOnly,
        validationState: validationState
    };
    let matchedBreakpoints = (0, $kmRpa$reactspectrumutils.useMatchedBreakpoints)(breakpoints);
    let filteredProps = {};
    Object.entries(currentProps).forEach(([key, value])=>value !== undefined && (filteredProps[key] = value));
    // Merge options with parent provider
    let context = Object.assign({}, prevContext, filteredProps);
    // Only wrap in a DOM node if the theme, colorScheme, or scale changed
    let contents = children;
    let domProps = (0, $kmRpa$reactariautils.filterDOMProps)(otherProps);
    let { styleProps: styleProps } = (0, $kmRpa$reactspectrumutils.useStyleProps)(otherProps, undefined, {
        matchedBreakpoints: matchedBreakpoints
    });
    if (!prevContext || props.locale || theme !== prevContext.theme || colorScheme !== prevContext.colorScheme || scale !== prevContext.scale || Object.keys(domProps).length > 0 || otherProps.UNSAFE_className || Object.keys(styleProps.style).length > 0) contents = /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).createElement($96cabdf5b0aed55b$var$ProviderWrapper, {
        ...props,
        UNSAFE_style: {
            isolation: !prevContext ? "isolate" : undefined,
            ...styleProps.style
        },
        ref: ref
    }, contents);
    if (router) contents = /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).createElement((0, $kmRpa$reactariautils.RouterProvider), router, contents);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).createElement((0, $a5b4f4e252b5a021$export$841858b892ce1f4c).Provider, {
        value: context
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).createElement((0, $kmRpa$reactariai18n.I18nProvider), {
        locale: locale
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).createElement((0, $kmRpa$reactspectrumutils.BreakpointProvider), {
        matchedBreakpoints: matchedBreakpoints
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).createElement((0, $kmRpa$reactariaoverlays.ModalProvider), null, contents))));
}
/**
 * Provider is the container for all React Spectrum applications.
 * It defines the theme, locale, and other application level settings,
 * and can also be used to provide common properties to a group of components.
 */ let $96cabdf5b0aed55b$export$2881499e37b75b9a = /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).forwardRef($96cabdf5b0aed55b$var$Provider);
const $96cabdf5b0aed55b$var$ProviderWrapper = /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).forwardRef(function ProviderWrapper(props, ref) {
    let { children: children, ...otherProps } = props;
    let { locale: locale, direction: direction } = (0, $kmRpa$reactariai18n.useLocale)();
    let { theme: theme, colorScheme: colorScheme, scale: scale } = $96cabdf5b0aed55b$export$693cdb10cec23617();
    let { modalProviderProps: modalProviderProps } = (0, $kmRpa$reactariaoverlays.useModalProvider)();
    let { styleProps: styleProps } = (0, $kmRpa$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $kmRpa$reactspectrumutils.useDOMRef)(ref);
    let themeKey = Object.keys(theme[colorScheme])[0];
    let scaleKey = Object.keys(theme[scale])[0];
    let className = (0, ($parcel$interopDefault($kmRpa$clsx)))(styleProps.className, (0, (/*@__PURE__*/$parcel$interopDefault($22365f0f5f2dcdd5$exports)))["spectrum"], (0, (/*@__PURE__*/$parcel$interopDefault($19fbfce217006b12$exports)))["spectrum"], Object.values(theme[colorScheme]), Object.values(theme[scale]), theme.global ? Object.values(theme.global) : null, {
        "react-spectrum-provider": (0, $kmRpa$reactspectrumutils.shouldKeepSpectrumClassNames),
        spectrum: (0, $kmRpa$reactspectrumutils.shouldKeepSpectrumClassNames),
        [themeKey]: (0, $kmRpa$reactspectrumutils.shouldKeepSpectrumClassNames),
        [scaleKey]: (0, $kmRpa$reactspectrumutils.shouldKeepSpectrumClassNames)
    });
    var _props_colorScheme, _ref;
    let style = {
        ...styleProps.style,
        // This ensures that browser native UI like scrollbars are rendered in the right color scheme.
        // See https://web.dev/color-scheme/.
        colorScheme: (_ref = (_props_colorScheme = props.colorScheme) !== null && _props_colorScheme !== void 0 ? _props_colorScheme : colorScheme) !== null && _ref !== void 0 ? _ref : Object.keys(theme).filter((k)=>k === "light" || k === "dark").join(" ")
    };
    let hasWarned = (0, $kmRpa$react.useRef)(false);
    (0, $kmRpa$react.useEffect)(()=>{
        if (direction && domRef.current) {
            let closestDir = domRef.current.parentElement.closest("[dir]");
            let dir = closestDir && closestDir.getAttribute("dir");
            if (dir && dir !== direction && !hasWarned.current) {
                console.warn(`Language directions cannot be nested. ${direction} inside ${dir}.`);
                hasWarned.current = true;
            }
        }
    }, [
        direction,
        domRef,
        hasWarned
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kmRpa$react))).createElement("div", {
        ...(0, $kmRpa$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ...modalProviderProps,
        className: className,
        style: style,
        lang: locale,
        dir: direction,
        ref: domRef
    }, children);
});
function $96cabdf5b0aed55b$export$693cdb10cec23617() {
    return (0, $kmRpa$react.useContext)((0, $a5b4f4e252b5a021$export$841858b892ce1f4c));
}
function $96cabdf5b0aed55b$export$521c373ccc32c300(props) {
    let context = $96cabdf5b0aed55b$export$693cdb10cec23617();
    if (!context) return props;
    return Object.assign({}, {
        isQuiet: context.isQuiet,
        isEmphasized: context.isEmphasized,
        isDisabled: context.isDisabled,
        isRequired: context.isRequired,
        isReadOnly: context.isReadOnly,
        validationState: context.validationState
    }, props);
}
