// @react-spectrum/checkbox/dist/main.js
"use client";

require("./main.css");
var $5Ft2u$spectrumiconsuiCheckmarkSmall = require("@spectrum-icons/ui/CheckmarkSmall");
var $5Ft2u$reactspectrumutils = require("@react-spectrum/utils");
var $5Ft2u$spectrumiconsuiDashSmall = require("@spectrum-icons/ui/DashSmall");
var $5Ft2u$reactariafocus = require("@react-aria/focus");
var $5Ft2u$react = require("react");
var $5Ft2u$reactariacheckbox = require("@react-aria/checkbox");
var $5Ft2u$reactspectrumform = require("@react-spectrum/form");
var $5Ft2u$reactariainteractions = require("@react-aria/interactions");
var $5Ft2u$reactspectrumprovider = require("@react-spectrum/provider");
var $5Ft2u$reactstatelytoggle = require("@react-stately/toggle");
var $5Ft2u$reactspectrumlabel = require("@react-spectrum/label");
var $5Ft2u$reactstatelycheckbox = require("@react-stately/checkbox");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Checkbox", () => $096ccd12d23dd6cc$export$48513f6b9f8ce62d);
$parcel$export(module.exports, "CheckboxGroup", () => $1151290755e16c44$export$4aa08d5625cb8ead);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $bdf3bf7fe71fe120$export$baf37c4be89255b8 = (0, ($parcel$interopDefault($5Ft2u$react))).createContext(null);







var $591f3ccba6339bdb$exports = {};

$parcel$export($591f3ccba6339bdb$exports, "focus-ring", () => $591f3ccba6339bdb$export$f39a09f249340e2a, (v) => $591f3ccba6339bdb$export$f39a09f249340e2a = v);
$parcel$export($591f3ccba6339bdb$exports, "i18nFontFamily", () => $591f3ccba6339bdb$export$8c4ee2c50c22c514, (v) => $591f3ccba6339bdb$export$8c4ee2c50c22c514 = v);
$parcel$export($591f3ccba6339bdb$exports, "is-checked", () => $591f3ccba6339bdb$export$3f46c891456641e7, (v) => $591f3ccba6339bdb$export$3f46c891456641e7 = v);
$parcel$export($591f3ccba6339bdb$exports, "is-disabled", () => $591f3ccba6339bdb$export$d35bc1e505d1ebbf, (v) => $591f3ccba6339bdb$export$d35bc1e505d1ebbf = v);
$parcel$export($591f3ccba6339bdb$exports, "is-hovered", () => $591f3ccba6339bdb$export$b8813cd5d7824ce7, (v) => $591f3ccba6339bdb$export$b8813cd5d7824ce7 = v);
$parcel$export($591f3ccba6339bdb$exports, "is-indeterminate", () => $591f3ccba6339bdb$export$682c8615600f6ee, (v) => $591f3ccba6339bdb$export$682c8615600f6ee = v);
$parcel$export($591f3ccba6339bdb$exports, "is-invalid", () => $591f3ccba6339bdb$export$14a30de2866685fa, (v) => $591f3ccba6339bdb$export$14a30de2866685fa = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-Checkbox", () => $591f3ccba6339bdb$export$f8ee3e99d637ccac, (v) => $591f3ccba6339bdb$export$f8ee3e99d637ccac = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-Checkbox--quiet", () => $591f3ccba6339bdb$export$3a2eb50fe5450439, (v) => $591f3ccba6339bdb$export$3a2eb50fe5450439 = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-Checkbox-box", () => $591f3ccba6339bdb$export$b07bd551df410935, (v) => $591f3ccba6339bdb$export$b07bd551df410935 = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-Checkbox-checkmark", () => $591f3ccba6339bdb$export$e152b25f70350d85, (v) => $591f3ccba6339bdb$export$e152b25f70350d85 = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-Checkbox-input", () => $591f3ccba6339bdb$export$fbf255e91152b086, (v) => $591f3ccba6339bdb$export$fbf255e91152b086 = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-Checkbox-label", () => $591f3ccba6339bdb$export$9244b14cfa532d4f, (v) => $591f3ccba6339bdb$export$9244b14cfa532d4f = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-Checkbox-partialCheckmark", () => $591f3ccba6339bdb$export$7f6ea68ad4b1fa69, (v) => $591f3ccba6339bdb$export$7f6ea68ad4b1fa69 = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-FocusRing-ring", () => $591f3ccba6339bdb$export$4109102f950813a6, (v) => $591f3ccba6339bdb$export$4109102f950813a6 = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-FocusRing", () => $591f3ccba6339bdb$export$24c7f46a6e3605dd, (v) => $591f3ccba6339bdb$export$24c7f46a6e3605dd = v);
$parcel$export($591f3ccba6339bdb$exports, "spectrum-FocusRing--quiet", () => $591f3ccba6339bdb$export$2927016961429360, (v) => $591f3ccba6339bdb$export$2927016961429360 = v);
var $591f3ccba6339bdb$export$f39a09f249340e2a;
var $591f3ccba6339bdb$export$8c4ee2c50c22c514;
var $591f3ccba6339bdb$export$3f46c891456641e7;
var $591f3ccba6339bdb$export$d35bc1e505d1ebbf;
var $591f3ccba6339bdb$export$b8813cd5d7824ce7;
var $591f3ccba6339bdb$export$682c8615600f6ee;
var $591f3ccba6339bdb$export$14a30de2866685fa;
var $591f3ccba6339bdb$export$f8ee3e99d637ccac;
var $591f3ccba6339bdb$export$3a2eb50fe5450439;
var $591f3ccba6339bdb$export$b07bd551df410935;
var $591f3ccba6339bdb$export$e152b25f70350d85;
var $591f3ccba6339bdb$export$fbf255e91152b086;
var $591f3ccba6339bdb$export$9244b14cfa532d4f;
var $591f3ccba6339bdb$export$7f6ea68ad4b1fa69;
var $591f3ccba6339bdb$export$4109102f950813a6;
var $591f3ccba6339bdb$export$24c7f46a6e3605dd;
var $591f3ccba6339bdb$export$2927016961429360;
$591f3ccba6339bdb$export$f39a09f249340e2a = `ISsn1a_focus-ring`;
$591f3ccba6339bdb$export$8c4ee2c50c22c514 = `ISsn1a_i18nFontFamily`;
$591f3ccba6339bdb$export$3f46c891456641e7 = `ISsn1a_is-checked`;
$591f3ccba6339bdb$export$d35bc1e505d1ebbf = `ISsn1a_is-disabled`;
$591f3ccba6339bdb$export$b8813cd5d7824ce7 = `ISsn1a_is-hovered`;
$591f3ccba6339bdb$export$682c8615600f6ee = `ISsn1a_is-indeterminate`;
$591f3ccba6339bdb$export$14a30de2866685fa = `ISsn1a_is-invalid`;
$591f3ccba6339bdb$export$f8ee3e99d637ccac = `ISsn1a_spectrum-Checkbox`;
$591f3ccba6339bdb$export$3a2eb50fe5450439 = `ISsn1a_spectrum-Checkbox--quiet`;
$591f3ccba6339bdb$export$b07bd551df410935 = `ISsn1a_spectrum-Checkbox-box`;
$591f3ccba6339bdb$export$e152b25f70350d85 = `ISsn1a_spectrum-Checkbox-checkmark`;
$591f3ccba6339bdb$export$fbf255e91152b086 = `ISsn1a_spectrum-Checkbox-input`;
$591f3ccba6339bdb$export$9244b14cfa532d4f = `ISsn1a_spectrum-Checkbox-label`;
$591f3ccba6339bdb$export$7f6ea68ad4b1fa69 = `ISsn1a_spectrum-Checkbox-partialCheckmark`;
$591f3ccba6339bdb$export$4109102f950813a6 = `ISsn1a_spectrum-FocusRing-ring`;
$591f3ccba6339bdb$export$24c7f46a6e3605dd = `ISsn1a_spectrum-FocusRing ${$591f3ccba6339bdb$export$4109102f950813a6}`;
$591f3ccba6339bdb$export$2927016961429360 = `ISsn1a_spectrum-FocusRing--quiet`;







function $096ccd12d23dd6cc$var$Checkbox(props, ref) {
    let originalProps = props;
    props = (0, $5Ft2u$reactspectrumprovider.useProviderProps)(props);
    props = (0, $5Ft2u$reactspectrumform.useFormProps)(props);
    let { isIndeterminate: isIndeterminate = false, isEmphasized: isEmphasized = false, autoFocus: autoFocus, children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $5Ft2u$reactspectrumutils.useStyleProps)(otherProps);
    let inputRef = (0, $5Ft2u$react.useRef)(null);
    let domRef = (0, $5Ft2u$reactspectrumutils.useFocusableRef)(ref, inputRef);
    // Swap hooks depending on whether this checkbox is inside a CheckboxGroup.
    // This is a bit unorthodox. Typically, hooks cannot be called in a conditional,
    // but since the checkbox won't move in and out of a group, it should be safe.
    let groupState = (0, $5Ft2u$react.useContext)((0, $bdf3bf7fe71fe120$export$baf37c4be89255b8));
    let { inputProps: inputProps, isInvalid: isInvalid, isDisabled: isDisabled } = groupState ? (0, $5Ft2u$reactariacheckbox.useCheckboxGroupItem)({
        ...props,
        // Value is optional for standalone checkboxes, but required for CheckboxGroup items;
        // it's passed explicitly here to avoid typescript error (requires ignore).
        // @ts-ignore
        value: props.value,
        // Only pass isRequired and validationState to react-aria if they came from
        // the props for this individual checkbox, and not from the group via context.
        isRequired: originalProps.isRequired,
        validationState: originalProps.validationState,
        isInvalid: originalProps.isInvalid
    }, groupState, inputRef) : (0, $5Ft2u$reactariacheckbox.useCheckbox)(props, (0, $5Ft2u$reactstatelytoggle.useToggleState)(props), inputRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $5Ft2u$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let markIcon = isIndeterminate ? /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement((0, ($parcel$interopDefault($5Ft2u$spectrumiconsuiDashSmall))), {
        UNSAFE_className: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($591f3ccba6339bdb$exports))), "spectrum-Checkbox-partialCheckmark")
    }) : /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement((0, ($parcel$interopDefault($5Ft2u$spectrumiconsuiCheckmarkSmall))), {
        UNSAFE_className: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($591f3ccba6339bdb$exports))), "spectrum-Checkbox-checkmark")
    });
    if (groupState) {
        for (let key of [
            "isSelected",
            "defaultSelected",
            "isEmphasized"
        ])if (originalProps[key] != null) console.warn(`${key} is unsupported on individual <Checkbox> elements within a <CheckboxGroup>. Please apply these props to the group instead.`);
        if (props.value == null) console.warn("A <Checkbox> element within a <CheckboxGroup> requires a `value` property.");
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($591f3ccba6339bdb$exports))), "spectrum-Checkbox", {
            "is-checked": inputProps.checked,
            "is-indeterminate": isIndeterminate,
            "spectrum-Checkbox--quiet": !isEmphasized,
            "is-invalid": isInvalid,
            "is-disabled": isDisabled,
            "is-hovered": isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement((0, $5Ft2u$reactariafocus.FocusRing), {
        focusRingClass: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($591f3ccba6339bdb$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($591f3ccba6339bdb$exports))), "spectrum-Checkbox-input")
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement("span", {
        className: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($591f3ccba6339bdb$exports))), "spectrum-Checkbox-box")
    }, markIcon), children && /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement("span", {
        className: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($591f3ccba6339bdb$exports))), "spectrum-Checkbox-label")
    }, children));
}
/**
 * Checkboxes allow users to select multiple items from a list of individual items,
 * or to mark one individual item as selected.
 */ let $096ccd12d23dd6cc$export$48513f6b9f8ce62d = /*#__PURE__*/ (0, $5Ft2u$react.forwardRef)($096ccd12d23dd6cc$var$Checkbox);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $53124a3a0fd56fca$exports = {};

$parcel$export($53124a3a0fd56fca$exports, "focus-ring", () => $53124a3a0fd56fca$export$f39a09f249340e2a, (v) => $53124a3a0fd56fca$export$f39a09f249340e2a = v);
$parcel$export($53124a3a0fd56fca$exports, "i18nFontFamily", () => $53124a3a0fd56fca$export$8c4ee2c50c22c514, (v) => $53124a3a0fd56fca$export$8c4ee2c50c22c514 = v);
$parcel$export($53124a3a0fd56fca$exports, "spectrum-FieldGroup", () => $53124a3a0fd56fca$export$75eb16cbfc18279c, (v) => $53124a3a0fd56fca$export$75eb16cbfc18279c = v);
$parcel$export($53124a3a0fd56fca$exports, "spectrum-FieldGroup-group", () => $53124a3a0fd56fca$export$984c7bbbfc7ddffc, (v) => $53124a3a0fd56fca$export$984c7bbbfc7ddffc = v);
$parcel$export($53124a3a0fd56fca$exports, "spectrum-FieldGroup-group--horizontal", () => $53124a3a0fd56fca$export$1ad8646bcba21c0e, (v) => $53124a3a0fd56fca$export$1ad8646bcba21c0e = v);
$parcel$export($53124a3a0fd56fca$exports, "spectrum-FocusRing-ring", () => $53124a3a0fd56fca$export$4109102f950813a6, (v) => $53124a3a0fd56fca$export$4109102f950813a6 = v);
$parcel$export($53124a3a0fd56fca$exports, "spectrum-FocusRing", () => $53124a3a0fd56fca$export$24c7f46a6e3605dd, (v) => $53124a3a0fd56fca$export$24c7f46a6e3605dd = v);
$parcel$export($53124a3a0fd56fca$exports, "spectrum-FocusRing--quiet", () => $53124a3a0fd56fca$export$2927016961429360, (v) => $53124a3a0fd56fca$export$2927016961429360 = v);
var $53124a3a0fd56fca$export$f39a09f249340e2a;
var $53124a3a0fd56fca$export$8c4ee2c50c22c514;
var $53124a3a0fd56fca$export$75eb16cbfc18279c;
var $53124a3a0fd56fca$export$984c7bbbfc7ddffc;
var $53124a3a0fd56fca$export$1ad8646bcba21c0e;
var $53124a3a0fd56fca$export$4109102f950813a6;
var $53124a3a0fd56fca$export$24c7f46a6e3605dd;
var $53124a3a0fd56fca$export$2927016961429360;
$53124a3a0fd56fca$export$f39a09f249340e2a = `hyn22G_focus-ring`;
$53124a3a0fd56fca$export$8c4ee2c50c22c514 = `hyn22G_i18nFontFamily`;
$53124a3a0fd56fca$export$75eb16cbfc18279c = `hyn22G_spectrum-FieldGroup`;
$53124a3a0fd56fca$export$984c7bbbfc7ddffc = `hyn22G_spectrum-FieldGroup-group`;
$53124a3a0fd56fca$export$1ad8646bcba21c0e = `hyn22G_spectrum-FieldGroup-group--horizontal`;
$53124a3a0fd56fca$export$4109102f950813a6 = `hyn22G_spectrum-FocusRing-ring`;
$53124a3a0fd56fca$export$24c7f46a6e3605dd = `hyn22G_spectrum-FocusRing ${$53124a3a0fd56fca$export$4109102f950813a6}`;
$53124a3a0fd56fca$export$2927016961429360 = `hyn22G_spectrum-FocusRing--quiet`;





function $1151290755e16c44$var$CheckboxGroup(props, ref) {
    props = (0, $5Ft2u$reactspectrumprovider.useProviderProps)(props);
    props = (0, $5Ft2u$reactspectrumform.useFormProps)(props);
    let { isEmphasized: isEmphasized, children: children, orientation: orientation = "vertical" } = props;
    let domRef = (0, $5Ft2u$reactspectrumutils.useDOMRef)(ref);
    let state = (0, $5Ft2u$reactstatelycheckbox.useCheckboxGroupState)(props);
    let { groupProps: groupProps, ...otherProps } = (0, $5Ft2u$reactariacheckbox.useCheckboxGroup)(props, state);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement((0, $5Ft2u$reactspectrumlabel.Field), {
        ...props,
        ...otherProps,
        ref: domRef,
        wrapperClassName: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($53124a3a0fd56fca$exports))), "spectrum-FieldGroup"),
        elementType: "span",
        includeNecessityIndicatorInAccessibilityName: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement("div", {
        ...groupProps,
        className: (0, $5Ft2u$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($53124a3a0fd56fca$exports))), "spectrum-FieldGroup-group", {
            "spectrum-FieldGroup-group--horizontal": orientation === "horizontal"
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement((0, $5Ft2u$reactspectrumprovider.Provider), {
        isEmphasized: isEmphasized
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).createElement((0, $bdf3bf7fe71fe120$export$baf37c4be89255b8).Provider, {
        value: state
    }, children))));
}
/**
 * A CheckboxGroup allows users to select one or more items from a list of choices.
 */ const $1151290755e16c44$export$4aa08d5625cb8ead = /*#__PURE__*/ (0, ($parcel$interopDefault($5Ft2u$react))).forwardRef($1151290755e16c44$var$CheckboxGroup);
