// @react-spectrum/textfield/dist/main.js
"use client";

require("./main.css");
var $74aPB$reactariautils = require("@react-aria/utils");
var $74aPB$react = require("react");
var $74aPB$reactstatelyutils = require("@react-stately/utils");
var $74aPB$reactspectrumform = require("@react-spectrum/form");
var $74aPB$reactspectrumprovider = require("@react-spectrum/provider");
var $74aPB$reactariatextfield = require("@react-aria/textfield");
var $74aPB$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $74aPB$spectrumiconsuiCheckmarkMedium = require("@spectrum-icons/ui/CheckmarkMedium");
var $74aPB$reactspectrumutils = require("@react-spectrum/utils");
var $74aPB$reactspectrumlabel = require("@react-spectrum/label");
var $74aPB$reactariafocus = require("@react-aria/focus");
var $74aPB$reactariainteractions = require("@react-aria/interactions");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "TextArea", () => $eec649e464f3832d$export$f5c9f3c2c4054eec);
$parcel$export(module.exports, "TextField", () => $34fd51d50eb4b056$export$2c73285ae9390cec);
$parcel$export(module.exports, "TextFieldBase", () => $86793250183ac29e$export$d22444a338b6e3c2);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





var $40cb3a00c193680f$exports = {};

$parcel$export($40cb3a00c193680f$exports, "focus-ring", () => $40cb3a00c193680f$export$f39a09f249340e2a, (v) => $40cb3a00c193680f$export$f39a09f249340e2a = v);
$parcel$export($40cb3a00c193680f$exports, "i18nFontFamily", () => $40cb3a00c193680f$export$8c4ee2c50c22c514, (v) => $40cb3a00c193680f$export$8c4ee2c50c22c514 = v);
$parcel$export($40cb3a00c193680f$exports, "is-disabled", () => $40cb3a00c193680f$export$d35bc1e505d1ebbf, (v) => $40cb3a00c193680f$export$d35bc1e505d1ebbf = v);
$parcel$export($40cb3a00c193680f$exports, "is-focused", () => $40cb3a00c193680f$export$e7dc768d35940237, (v) => $40cb3a00c193680f$export$e7dc768d35940237 = v);
$parcel$export($40cb3a00c193680f$exports, "is-hovered", () => $40cb3a00c193680f$export$b8813cd5d7824ce7, (v) => $40cb3a00c193680f$export$b8813cd5d7824ce7 = v);
$parcel$export($40cb3a00c193680f$exports, "is-placeholder", () => $40cb3a00c193680f$export$e5b2f5233e4e5194, (v) => $40cb3a00c193680f$export$e5b2f5233e4e5194 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-FocusRing-ring", () => $40cb3a00c193680f$export$4109102f950813a6, (v) => $40cb3a00c193680f$export$4109102f950813a6 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-FocusRing", () => $40cb3a00c193680f$export$24c7f46a6e3605dd, (v) => $40cb3a00c193680f$export$24c7f46a6e3605dd = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-FocusRing--quiet", () => $40cb3a00c193680f$export$2927016961429360, (v) => $40cb3a00c193680f$export$2927016961429360 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield", () => $40cb3a00c193680f$export$2c8a10299d8b3418, (v) => $40cb3a00c193680f$export$2c8a10299d8b3418 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--invalid", () => $40cb3a00c193680f$export$58e50b10d30123df, (v) => $40cb3a00c193680f$export$58e50b10d30123df = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--loadable", () => $40cb3a00c193680f$export$14fd30fea6f1348c, (v) => $40cb3a00c193680f$export$14fd30fea6f1348c = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--multiline", () => $40cb3a00c193680f$export$9699f1c538a448a0, (v) => $40cb3a00c193680f$export$9699f1c538a448a0 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--quiet", () => $40cb3a00c193680f$export$bffd5e3d61a81737, (v) => $40cb3a00c193680f$export$bffd5e3d61a81737 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield--valid", () => $40cb3a00c193680f$export$c2abbc3f970170b7, (v) => $40cb3a00c193680f$export$c2abbc3f970170b7 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-circleLoader", () => $40cb3a00c193680f$export$82a249fb4d6127, (v) => $40cb3a00c193680f$export$82a249fb4d6127 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-icon", () => $40cb3a00c193680f$export$222d1ccd1870be1d, (v) => $40cb3a00c193680f$export$222d1ccd1870be1d = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-input", () => $40cb3a00c193680f$export$1ac009feb2b0bd0c, (v) => $40cb3a00c193680f$export$1ac009feb2b0bd0c = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-inputIcon", () => $40cb3a00c193680f$export$c1cdcaa5fe76a871, (v) => $40cb3a00c193680f$export$c1cdcaa5fe76a871 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-validationIcon", () => $40cb3a00c193680f$export$cf0ead5b44db0da3, (v) => $40cb3a00c193680f$export$cf0ead5b44db0da3 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-wrapper", () => $40cb3a00c193680f$export$dd1165e0e6012973, (v) => $40cb3a00c193680f$export$dd1165e0e6012973 = v);
$parcel$export($40cb3a00c193680f$exports, "spectrum-Textfield-wrapper--quiet", () => $40cb3a00c193680f$export$9fcb3d6af94a9d12, (v) => $40cb3a00c193680f$export$9fcb3d6af94a9d12 = v);
var $40cb3a00c193680f$export$f39a09f249340e2a;
var $40cb3a00c193680f$export$8c4ee2c50c22c514;
var $40cb3a00c193680f$export$d35bc1e505d1ebbf;
var $40cb3a00c193680f$export$e7dc768d35940237;
var $40cb3a00c193680f$export$b8813cd5d7824ce7;
var $40cb3a00c193680f$export$e5b2f5233e4e5194;
var $40cb3a00c193680f$export$4109102f950813a6;
var $40cb3a00c193680f$export$24c7f46a6e3605dd;
var $40cb3a00c193680f$export$2927016961429360;
var $40cb3a00c193680f$export$2c8a10299d8b3418;
var $40cb3a00c193680f$export$58e50b10d30123df;
var $40cb3a00c193680f$export$14fd30fea6f1348c;
var $40cb3a00c193680f$export$9699f1c538a448a0;
var $40cb3a00c193680f$export$bffd5e3d61a81737;
var $40cb3a00c193680f$export$c2abbc3f970170b7;
var $40cb3a00c193680f$export$82a249fb4d6127;
var $40cb3a00c193680f$export$222d1ccd1870be1d;
var $40cb3a00c193680f$export$1ac009feb2b0bd0c;
var $40cb3a00c193680f$export$c1cdcaa5fe76a871;
var $40cb3a00c193680f$export$cf0ead5b44db0da3;
var $40cb3a00c193680f$export$dd1165e0e6012973;
var $40cb3a00c193680f$export$9fcb3d6af94a9d12;
$40cb3a00c193680f$export$f39a09f249340e2a = `YO3Nla_focus-ring`;
$40cb3a00c193680f$export$8c4ee2c50c22c514 = `YO3Nla_i18nFontFamily`;
$40cb3a00c193680f$export$d35bc1e505d1ebbf = `YO3Nla_is-disabled`;
$40cb3a00c193680f$export$e7dc768d35940237 = `YO3Nla_is-focused`;
$40cb3a00c193680f$export$b8813cd5d7824ce7 = `YO3Nla_is-hovered`;
$40cb3a00c193680f$export$e5b2f5233e4e5194 = `YO3Nla_is-placeholder`;
$40cb3a00c193680f$export$4109102f950813a6 = `YO3Nla_spectrum-FocusRing-ring`;
$40cb3a00c193680f$export$24c7f46a6e3605dd = `YO3Nla_spectrum-FocusRing ${$40cb3a00c193680f$export$4109102f950813a6}`;
$40cb3a00c193680f$export$2927016961429360 = `YO3Nla_spectrum-FocusRing--quiet`;
$40cb3a00c193680f$export$2c8a10299d8b3418 = `YO3Nla_spectrum-Textfield ${$40cb3a00c193680f$export$24c7f46a6e3605dd}`;
$40cb3a00c193680f$export$58e50b10d30123df = `YO3Nla_spectrum-Textfield--invalid`;
$40cb3a00c193680f$export$14fd30fea6f1348c = `YO3Nla_spectrum-Textfield--loadable`;
$40cb3a00c193680f$export$9699f1c538a448a0 = `YO3Nla_spectrum-Textfield--multiline`;
$40cb3a00c193680f$export$bffd5e3d61a81737 = `YO3Nla_spectrum-Textfield--quiet ${$40cb3a00c193680f$export$2927016961429360}`;
$40cb3a00c193680f$export$c2abbc3f970170b7 = `YO3Nla_spectrum-Textfield--valid`;
$40cb3a00c193680f$export$82a249fb4d6127 = `YO3Nla_spectrum-Textfield-circleLoader`;
$40cb3a00c193680f$export$222d1ccd1870be1d = `YO3Nla_spectrum-Textfield-icon`;
$40cb3a00c193680f$export$1ac009feb2b0bd0c = `YO3Nla_spectrum-Textfield-input ${$40cb3a00c193680f$export$8c4ee2c50c22c514}`;
$40cb3a00c193680f$export$c1cdcaa5fe76a871 = `YO3Nla_spectrum-Textfield-inputIcon`;
$40cb3a00c193680f$export$cf0ead5b44db0da3 = `YO3Nla_spectrum-Textfield-validationIcon`;
$40cb3a00c193680f$export$dd1165e0e6012973 = `YO3Nla_spectrum-Textfield-wrapper`;
$40cb3a00c193680f$export$9fcb3d6af94a9d12 = `YO3Nla_spectrum-Textfield-wrapper--quiet`;




function $86793250183ac29e$var$TextFieldBase(props, ref) {
    let { validationState: validationState = props.isInvalid ? "invalid" : null, icon: icon, isQuiet: isQuiet = false, isDisabled: isDisabled, multiLine: multiLine, autoFocus: autoFocus, inputClassName: inputClassName, wrapperChildren: wrapperChildren, labelProps: labelProps, inputProps: inputProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps, inputRef: userInputRef, isLoading: isLoading, loadingIndicator: loadingIndicator, validationIconClassName: validationIconClassName, disableFocusRing: disableFocusRing } = props;
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $74aPB$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let domRef = (0, $74aPB$react.useRef)(null);
    let defaultInputRef = (0, $74aPB$react.useRef)(null);
    let inputRef = userInputRef || defaultInputRef;
    // Expose imperative interface for ref
    (0, $74aPB$react.useImperativeHandle)(ref, ()=>({
            ...(0, $74aPB$reactspectrumutils.createFocusableRef)(domRef, inputRef),
            select () {
                if (inputRef.current) inputRef.current.select();
            },
            getInputElement () {
                return inputRef.current;
            }
        }));
    let ElementType = multiLine ? "textarea" : "input";
    let isInvalid = validationState === "invalid" && !isDisabled;
    if (icon) {
        let UNSAFE_className = (0, $74aPB$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports))), icon.props && icon.props.UNSAFE_className, "spectrum-Textfield-icon");
        icon = /*#__PURE__*/ (0, $74aPB$react.cloneElement)(icon, {
            UNSAFE_className: UNSAFE_className,
            size: "S"
        });
    }
    let validationIcon = isInvalid ? /*#__PURE__*/ (0, ($parcel$interopDefault($74aPB$react))).createElement((0, ($parcel$interopDefault($74aPB$spectrumiconsuiAlertMedium))), null) : /*#__PURE__*/ (0, ($parcel$interopDefault($74aPB$react))).createElement((0, ($parcel$interopDefault($74aPB$spectrumiconsuiCheckmarkMedium))), null);
    let validation = /*#__PURE__*/ (0, $74aPB$react.cloneElement)(validationIcon, {
        UNSAFE_className: (0, $74aPB$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports))), "spectrum-Textfield-validationIcon", validationIconClassName)
    });
    let { focusProps: focusProps, isFocusVisible: isFocusVisible } = (0, $74aPB$reactariafocus.useFocusRing)({
        isTextInput: true,
        autoFocus: autoFocus
    });
    let textField = /*#__PURE__*/ (0, ($parcel$interopDefault($74aPB$react))).createElement("div", {
        className: (0, $74aPB$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports))), "spectrum-Textfield", {
            "spectrum-Textfield--invalid": isInvalid,
            "spectrum-Textfield--valid": validationState === "valid" && !isDisabled,
            "spectrum-Textfield--loadable": loadingIndicator,
            "spectrum-Textfield--quiet": isQuiet,
            "spectrum-Textfield--multiline": multiLine,
            "focus-ring": !disableFocusRing && isFocusVisible
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($74aPB$react))).createElement(ElementType, {
        ...(0, $74aPB$reactariautils.mergeProps)(inputProps, hoverProps, focusProps),
        ref: inputRef,
        rows: multiLine ? 1 : undefined,
        className: (0, $74aPB$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports))), "spectrum-Textfield-input", {
            "spectrum-Textfield-inputIcon": icon,
            "is-hovered": isHovered
        }, inputClassName)
    }), icon, validationState && !isLoading && !isDisabled ? validation : null, isLoading && loadingIndicator, wrapperChildren);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($74aPB$react))).createElement((0, $74aPB$reactspectrumlabel.Field), {
        ...props,
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        wrapperClassName: (0, $74aPB$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($40cb3a00c193680f$exports))), "spectrum-Textfield-wrapper", {
            "spectrum-Textfield-wrapper--quiet": isQuiet
        }),
        showErrorIcon: false,
        ref: domRef
    }, textField);
}
const $86793250183ac29e$export$d22444a338b6e3c2 = /*#__PURE__*/ (0, $74aPB$react.forwardRef)($86793250183ac29e$var$TextFieldBase);






function $eec649e464f3832d$var$TextArea(props, ref) {
    props = (0, $74aPB$reactspectrumprovider.useProviderProps)(props);
    props = (0, $74aPB$reactspectrumform.useFormProps)(props);
    let { isDisabled: isDisabled = false, isQuiet: isQuiet = false, isReadOnly: isReadOnly = false, isRequired: isRequired = false, onChange: onChange, ...otherProps } = props;
    var _props_defaultValue;
    // not in stately because this is so we know when to re-measure, which is a spectrum design
    let [inputValue, setInputValue] = (0, $74aPB$reactstatelyutils.useControlledState)(props.value, (_props_defaultValue = props.defaultValue) !== null && _props_defaultValue !== void 0 ? _props_defaultValue : "", ()=>{});
    let inputRef = (0, $74aPB$react.useRef)(null);
    let onHeightChange = (0, $74aPB$react.useCallback)(()=>{
        // Quiet textareas always grow based on their text content.
        // Standard textareas also grow by default, unless an explicit height is set.
        if ((isQuiet || !props.height) && inputRef.current) {
            let input = inputRef.current;
            let prevAlignment = input.style.alignSelf;
            let prevOverflow = input.style.overflow;
            // Firefox scroll position is lost when overflow: 'hidden' is applied so we skip applying it.
            // The measure/applied height is also incorrect/reset if we turn on and off
            // overflow: hidden in Firefox https://bugzilla.mozilla.org/show_bug.cgi?id=1787062
            let isFirefox = "MozAppearance" in input.style;
            if (!isFirefox) input.style.overflow = "hidden";
            input.style.alignSelf = "start";
            input.style.height = "auto";
            // offsetHeight - clientHeight accounts for the border/padding.
            input.style.height = `${input.scrollHeight + (input.offsetHeight - input.clientHeight)}px`;
            input.style.overflow = prevOverflow;
            input.style.alignSelf = prevAlignment;
        }
    }, [
        isQuiet,
        inputRef,
        props.height
    ]);
    (0, $74aPB$reactariautils.useLayoutEffect)(()=>{
        if (inputRef.current) onHeightChange();
    }, [
        onHeightChange,
        inputValue,
        inputRef
    ]);
    if (props.placeholder) console.warn("Placeholders are deprecated due to accessibility issues. Please use help text instead. See the docs for details: https://react-spectrum.adobe.com/react-spectrum/TextArea.html#help-text");
    let result = (0, $74aPB$reactariatextfield.useTextField)({
        ...props,
        onChange: (0, $74aPB$reactariautils.chain)(onChange, setInputValue),
        inputElementType: "textarea"
    }, inputRef);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($74aPB$react))).createElement((0, $86793250183ac29e$export$d22444a338b6e3c2), {
        ...otherProps,
        ref: ref,
        inputRef: inputRef,
        ...result,
        multiLine: true,
        isDisabled: isDisabled,
        isQuiet: isQuiet,
        isReadOnly: isReadOnly,
        isRequired: isRequired
    });
}
/**
 * TextAreas are multiline text inputs, useful for cases where users have
 * a sizable amount of text to enter. They allow for all customizations that
 * are available to text fields.
 */ let $eec649e464f3832d$export$f5c9f3c2c4054eec = /*#__PURE__*/ (0, ($parcel$interopDefault($74aPB$react))).forwardRef($eec649e464f3832d$var$TextArea);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $34fd51d50eb4b056$var$TextField(props, ref) {
    props = (0, $74aPB$reactspectrumprovider.useProviderProps)(props);
    props = (0, $74aPB$reactspectrumform.useFormProps)(props);
    let inputRef = (0, $74aPB$react.useRef)(null);
    let result = (0, $74aPB$reactariatextfield.useTextField)(props, inputRef);
    if (props.placeholder) console.warn("Placeholders are deprecated due to accessibility issues. Please use help text instead. See the docs for details: https://react-spectrum.adobe.com/react-spectrum/TextField.html#help-text");
    return /*#__PURE__*/ (0, ($parcel$interopDefault($74aPB$react))).createElement((0, $86793250183ac29e$export$d22444a338b6e3c2), {
        ...props,
        ...result,
        ref: ref,
        inputRef: inputRef
    });
}
/**
 * TextFields are text inputs that allow users to input custom text entries
 * with a keyboard. Various decorations can be displayed around the field to
 * communicate the entry requirements.
 */ const $34fd51d50eb4b056$export$2c73285ae9390cec = /*#__PURE__*/ (0, $74aPB$react.forwardRef)($34fd51d50eb4b056$var$TextField);
