// @react-spectrum/table/dist/import.mjs
"use client";

import "./main.css";
import $daf541512f75fd9b$exports from "./ar-AE.mjs";
import $50819411ab066aa0$exports from "./bg-BG.mjs";
import $4f747f794f1bcba7$exports from "./cs-CZ.mjs";
import $9b4578f7fea36fce$exports from "./da-DK.mjs";
import $d1341bbc3e1d2378$exports from "./de-DE.mjs";
import $62dc02a47a5c68a9$exports from "./el-GR.mjs";
import $4040b78666bdeb49$exports from "./en-US.mjs";
import $b77a23040d0637b4$exports from "./es-ES.mjs";
import $54ccb4bcf3ebc0a8$exports from "./et-EE.mjs";
import $37e671b8f89037ca$exports from "./fi-FI.mjs";
import $8d65c5abdb6b7cbc$exports from "./fr-FR.mjs";
import $a87fac8df5d40e78$exports from "./he-IL.mjs";
import $a7ee65f0f8055b83$exports from "./hr-HR.mjs";
import $e6091fd2263a451f$exports from "./hu-HU.mjs";
import $c8d9d250bf8952b2$exports from "./it-IT.mjs";
import $df07a03bb9f879ed$exports from "./ja-JP.mjs";
import $40b3ef3504f4b1d0$exports from "./ko-KR.mjs";
import $c6959b6b120377f9$exports from "./lt-LT.mjs";
import $8fcc23e9320cffd0$exports from "./lv-LV.mjs";
import $ec9bb7644b9b61e5$exports from "./nb-NO.mjs";
import $049ab8e2dc69b622$exports from "./nl-NL.mjs";
import $480b88f7afcf23a9$exports from "./pl-PL.mjs";
import $0c053d6ef0255c08$exports from "./pt-BR.mjs";
import $3aff8d1b1fa24532$exports from "./pt-PT.mjs";
import $90e0c3e69dc73a74$exports from "./ro-RO.mjs";
import $248be0f7eca0ea1f$exports from "./ru-RU.mjs";
import $3142c640504df85a$exports from "./sk-SK.mjs";
import $7b3b7f28a30b2687$exports from "./sl-SI.mjs";
import $06c28f958a58f366$exports from "./sr-SP.mjs";
import $b45c13a581472a6c$exports from "./sv-SE.mjs";
import $785ea1a5e7a35eb5$exports from "./tr-TR.mjs";
import $2e46541801441421$exports from "./uk-UA.mjs";
import $8de129c51e22709f$exports from "./zh-CN.mjs";
import $e57af0887d00746b$exports from "./zh-TW.mjs";
import {Column as $gyhAV$Column, TableHeader as $e668f3d0f378a712$re_export$TableHeader, TableBody as $e668f3d0f378a712$re_export$TableBody, Section as $e668f3d0f378a712$re_export$Section, Row as $e668f3d0f378a712$re_export$Row, Cell as $e668f3d0f378a712$re_export$Cell, useTableState as $gyhAV$useTableState, TableColumnLayout as $gyhAV$TableColumnLayout, UNSTABLE_useTreeGridState as $gyhAV$UNSTABLE_useTreeGridState} from "@react-stately/table";
import $gyhAV$react, {useState as $gyhAV$useState, useContext as $gyhAV$useContext, useRef as $gyhAV$useRef, useEffect as $gyhAV$useEffect, useCallback as $gyhAV$useCallback, useMemo as $gyhAV$useMemo} from "react";
import {tableNestedRows as $gyhAV$tableNestedRows} from "@react-stately/flags";
import $gyhAV$spectrumiconsuiArrowDownSmall from "@spectrum-icons/ui/ArrowDownSmall";
import {mergeProps as $gyhAV$mergeProps, scrollIntoView as $gyhAV$scrollIntoView, scrollIntoViewport as $gyhAV$scrollIntoViewport, chain as $gyhAV$chain, isAndroid as $gyhAV$isAndroid, isWebKit as $gyhAV$isWebKit} from "@react-aria/utils";
import {Checkbox as $gyhAV$Checkbox} from "@react-spectrum/checkbox";
import $gyhAV$spectrumiconsuiChevronDownMedium from "@spectrum-icons/ui/ChevronDownMedium";
import $gyhAV$spectrumiconsuiChevronLeftMedium from "@spectrum-icons/ui/ChevronLeftMedium";
import $gyhAV$spectrumiconsuiChevronRightMedium from "@spectrum-icons/ui/ChevronRightMedium";
import {useStyleProps as $gyhAV$useStyleProps, useDOMRef as $gyhAV$useDOMRef, classNames as $gyhAV$classNames, useFocusableRef as $gyhAV$useFocusableRef, useUnwrapDOMRef as $gyhAV$useUnwrapDOMRef} from "@react-spectrum/utils";
import {useFocusRing as $gyhAV$useFocusRing, FocusScope as $gyhAV$FocusScope, FocusRing as $gyhAV$FocusRing} from "@react-aria/focus";
import {getInteractionModality as $gyhAV$getInteractionModality, usePress as $gyhAV$usePress, useHover as $gyhAV$useHover, isFocusVisible as $gyhAV$isFocusVisible} from "@react-aria/interactions";
import {MenuTrigger as $gyhAV$MenuTrigger, Menu as $gyhAV$Menu, Item as $gyhAV$Item} from "@react-spectrum/menu";
import {layoutInfoToStyle as $gyhAV$layoutInfoToStyle, VirtualizerItem as $gyhAV$VirtualizerItem, setScrollLeft as $gyhAV$setScrollLeft, useVirtualizer as $gyhAV$useVirtualizer, ScrollView as $gyhAV$ScrollView} from "@react-aria/virtualizer";
import $gyhAV$spectrumiconsuiListGripper from "@spectrum-icons/ui/ListGripper";
import {ProgressCircle as $gyhAV$ProgressCircle} from "@react-spectrum/progress";
import {useVirtualizerState as $gyhAV$useVirtualizerState} from "@react-stately/virtualizer";
import {TableLayout as $gyhAV$TableLayout} from "@react-stately/layout";
import {TooltipTrigger as $gyhAV$TooltipTrigger, Tooltip as $gyhAV$Tooltip} from "@react-spectrum/tooltip";
import {useButton as $gyhAV$useButton} from "@react-aria/button";
import {useLocale as $gyhAV$useLocale, useLocalizedStringFormatter as $gyhAV$useLocalizedStringFormatter} from "@react-aria/i18n";
import {useProviderProps as $gyhAV$useProviderProps, useProvider as $gyhAV$useProvider} from "@react-spectrum/provider";
import {useTable as $gyhAV$useTable, useTableRowGroup as $gyhAV$useTableRowGroup, useTableColumnHeader as $gyhAV$useTableColumnHeader, useTableSelectAllCheckbox as $gyhAV$useTableSelectAllCheckbox, useTableRow as $gyhAV$useTableRow, useTableHeaderRow as $gyhAV$useTableHeaderRow, useTableCell as $gyhAV$useTableCell, useTableSelectionCheckbox as $gyhAV$useTableSelectionCheckbox, useTableColumnResize as $gyhAV$useTableColumnResize} from "@react-aria/table";
import {VisuallyHidden as $gyhAV$VisuallyHidden, useVisuallyHidden as $gyhAV$useVisuallyHidden} from "@react-aria/visually-hidden";
import $gyhAV$reactdom from "react-dom";
import {Flex as $gyhAV$Flex} from "@react-spectrum/layout";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 








/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

var $325bc3d43b25708b$exports = {};

$parcel$export($325bc3d43b25708b$exports, "focus-ring", () => $325bc3d43b25708b$export$f39a09f249340e2a, (v) => $325bc3d43b25708b$export$f39a09f249340e2a = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table", () => $325bc3d43b25708b$export$8cc7e6e5f955c192, (v) => $325bc3d43b25708b$export$8cc7e6e5f955c192 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-InsertionIndicator", () => $325bc3d43b25708b$export$e0940649899bb056, (v) => $325bc3d43b25708b$export$e0940649899bb056 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-InsertionIndicator--dropTarget", () => $325bc3d43b25708b$export$e246eff99e77d045, (v) => $325bc3d43b25708b$export$e246eff99e77d045 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-body", () => $325bc3d43b25708b$export$7c43a9cc5d8e99e7, (v) => $325bc3d43b25708b$export$7c43a9cc5d8e99e7 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-body--dropTarget", () => $325bc3d43b25708b$export$2698e8974a57cb61, (v) => $325bc3d43b25708b$export$2698e8974a57cb61 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-cell", () => $325bc3d43b25708b$export$19015e98090cea63, (v) => $325bc3d43b25708b$export$19015e98090cea63 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-cell--alignCenter", () => $325bc3d43b25708b$export$14101ea2699002fc, (v) => $325bc3d43b25708b$export$14101ea2699002fc = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-cell--alignEnd", () => $325bc3d43b25708b$export$c594335f8535fc47, (v) => $325bc3d43b25708b$export$c594335f8535fc47 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-cell--alignStart", () => $325bc3d43b25708b$export$25d7f56ba6d0d442, (v) => $325bc3d43b25708b$export$25d7f56ba6d0d442 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-cellWrapper", () => $325bc3d43b25708b$export$ea1d9f40910c509, (v) => $325bc3d43b25708b$export$ea1d9f40910c509 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-cellWrapper--dropTarget", () => $325bc3d43b25708b$export$374244a7e6e6e484, (v) => $325bc3d43b25708b$export$374244a7e6e6e484 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-centeredWrapper", () => $325bc3d43b25708b$export$8bffb31c98aaf8d5, (v) => $325bc3d43b25708b$export$8bffb31c98aaf8d5 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-dragButton", () => $325bc3d43b25708b$export$d4b8138d7ce0ca37, (v) => $325bc3d43b25708b$export$d4b8138d7ce0ca37 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-dragButtonCell", () => $325bc3d43b25708b$export$8d6d420175da304f, (v) => $325bc3d43b25708b$export$8d6d420175da304f = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-dragButtonHeadCell", () => $325bc3d43b25708b$export$211bbf0547a730b6, (v) => $325bc3d43b25708b$export$211bbf0547a730b6 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-headCell", () => $325bc3d43b25708b$export$6fe7d8d8ce0454eb, (v) => $325bc3d43b25708b$export$6fe7d8d8ce0454eb = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-row", () => $325bc3d43b25708b$export$a2af12d9c37814bd, (v) => $325bc3d43b25708b$export$a2af12d9c37814bd = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-row--dropTarget", () => $325bc3d43b25708b$export$5e4ce8d9f081b812, (v) => $325bc3d43b25708b$export$5e4ce8d9f081b812 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-row-badge", () => $325bc3d43b25708b$export$3559001d7e84a6da, (v) => $325bc3d43b25708b$export$3559001d7e84a6da = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-row-dragPreview", () => $325bc3d43b25708b$export$ba71e6bcd70242c7, (v) => $325bc3d43b25708b$export$ba71e6bcd70242c7 = v);
$parcel$export($325bc3d43b25708b$exports, "react-spectrum-Table-row-dragPreview--multiple", () => $325bc3d43b25708b$export$3d73889014dd20dd, (v) => $325bc3d43b25708b$export$3d73889014dd20dd = v);
var $325bc3d43b25708b$export$f39a09f249340e2a;
var $325bc3d43b25708b$export$8cc7e6e5f955c192;
var $325bc3d43b25708b$export$e0940649899bb056;
var $325bc3d43b25708b$export$e246eff99e77d045;
var $325bc3d43b25708b$export$7c43a9cc5d8e99e7;
var $325bc3d43b25708b$export$2698e8974a57cb61;
var $325bc3d43b25708b$export$19015e98090cea63;
var $325bc3d43b25708b$export$14101ea2699002fc;
var $325bc3d43b25708b$export$c594335f8535fc47;
var $325bc3d43b25708b$export$25d7f56ba6d0d442;
var $325bc3d43b25708b$export$ea1d9f40910c509;
var $325bc3d43b25708b$export$374244a7e6e6e484;
var $325bc3d43b25708b$export$8bffb31c98aaf8d5;
var $325bc3d43b25708b$export$d4b8138d7ce0ca37;
var $325bc3d43b25708b$export$8d6d420175da304f;
var $325bc3d43b25708b$export$211bbf0547a730b6;
var $325bc3d43b25708b$export$6fe7d8d8ce0454eb;
var $325bc3d43b25708b$export$a2af12d9c37814bd;
var $325bc3d43b25708b$export$5e4ce8d9f081b812;
var $325bc3d43b25708b$export$3559001d7e84a6da;
var $325bc3d43b25708b$export$ba71e6bcd70242c7;
var $325bc3d43b25708b$export$3d73889014dd20dd;
$325bc3d43b25708b$export$f39a09f249340e2a = `EfWbna_focus-ring`;
$325bc3d43b25708b$export$8cc7e6e5f955c192 = `EfWbna_react-spectrum-Table`;
$325bc3d43b25708b$export$e0940649899bb056 = `EfWbna_react-spectrum-Table-InsertionIndicator`;
$325bc3d43b25708b$export$e246eff99e77d045 = `EfWbna_react-spectrum-Table-InsertionIndicator--dropTarget`;
$325bc3d43b25708b$export$7c43a9cc5d8e99e7 = `EfWbna_react-spectrum-Table-body`;
$325bc3d43b25708b$export$2698e8974a57cb61 = `EfWbna_react-spectrum-Table-body--dropTarget`;
$325bc3d43b25708b$export$19015e98090cea63 = `EfWbna_react-spectrum-Table-cell`;
$325bc3d43b25708b$export$14101ea2699002fc = `EfWbna_react-spectrum-Table-cell--alignCenter`;
$325bc3d43b25708b$export$c594335f8535fc47 = `EfWbna_react-spectrum-Table-cell--alignEnd`;
$325bc3d43b25708b$export$25d7f56ba6d0d442 = `EfWbna_react-spectrum-Table-cell--alignStart`;
$325bc3d43b25708b$export$ea1d9f40910c509 = `EfWbna_react-spectrum-Table-cellWrapper`;
$325bc3d43b25708b$export$374244a7e6e6e484 = `EfWbna_react-spectrum-Table-cellWrapper--dropTarget`;
$325bc3d43b25708b$export$8bffb31c98aaf8d5 = `EfWbna_react-spectrum-Table-centeredWrapper`;
$325bc3d43b25708b$export$d4b8138d7ce0ca37 = `EfWbna_react-spectrum-Table-dragButton`;
$325bc3d43b25708b$export$8d6d420175da304f = `EfWbna_react-spectrum-Table-dragButtonCell`;
$325bc3d43b25708b$export$211bbf0547a730b6 = `EfWbna_react-spectrum-Table-dragButtonHeadCell`;
$325bc3d43b25708b$export$6fe7d8d8ce0454eb = `EfWbna_react-spectrum-Table-headCell`;
$325bc3d43b25708b$export$a2af12d9c37814bd = `EfWbna_react-spectrum-Table-row`;
$325bc3d43b25708b$export$5e4ce8d9f081b812 = `EfWbna_react-spectrum-Table-row--dropTarget`;
$325bc3d43b25708b$export$3559001d7e84a6da = `EfWbna_react-spectrum-Table-row-badge`;
$325bc3d43b25708b$export$ba71e6bcd70242c7 = `EfWbna_react-spectrum-Table-row-dragPreview`;
$325bc3d43b25708b$export$3d73889014dd20dd = `EfWbna_react-spectrum-Table-row-dragPreview--multiple`;




function $7ee9a922ee4e8032$export$2c0bab5914a9d088(props) {
    let { dropState: dropState, dragAndDropHooks: dragAndDropHooks } = (0, $bd013581c0a4b065$export$3cb274deb6c2d854)();
    const { target: target, rowProps: rowProps } = props;
    let ref = (0, $gyhAV$useRef)();
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator(props, dropState, ref);
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $gyhAV$useVisuallyHidden)();
    let isDropTarget = dropState.isDropTarget(target);
    if (!isDropTarget && dropIndicatorProps["aria-hidden"]) return null;
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        style: {
            position: "absolute",
            top: rowProps.style.top + (target.dropPosition === "after" ? rowProps.style.height : 0),
            width: rowProps.style.width
        },
        role: "row",
        "aria-hidden": dropIndicatorProps["aria-hidden"]
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "gridcell",
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-InsertionIndicator", {
            "react-spectrum-Table-InsertionIndicator--dropTarget": isDropTarget
        })
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...visuallyHiddenProps,
        role: "button",
        ...dropIndicatorProps,
        ref: ref
    })));
}


var $cc43b2d4f6011c04$exports = {};


































$cc43b2d4f6011c04$exports = {
    "ar-AE": $daf541512f75fd9b$exports,
    "bg-BG": $50819411ab066aa0$exports,
    "cs-CZ": $4f747f794f1bcba7$exports,
    "da-DK": $9b4578f7fea36fce$exports,
    "de-DE": $d1341bbc3e1d2378$exports,
    "el-GR": $62dc02a47a5c68a9$exports,
    "en-US": $4040b78666bdeb49$exports,
    "es-ES": $b77a23040d0637b4$exports,
    "et-EE": $54ccb4bcf3ebc0a8$exports,
    "fi-FI": $37e671b8f89037ca$exports,
    "fr-FR": $8d65c5abdb6b7cbc$exports,
    "he-IL": $a87fac8df5d40e78$exports,
    "hr-HR": $a7ee65f0f8055b83$exports,
    "hu-HU": $e6091fd2263a451f$exports,
    "it-IT": $c8d9d250bf8952b2$exports,
    "ja-JP": $df07a03bb9f879ed$exports,
    "ko-KR": $40b3ef3504f4b1d0$exports,
    "lt-LT": $c6959b6b120377f9$exports,
    "lv-LV": $8fcc23e9320cffd0$exports,
    "nb-NO": $ec9bb7644b9b61e5$exports,
    "nl-NL": $049ab8e2dc69b622$exports,
    "pl-PL": $480b88f7afcf23a9$exports,
    "pt-BR": $0c053d6ef0255c08$exports,
    "pt-PT": $3aff8d1b1fa24532$exports,
    "ro-RO": $90e0c3e69dc73a74$exports,
    "ru-RU": $248be0f7eca0ea1f$exports,
    "sk-SK": $3142c640504df85a$exports,
    "sl-SI": $7b3b7f28a30b2687$exports,
    "sr-SP": $06c28f958a58f366$exports,
    "sv-SE": $b45c13a581472a6c$exports,
    "tr-TR": $785ea1a5e7a35eb5$exports,
    "uk-UA": $2e46541801441421$exports,
    "zh-CN": $8de129c51e22709f$exports,
    "zh-TW": $e57af0887d00746b$exports
};





/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
function $cb7d341e8868d81d$export$d9658cdf8c86807() {
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("svg", {
        xmlns: "http://www.w3.org/2000/svg",
        width: "16",
        height: "16",
        viewBox: "0 0 16 16"
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("g", {
        fill: "var(--spectrum-global-color-blue-600)",
        stroke: "var(--spectrum-global-color-blue-600)",
        strokeWidth: "2"
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("circle", {
        cx: "8",
        cy: "8",
        r: "8",
        stroke: "none"
    }), /*#__PURE__*/ (0, $gyhAV$react).createElement("circle", {
        cx: "8",
        cy: "8",
        r: "7",
        fill: "none"
    })), /*#__PURE__*/ (0, $gyhAV$react).createElement("path", {
        d: "M-2106-7380.263v5l2.5-2.551Z",
        transform: "translate(2116 7385.763)",
        fill: "#fff",
        stroke: "#fff",
        strokeLinejoin: "round",
        strokeWidth: "2"
    }), /*#__PURE__*/ (0, $gyhAV$react).createElement("path", {
        d: "M-2106-7380.263v5l2.5-2.551Z",
        transform: "translate(-2100 -7369.763) rotate(180)",
        fill: "#fff",
        stroke: "#fff",
        strokeLinejoin: "round",
        strokeWidth: "2"
    }));
}




/* eslint-disable jsx-a11y/role-supports-aria-props */ 
var $2fc56f9fdd93be52$exports = {};
$2fc56f9fdd93be52$exports = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"32\" height=\"32\" viewBox=\"-1 0 31 32\">\n  <filter id=\"shadow\">\n    <feDropShadow dx=\"0\" dy=\"0.5\" stdDeviation=\"1.2\" flood-opacity=\"0.7\"></feDropShadow>\n  </filter>\n  <g filter=\"url('#shadow')\" transform=\"translate(.5 0)\">\n    <path d=\"M9.5,2c.27614,0,.5,.22388,.5,.5v6.5h3v-1.74951c0-.14893,.12215-.25049,.2513-.25049,.05457,0,.1104,.01819,.15875,.05847l2.58995,2.44153-2.58995,2.44153c-.04837,.04028-.10419,.05847-.15875,.05847-.12914,0-.2513-.10156-.2513-.25049v-1.74951h-3v6.5c0,.27612-.22386,.5-.5,.5h-1c-.27614,0-.5-.22388-.5-.5V2.5c0-.27612,.22386-.5,.5-.5h1Zm0-1h-1c-.82843,0-1.5,.67157-1.5,1.5v14c0,.82843,.67157,1.5,1.5,1.5h1c.82843,0,1.5-.67157,1.5-1.5v-5.5h1v.67757c0,.40416,.1632,.80501,.48197,1.05347,.49639,.3869,1.14001,.33571,1.56838-.02144l.02344-.01953,.0222-.02087,2.97589-2.80537c.20937-.19737,.20937-.53028,0-.72765l-2.97589-2.80537-.0222-.02087-.02345-.01965c-.22456-.18713-.5083-.29028-.79895-.29028-.3208,0-.62624,.12134-.86003,.34167-.24872,.23438-.39136,.56567-.39136,.90881v.74951h-1V2.5c0-.82843-.67157-1.5-1.5-1.5Z\" fill=\"#fff\"></path>\n    <path d=\"M10,16.5v-6.5h3v1.74952c0,.21196,.24721,.32775,.41005,.19206l2.58995-2.44158-2.58995-2.44158c-.16283-.13569-.41005-.0199-.41005,.19206v1.74952h-3V2.5c0-.27614-.22386-.5-.5-.5h-1c-.27614,0-.5,.22386-.5,.5v14c0,.27614,.22386,.5,.5,.5h1c.27614,0,.5-.22386,.5-.5Z\"></path>\n  </g>\n</svg>\n";


var $ff1f3562a1be2216$exports = {};
$ff1f3562a1be2216$exports = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"32\" height=\"32\" viewBox=\"-1 0 31 32\">\n  <filter id=\"shadow\">\n    <feDropShadow dx=\"0\" dy=\"0.5\" stdDeviation=\"1.2\" flood-opacity=\"0.7\"></feDropShadow>\n  </filter>\n  <g filter=\"url('#shadow')\" transform=\"translate(.5 0)\">\n    <path fill=\"#fff\" d=\"M9.5,2a.5.5,0,0,1,.5.5V9h3V7.2505A.25014.25014,0,0,1,13.24978,7h.00152a.246.246,0,0,1,.15875.05847L16,9.5l-2.58995,2.44153A.24611.24611,0,0,1,13.2513,12,.25016.25016,0,0,1,13,11.751V10H10v6.5a.5.5,0,0,1-.5.5h-1a.5.5,0,0,1-.5-.5V10H5v1.7495A.25014.25014,0,0,1,4.75022,12H4.7487A.246.246,0,0,1,4.59,11.94153L2,9.5,4.59,7.05847A.246.246,0,0,1,4.7487,7,.25016.25016,0,0,1,5,7.249V9H8V2.5A.5.5,0,0,1,8.5,2Zm0-1h-1A1.5,1.5,0,0,0,7,2.5V8H6V7.32389a1.33,1.33,0,0,0-.4816-1.05444,1.23962,1.23962,0,0,0-1.56876.021l-.02344.01953-.0222.02088L.92812,9.13618A.5.5,0,0,0,.90726,9.843l.02086.02085L3.904,12.66919l.02221.02088.02345.01965a1.23986,1.23986,0,0,0,1.56836.02116A1.3294,1.3294,0,0,0,6,11.6765V11H7v5.5A1.5,1.5,0,0,0,8.5,18h1A1.5,1.5,0,0,0,11,16.5V11h1v.67757a1.32792,1.32792,0,0,0,.482,1.05347,1.24,1.24,0,0,0,1.56839-.02144l.02343-.01953.02221-.02088,2.97588-2.80537a.5.5,0,0,0,.02086-.70679l-.02086-.02085L14.096,6.33081l-.02221-.02087-.02345-.01965A1.24978,1.24978,0,0,0,12,7.2505V8H11V2.5A1.5,1.5,0,0,0,9.5,1Z\"></path>\n    <path d=\"M10,16.5V10h3v1.7495a.25.25,0,0,0,.41.19206L16,9.5,13.41,7.05842A.25.25,0,0,0,13,7.2505V9H10V2.5A.5.5,0,0,0,9.5,2h-1a.5.5,0,0,0-.5.5V9H5V7.2505a.25.25,0,0,0-.41-.19206L2,9.5l2.59,2.44158A.25.25,0,0,0,5,11.7495V10H8v6.5a.5.5,0,0,0,.5.5h1A.5.5,0,0,0,10,16.5Z\"></path>\n  </g>\n</svg>\n";







var $bd5d41f647a296c3$exports = {};

$parcel$export($bd5d41f647a296c3$exports, "contents", () => $bd5d41f647a296c3$export$fd59df041b888442, (v) => $bd5d41f647a296c3$export$fd59df041b888442 = v);
$parcel$export($bd5d41f647a296c3$exports, "expand-button", () => $bd5d41f647a296c3$export$8537c4b9f826a67e, (v) => $bd5d41f647a296c3$export$8537c4b9f826a67e = v);
$parcel$export($bd5d41f647a296c3$exports, "focus-ring", () => $bd5d41f647a296c3$export$f39a09f249340e2a, (v) => $bd5d41f647a296c3$export$f39a09f249340e2a = v);
$parcel$export($bd5d41f647a296c3$exports, "i18nFontFamily", () => $bd5d41f647a296c3$export$8c4ee2c50c22c514, (v) => $bd5d41f647a296c3$export$8c4ee2c50c22c514 = v);
$parcel$export($bd5d41f647a296c3$exports, "is-active", () => $bd5d41f647a296c3$export$20fd0f7cd4e6112f, (v) => $bd5d41f647a296c3$export$20fd0f7cd4e6112f = v);
$parcel$export($bd5d41f647a296c3$exports, "is-disabled", () => $bd5d41f647a296c3$export$d35bc1e505d1ebbf, (v) => $bd5d41f647a296c3$export$d35bc1e505d1ebbf = v);
$parcel$export($bd5d41f647a296c3$exports, "is-drop-target", () => $bd5d41f647a296c3$export$d8467f38d2d00bd0, (v) => $bd5d41f647a296c3$export$d8467f38d2d00bd0 = v);
$parcel$export($bd5d41f647a296c3$exports, "is-focused", () => $bd5d41f647a296c3$export$e7dc768d35940237, (v) => $bd5d41f647a296c3$export$e7dc768d35940237 = v);
$parcel$export($bd5d41f647a296c3$exports, "is-hovered", () => $bd5d41f647a296c3$export$b8813cd5d7824ce7, (v) => $bd5d41f647a296c3$export$b8813cd5d7824ce7 = v);
$parcel$export($bd5d41f647a296c3$exports, "is-next-selected", () => $bd5d41f647a296c3$export$53aa811f86efbfad, (v) => $bd5d41f647a296c3$export$53aa811f86efbfad = v);
$parcel$export($bd5d41f647a296c3$exports, "is-open", () => $bd5d41f647a296c3$export$a9781837241c946d, (v) => $bd5d41f647a296c3$export$a9781837241c946d = v);
$parcel$export($bd5d41f647a296c3$exports, "is-resizable", () => $bd5d41f647a296c3$export$d20629d2e587d4f, (v) => $bd5d41f647a296c3$export$d20629d2e587d4f = v);
$parcel$export($bd5d41f647a296c3$exports, "is-selected", () => $bd5d41f647a296c3$export$1e0fb04f31d3c22a, (v) => $bd5d41f647a296c3$export$1e0fb04f31d3c22a = v);
$parcel$export($bd5d41f647a296c3$exports, "is-sortable", () => $bd5d41f647a296c3$export$c5f84948d72c03fb, (v) => $bd5d41f647a296c3$export$c5f84948d72c03fb = v);
$parcel$export($bd5d41f647a296c3$exports, "is-sorted-asc", () => $bd5d41f647a296c3$export$24499792d6b87cb, (v) => $bd5d41f647a296c3$export$24499792d6b87cb = v);
$parcel$export($bd5d41f647a296c3$exports, "is-sorted-desc", () => $bd5d41f647a296c3$export$cda6d7b632f895cc, (v) => $bd5d41f647a296c3$export$cda6d7b632f895cc = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-FocusRing-ring", () => $bd5d41f647a296c3$export$4109102f950813a6, (v) => $bd5d41f647a296c3$export$4109102f950813a6 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-FocusRing", () => $bd5d41f647a296c3$export$24c7f46a6e3605dd, (v) => $bd5d41f647a296c3$export$24c7f46a6e3605dd = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-FocusRing--quiet", () => $bd5d41f647a296c3$export$2927016961429360, (v) => $bd5d41f647a296c3$export$2927016961429360 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table", () => $bd5d41f647a296c3$export$455c59d2b5bafef0, (v) => $bd5d41f647a296c3$export$455c59d2b5bafef0 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table--compact", () => $bd5d41f647a296c3$export$bcb43fe016211cab, (v) => $bd5d41f647a296c3$export$bcb43fe016211cab = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table--isHorizontalScrollbarVisible", () => $bd5d41f647a296c3$export$eb2b06f9a223c9ae, (v) => $bd5d41f647a296c3$export$eb2b06f9a223c9ae = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table--isVerticalScrollbarVisible", () => $bd5d41f647a296c3$export$f3cbf113923c0570, (v) => $bd5d41f647a296c3$export$f3cbf113923c0570 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table--loadingMore", () => $bd5d41f647a296c3$export$51b307035a9062f1, (v) => $bd5d41f647a296c3$export$51b307035a9062f1 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table--quiet", () => $bd5d41f647a296c3$export$80f179961079d85e, (v) => $bd5d41f647a296c3$export$80f179961079d85e = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table--regular", () => $bd5d41f647a296c3$export$3be3e4ba813e9589, (v) => $bd5d41f647a296c3$export$3be3e4ba813e9589 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table--spacious", () => $bd5d41f647a296c3$export$eef21b0715bc0914, (v) => $bd5d41f647a296c3$export$eef21b0715bc0914 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table--wrap", () => $bd5d41f647a296c3$export$9d4c255d6285f0ec, (v) => $bd5d41f647a296c3$export$9d4c255d6285f0ec = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-body", () => $bd5d41f647a296c3$export$c841a58af70827c7, (v) => $bd5d41f647a296c3$export$c841a58af70827c7 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-body--resizerAtTableEdge", () => $bd5d41f647a296c3$export$ec8d96850ac2415e, (v) => $bd5d41f647a296c3$export$ec8d96850ac2415e = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-bodyResizeIndicator", () => $bd5d41f647a296c3$export$9a870fddd16be630, (v) => $bd5d41f647a296c3$export$9a870fddd16be630 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-cell", () => $bd5d41f647a296c3$export$c073926c283bb618, (v) => $bd5d41f647a296c3$export$c073926c283bb618 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-cell--alignCenter", () => $bd5d41f647a296c3$export$7e9bb4c3d8ec70e2, (v) => $bd5d41f647a296c3$export$7e9bb4c3d8ec70e2 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-cell--alignEnd", () => $bd5d41f647a296c3$export$94db5e2e979379b6, (v) => $bd5d41f647a296c3$export$94db5e2e979379b6 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-cell--divider", () => $bd5d41f647a296c3$export$9531650467e5785b, (v) => $bd5d41f647a296c3$export$9531650467e5785b = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-cell--hasExpandCollapseButton", () => $bd5d41f647a296c3$export$466223eab8fe7a73, (v) => $bd5d41f647a296c3$export$466223eab8fe7a73 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-cell--hideHeader", () => $bd5d41f647a296c3$export$8ff2ab45f5f0045d, (v) => $bd5d41f647a296c3$export$8ff2ab45f5f0045d = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-cellContents", () => $bd5d41f647a296c3$export$683061182a3353d3, (v) => $bd5d41f647a296c3$export$683061182a3353d3 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-cellWrapper", () => $bd5d41f647a296c3$export$fe4389976827e927, (v) => $bd5d41f647a296c3$export$fe4389976827e927 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-checkbox", () => $bd5d41f647a296c3$export$9c494845d6df55df, (v) => $bd5d41f647a296c3$export$9c494845d6df55df = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-checkboxCell", () => $bd5d41f647a296c3$export$ae17a7501a9fd0b3, (v) => $bd5d41f647a296c3$export$ae17a7501a9fd0b3 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-colResizeIndicator", () => $bd5d41f647a296c3$export$3a270f8fab5daa8b, (v) => $bd5d41f647a296c3$export$3a270f8fab5daa8b = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-colResizeIndicator--resizing", () => $bd5d41f647a296c3$export$d3861fcb1e5f6a2b, (v) => $bd5d41f647a296c3$export$d3861fcb1e5f6a2b = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-colResizeIndicator--visible", () => $bd5d41f647a296c3$export$b0bdf5d269bf857, (v) => $bd5d41f647a296c3$export$b0bdf5d269bf857 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-colResizeNubbin", () => $bd5d41f647a296c3$export$12c7b2fb4df461f8, (v) => $bd5d41f647a296c3$export$12c7b2fb4df461f8 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-colResizeNubbin--visible", () => $bd5d41f647a296c3$export$56a3c5ebd77d19bf, (v) => $bd5d41f647a296c3$export$56a3c5ebd77d19bf = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-columnResizer", () => $bd5d41f647a296c3$export$a1a03b9bd1868758, (v) => $bd5d41f647a296c3$export$a1a03b9bd1868758 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-columnResizerPlaceholder", () => $bd5d41f647a296c3$export$dbbe6828fb150c0d, (v) => $bd5d41f647a296c3$export$dbbe6828fb150c0d = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-expandButton", () => $bd5d41f647a296c3$export$ce10510e27df0512, (v) => $bd5d41f647a296c3$export$ce10510e27df0512 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-headCell", () => $bd5d41f647a296c3$export$f64307d6c29e2f55, (v) => $bd5d41f647a296c3$export$f64307d6c29e2f55 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-headCellButton", () => $bd5d41f647a296c3$export$231a910aceaf76df, (v) => $bd5d41f647a296c3$export$231a910aceaf76df = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-headCellButton--alignCenter", () => $bd5d41f647a296c3$export$75a2aad672a61387, (v) => $bd5d41f647a296c3$export$75a2aad672a61387 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-headCellButton--alignEnd", () => $bd5d41f647a296c3$export$7d0b6721a7bc5a4e, (v) => $bd5d41f647a296c3$export$7d0b6721a7bc5a4e = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-headCellButton--alignStart", () => $bd5d41f647a296c3$export$b1f0f17b71ca32b7, (v) => $bd5d41f647a296c3$export$b1f0f17b71ca32b7 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-headCellContents", () => $bd5d41f647a296c3$export$ce66f3b8316b228e, (v) => $bd5d41f647a296c3$export$ce66f3b8316b228e = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-headWrapper", () => $bd5d41f647a296c3$export$a08929593e9a2fb2, (v) => $bd5d41f647a296c3$export$a08929593e9a2fb2 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-headerCellText", () => $bd5d41f647a296c3$export$3d7b5f594896430, (v) => $bd5d41f647a296c3$export$3d7b5f594896430 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-menuChevron", () => $bd5d41f647a296c3$export$da1b2523b4f18b96, (v) => $bd5d41f647a296c3$export$da1b2523b4f18b96 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-row", () => $bd5d41f647a296c3$export$fefa57dcd4648052, (v) => $bd5d41f647a296c3$export$fefa57dcd4648052 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-row--firstRow", () => $bd5d41f647a296c3$export$71150fb6934bcef6, (v) => $bd5d41f647a296c3$export$71150fb6934bcef6 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-row--isFlushBottom", () => $bd5d41f647a296c3$export$c544a9cedcdf7e74, (v) => $bd5d41f647a296c3$export$c544a9cedcdf7e74 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-row--lastRow", () => $bd5d41f647a296c3$export$6bc25dc2c214dc40, (v) => $bd5d41f647a296c3$export$6bc25dc2c214dc40 = v);
$parcel$export($bd5d41f647a296c3$exports, "spectrum-Table-sortedIcon", () => $bd5d41f647a296c3$export$7d48ded8ceef1e62, (v) => $bd5d41f647a296c3$export$7d48ded8ceef1e62 = v);
var $bd5d41f647a296c3$export$fd59df041b888442;
var $bd5d41f647a296c3$export$8537c4b9f826a67e;
var $bd5d41f647a296c3$export$f39a09f249340e2a;
var $bd5d41f647a296c3$export$8c4ee2c50c22c514;
var $bd5d41f647a296c3$export$20fd0f7cd4e6112f;
var $bd5d41f647a296c3$export$d35bc1e505d1ebbf;
var $bd5d41f647a296c3$export$d8467f38d2d00bd0;
var $bd5d41f647a296c3$export$e7dc768d35940237;
var $bd5d41f647a296c3$export$b8813cd5d7824ce7;
var $bd5d41f647a296c3$export$53aa811f86efbfad;
var $bd5d41f647a296c3$export$a9781837241c946d;
var $bd5d41f647a296c3$export$d20629d2e587d4f;
var $bd5d41f647a296c3$export$1e0fb04f31d3c22a;
var $bd5d41f647a296c3$export$c5f84948d72c03fb;
var $bd5d41f647a296c3$export$24499792d6b87cb;
var $bd5d41f647a296c3$export$cda6d7b632f895cc;
var $bd5d41f647a296c3$export$4109102f950813a6;
var $bd5d41f647a296c3$export$24c7f46a6e3605dd;
var $bd5d41f647a296c3$export$2927016961429360;
var $bd5d41f647a296c3$export$455c59d2b5bafef0;
var $bd5d41f647a296c3$export$bcb43fe016211cab;
var $bd5d41f647a296c3$export$eb2b06f9a223c9ae;
var $bd5d41f647a296c3$export$f3cbf113923c0570;
var $bd5d41f647a296c3$export$51b307035a9062f1;
var $bd5d41f647a296c3$export$80f179961079d85e;
var $bd5d41f647a296c3$export$3be3e4ba813e9589;
var $bd5d41f647a296c3$export$eef21b0715bc0914;
var $bd5d41f647a296c3$export$9d4c255d6285f0ec;
var $bd5d41f647a296c3$export$c841a58af70827c7;
var $bd5d41f647a296c3$export$ec8d96850ac2415e;
var $bd5d41f647a296c3$export$9a870fddd16be630;
var $bd5d41f647a296c3$export$c073926c283bb618;
var $bd5d41f647a296c3$export$7e9bb4c3d8ec70e2;
var $bd5d41f647a296c3$export$94db5e2e979379b6;
var $bd5d41f647a296c3$export$9531650467e5785b;
var $bd5d41f647a296c3$export$466223eab8fe7a73;
var $bd5d41f647a296c3$export$8ff2ab45f5f0045d;
var $bd5d41f647a296c3$export$683061182a3353d3;
var $bd5d41f647a296c3$export$fe4389976827e927;
var $bd5d41f647a296c3$export$9c494845d6df55df;
var $bd5d41f647a296c3$export$ae17a7501a9fd0b3;
var $bd5d41f647a296c3$export$3a270f8fab5daa8b;
var $bd5d41f647a296c3$export$d3861fcb1e5f6a2b;
var $bd5d41f647a296c3$export$b0bdf5d269bf857;
var $bd5d41f647a296c3$export$12c7b2fb4df461f8;
var $bd5d41f647a296c3$export$56a3c5ebd77d19bf;
var $bd5d41f647a296c3$export$a1a03b9bd1868758;
var $bd5d41f647a296c3$export$dbbe6828fb150c0d;
var $bd5d41f647a296c3$export$ce10510e27df0512;
var $bd5d41f647a296c3$export$f64307d6c29e2f55;
var $bd5d41f647a296c3$export$231a910aceaf76df;
var $bd5d41f647a296c3$export$75a2aad672a61387;
var $bd5d41f647a296c3$export$7d0b6721a7bc5a4e;
var $bd5d41f647a296c3$export$b1f0f17b71ca32b7;
var $bd5d41f647a296c3$export$ce66f3b8316b228e;
var $bd5d41f647a296c3$export$a08929593e9a2fb2;
var $bd5d41f647a296c3$export$3d7b5f594896430;
var $bd5d41f647a296c3$export$da1b2523b4f18b96;
var $bd5d41f647a296c3$export$fefa57dcd4648052;
var $bd5d41f647a296c3$export$71150fb6934bcef6;
var $bd5d41f647a296c3$export$c544a9cedcdf7e74;
var $bd5d41f647a296c3$export$6bc25dc2c214dc40;
var $bd5d41f647a296c3$export$7d48ded8ceef1e62;
$bd5d41f647a296c3$export$fd59df041b888442 = `_5uzlQq_contents`;
$bd5d41f647a296c3$export$8537c4b9f826a67e = `_5uzlQq_expand-button`;
$bd5d41f647a296c3$export$f39a09f249340e2a = `_5uzlQq_focus-ring`;
$bd5d41f647a296c3$export$8c4ee2c50c22c514 = `_5uzlQq_i18nFontFamily`;
$bd5d41f647a296c3$export$20fd0f7cd4e6112f = `_5uzlQq_is-active`;
$bd5d41f647a296c3$export$d35bc1e505d1ebbf = `_5uzlQq_is-disabled`;
$bd5d41f647a296c3$export$d8467f38d2d00bd0 = `_5uzlQq_is-drop-target`;
$bd5d41f647a296c3$export$e7dc768d35940237 = `_5uzlQq_is-focused`;
$bd5d41f647a296c3$export$b8813cd5d7824ce7 = `_5uzlQq_is-hovered`;
$bd5d41f647a296c3$export$53aa811f86efbfad = `_5uzlQq_is-next-selected`;
$bd5d41f647a296c3$export$a9781837241c946d = `_5uzlQq_is-open`;
$bd5d41f647a296c3$export$d20629d2e587d4f = `_5uzlQq_is-resizable`;
$bd5d41f647a296c3$export$1e0fb04f31d3c22a = `_5uzlQq_is-selected`;
$bd5d41f647a296c3$export$c5f84948d72c03fb = `_5uzlQq_is-sortable`;
$bd5d41f647a296c3$export$24499792d6b87cb = `_5uzlQq_is-sorted-asc`;
$bd5d41f647a296c3$export$cda6d7b632f895cc = `_5uzlQq_is-sorted-desc`;
$bd5d41f647a296c3$export$4109102f950813a6 = `_5uzlQq_spectrum-FocusRing-ring`;
$bd5d41f647a296c3$export$24c7f46a6e3605dd = `_5uzlQq_spectrum-FocusRing ${$bd5d41f647a296c3$export$4109102f950813a6}`;
$bd5d41f647a296c3$export$2927016961429360 = `_5uzlQq_spectrum-FocusRing--quiet`;
$bd5d41f647a296c3$export$455c59d2b5bafef0 = `_5uzlQq_spectrum-Table`;
$bd5d41f647a296c3$export$bcb43fe016211cab = `_5uzlQq_spectrum-Table--compact`;
$bd5d41f647a296c3$export$eb2b06f9a223c9ae = `_5uzlQq_spectrum-Table--isHorizontalScrollbarVisible`;
$bd5d41f647a296c3$export$f3cbf113923c0570 = `_5uzlQq_spectrum-Table--isVerticalScrollbarVisible`;
$bd5d41f647a296c3$export$51b307035a9062f1 = `_5uzlQq_spectrum-Table--loadingMore`;
$bd5d41f647a296c3$export$80f179961079d85e = `_5uzlQq_spectrum-Table--quiet`;
$bd5d41f647a296c3$export$3be3e4ba813e9589 = `_5uzlQq_spectrum-Table--regular`;
$bd5d41f647a296c3$export$eef21b0715bc0914 = `_5uzlQq_spectrum-Table--spacious`;
$bd5d41f647a296c3$export$9d4c255d6285f0ec = `_5uzlQq_spectrum-Table--wrap`;
$bd5d41f647a296c3$export$c841a58af70827c7 = `_5uzlQq_spectrum-Table-body`;
$bd5d41f647a296c3$export$ec8d96850ac2415e = `_5uzlQq_spectrum-Table-body--resizerAtTableEdge`;
$bd5d41f647a296c3$export$9a870fddd16be630 = `_5uzlQq_spectrum-Table-bodyResizeIndicator`;
$bd5d41f647a296c3$export$c073926c283bb618 = `_5uzlQq_spectrum-Table-cell`;
$bd5d41f647a296c3$export$7e9bb4c3d8ec70e2 = `_5uzlQq_spectrum-Table-cell--alignCenter`;
$bd5d41f647a296c3$export$94db5e2e979379b6 = `_5uzlQq_spectrum-Table-cell--alignEnd`;
$bd5d41f647a296c3$export$9531650467e5785b = `_5uzlQq_spectrum-Table-cell--divider`;
$bd5d41f647a296c3$export$466223eab8fe7a73 = `_5uzlQq_spectrum-Table-cell--hasExpandCollapseButton`;
$bd5d41f647a296c3$export$8ff2ab45f5f0045d = `_5uzlQq_spectrum-Table-cell--hideHeader`;
$bd5d41f647a296c3$export$683061182a3353d3 = `_5uzlQq_spectrum-Table-cellContents`;
$bd5d41f647a296c3$export$fe4389976827e927 = `_5uzlQq_spectrum-Table-cellWrapper`;
$bd5d41f647a296c3$export$9c494845d6df55df = `_5uzlQq_spectrum-Table-checkbox`;
$bd5d41f647a296c3$export$ae17a7501a9fd0b3 = `_5uzlQq_spectrum-Table-checkboxCell`;
$bd5d41f647a296c3$export$3a270f8fab5daa8b = `_5uzlQq_spectrum-Table-colResizeIndicator`;
$bd5d41f647a296c3$export$d3861fcb1e5f6a2b = `_5uzlQq_spectrum-Table-colResizeIndicator--resizing`;
$bd5d41f647a296c3$export$b0bdf5d269bf857 = `_5uzlQq_spectrum-Table-colResizeIndicator--visible`;
$bd5d41f647a296c3$export$12c7b2fb4df461f8 = `_5uzlQq_spectrum-Table-colResizeNubbin`;
$bd5d41f647a296c3$export$56a3c5ebd77d19bf = `_5uzlQq_spectrum-Table-colResizeNubbin--visible`;
$bd5d41f647a296c3$export$a1a03b9bd1868758 = `_5uzlQq_spectrum-Table-columnResizer`;
$bd5d41f647a296c3$export$dbbe6828fb150c0d = `_5uzlQq_spectrum-Table-columnResizerPlaceholder`;
$bd5d41f647a296c3$export$ce10510e27df0512 = `_5uzlQq_spectrum-Table-expandButton`;
$bd5d41f647a296c3$export$f64307d6c29e2f55 = `_5uzlQq_spectrum-Table-headCell`;
$bd5d41f647a296c3$export$231a910aceaf76df = `_5uzlQq_spectrum-Table-headCellButton`;
$bd5d41f647a296c3$export$75a2aad672a61387 = `_5uzlQq_spectrum-Table-headCellButton--alignCenter`;
$bd5d41f647a296c3$export$7d0b6721a7bc5a4e = `_5uzlQq_spectrum-Table-headCellButton--alignEnd`;
$bd5d41f647a296c3$export$b1f0f17b71ca32b7 = `_5uzlQq_spectrum-Table-headCellButton--alignStart`;
$bd5d41f647a296c3$export$ce66f3b8316b228e = `_5uzlQq_spectrum-Table-headCellContents`;
$bd5d41f647a296c3$export$a08929593e9a2fb2 = `_5uzlQq_spectrum-Table-headWrapper`;
$bd5d41f647a296c3$export$3d7b5f594896430 = `_5uzlQq_spectrum-Table-headerCellText`;
$bd5d41f647a296c3$export$da1b2523b4f18b96 = `_5uzlQq_spectrum-Table-menuChevron`;
$bd5d41f647a296c3$export$fefa57dcd4648052 = `_5uzlQq_spectrum-Table-row`;
$bd5d41f647a296c3$export$71150fb6934bcef6 = `_5uzlQq_spectrum-Table-row--firstRow`;
$bd5d41f647a296c3$export$c544a9cedcdf7e74 = `_5uzlQq_spectrum-Table-row--isFlushBottom`;
$bd5d41f647a296c3$export$6bc25dc2c214dc40 = `_5uzlQq_spectrum-Table-row--lastRow`;
$bd5d41f647a296c3$export$7d48ded8ceef1e62 = `_5uzlQq_spectrum-Table-sortedIcon`;





var $58407c2d4eb33511$exports = {};
$58407c2d4eb33511$exports = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"32\" height=\"32\" viewBox=\"-1 0 31 32\">\n  <filter id=\"shadow\">\n    <feDropShadow dx=\"0\" dy=\"0.5\" stdDeviation=\"1.2\" flood-opacity=\"0.7\"></feDropShadow>\n  </filter>\n  <g filter=\"url('#shadow')\" transform=\"translate(.5 0)\">\n    <path d=\"M9.5,2c.27614,0,.5,.22388,.5,.5v14c0,.27612-.22386,.5-.5,.5h-1c-.27614,0-.5-.22388-.5-.5v-6.5h-3v1.74951c0,.14893-.12215,.25049-.2513,.25049-.05457,0-.1104-.01819-.15875-.05847l-2.58995-2.44153,2.58995-2.44153c.04837-.04028,.10419-.05847,.15875-.05847,.12914,0,.2513,.10156,.2513,.25049v1.74951h3V2.5c0-.27612,.22386-.5,.5-.5h1Zm0-1h-1c-.82843,0-1.5,.67157-1.5,1.5v5.5h-1v-.67611c0-.40436-.1629-.80555-.48159-1.05444-.4961-.38744-1.14015-.33635-1.56876,.02095l-.02344,.01953-.0222,.02087L.92811,9.13618c-.20937,.19737-.20937,.53028,0,.72765l2.97589,2.80537,.0222,.02087,.02345,.01965c.42839,.35694,1.07208,.40833,1.56837,.02116,.31886-.24875,.48198-.64994,.48198-1.05435v-.67652s.99999,0,.99999,0v5.5c0,.82843,.67157,1.5,1.5,1.5h1c.82843,0,1.5-.67157,1.5-1.5V2.5c0-.82843-.67157-1.5-1.5-1.5Z\" fill=\"#fff\"></path>\n    <path d=\"M10,16.5V2.5c0-.27614-.22386-.5-.5-.5h-1c-.27614,0-.5,.22386-.5,.5v6.5h-3v-1.74952c0-.21196-.24721-.32775-.41005-.19206l-2.58995,2.44158,2.58995,2.44158c.16283,.13569,.41005,.0199,.41005-.19206v-1.74952h3v6.5c0,.27614,.22386,.5,.5,.5h1c.27614,0,.5-.22386,.5-.5Z\"></path>\n  </g>\n</svg>\n";


function $fc695d2eafc2b351$var$getCursor(svg, fallback) {
    // WebKit renders SVG cursors blurry on 2x screens: https://bugs.webkit.org/show_bug.cgi?id=160657
    // To work around this, we generate two SVGs at different sizes and use image-set to pick between them.
    // Only do this in WebKit to avoid Firefox rendering the cursor at twice the size.
    if ((0, $gyhAV$isWebKit)()) return `image-set(url("data:image/svg+xml,${encodeURIComponent(svg)}") 1x, url("data:image/svg+xml,${encodeURIComponent(svg.replace('width="32" height="32"', 'width="64" height="64"'))}") 2x) 8 8, ${fallback}`;
    else return `url("data:image/svg+xml,${encodeURIComponent(svg)}") 8 8, ${fallback}`;
}
const $fc695d2eafc2b351$var$CURSORS = {
    ew: $fc695d2eafc2b351$var$getCursor((0, (/*@__PURE__*/$parcel$interopDefault($ff1f3562a1be2216$exports))), "ew-resize"),
    w: $fc695d2eafc2b351$var$getCursor((0, (/*@__PURE__*/$parcel$interopDefault($58407c2d4eb33511$exports))), "w-resize"),
    e: $fc695d2eafc2b351$var$getCursor((0, (/*@__PURE__*/$parcel$interopDefault($2fc56f9fdd93be52$exports))), "e-resize")
};
function $fc695d2eafc2b351$var$Resizer(props, ref) {
    let { column: column, showResizer: showResizer } = props;
    let { isEmpty: isEmpty, layout: layout, onFocusedResizer: onFocusedResizer } = (0, $bd013581c0a4b065$export$3cb274deb6c2d854)();
    // Virtualizer re-renders, but these components are all cached
    // in order to get around that and cause a rerender here, we use context
    // but we don't actually need any value, they are available on the layout object
    (0, $bd013581c0a4b065$export$3f8f74b6bfd2c5df)();
    let stringFormatter = (0, $gyhAV$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($cc43b2d4f6011c04$exports))), "@react-spectrum/table");
    let { direction: direction } = (0, $gyhAV$useLocale)();
    let [isPointerDown, setIsPointerDown] = (0, $gyhAV$useState)(false);
    (0, $gyhAV$useEffect)(()=>{
        let setDown = (e)=>{
            if (e.pointerType === "mouse") setIsPointerDown(true);
        };
        let setUp = (e)=>{
            if (e.pointerType === "mouse") setIsPointerDown(false);
        };
        document.addEventListener("pointerdown", setDown, {
            capture: true
        });
        document.addEventListener("pointerup", setUp, {
            capture: true
        });
        return ()=>{
            document.removeEventListener("pointerdown", setDown, {
                capture: true
            });
            document.removeEventListener("pointerup", setUp, {
                capture: true
            });
        };
    }, []);
    let { inputProps: inputProps, resizerProps: resizerProps } = (0, $gyhAV$useTableColumnResize)((0, $gyhAV$mergeProps)(props, {
        "aria-label": stringFormatter.format("columnResizer"),
        isDisabled: isEmpty
    }), layout, ref);
    let isEResizable = layout.getColumnMinWidth(column.key) >= layout.getColumnWidth(column.key);
    let isWResizable = layout.getColumnMaxWidth(column.key) <= layout.getColumnWidth(column.key);
    let isResizing = layout.resizingColumn === column.key;
    let cursor = "";
    if (isEResizable) cursor = direction === "rtl" ? $fc695d2eafc2b351$var$CURSORS.w : $fc695d2eafc2b351$var$CURSORS.e;
    else if (isWResizable) cursor = direction === "rtl" ? $fc695d2eafc2b351$var$CURSORS.e : $fc695d2eafc2b351$var$CURSORS.w;
    else cursor = $fc695d2eafc2b351$var$CURSORS.ew;
    let style = {
        ...resizerProps.style,
        height: "100%",
        display: showResizer ? undefined : "none",
        cursor: cursor
    };
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$react).Fragment, null, /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        within: true,
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...resizerProps,
        role: "presentation",
        style: style,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-columnResizer")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("input", {
        ref: ref,
        ...(0, $gyhAV$mergeProps)(inputProps, {
            onFocus: onFocusedResizer
        })
    }))), /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        "aria-hidden": true,
        role: "presentation",
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-columnResizerPlaceholder")
    }), /*#__PURE__*/ (0, $gyhAV$react).createElement($fc695d2eafc2b351$var$CursorOverlay, {
        show: isResizing && isPointerDown
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        style: {
            position: "fixed",
            top: 0,
            left: 0,
            bottom: 0,
            right: 0,
            cursor: cursor
        }
    })));
}
function $fc695d2eafc2b351$var$CursorOverlay(props) {
    let { show: show, children: children } = props;
    return show ? /*#__PURE__*/ (0, $gyhAV$reactdom).createPortal(children, document.body) : null;
}
const $fc695d2eafc2b351$export$48a76196cafe3b93 = /*#__PURE__*/ (0, $gyhAV$react).forwardRef($fc695d2eafc2b351$var$Resizer);



/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $4e6fc36ba71ba405$export$d30a7814cfd4033e() {
    let { dropState: dropState, dragAndDropHooks: dragAndDropHooks, state: state } = (0, $bd013581c0a4b065$export$3cb274deb6c2d854)();
    let ref = (0, $gyhAV$useRef)();
    let { dropIndicatorProps: dropIndicatorProps } = dragAndDropHooks.useDropIndicator({
        target: {
            type: "root"
        }
    }, dropState, ref);
    let isDropTarget = dropState.isDropTarget({
        type: "root"
    });
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $gyhAV$useVisuallyHidden)();
    if (!isDropTarget && dropIndicatorProps["aria-hidden"]) return null;
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "row",
        "aria-hidden": dropIndicatorProps["aria-hidden"]
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "gridcell",
        "aria-selected": "false",
        "aria-colspan": state.collection.columns.length
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "button",
        ...visuallyHiddenProps,
        ...dropIndicatorProps,
        ref: ref
    })));
}


/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




function $daa6ead3d9d3506a$export$905ab40ac2179daa(props) {
    let { itemText: itemText, itemCount: itemCount, height: height, maxWidth: maxWidth } = props;
    let isDraggingMultiple = itemCount > 1;
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$Flex), {
        justifyContent: "space-between",
        height: height,
        maxWidth: maxWidth,
        UNSAFE_className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-row", (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-row", "react-spectrum-Table-row-dragPreview", {
            "react-spectrum-Table-row-dragPreview--multiple": isDraggingMultiple
        }))
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-cell", (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-cell"))
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("span", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-cellContents")
    }, itemText)), isDraggingMultiple && /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-row-badge")
    }, itemCount));
}












const $bd013581c0a4b065$var$DEFAULT_HEADER_HEIGHT = {
    medium: 34,
    large: 40
};
const $bd013581c0a4b065$var$DEFAULT_HIDE_HEADER_CELL_WIDTH = {
    medium: 38,
    large: 46
};
const $bd013581c0a4b065$var$ROW_HEIGHTS = {
    compact: {
        medium: 32,
        large: 40
    },
    regular: {
        medium: 40,
        large: 50
    },
    spacious: {
        medium: 48,
        large: 60
    }
};
const $bd013581c0a4b065$var$SELECTION_CELL_DEFAULT_WIDTH = {
    medium: 38,
    large: 48
};
const $bd013581c0a4b065$var$DRAG_BUTTON_CELL_DEFAULT_WIDTH = {
    medium: 16,
    large: 20
};
const $bd013581c0a4b065$var$LEVEL_OFFSET_WIDTH = {
    medium: 16,
    large: 20
};
const $bd013581c0a4b065$export$93e4b0b2cc49b648 = /*#__PURE__*/ (0, $gyhAV$react).createContext(null);
function $bd013581c0a4b065$export$3cb274deb6c2d854() {
    return (0, $gyhAV$useContext)($bd013581c0a4b065$export$93e4b0b2cc49b648);
}
const $bd013581c0a4b065$export$d288a7dd40372bc = /*#__PURE__*/ (0, $gyhAV$react).createContext(null);
function $bd013581c0a4b065$export$3f8f74b6bfd2c5df() {
    return (0, $gyhAV$useContext)($bd013581c0a4b065$export$d288a7dd40372bc);
}
function $bd013581c0a4b065$var$TableViewBase(props, ref) {
    var _dropState_target;
    props = (0, $gyhAV$useProviderProps)(props);
    let { isQuiet: isQuiet, onAction: onAction, onResizeStart: propsOnResizeStart, onResizeEnd: propsOnResizeEnd, dragAndDropHooks: dragAndDropHooks, state: state } = props;
    let isTableDraggable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDraggableCollectionState);
    let isTableDroppable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDroppableCollectionState);
    let dragHooksProvided = (0, $gyhAV$useRef)(isTableDraggable);
    let dropHooksProvided = (0, $gyhAV$useRef)(isTableDroppable);
    (0, $gyhAV$useEffect)(()=>{
        if (dragHooksProvided.current !== isTableDraggable) console.warn("Drag hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.");
        if (dropHooksProvided.current !== isTableDroppable) console.warn("Drop hooks were provided during one render, but not another. This should be avoided as it may produce unexpected behavior.");
        if ("expandedKeys" in state && (isTableDraggable || isTableDroppable)) console.warn("Drag and drop is not yet fully supported with expandable rows and may produce unexpected results.");
    }, [
        isTableDraggable,
        isTableDroppable,
        state
    ]);
    let { styleProps: styleProps } = (0, $gyhAV$useStyleProps)(props);
    let { direction: direction } = (0, $gyhAV$useLocale)();
    let { scale: scale } = (0, $gyhAV$useProvider)();
    const getDefaultWidth = (0, $gyhAV$useCallback)(({ props: { hideHeader: hideHeader, isSelectionCell: isSelectionCell, showDivider: showDivider, isDragButtonCell: isDragButtonCell } })=>{
        if (hideHeader) {
            let width = $bd013581c0a4b065$var$DEFAULT_HIDE_HEADER_CELL_WIDTH[scale];
            return showDivider ? width + 1 : width;
        } else if (isSelectionCell) return $bd013581c0a4b065$var$SELECTION_CELL_DEFAULT_WIDTH[scale];
        else if (isDragButtonCell) return $bd013581c0a4b065$var$DRAG_BUTTON_CELL_DEFAULT_WIDTH[scale];
    }, [
        scale
    ]);
    const getDefaultMinWidth = (0, $gyhAV$useCallback)(({ props: { hideHeader: hideHeader, isSelectionCell: isSelectionCell, showDivider: showDivider, isDragButtonCell: isDragButtonCell } })=>{
        if (hideHeader) {
            let width = $bd013581c0a4b065$var$DEFAULT_HIDE_HEADER_CELL_WIDTH[scale];
            return showDivider ? width + 1 : width;
        } else if (isSelectionCell) return $bd013581c0a4b065$var$SELECTION_CELL_DEFAULT_WIDTH[scale];
        else if (isDragButtonCell) return $bd013581c0a4b065$var$DRAG_BUTTON_CELL_DEFAULT_WIDTH[scale];
        return 75;
    }, [
        scale
    ]);
    // Starts when the user selects resize from the menu, ends when resizing ends
    // used to control the visibility of the resizer Nubbin
    let [isInResizeMode, setIsInResizeMode] = (0, $gyhAV$useState)(false);
    // Starts when the resizer is actually moved
    // entering resizing/exiting resizing doesn't trigger a render
    // with table layout, so we need to track it here
    let [, setIsResizing] = (0, $gyhAV$useState)(false);
    let domRef = (0, $gyhAV$useDOMRef)(ref);
    let headerRef = (0, $gyhAV$useRef)();
    let bodyRef = (0, $gyhAV$useRef)();
    let stringFormatter = (0, $gyhAV$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($cc43b2d4f6011c04$exports))), "@react-spectrum/table");
    let density = props.density || "regular";
    let columnLayout = (0, $gyhAV$useMemo)(()=>new (0, $gyhAV$TableColumnLayout)({
            getDefaultWidth: getDefaultWidth,
            getDefaultMinWidth: getDefaultMinWidth
        }), [
        getDefaultWidth,
        getDefaultMinWidth
    ]);
    let tableLayout = (0, $gyhAV$useMemo)(()=>new (0, $gyhAV$TableLayout)({
            // If props.rowHeight is auto, then use estimated heights based on scale, otherwise use fixed heights.
            rowHeight: props.overflowMode === "wrap" ? null : $bd013581c0a4b065$var$ROW_HEIGHTS[density][scale],
            estimatedRowHeight: props.overflowMode === "wrap" ? $bd013581c0a4b065$var$ROW_HEIGHTS[density][scale] : null,
            headingHeight: props.overflowMode === "wrap" ? null : $bd013581c0a4b065$var$DEFAULT_HEADER_HEIGHT[scale],
            estimatedHeadingHeight: props.overflowMode === "wrap" ? $bd013581c0a4b065$var$DEFAULT_HEADER_HEIGHT[scale] : null,
            columnLayout: columnLayout,
            initialCollection: state.collection
        }), // don't recompute when state.collection changes, only used for initial value
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [
        props.overflowMode,
        scale,
        density,
        columnLayout
    ]);
    // Use a proxy so that a new object is created for each render so that alternate instances aren't affected by mutation.
    // This can be thought of as equivalent to `{tableLayout, tableState: state}`, but works with classes as well.
    let layout = (0, $gyhAV$useMemo)(()=>{
        let proxy = new Proxy(tableLayout, {
            get (target, prop, receiver) {
                return prop === "tableState" ? state : Reflect.get(target, prop, receiver);
            }
        });
        return proxy;
    }, [
        state,
        tableLayout
    ]);
    let dragState;
    let preview = (0, $gyhAV$useRef)(null);
    if (isTableDraggable) {
        dragState = dragAndDropHooks.useDraggableCollectionState({
            collection: state.collection,
            selectionManager: state.selectionManager,
            preview: preview
        });
        dragAndDropHooks.useDraggableCollection({}, dragState, domRef);
    }
    let DragPreview = dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.DragPreview;
    let dropState;
    let droppableCollection;
    let isRootDropTarget;
    if (isTableDroppable) {
        dropState = dragAndDropHooks.useDroppableCollectionState({
            collection: state.collection,
            selectionManager: state.selectionManager
        });
        droppableCollection = dragAndDropHooks.useDroppableCollection({
            keyboardDelegate: layout,
            dropTargetDelegate: layout
        }, dropState, domRef);
        isRootDropTarget = dropState.isDropTarget({
            type: "root"
        });
    }
    let { gridProps: gridProps } = (0, $gyhAV$useTable)({
        ...props,
        isVirtualized: true,
        layout: layout,
        onRowAction: onAction
    }, state, domRef);
    let [headerMenuOpen, setHeaderMenuOpen] = (0, $gyhAV$useState)(false);
    let [headerRowHovered, setHeaderRowHovered] = (0, $gyhAV$useState)(false);
    let renderWrapper = (parent, reusableView, children, renderChildren)=>{
        let style = (0, $gyhAV$layoutInfoToStyle)(reusableView.layoutInfo, direction, parent && parent.layoutInfo);
        if (style.overflow === "hidden") style.overflow = "visible"; // needed to support position: sticky
        if (reusableView.viewType === "rowgroup") return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableRowGroup, {
            key: reusableView.key,
            style: style
        }, isTableDroppable && /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $4e6fc36ba71ba405$export$d30a7814cfd4033e), {
            key: "root"
        }), renderChildren(children));
        if (reusableView.viewType === "header") return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableHeader, {
            key: reusableView.key,
            style: style
        }, renderChildren(children));
        if (reusableView.viewType === "row") return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableRow, {
            key: reusableView.key,
            item: reusableView.content,
            style: style,
            hasActions: onAction,
            isTableDroppable: isTableDroppable,
            isTableDraggable: isTableDraggable
        }, renderChildren(children));
        if (reusableView.viewType === "headerrow") return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableHeaderRow, {
            onHoverChange: setHeaderRowHovered,
            key: reusableView.key,
            style: style,
            item: reusableView.content
        }, renderChildren(children));
        let isDropTarget;
        let isRootDroptarget;
        if (isTableDroppable) {
            if (parent.content) isDropTarget = dropState.isDropTarget({
                type: "item",
                dropPosition: "on",
                key: parent.content.key
            });
            isRootDroptarget = dropState.isDropTarget({
                type: "root"
            });
        }
        return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$VirtualizerItem), {
            key: reusableView.key,
            layoutInfo: reusableView.layoutInfo,
            virtualizer: reusableView.virtualizer,
            parent: parent === null || parent === void 0 ? void 0 : parent.layoutInfo,
            className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-cellWrapper", (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), {
                "react-spectrum-Table-cellWrapper": !reusableView.layoutInfo.estimatedSize,
                "react-spectrum-Table-cellWrapper--dropTarget": isDropTarget || isRootDroptarget
            }))
        }, reusableView.rendered);
    };
    let renderView = (type, item)=>{
        switch(type){
            case "header":
            case "rowgroup":
            case "section":
            case "row":
            case "headerrow":
                return null;
            case "cell":
                if (item.props.isSelectionCell) return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableCheckboxCell, {
                    cell: item
                });
                if (item.props.isDragButtonCell) return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableDragCell, {
                    cell: item
                });
                return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableCell, {
                    cell: item
                });
            case "placeholder":
                // TODO: move to react-aria?
                return /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
                    role: "gridcell",
                    "aria-colindex": item.index + 1,
                    "aria-colspan": item.colspan > 1 ? item.colspan : null
                });
            case "column":
                if (item.props.isSelectionCell) return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableSelectAllCell, {
                    column: item
                });
                if (item.props.isDragButtonCell) return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableDragHeaderCell, {
                    column: item
                });
                // TODO: consider this case, what if we have hidden headers and a empty table
                if (item.props.hideHeader) return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$TooltipTrigger), {
                    placement: "top",
                    trigger: "focus"
                }, /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableColumnHeader, {
                    column: item
                }), /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$Tooltip), {
                    placement: "top"
                }, item.rendered));
                if (item.props.allowsResizing && !item.hasChildNodes) return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$ResizableTableColumnHeader, {
                    tableRef: domRef,
                    column: item
                });
                return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableColumnHeader, {
                    column: item
                });
            case "loader":
                return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$CenteredWrapper, null, /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$ProgressCircle), {
                    isIndeterminate: true,
                    "aria-label": state.collection.size > 0 ? stringFormatter.format("loadingMore") : stringFormatter.format("loading")
                }));
            case "empty":
                {
                    let emptyState = props.renderEmptyState ? props.renderEmptyState() : null;
                    if (emptyState == null) return null;
                    return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$CenteredWrapper, null, emptyState);
                }
        }
    };
    let [isVerticalScrollbarVisible, setVerticalScollbarVisible] = (0, $gyhAV$useState)(false);
    let [isHorizontalScrollbarVisible, setHorizontalScollbarVisible] = (0, $gyhAV$useState)(false);
    let viewport = (0, $gyhAV$useRef)({
        x: 0,
        y: 0,
        width: 0,
        height: 0
    });
    let onVisibleRectChange = (0, $gyhAV$useCallback)((e)=>{
        if (viewport.current.width === e.width && viewport.current.height === e.height) return;
        viewport.current = e;
        if (bodyRef.current) {
            setVerticalScollbarVisible(bodyRef.current.clientWidth + 2 < bodyRef.current.offsetWidth);
            setHorizontalScollbarVisible(bodyRef.current.clientHeight + 2 < bodyRef.current.offsetHeight);
        }
    }, []);
    let { isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $gyhAV$useFocusRing)();
    let isEmpty = state.collection.size === 0;
    let onFocusedResizer = ()=>{
        bodyRef.current.scrollLeft = headerRef.current.scrollLeft;
    };
    let onResizeStart = (0, $gyhAV$useCallback)((widths)=>{
        setIsResizing(true);
        propsOnResizeStart === null || propsOnResizeStart === void 0 ? void 0 : propsOnResizeStart(widths);
    }, [
        setIsResizing,
        propsOnResizeStart
    ]);
    let onResizeEnd = (0, $gyhAV$useCallback)((widths)=>{
        setIsInResizeMode(false);
        setIsResizing(false);
        propsOnResizeEnd === null || propsOnResizeEnd === void 0 ? void 0 : propsOnResizeEnd(widths);
    }, [
        propsOnResizeEnd,
        setIsInResizeMode,
        setIsResizing
    ]);
    let focusedKey = state.selectionManager.focusedKey;
    if ((dropState === null || dropState === void 0 ? void 0 : (_dropState_target = dropState.target) === null || _dropState_target === void 0 ? void 0 : _dropState_target.type) === "item") focusedKey = dropState.target.key;
    let mergedProps = (0, $gyhAV$mergeProps)(isTableDroppable && (droppableCollection === null || droppableCollection === void 0 ? void 0 : droppableCollection.collectionProps), gridProps, focusProps, (dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.isVirtualDragging()) && {
        tabIndex: null
    });
    return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$export$93e4b0b2cc49b648.Provider, {
        value: {
            state: state,
            dragState: dragState,
            dropState: dropState,
            dragAndDropHooks: dragAndDropHooks,
            isTableDraggable: isTableDraggable,
            isTableDroppable: isTableDroppable,
            layout: layout,
            onResizeStart: onResizeStart,
            onResize: props.onResize,
            onResizeEnd: onResizeEnd,
            headerRowHovered: headerRowHovered,
            isInResizeMode: isInResizeMode,
            setIsInResizeMode: setIsInResizeMode,
            isEmpty: isEmpty,
            onFocusedResizer: onFocusedResizer,
            headerMenuOpen: headerMenuOpen,
            setHeaderMenuOpen: setHeaderMenuOpen
        }
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableVirtualizer, {
        ...mergedProps,
        ...styleProps,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table", `spectrum-Table--${density}`, {
            "spectrum-Table--quiet": isQuiet,
            "spectrum-Table--wrap": props.overflowMode === "wrap",
            "spectrum-Table--loadingMore": state.collection.body.props.loadingState === "loadingMore",
            "spectrum-Table--isVerticalScrollbarVisible": isVerticalScrollbarVisible,
            "spectrum-Table--isHorizontalScrollbarVisible": isHorizontalScrollbarVisible
        }, (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table"), styleProps.className),
        layout: layout,
        collection: state.collection,
        focusedKey: focusedKey,
        renderView: renderView,
        renderWrapper: renderWrapper,
        onVisibleRectChange: onVisibleRectChange,
        domRef: domRef,
        headerRef: headerRef,
        bodyRef: bodyRef,
        isFocusVisible: isFocusVisible,
        isVirtualDragging: dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.isVirtualDragging(),
        isRootDropTarget: isRootDropTarget
    }), DragPreview && isTableDraggable && /*#__PURE__*/ (0, $gyhAV$react).createElement(DragPreview, {
        ref: preview
    }, ()=>{
        if (dragAndDropHooks.renderPreview) return dragAndDropHooks.renderPreview(dragState.draggingKeys, dragState.draggedKey);
        let itemCount = dragState.draggingKeys.size;
        let maxWidth = bodyRef.current.getBoundingClientRect().width;
        let height = $bd013581c0a4b065$var$ROW_HEIGHTS[density][scale];
        let itemText = state.collection.getTextValue(dragState.draggedKey);
        return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $daa6ead3d9d3506a$export$905ab40ac2179daa), {
            itemText: itemText,
            itemCount: itemCount,
            height: height,
            maxWidth: maxWidth
        });
    }));
}
// This is a custom Virtualizer that also has a header that syncs its scroll position with the body.
function $bd013581c0a4b065$var$TableVirtualizer(props) {
    var _layout_getLayoutInfo;
    let { layout: layout, collection: collection, focusedKey: focusedKey, renderView: renderView, renderWrapper: renderWrapper, domRef: domRef, bodyRef: bodyRef, headerRef: headerRef, onVisibleRectChange: onVisibleRectChangeProp, isFocusVisible: isFocusVisible, isVirtualDragging: isVirtualDragging, isRootDropTarget: isRootDropTarget, ...otherProps } = props;
    let { direction: direction } = (0, $gyhAV$useLocale)();
    let loadingState = collection.body.props.loadingState;
    let isLoading = loadingState === "loading" || loadingState === "loadingMore";
    let onLoadMore = collection.body.props.onLoadMore;
    let transitionDuration = 220;
    if (isLoading) transitionDuration = 160;
    if (layout.resizingColumn != null) // while resizing, prop changes should not cause animations
    transitionDuration = 0;
    let state = (0, $gyhAV$useVirtualizerState)({
        layout: layout,
        collection: collection,
        renderView: renderView,
        renderWrapper: renderWrapper,
        onVisibleRectChange (rect) {
            bodyRef.current.scrollTop = rect.y;
            (0, $gyhAV$setScrollLeft)(bodyRef.current, direction, rect.x);
        },
        transitionDuration: transitionDuration
    });
    let scrollToItem = (0, $gyhAV$useCallback)((key)=>{
        let item = collection.getItem(key);
        let column = collection.columns[0];
        let virtualizer = state.virtualizer;
        virtualizer.scrollToItem(key, {
            duration: 0,
            // Prevent scrolling to the top when clicking on column headers.
            shouldScrollY: (item === null || item === void 0 ? void 0 : item.type) !== "column",
            // Offset scroll position by width of selection cell
            // (which is sticky and will overlap the cell we're scrolling to).
            offsetX: column.props.isSelectionCell || column.props.isDragButtonCell ? layout.getColumnWidth(column.key) : 0
        });
        // Sync the scroll positions of the column headers and the body so scrollIntoViewport can
        // properly decide if the column is outside the viewport or not
        headerRef.current.scrollLeft = bodyRef.current.scrollLeft;
    }, [
        collection,
        bodyRef,
        headerRef,
        layout,
        state.virtualizer
    ]);
    let memoedVirtualizerProps = (0, $gyhAV$useMemo)(()=>({
            tabIndex: otherProps.tabIndex,
            focusedKey: focusedKey,
            scrollToItem: scrollToItem,
            isLoading: isLoading,
            onLoadMore: onLoadMore
        }), [
        otherProps.tabIndex,
        focusedKey,
        scrollToItem,
        isLoading,
        onLoadMore
    ]);
    let { virtualizerProps: virtualizerProps, scrollViewProps: { onVisibleRectChange: onVisibleRectChange } } = (0, $gyhAV$useVirtualizer)(memoedVirtualizerProps, state, domRef);
    // this effect runs whenever the contentSize changes, it doesn't matter what the content size is
    // only that it changes in a resize, and when that happens, we want to sync the body to the
    // header scroll position
    (0, $gyhAV$useEffect)(()=>{
        if ((0, $gyhAV$getInteractionModality)() === "keyboard" && headerRef.current.contains(document.activeElement)) {
            (0, $gyhAV$scrollIntoView)(headerRef.current, document.activeElement);
            (0, $gyhAV$scrollIntoViewport)(document.activeElement, {
                containingElement: domRef.current
            });
            bodyRef.current.scrollLeft = headerRef.current.scrollLeft;
        }
    }, [
        state.contentSize,
        headerRef,
        bodyRef,
        domRef
    ]);
    let headerHeight = ((_layout_getLayoutInfo = layout.getLayoutInfo("header")) === null || _layout_getLayoutInfo === void 0 ? void 0 : _layout_getLayoutInfo.rect.height) || 0;
    // Sync the scroll position from the table body to the header container.
    let onScroll = (0, $gyhAV$useCallback)(()=>{
        headerRef.current.scrollLeft = bodyRef.current.scrollLeft;
    }, [
        bodyRef,
        headerRef
    ]);
    let resizerPosition = layout.getResizerPosition() - 2;
    let resizerAtEdge = resizerPosition > Math.max(state.virtualizer.contentSize.width, state.virtualizer.visibleRect.width) - 3;
    // this should be fine, every movement of the resizer causes a rerender
    // scrolling can cause it to lag for a moment, but it's always updated
    let resizerInVisibleRegion = resizerPosition < state.virtualizer.visibleRect.maxX;
    let shouldHardCornerResizeCorner = resizerAtEdge && resizerInVisibleRegion;
    // minimize re-render caused on Resizers by memoing this
    let resizingColumnWidth = layout.getColumnWidth(layout.resizingColumn);
    let resizingColumn = (0, $gyhAV$useMemo)(()=>({
            width: resizingColumnWidth,
            key: layout.resizingColumn
        }), [
        resizingColumnWidth,
        layout.resizingColumn
    ]);
    let mergedProps = (0, $gyhAV$mergeProps)(otherProps, virtualizerProps, isVirtualDragging && {
        tabIndex: null
    });
    return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$export$d288a7dd40372bc.Provider, {
        value: resizingColumn
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusScope), null, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...mergedProps,
        ref: domRef
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "presentation",
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headWrapper"),
        style: {
            height: headerHeight,
            overflow: "hidden",
            position: "relative",
            willChange: state.isScrolling ? "scroll-position" : undefined,
            transition: state.isAnimating ? `none ${state.virtualizer.transitionDuration}ms` : undefined
        },
        ref: headerRef
    }, state.visibleViews[0]), /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$ScrollView), {
        role: "presentation",
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-body", {
            "focus-ring": isFocusVisible,
            "spectrum-Table-body--resizerAtTableEdge": shouldHardCornerResizeCorner
        }, (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-body", {
            "react-spectrum-Table-body--dropTarget": !!isRootDropTarget
        })),
        tabIndex: isVirtualDragging ? null : -1,
        style: {
            flex: 1
        },
        innerStyle: {
            overflow: "visible",
            transition: state.isAnimating ? `none ${state.virtualizer.transitionDuration}ms` : undefined
        },
        ref: bodyRef,
        contentSize: state.contentSize,
        onVisibleRectChange: (0, $gyhAV$chain)(onVisibleRectChange, onVisibleRectChangeProp),
        onScrollStart: state.startScrolling,
        onScrollEnd: state.endScrolling,
        onScroll: onScroll
    }, state.visibleViews[1], /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-bodyResizeIndicator"),
        style: {
            [direction === "ltr" ? "left" : "right"]: `${resizerPosition}px`,
            height: `${Math.max(state.virtualizer.contentSize.height, state.virtualizer.visibleRect.height)}px`,
            display: layout.resizingColumn ? "block" : "none"
        }
    })))));
}
function $bd013581c0a4b065$var$TableHeader({ children: children, ...otherProps }) {
    let { rowGroupProps: rowGroupProps } = (0, $gyhAV$useTableRowGroup)();
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...rowGroupProps,
        ...otherProps,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-head")
    }, children);
}
function $bd013581c0a4b065$var$TableColumnHeader(props) {
    var _state_sortDescriptor, _state_sortDescriptor1, _state_sortDescriptor2, _state_sortDescriptor3;
    let { column: column } = props;
    let ref = (0, $gyhAV$useRef)(null);
    let { state: state, isEmpty: isEmpty } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let { pressProps: pressProps, isPressed: isPressed } = (0, $gyhAV$usePress)({
        isDisabled: isEmpty
    });
    let columnProps = column.props;
    (0, $gyhAV$useEffect)(()=>{
        if (column.hasChildNodes && columnProps.allowsResizing) console.warn(`Column key: ${column.key}. Columns with child columns don't allow resizing.`);
    }, [
        column.hasChildNodes,
        column.key,
        columnProps.allowsResizing
    ]);
    let { columnHeaderProps: columnHeaderProps } = (0, $gyhAV$useTableColumnHeader)({
        node: column,
        isVirtualized: true
    }, state, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gyhAV$useHover)({
        ...props,
        isDisabled: isEmpty
    });
    const allProps = [
        columnHeaderProps,
        hoverProps,
        pressProps
    ];
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...(0, $gyhAV$mergeProps)(...allProps),
        ref: ref,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headCell", {
            "is-active": isPressed,
            "is-sortable": columnProps.allowsSorting,
            "is-sorted-desc": ((_state_sortDescriptor = state.sortDescriptor) === null || _state_sortDescriptor === void 0 ? void 0 : _state_sortDescriptor.column) === column.key && ((_state_sortDescriptor1 = state.sortDescriptor) === null || _state_sortDescriptor1 === void 0 ? void 0 : _state_sortDescriptor1.direction) === "descending",
            "is-sorted-asc": ((_state_sortDescriptor2 = state.sortDescriptor) === null || _state_sortDescriptor2 === void 0 ? void 0 : _state_sortDescriptor2.column) === column.key && ((_state_sortDescriptor3 = state.sortDescriptor) === null || _state_sortDescriptor3 === void 0 ? void 0 : _state_sortDescriptor3.direction) === "ascending",
            "is-hovered": isHovered,
            "spectrum-Table-cell--hideHeader": columnProps.hideHeader
        }, (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-cell", {
            "react-spectrum-Table-cell--alignCenter": columnProps.align === "center" || column.colspan > 1,
            "react-spectrum-Table-cell--alignEnd": columnProps.align === "end"
        }))
    }, columnProps.allowsSorting && /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$spectrumiconsuiArrowDownSmall), {
        UNSAFE_className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-sortedIcon")
    }), columnProps.hideHeader ? /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$VisuallyHidden), null, column.rendered) : /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headCellContents")
    }, column.rendered)));
}
let $bd013581c0a4b065$var$_TableColumnHeaderButton = (props, ref)=>{
    let { focusProps: focusProps, alignment: alignment, ...otherProps } = props;
    let { isEmpty: isEmpty } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let domRef = (0, $gyhAV$useFocusableRef)(ref);
    let { buttonProps: buttonProps } = (0, $gyhAV$useButton)({
        ...otherProps,
        elementType: "div",
        isDisabled: isEmpty
    }, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gyhAV$useHover)({
        ...otherProps,
        isDisabled: isEmpty
    });
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headCellContents", {
            "is-hovered": isHovered
        }),
        ...hoverProps
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headCellButton", {
            "spectrum-Table-headCellButton--alignStart": alignment === "start",
            "spectrum-Table-headCellButton--alignCenter": alignment === "center",
            "spectrum-Table-headCellButton--alignEnd": alignment === "end"
        }),
        ...(0, $gyhAV$mergeProps)(buttonProps, focusProps),
        ref: domRef
    }, props.children));
};
let $bd013581c0a4b065$var$TableColumnHeaderButton = /*#__PURE__*/ (0, $gyhAV$react).forwardRef($bd013581c0a4b065$var$_TableColumnHeaderButton);
function $bd013581c0a4b065$var$ResizableTableColumnHeader(props) {
    var _column_props, _state_sortDescriptor, _state_sortDescriptor1, _state_sortDescriptor2, _state_sortDescriptor3;
    let { column: column } = props;
    let ref = (0, $gyhAV$useRef)(null);
    let triggerRef = (0, $gyhAV$useRef)(null);
    let resizingRef = (0, $gyhAV$useRef)(null);
    let { state: state, layout: layout, onResizeStart: onResizeStart, onResize: onResize, onResizeEnd: onResizeEnd, headerRowHovered: headerRowHovered, setIsInResizeMode: setIsInResizeMode, isEmpty: isEmpty, isInResizeMode: isInResizeMode, headerMenuOpen: headerMenuOpen, setHeaderMenuOpen: setHeaderMenuOpen } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let stringFormatter = (0, $gyhAV$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($cc43b2d4f6011c04$exports))), "@react-spectrum/table");
    let { pressProps: pressProps, isPressed: isPressed } = (0, $gyhAV$usePress)({
        isDisabled: isEmpty
    });
    let { columnHeaderProps: columnHeaderProps } = (0, $gyhAV$useTableColumnHeader)({
        node: column,
        isVirtualized: true
    }, state, ref);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gyhAV$useHover)({
        ...props,
        isDisabled: isEmpty || headerMenuOpen
    });
    const allProps = [
        columnHeaderProps,
        pressProps,
        hoverProps
    ];
    let columnProps = column.props;
    let { isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $gyhAV$useFocusRing)();
    const onMenuSelect = (key)=>{
        switch(key){
            case "sort-asc":
                state.sort(column.key, "ascending");
                break;
            case "sort-desc":
                state.sort(column.key, "descending");
                break;
            case "resize":
                layout.startResize(column.key);
                setIsInResizeMode(true);
                state.setKeyboardNavigationDisabled(true);
                break;
        }
    };
    let allowsSorting = (_column_props = column.props) === null || _column_props === void 0 ? void 0 : _column_props.allowsSorting;
    let items = (0, $gyhAV$useMemo)(()=>{
        let options = [
            allowsSorting ? {
                label: stringFormatter.format("sortAscending"),
                id: "sort-asc"
            } : undefined,
            allowsSorting ? {
                label: stringFormatter.format("sortDescending"),
                id: "sort-desc"
            } : undefined,
            {
                label: stringFormatter.format("resizeColumn"),
                id: "resize"
            }
        ];
        return options;
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [
        allowsSorting
    ]);
    let resizingColumn = layout.resizingColumn;
    let showResizer = !isEmpty && (headerRowHovered && (0, $gyhAV$getInteractionModality)() !== "keyboard" || resizingColumn != null);
    let alignment = "start";
    let menuAlign = "start";
    if (columnProps.align === "center" || column.colspan > 1) alignment = "center";
    else if (columnProps.align === "end") {
        alignment = "end";
        menuAlign = "end";
    }
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...(0, $gyhAV$mergeProps)(...allProps),
        ref: ref,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headCell", {
            "is-active": isPressed,
            "is-resizable": columnProps.allowsResizing,
            "is-sortable": columnProps.allowsSorting,
            "is-sorted-desc": ((_state_sortDescriptor = state.sortDescriptor) === null || _state_sortDescriptor === void 0 ? void 0 : _state_sortDescriptor.column) === column.key && ((_state_sortDescriptor1 = state.sortDescriptor) === null || _state_sortDescriptor1 === void 0 ? void 0 : _state_sortDescriptor1.direction) === "descending",
            "is-sorted-asc": ((_state_sortDescriptor2 = state.sortDescriptor) === null || _state_sortDescriptor2 === void 0 ? void 0 : _state_sortDescriptor2.column) === column.key && ((_state_sortDescriptor3 = state.sortDescriptor) === null || _state_sortDescriptor3 === void 0 ? void 0 : _state_sortDescriptor3.direction) === "ascending",
            "is-hovered": isHovered,
            "focus-ring": isFocusVisible,
            "spectrum-Table-cell--hideHeader": columnProps.hideHeader
        }, (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-cell", {
            "react-spectrum-Table-cell--alignCenter": alignment === "center",
            "react-spectrum-Table-cell--alignEnd": alignment === "end"
        }))
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$MenuTrigger), {
        onOpenChange: setHeaderMenuOpen,
        align: menuAlign
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableColumnHeaderButton, {
        alignment: alignment,
        ref: triggerRef,
        focusProps: focusProps
    }, columnProps.allowsSorting && /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$spectrumiconsuiArrowDownSmall), {
        UNSAFE_className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-sortedIcon")
    }), columnProps.hideHeader ? /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$VisuallyHidden), null, column.rendered) : /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headerCellText")
    }, column.rendered), columnProps.allowsResizing && /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$spectrumiconsuiChevronDownMedium), {
        UNSAFE_className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-menuChevron")
    })), /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$Menu), {
        onAction: onMenuSelect,
        minWidth: "size-2000",
        items: items
    }, (item)=>/*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$Item), null, item.label))), /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $fc695d2eafc2b351$export$48a76196cafe3b93), {
        ref: resizingRef,
        column: column,
        showResizer: showResizer,
        onResizeStart: onResizeStart,
        onResize: onResize,
        onResizeEnd: onResizeEnd,
        triggerRef: (0, $gyhAV$useUnwrapDOMRef)(triggerRef)
    }), /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        "aria-hidden": true,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-colResizeIndicator", {
            "spectrum-Table-colResizeIndicator--visible": resizingColumn != null,
            "spectrum-Table-colResizeIndicator--resizing": resizingColumn === column.key
        })
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-colResizeNubbin", {
            "spectrum-Table-colResizeNubbin--visible": isInResizeMode && resizingColumn === column.key
        })
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $cb7d341e8868d81d$export$d9658cdf8c86807), null)))));
}
function $bd013581c0a4b065$var$TableSelectAllCell({ column: column }) {
    let ref = (0, $gyhAV$useRef)();
    let { state: state } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let isSingleSelectionMode = state.selectionManager.selectionMode === "single";
    let { columnHeaderProps: columnHeaderProps } = (0, $gyhAV$useTableColumnHeader)({
        node: column,
        isVirtualized: true
    }, state, ref);
    let { checkboxProps: checkboxProps } = (0, $gyhAV$useTableSelectAllCheckbox)(state);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gyhAV$useHover)({});
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...(0, $gyhAV$mergeProps)(columnHeaderProps, hoverProps),
        ref: ref,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headCell", "spectrum-Table-checkboxCell", {
            "is-hovered": isHovered
        })
    }, /*
            In single selection mode, the checkbox will be hidden.
            So to avoid leaving a column header with no accessible content,
            we use a VisuallyHidden component to include the aria-label from the checkbox,
            which for single selection will be "Select."
          */ isSingleSelectionMode && /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$VisuallyHidden), null, checkboxProps["aria-label"]), /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$Checkbox), {
        ...checkboxProps,
        isEmphasized: true,
        UNSAFE_style: isSingleSelectionMode ? {
            visibility: "hidden"
        } : undefined,
        UNSAFE_className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-checkbox")
    })));
}
function $bd013581c0a4b065$var$TableDragHeaderCell({ column: column }) {
    let ref = (0, $gyhAV$useRef)();
    let { state: state } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let { columnHeaderProps: columnHeaderProps } = (0, $gyhAV$useTableColumnHeader)({
        node: column,
        isVirtualized: true
    }, state, ref);
    let stringFormatter = (0, $gyhAV$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($cc43b2d4f6011c04$exports))), "@react-spectrum/table");
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...columnHeaderProps,
        ref: ref,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-headCell", (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-headCell", "react-spectrum-Table-dragButtonHeadCell"))
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$VisuallyHidden), null, stringFormatter.format("drag"))));
}
function $bd013581c0a4b065$var$TableRowGroup({ children: children, ...otherProps }) {
    let { rowGroupProps: rowGroupProps } = (0, $gyhAV$useTableRowGroup)();
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...rowGroupProps,
        ...otherProps
    }, children);
}
function $bd013581c0a4b065$var$DragButton() {
    let { dragButtonProps: dragButtonProps, dragButtonRef: dragButtonRef, isFocusVisibleWithin: isFocusVisibleWithin } = $bd013581c0a4b065$export$cd7c5802f9e21187();
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $gyhAV$useVisuallyHidden)();
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...dragButtonProps,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-dragButton"),
        style: !isFocusVisibleWithin ? {
            ...visuallyHiddenProps.style
        } : {},
        ref: dragButtonRef,
        draggable: "true"
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$spectrumiconsuiListGripper), {
        UNSAFE_className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))))
    })));
}
const $bd013581c0a4b065$var$TableRowContext = /*#__PURE__*/ (0, $gyhAV$react).createContext(null);
function $bd013581c0a4b065$export$cd7c5802f9e21187() {
    return (0, $gyhAV$useContext)($bd013581c0a4b065$var$TableRowContext);
}
function $bd013581c0a4b065$var$TableRow({ item: item, children: children, hasActions: hasActions, isTableDraggable: isTableDraggable, isTableDroppable: isTableDroppable, ...otherProps }) {
    var _state_collection_rows_find, _layout_getContentSize, _layout_virtualizer;
    let ref = (0, $gyhAV$useRef)();
    let { state: state, layout: layout, dragAndDropHooks: dragAndDropHooks, dragState: dragState, dropState: dropState } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let allowsInteraction = state.selectionManager.selectionMode !== "none" || hasActions;
    let isDisabled = !allowsInteraction || state.disabledKeys.has(item.key);
    let isDroppable = isTableDroppable && !isDisabled;
    let isSelected = state.selectionManager.isSelected(item.key);
    let { rowProps: rowProps } = (0, $gyhAV$useTableRow)({
        node: item,
        isVirtualized: true,
        shouldSelectOnPressUp: isTableDraggable
    }, state, ref);
    let { pressProps: pressProps, isPressed: isPressed } = (0, $gyhAV$usePress)({
        isDisabled: isDisabled
    });
    // The row should show the focus background style when any cell inside it is focused.
    // If the row itself is focused, then it should have a blue focus indicator on the left.
    let { isFocusVisible: isFocusVisibleWithin, focusProps: focusWithinProps } = (0, $gyhAV$useFocusRing)({
        within: true
    });
    let { isFocusVisible: isFocusVisible, focusProps: focusProps } = (0, $gyhAV$useFocusRing)();
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $gyhAV$useHover)({
        isDisabled: isDisabled
    });
    let isFirstRow = ((_state_collection_rows_find = state.collection.rows.find((row)=>row.level === 1)) === null || _state_collection_rows_find === void 0 ? void 0 : _state_collection_rows_find.key) === item.key;
    let isLastRow = item.nextKey == null;
    // Figure out if the TableView content is equal or greater in height to the container. If so, we'll need to round the bottom
    // border corners of the last row when selected.
    let isFlushWithContainerBottom = false;
    if (isLastRow) {
        if (((_layout_getContentSize = layout.getContentSize()) === null || _layout_getContentSize === void 0 ? void 0 : _layout_getContentSize.height) >= ((_layout_virtualizer = layout.virtualizer) === null || _layout_virtualizer === void 0 ? void 0 : _layout_virtualizer.getVisibleRect().height)) isFlushWithContainerBottom = true;
    }
    let draggableItem;
    if (isTableDraggable) {
        // eslint-disable-next-line react-hooks/rules-of-hooks
        draggableItem = dragAndDropHooks.useDraggableItem({
            key: item.key,
            hasDragButton: true
        }, dragState);
        if (isDisabled) draggableItem = null;
    }
    let droppableItem;
    let isDropTarget;
    let dropIndicator;
    let dropIndicatorRef = (0, $gyhAV$useRef)();
    if (isTableDroppable) {
        let target = {
            type: "item",
            key: item.key,
            dropPosition: "on"
        };
        isDropTarget = dropState.isDropTarget(target);
        // eslint-disable-next-line react-hooks/rules-of-hooks
        dropIndicator = dragAndDropHooks.useDropIndicator({
            target: target
        }, dropState, dropIndicatorRef);
    }
    let dragButtonRef = (0, $gyhAV$react).useRef();
    let { buttonProps: dragButtonProps } = (0, $gyhAV$useButton)({
        ...draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragButtonProps,
        elementType: "div"
    }, dragButtonRef);
    let props = (0, $gyhAV$mergeProps)(rowProps, otherProps, focusWithinProps, focusProps, hoverProps, pressProps, draggableItem === null || draggableItem === void 0 ? void 0 : draggableItem.dragProps, // Remove tab index from list row if performing a screenreader drag. This prevents TalkBack from focusing the row,
    // allowing for single swipe navigation between row drop indicator
    (dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.isVirtualDragging()) && {
        tabIndex: null
    });
    let dropProps = isDroppable ? droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps : {
        "aria-hidden": droppableItem === null || droppableItem === void 0 ? void 0 : droppableItem.dropProps["aria-hidden"]
    };
    let { visuallyHiddenProps: visuallyHiddenProps } = (0, $gyhAV$useVisuallyHidden)();
    return /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$TableRowContext.Provider, {
        value: {
            dragButtonProps: dragButtonProps,
            dragButtonRef: dragButtonRef,
            isFocusVisibleWithin: isFocusVisibleWithin
        }
    }, isTableDroppable && isFirstRow && /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $7ee9a922ee4e8032$export$2c0bab5914a9d088), {
        rowProps: props,
        key: `${item.key}-before`,
        target: {
            key: item.key,
            type: "item",
            dropPosition: "before"
        }
    }), isTableDroppable && !(dropIndicator === null || dropIndicator === void 0 ? void 0 : dropIndicator.isHidden) && /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "row",
        ...visuallyHiddenProps
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "gridcell"
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "button",
        ...dropIndicator === null || dropIndicator === void 0 ? void 0 : dropIndicator.dropIndicatorProps,
        ref: dropIndicatorRef
    }))), /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...(0, $gyhAV$mergeProps)(props, dropProps),
        ref: ref,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-row", {
            "is-active": isPressed,
            "is-selected": isSelected,
            "spectrum-Table-row--highlightSelection": state.selectionManager.selectionBehavior === "replace",
            "is-next-selected": state.selectionManager.isSelected(item.nextKey),
            "is-focused": isFocusVisibleWithin,
            "focus-ring": isFocusVisible,
            "is-hovered": isHovered,
            "is-disabled": isDisabled,
            "spectrum-Table-row--firstRow": isFirstRow,
            "spectrum-Table-row--lastRow": isLastRow,
            "spectrum-Table-row--isFlushBottom": isFlushWithContainerBottom
        }, (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-row", {
            "react-spectrum-Table-row--dropTarget": isDropTarget
        }))
    }, children), isTableDroppable && /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $7ee9a922ee4e8032$export$2c0bab5914a9d088), {
        rowProps: props,
        key: `${item.key}-after`,
        target: {
            key: item.key,
            type: "item",
            dropPosition: "after"
        }
    }));
}
function $bd013581c0a4b065$var$TableHeaderRow({ item: item, children: children, style: style, ...props }) {
    let { state: state, headerMenuOpen: headerMenuOpen } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let ref = (0, $gyhAV$useRef)();
    let { rowProps: rowProps } = (0, $gyhAV$useTableHeaderRow)({
        node: item,
        isVirtualized: true
    }, state, ref);
    let { hoverProps: hoverProps } = (0, $gyhAV$useHover)({
        ...props,
        isDisabled: headerMenuOpen
    });
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...(0, $gyhAV$mergeProps)(rowProps, hoverProps),
        ref: ref,
        style: style
    }, children);
}
function $bd013581c0a4b065$var$TableDragCell({ cell: cell }) {
    let ref = (0, $gyhAV$useRef)();
    let { state: state, isTableDraggable: isTableDraggable } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let isDisabled = state.disabledKeys.has(cell.parentKey);
    let { gridCellProps: gridCellProps } = (0, $gyhAV$useTableCell)({
        node: cell,
        isVirtualized: true
    }, state, ref);
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...gridCellProps,
        ref: ref,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-cell", {
            "is-disabled": isDisabled
        }, (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-cell", "react-spectrum-Table-dragButtonCell"))
    }, isTableDraggable && !isDisabled && /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$DragButton, null)));
}
function $bd013581c0a4b065$var$TableCheckboxCell({ cell: cell }) {
    let ref = (0, $gyhAV$useRef)();
    let { state: state } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let isDisabled = state.disabledKeys.has(cell.parentKey);
    let { gridCellProps: gridCellProps } = (0, $gyhAV$useTableCell)({
        node: cell,
        isVirtualized: true
    }, state, ref);
    let { checkboxProps: checkboxProps } = (0, $gyhAV$useTableSelectionCheckbox)({
        key: cell.parentKey
    }, state);
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...gridCellProps,
        ref: ref,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-cell", "spectrum-Table-checkboxCell", {
            "is-disabled": isDisabled
        }, (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-cell"))
    }, state.selectionManager.selectionMode !== "none" && /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$Checkbox), {
        ...checkboxProps,
        isEmphasized: true,
        isDisabled: isDisabled,
        UNSAFE_className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-checkbox")
    })));
}
function $bd013581c0a4b065$var$TableCell({ cell: cell }) {
    let { scale: scale } = (0, $gyhAV$useProvider)();
    let { state: state } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let isExpandableTable = "expandedKeys" in state;
    let ref = (0, $gyhAV$useRef)();
    let columnProps = cell.column.props;
    let isDisabled = state.disabledKeys.has(cell.parentKey);
    let { gridCellProps: gridCellProps } = (0, $gyhAV$useTableCell)({
        node: cell,
        isVirtualized: true
    }, state, ref);
    let { id: id, ...otherGridCellProps } = gridCellProps;
    let isFirstRowHeaderCell = state.collection.rowHeaderColumnKeys.keys().next().value === cell.column.key;
    let isRowExpandable = false;
    let showExpandCollapseButton = false;
    let levelOffset = 0;
    if ("expandedKeys" in state) {
        var _state_keyMap_get_props_UNSTABLE_childItems, _state_keyMap_get, _state_keyMap_get_props_children, _state_keyMap_get_props, _state_keyMap_get1;
        isRowExpandable = ((_state_keyMap_get = state.keyMap.get(cell.parentKey)) === null || _state_keyMap_get === void 0 ? void 0 : (_state_keyMap_get_props_UNSTABLE_childItems = _state_keyMap_get.props.UNSTABLE_childItems) === null || _state_keyMap_get_props_UNSTABLE_childItems === void 0 ? void 0 : _state_keyMap_get_props_UNSTABLE_childItems.length) > 0 || ((_state_keyMap_get1 = state.keyMap.get(cell.parentKey)) === null || _state_keyMap_get1 === void 0 ? void 0 : (_state_keyMap_get_props = _state_keyMap_get1.props) === null || _state_keyMap_get_props === void 0 ? void 0 : (_state_keyMap_get_props_children = _state_keyMap_get_props.children) === null || _state_keyMap_get_props_children === void 0 ? void 0 : _state_keyMap_get_props_children.length) > state.userColumnCount;
        showExpandCollapseButton = isFirstRowHeaderCell && isRowExpandable;
        // Offset based on level, and add additional offset if there is no expand/collapse button on a row
        levelOffset = (cell.level - 2) * $bd013581c0a4b065$var$LEVEL_OFFSET_WIDTH[scale] + (!showExpandCollapseButton ? $bd013581c0a4b065$var$LEVEL_OFFSET_WIDTH[scale] * 2 : 0);
    }
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$FocusRing), {
        focusRingClass: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        ...otherGridCellProps,
        "aria-labelledby": id,
        ref: ref,
        style: isExpandableTable && isFirstRowHeaderCell ? {
            paddingInlineStart: levelOffset
        } : {},
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-cell", {
            "spectrum-Table-cell--divider": columnProps.showDivider && cell.column.nextKey !== null,
            "spectrum-Table-cell--hideHeader": columnProps.hideHeader,
            "spectrum-Table-cell--hasExpandCollapseButton": showExpandCollapseButton,
            "is-disabled": isDisabled
        }, (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-cell", {
            "react-spectrum-Table-cell--alignStart": columnProps.align === "start",
            "react-spectrum-Table-cell--alignCenter": columnProps.align === "center",
            "react-spectrum-Table-cell--alignEnd": columnProps.align === "end"
        }))
    }, showExpandCollapseButton && /*#__PURE__*/ (0, $gyhAV$react).createElement($bd013581c0a4b065$var$ExpandableRowChevron, {
        cell: cell
    }), /*#__PURE__*/ (0, $gyhAV$react).createElement("span", {
        id: id,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-cellContents")
    }, cell.rendered)));
}
function $bd013581c0a4b065$var$ExpandableRowChevron({ cell: cell }) {
    // TODO: move some/all of the chevron button setup into a separate hook?
    let { direction: direction } = (0, $gyhAV$useLocale)();
    let { state: state } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let expandButtonRef = (0, $gyhAV$useRef)();
    let stringFormatter = (0, $gyhAV$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($cc43b2d4f6011c04$exports))), "@react-spectrum/table");
    let isExpanded;
    if ("expandedKeys" in state) isExpanded = state.expandedKeys === "all" || state.expandedKeys.has(cell.parentKey);
    // Will need to keep the chevron as a button for iOS VO at all times since VO doesn't focus the cell. Also keep as button if cellAction is defined by the user in the future
    let { buttonProps: buttonProps } = (0, $gyhAV$useButton)({
        // Desktop and mobile both toggle expansion of a native expandable row on mouse/touch up
        onPress: ()=>{
            state.toggleKey(cell.parentKey);
            if (!(0, $gyhAV$isFocusVisible)()) {
                state.selectionManager.setFocused(true);
                state.selectionManager.setFocusedKey(cell.parentKey);
            }
        },
        elementType: "span",
        "aria-label": isExpanded ? stringFormatter.format("collapse") : stringFormatter.format("expand")
    }, expandButtonRef);
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("span", {
        ...buttonProps,
        ref: expandButtonRef,
        // Override tabindex so that grid keyboard nav skips over it. Needs -1 so android talkback can actually "focus" it
        tabIndex: (0, $gyhAV$isAndroid)() ? -1 : undefined,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($bd5d41f647a296c3$exports))), "spectrum-Table-expandButton", {
            "is-open": isExpanded
        })
    }, direction === "ltr" ? /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$spectrumiconsuiChevronRightMedium), null) : /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $gyhAV$spectrumiconsuiChevronLeftMedium), null));
}
function $bd013581c0a4b065$var$CenteredWrapper({ children: children }) {
    let { state: state } = $bd013581c0a4b065$export$3cb274deb6c2d854();
    let rowProps;
    if ("expandedKeys" in state) {
        let topLevelRowCount = [
            ...state.keyMap.get(state.collection.body.key).childNodes
        ].length;
        rowProps = {
            "aria-level": 1,
            "aria-posinset": topLevelRowCount + 1,
            "aria-setsize": topLevelRowCount + 1
        };
    } else rowProps = {
        "aria-rowindex": state.collection.headerRows.length + state.collection.size + 1
    };
    return /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "row",
        ...rowProps,
        className: (0, $gyhAV$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($325bc3d43b25708b$exports))), "react-spectrum-Table-centeredWrapper")
    }, /*#__PURE__*/ (0, $gyhAV$react).createElement("div", {
        role: "rowheader",
        "aria-colspan": state.collection.columns.length
    }, children));
}
const $bd013581c0a4b065$export$517e02184d273d69 = /*#__PURE__*/ (0, $gyhAV$react).forwardRef($bd013581c0a4b065$var$TableViewBase);



function $d14697b14e638114$var$TableView(props, ref) {
    let { selectionStyle: selectionStyle, dragAndDropHooks: dragAndDropHooks } = props;
    let [showSelectionCheckboxes, setShowSelectionCheckboxes] = (0, $gyhAV$useState)(selectionStyle !== "highlight");
    let isTableDraggable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDraggableCollectionState);
    let state = (0, $gyhAV$useTableState)({
        ...props,
        showSelectionCheckboxes: showSelectionCheckboxes,
        showDragButtons: isTableDraggable,
        selectionBehavior: props.selectionStyle === "highlight" ? "replace" : "toggle"
    });
    // If the selection behavior changes in state, we need to update showSelectionCheckboxes here due to the circular dependency...
    let shouldShowCheckboxes = state.selectionManager.selectionBehavior !== "replace";
    if (shouldShowCheckboxes !== showSelectionCheckboxes) setShowSelectionCheckboxes(shouldShowCheckboxes);
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $bd013581c0a4b065$export$517e02184d273d69), {
        ...props,
        state: state,
        ref: ref
    });
}
const $d14697b14e638114$export$b3c27e869d856b7 = /*#__PURE__*/ (0, $gyhAV$react).forwardRef($d14697b14e638114$var$TableView);


/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $be90f60881eac8aa$var$TreeGridTableView(props, ref) {
    let { selectionStyle: selectionStyle, dragAndDropHooks: dragAndDropHooks } = props;
    let [showSelectionCheckboxes, setShowSelectionCheckboxes] = (0, $gyhAV$useState)(selectionStyle !== "highlight");
    let isTableDraggable = !!(dragAndDropHooks === null || dragAndDropHooks === void 0 ? void 0 : dragAndDropHooks.useDraggableCollectionState);
    let state = (0, $gyhAV$UNSTABLE_useTreeGridState)({
        ...props,
        showSelectionCheckboxes: showSelectionCheckboxes,
        showDragButtons: isTableDraggable,
        selectionBehavior: props.selectionStyle === "highlight" ? "replace" : "toggle"
    });
    // If the selection behavior changes in state, we need to update showSelectionCheckboxes here due to the circular dependency...
    let shouldShowCheckboxes = state.selectionManager.selectionBehavior !== "replace";
    if (shouldShowCheckboxes !== showSelectionCheckboxes) setShowSelectionCheckboxes(shouldShowCheckboxes);
    return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $bd013581c0a4b065$export$517e02184d273d69), {
        ...props,
        state: state,
        ref: ref
    });
}
const $be90f60881eac8aa$export$5669566ac2c90964 = /*#__PURE__*/ (0, $gyhAV$react).forwardRef($be90f60881eac8aa$var$TreeGridTableView);


function $9f67189e3f884740$var$TableViewWrapper(props, ref) {
    let { UNSTABLE_allowsExpandableRows: UNSTABLE_allowsExpandableRows, ...otherProps } = props;
    if ((0, $gyhAV$tableNestedRows)() && UNSTABLE_allowsExpandableRows) return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $be90f60881eac8aa$export$5669566ac2c90964), {
        ...otherProps,
        ref: ref
    });
    else return /*#__PURE__*/ (0, $gyhAV$react).createElement((0, $d14697b14e638114$export$b3c27e869d856b7), {
        ...otherProps,
        ref: ref
    });
}
/**
 * Tables are containers for displaying information. They allow users to quickly scan, sort, compare, and take action on large amounts of data.
 */ const $9f67189e3f884740$export$b3c27e869d856b7 = /*#__PURE__*/ (0, $gyhAV$react).forwardRef($9f67189e3f884740$var$TableViewWrapper);



// Override TS for Column to support spectrum specific props.
const $e668f3d0f378a712$export$816b5d811295e6bc = (0, $gyhAV$Column);


export {$e668f3d0f378a712$export$816b5d811295e6bc as Column, $9f67189e3f884740$export$b3c27e869d856b7 as TableView, $e668f3d0f378a712$re_export$TableHeader as TableHeader, $e668f3d0f378a712$re_export$TableBody as TableBody, $e668f3d0f378a712$re_export$Section as Section, $e668f3d0f378a712$re_export$Row as Row, $e668f3d0f378a712$re_export$Cell as Cell};
