// @react-spectrum/statuslight/dist/main.js
"use client";

require("./main.css");
var $cYKEG$reactspectrumutils = require("@react-spectrum/utils");
var $cYKEG$reactariautils = require("@react-aria/utils");
var $cYKEG$react = require("react");
var $cYKEG$reactspectrumprovider = require("@react-spectrum/provider");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "StatusLight", () => $ba01d1490c4da3ea$export$5f84c37a31c6e41c);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $7e20c9d2bb376434$exports = {};

$parcel$export($7e20c9d2bb376434$exports, "focus-ring", () => $7e20c9d2bb376434$export$f39a09f249340e2a, (v) => $7e20c9d2bb376434$export$f39a09f249340e2a = v);
$parcel$export($7e20c9d2bb376434$exports, "i18nFontFamily", () => $7e20c9d2bb376434$export$8c4ee2c50c22c514, (v) => $7e20c9d2bb376434$export$8c4ee2c50c22c514 = v);
$parcel$export($7e20c9d2bb376434$exports, "is-disabled", () => $7e20c9d2bb376434$export$d35bc1e505d1ebbf, (v) => $7e20c9d2bb376434$export$d35bc1e505d1ebbf = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-FocusRing-ring", () => $7e20c9d2bb376434$export$4109102f950813a6, (v) => $7e20c9d2bb376434$export$4109102f950813a6 = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-FocusRing", () => $7e20c9d2bb376434$export$24c7f46a6e3605dd, (v) => $7e20c9d2bb376434$export$24c7f46a6e3605dd = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-FocusRing--quiet", () => $7e20c9d2bb376434$export$2927016961429360, (v) => $7e20c9d2bb376434$export$2927016961429360 = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight", () => $7e20c9d2bb376434$export$11d3697e7e5c1bef, (v) => $7e20c9d2bb376434$export$11d3697e7e5c1bef = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--active", () => $7e20c9d2bb376434$export$b40c603f4b579265, (v) => $7e20c9d2bb376434$export$b40c603f4b579265 = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--celery", () => $7e20c9d2bb376434$export$e9a22f483095fe3e, (v) => $7e20c9d2bb376434$export$e9a22f483095fe3e = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--chartreuse", () => $7e20c9d2bb376434$export$f05bd1f10c0cc15f, (v) => $7e20c9d2bb376434$export$f05bd1f10c0cc15f = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--fuchsia", () => $7e20c9d2bb376434$export$cbaba642d0afd312, (v) => $7e20c9d2bb376434$export$cbaba642d0afd312 = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--indigo", () => $7e20c9d2bb376434$export$8b77e0c07c11635a, (v) => $7e20c9d2bb376434$export$8b77e0c07c11635a = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--info", () => $7e20c9d2bb376434$export$945a5d9a4e8494e6, (v) => $7e20c9d2bb376434$export$945a5d9a4e8494e6 = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--magenta", () => $7e20c9d2bb376434$export$43990cc5ba9a0059, (v) => $7e20c9d2bb376434$export$43990cc5ba9a0059 = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--negative", () => $7e20c9d2bb376434$export$1974c2646528dc4f, (v) => $7e20c9d2bb376434$export$1974c2646528dc4f = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--neutral", () => $7e20c9d2bb376434$export$bbf75dd99797a93e, (v) => $7e20c9d2bb376434$export$bbf75dd99797a93e = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--notice", () => $7e20c9d2bb376434$export$30a5e512ed4618b, (v) => $7e20c9d2bb376434$export$30a5e512ed4618b = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--positive", () => $7e20c9d2bb376434$export$141dae5af9534f54, (v) => $7e20c9d2bb376434$export$141dae5af9534f54 = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--purple", () => $7e20c9d2bb376434$export$975baced4f8182b4, (v) => $7e20c9d2bb376434$export$975baced4f8182b4 = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--seafoam", () => $7e20c9d2bb376434$export$d14bcd2d3ae8238c, (v) => $7e20c9d2bb376434$export$d14bcd2d3ae8238c = v);
$parcel$export($7e20c9d2bb376434$exports, "spectrum-StatusLight--yellow", () => $7e20c9d2bb376434$export$b22c4af54e28c6c, (v) => $7e20c9d2bb376434$export$b22c4af54e28c6c = v);
var $7e20c9d2bb376434$export$f39a09f249340e2a;
var $7e20c9d2bb376434$export$8c4ee2c50c22c514;
var $7e20c9d2bb376434$export$d35bc1e505d1ebbf;
var $7e20c9d2bb376434$export$4109102f950813a6;
var $7e20c9d2bb376434$export$24c7f46a6e3605dd;
var $7e20c9d2bb376434$export$2927016961429360;
var $7e20c9d2bb376434$export$11d3697e7e5c1bef;
var $7e20c9d2bb376434$export$b40c603f4b579265;
var $7e20c9d2bb376434$export$e9a22f483095fe3e;
var $7e20c9d2bb376434$export$f05bd1f10c0cc15f;
var $7e20c9d2bb376434$export$cbaba642d0afd312;
var $7e20c9d2bb376434$export$8b77e0c07c11635a;
var $7e20c9d2bb376434$export$945a5d9a4e8494e6;
var $7e20c9d2bb376434$export$43990cc5ba9a0059;
var $7e20c9d2bb376434$export$1974c2646528dc4f;
var $7e20c9d2bb376434$export$bbf75dd99797a93e;
var $7e20c9d2bb376434$export$30a5e512ed4618b;
var $7e20c9d2bb376434$export$141dae5af9534f54;
var $7e20c9d2bb376434$export$975baced4f8182b4;
var $7e20c9d2bb376434$export$d14bcd2d3ae8238c;
var $7e20c9d2bb376434$export$b22c4af54e28c6c;
$7e20c9d2bb376434$export$f39a09f249340e2a = `j59EQa_focus-ring`;
$7e20c9d2bb376434$export$8c4ee2c50c22c514 = `j59EQa_i18nFontFamily`;
$7e20c9d2bb376434$export$d35bc1e505d1ebbf = `j59EQa_is-disabled`;
$7e20c9d2bb376434$export$4109102f950813a6 = `j59EQa_spectrum-FocusRing-ring`;
$7e20c9d2bb376434$export$24c7f46a6e3605dd = `j59EQa_spectrum-FocusRing ${$7e20c9d2bb376434$export$4109102f950813a6}`;
$7e20c9d2bb376434$export$2927016961429360 = `j59EQa_spectrum-FocusRing--quiet`;
$7e20c9d2bb376434$export$11d3697e7e5c1bef = `j59EQa_spectrum-StatusLight`;
$7e20c9d2bb376434$export$b40c603f4b579265 = `j59EQa_spectrum-StatusLight--active`;
$7e20c9d2bb376434$export$e9a22f483095fe3e = `j59EQa_spectrum-StatusLight--celery`;
$7e20c9d2bb376434$export$f05bd1f10c0cc15f = `j59EQa_spectrum-StatusLight--chartreuse`;
$7e20c9d2bb376434$export$cbaba642d0afd312 = `j59EQa_spectrum-StatusLight--fuchsia`;
$7e20c9d2bb376434$export$8b77e0c07c11635a = `j59EQa_spectrum-StatusLight--indigo`;
$7e20c9d2bb376434$export$945a5d9a4e8494e6 = `j59EQa_spectrum-StatusLight--info`;
$7e20c9d2bb376434$export$43990cc5ba9a0059 = `j59EQa_spectrum-StatusLight--magenta`;
$7e20c9d2bb376434$export$1974c2646528dc4f = `j59EQa_spectrum-StatusLight--negative`;
$7e20c9d2bb376434$export$bbf75dd99797a93e = `j59EQa_spectrum-StatusLight--neutral`;
$7e20c9d2bb376434$export$30a5e512ed4618b = `j59EQa_spectrum-StatusLight--notice`;
$7e20c9d2bb376434$export$141dae5af9534f54 = `j59EQa_spectrum-StatusLight--positive`;
$7e20c9d2bb376434$export$975baced4f8182b4 = `j59EQa_spectrum-StatusLight--purple`;
$7e20c9d2bb376434$export$d14bcd2d3ae8238c = `j59EQa_spectrum-StatusLight--seafoam`;
$7e20c9d2bb376434$export$b22c4af54e28c6c = `j59EQa_spectrum-StatusLight--yellow`;



function $ba01d1490c4da3ea$var$StatusLight(props, ref) {
    let { variant: variant, children: children, isDisabled: isDisabled, role: role, ...otherProps } = (0, $cYKEG$reactspectrumprovider.useProviderProps)(props);
    let domRef = (0, $cYKEG$reactspectrumutils.useDOMRef)(ref);
    let { styleProps: styleProps } = (0, $cYKEG$reactspectrumutils.useStyleProps)(otherProps);
    if (!children && !props["aria-label"]) console.warn("If no children are provided, an aria-label must be specified");
    if (!role && (props["aria-label"] || props["aria-labelledby"])) console.warn("A labelled StatusLight must have a role.");
    return /*#__PURE__*/ (0, ($parcel$interopDefault($cYKEG$react))).createElement("div", {
        ...(0, $cYKEG$reactariautils.filterDOMProps)(otherProps, {
            labelable: !!role
        }),
        ...styleProps,
        role: role,
        className: (0, $cYKEG$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7e20c9d2bb376434$exports))), "spectrum-StatusLight", `spectrum-StatusLight--${variant}`, {
            "is-disabled": isDisabled
        }, styleProps.className),
        ref: domRef
    }, children);
}
/**
 * Status lights are used to color code categories and labels commonly found in data visualization.
 * When status lights have a semantic meaning, they should use semantic variant colors.
 */ let $ba01d1490c4da3ea$export$5f84c37a31c6e41c = /*#__PURE__*/ (0, $cYKEG$react.forwardRef)($ba01d1490c4da3ea$var$StatusLight);
