// @react-spectrum/progress/dist/main.js
"use client";

require("./main.css");
var $cIANI$reactspectrumutils = require("@react-spectrum/utils");
var $cIANI$react = require("react");
var $cIANI$reactariaprogress = require("@react-aria/progress");
var $cIANI$reactariautils = require("@react-aria/utils");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "ProgressBar", () => $08d0cdc7f7a792a7$export$c17561cb55d4db30);
$parcel$export(module.exports, "ProgressCircle", () => $365887133501f5a1$export$c79b9d6b4cc92af7);
$parcel$export(module.exports, "ProgressBarBase", () => $3d86ae6a9356cc8a$export$7c6ed87244065f3a);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $9f78a6c009f32d0a$exports = {};

$parcel$export($9f78a6c009f32d0a$exports, "focus-ring", () => $9f78a6c009f32d0a$export$f39a09f249340e2a, (v) => $9f78a6c009f32d0a$export$f39a09f249340e2a = v);
$parcel$export($9f78a6c009f32d0a$exports, "i18nFontFamily", () => $9f78a6c009f32d0a$export$8c4ee2c50c22c514, (v) => $9f78a6c009f32d0a$export$8c4ee2c50c22c514 = v);
$parcel$export($9f78a6c009f32d0a$exports, "indeterminate-loop-ltr", () => $9f78a6c009f32d0a$export$a889038a25113dd4, (v) => $9f78a6c009f32d0a$export$a889038a25113dd4 = v);
$parcel$export($9f78a6c009f32d0a$exports, "indeterminate-loop-rtl", () => $9f78a6c009f32d0a$export$aeff243cf3356f00, (v) => $9f78a6c009f32d0a$export$aeff243cf3356f00 = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-critical", () => $9f78a6c009f32d0a$export$a1f608a9810da50f, (v) => $9f78a6c009f32d0a$export$a1f608a9810da50f = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-positive", () => $9f78a6c009f32d0a$export$aea9fd919467f03a, (v) => $9f78a6c009f32d0a$export$aea9fd919467f03a = v);
$parcel$export($9f78a6c009f32d0a$exports, "is-warning", () => $9f78a6c009f32d0a$export$d4ebda6218901908, (v) => $9f78a6c009f32d0a$export$d4ebda6218901908 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader", () => $9f78a6c009f32d0a$export$162ba336fb0c239f, (v) => $9f78a6c009f32d0a$export$162ba336fb0c239f = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--indeterminate", () => $9f78a6c009f32d0a$export$1249dfbb98223794, (v) => $9f78a6c009f32d0a$export$1249dfbb98223794 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--overBackground", () => $9f78a6c009f32d0a$export$55612d0bd9b5d41, (v) => $9f78a6c009f32d0a$export$55612d0bd9b5d41 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--sideLabel", () => $9f78a6c009f32d0a$export$99af8802cc2a10dc, (v) => $9f78a6c009f32d0a$export$99af8802cc2a10dc = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--small", () => $9f78a6c009f32d0a$export$b945305960ad74a7, (v) => $9f78a6c009f32d0a$export$b945305960ad74a7 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--staticBlack", () => $9f78a6c009f32d0a$export$f45257a01e7aabeb, (v) => $9f78a6c009f32d0a$export$f45257a01e7aabeb = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader--staticWhite", () => $9f78a6c009f32d0a$export$734e9273ae1f542e, (v) => $9f78a6c009f32d0a$export$734e9273ae1f542e = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-fill", () => $9f78a6c009f32d0a$export$84c9ee7bc724b71b, (v) => $9f78a6c009f32d0a$export$84c9ee7bc724b71b = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-label", () => $9f78a6c009f32d0a$export$5379ba198163be86, (v) => $9f78a6c009f32d0a$export$5379ba198163be86 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-percentage", () => $9f78a6c009f32d0a$export$ecc9c825de9d1a18, (v) => $9f78a6c009f32d0a$export$ecc9c825de9d1a18 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-BarLoader-track", () => $9f78a6c009f32d0a$export$3e2e13fb57b5c74b, (v) => $9f78a6c009f32d0a$export$3e2e13fb57b5c74b = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-FocusRing-ring", () => $9f78a6c009f32d0a$export$4109102f950813a6, (v) => $9f78a6c009f32d0a$export$4109102f950813a6 = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-FocusRing", () => $9f78a6c009f32d0a$export$24c7f46a6e3605dd, (v) => $9f78a6c009f32d0a$export$24c7f46a6e3605dd = v);
$parcel$export($9f78a6c009f32d0a$exports, "spectrum-FocusRing--quiet", () => $9f78a6c009f32d0a$export$2927016961429360, (v) => $9f78a6c009f32d0a$export$2927016961429360 = v);
var $9f78a6c009f32d0a$export$f39a09f249340e2a;
var $9f78a6c009f32d0a$export$8c4ee2c50c22c514;
var $9f78a6c009f32d0a$export$a889038a25113dd4;
var $9f78a6c009f32d0a$export$aeff243cf3356f00;
var $9f78a6c009f32d0a$export$a1f608a9810da50f;
var $9f78a6c009f32d0a$export$aea9fd919467f03a;
var $9f78a6c009f32d0a$export$d4ebda6218901908;
var $9f78a6c009f32d0a$export$162ba336fb0c239f;
var $9f78a6c009f32d0a$export$1249dfbb98223794;
var $9f78a6c009f32d0a$export$55612d0bd9b5d41;
var $9f78a6c009f32d0a$export$99af8802cc2a10dc;
var $9f78a6c009f32d0a$export$b945305960ad74a7;
var $9f78a6c009f32d0a$export$f45257a01e7aabeb;
var $9f78a6c009f32d0a$export$734e9273ae1f542e;
var $9f78a6c009f32d0a$export$84c9ee7bc724b71b;
var $9f78a6c009f32d0a$export$5379ba198163be86;
var $9f78a6c009f32d0a$export$ecc9c825de9d1a18;
var $9f78a6c009f32d0a$export$3e2e13fb57b5c74b;
var $9f78a6c009f32d0a$export$4109102f950813a6;
var $9f78a6c009f32d0a$export$24c7f46a6e3605dd;
var $9f78a6c009f32d0a$export$2927016961429360;
$9f78a6c009f32d0a$export$f39a09f249340e2a = `uFrYBa_focus-ring`;
$9f78a6c009f32d0a$export$8c4ee2c50c22c514 = `uFrYBa_i18nFontFamily`;
$9f78a6c009f32d0a$export$a889038a25113dd4 = `uFrYBa_indeterminate-loop-ltr`;
$9f78a6c009f32d0a$export$a889038a25113dd4;
$9f78a6c009f32d0a$export$aeff243cf3356f00 = `uFrYBa_indeterminate-loop-rtl`;
$9f78a6c009f32d0a$export$aeff243cf3356f00;
$9f78a6c009f32d0a$export$a1f608a9810da50f = `uFrYBa_is-critical`;
$9f78a6c009f32d0a$export$aea9fd919467f03a = `uFrYBa_is-positive`;
$9f78a6c009f32d0a$export$d4ebda6218901908 = `uFrYBa_is-warning`;
$9f78a6c009f32d0a$export$162ba336fb0c239f = `uFrYBa_spectrum-BarLoader`;
$9f78a6c009f32d0a$export$1249dfbb98223794 = `uFrYBa_spectrum-BarLoader--indeterminate`;
$9f78a6c009f32d0a$export$55612d0bd9b5d41 = `uFrYBa_spectrum-BarLoader--overBackground`;
$9f78a6c009f32d0a$export$99af8802cc2a10dc = `uFrYBa_spectrum-BarLoader--sideLabel`;
$9f78a6c009f32d0a$export$b945305960ad74a7 = `uFrYBa_spectrum-BarLoader--small`;
$9f78a6c009f32d0a$export$f45257a01e7aabeb = `uFrYBa_spectrum-BarLoader--staticBlack`;
$9f78a6c009f32d0a$export$734e9273ae1f542e = `uFrYBa_spectrum-BarLoader--staticWhite`;
$9f78a6c009f32d0a$export$84c9ee7bc724b71b = `uFrYBa_spectrum-BarLoader-fill`;
$9f78a6c009f32d0a$export$5379ba198163be86 = `uFrYBa_spectrum-BarLoader-label`;
$9f78a6c009f32d0a$export$ecc9c825de9d1a18 = `uFrYBa_spectrum-BarLoader-percentage`;
$9f78a6c009f32d0a$export$3e2e13fb57b5c74b = `uFrYBa_spectrum-BarLoader-track`;
$9f78a6c009f32d0a$export$4109102f950813a6 = `uFrYBa_spectrum-FocusRing-ring`;
$9f78a6c009f32d0a$export$24c7f46a6e3605dd = `uFrYBa_spectrum-FocusRing ${$9f78a6c009f32d0a$export$4109102f950813a6}`;
$9f78a6c009f32d0a$export$2927016961429360 = `uFrYBa_spectrum-FocusRing--quiet`;


// Base ProgressBar component shared with Meter.
function $3d86ae6a9356cc8a$var$ProgressBarBase(props, ref) {
    let { value: value = 0, minValue: minValue = 0, maxValue: maxValue = 100, size: size = "L", label: label, barClassName: barClassName, showValueLabel: showValueLabel = !!label, labelPosition: labelPosition = "top", isIndeterminate: isIndeterminate = false, barProps: barProps, labelProps: labelProps, "aria-label": ariaLabel, "aria-labelledby": ariaLabelledby, ...otherProps } = props;
    let domRef = (0, $cIANI$reactspectrumutils.useDOMRef)(ref);
    let { styleProps: styleProps } = (0, $cIANI$reactspectrumutils.useStyleProps)(otherProps);
    value = (0, $cIANI$reactariautils.clamp)(value, minValue, maxValue);
    let barStyle = {};
    if (!isIndeterminate) {
        let percentage = (value - minValue) / (maxValue - minValue);
        barStyle.width = `${Math.round(percentage * 100)}%`;
    }
    // Ideally this should be in useProgressBar, but children
    // are not supported in ProgressCircle which shares that hook...
    if (!label && !ariaLabel && !ariaLabelledby) console.warn("If you do not provide a visible label via children, you must specify an aria-label or aria-labelledby attribute for accessibility");
    // use inline style for fit-content because cssnano is too smart for us and will strip out the -moz prefix in css files
    return /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        ...barProps,
        ref: domRef,
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($9f78a6c009f32d0a$exports))), "spectrum-BarLoader", {
            "spectrum-BarLoader--small": size === "S",
            "spectrum-BarLoader--large": size === "L",
            "spectrum-BarLoader--indeterminate": isIndeterminate,
            "spectrum-BarLoader--sideLabel": labelPosition === "side"
        }, barClassName, styleProps.className),
        style: {
            minWidth: "-moz-fit-content",
            ...styleProps.style
        }
    }, label && /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("span", {
        ...labelProps,
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($9f78a6c009f32d0a$exports))), "spectrum-BarLoader-label")
    }, label), showValueLabel && barProps && /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($9f78a6c009f32d0a$exports))), "spectrum-BarLoader-percentage")
    }, barProps["aria-valuetext"]), /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($9f78a6c009f32d0a$exports))), "spectrum-BarLoader-track")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($9f78a6c009f32d0a$exports))), "spectrum-BarLoader-fill"),
        style: barStyle
    })));
}
let $3d86ae6a9356cc8a$export$7c6ed87244065f3a = /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).forwardRef($3d86ae6a9356cc8a$var$ProgressBarBase);





function $08d0cdc7f7a792a7$var$ProgressBar(props, ref) {
    let { staticColor: staticColor, variant: variant, ...otherProps } = props;
    const { progressBarProps: progressBarProps, labelProps: labelProps } = (0, $cIANI$reactariaprogress.useProgressBar)(props);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement((0, $3d86ae6a9356cc8a$export$7c6ed87244065f3a), {
        ...otherProps,
        ref: ref,
        barProps: progressBarProps,
        labelProps: labelProps,
        barClassName: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($9f78a6c009f32d0a$exports))), {
            "spectrum-BarLoader--overBackground": variant === "overBackground",
            "spectrum-BarLoader--staticWhite": staticColor === "white",
            "spectrum-BarLoader--staticBlack": staticColor === "black"
        })
    });
}
/**
 * ProgressBars show the progression of a system operation: downloading, uploading, processing, etc., in a visual way.
 * They can represent either determinate or indeterminate progress.
 */ let $08d0cdc7f7a792a7$export$c17561cb55d4db30 = /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).forwardRef($08d0cdc7f7a792a7$var$ProgressBar);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $f4d8d13edf2d3ca2$exports = {};

$parcel$export($f4d8d13edf2d3ca2$exports, "focus-ring", () => $f4d8d13edf2d3ca2$export$f39a09f249340e2a, (v) => $f4d8d13edf2d3ca2$export$f39a09f249340e2a = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "i18nFontFamily", () => $f4d8d13edf2d3ca2$export$8c4ee2c50c22c514, (v) => $f4d8d13edf2d3ca2$export$8c4ee2c50c22c514 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader", () => $f4d8d13edf2d3ca2$export$9633d311d6d6830b, (v) => $f4d8d13edf2d3ca2$export$9633d311d6d6830b = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader--indeterminate", () => $f4d8d13edf2d3ca2$export$79b7e5abeef5134f, (v) => $f4d8d13edf2d3ca2$export$79b7e5abeef5134f = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader--indeterminate-fill-submask-2", () => $f4d8d13edf2d3ca2$export$f7083dbeadcbb2bd, (v) => $f4d8d13edf2d3ca2$export$f7083dbeadcbb2bd = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader--large", () => $f4d8d13edf2d3ca2$export$c8c34e5e37bf511c, (v) => $f4d8d13edf2d3ca2$export$c8c34e5e37bf511c = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader--overBackground", () => $f4d8d13edf2d3ca2$export$6668708cb539f403, (v) => $f4d8d13edf2d3ca2$export$6668708cb539f403 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader--small", () => $f4d8d13edf2d3ca2$export$79819ab9844dbe9b, (v) => $f4d8d13edf2d3ca2$export$79819ab9844dbe9b = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader--staticBlack", () => $f4d8d13edf2d3ca2$export$7f8072f8867b580e, (v) => $f4d8d13edf2d3ca2$export$7f8072f8867b580e = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader--staticWhite", () => $f4d8d13edf2d3ca2$export$f2e90b77c21d4ec7, (v) => $f4d8d13edf2d3ca2$export$f2e90b77c21d4ec7 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader-fill", () => $f4d8d13edf2d3ca2$export$58fb9ac68cf8e784, (v) => $f4d8d13edf2d3ca2$export$58fb9ac68cf8e784 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader-fillMask1", () => $f4d8d13edf2d3ca2$export$b676cd958401a736, (v) => $f4d8d13edf2d3ca2$export$b676cd958401a736 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader-fillMask2", () => $f4d8d13edf2d3ca2$export$8c428f4f13e424b, (v) => $f4d8d13edf2d3ca2$export$8c428f4f13e424b = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader-fillSubMask1", () => $f4d8d13edf2d3ca2$export$98ba5ef650aeb8ca, (v) => $f4d8d13edf2d3ca2$export$98ba5ef650aeb8ca = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader-fillSubMask2", () => $f4d8d13edf2d3ca2$export$dc70b48c897b0f1a, (v) => $f4d8d13edf2d3ca2$export$dc70b48c897b0f1a = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader-fills", () => $f4d8d13edf2d3ca2$export$3038f4c0c86f539b, (v) => $f4d8d13edf2d3ca2$export$3038f4c0c86f539b = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-CircleLoader-track", () => $f4d8d13edf2d3ca2$export$278efdb85ed802da, (v) => $f4d8d13edf2d3ca2$export$278efdb85ed802da = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-FocusRing-ring", () => $f4d8d13edf2d3ca2$export$4109102f950813a6, (v) => $f4d8d13edf2d3ca2$export$4109102f950813a6 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-FocusRing", () => $f4d8d13edf2d3ca2$export$24c7f46a6e3605dd, (v) => $f4d8d13edf2d3ca2$export$24c7f46a6e3605dd = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-FocusRing--quiet", () => $f4d8d13edf2d3ca2$export$2927016961429360, (v) => $f4d8d13edf2d3ca2$export$2927016961429360 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-fill-mask-1", () => $f4d8d13edf2d3ca2$export$524c81e63e1d64f5, (v) => $f4d8d13edf2d3ca2$export$524c81e63e1d64f5 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-fill-mask-2", () => $f4d8d13edf2d3ca2$export$7219832bea80eb54, (v) => $f4d8d13edf2d3ca2$export$7219832bea80eb54 = v);
$parcel$export($f4d8d13edf2d3ca2$exports, "spectrum-fills-rotate", () => $f4d8d13edf2d3ca2$export$3663cbdb240663ec, (v) => $f4d8d13edf2d3ca2$export$3663cbdb240663ec = v);
var $f4d8d13edf2d3ca2$export$f39a09f249340e2a;
var $f4d8d13edf2d3ca2$export$8c4ee2c50c22c514;
var $f4d8d13edf2d3ca2$export$9633d311d6d6830b;
var $f4d8d13edf2d3ca2$export$79b7e5abeef5134f;
var $f4d8d13edf2d3ca2$export$f7083dbeadcbb2bd;
var $f4d8d13edf2d3ca2$export$c8c34e5e37bf511c;
var $f4d8d13edf2d3ca2$export$6668708cb539f403;
var $f4d8d13edf2d3ca2$export$79819ab9844dbe9b;
var $f4d8d13edf2d3ca2$export$7f8072f8867b580e;
var $f4d8d13edf2d3ca2$export$f2e90b77c21d4ec7;
var $f4d8d13edf2d3ca2$export$58fb9ac68cf8e784;
var $f4d8d13edf2d3ca2$export$b676cd958401a736;
var $f4d8d13edf2d3ca2$export$8c428f4f13e424b;
var $f4d8d13edf2d3ca2$export$98ba5ef650aeb8ca;
var $f4d8d13edf2d3ca2$export$dc70b48c897b0f1a;
var $f4d8d13edf2d3ca2$export$3038f4c0c86f539b;
var $f4d8d13edf2d3ca2$export$278efdb85ed802da;
var $f4d8d13edf2d3ca2$export$4109102f950813a6;
var $f4d8d13edf2d3ca2$export$24c7f46a6e3605dd;
var $f4d8d13edf2d3ca2$export$2927016961429360;
var $f4d8d13edf2d3ca2$export$524c81e63e1d64f5;
var $f4d8d13edf2d3ca2$export$7219832bea80eb54;
var $f4d8d13edf2d3ca2$export$3663cbdb240663ec;
$f4d8d13edf2d3ca2$export$f39a09f249340e2a = `EQYv7q_focus-ring`;
$f4d8d13edf2d3ca2$export$8c4ee2c50c22c514 = `EQYv7q_i18nFontFamily`;
$f4d8d13edf2d3ca2$export$9633d311d6d6830b = `EQYv7q_spectrum-CircleLoader`;
$f4d8d13edf2d3ca2$export$79b7e5abeef5134f = `EQYv7q_spectrum-CircleLoader--indeterminate`;
$f4d8d13edf2d3ca2$export$f7083dbeadcbb2bd = `EQYv7q_spectrum-CircleLoader--indeterminate-fill-submask-2`;
$f4d8d13edf2d3ca2$export$c8c34e5e37bf511c = `EQYv7q_spectrum-CircleLoader--large`;
$f4d8d13edf2d3ca2$export$6668708cb539f403 = `EQYv7q_spectrum-CircleLoader--overBackground`;
$f4d8d13edf2d3ca2$export$79819ab9844dbe9b = `EQYv7q_spectrum-CircleLoader--small`;
$f4d8d13edf2d3ca2$export$7f8072f8867b580e = `EQYv7q_spectrum-CircleLoader--staticBlack`;
$f4d8d13edf2d3ca2$export$f2e90b77c21d4ec7 = `EQYv7q_spectrum-CircleLoader--staticWhite`;
$f4d8d13edf2d3ca2$export$58fb9ac68cf8e784 = `EQYv7q_spectrum-CircleLoader-fill`;
$f4d8d13edf2d3ca2$export$b676cd958401a736 = `EQYv7q_spectrum-CircleLoader-fillMask1`;
$f4d8d13edf2d3ca2$export$8c428f4f13e424b = `EQYv7q_spectrum-CircleLoader-fillMask2`;
$f4d8d13edf2d3ca2$export$98ba5ef650aeb8ca = `EQYv7q_spectrum-CircleLoader-fillSubMask1`;
$f4d8d13edf2d3ca2$export$dc70b48c897b0f1a = `EQYv7q_spectrum-CircleLoader-fillSubMask2`;
$f4d8d13edf2d3ca2$export$3038f4c0c86f539b = `EQYv7q_spectrum-CircleLoader-fills`;
$f4d8d13edf2d3ca2$export$278efdb85ed802da = `EQYv7q_spectrum-CircleLoader-track`;
$f4d8d13edf2d3ca2$export$4109102f950813a6 = `EQYv7q_spectrum-FocusRing-ring`;
$f4d8d13edf2d3ca2$export$24c7f46a6e3605dd = `EQYv7q_spectrum-FocusRing ${$f4d8d13edf2d3ca2$export$4109102f950813a6}`;
$f4d8d13edf2d3ca2$export$2927016961429360 = `EQYv7q_spectrum-FocusRing--quiet`;
$f4d8d13edf2d3ca2$export$524c81e63e1d64f5 = `EQYv7q_spectrum-fill-mask-1`;
$f4d8d13edf2d3ca2$export$524c81e63e1d64f5;
$f4d8d13edf2d3ca2$export$7219832bea80eb54 = `EQYv7q_spectrum-fill-mask-2`;
$f4d8d13edf2d3ca2$export$7219832bea80eb54;
$f4d8d13edf2d3ca2$export$3663cbdb240663ec = `EQYv7q_spectrum-fills-rotate`;
$f4d8d13edf2d3ca2$export$3663cbdb240663ec;



function $365887133501f5a1$var$ProgressCircle(props, ref) {
    let { value: value = 0, minValue: minValue = 0, maxValue: maxValue = 100, size: size = "M", staticColor: staticColor, variant: variant, isIndeterminate: isIndeterminate = false, "aria-label": ariaLabel, "aria-labelledby": ariaLabelledby, ...otherProps } = props;
    let domRef = (0, $cIANI$reactspectrumutils.useDOMRef)(ref);
    let { styleProps: styleProps } = (0, $cIANI$reactspectrumutils.useStyleProps)(otherProps);
    value = (0, $cIANI$reactariautils.clamp)(value, minValue, maxValue);
    let { progressBarProps: progressBarProps } = (0, $cIANI$reactariaprogress.useProgressBar)({
        ...props,
        value: value
    });
    let subMask1Style = {};
    let subMask2Style = {};
    if (!isIndeterminate) {
        let percentage = (value - minValue) / (maxValue - minValue) * 100;
        let angle;
        if (percentage > 0 && percentage <= 50) {
            angle = -180 + percentage / 50 * 180;
            subMask1Style.transform = `rotate(${angle}deg)`;
            subMask2Style.transform = "rotate(-180deg)";
        } else if (percentage > 50) {
            angle = -180 + (percentage - 50) / 50 * 180;
            subMask1Style.transform = "rotate(0deg)";
            subMask2Style.transform = `rotate(${angle}deg)`;
        }
    }
    if (!ariaLabel && !ariaLabelledby) console.warn("ProgressCircle requires an aria-label or aria-labelledby attribute for accessibility");
    return /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        ...styleProps,
        ...progressBarProps,
        ref: domRef,
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader", {
            "spectrum-CircleLoader--indeterminate": isIndeterminate,
            "spectrum-CircleLoader--small": size === "S",
            "spectrum-CircleLoader--large": size === "L",
            "spectrum-CircleLoader--overBackground": variant === "overBackground",
            "spectrum-CircleLoader--staticWhite": staticColor === "white",
            "spectrum-CircleLoader--staticBlack": staticColor === "black"
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader-track")
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader-fills")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader-fillMask1")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader-fillSubMask1"),
        "data-testid": "fillSubMask1",
        style: subMask1Style
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader-fill")
    }))), /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader-fillMask2")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader-fillSubMask2"),
        "data-testid": "fillSubMask2",
        style: subMask2Style
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).createElement("div", {
        className: (0, $cIANI$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($f4d8d13edf2d3ca2$exports))), "spectrum-CircleLoader-fill")
    })))));
}
/**
 * ProgressCircles show the progression of a system operation such as downloading, uploading, or processing, in a visual way.
 * They can represent determinate or indeterminate progress.
 */ let $365887133501f5a1$export$c79b9d6b4cc92af7 = /*#__PURE__*/ (0, ($parcel$interopDefault($cIANI$react))).forwardRef($365887133501f5a1$var$ProgressCircle);
