// @react-spectrum/tooltip/dist/main.js
"use client";

require("./main.css");
var $kAvU8$spectrumiconsuiAlertSmall = require("@spectrum-icons/ui/AlertSmall");
var $kAvU8$reactspectrumutils = require("@react-spectrum/utils");
var $kAvU8$spectrumiconsuiInfoSmall = require("@spectrum-icons/ui/InfoSmall");
var $kAvU8$reactariautils = require("@react-aria/utils");
var $kAvU8$react = require("react");
var $kAvU8$spectrumiconsuiSuccessSmall = require("@spectrum-icons/ui/SuccessSmall");
var $kAvU8$reactariatooltip = require("@react-aria/tooltip");
var $kAvU8$reactariafocus = require("@react-aria/focus");
var $kAvU8$reactspectrumoverlays = require("@react-spectrum/overlays");
var $kAvU8$reactariaoverlays = require("@react-aria/overlays");
var $kAvU8$reactstatelytooltip = require("@react-stately/tooltip");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Tooltip", () => $55bd11986fcffc6f$export$28c660c63b792dea);
$parcel$export(module.exports, "TooltipTrigger", () => $19f06480c1c1a932$export$8c610744efcf8a1d);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $ba73465d6d9b6556$exports = {};

$parcel$export($ba73465d6d9b6556$exports, "focus-ring", () => $ba73465d6d9b6556$export$f39a09f249340e2a, (v) => $ba73465d6d9b6556$export$f39a09f249340e2a = v);
$parcel$export($ba73465d6d9b6556$exports, "i18nFontFamily", () => $ba73465d6d9b6556$export$8c4ee2c50c22c514, (v) => $ba73465d6d9b6556$export$8c4ee2c50c22c514 = v);
$parcel$export($ba73465d6d9b6556$exports, "is-focused", () => $ba73465d6d9b6556$export$e7dc768d35940237, (v) => $ba73465d6d9b6556$export$e7dc768d35940237 = v);
$parcel$export($ba73465d6d9b6556$exports, "is-hovered", () => $ba73465d6d9b6556$export$b8813cd5d7824ce7, (v) => $ba73465d6d9b6556$export$b8813cd5d7824ce7 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-overlay--open", () => $ba73465d6d9b6556$export$a397eb58038aa381, (v) => $ba73465d6d9b6556$export$a397eb58038aa381 = v);
$parcel$export($ba73465d6d9b6556$exports, "is-open", () => $ba73465d6d9b6556$export$a9781837241c946d, (v) => $ba73465d6d9b6556$export$a9781837241c946d = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-overlay--bottom--open", () => $ba73465d6d9b6556$export$a96755f239de61ee, (v) => $ba73465d6d9b6556$export$a96755f239de61ee = v);
$parcel$export($ba73465d6d9b6556$exports, "is-open--bottom", () => $ba73465d6d9b6556$export$1f160596ffe52357, (v) => $ba73465d6d9b6556$export$1f160596ffe52357 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-overlay--left--open", () => $ba73465d6d9b6556$export$a85c0c6b6cf8598c, (v) => $ba73465d6d9b6556$export$a85c0c6b6cf8598c = v);
$parcel$export($ba73465d6d9b6556$exports, "is-open--left", () => $ba73465d6d9b6556$export$d6106cb3acaf56ae, (v) => $ba73465d6d9b6556$export$d6106cb3acaf56ae = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-overlay--right--open", () => $ba73465d6d9b6556$export$85aaa87e65e040bf, (v) => $ba73465d6d9b6556$export$85aaa87e65e040bf = v);
$parcel$export($ba73465d6d9b6556$exports, "is-open--right", () => $ba73465d6d9b6556$export$8935d072812c3169, (v) => $ba73465d6d9b6556$export$8935d072812c3169 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-overlay--top--open", () => $ba73465d6d9b6556$export$b9768db7ffc1bc2, (v) => $ba73465d6d9b6556$export$b9768db7ffc1bc2 = v);
$parcel$export($ba73465d6d9b6556$exports, "is-open--top", () => $ba73465d6d9b6556$export$cd4bf1e07ae86214, (v) => $ba73465d6d9b6556$export$cd4bf1e07ae86214 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-FocusRing-ring", () => $ba73465d6d9b6556$export$4109102f950813a6, (v) => $ba73465d6d9b6556$export$4109102f950813a6 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-FocusRing", () => $ba73465d6d9b6556$export$24c7f46a6e3605dd, (v) => $ba73465d6d9b6556$export$24c7f46a6e3605dd = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-FocusRing--quiet", () => $ba73465d6d9b6556$export$2927016961429360, (v) => $ba73465d6d9b6556$export$2927016961429360 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-overlay", () => $ba73465d6d9b6556$export$39eed38e2296a1d6, (v) => $ba73465d6d9b6556$export$39eed38e2296a1d6 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip", () => $ba73465d6d9b6556$export$f7bf7078d3d03f05, (v) => $ba73465d6d9b6556$export$f7bf7078d3d03f05 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--bottom", () => $ba73465d6d9b6556$export$4f420a9bbb442953, (v) => $ba73465d6d9b6556$export$4f420a9bbb442953 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--error", () => $ba73465d6d9b6556$export$2b291890a5577893, (v) => $ba73465d6d9b6556$export$2b291890a5577893 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--help", () => $ba73465d6d9b6556$export$d33ca445ae528241, (v) => $ba73465d6d9b6556$export$d33ca445ae528241 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--info", () => $ba73465d6d9b6556$export$e995d5780591d0a8, (v) => $ba73465d6d9b6556$export$e995d5780591d0a8 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--left", () => $ba73465d6d9b6556$export$47f060b68b4cab2c, (v) => $ba73465d6d9b6556$export$47f060b68b4cab2c = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--negative", () => $ba73465d6d9b6556$export$92fe45b4260594f6, (v) => $ba73465d6d9b6556$export$92fe45b4260594f6 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--positive", () => $ba73465d6d9b6556$export$11cbf31e1b6e51ac, (v) => $ba73465d6d9b6556$export$11cbf31e1b6e51ac = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--right", () => $ba73465d6d9b6556$export$c094187772823b2e, (v) => $ba73465d6d9b6556$export$c094187772823b2e = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--success", () => $ba73465d6d9b6556$export$4eae622bc9b08cac, (v) => $ba73465d6d9b6556$export$4eae622bc9b08cac = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--top", () => $ba73465d6d9b6556$export$d7516df59e198fc7, (v) => $ba73465d6d9b6556$export$d7516df59e198fc7 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip-label", () => $ba73465d6d9b6556$export$f4096e0dc445b255, (v) => $ba73465d6d9b6556$export$f4096e0dc445b255 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip-tip", () => $ba73465d6d9b6556$export$9deb89a7a91de30, (v) => $ba73465d6d9b6556$export$9deb89a7a91de30 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip-typeIcon", () => $ba73465d6d9b6556$export$ff23d46fee8272e4, (v) => $ba73465d6d9b6556$export$ff23d46fee8272e4 = v);
$parcel$export($ba73465d6d9b6556$exports, "u-tooltip-showOnHover", () => $ba73465d6d9b6556$export$6ea3237f62800d00, (v) => $ba73465d6d9b6556$export$6ea3237f62800d00 = v);
var $ba73465d6d9b6556$export$f39a09f249340e2a;
var $ba73465d6d9b6556$export$8c4ee2c50c22c514;
var $ba73465d6d9b6556$export$e7dc768d35940237;
var $ba73465d6d9b6556$export$b8813cd5d7824ce7;
var $ba73465d6d9b6556$export$a397eb58038aa381;
var $ba73465d6d9b6556$export$a9781837241c946d;
var $ba73465d6d9b6556$export$a96755f239de61ee;
var $ba73465d6d9b6556$export$1f160596ffe52357;
var $ba73465d6d9b6556$export$a85c0c6b6cf8598c;
var $ba73465d6d9b6556$export$d6106cb3acaf56ae;
var $ba73465d6d9b6556$export$85aaa87e65e040bf;
var $ba73465d6d9b6556$export$8935d072812c3169;
var $ba73465d6d9b6556$export$b9768db7ffc1bc2;
var $ba73465d6d9b6556$export$cd4bf1e07ae86214;
var $ba73465d6d9b6556$export$4109102f950813a6;
var $ba73465d6d9b6556$export$24c7f46a6e3605dd;
var $ba73465d6d9b6556$export$2927016961429360;
var $ba73465d6d9b6556$export$39eed38e2296a1d6;
var $ba73465d6d9b6556$export$f7bf7078d3d03f05;
var $ba73465d6d9b6556$export$4f420a9bbb442953;
var $ba73465d6d9b6556$export$2b291890a5577893;
var $ba73465d6d9b6556$export$d33ca445ae528241;
var $ba73465d6d9b6556$export$e995d5780591d0a8;
var $ba73465d6d9b6556$export$47f060b68b4cab2c;
var $ba73465d6d9b6556$export$92fe45b4260594f6;
var $ba73465d6d9b6556$export$11cbf31e1b6e51ac;
var $ba73465d6d9b6556$export$c094187772823b2e;
var $ba73465d6d9b6556$export$4eae622bc9b08cac;
var $ba73465d6d9b6556$export$d7516df59e198fc7;
var $ba73465d6d9b6556$export$f4096e0dc445b255;
var $ba73465d6d9b6556$export$9deb89a7a91de30;
var $ba73465d6d9b6556$export$ff23d46fee8272e4;
var $ba73465d6d9b6556$export$6ea3237f62800d00;
$ba73465d6d9b6556$export$f39a09f249340e2a = `KF3Kcq_focus-ring`;
$ba73465d6d9b6556$export$8c4ee2c50c22c514 = `KF3Kcq_i18nFontFamily`;
$ba73465d6d9b6556$export$e7dc768d35940237 = `KF3Kcq_is-focused`;
$ba73465d6d9b6556$export$b8813cd5d7824ce7 = `KF3Kcq_is-hovered`;
$ba73465d6d9b6556$export$a397eb58038aa381 = `KF3Kcq_spectrum-overlay--open`;
$ba73465d6d9b6556$export$a9781837241c946d = `KF3Kcq_is-open ${$ba73465d6d9b6556$export$a397eb58038aa381}`;
$ba73465d6d9b6556$export$a96755f239de61ee = `KF3Kcq_spectrum-overlay--bottom--open`;
$ba73465d6d9b6556$export$1f160596ffe52357 = `KF3Kcq_is-open--bottom ${$ba73465d6d9b6556$export$a96755f239de61ee}`;
$ba73465d6d9b6556$export$a85c0c6b6cf8598c = `KF3Kcq_spectrum-overlay--left--open`;
$ba73465d6d9b6556$export$d6106cb3acaf56ae = `KF3Kcq_is-open--left ${$ba73465d6d9b6556$export$a85c0c6b6cf8598c}`;
$ba73465d6d9b6556$export$85aaa87e65e040bf = `KF3Kcq_spectrum-overlay--right--open`;
$ba73465d6d9b6556$export$8935d072812c3169 = `KF3Kcq_is-open--right ${$ba73465d6d9b6556$export$85aaa87e65e040bf}`;
$ba73465d6d9b6556$export$b9768db7ffc1bc2 = `KF3Kcq_spectrum-overlay--top--open`;
$ba73465d6d9b6556$export$cd4bf1e07ae86214 = `KF3Kcq_is-open--top ${$ba73465d6d9b6556$export$b9768db7ffc1bc2}`;
$ba73465d6d9b6556$export$4109102f950813a6 = `KF3Kcq_spectrum-FocusRing-ring`;
$ba73465d6d9b6556$export$24c7f46a6e3605dd = `KF3Kcq_spectrum-FocusRing ${$ba73465d6d9b6556$export$4109102f950813a6}`;
$ba73465d6d9b6556$export$2927016961429360 = `KF3Kcq_spectrum-FocusRing--quiet`;
$ba73465d6d9b6556$export$39eed38e2296a1d6 = `KF3Kcq_spectrum-overlay`;
$ba73465d6d9b6556$export$f7bf7078d3d03f05 = `KF3Kcq_spectrum-Tooltip ${$ba73465d6d9b6556$export$39eed38e2296a1d6}`;
$ba73465d6d9b6556$export$4f420a9bbb442953 = `KF3Kcq_spectrum-Tooltip--bottom`;
$ba73465d6d9b6556$export$2b291890a5577893 = `KF3Kcq_spectrum-Tooltip--error`;
$ba73465d6d9b6556$export$d33ca445ae528241 = `KF3Kcq_spectrum-Tooltip--help`;
$ba73465d6d9b6556$export$e995d5780591d0a8 = `KF3Kcq_spectrum-Tooltip--info`;
$ba73465d6d9b6556$export$47f060b68b4cab2c = `KF3Kcq_spectrum-Tooltip--left`;
$ba73465d6d9b6556$export$92fe45b4260594f6 = `KF3Kcq_spectrum-Tooltip--negative`;
$ba73465d6d9b6556$export$11cbf31e1b6e51ac = `KF3Kcq_spectrum-Tooltip--positive`;
$ba73465d6d9b6556$export$c094187772823b2e = `KF3Kcq_spectrum-Tooltip--right`;
$ba73465d6d9b6556$export$4eae622bc9b08cac = `KF3Kcq_spectrum-Tooltip--success`;
$ba73465d6d9b6556$export$d7516df59e198fc7 = `KF3Kcq_spectrum-Tooltip--top`;
$ba73465d6d9b6556$export$f4096e0dc445b255 = `KF3Kcq_spectrum-Tooltip-label`;
$ba73465d6d9b6556$export$9deb89a7a91de30 = `KF3Kcq_spectrum-Tooltip-tip`;
$ba73465d6d9b6556$export$ff23d46fee8272e4 = `KF3Kcq_spectrum-Tooltip-typeIcon`;
$ba73465d6d9b6556$export$6ea3237f62800d00 = `KF3Kcq_u-tooltip-showOnHover`;



/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $ed2e56c8503433b4$export$39ae08fa83328b12 = (0, ($parcel$interopDefault($kAvU8$react))).createContext({});



let $55bd11986fcffc6f$var$iconMap = {
    info: (0, ($parcel$interopDefault($kAvU8$spectrumiconsuiInfoSmall))),
    positive: (0, ($parcel$interopDefault($kAvU8$spectrumiconsuiSuccessSmall))),
    negative: (0, ($parcel$interopDefault($kAvU8$spectrumiconsuiAlertSmall)))
};
function $55bd11986fcffc6f$var$Tooltip(props, ref) {
    let { ref: overlayRef, arrowProps: arrowProps, state: state, arrowRef: arrowRef, ...tooltipProviderProps } = (0, $kAvU8$react.useContext)((0, $ed2e56c8503433b4$export$39ae08fa83328b12));
    let defaultRef = (0, $kAvU8$react.useRef)();
    overlayRef = overlayRef || defaultRef;
    props = (0, $kAvU8$reactariautils.mergeProps)(props, tooltipProviderProps);
    let { variant: variant = "neutral", placement: placement = "top", isOpen: isOpen, showIcon: showIcon, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $kAvU8$reactspectrumutils.useStyleProps)(otherProps);
    let { tooltipProps: tooltipProps } = (0, $kAvU8$reactariatooltip.useTooltip)(props, state);
    // Sync ref with overlayRef from context.
    (0, $kAvU8$react.useImperativeHandle)(ref, ()=>(0, $kAvU8$reactspectrumutils.createDOMRef)(overlayRef));
    let Icon = $55bd11986fcffc6f$var$iconMap[variant];
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement("div", {
        ...styleProps,
        ...tooltipProps,
        className: (0, $kAvU8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba73465d6d9b6556$exports))), "spectrum-Tooltip", `spectrum-Tooltip--${variant}`, `spectrum-Tooltip--${placement}`, {
            "is-open": isOpen,
            [`is-open--${placement}`]: isOpen
        }, styleProps.className),
        ref: overlayRef
    }, showIcon && variant !== "neutral" && /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement(Icon, {
        UNSAFE_className: (0, $kAvU8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba73465d6d9b6556$exports))), "spectrum-Tooltip-typeIcon"),
        "aria-hidden": true
    }), props.children && /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement("span", {
        className: (0, $kAvU8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba73465d6d9b6556$exports))), "spectrum-Tooltip-label")
    }, props.children), /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement("span", {
        ...arrowProps,
        ref: arrowRef,
        className: (0, $kAvU8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba73465d6d9b6556$exports))), "spectrum-Tooltip-tip")
    }));
}
/**
 * Display container for Tooltip content. Has a directional arrow dependent on its placement.
 */ let $55bd11986fcffc6f$export$28c660c63b792dea = /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).forwardRef($55bd11986fcffc6f$var$Tooltip);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







const $19f06480c1c1a932$var$DEFAULT_OFFSET = -1; // Offset needed to reach 4px/5px (med/large) distance between tooltip and trigger button
const $19f06480c1c1a932$var$DEFAULT_CROSS_OFFSET = 0;
function $19f06480c1c1a932$var$TooltipTrigger(props) {
    let { children: children, crossOffset: crossOffset = $19f06480c1c1a932$var$DEFAULT_CROSS_OFFSET, isDisabled: isDisabled, offset: offset = $19f06480c1c1a932$var$DEFAULT_OFFSET, trigger: triggerAction } = props;
    let [trigger, tooltip] = (0, ($parcel$interopDefault($kAvU8$react))).Children.toArray(children);
    let state = (0, $kAvU8$reactstatelytooltip.useTooltipTriggerState)(props);
    let tooltipTriggerRef = (0, $kAvU8$react.useRef)();
    let overlayRef = (0, $kAvU8$react.useRef)();
    let { triggerProps: triggerProps, tooltipProps: tooltipProps } = (0, $kAvU8$reactariatooltip.useTooltipTrigger)({
        isDisabled: isDisabled,
        trigger: triggerAction
    }, state, tooltipTriggerRef);
    let [borderRadius, setBorderRadius] = (0, $kAvU8$react.useState)(0);
    (0, $kAvU8$reactariautils.useLayoutEffect)(()=>{
        if (overlayRef.current && state.isOpen) {
            let spectrumBorderRadius = window.getComputedStyle(overlayRef.current).borderRadius;
            if (spectrumBorderRadius !== "") setBorderRadius(parseInt(spectrumBorderRadius, 10));
        }
    }, [
        state.isOpen,
        overlayRef
    ]);
    let arrowRef = (0, $kAvU8$react.useRef)(null);
    let [arrowWidth, setArrowWidth] = (0, $kAvU8$react.useState)(0);
    (0, $kAvU8$reactariautils.useLayoutEffect)(()=>{
        if (arrowRef.current && state.isOpen) setArrowWidth(arrowRef.current.getBoundingClientRect().width);
    }, [
        state.isOpen,
        arrowRef
    ]);
    let { overlayProps: overlayProps, arrowProps: arrowProps, placement: placement } = (0, $kAvU8$reactariaoverlays.useOverlayPosition)({
        placement: props.placement || "top",
        targetRef: tooltipTriggerRef,
        overlayRef: overlayRef,
        offset: offset,
        crossOffset: crossOffset,
        isOpen: state.isOpen,
        shouldFlip: props.shouldFlip,
        containerPadding: props.containerPadding,
        arrowSize: arrowWidth,
        arrowBoundaryOffset: borderRadius
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement((0, $kAvU8$reactariafocus.FocusableProvider), {
        ...triggerProps,
        ref: tooltipTriggerRef
    }, trigger, /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement((0, $ed2e56c8503433b4$export$39ae08fa83328b12).Provider, {
        value: {
            state: state,
            placement: placement,
            ref: overlayRef,
            UNSAFE_style: overlayProps.style,
            arrowProps: arrowProps,
            arrowRef: arrowRef,
            ...tooltipProps
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement((0, $kAvU8$reactspectrumoverlays.Overlay), {
        isOpen: state.isOpen,
        nodeRef: overlayRef
    }, tooltip)));
}
// Support TooltipTrigger inside components using CollectionBuilder.
$19f06480c1c1a932$var$TooltipTrigger.getCollectionNode = function*(props) {
    // Replaced the use of React.Childern.toArray because it mutates the key prop.
    let childArray = [];
    (0, ($parcel$interopDefault($kAvU8$react))).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).isValidElement(child)) childArray.push(child);
    });
    let [trigger, tooltip] = childArray;
    yield {
        element: trigger,
        wrapper: (element)=>/*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement($19f06480c1c1a932$var$TooltipTrigger, {
                key: element.key,
                ...props
            }, element, tooltip)
    };
};
/**
 * TooltipTrigger wraps around a trigger element and a Tooltip. It handles opening and closing
 * the Tooltip when the user hovers over or focuses the trigger, and positioning the Tooltip
 * relative to the trigger.
 */ // We don't want getCollectionNode to show up in the type definition
let $19f06480c1c1a932$export$8c610744efcf8a1d = $19f06480c1c1a932$var$TooltipTrigger;
