// @react-spectrum/contextualhelp/dist/import.mjs
"use client";

import "./main.css";
import $9cc8b85ee8327c05$exports from "./ar-AE.mjs";
import $56f7d9be41e5dc75$exports from "./bg-BG.mjs";
import $c427e4a9ab6700b2$exports from "./cs-CZ.mjs";
import $ee2212e66656667f$exports from "./da-DK.mjs";
import $112e467ad7137275$exports from "./de-DE.mjs";
import $7ad0d6a9c54bdc43$exports from "./el-GR.mjs";
import $704e90e35332a9cb$exports from "./en-US.mjs";
import $f2e0410dbb606550$exports from "./es-ES.mjs";
import $f4852a97714b8951$exports from "./et-EE.mjs";
import $fb4baf837c8a96f6$exports from "./fi-FI.mjs";
import $28b0fbf00803a68f$exports from "./fr-FR.mjs";
import $77110a8f3000e559$exports from "./he-IL.mjs";
import $dcc3492993669fc3$exports from "./hr-HR.mjs";
import $d376dd5b16d08137$exports from "./hu-HU.mjs";
import $45f88f0321bb5a8f$exports from "./it-IT.mjs";
import $8f643aac2e9e9e26$exports from "./ja-JP.mjs";
import $784ec4e789b99a71$exports from "./ko-KR.mjs";
import $fcda0e0582088650$exports from "./lt-LT.mjs";
import $9c3b35ecd7c5e348$exports from "./lv-LV.mjs";
import $b6518647ee722bc3$exports from "./nb-NO.mjs";
import $5cb1939fecdd5f1a$exports from "./nl-NL.mjs";
import $3ed876b500650109$exports from "./pl-PL.mjs";
import $999b0348d4f9e0ae$exports from "./pt-BR.mjs";
import $01e38e0d24e067b8$exports from "./pt-PT.mjs";
import $130ce4093fe216d5$exports from "./ro-RO.mjs";
import $3b25696a80a9bfd8$exports from "./ru-RU.mjs";
import $70037a741b152768$exports from "./sk-SK.mjs";
import $4ce06844b60c40b5$exports from "./sl-SI.mjs";
import $78c0c0c4796561ef$exports from "./sr-SP.mjs";
import $a701758f24293d53$exports from "./sv-SE.mjs";
import $ac0041dae2f30090$exports from "./tr-TR.mjs";
import $e3e68b40fcb16018$exports from "./uk-UA.mjs";
import $befba97b83d35197$exports from "./zh-CN.mjs";
import $6051e1035ff73e7b$exports from "./zh-TW.mjs";
import {ActionButton as $2FtBM$ActionButton} from "@react-spectrum/button";
import {classNames as $2FtBM$classNames, ClearSlots as $2FtBM$ClearSlots, SlotProvider as $2FtBM$SlotProvider} from "@react-spectrum/utils";
import {DialogTrigger as $2FtBM$DialogTrigger, Dialog as $2FtBM$Dialog} from "@react-spectrum/dialog";
import $2FtBM$spectrumiconsworkflowHelpOutline from "@spectrum-icons/workflow/HelpOutline";
import $2FtBM$spectrumiconsworkflowInfoOutline from "@spectrum-icons/workflow/InfoOutline";
import {useLabels as $2FtBM$useLabels, mergeProps as $2FtBM$mergeProps} from "@react-aria/utils";
import $2FtBM$react from "react";
import {useLocalizedStringFormatter as $2FtBM$useLocalizedStringFormatter} from "@react-aria/i18n";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $c487b1d738ebeac2$exports = {};

$parcel$export($c487b1d738ebeac2$exports, "react-spectrum-ContextualHelp-button", () => $c487b1d738ebeac2$export$20d9583c9f1529c1, (v) => $c487b1d738ebeac2$export$20d9583c9f1529c1 = v);
$parcel$export($c487b1d738ebeac2$exports, "react-spectrum-ContextualHelp-content", () => $c487b1d738ebeac2$export$a1f805890cf58c4, (v) => $c487b1d738ebeac2$export$a1f805890cf58c4 = v);
$parcel$export($c487b1d738ebeac2$exports, "react-spectrum-ContextualHelp-dialog", () => $c487b1d738ebeac2$export$ab2d873674d20d28, (v) => $c487b1d738ebeac2$export$ab2d873674d20d28 = v);
$parcel$export($c487b1d738ebeac2$exports, "react-spectrum-ContextualHelp-dialog--isMobile", () => $c487b1d738ebeac2$export$a438776606a9a3cc, (v) => $c487b1d738ebeac2$export$a438776606a9a3cc = v);
$parcel$export($c487b1d738ebeac2$exports, "react-spectrum-ContextualHelp-footer", () => $c487b1d738ebeac2$export$d901bc3fa82b87a6, (v) => $c487b1d738ebeac2$export$d901bc3fa82b87a6 = v);
var $c487b1d738ebeac2$export$20d9583c9f1529c1;
var $c487b1d738ebeac2$export$a1f805890cf58c4;
var $c487b1d738ebeac2$export$ab2d873674d20d28;
var $c487b1d738ebeac2$export$a438776606a9a3cc;
var $c487b1d738ebeac2$export$d901bc3fa82b87a6;
$c487b1d738ebeac2$export$20d9583c9f1529c1 = `L6gUaa_react-spectrum-ContextualHelp-button`;
$c487b1d738ebeac2$export$a1f805890cf58c4 = `L6gUaa_react-spectrum-ContextualHelp-content`;
$c487b1d738ebeac2$export$ab2d873674d20d28 = `L6gUaa_react-spectrum-ContextualHelp-dialog`;
$c487b1d738ebeac2$export$a438776606a9a3cc = `L6gUaa_react-spectrum-ContextualHelp-dialog--isMobile`;
$c487b1d738ebeac2$export$d901bc3fa82b87a6 = `L6gUaa_react-spectrum-ContextualHelp-footer`;



var $ff7022a435a048e3$exports = {};


































$ff7022a435a048e3$exports = {
    "ar-AE": $9cc8b85ee8327c05$exports,
    "bg-BG": $56f7d9be41e5dc75$exports,
    "cs-CZ": $c427e4a9ab6700b2$exports,
    "da-DK": $ee2212e66656667f$exports,
    "de-DE": $112e467ad7137275$exports,
    "el-GR": $7ad0d6a9c54bdc43$exports,
    "en-US": $704e90e35332a9cb$exports,
    "es-ES": $f2e0410dbb606550$exports,
    "et-EE": $f4852a97714b8951$exports,
    "fi-FI": $fb4baf837c8a96f6$exports,
    "fr-FR": $28b0fbf00803a68f$exports,
    "he-IL": $77110a8f3000e559$exports,
    "hr-HR": $dcc3492993669fc3$exports,
    "hu-HU": $d376dd5b16d08137$exports,
    "it-IT": $45f88f0321bb5a8f$exports,
    "ja-JP": $8f643aac2e9e9e26$exports,
    "ko-KR": $784ec4e789b99a71$exports,
    "lt-LT": $fcda0e0582088650$exports,
    "lv-LV": $9c3b35ecd7c5e348$exports,
    "nb-NO": $b6518647ee722bc3$exports,
    "nl-NL": $5cb1939fecdd5f1a$exports,
    "pl-PL": $3ed876b500650109$exports,
    "pt-BR": $999b0348d4f9e0ae$exports,
    "pt-PT": $01e38e0d24e067b8$exports,
    "ro-RO": $130ce4093fe216d5$exports,
    "ru-RU": $3b25696a80a9bfd8$exports,
    "sk-SK": $70037a741b152768$exports,
    "sl-SI": $4ce06844b60c40b5$exports,
    "sr-SP": $78c0c0c4796561ef$exports,
    "sv-SE": $a701758f24293d53$exports,
    "tr-TR": $ac0041dae2f30090$exports,
    "uk-UA": $e3e68b40fcb16018$exports,
    "zh-CN": $befba97b83d35197$exports,
    "zh-TW": $6051e1035ff73e7b$exports
};





function $9d9069cfc9959198$var$ContextualHelp(props, ref) {
    let { variant: variant = "help", placement: placement = "bottom start", children: children, ...otherProps } = props;
    let stringFormatter = (0, $2FtBM$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($ff7022a435a048e3$exports))), "@react-spectrum/contextualhelp");
    let icon = variant === "info" ? /*#__PURE__*/ (0, $2FtBM$react).createElement((0, $2FtBM$spectrumiconsworkflowInfoOutline), null) : /*#__PURE__*/ (0, $2FtBM$react).createElement((0, $2FtBM$spectrumiconsworkflowHelpOutline), null);
    let slots = {
        content: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c487b1d738ebeac2$exports)))["react-spectrum-ContextualHelp-content"]
        },
        footer: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($c487b1d738ebeac2$exports)))["react-spectrum-ContextualHelp-footer"]
        }
    };
    let labelProps = (0, $2FtBM$useLabels)(otherProps, stringFormatter.format(variant));
    return /*#__PURE__*/ (0, $2FtBM$react).createElement((0, $2FtBM$DialogTrigger), {
        ...otherProps,
        type: "popover",
        placement: placement,
        hideArrow: true
    }, /*#__PURE__*/ (0, $2FtBM$react).createElement((0, $2FtBM$ActionButton), {
        ...(0, $2FtBM$mergeProps)(otherProps, labelProps, {
            isDisabled: false
        }),
        ref: ref,
        UNSAFE_className: (0, $2FtBM$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c487b1d738ebeac2$exports))), "react-spectrum-ContextualHelp-button", otherProps.UNSAFE_className),
        isQuiet: true
    }, icon), /*#__PURE__*/ (0, $2FtBM$react).createElement((0, $2FtBM$ClearSlots), null, /*#__PURE__*/ (0, $2FtBM$react).createElement((0, $2FtBM$SlotProvider), {
        slots: slots
    }, /*#__PURE__*/ (0, $2FtBM$react).createElement((0, $2FtBM$Dialog), {
        UNSAFE_className: (0, $2FtBM$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($c487b1d738ebeac2$exports))), "react-spectrum-ContextualHelp-dialog")
    }, children))));
}
/**
 * Contextual help shows a user extra information about the state of an adjacent component, or a total view.
 */ let $9d9069cfc9959198$export$7d3cdb256c2ba320 = /*#__PURE__*/ (0, $2FtBM$react).forwardRef($9d9069cfc9959198$var$ContextualHelp);




export {$9d9069cfc9959198$export$7d3cdb256c2ba320 as ContextualHelp};
