// @react-spectrum/contextualhelp/dist/main.js
"use client";

require("./main.css");
var $acb7f58c98e2cf06$exports = require("./ar-AE.main.js");
var $2bbfbfd282c7ac2e$exports = require("./bg-BG.main.js");
var $fb114849c2cd165d$exports = require("./cs-CZ.main.js");
var $d915a218ab0c9e80$exports = require("./da-DK.main.js");
var $b967195fa197fdf6$exports = require("./de-DE.main.js");
var $2e327399cfbbb22c$exports = require("./el-GR.main.js");
var $76d69c9d92b7631c$exports = require("./en-US.main.js");
var $f16d20dee3bf2650$exports = require("./es-ES.main.js");
var $681c6c332d3324fe$exports = require("./et-EE.main.js");
var $aaae4f968feee6a5$exports = require("./fi-FI.main.js");
var $6ca7aa54fe17b013$exports = require("./fr-FR.main.js");
var $eead079849c849a8$exports = require("./he-IL.main.js");
var $c595ccd23af366f1$exports = require("./hr-HR.main.js");
var $13646d03ad3676f9$exports = require("./hu-HU.main.js");
var $d288177f1ed0c023$exports = require("./it-IT.main.js");
var $97fad5409bb5ac4b$exports = require("./ja-JP.main.js");
var $8c367af4c98bf9ef$exports = require("./ko-KR.main.js");
var $ff404be8bf7c1541$exports = require("./lt-LT.main.js");
var $8f9381b231c1b1e7$exports = require("./lv-LV.main.js");
var $66586c79322de90c$exports = require("./nb-NO.main.js");
var $2707c2e216fd1fb0$exports = require("./nl-NL.main.js");
var $1b8a6c42bf8a93cf$exports = require("./pl-PL.main.js");
var $21519e2cdfbf2418$exports = require("./pt-BR.main.js");
var $af034b95fde6a053$exports = require("./pt-PT.main.js");
var $4881897691d95dcd$exports = require("./ro-RO.main.js");
var $697e58aab1351cb6$exports = require("./ru-RU.main.js");
var $18d1df2cd1fd4664$exports = require("./sk-SK.main.js");
var $5923d5e92381fe28$exports = require("./sl-SI.main.js");
var $0ef592839a88ef41$exports = require("./sr-SP.main.js");
var $c254a7ac0f4821d7$exports = require("./sv-SE.main.js");
var $a520943f16b6160a$exports = require("./tr-TR.main.js");
var $679d794c1ece5065$exports = require("./uk-UA.main.js");
var $a531570c7a651d45$exports = require("./zh-CN.main.js");
var $487db55c4b552587$exports = require("./zh-TW.main.js");
var $ST3N8$reactspectrumbutton = require("@react-spectrum/button");
var $ST3N8$reactspectrumutils = require("@react-spectrum/utils");
var $ST3N8$reactspectrumdialog = require("@react-spectrum/dialog");
var $ST3N8$spectrumiconsworkflowHelpOutline = require("@spectrum-icons/workflow/HelpOutline");
var $ST3N8$spectrumiconsworkflowInfoOutline = require("@spectrum-icons/workflow/InfoOutline");
var $ST3N8$reactariautils = require("@react-aria/utils");
var $ST3N8$react = require("react");
var $ST3N8$reactariai18n = require("@react-aria/i18n");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "ContextualHelp", () => $05e6407aa891f27c$export$7d3cdb256c2ba320);
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $a10e565ec42bcf96$exports = {};

$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-button", () => $a10e565ec42bcf96$export$20d9583c9f1529c1, (v) => $a10e565ec42bcf96$export$20d9583c9f1529c1 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-content", () => $a10e565ec42bcf96$export$a1f805890cf58c4, (v) => $a10e565ec42bcf96$export$a1f805890cf58c4 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-dialog", () => $a10e565ec42bcf96$export$ab2d873674d20d28, (v) => $a10e565ec42bcf96$export$ab2d873674d20d28 = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-dialog--isMobile", () => $a10e565ec42bcf96$export$a438776606a9a3cc, (v) => $a10e565ec42bcf96$export$a438776606a9a3cc = v);
$parcel$export($a10e565ec42bcf96$exports, "react-spectrum-ContextualHelp-footer", () => $a10e565ec42bcf96$export$d901bc3fa82b87a6, (v) => $a10e565ec42bcf96$export$d901bc3fa82b87a6 = v);
var $a10e565ec42bcf96$export$20d9583c9f1529c1;
var $a10e565ec42bcf96$export$a1f805890cf58c4;
var $a10e565ec42bcf96$export$ab2d873674d20d28;
var $a10e565ec42bcf96$export$a438776606a9a3cc;
var $a10e565ec42bcf96$export$d901bc3fa82b87a6;
$a10e565ec42bcf96$export$20d9583c9f1529c1 = `L6gUaa_react-spectrum-ContextualHelp-button`;
$a10e565ec42bcf96$export$a1f805890cf58c4 = `L6gUaa_react-spectrum-ContextualHelp-content`;
$a10e565ec42bcf96$export$ab2d873674d20d28 = `L6gUaa_react-spectrum-ContextualHelp-dialog`;
$a10e565ec42bcf96$export$a438776606a9a3cc = `L6gUaa_react-spectrum-ContextualHelp-dialog--isMobile`;
$a10e565ec42bcf96$export$d901bc3fa82b87a6 = `L6gUaa_react-spectrum-ContextualHelp-footer`;



var $ab07f09ac8b48f11$exports = {};


































$ab07f09ac8b48f11$exports = {
    "ar-AE": $acb7f58c98e2cf06$exports,
    "bg-BG": $2bbfbfd282c7ac2e$exports,
    "cs-CZ": $fb114849c2cd165d$exports,
    "da-DK": $d915a218ab0c9e80$exports,
    "de-DE": $b967195fa197fdf6$exports,
    "el-GR": $2e327399cfbbb22c$exports,
    "en-US": $76d69c9d92b7631c$exports,
    "es-ES": $f16d20dee3bf2650$exports,
    "et-EE": $681c6c332d3324fe$exports,
    "fi-FI": $aaae4f968feee6a5$exports,
    "fr-FR": $6ca7aa54fe17b013$exports,
    "he-IL": $eead079849c849a8$exports,
    "hr-HR": $c595ccd23af366f1$exports,
    "hu-HU": $13646d03ad3676f9$exports,
    "it-IT": $d288177f1ed0c023$exports,
    "ja-JP": $97fad5409bb5ac4b$exports,
    "ko-KR": $8c367af4c98bf9ef$exports,
    "lt-LT": $ff404be8bf7c1541$exports,
    "lv-LV": $8f9381b231c1b1e7$exports,
    "nb-NO": $66586c79322de90c$exports,
    "nl-NL": $2707c2e216fd1fb0$exports,
    "pl-PL": $1b8a6c42bf8a93cf$exports,
    "pt-BR": $21519e2cdfbf2418$exports,
    "pt-PT": $af034b95fde6a053$exports,
    "ro-RO": $4881897691d95dcd$exports,
    "ru-RU": $697e58aab1351cb6$exports,
    "sk-SK": $18d1df2cd1fd4664$exports,
    "sl-SI": $5923d5e92381fe28$exports,
    "sr-SP": $0ef592839a88ef41$exports,
    "sv-SE": $c254a7ac0f4821d7$exports,
    "tr-TR": $a520943f16b6160a$exports,
    "uk-UA": $679d794c1ece5065$exports,
    "zh-CN": $a531570c7a651d45$exports,
    "zh-TW": $487db55c4b552587$exports
};





function $05e6407aa891f27c$var$ContextualHelp(props, ref) {
    let { variant: variant = "help", placement: placement = "bottom start", children: children, ...otherProps } = props;
    let stringFormatter = (0, $ST3N8$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($ab07f09ac8b48f11$exports))), "@react-spectrum/contextualhelp");
    let icon = variant === "info" ? /*#__PURE__*/ (0, ($parcel$interopDefault($ST3N8$react))).createElement((0, ($parcel$interopDefault($ST3N8$spectrumiconsworkflowInfoOutline))), null) : /*#__PURE__*/ (0, ($parcel$interopDefault($ST3N8$react))).createElement((0, ($parcel$interopDefault($ST3N8$spectrumiconsworkflowHelpOutline))), null);
    let slots = {
        content: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports)))["react-spectrum-ContextualHelp-content"]
        },
        footer: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports)))["react-spectrum-ContextualHelp-footer"]
        }
    };
    let labelProps = (0, $ST3N8$reactariautils.useLabels)(otherProps, stringFormatter.format(variant));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ST3N8$react))).createElement((0, $ST3N8$reactspectrumdialog.DialogTrigger), {
        ...otherProps,
        type: "popover",
        placement: placement,
        hideArrow: true
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ST3N8$react))).createElement((0, $ST3N8$reactspectrumbutton.ActionButton), {
        ...(0, $ST3N8$reactariautils.mergeProps)(otherProps, labelProps, {
            isDisabled: false
        }),
        ref: ref,
        UNSAFE_className: (0, $ST3N8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports))), "react-spectrum-ContextualHelp-button", otherProps.UNSAFE_className),
        isQuiet: true
    }, icon), /*#__PURE__*/ (0, ($parcel$interopDefault($ST3N8$react))).createElement((0, $ST3N8$reactspectrumutils.ClearSlots), null, /*#__PURE__*/ (0, ($parcel$interopDefault($ST3N8$react))).createElement((0, $ST3N8$reactspectrumutils.SlotProvider), {
        slots: slots
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ST3N8$react))).createElement((0, $ST3N8$reactspectrumdialog.Dialog), {
        UNSAFE_className: (0, $ST3N8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a10e565ec42bcf96$exports))), "react-spectrum-ContextualHelp-dialog")
    }, children))));
}
/**
 * Contextual help shows a user extra information about the state of an adjacent component, or a total view.
 */ let $05e6407aa891f27c$export$7d3cdb256c2ba320 = /*#__PURE__*/ (0, ($parcel$interopDefault($ST3N8$react))).forwardRef($05e6407aa891f27c$var$ContextualHelp);
