// @react-spectrum/label/dist/main.js
"use client";

require("./main.css");
var $be852afcd8987832$exports = require("./ar-AE.main.js");
var $aa4ffc3da65655ef$exports = require("./bg-BG.main.js");
var $70336543455a8968$exports = require("./cs-CZ.main.js");
var $da721886d3a532d3$exports = require("./da-DK.main.js");
var $8d18e19dccc28394$exports = require("./de-DE.main.js");
var $c70340f3f366f0c9$exports = require("./el-GR.main.js");
var $100a4f0d92ab2685$exports = require("./en-US.main.js");
var $376a75ae5e97584b$exports = require("./es-ES.main.js");
var $ea0a534428756103$exports = require("./et-EE.main.js");
var $2176594cdf3907b0$exports = require("./fi-FI.main.js");
var $1190b8ce746b4c70$exports = require("./fr-FR.main.js");
var $b78e8997233e1d53$exports = require("./he-IL.main.js");
var $1788058e8b1338cf$exports = require("./hr-HR.main.js");
var $5e7e524c4d77d357$exports = require("./hu-HU.main.js");
var $8545837977fb83c7$exports = require("./it-IT.main.js");
var $b0b5a068cf5e941f$exports = require("./ja-JP.main.js");
var $87a7a70faf1c577f$exports = require("./ko-KR.main.js");
var $c3f2142cd0e94e15$exports = require("./lt-LT.main.js");
var $008ebeb5a8039d3b$exports = require("./lv-LV.main.js");
var $3ba626fbf1933ce1$exports = require("./nb-NO.main.js");
var $78ed57b244f3862b$exports = require("./nl-NL.main.js");
var $391247f29fc5b238$exports = require("./pl-PL.main.js");
var $c03311fdea551db1$exports = require("./pt-BR.main.js");
var $82f94bdcbdbace10$exports = require("./pt-PT.main.js");
var $459c5159c6836bf9$exports = require("./ro-RO.main.js");
var $e94cc84d0b96cae7$exports = require("./ru-RU.main.js");
var $f9f957260773ddeb$exports = require("./sk-SK.main.js");
var $9775fc82dcd5cb58$exports = require("./sl-SI.main.js");
var $f735f63121c09841$exports = require("./sr-SP.main.js");
var $bbe1e12be1e3dd67$exports = require("./sv-SE.main.js");
var $c8baaffeccc5d05c$exports = require("./tr-TR.main.js");
var $849858b96dfabeac$exports = require("./uk-UA.main.js");
var $3135581a6ef2f6d9$exports = require("./zh-CN.main.js");
var $fb217acd414dfdf8$exports = require("./zh-TW.main.js");
var $fXO85$reactspectrumutils = require("@react-spectrum/utils");
var $fXO85$reactspectrumlayout = require("@react-spectrum/layout");
var $fXO85$reactariautils = require("@react-aria/utils");
var $fXO85$react = require("react");
var $fXO85$reactspectrumform = require("@react-spectrum/form");
var $fXO85$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $fXO85$spectrumiconsuiAsterisk = require("@spectrum-icons/ui/Asterisk");
var $fXO85$reactariai18n = require("@react-aria/i18n");
var $fXO85$reactspectrumprovider = require("@react-spectrum/provider");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Field", () => $0e5410371a445d59$export$a455218a85c89869);
$parcel$export(module.exports, "Label", () => $38a9afbd39356629$export$b04be29aa201d4f5);
$parcel$export(module.exports, "HelpText", () => $2d1b5114e5b358b6$export$a67c0bc59081311a);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $680ab92507bf49aa$exports = {};

$parcel$export($680ab92507bf49aa$exports, "focus-ring", () => $680ab92507bf49aa$export$f39a09f249340e2a, (v) => $680ab92507bf49aa$export$f39a09f249340e2a = v);
$parcel$export($680ab92507bf49aa$exports, "i18nFontFamily", () => $680ab92507bf49aa$export$8c4ee2c50c22c514, (v) => $680ab92507bf49aa$export$8c4ee2c50c22c514 = v);
$parcel$export($680ab92507bf49aa$exports, "is-disabled", () => $680ab92507bf49aa$export$d35bc1e505d1ebbf, (v) => $680ab92507bf49aa$export$d35bc1e505d1ebbf = v);
$parcel$export($680ab92507bf49aa$exports, "spectrum-FocusRing-ring", () => $680ab92507bf49aa$export$4109102f950813a6, (v) => $680ab92507bf49aa$export$4109102f950813a6 = v);
$parcel$export($680ab92507bf49aa$exports, "spectrum-FocusRing", () => $680ab92507bf49aa$export$24c7f46a6e3605dd, (v) => $680ab92507bf49aa$export$24c7f46a6e3605dd = v);
$parcel$export($680ab92507bf49aa$exports, "spectrum-FocusRing--quiet", () => $680ab92507bf49aa$export$2927016961429360, (v) => $680ab92507bf49aa$export$2927016961429360 = v);
$parcel$export($680ab92507bf49aa$exports, "spectrum-HelpText", () => $680ab92507bf49aa$export$c0875e83c57c7736, (v) => $680ab92507bf49aa$export$c0875e83c57c7736 = v);
$parcel$export($680ab92507bf49aa$exports, "spectrum-HelpText--negative", () => $680ab92507bf49aa$export$d31dd875d69daee0, (v) => $680ab92507bf49aa$export$d31dd875d69daee0 = v);
$parcel$export($680ab92507bf49aa$exports, "spectrum-HelpText--neutral", () => $680ab92507bf49aa$export$6e9351b1ad13c, (v) => $680ab92507bf49aa$export$6e9351b1ad13c = v);
$parcel$export($680ab92507bf49aa$exports, "spectrum-HelpText-text", () => $680ab92507bf49aa$export$a512452beb4231bf, (v) => $680ab92507bf49aa$export$a512452beb4231bf = v);
$parcel$export($680ab92507bf49aa$exports, "spectrum-HelpText-validationIcon", () => $680ab92507bf49aa$export$527776642b333691, (v) => $680ab92507bf49aa$export$527776642b333691 = v);
var $680ab92507bf49aa$export$f39a09f249340e2a;
var $680ab92507bf49aa$export$8c4ee2c50c22c514;
var $680ab92507bf49aa$export$d35bc1e505d1ebbf;
var $680ab92507bf49aa$export$4109102f950813a6;
var $680ab92507bf49aa$export$24c7f46a6e3605dd;
var $680ab92507bf49aa$export$2927016961429360;
var $680ab92507bf49aa$export$c0875e83c57c7736;
var $680ab92507bf49aa$export$d31dd875d69daee0;
var $680ab92507bf49aa$export$6e9351b1ad13c;
var $680ab92507bf49aa$export$a512452beb4231bf;
var $680ab92507bf49aa$export$527776642b333691;
$680ab92507bf49aa$export$f39a09f249340e2a = `a1Qy2q_focus-ring`;
$680ab92507bf49aa$export$8c4ee2c50c22c514 = `a1Qy2q_i18nFontFamily`;
$680ab92507bf49aa$export$d35bc1e505d1ebbf = `a1Qy2q_is-disabled`;
$680ab92507bf49aa$export$4109102f950813a6 = `a1Qy2q_spectrum-FocusRing-ring`;
$680ab92507bf49aa$export$24c7f46a6e3605dd = `a1Qy2q_spectrum-FocusRing ${$680ab92507bf49aa$export$4109102f950813a6}`;
$680ab92507bf49aa$export$2927016961429360 = `a1Qy2q_spectrum-FocusRing--quiet`;
$680ab92507bf49aa$export$c0875e83c57c7736 = `a1Qy2q_spectrum-HelpText`;
$680ab92507bf49aa$export$d31dd875d69daee0 = `a1Qy2q_spectrum-HelpText--negative`;
$680ab92507bf49aa$export$6e9351b1ad13c = `a1Qy2q_spectrum-HelpText--neutral`;
$680ab92507bf49aa$export$a512452beb4231bf = `a1Qy2q_spectrum-HelpText-text`;
$680ab92507bf49aa$export$527776642b333691 = `a1Qy2q_spectrum-HelpText-validationIcon`;


function $2d1b5114e5b358b6$var$HelpText(props, ref) {
    let { description: description, errorMessage: errorMessage, validationState: validationState, isInvalid: isInvalid, isDisabled: isDisabled, showErrorIcon: showErrorIcon, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps } = props;
    let domRef = (0, $fXO85$reactspectrumutils.useDOMRef)(ref);
    let isErrorMessage = errorMessage && (isInvalid || validationState === "invalid");
    let { styleProps: styleProps } = (0, $fXO85$reactspectrumutils.useStyleProps)(props);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement("div", {
        ...styleProps,
        className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($680ab92507bf49aa$exports))), "spectrum-HelpText", `spectrum-HelpText--${isErrorMessage ? "negative" : "neutral"}`, {
            "is-disabled": isDisabled
        }, styleProps.className),
        ref: domRef
    }, isErrorMessage ? /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, ($parcel$interopDefault($fXO85$react))).Fragment, null, showErrorIcon && /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, ($parcel$interopDefault($fXO85$spectrumiconsuiAlertMedium))), {
        UNSAFE_className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($680ab92507bf49aa$exports))), "spectrum-HelpText-validationIcon")
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement("div", {
        ...errorMessageProps,
        className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($680ab92507bf49aa$exports))), "spectrum-HelpText-text")
    }, errorMessage)) : /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement("div", {
        ...descriptionProps,
        className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($680ab92507bf49aa$exports))), "spectrum-HelpText-text")
    }, description));
}
/**
 * Help text provides either an informative description or an error message that gives more context about what a user needs to input. It's commonly used in forms.
 */ const $2d1b5114e5b358b6$export$a67c0bc59081311a = /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).forwardRef($2d1b5114e5b358b6$var$HelpText);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $503911526ad50850$exports = {};


































$503911526ad50850$exports = {
    "ar-AE": $be852afcd8987832$exports,
    "bg-BG": $aa4ffc3da65655ef$exports,
    "cs-CZ": $70336543455a8968$exports,
    "da-DK": $da721886d3a532d3$exports,
    "de-DE": $8d18e19dccc28394$exports,
    "el-GR": $c70340f3f366f0c9$exports,
    "en-US": $100a4f0d92ab2685$exports,
    "es-ES": $376a75ae5e97584b$exports,
    "et-EE": $ea0a534428756103$exports,
    "fi-FI": $2176594cdf3907b0$exports,
    "fr-FR": $1190b8ce746b4c70$exports,
    "he-IL": $b78e8997233e1d53$exports,
    "hr-HR": $1788058e8b1338cf$exports,
    "hu-HU": $5e7e524c4d77d357$exports,
    "it-IT": $8545837977fb83c7$exports,
    "ja-JP": $b0b5a068cf5e941f$exports,
    "ko-KR": $87a7a70faf1c577f$exports,
    "lt-LT": $c3f2142cd0e94e15$exports,
    "lv-LV": $008ebeb5a8039d3b$exports,
    "nb-NO": $3ba626fbf1933ce1$exports,
    "nl-NL": $78ed57b244f3862b$exports,
    "pl-PL": $391247f29fc5b238$exports,
    "pt-BR": $c03311fdea551db1$exports,
    "pt-PT": $82f94bdcbdbace10$exports,
    "ro-RO": $459c5159c6836bf9$exports,
    "ru-RU": $e94cc84d0b96cae7$exports,
    "sk-SK": $f9f957260773ddeb$exports,
    "sl-SI": $9775fc82dcd5cb58$exports,
    "sr-SP": $f735f63121c09841$exports,
    "sv-SE": $bbe1e12be1e3dd67$exports,
    "tr-TR": $c8baaffeccc5d05c$exports,
    "uk-UA": $849858b96dfabeac$exports,
    "zh-CN": $3135581a6ef2f6d9$exports,
    "zh-TW": $fb217acd414dfdf8$exports
};



var $eb2fc0ee655eff6b$exports = {};

$parcel$export($eb2fc0ee655eff6b$exports, "contextualHelp", () => $eb2fc0ee655eff6b$export$4ba534aba1836d, (v) => $eb2fc0ee655eff6b$export$4ba534aba1836d = v);
$parcel$export($eb2fc0ee655eff6b$exports, "field", () => $eb2fc0ee655eff6b$export$e0f35d825088c098, (v) => $eb2fc0ee655eff6b$export$e0f35d825088c098 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "focus-ring", () => $eb2fc0ee655eff6b$export$f39a09f249340e2a, (v) => $eb2fc0ee655eff6b$export$f39a09f249340e2a = v);
$parcel$export($eb2fc0ee655eff6b$exports, "helpText", () => $eb2fc0ee655eff6b$export$431a3b688422af44, (v) => $eb2fc0ee655eff6b$export$431a3b688422af44 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "i18nFontFamily", () => $eb2fc0ee655eff6b$export$8c4ee2c50c22c514, (v) => $eb2fc0ee655eff6b$export$8c4ee2c50c22c514 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "is-disabled", () => $eb2fc0ee655eff6b$export$d35bc1e505d1ebbf, (v) => $eb2fc0ee655eff6b$export$d35bc1e505d1ebbf = v);
$parcel$export($eb2fc0ee655eff6b$exports, "label", () => $eb2fc0ee655eff6b$export$1237798dc640739a, (v) => $eb2fc0ee655eff6b$export$1237798dc640739a = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field", () => $eb2fc0ee655eff6b$export$f6d480ae1e56eba0, (v) => $eb2fc0ee655eff6b$export$f6d480ae1e56eba0 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field--alignEnd", () => $eb2fc0ee655eff6b$export$5354c9be7ada5f83, (v) => $eb2fc0ee655eff6b$export$5354c9be7ada5f83 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field--hasContextualHelp", () => $eb2fc0ee655eff6b$export$9ba23707a252cdd4, (v) => $eb2fc0ee655eff6b$export$9ba23707a252cdd4 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field--positionSide", () => $eb2fc0ee655eff6b$export$2963225c91179589, (v) => $eb2fc0ee655eff6b$export$2963225c91179589 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field--positionTop", () => $eb2fc0ee655eff6b$export$a4ea780a9064d7f9, (v) => $eb2fc0ee655eff6b$export$a4ea780a9064d7f9 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-contextualHelp", () => $eb2fc0ee655eff6b$export$24fb860d16dd2d3f, (v) => $eb2fc0ee655eff6b$export$24fb860d16dd2d3f = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-field", () => $eb2fc0ee655eff6b$export$3ab8a3dc7f0563d, (v) => $eb2fc0ee655eff6b$export$3ab8a3dc7f0563d = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-labelCell", () => $eb2fc0ee655eff6b$export$9cde5aecff012641, (v) => $eb2fc0ee655eff6b$export$9cde5aecff012641 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-labelWrapper", () => $eb2fc0ee655eff6b$export$a872a8c78b26c9e0, (v) => $eb2fc0ee655eff6b$export$a872a8c78b26c9e0 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-wrapper", () => $eb2fc0ee655eff6b$export$127df0b7290eb8ef, (v) => $eb2fc0ee655eff6b$export$127df0b7290eb8ef = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FieldLabel", () => $eb2fc0ee655eff6b$export$493d18e796ae054, (v) => $eb2fc0ee655eff6b$export$493d18e796ae054 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FieldLabel--alignEnd", () => $eb2fc0ee655eff6b$export$885efcc08143a987, (v) => $eb2fc0ee655eff6b$export$885efcc08143a987 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FieldLabel--positionSide", () => $eb2fc0ee655eff6b$export$9e6f19dc21f22f2e, (v) => $eb2fc0ee655eff6b$export$9e6f19dc21f22f2e = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FieldLabel-requiredIcon", () => $eb2fc0ee655eff6b$export$7c47642c7d46f3c9, (v) => $eb2fc0ee655eff6b$export$7c47642c7d46f3c9 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FocusRing-ring", () => $eb2fc0ee655eff6b$export$4109102f950813a6, (v) => $eb2fc0ee655eff6b$export$4109102f950813a6 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FocusRing", () => $eb2fc0ee655eff6b$export$24c7f46a6e3605dd, (v) => $eb2fc0ee655eff6b$export$24c7f46a6e3605dd = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FocusRing--quiet", () => $eb2fc0ee655eff6b$export$2927016961429360, (v) => $eb2fc0ee655eff6b$export$2927016961429360 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Form", () => $eb2fc0ee655eff6b$export$94d2bcc94afabd89, (v) => $eb2fc0ee655eff6b$export$94d2bcc94afabd89 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Form--positionSide", () => $eb2fc0ee655eff6b$export$7f44db659563c8f4, (v) => $eb2fc0ee655eff6b$export$7f44db659563c8f4 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Form--positionTop", () => $eb2fc0ee655eff6b$export$11b8bc81fe551057, (v) => $eb2fc0ee655eff6b$export$11b8bc81fe551057 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Form-itemLabel", () => $eb2fc0ee655eff6b$export$51ec27881a1e3928, (v) => $eb2fc0ee655eff6b$export$51ec27881a1e3928 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-LabeledValue", () => $eb2fc0ee655eff6b$export$79f9a5e9dd010c8d, (v) => $eb2fc0ee655eff6b$export$79f9a5e9dd010c8d = v);
var $eb2fc0ee655eff6b$export$4ba534aba1836d;
var $eb2fc0ee655eff6b$export$e0f35d825088c098;
var $eb2fc0ee655eff6b$export$f39a09f249340e2a;
var $eb2fc0ee655eff6b$export$431a3b688422af44;
var $eb2fc0ee655eff6b$export$8c4ee2c50c22c514;
var $eb2fc0ee655eff6b$export$d35bc1e505d1ebbf;
var $eb2fc0ee655eff6b$export$1237798dc640739a;
var $eb2fc0ee655eff6b$export$f6d480ae1e56eba0;
var $eb2fc0ee655eff6b$export$5354c9be7ada5f83;
var $eb2fc0ee655eff6b$export$9ba23707a252cdd4;
var $eb2fc0ee655eff6b$export$2963225c91179589;
var $eb2fc0ee655eff6b$export$a4ea780a9064d7f9;
var $eb2fc0ee655eff6b$export$24fb860d16dd2d3f;
var $eb2fc0ee655eff6b$export$3ab8a3dc7f0563d;
var $eb2fc0ee655eff6b$export$9cde5aecff012641;
var $eb2fc0ee655eff6b$export$a872a8c78b26c9e0;
var $eb2fc0ee655eff6b$export$127df0b7290eb8ef;
var $eb2fc0ee655eff6b$export$493d18e796ae054;
var $eb2fc0ee655eff6b$export$885efcc08143a987;
var $eb2fc0ee655eff6b$export$9e6f19dc21f22f2e;
var $eb2fc0ee655eff6b$export$7c47642c7d46f3c9;
var $eb2fc0ee655eff6b$export$4109102f950813a6;
var $eb2fc0ee655eff6b$export$24c7f46a6e3605dd;
var $eb2fc0ee655eff6b$export$2927016961429360;
var $eb2fc0ee655eff6b$export$94d2bcc94afabd89;
var $eb2fc0ee655eff6b$export$7f44db659563c8f4;
var $eb2fc0ee655eff6b$export$11b8bc81fe551057;
var $eb2fc0ee655eff6b$export$51ec27881a1e3928;
var $eb2fc0ee655eff6b$export$79f9a5e9dd010c8d;
$eb2fc0ee655eff6b$export$4ba534aba1836d = `A-HlBa_contextualHelp`;
$eb2fc0ee655eff6b$export$e0f35d825088c098 = `A-HlBa_field`;
$eb2fc0ee655eff6b$export$f39a09f249340e2a = `A-HlBa_focus-ring`;
$eb2fc0ee655eff6b$export$431a3b688422af44 = `A-HlBa_helpText`;
$eb2fc0ee655eff6b$export$8c4ee2c50c22c514 = `A-HlBa_i18nFontFamily`;
$eb2fc0ee655eff6b$export$d35bc1e505d1ebbf = `A-HlBa_is-disabled`;
$eb2fc0ee655eff6b$export$1237798dc640739a = `A-HlBa_label`;
$eb2fc0ee655eff6b$export$f6d480ae1e56eba0 = `A-HlBa_spectrum-Field`;
$eb2fc0ee655eff6b$export$5354c9be7ada5f83 = `A-HlBa_spectrum-Field--alignEnd`;
$eb2fc0ee655eff6b$export$9ba23707a252cdd4 = `A-HlBa_spectrum-Field--hasContextualHelp`;
$eb2fc0ee655eff6b$export$2963225c91179589 = `A-HlBa_spectrum-Field--positionSide`;
$eb2fc0ee655eff6b$export$a4ea780a9064d7f9 = `A-HlBa_spectrum-Field--positionTop`;
$eb2fc0ee655eff6b$export$24fb860d16dd2d3f = `A-HlBa_spectrum-Field-contextualHelp`;
$eb2fc0ee655eff6b$export$3ab8a3dc7f0563d = `A-HlBa_spectrum-Field-field`;
$eb2fc0ee655eff6b$export$9cde5aecff012641 = `A-HlBa_spectrum-Field-labelCell`;
$eb2fc0ee655eff6b$export$a872a8c78b26c9e0 = `A-HlBa_spectrum-Field-labelWrapper`;
$eb2fc0ee655eff6b$export$127df0b7290eb8ef = `A-HlBa_spectrum-Field-wrapper`;
$eb2fc0ee655eff6b$export$493d18e796ae054 = `A-HlBa_spectrum-FieldLabel`;
$eb2fc0ee655eff6b$export$885efcc08143a987 = `A-HlBa_spectrum-FieldLabel--alignEnd`;
$eb2fc0ee655eff6b$export$9e6f19dc21f22f2e = `A-HlBa_spectrum-FieldLabel--positionSide`;
$eb2fc0ee655eff6b$export$7c47642c7d46f3c9 = `A-HlBa_spectrum-FieldLabel-requiredIcon`;
$eb2fc0ee655eff6b$export$4109102f950813a6 = `A-HlBa_spectrum-FocusRing-ring`;
$eb2fc0ee655eff6b$export$24c7f46a6e3605dd = `A-HlBa_spectrum-FocusRing ${$eb2fc0ee655eff6b$export$4109102f950813a6}`;
$eb2fc0ee655eff6b$export$2927016961429360 = `A-HlBa_spectrum-FocusRing--quiet`;
$eb2fc0ee655eff6b$export$94d2bcc94afabd89 = `A-HlBa_spectrum-Form`;
$eb2fc0ee655eff6b$export$7f44db659563c8f4 = `A-HlBa_spectrum-Form--positionSide`;
$eb2fc0ee655eff6b$export$11b8bc81fe551057 = `A-HlBa_spectrum-Form--positionTop`;
$eb2fc0ee655eff6b$export$51ec27881a1e3928 = `A-HlBa_spectrum-Form-itemLabel`;
$eb2fc0ee655eff6b$export$79f9a5e9dd010c8d = `A-HlBa_spectrum-LabeledValue`;




function $38a9afbd39356629$var$Label(props, ref) {
    props = (0, $fXO85$reactspectrumprovider.useProviderProps)(props);
    let { children: children, labelPosition: labelPosition = "top", labelAlign: labelAlign = labelPosition === "side" ? "start" : null, isRequired: isRequired, necessityIndicator: necessityIndicator = isRequired != null ? "icon" : null, includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName = false, htmlFor: htmlFor, for: labelFor, elementType: ElementType = "label", onClick: onClick, ...otherProps } = props;
    let domRef = (0, $fXO85$reactspectrumutils.useDOMRef)(ref);
    let { styleProps: styleProps } = (0, $fXO85$reactspectrumutils.useStyleProps)(otherProps);
    let stringFormatter = (0, $fXO85$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($503911526ad50850$exports))), "@react-spectrum/label");
    let necessityLabel = isRequired ? stringFormatter.format("(required)") : stringFormatter.format("(optional)");
    let icon = /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, ($parcel$interopDefault($fXO85$spectrumiconsuiAsterisk))), {
        UNSAFE_className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-FieldLabel-requiredIcon"),
        "aria-label": includeNecessityIndicatorInAccessibilityName ? stringFormatter.format("(required)") : undefined
    });
    let labelClassNames = (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-FieldLabel", {
        "spectrum-FieldLabel--positionSide": labelPosition === "side",
        "spectrum-FieldLabel--alignEnd": labelAlign === "end"
    }, styleProps.className);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement(ElementType, {
        ...(0, $fXO85$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        onClick: onClick,
        ref: domRef,
        className: labelClassNames,
        htmlFor: ElementType === "label" ? labelFor || htmlFor : undefined
    }, children, (necessityIndicator === "label" || necessityIndicator === "icon" && isRequired) && " \u200B", necessityIndicator === "label" && /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement("span", {
        "aria-hidden": !includeNecessityIndicatorInAccessibilityName ? isRequired : undefined
    }, necessityLabel), necessityIndicator === "icon" && isRequired && icon);
}
let $38a9afbd39356629$export$b04be29aa201d4f5 = /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).forwardRef($38a9afbd39356629$var$Label);






function $0e5410371a445d59$var$Field(props, ref) {
    let formProps = (0, $fXO85$reactspectrumform.useFormProps)(props);
    let isInForm = formProps !== props;
    props = formProps;
    let { label: label, labelPosition: labelPosition = "top", labelAlign: labelAlign, isRequired: isRequired, necessityIndicator: necessityIndicator, includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName, validationState: validationState, isInvalid: isInvalid, description: description, errorMessage: errorMessage = (e)=>e.validationErrors.join(" "), validationErrors: validationErrors, validationDetails: validationDetails, isDisabled: isDisabled, showErrorIcon: showErrorIcon, contextualHelp: contextualHelp, children: children, labelProps: labelProps = {}, descriptionProps: // Not every component that uses <Field> supports help text.
    descriptionProps = {}, errorMessageProps: errorMessageProps = {}, elementType: elementType, wrapperClassName: wrapperClassName, wrapperProps: wrapperProps = {}, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $fXO85$reactspectrumutils.useStyleProps)(otherProps);
    let errorMessageString = null;
    if (typeof errorMessage === "function") errorMessageString = isInvalid != null && validationErrors != null && validationDetails != null ? errorMessage({
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails
    }) : null;
    else errorMessageString = errorMessage;
    let hasHelpText = !!description || errorMessageString && (isInvalid || validationState === "invalid");
    let contextualHelpId = (0, $fXO85$reactariautils.useId)();
    let fallbackLabelPropsId = (0, $fXO85$reactariautils.useId)();
    if (label && contextualHelp && !labelProps.id) labelProps.id = fallbackLabelPropsId;
    let labelWrapperClass = (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-Field", {
        "spectrum-Field--positionTop": labelPosition === "top",
        "spectrum-Field--positionSide": labelPosition === "side",
        "spectrum-Field--alignEnd": labelAlign === "end",
        "spectrum-Field--hasContextualHelp": !!props.contextualHelp
    }, styleProps.className, wrapperClassName);
    children = /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).cloneElement(children, (0, $fXO85$reactariautils.mergeProps)(children.props, {
        className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-Field-field")
    }));
    let renderHelpText = ()=>/*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, $2d1b5114e5b358b6$export$a67c0bc59081311a), {
            descriptionProps: descriptionProps,
            errorMessageProps: errorMessageProps,
            description: description,
            errorMessage: errorMessageString,
            validationState: validationState,
            isInvalid: isInvalid,
            isDisabled: isDisabled,
            showErrorIcon: showErrorIcon,
            gridArea: (0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))).helpText
        });
    let renderChildren = ()=>{
        if (labelPosition === "side") return /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, $fXO85$reactspectrumlayout.Flex), {
            direction: "column",
            UNSAFE_className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-Field-wrapper")
        }, children, hasHelpText && renderHelpText());
        return /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, ($parcel$interopDefault($fXO85$react))).Fragment, null, children, hasHelpText && renderHelpText());
    };
    let labelAndContextualHelp = /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, ($parcel$interopDefault($fXO85$react))).Fragment, null, label && /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, $38a9afbd39356629$export$b04be29aa201d4f5), {
        ...labelProps,
        labelPosition: labelPosition,
        labelAlign: labelAlign,
        isRequired: isRequired,
        necessityIndicator: necessityIndicator,
        includeNecessityIndicatorInAccessibilityName: includeNecessityIndicatorInAccessibilityName,
        elementType: elementType
    }, label), label && contextualHelp && /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement((0, $fXO85$reactspectrumutils.SlotProvider), {
        slots: {
            actionButton: {
                UNSAFE_className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-Field-contextualHelp"),
                id: contextualHelpId,
                "aria-labelledby": (labelProps === null || labelProps === void 0 ? void 0 : labelProps.id) ? `${labelProps.id} ${contextualHelpId}` : undefined
            }
        }
    }, contextualHelp));
    // Need to add an extra wrapper for the label and contextual help if labelPosition is side,
    // so that the table layout works inside forms.
    if (isInForm && labelPosition === "side" && label && contextualHelp) labelAndContextualHelp = /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement("div", {
        className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-Field-labelCell")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement("div", {
        className: (0, $fXO85$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-Field-labelWrapper")
    }, labelAndContextualHelp));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).createElement("div", {
        ...styleProps,
        ...wrapperProps,
        ref: ref,
        className: labelWrapperClass
    }, labelAndContextualHelp, renderChildren());
}
let $0e5410371a445d59$export$a455218a85c89869 = /*#__PURE__*/ (0, ($parcel$interopDefault($fXO85$react))).forwardRef($0e5410371a445d59$var$Field);
