// @react-spectrum/calendar/dist/main.js
"use client";

require("./main.css");
var $ce42e12d543b7e38$exports = require("./ar-AE.main.js");
var $cf1746301f7b3a2c$exports = require("./bg-BG.main.js");
var $2314514bf1e3a606$exports = require("./cs-CZ.main.js");
var $949c212f0c3f109b$exports = require("./da-DK.main.js");
var $7fd2c2494bdbc301$exports = require("./de-DE.main.js");
var $09ce79b5edaa2da5$exports = require("./el-GR.main.js");
var $5e9d23939f79d1b1$exports = require("./en-US.main.js");
var $f225659383b6c49c$exports = require("./es-ES.main.js");
var $c33e8e5f03cbf661$exports = require("./et-EE.main.js");
var $c4cafdbad7206639$exports = require("./fi-FI.main.js");
var $a99d0002fed1587c$exports = require("./fr-FR.main.js");
var $da92e613c17f6315$exports = require("./he-IL.main.js");
var $813c89c3343bf5d8$exports = require("./hr-HR.main.js");
var $e55eca3816269da0$exports = require("./hu-HU.main.js");
var $5bc7943c70fea77d$exports = require("./it-IT.main.js");
var $a229bcf1010fe509$exports = require("./ja-JP.main.js");
var $7a29f4d5c7513499$exports = require("./ko-KR.main.js");
var $cc5486cad5241d94$exports = require("./lt-LT.main.js");
var $c1c7136afde04f9d$exports = require("./lv-LV.main.js");
var $136b7b39137652a0$exports = require("./nb-NO.main.js");
var $12ce1e87f3d70b90$exports = require("./nl-NL.main.js");
var $85cf17eb58ae9432$exports = require("./pl-PL.main.js");
var $26ac8703247f4b49$exports = require("./pt-BR.main.js");
var $5afb437497ade41d$exports = require("./pt-PT.main.js");
var $db350c958ef4beb2$exports = require("./ro-RO.main.js");
var $77640ce2df581e8d$exports = require("./ru-RU.main.js");
var $fdbb191ea38137b2$exports = require("./sk-SK.main.js");
var $c0a1d77780011254$exports = require("./sl-SI.main.js");
var $178eeb83a4f423ba$exports = require("./sr-SP.main.js");
var $2d462724002952ef$exports = require("./sv-SE.main.js");
var $b81def5695dcf161$exports = require("./tr-TR.main.js");
var $15cf3fb890bd15fc$exports = require("./uk-UA.main.js");
var $7e919322f8380a78$exports = require("./zh-CN.main.js");
var $fdf4dbfb11dfcef7$exports = require("./zh-TW.main.js");
var $4BAFd$internationalizeddate = require("@internationalized/date");
var $4BAFd$reactspectrumutils = require("@react-spectrum/utils");
var $4BAFd$react = require("react");
var $4BAFd$reactariacalendar = require("@react-aria/calendar");
var $4BAFd$reactstatelycalendar = require("@react-stately/calendar");
var $4BAFd$reactariai18n = require("@react-aria/i18n");
var $4BAFd$reactspectrumprovider = require("@react-spectrum/provider");
var $4BAFd$reactspectrumbutton = require("@react-spectrum/button");
var $4BAFd$spectrumiconsuiChevronLeftLarge = require("@spectrum-icons/ui/ChevronLeftLarge");
var $4BAFd$spectrumiconsuiChevronRightLarge = require("@spectrum-icons/ui/ChevronRightLarge");
var $4BAFd$reactspectrumlabel = require("@react-spectrum/label");
var $4BAFd$reactariavisuallyhidden = require("@react-aria/visually-hidden");
var $4BAFd$reactariautils = require("@react-aria/utils");
var $4BAFd$reactariafocus = require("@react-aria/focus");
var $4BAFd$reactariainteractions = require("@react-aria/interactions");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Calendar", () => $d9490b77e8681e44$export$e1aef45b828286de);
$parcel$export(module.exports, "RangeCalendar", () => $4d045066f2092e3b$export$a4f5c8b89d277a8d);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $915891ae6206caad$exports = {};

$parcel$export($915891ae6206caad$exports, "focus-ring", () => $915891ae6206caad$export$f39a09f249340e2a, (v) => $915891ae6206caad$export$f39a09f249340e2a = v);
$parcel$export($915891ae6206caad$exports, "i18nFontFamily", () => $915891ae6206caad$export$8c4ee2c50c22c514, (v) => $915891ae6206caad$export$8c4ee2c50c22c514 = v);
$parcel$export($915891ae6206caad$exports, "is-disabled", () => $915891ae6206caad$export$d35bc1e505d1ebbf, (v) => $915891ae6206caad$export$d35bc1e505d1ebbf = v);
$parcel$export($915891ae6206caad$exports, "is-focused", () => $915891ae6206caad$export$e7dc768d35940237, (v) => $915891ae6206caad$export$e7dc768d35940237 = v);
$parcel$export($915891ae6206caad$exports, "is-hovered", () => $915891ae6206caad$export$b8813cd5d7824ce7, (v) => $915891ae6206caad$export$b8813cd5d7824ce7 = v);
$parcel$export($915891ae6206caad$exports, "is-invalid", () => $915891ae6206caad$export$14a30de2866685fa, (v) => $915891ae6206caad$export$14a30de2866685fa = v);
$parcel$export($915891ae6206caad$exports, "is-outsideMonth", () => $915891ae6206caad$export$93ff5b06817b3d41, (v) => $915891ae6206caad$export$93ff5b06817b3d41 = v);
$parcel$export($915891ae6206caad$exports, "is-pressed", () => $915891ae6206caad$export$15b73a232798b8d3, (v) => $915891ae6206caad$export$15b73a232798b8d3 = v);
$parcel$export($915891ae6206caad$exports, "is-range-end", () => $915891ae6206caad$export$52890a3cb1ab46b, (v) => $915891ae6206caad$export$52890a3cb1ab46b = v);
$parcel$export($915891ae6206caad$exports, "is-range-selection", () => $915891ae6206caad$export$cd2839d8bb0c6407, (v) => $915891ae6206caad$export$cd2839d8bb0c6407 = v);
$parcel$export($915891ae6206caad$exports, "is-range-start", () => $915891ae6206caad$export$4ce7d3d24e42e411, (v) => $915891ae6206caad$export$4ce7d3d24e42e411 = v);
$parcel$export($915891ae6206caad$exports, "is-selected", () => $915891ae6206caad$export$1e0fb04f31d3c22a, (v) => $915891ae6206caad$export$1e0fb04f31d3c22a = v);
$parcel$export($915891ae6206caad$exports, "is-selection-end", () => $915891ae6206caad$export$50722ae5663230f9, (v) => $915891ae6206caad$export$50722ae5663230f9 = v);
$parcel$export($915891ae6206caad$exports, "is-selection-start", () => $915891ae6206caad$export$f0091c9d6b8c19d3, (v) => $915891ae6206caad$export$f0091c9d6b8c19d3 = v);
$parcel$export($915891ae6206caad$exports, "is-today", () => $915891ae6206caad$export$189f6cc7eaf68d8b, (v) => $915891ae6206caad$export$189f6cc7eaf68d8b = v);
$parcel$export($915891ae6206caad$exports, "is-unavailable", () => $915891ae6206caad$export$38b08f1df825800e, (v) => $915891ae6206caad$export$38b08f1df825800e = v);
$parcel$export($915891ae6206caad$exports, "next", () => $915891ae6206caad$export$48cfd1e771a65c4f, (v) => $915891ae6206caad$export$48cfd1e771a65c4f = v);
$parcel$export($915891ae6206caad$exports, "prev", () => $915891ae6206caad$export$232faf9add678146, (v) => $915891ae6206caad$export$232faf9add678146 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar", () => $915891ae6206caad$export$2d9837ea36f7a875, (v) => $915891ae6206caad$export$2d9837ea36f7a875 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar--padded", () => $915891ae6206caad$export$8d90e3a3d21fe428, (v) => $915891ae6206caad$export$8d90e3a3d21fe428 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-body", () => $915891ae6206caad$export$33407a73687c0e62, (v) => $915891ae6206caad$export$33407a73687c0e62 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-date", () => $915891ae6206caad$export$c20fe43d14f8fa36, (v) => $915891ae6206caad$export$c20fe43d14f8fa36 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-dateText", () => $915891ae6206caad$export$cc48a6e73fdb7719, (v) => $915891ae6206caad$export$cc48a6e73fdb7719 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-dayOfWeek", () => $915891ae6206caad$export$7ff21fe81b21fa51, (v) => $915891ae6206caad$export$7ff21fe81b21fa51 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-header", () => $915891ae6206caad$export$7efb9b56af6e2d09, (v) => $915891ae6206caad$export$7efb9b56af6e2d09 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-monthHeader", () => $915891ae6206caad$export$185018e73e57145f, (v) => $915891ae6206caad$export$185018e73e57145f = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-months", () => $915891ae6206caad$export$5dacd232bfef6687, (v) => $915891ae6206caad$export$5dacd232bfef6687 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-nextMonth", () => $915891ae6206caad$export$7876229ab722b448, (v) => $915891ae6206caad$export$7876229ab722b448 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-prevMonth", () => $915891ae6206caad$export$d4c1c2a202a50db2, (v) => $915891ae6206caad$export$d4c1c2a202a50db2 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-table", () => $915891ae6206caad$export$27f4c38933cceedf, (v) => $915891ae6206caad$export$27f4c38933cceedf = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-tableCell", () => $915891ae6206caad$export$292b81cfd590634d, (v) => $915891ae6206caad$export$292b81cfd590634d = v);
$parcel$export($915891ae6206caad$exports, "spectrum-Calendar-title", () => $915891ae6206caad$export$98ee27555a2b7c01, (v) => $915891ae6206caad$export$98ee27555a2b7c01 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-FocusRing-ring", () => $915891ae6206caad$export$4109102f950813a6, (v) => $915891ae6206caad$export$4109102f950813a6 = v);
$parcel$export($915891ae6206caad$exports, "spectrum-FocusRing", () => $915891ae6206caad$export$24c7f46a6e3605dd, (v) => $915891ae6206caad$export$24c7f46a6e3605dd = v);
$parcel$export($915891ae6206caad$exports, "spectrum-FocusRing--quiet", () => $915891ae6206caad$export$2927016961429360, (v) => $915891ae6206caad$export$2927016961429360 = v);
$parcel$export($915891ae6206caad$exports, "title", () => $915891ae6206caad$export$fb184b623420d9be, (v) => $915891ae6206caad$export$fb184b623420d9be = v);
var $915891ae6206caad$export$f39a09f249340e2a;
var $915891ae6206caad$export$8c4ee2c50c22c514;
var $915891ae6206caad$export$d35bc1e505d1ebbf;
var $915891ae6206caad$export$e7dc768d35940237;
var $915891ae6206caad$export$b8813cd5d7824ce7;
var $915891ae6206caad$export$14a30de2866685fa;
var $915891ae6206caad$export$93ff5b06817b3d41;
var $915891ae6206caad$export$15b73a232798b8d3;
var $915891ae6206caad$export$52890a3cb1ab46b;
var $915891ae6206caad$export$cd2839d8bb0c6407;
var $915891ae6206caad$export$4ce7d3d24e42e411;
var $915891ae6206caad$export$1e0fb04f31d3c22a;
var $915891ae6206caad$export$50722ae5663230f9;
var $915891ae6206caad$export$f0091c9d6b8c19d3;
var $915891ae6206caad$export$189f6cc7eaf68d8b;
var $915891ae6206caad$export$38b08f1df825800e;
var $915891ae6206caad$export$48cfd1e771a65c4f;
var $915891ae6206caad$export$232faf9add678146;
var $915891ae6206caad$export$2d9837ea36f7a875;
var $915891ae6206caad$export$8d90e3a3d21fe428;
var $915891ae6206caad$export$33407a73687c0e62;
var $915891ae6206caad$export$c20fe43d14f8fa36;
var $915891ae6206caad$export$cc48a6e73fdb7719;
var $915891ae6206caad$export$7ff21fe81b21fa51;
var $915891ae6206caad$export$7efb9b56af6e2d09;
var $915891ae6206caad$export$185018e73e57145f;
var $915891ae6206caad$export$5dacd232bfef6687;
var $915891ae6206caad$export$7876229ab722b448;
var $915891ae6206caad$export$d4c1c2a202a50db2;
var $915891ae6206caad$export$27f4c38933cceedf;
var $915891ae6206caad$export$292b81cfd590634d;
var $915891ae6206caad$export$98ee27555a2b7c01;
var $915891ae6206caad$export$4109102f950813a6;
var $915891ae6206caad$export$24c7f46a6e3605dd;
var $915891ae6206caad$export$2927016961429360;
var $915891ae6206caad$export$fb184b623420d9be;
$915891ae6206caad$export$f39a09f249340e2a = `EuwYBq_focus-ring`;
$915891ae6206caad$export$8c4ee2c50c22c514 = `EuwYBq_i18nFontFamily`;
$915891ae6206caad$export$d35bc1e505d1ebbf = `EuwYBq_is-disabled`;
$915891ae6206caad$export$e7dc768d35940237 = `EuwYBq_is-focused`;
$915891ae6206caad$export$b8813cd5d7824ce7 = `EuwYBq_is-hovered`;
$915891ae6206caad$export$14a30de2866685fa = `EuwYBq_is-invalid`;
$915891ae6206caad$export$93ff5b06817b3d41 = `EuwYBq_is-outsideMonth`;
$915891ae6206caad$export$15b73a232798b8d3 = `EuwYBq_is-pressed`;
$915891ae6206caad$export$52890a3cb1ab46b = `EuwYBq_is-range-end`;
$915891ae6206caad$export$cd2839d8bb0c6407 = `EuwYBq_is-range-selection`;
$915891ae6206caad$export$4ce7d3d24e42e411 = `EuwYBq_is-range-start`;
$915891ae6206caad$export$1e0fb04f31d3c22a = `EuwYBq_is-selected`;
$915891ae6206caad$export$50722ae5663230f9 = `EuwYBq_is-selection-end`;
$915891ae6206caad$export$f0091c9d6b8c19d3 = `EuwYBq_is-selection-start`;
$915891ae6206caad$export$189f6cc7eaf68d8b = `EuwYBq_is-today`;
$915891ae6206caad$export$38b08f1df825800e = `EuwYBq_is-unavailable`;
$915891ae6206caad$export$48cfd1e771a65c4f = `EuwYBq_next`;
$915891ae6206caad$export$232faf9add678146 = `EuwYBq_prev`;
$915891ae6206caad$export$2d9837ea36f7a875 = `EuwYBq_spectrum-Calendar`;
$915891ae6206caad$export$8d90e3a3d21fe428 = `EuwYBq_spectrum-Calendar--padded`;
$915891ae6206caad$export$33407a73687c0e62 = `EuwYBq_spectrum-Calendar-body`;
$915891ae6206caad$export$c20fe43d14f8fa36 = `EuwYBq_spectrum-Calendar-date`;
$915891ae6206caad$export$cc48a6e73fdb7719 = `EuwYBq_spectrum-Calendar-dateText`;
$915891ae6206caad$export$7ff21fe81b21fa51 = `EuwYBq_spectrum-Calendar-dayOfWeek`;
$915891ae6206caad$export$7efb9b56af6e2d09 = `EuwYBq_spectrum-Calendar-header`;
$915891ae6206caad$export$185018e73e57145f = `EuwYBq_spectrum-Calendar-monthHeader`;
$915891ae6206caad$export$5dacd232bfef6687 = `EuwYBq_spectrum-Calendar-months`;
$915891ae6206caad$export$7876229ab722b448 = `EuwYBq_spectrum-Calendar-nextMonth`;
$915891ae6206caad$export$d4c1c2a202a50db2 = `EuwYBq_spectrum-Calendar-prevMonth`;
$915891ae6206caad$export$27f4c38933cceedf = `EuwYBq_spectrum-Calendar-table`;
$915891ae6206caad$export$292b81cfd590634d = `EuwYBq_spectrum-Calendar-tableCell`;
$915891ae6206caad$export$98ee27555a2b7c01 = `EuwYBq_spectrum-Calendar-title`;
$915891ae6206caad$export$4109102f950813a6 = `EuwYBq_spectrum-FocusRing-ring`;
$915891ae6206caad$export$24c7f46a6e3605dd = `EuwYBq_spectrum-FocusRing ${$915891ae6206caad$export$4109102f950813a6}`;
$915891ae6206caad$export$2927016961429360 = `EuwYBq_spectrum-FocusRing--quiet`;
$915891ae6206caad$export$fb184b623420d9be = `EuwYBq_title`;





function $52c7c3c887827b64$export$5d847498420df57b({ state: state, currentMonth: currentMonth, ...props }) {
    let ref = (0, $4BAFd$react.useRef)();
    let { cellProps: cellProps, buttonProps: buttonProps, isPressed: isPressed, isSelected: isSelected, isDisabled: isDisabled, isFocused: isFocused, isInvalid: isInvalid, formattedDate: formattedDate } = (0, $4BAFd$reactariacalendar.useCalendarCell)({
        ...props,
        isDisabled: !(0, $4BAFd$internationalizeddate.isSameMonth)(props.date, currentMonth)
    }, state, ref);
    let isUnavailable = state.isCellUnavailable(props.date) && !isDisabled;
    let isLastSelectedBeforeDisabled = !isDisabled && !isInvalid && state.isCellUnavailable(props.date.add({
        days: 1
    }));
    let isFirstSelectedAfterDisabled = !isDisabled && !isInvalid && state.isCellUnavailable(props.date.subtract({
        days: 1
    }));
    let highlightedRange = "highlightedRange" in state && state.highlightedRange;
    let isSelectionStart = isSelected && highlightedRange && (0, $4BAFd$internationalizeddate.isSameDay)(props.date, highlightedRange.start);
    let isSelectionEnd = isSelected && highlightedRange && (0, $4BAFd$internationalizeddate.isSameDay)(props.date, highlightedRange.end);
    let { locale: locale } = (0, $4BAFd$reactariai18n.useLocale)();
    let dayOfWeek = (0, $4BAFd$internationalizeddate.getDayOfWeek)(props.date, locale);
    let isRangeStart = isSelected && (isFirstSelectedAfterDisabled || dayOfWeek === 0 || props.date.day === 1);
    let isRangeEnd = isSelected && (isLastSelectedBeforeDisabled || dayOfWeek === 6 || props.date.day === currentMonth.calendar.getDaysInMonth(currentMonth));
    let { focusProps: focusProps, isFocusVisible: isFocusVisible } = (0, $4BAFd$reactariafocus.useFocusRing)();
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $4BAFd$reactariainteractions.useHover)({
        isDisabled: isDisabled || isUnavailable || state.isReadOnly
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("td", {
        ...cellProps,
        className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-tableCell")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("span", {
        ...(0, $4BAFd$reactariautils.mergeProps)(buttonProps, hoverProps, focusProps),
        ref: ref,
        className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-date", {
            "is-today": (0, $4BAFd$internationalizeddate.isToday)(props.date, state.timeZone),
            "is-selected": isSelected,
            "is-focused": isFocused && isFocusVisible,
            // Style disabled (i.e. out of min/max range), but selected dates as unavailable
            // since it is more clear than trying to dim the selection.
            "is-disabled": isDisabled && !isInvalid,
            "is-unavailable": isUnavailable || isInvalid && isDisabled,
            "is-outsideMonth": !(0, $4BAFd$internationalizeddate.isSameMonth)(props.date, currentMonth),
            "is-range-start": isRangeStart,
            "is-range-end": isRangeEnd,
            "is-range-selection": isSelected && "highlightedRange" in state,
            "is-selection-start": isSelectionStart,
            "is-selection-end": isSelectionEnd,
            "is-hovered": isHovered,
            "is-pressed": isPressed && !state.isReadOnly,
            "is-invalid": isInvalid
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("span", {
        className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-dateText")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("span", null, formattedDate))));
}








function $02c19eea15a72a7b$export$26e2752316b9a375(props) {
    let { state: state, startDate: startDate } = props;
    let { gridProps: gridProps, headerProps: headerProps, weekDays: weekDays } = (0, $4BAFd$reactariacalendar.useCalendarGrid)({
        ...props,
        endDate: (0, $4BAFd$internationalizeddate.endOfMonth)(startDate)
    }, state);
    let { locale: locale } = (0, $4BAFd$reactariai18n.useLocale)();
    let weeksInMonth = (0, $4BAFd$internationalizeddate.getWeeksInMonth)(startDate, locale);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("table", {
        ...gridProps,
        className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-body", "spectrum-Calendar-table")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("thead", headerProps, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("tr", null, weekDays.map((day, index)=>/*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("th", {
            key: index,
            className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-tableCell")
        }, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("span", {
            className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-dayOfWeek")
        }, day))))), /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("tbody", null, [
        ...new Array(weeksInMonth).keys()
    ].map((weekIndex)=>/*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("tr", {
            key: weekIndex
        }, state.getDatesInWeek(weekIndex, startDate).map((date, i)=>date ? /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $52c7c3c887827b64$export$5d847498420df57b), {
                key: i,
                state: state,
                date: date,
                currentMonth: startDate
            }) : /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("td", {
                key: i
            }))))));
}






var $7e7c3b1f5233ce23$exports = {};


































$7e7c3b1f5233ce23$exports = {
    "ar-AE": $ce42e12d543b7e38$exports,
    "bg-BG": $cf1746301f7b3a2c$exports,
    "cs-CZ": $2314514bf1e3a606$exports,
    "da-DK": $949c212f0c3f109b$exports,
    "de-DE": $7fd2c2494bdbc301$exports,
    "el-GR": $09ce79b5edaa2da5$exports,
    "en-US": $5e9d23939f79d1b1$exports,
    "es-ES": $f225659383b6c49c$exports,
    "et-EE": $c33e8e5f03cbf661$exports,
    "fi-FI": $c4cafdbad7206639$exports,
    "fr-FR": $a99d0002fed1587c$exports,
    "he-IL": $da92e613c17f6315$exports,
    "hr-HR": $813c89c3343bf5d8$exports,
    "hu-HU": $e55eca3816269da0$exports,
    "it-IT": $5bc7943c70fea77d$exports,
    "ja-JP": $a229bcf1010fe509$exports,
    "ko-KR": $7a29f4d5c7513499$exports,
    "lt-LT": $cc5486cad5241d94$exports,
    "lv-LV": $c1c7136afde04f9d$exports,
    "nb-NO": $136b7b39137652a0$exports,
    "nl-NL": $12ce1e87f3d70b90$exports,
    "pl-PL": $85cf17eb58ae9432$exports,
    "pt-BR": $26ac8703247f4b49$exports,
    "pt-PT": $5afb437497ade41d$exports,
    "ro-RO": $db350c958ef4beb2$exports,
    "ru-RU": $77640ce2df581e8d$exports,
    "sk-SK": $fdbb191ea38137b2$exports,
    "sl-SI": $c0a1d77780011254$exports,
    "sr-SP": $178eeb83a4f423ba$exports,
    "sv-SE": $2d462724002952ef$exports,
    "tr-TR": $b81def5695dcf161$exports,
    "uk-UA": $15cf3fb890bd15fc$exports,
    "zh-CN": $7e919322f8380a78$exports,
    "zh-TW": $fdf4dbfb11dfcef7$exports
};






function $6d416018158e3a26$export$bfd52a43017368fe(props) {
    let { state: state, calendarProps: calendarProps, nextButtonProps: nextButtonProps, prevButtonProps: prevButtonProps, errorMessageProps: errorMessageProps, calendarRef: ref, visibleMonths: visibleMonths = 1 } = props;
    let { styleProps: styleProps } = (0, $4BAFd$reactspectrumutils.useStyleProps)(props);
    let stringFormatter = (0, $4BAFd$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($7e7c3b1f5233ce23$exports))), "@react-spectrum/calendar");
    let { direction: direction } = (0, $4BAFd$reactariai18n.useLocale)();
    let currentMonth = state.visibleRange.start;
    let monthDateFormatter = (0, $4BAFd$reactariai18n.useDateFormatter)({
        month: "long",
        year: "numeric",
        era: currentMonth.calendar.identifier === "gregory" && currentMonth.era === "BC" ? "short" : undefined,
        calendar: currentMonth.calendar.identifier,
        timeZone: state.timeZone
    });
    let titles = [];
    let calendars = [];
    for(let i = 0; i < visibleMonths; i++){
        let d = currentMonth.add({
            months: i
        });
        titles.push(/*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("div", {
            key: i,
            className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-monthHeader")
        }, i === 0 && /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $4BAFd$reactspectrumbutton.ActionButton), {
            ...prevButtonProps,
            UNSAFE_className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-prevMonth"),
            isQuiet: true
        }, direction === "rtl" ? /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, ($parcel$interopDefault($4BAFd$spectrumiconsuiChevronRightLarge))), null) : /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, ($parcel$interopDefault($4BAFd$spectrumiconsuiChevronLeftLarge))), null)), /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("h2", {
            // We have a visually hidden heading describing the entire visible range,
            // and the calendar itself describes the individual month
            // so we don't need to repeat that here for screen reader users.
            "aria-hidden": true,
            className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-title")
        }, monthDateFormatter.format(d.toDate(state.timeZone))), i === visibleMonths - 1 && /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $4BAFd$reactspectrumbutton.ActionButton), {
            ...nextButtonProps,
            UNSAFE_className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-nextMonth"),
            isQuiet: true
        }, direction === "rtl" ? /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, ($parcel$interopDefault($4BAFd$spectrumiconsuiChevronLeftLarge))), null) : /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, ($parcel$interopDefault($4BAFd$spectrumiconsuiChevronRightLarge))), null))));
        calendars.push(/*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $02c19eea15a72a7b$export$26e2752316b9a375), {
            ...props,
            key: i,
            state: state,
            startDate: d
        }));
    }
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("div", {
        ...styleProps,
        ...calendarProps,
        ref: ref,
        className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar", styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $4BAFd$reactariavisuallyhidden.VisuallyHidden), null, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("h2", null, calendarProps["aria-label"])), /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("div", {
        className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-header")
    }, titles), /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("div", {
        className: (0, $4BAFd$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($915891ae6206caad$exports))), "spectrum-Calendar-months")
    }, calendars), /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $4BAFd$reactariavisuallyhidden.VisuallyHidden), null, /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement("button", {
        "aria-label": nextButtonProps["aria-label"],
        disabled: nextButtonProps.isDisabled,
        onClick: ()=>state.focusNextPage(),
        tabIndex: -1
    })), state.isValueInvalid && /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $4BAFd$reactspectrumlabel.HelpText), {
        showErrorIcon: true,
        errorMessage: props.errorMessage || stringFormatter.format("invalidSelection", {
            selectedCount: "highlightedRange" in state ? 2 : 1
        }),
        errorMessageProps: errorMessageProps,
        isInvalid: true,
        // Intentionally a global class name so it can be targeted in DatePicker CSS...
        UNSAFE_className: "spectrum-Calendar-helpText"
    }));
}









function $d9490b77e8681e44$var$Calendar(props, ref) {
    props = (0, $4BAFd$reactspectrumprovider.useProviderProps)(props);
    let { visibleMonths: visibleMonths = 1 } = props;
    visibleMonths = Math.max(visibleMonths, 1);
    let visibleDuration = (0, $4BAFd$react.useMemo)(()=>({
            months: visibleMonths
        }), [
        visibleMonths
    ]);
    let { locale: locale } = (0, $4BAFd$reactariai18n.useLocale)();
    let state = (0, $4BAFd$reactstatelycalendar.useCalendarState)({
        ...props,
        locale: locale,
        visibleDuration: visibleDuration,
        createCalendar: $4BAFd$internationalizeddate.createCalendar
    });
    let domRef = (0, $4BAFd$react.useRef)();
    (0, $4BAFd$react.useImperativeHandle)(ref, ()=>({
            ...(0, $4BAFd$reactspectrumutils.createDOMRef)(domRef),
            focus () {
                state.setFocused(true);
            }
        }));
    let { calendarProps: calendarProps, prevButtonProps: prevButtonProps, nextButtonProps: nextButtonProps, errorMessageProps: errorMessageProps } = (0, $4BAFd$reactariacalendar.useCalendar)(props, state);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $6d416018158e3a26$export$bfd52a43017368fe), {
        ...props,
        visibleMonths: visibleMonths,
        state: state,
        calendarRef: domRef,
        calendarProps: calendarProps,
        prevButtonProps: prevButtonProps,
        nextButtonProps: nextButtonProps,
        errorMessageProps: errorMessageProps
    });
}
/**
 * Calendars display a grid of days in one or more months and allow users to select a single date.
 */ const $d9490b77e8681e44$export$e1aef45b828286de = /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).forwardRef($d9490b77e8681e44$var$Calendar);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







function $4d045066f2092e3b$var$RangeCalendar(props, ref) {
    props = (0, $4BAFd$reactspectrumprovider.useProviderProps)(props);
    let { visibleMonths: visibleMonths = 1 } = props;
    visibleMonths = Math.max(visibleMonths, 1);
    let visibleDuration = (0, $4BAFd$react.useMemo)(()=>({
            months: visibleMonths
        }), [
        visibleMonths
    ]);
    let { locale: locale } = (0, $4BAFd$reactariai18n.useLocale)();
    let state = (0, $4BAFd$reactstatelycalendar.useRangeCalendarState)({
        ...props,
        locale: locale,
        visibleDuration: visibleDuration,
        createCalendar: $4BAFd$internationalizeddate.createCalendar
    });
    let domRef = (0, $4BAFd$react.useRef)();
    (0, $4BAFd$react.useImperativeHandle)(ref, ()=>({
            ...(0, $4BAFd$reactspectrumutils.createDOMRef)(domRef),
            focus () {
                state.setFocused(true);
            }
        }));
    let { calendarProps: calendarProps, prevButtonProps: prevButtonProps, nextButtonProps: nextButtonProps, errorMessageProps: errorMessageProps } = (0, $4BAFd$reactariacalendar.useRangeCalendar)(props, state, domRef);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).createElement((0, $6d416018158e3a26$export$bfd52a43017368fe), {
        ...props,
        visibleMonths: visibleMonths,
        state: state,
        calendarRef: domRef,
        calendarProps: calendarProps,
        prevButtonProps: prevButtonProps,
        nextButtonProps: nextButtonProps,
        errorMessageProps: errorMessageProps
    });
}
/**
 * RangeCalendars display a grid of days in one or more months and allow users to select a contiguous range of dates.
 */ const $4d045066f2092e3b$export$a4f5c8b89d277a8d = /*#__PURE__*/ (0, ($parcel$interopDefault($4BAFd$react))).forwardRef($4d045066f2092e3b$var$RangeCalendar);
