// @react-spectrum/avatar/dist/main.js
"use client";

require("./main.css");
var $hTGwy$reactspectrumutils = require("@react-spectrum/utils");
var $hTGwy$reactariautils = require("@react-aria/utils");
var $hTGwy$react = require("react");
var $hTGwy$reactspectrumprovider = require("@react-spectrum/provider");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Avatar", () => $cc830936d47bae2b$export$e2255cf6045e8d47);
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2021 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $d22788318b5f79eb$exports = {};

$parcel$export($d22788318b5f79eb$exports, "is-disabled", () => $d22788318b5f79eb$export$d35bc1e505d1ebbf, (v) => $d22788318b5f79eb$export$d35bc1e505d1ebbf = v);
$parcel$export($d22788318b5f79eb$exports, "spectrum-Avatar", () => $d22788318b5f79eb$export$2e049b8d031600fb, (v) => $d22788318b5f79eb$export$2e049b8d031600fb = v);
var $d22788318b5f79eb$export$d35bc1e505d1ebbf;
var $d22788318b5f79eb$export$2e049b8d031600fb;
$d22788318b5f79eb$export$d35bc1e505d1ebbf = `IFaZJG_is-disabled`;
$d22788318b5f79eb$export$2e049b8d031600fb = `IFaZJG_spectrum-Avatar`;



const $cc830936d47bae2b$var$DEFAULT_SIZE = "avatar-size-100";
const $cc830936d47bae2b$var$SIZE_RE = /^size-\d+/;
function $cc830936d47bae2b$var$Avatar(props, ref) {
    props = (0, $hTGwy$reactspectrumutils.useSlotProps)(props, "avatar");
    const { alt: alt = "", isDisabled: isDisabled, size: size = $cc830936d47bae2b$var$DEFAULT_SIZE, src: src, ...otherProps } = (0, $hTGwy$reactspectrumprovider.useProviderProps)(props);
    const { styleProps: styleProps } = (0, $hTGwy$reactspectrumutils.useStyleProps)(otherProps);
    const domRef = (0, $hTGwy$reactspectrumutils.useDOMRef)(ref);
    const domProps = (0, $hTGwy$reactariautils.filterDOMProps)(otherProps);
    // Casting `size` as `any` since `isNaN` expects a `number`, but we want it
    // to handle `string` numbers; e.g. '300' as opposed to 300
    const sizeValue = typeof size !== "number" && ($cc830936d47bae2b$var$SIZE_RE.test(size) || !isNaN(size)) ? (0, $hTGwy$reactspectrumutils.dimensionValue)($cc830936d47bae2b$var$DEFAULT_SIZE) // override disallowed size values
     : (0, $hTGwy$reactspectrumutils.dimensionValue)(size || $cc830936d47bae2b$var$DEFAULT_SIZE);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($hTGwy$react))).createElement("img", {
        ...styleProps,
        ...domProps,
        alt: alt,
        className: (0, $hTGwy$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($d22788318b5f79eb$exports))), "spectrum-Avatar", {
            "is-disabled": isDisabled
        }, styleProps.className),
        ref: domRef,
        src: src,
        style: {
            ...styleProps.style,
            ...sizeValue && {
                height: sizeValue,
                width: sizeValue
            }
        }
    });
}
/**
 * An avatar is a thumbnail representation of an entity, such as a user or an organization.
 */ const $cc830936d47bae2b$export$e2255cf6045e8d47 = /*#__PURE__*/ (0, $hTGwy$react.forwardRef)($cc830936d47bae2b$var$Avatar);
