// @react-spectrum/view/dist/main.js
"use client";

var $3YgEH$reactspectrumutils = require("@react-spectrum/utils");
var $3YgEH$reactariautils = require("@react-aria/utils");
var $3YgEH$react = require("react");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "View", () => $6c848be8e8bffe98$export$27a5bd065ad55220);
$parcel$export(module.exports, "Content", () => $2d8ae89d4e4c1d41$export$7c6e2c02157bb7d2);
$parcel$export(module.exports, "Footer", () => $e266c0b33fd8525d$export$a06f1c675e846f6f);
$parcel$export(module.exports, "Header", () => $62d558b43ecd7a77$export$8b251419efc915eb);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $6c848be8e8bffe98$var$View(props, ref) {
    props = (0, $3YgEH$reactspectrumutils.useSlotProps)(props);
    let { elementType: ElementType = "div", children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $3YgEH$reactspectrumutils.useStyleProps)(props, (0, $3YgEH$reactspectrumutils.viewStyleProps));
    let domRef = (0, $3YgEH$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3YgEH$react))).createElement(ElementType, {
        ...(0, $3YgEH$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3YgEH$react))).createElement((0, $3YgEH$reactspectrumutils.ClearSlots), null, children));
}
/**
 * View is a general purpose container with no specific semantics that can be used for custom styling purposes.
 * It supports Spectrum style props to ensure consistency with other Spectrum components.
 */ const $6c848be8e8bffe98$export$27a5bd065ad55220 = /*#__PURE__*/ (0, $3YgEH$react.forwardRef)($6c848be8e8bffe98$var$View);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $2d8ae89d4e4c1d41$var$Content(props, ref) {
    props = (0, $3YgEH$reactspectrumutils.useSlotProps)(props, "content");
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $3YgEH$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $3YgEH$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3YgEH$react))).createElement("section", {
        ...(0, $3YgEH$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3YgEH$react))).createElement((0, $3YgEH$reactspectrumutils.ClearSlots), null, children));
}
/**
 * Content represents the primary content within a Spectrum container.
 */ const $2d8ae89d4e4c1d41$export$7c6e2c02157bb7d2 = /*#__PURE__*/ (0, $3YgEH$react.forwardRef)($2d8ae89d4e4c1d41$var$Content);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $e266c0b33fd8525d$var$Footer(props, ref) {
    props = (0, $3YgEH$reactspectrumutils.useSlotProps)(props, "footer");
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $3YgEH$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $3YgEH$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3YgEH$react))).createElement("footer", {
        ...(0, $3YgEH$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3YgEH$react))).createElement((0, $3YgEH$reactspectrumutils.ClearSlots), null, children));
}
/**
 * Footer represents a footer within a Spectrum container.
 */ const $e266c0b33fd8525d$export$a06f1c675e846f6f = /*#__PURE__*/ (0, $3YgEH$react.forwardRef)($e266c0b33fd8525d$var$Footer);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


function $62d558b43ecd7a77$var$Header(props, ref) {
    props = (0, $3YgEH$reactspectrumutils.useSlotProps)(props, "header");
    let { children: children, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $3YgEH$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $3YgEH$reactspectrumutils.useDOMRef)(ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3YgEH$react))).createElement("header", {
        ...(0, $3YgEH$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3YgEH$react))).createElement((0, $3YgEH$reactspectrumutils.ClearSlots), null, children));
}
/**
 * Header represents a header within a Spectrum container.
 */ const $62d558b43ecd7a77$export$8b251419efc915eb = /*#__PURE__*/ (0, $3YgEH$react.forwardRef)($62d558b43ecd7a77$var$Header);
