// @react-spectrum/badge/dist/main.js
"use client";

require("./main.css");
var $jLeOL$reactspectrumutils = require("@react-spectrum/utils");
var $jLeOL$reactariautils = require("@react-aria/utils");
var $jLeOL$react = require("react");
var $jLeOL$reactspectrumtext = require("@react-spectrum/text");
var $jLeOL$reactspectrumprovider = require("@react-spectrum/provider");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Badge", () => $f243611f87a6fe61$export$37acb3580601e69a);
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $0f362c72c73357fc$exports = {};

$parcel$export($0f362c72c73357fc$exports, "focus-ring", () => $0f362c72c73357fc$export$f39a09f249340e2a, (v) => $0f362c72c73357fc$export$f39a09f249340e2a = v);
$parcel$export($0f362c72c73357fc$exports, "i18nFontFamily", () => $0f362c72c73357fc$export$8c4ee2c50c22c514, (v) => $0f362c72c73357fc$export$8c4ee2c50c22c514 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge", () => $0f362c72c73357fc$export$acb311e43ff4d1, (v) => $0f362c72c73357fc$export$acb311e43ff4d1 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--fuchsia", () => $0f362c72c73357fc$export$17d8c34537918c7, (v) => $0f362c72c73357fc$export$17d8c34537918c7 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--indigo", () => $0f362c72c73357fc$export$cf21c81baa11bb0f, (v) => $0f362c72c73357fc$export$cf21c81baa11bb0f = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--info", () => $0f362c72c73357fc$export$a276b7cecb1e7608, (v) => $0f362c72c73357fc$export$a276b7cecb1e7608 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--large", () => $0f362c72c73357fc$export$8e48598848736b6e, (v) => $0f362c72c73357fc$export$8e48598848736b6e = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--magenta", () => $0f362c72c73357fc$export$a6258e5c939933e, (v) => $0f362c72c73357fc$export$a6258e5c939933e = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--negative", () => $0f362c72c73357fc$export$250de4fc6219f4ba, (v) => $0f362c72c73357fc$export$250de4fc6219f4ba = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--neutral", () => $0f362c72c73357fc$export$ce0abe6a13066aee, (v) => $0f362c72c73357fc$export$ce0abe6a13066aee = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--positive", () => $0f362c72c73357fc$export$9cb54bc972cacc06, (v) => $0f362c72c73357fc$export$9cb54bc972cacc06 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--purple", () => $0f362c72c73357fc$export$67ff21277f017a3f, (v) => $0f362c72c73357fc$export$67ff21277f017a3f = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--seafoam", () => $0f362c72c73357fc$export$b9072ec5c3c3196e, (v) => $0f362c72c73357fc$export$b9072ec5c3c3196e = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--small", () => $0f362c72c73357fc$export$e76d1f315203cda4, (v) => $0f362c72c73357fc$export$e76d1f315203cda4 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge--yellow", () => $0f362c72c73357fc$export$f39824621eafe7d1, (v) => $0f362c72c73357fc$export$f39824621eafe7d1 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge-icon", () => $0f362c72c73357fc$export$70ebbcb6132f4aa8, (v) => $0f362c72c73357fc$export$70ebbcb6132f4aa8 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-Badge-label", () => $0f362c72c73357fc$export$87cc42d7dd800a50, (v) => $0f362c72c73357fc$export$87cc42d7dd800a50 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-FocusRing-ring", () => $0f362c72c73357fc$export$4109102f950813a6, (v) => $0f362c72c73357fc$export$4109102f950813a6 = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-FocusRing", () => $0f362c72c73357fc$export$24c7f46a6e3605dd, (v) => $0f362c72c73357fc$export$24c7f46a6e3605dd = v);
$parcel$export($0f362c72c73357fc$exports, "spectrum-FocusRing--quiet", () => $0f362c72c73357fc$export$2927016961429360, (v) => $0f362c72c73357fc$export$2927016961429360 = v);
var $0f362c72c73357fc$export$f39a09f249340e2a;
var $0f362c72c73357fc$export$8c4ee2c50c22c514;
var $0f362c72c73357fc$export$acb311e43ff4d1;
var $0f362c72c73357fc$export$17d8c34537918c7;
var $0f362c72c73357fc$export$cf21c81baa11bb0f;
var $0f362c72c73357fc$export$a276b7cecb1e7608;
var $0f362c72c73357fc$export$8e48598848736b6e;
var $0f362c72c73357fc$export$a6258e5c939933e;
var $0f362c72c73357fc$export$250de4fc6219f4ba;
var $0f362c72c73357fc$export$ce0abe6a13066aee;
var $0f362c72c73357fc$export$9cb54bc972cacc06;
var $0f362c72c73357fc$export$67ff21277f017a3f;
var $0f362c72c73357fc$export$b9072ec5c3c3196e;
var $0f362c72c73357fc$export$e76d1f315203cda4;
var $0f362c72c73357fc$export$f39824621eafe7d1;
var $0f362c72c73357fc$export$70ebbcb6132f4aa8;
var $0f362c72c73357fc$export$87cc42d7dd800a50;
var $0f362c72c73357fc$export$4109102f950813a6;
var $0f362c72c73357fc$export$24c7f46a6e3605dd;
var $0f362c72c73357fc$export$2927016961429360;
$0f362c72c73357fc$export$f39a09f249340e2a = `CIB44a_focus-ring`;
$0f362c72c73357fc$export$8c4ee2c50c22c514 = `CIB44a_i18nFontFamily`;
$0f362c72c73357fc$export$acb311e43ff4d1 = `CIB44a_spectrum-Badge`;
$0f362c72c73357fc$export$17d8c34537918c7 = `CIB44a_spectrum-Badge--fuchsia`;
$0f362c72c73357fc$export$cf21c81baa11bb0f = `CIB44a_spectrum-Badge--indigo`;
$0f362c72c73357fc$export$a276b7cecb1e7608 = `CIB44a_spectrum-Badge--info`;
$0f362c72c73357fc$export$8e48598848736b6e = `CIB44a_spectrum-Badge--large`;
$0f362c72c73357fc$export$a6258e5c939933e = `CIB44a_spectrum-Badge--magenta`;
$0f362c72c73357fc$export$250de4fc6219f4ba = `CIB44a_spectrum-Badge--negative`;
$0f362c72c73357fc$export$ce0abe6a13066aee = `CIB44a_spectrum-Badge--neutral`;
$0f362c72c73357fc$export$9cb54bc972cacc06 = `CIB44a_spectrum-Badge--positive`;
$0f362c72c73357fc$export$67ff21277f017a3f = `CIB44a_spectrum-Badge--purple`;
$0f362c72c73357fc$export$b9072ec5c3c3196e = `CIB44a_spectrum-Badge--seafoam`;
$0f362c72c73357fc$export$e76d1f315203cda4 = `CIB44a_spectrum-Badge--small`;
$0f362c72c73357fc$export$f39824621eafe7d1 = `CIB44a_spectrum-Badge--yellow`;
$0f362c72c73357fc$export$70ebbcb6132f4aa8 = `CIB44a_spectrum-Badge-icon`;
$0f362c72c73357fc$export$87cc42d7dd800a50 = `CIB44a_spectrum-Badge-label`;
$0f362c72c73357fc$export$4109102f950813a6 = `CIB44a_spectrum-FocusRing-ring`;
$0f362c72c73357fc$export$24c7f46a6e3605dd = `CIB44a_spectrum-FocusRing ${$0f362c72c73357fc$export$4109102f950813a6}`;
$0f362c72c73357fc$export$2927016961429360 = `CIB44a_spectrum-FocusRing--quiet`;




function $f243611f87a6fe61$var$Badge(props, ref) {
    let { children: children, variant: variant, ...otherProps } = (0, $jLeOL$reactspectrumprovider.useProviderProps)(props);
    let domRef = (0, $jLeOL$reactspectrumutils.useDOMRef)(ref);
    let { styleProps: styleProps } = (0, $jLeOL$reactspectrumutils.useStyleProps)(otherProps);
    let isTextOnly = (0, ($parcel$interopDefault($jLeOL$react))).Children.toArray(props.children).every((c)=>!/*#__PURE__*/ (0, ($parcel$interopDefault($jLeOL$react))).isValidElement(c));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($jLeOL$react))).createElement("span", {
        ...(0, $jLeOL$reactariautils.filterDOMProps)(otherProps),
        ...styleProps,
        role: "presentation",
        className: (0, $jLeOL$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($0f362c72c73357fc$exports))), "spectrum-Badge", {
            [`spectrum-Badge--${variant}`]: variant
        }, styleProps.className),
        ref: domRef
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($jLeOL$react))).createElement((0, $jLeOL$reactspectrumutils.SlotProvider), {
        slots: {
            icon: {
                size: "S",
                UNSAFE_className: (0, $jLeOL$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($0f362c72c73357fc$exports))), "spectrum-Badge-icon")
            },
            text: {
                UNSAFE_className: (0, $jLeOL$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($0f362c72c73357fc$exports))), "spectrum-Badge-label")
            }
        }
    }, typeof children === "string" || isTextOnly ? /*#__PURE__*/ (0, ($parcel$interopDefault($jLeOL$react))).createElement((0, $jLeOL$reactspectrumtext.Text), null, children) : children));
}
/**
 * Badges are used for showing a small amount of color-categorized metadata, ideal for getting a user's attention.
 */ let $f243611f87a6fe61$export$37acb3580601e69a = /*#__PURE__*/ (0, $jLeOL$react.forwardRef)($f243611f87a6fe61$var$Badge);
