// @react-aria/numberfield/dist/main.js
var $15c26e9e75543508$exports = require("./ar-AE.main.js");
var $2d346802965e5c25$exports = require("./bg-BG.main.js");
var $09893e85abdd2974$exports = require("./cs-CZ.main.js");
var $413bf45f477fb4e1$exports = require("./da-DK.main.js");
var $2727845e2efab680$exports = require("./de-DE.main.js");
var $11030a3afacdbc2e$exports = require("./el-GR.main.js");
var $1e48617299cb10e8$exports = require("./en-US.main.js");
var $335176f389a02847$exports = require("./es-ES.main.js");
var $b08e7fdc34a812f3$exports = require("./et-EE.main.js");
var $ce4c5a1eb7e6fc96$exports = require("./fi-FI.main.js");
var $341f988ce23c559c$exports = require("./fr-FR.main.js");
var $950fd1a1d57d2f86$exports = require("./he-IL.main.js");
var $7b949fa64cf10a07$exports = require("./hr-HR.main.js");
var $a945ff6d5cb9cf90$exports = require("./hu-HU.main.js");
var $f3393075c1f078d7$exports = require("./it-IT.main.js");
var $a12655e072ef354d$exports = require("./ja-JP.main.js");
var $880dbee0519c1bad$exports = require("./ko-KR.main.js");
var $d4a9630b079105aa$exports = require("./lt-LT.main.js");
var $e358cd3315986eb7$exports = require("./lv-LV.main.js");
var $a8d0ee8ec28363c7$exports = require("./nb-NO.main.js");
var $0073957e0c754983$exports = require("./nl-NL.main.js");
var $51626cc4fccef54e$exports = require("./pl-PL.main.js");
var $60b1cee412b56d27$exports = require("./pt-BR.main.js");
var $df9c7cc4e8072a0f$exports = require("./pt-PT.main.js");
var $aea54857f9df076d$exports = require("./ro-RO.main.js");
var $8337ebc47a7e61a3$exports = require("./ru-RU.main.js");
var $4703837f8f4224f8$exports = require("./sk-SK.main.js");
var $2b9c8b37ab5bbcca$exports = require("./sl-SI.main.js");
var $2c78665a3b81ed56$exports = require("./sr-SP.main.js");
var $8da5dab2105c0d01$exports = require("./sv-SE.main.js");
var $73a2418cb50e8fda$exports = require("./tr-TR.main.js");
var $9c85d1989bb8c41d$exports = require("./uk-UA.main.js");
var $a7db09d81aa41beb$exports = require("./zh-CN.main.js");
var $edd6ee4675b7a1cd$exports = require("./zh-TW.main.js");
var $2wEZH$reactariautils = require("@react-aria/utils");
var $2wEZH$react = require("react");
var $2wEZH$reactstatelyform = require("@react-stately/form");
var $2wEZH$reactariainteractions = require("@react-aria/interactions");
var $2wEZH$reactariatextfield = require("@react-aria/textfield");
var $2wEZH$reactariai18n = require("@react-aria/i18n");
var $2wEZH$reactariaspinbutton = require("@react-aria/spinbutton");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "useNumberField", () => $fa863e9b015ae839$export$23f548e970bdf099);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

var $4932e21065cdc2cd$exports = {};


































$4932e21065cdc2cd$exports = {
    "ar-AE": $15c26e9e75543508$exports,
    "bg-BG": $2d346802965e5c25$exports,
    "cs-CZ": $09893e85abdd2974$exports,
    "da-DK": $413bf45f477fb4e1$exports,
    "de-DE": $2727845e2efab680$exports,
    "el-GR": $11030a3afacdbc2e$exports,
    "en-US": $1e48617299cb10e8$exports,
    "es-ES": $335176f389a02847$exports,
    "et-EE": $b08e7fdc34a812f3$exports,
    "fi-FI": $ce4c5a1eb7e6fc96$exports,
    "fr-FR": $341f988ce23c559c$exports,
    "he-IL": $950fd1a1d57d2f86$exports,
    "hr-HR": $7b949fa64cf10a07$exports,
    "hu-HU": $a945ff6d5cb9cf90$exports,
    "it-IT": $f3393075c1f078d7$exports,
    "ja-JP": $a12655e072ef354d$exports,
    "ko-KR": $880dbee0519c1bad$exports,
    "lt-LT": $d4a9630b079105aa$exports,
    "lv-LV": $e358cd3315986eb7$exports,
    "nb-NO": $a8d0ee8ec28363c7$exports,
    "nl-NL": $0073957e0c754983$exports,
    "pl-PL": $51626cc4fccef54e$exports,
    "pt-BR": $60b1cee412b56d27$exports,
    "pt-PT": $df9c7cc4e8072a0f$exports,
    "ro-RO": $aea54857f9df076d$exports,
    "ru-RU": $8337ebc47a7e61a3$exports,
    "sk-SK": $4703837f8f4224f8$exports,
    "sl-SI": $2b9c8b37ab5bbcca$exports,
    "sr-SP": $2c78665a3b81ed56$exports,
    "sv-SE": $8da5dab2105c0d01$exports,
    "tr-TR": $73a2418cb50e8fda$exports,
    "uk-UA": $9c85d1989bb8c41d$exports,
    "zh-CN": $a7db09d81aa41beb$exports,
    "zh-TW": $edd6ee4675b7a1cd$exports
};







function $fa863e9b015ae839$export$23f548e970bdf099(props, state, inputRef) {
    let { id: id, decrementAriaLabel: decrementAriaLabel, incrementAriaLabel: incrementAriaLabel, isDisabled: isDisabled, isReadOnly: isReadOnly, isRequired: isRequired, minValue: minValue, maxValue: maxValue, autoFocus: autoFocus, label: label, formatOptions: formatOptions, onBlur: onBlur = ()=>{}, onFocus: onFocus, onFocusChange: onFocusChange, onKeyDown: onKeyDown, onKeyUp: onKeyUp, description: description, errorMessage: errorMessage, ...otherProps } = props;
    let { increment: increment, incrementToMax: incrementToMax, decrement: decrement, decrementToMin: decrementToMin, numberValue: numberValue, inputValue: inputValue, commit: commit, commitValidation: commitValidation } = state;
    const stringFormatter = (0, $2wEZH$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($4932e21065cdc2cd$exports))), "@react-aria/numberfield");
    let inputId = (0, $2wEZH$reactariautils.useId)(id);
    let { focusProps: focusProps } = (0, $2wEZH$reactariainteractions.useFocus)({
        onBlur () {
            // Set input value to normalized valid value
            commit();
        }
    });
    let numberFormatter = (0, $2wEZH$reactariai18n.useNumberFormatter)(formatOptions);
    let intlOptions = (0, $2wEZH$react.useMemo)(()=>numberFormatter.resolvedOptions(), [
        numberFormatter
    ]);
    // Replace negative textValue formatted using currencySign: 'accounting'
    // with a textValue that can be announced using a minus sign.
    let textValueFormatter = (0, $2wEZH$reactariai18n.useNumberFormatter)({
        ...formatOptions,
        currencySign: undefined
    });
    let textValue = (0, $2wEZH$react.useMemo)(()=>isNaN(numberValue) ? "" : textValueFormatter.format(numberValue), [
        textValueFormatter,
        numberValue
    ]);
    let { spinButtonProps: spinButtonProps, incrementButtonProps: incButtonProps, decrementButtonProps: decButtonProps } = (0, $2wEZH$reactariaspinbutton.useSpinButton)({
        isDisabled: isDisabled,
        isReadOnly: isReadOnly,
        isRequired: isRequired,
        maxValue: maxValue,
        minValue: minValue,
        onIncrement: increment,
        onIncrementToMax: incrementToMax,
        onDecrement: decrement,
        onDecrementToMin: decrementToMin,
        value: numberValue,
        textValue: textValue
    });
    let [focusWithin, setFocusWithin] = (0, $2wEZH$react.useState)(false);
    let { focusWithinProps: focusWithinProps } = (0, $2wEZH$reactariainteractions.useFocusWithin)({
        isDisabled: isDisabled,
        onFocusWithinChange: setFocusWithin
    });
    let onWheel = (0, $2wEZH$react.useCallback)((e)=>{
        // if on a trackpad, users can scroll in both X and Y at once, check the magnitude of the change
        // if it's mostly in the X direction, then just return, the user probably doesn't mean to inc/dec
        // this isn't perfect, events come in fast with small deltas and a part of the scroll may give a false indication
        // especially if the user is scrolling near 45deg
        if (Math.abs(e.deltaY) <= Math.abs(e.deltaX)) return;
        if (e.deltaY > 0) increment();
        else if (e.deltaY < 0) decrement();
    }, [
        decrement,
        increment
    ]);
    // If the input isn't supposed to receive input, disable scrolling.
    let scrollingDisabled = isDisabled || isReadOnly || !focusWithin;
    (0, $2wEZH$reactariainteractions.useScrollWheel)({
        onScroll: onWheel,
        isDisabled: scrollingDisabled
    }, inputRef);
    // The inputMode attribute influences the software keyboard that is shown on touch devices.
    // Browsers and operating systems are quite inconsistent about what keys are available, however.
    // We choose between numeric and decimal based on whether we allow negative and fractional numbers,
    // and based on testing on various devices to determine what keys are available in each inputMode.
    let hasDecimals = intlOptions.maximumFractionDigits > 0;
    let hasNegative = isNaN(state.minValue) || state.minValue < 0;
    let inputMode = "numeric";
    if ((0, $2wEZH$reactariautils.isIPhone)()) {
        // iPhone doesn't have a minus sign in either numeric or decimal.
        // Note this is only for iPhone, not iPad, which always has both
        // minus and decimal in numeric.
        if (hasNegative) inputMode = "text";
        else if (hasDecimals) inputMode = "decimal";
    } else if ((0, $2wEZH$reactariautils.isAndroid)()) {
        // Android numeric has both a decimal point and minus key.
        // decimal does not have a minus key.
        if (hasNegative) inputMode = "numeric";
        else if (hasDecimals) inputMode = "decimal";
    }
    let onChange = (value)=>{
        if (state.validate(value)) state.setInputValue(value);
    };
    let domProps = (0, $2wEZH$reactariautils.filterDOMProps)(props);
    let onKeyDownEnter = (0, $2wEZH$react.useCallback)((e)=>{
        if (e.key === "Enter") {
            commit();
            commitValidation();
        } else e.continuePropagation();
    }, [
        commit,
        commitValidation
    ]);
    let { isInvalid: isInvalid, validationErrors: validationErrors, validationDetails: validationDetails } = state.displayValidation;
    let { labelProps: labelProps, inputProps: textFieldProps, descriptionProps: descriptionProps, errorMessageProps: errorMessageProps } = (0, $2wEZH$reactariatextfield.useFormattedTextField)({
        ...otherProps,
        ...domProps,
        name: undefined,
        label: label,
        autoFocus: autoFocus,
        isDisabled: isDisabled,
        isReadOnly: isReadOnly,
        isRequired: isRequired,
        validate: undefined,
        [(0, $2wEZH$reactstatelyform.privateValidationStateProp)]: state,
        value: inputValue,
        defaultValue: undefined,
        autoComplete: "off",
        "aria-label": props["aria-label"] || null,
        "aria-labelledby": props["aria-labelledby"] || null,
        id: inputId,
        type: "text",
        inputMode: inputMode,
        onChange: onChange,
        onBlur: onBlur,
        onFocus: onFocus,
        onFocusChange: onFocusChange,
        onKeyDown: (0, $2wEZH$react.useMemo)(()=>(0, $2wEZH$reactariautils.chain)(onKeyDownEnter, onKeyDown), [
            onKeyDownEnter,
            onKeyDown
        ]),
        onKeyUp: onKeyUp,
        description: description,
        errorMessage: errorMessage
    }, state, inputRef);
    (0, $2wEZH$reactariautils.useFormReset)(inputRef, state.numberValue, state.setNumberValue);
    let inputProps = (0, $2wEZH$reactariautils.mergeProps)(spinButtonProps, focusProps, textFieldProps, {
        // override the spinbutton role, we can't focus a spin button with VO
        role: null,
        // ignore aria-roledescription on iOS so that required state will announce when it is present
        "aria-roledescription": !(0, $2wEZH$reactariautils.isIOS)() ? stringFormatter.format("numberField") : null,
        "aria-valuemax": null,
        "aria-valuemin": null,
        "aria-valuenow": null,
        "aria-valuetext": null,
        autoCorrect: "off",
        spellCheck: "false"
    });
    if (props.validationBehavior === "native") inputProps["aria-required"] = undefined;
    let onButtonPressStart = (e)=>{
        // If focus is already on the input, keep it there so we don't hide the
        // software keyboard when tapping the increment/decrement buttons.
        if (document.activeElement === inputRef.current) return;
        // Otherwise, when using a mouse, move focus to the input.
        // On touch, or with a screen reader, focus the button so that the software
        // keyboard does not appear and the screen reader cursor is not moved off the button.
        if (e.pointerType === "mouse") inputRef.current.focus();
        else e.target.focus();
    };
    // Determine the label for the increment and decrement buttons. There are 4 cases:
    //
    // 1. With a visible label that is a string: aria-label: `Increase ${props.label}`
    // 2. With a visible label that is JSX: aria-label: 'Increase', aria-labelledby: '${incrementId} ${labelId}'
    // 3. With an aria-label: aria-label: `Increase ${props['aria-label']}`
    // 4. With an aria-labelledby: aria-label: 'Increase', aria-labelledby: `${incrementId} ${props['aria-labelledby']}`
    //
    // (1) and (2) could possibly be combined and both use aria-labelledby. However, placing the label in
    // the aria-label string rather than using aria-labelledby gives more flexibility to translators to change
    // the order or add additional words around the label if needed.
    let fieldLabel = props["aria-label"] || (typeof props.label === "string" ? props.label : "");
    let ariaLabelledby;
    if (!fieldLabel) ariaLabelledby = props.label != null ? labelProps.id : props["aria-labelledby"];
    let incrementId = (0, $2wEZH$reactariautils.useId)();
    let decrementId = (0, $2wEZH$reactariautils.useId)();
    let incrementButtonProps = (0, $2wEZH$reactariautils.mergeProps)(incButtonProps, {
        "aria-label": incrementAriaLabel || stringFormatter.format("increase", {
            fieldLabel: fieldLabel
        }).trim(),
        id: ariaLabelledby && !incrementAriaLabel ? incrementId : null,
        "aria-labelledby": ariaLabelledby && !incrementAriaLabel ? `${incrementId} ${ariaLabelledby}` : null,
        "aria-controls": inputId,
        excludeFromTabOrder: true,
        preventFocusOnPress: true,
        allowFocusWhenDisabled: true,
        isDisabled: !state.canIncrement,
        onPressStart: onButtonPressStart
    });
    let decrementButtonProps = (0, $2wEZH$reactariautils.mergeProps)(decButtonProps, {
        "aria-label": decrementAriaLabel || stringFormatter.format("decrease", {
            fieldLabel: fieldLabel
        }).trim(),
        id: ariaLabelledby && !decrementAriaLabel ? decrementId : null,
        "aria-labelledby": ariaLabelledby && !decrementAriaLabel ? `${decrementId} ${ariaLabelledby}` : null,
        "aria-controls": inputId,
        excludeFromTabOrder: true,
        preventFocusOnPress: true,
        allowFocusWhenDisabled: true,
        isDisabled: !state.canDecrement,
        onPressStart: onButtonPressStart
    });
    return {
        groupProps: {
            ...focusWithinProps,
            role: "group",
            "aria-disabled": isDisabled,
            "aria-invalid": isInvalid ? "true" : undefined
        },
        labelProps: labelProps,
        inputProps: inputProps,
        incrementButtonProps: incrementButtonProps,
        decrementButtonProps: decrementButtonProps,
        errorMessageProps: errorMessageProps,
        descriptionProps: descriptionProps,
        isInvalid: isInvalid,
        validationErrors: validationErrors,
        validationDetails: validationDetails
    };
}
