// @aemforms/af-core/lib/rules/RuleEngine.js
"use strict";
/*
 * Copyright 2022 Adobe, Inc.
 *
 * Your access and use of this software is governed by the Adobe Customer Feedback Program Terms and Conditions or other Beta License Agreement signed by your employer and Adobe, Inc.. This software is NOT open source and may not be used without one of the foregoing licenses. Even with a foregoing license, your access and use of this file is limited to the earlier of (a) 180 days, (b) general availability of the product(s) which utilize this software (i.e. AEM Forms), (c) January 1, 2023, (d) Adobe providing notice to you that you may no longer use the software or that your beta trial has otherwise ended.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL ADOBE NOR ITS THIRD PARTY PROVIDERS AND PARTNERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Controller_1 = require("../controller/Controller");
const json_formula_1 = require("@adobe/json-formula");
const FunctionRuntime_1 = __importDefault(require("./FunctionRuntime"));
/**
 * Implementation of AddDependant event
 * @private
 */
class AddDependent extends Controller_1.ActionImpl {
    constructor(payload) {
        super(payload, 'addDependent');
    }
    payloadToJson() {
        return this.payload.getState();
    }
}
/**
 * Implementation of rule engine
 * @private
 */
class RuleEngine {
    constructor() {
        this._globalNames = [
            '$form',
            '$field',
            '$event'
        ];
    }
    compileRule(rule) {
        const customFunctions = FunctionRuntime_1.default.getFunctions();
        return new json_formula_1.Formula(rule, customFunctions, undefined, this._globalNames);
    }
    execute(node, data, globals, useValueOf = false) {
        const oldContext = this._context;
        this._context = globals;
        const res = node.search(data, globals);
        let finalRes = res;
        if (useValueOf) {
            if (typeof res === 'object' && res !== null) {
                finalRes = Object.getPrototypeOf(res).valueOf.call(res);
            }
        }
        this._context = oldContext;
        return finalRes;
    }
    /**
     * Listen to subscriber for
     * @param subscriber
     */
    trackDependency(subscriber) {
        if (this._context && this._context.field !== undefined && this._context.field !== subscriber) {
            subscriber.dispatch(new AddDependent(this._context.field));
        }
    }
}
exports.default = RuleEngine;
