// @react-spectrum/inlinealert/dist/import.mjs
"use client";

import $7c06d136a7db332f$exports from "./ar-AE.mjs";
import $7572fcde1b201200$exports from "./bg-BG.mjs";
import $4baaa05e7e36c8d5$exports from "./cs-CZ.mjs";
import $67391f9ec3919bbd$exports from "./da-DK.mjs";
import $ba4d55b7a25dbd70$exports from "./de-DE.mjs";
import $8f11cbfb87576513$exports from "./el-GR.mjs";
import $a5983f3344cffe9c$exports from "./en-US.mjs";
import $5b00f4541c9c46ff$exports from "./es-ES.mjs";
import $aa360d77330765aa$exports from "./et-EE.mjs";
import $03e931e04f228c9f$exports from "./fi-FI.mjs";
import $9d5074a25853ca6e$exports from "./fr-FR.mjs";
import $2c2676389f361ece$exports from "./he-IL.mjs";
import $5fdda27174ef2fc7$exports from "./hr-HR.mjs";
import $dc458e3abea04ded$exports from "./hu-HU.mjs";
import $2ebc0e35fdc040b7$exports from "./it-IT.mjs";
import $44f02920600efced$exports from "./ja-JP.mjs";
import $eeee9839fa747b1b$exports from "./ko-KR.mjs";
import $7889661aab511c26$exports from "./lt-LT.mjs";
import $f73d040d5b290afc$exports from "./lv-LV.mjs";
import $eae0cb6abb6f7592$exports from "./nb-NO.mjs";
import $1e0495a2c0782298$exports from "./nl-NL.mjs";
import $06b952e4a8b5ac61$exports from "./pl-PL.mjs";
import $d6a05b7bb1801915$exports from "./pt-BR.mjs";
import $83d519ea719f843b$exports from "./pt-PT.mjs";
import $3fde5b55008eef3f$exports from "./ro-RO.mjs";
import $96d433eefb3f8906$exports from "./ru-RU.mjs";
import $0893538688618f33$exports from "./sk-SK.mjs";
import $254b6c961e4f0e80$exports from "./sl-SI.mjs";
import $680d309814a11fc5$exports from "./sr-SP.mjs";
import $f9418595c1e772cf$exports from "./sv-SE.mjs";
import $f7b3644c5b6100d7$exports from "./tr-TR.mjs";
import $74a01294f8d465f8$exports from "./uk-UA.mjs";
import $a1438ca118687381$exports from "./zh-CN.mjs";
import $948f0650714aa894$exports from "./zh-TW.mjs";
import "./main.css";
import $d32NX$spectrumiconsuiAlertMedium from "@spectrum-icons/ui/AlertMedium";
import {useStyleProps as $d32NX$useStyleProps, useDOMRef as $d32NX$useDOMRef, classNames as $d32NX$classNames, SlotProvider as $d32NX$SlotProvider} from "@react-spectrum/utils";
import {filterDOMProps as $d32NX$filterDOMProps} from "@react-aria/utils";
import {FocusRing as $d32NX$FocusRing} from "@react-aria/focus";
import {Grid as $d32NX$Grid} from "@react-spectrum/layout";
import $d32NX$spectrumiconsuiInfoMedium from "@spectrum-icons/ui/InfoMedium";
import $d32NX$react, {useRef as $d32NX$useRef, useEffect as $d32NX$useEffect} from "react";
import $d32NX$spectrumiconsuiSuccessMedium from "@spectrum-icons/ui/SuccessMedium";
import {useLocalizedStringFormatter as $d32NX$useLocalizedStringFormatter} from "@react-aria/i18n";
import {useProviderProps as $d32NX$useProviderProps} from "@react-spectrum/provider";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





var $a0360622247416a9$exports = {};


































$a0360622247416a9$exports = {
    "ar-AE": $7c06d136a7db332f$exports,
    "bg-BG": $7572fcde1b201200$exports,
    "cs-CZ": $4baaa05e7e36c8d5$exports,
    "da-DK": $67391f9ec3919bbd$exports,
    "de-DE": $ba4d55b7a25dbd70$exports,
    "el-GR": $8f11cbfb87576513$exports,
    "en-US": $a5983f3344cffe9c$exports,
    "es-ES": $5b00f4541c9c46ff$exports,
    "et-EE": $aa360d77330765aa$exports,
    "fi-FI": $03e931e04f228c9f$exports,
    "fr-FR": $9d5074a25853ca6e$exports,
    "he-IL": $2c2676389f361ece$exports,
    "hr-HR": $5fdda27174ef2fc7$exports,
    "hu-HU": $dc458e3abea04ded$exports,
    "it-IT": $2ebc0e35fdc040b7$exports,
    "ja-JP": $44f02920600efced$exports,
    "ko-KR": $eeee9839fa747b1b$exports,
    "lt-LT": $7889661aab511c26$exports,
    "lv-LV": $f73d040d5b290afc$exports,
    "nb-NO": $eae0cb6abb6f7592$exports,
    "nl-NL": $1e0495a2c0782298$exports,
    "pl-PL": $06b952e4a8b5ac61$exports,
    "pt-BR": $d6a05b7bb1801915$exports,
    "pt-PT": $83d519ea719f843b$exports,
    "ro-RO": $3fde5b55008eef3f$exports,
    "ru-RU": $96d433eefb3f8906$exports,
    "sk-SK": $0893538688618f33$exports,
    "sl-SI": $254b6c961e4f0e80$exports,
    "sr-SP": $680d309814a11fc5$exports,
    "sv-SE": $f9418595c1e772cf$exports,
    "tr-TR": $f7b3644c5b6100d7$exports,
    "uk-UA": $74a01294f8d465f8$exports,
    "zh-CN": $a1438ca118687381$exports,
    "zh-TW": $948f0650714aa894$exports
};



var $48fe1f4c438508c0$exports = {};

$parcel$export($48fe1f4c438508c0$exports, "content", () => $48fe1f4c438508c0$export$a7db06668cad9adb, (v) => $48fe1f4c438508c0$export$a7db06668cad9adb = v);
$parcel$export($48fe1f4c438508c0$exports, "focus-ring", () => $48fe1f4c438508c0$export$f39a09f249340e2a, (v) => $48fe1f4c438508c0$export$f39a09f249340e2a = v);
$parcel$export($48fe1f4c438508c0$exports, "heading", () => $48fe1f4c438508c0$export$46e4a324ac90507f, (v) => $48fe1f4c438508c0$export$46e4a324ac90507f = v);
$parcel$export($48fe1f4c438508c0$exports, "i18nFontFamily", () => $48fe1f4c438508c0$export$8c4ee2c50c22c514, (v) => $48fe1f4c438508c0$export$8c4ee2c50c22c514 = v);
$parcel$export($48fe1f4c438508c0$exports, "icon", () => $48fe1f4c438508c0$export$1ca1ec8b29a4ce27, (v) => $48fe1f4c438508c0$export$1ca1ec8b29a4ce27 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-FocusRing-ring", () => $48fe1f4c438508c0$export$4109102f950813a6, (v) => $48fe1f4c438508c0$export$4109102f950813a6 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-FocusRing", () => $48fe1f4c438508c0$export$24c7f46a6e3605dd, (v) => $48fe1f4c438508c0$export$24c7f46a6e3605dd = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-FocusRing--quiet", () => $48fe1f4c438508c0$export$2927016961429360, (v) => $48fe1f4c438508c0$export$2927016961429360 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert", () => $48fe1f4c438508c0$export$73769bb5d26d9ff7, (v) => $48fe1f4c438508c0$export$73769bb5d26d9ff7 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--info", () => $48fe1f4c438508c0$export$71c1df0099076ffd, (v) => $48fe1f4c438508c0$export$71c1df0099076ffd = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--negative", () => $48fe1f4c438508c0$export$22162c465dd1f294, (v) => $48fe1f4c438508c0$export$22162c465dd1f294 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--neutral", () => $48fe1f4c438508c0$export$1d6dd31283ff6c5c, (v) => $48fe1f4c438508c0$export$1d6dd31283ff6c5c = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--notice", () => $48fe1f4c438508c0$export$29b25aea79cc7183, (v) => $48fe1f4c438508c0$export$29b25aea79cc7183 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert--positive", () => $48fe1f4c438508c0$export$fe577b621f09240b, (v) => $48fe1f4c438508c0$export$fe577b621f09240b = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert-content", () => $48fe1f4c438508c0$export$b9bc26a202328584, (v) => $48fe1f4c438508c0$export$b9bc26a202328584 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert-grid", () => $48fe1f4c438508c0$export$c31ea092819cec48, (v) => $48fe1f4c438508c0$export$c31ea092819cec48 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert-heading", () => $48fe1f4c438508c0$export$5f11e05135494154, (v) => $48fe1f4c438508c0$export$5f11e05135494154 = v);
$parcel$export($48fe1f4c438508c0$exports, "spectrum-InLineAlert-icon", () => $48fe1f4c438508c0$export$3be02ea40f92bdfc, (v) => $48fe1f4c438508c0$export$3be02ea40f92bdfc = v);
var $48fe1f4c438508c0$export$a7db06668cad9adb;
var $48fe1f4c438508c0$export$f39a09f249340e2a;
var $48fe1f4c438508c0$export$46e4a324ac90507f;
var $48fe1f4c438508c0$export$8c4ee2c50c22c514;
var $48fe1f4c438508c0$export$1ca1ec8b29a4ce27;
var $48fe1f4c438508c0$export$4109102f950813a6;
var $48fe1f4c438508c0$export$24c7f46a6e3605dd;
var $48fe1f4c438508c0$export$2927016961429360;
var $48fe1f4c438508c0$export$73769bb5d26d9ff7;
var $48fe1f4c438508c0$export$71c1df0099076ffd;
var $48fe1f4c438508c0$export$22162c465dd1f294;
var $48fe1f4c438508c0$export$1d6dd31283ff6c5c;
var $48fe1f4c438508c0$export$29b25aea79cc7183;
var $48fe1f4c438508c0$export$fe577b621f09240b;
var $48fe1f4c438508c0$export$b9bc26a202328584;
var $48fe1f4c438508c0$export$c31ea092819cec48;
var $48fe1f4c438508c0$export$5f11e05135494154;
var $48fe1f4c438508c0$export$3be02ea40f92bdfc;
$48fe1f4c438508c0$export$a7db06668cad9adb = `G_feca_content`;
$48fe1f4c438508c0$export$f39a09f249340e2a = `G_feca_focus-ring`;
$48fe1f4c438508c0$export$46e4a324ac90507f = `G_feca_heading`;
$48fe1f4c438508c0$export$8c4ee2c50c22c514 = `G_feca_i18nFontFamily`;
$48fe1f4c438508c0$export$1ca1ec8b29a4ce27 = `G_feca_icon`;
$48fe1f4c438508c0$export$4109102f950813a6 = `G_feca_spectrum-FocusRing-ring`;
$48fe1f4c438508c0$export$24c7f46a6e3605dd = `G_feca_spectrum-FocusRing ${$48fe1f4c438508c0$export$4109102f950813a6}`;
$48fe1f4c438508c0$export$2927016961429360 = `G_feca_spectrum-FocusRing--quiet`;
$48fe1f4c438508c0$export$73769bb5d26d9ff7 = `G_feca_spectrum-InLineAlert ${$48fe1f4c438508c0$export$24c7f46a6e3605dd}`;
$48fe1f4c438508c0$export$71c1df0099076ffd = `G_feca_spectrum-InLineAlert--info`;
$48fe1f4c438508c0$export$22162c465dd1f294 = `G_feca_spectrum-InLineAlert--negative`;
$48fe1f4c438508c0$export$1d6dd31283ff6c5c = `G_feca_spectrum-InLineAlert--neutral`;
$48fe1f4c438508c0$export$29b25aea79cc7183 = `G_feca_spectrum-InLineAlert--notice`;
$48fe1f4c438508c0$export$fe577b621f09240b = `G_feca_spectrum-InLineAlert--positive`;
$48fe1f4c438508c0$export$b9bc26a202328584 = `G_feca_spectrum-InLineAlert-content`;
$48fe1f4c438508c0$export$c31ea092819cec48 = `G_feca_spectrum-InLineAlert-grid`;
$48fe1f4c438508c0$export$5f11e05135494154 = `G_feca_spectrum-InLineAlert-heading`;
$48fe1f4c438508c0$export$3be02ea40f92bdfc = `G_feca_spectrum-InLineAlert-icon`;





let $abf1e6290496dda5$var$ICONS = {
    info: (0, $d32NX$spectrumiconsuiInfoMedium),
    positive: (0, $d32NX$spectrumiconsuiSuccessMedium),
    notice: (0, $d32NX$spectrumiconsuiAlertMedium),
    negative: (0, $d32NX$spectrumiconsuiAlertMedium)
};
function $abf1e6290496dda5$var$InlineAlert(props, ref) {
    props = (0, $d32NX$useProviderProps)(props);
    let { children: children, variant: variant = "neutral", autoFocus: autoFocus, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $d32NX$useStyleProps)(otherProps);
    let domRef = (0, $d32NX$useDOMRef)(ref);
    let slots = {
        heading: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["spectrum-InLineAlert-heading"]
        },
        content: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["spectrum-InLineAlert-content"]
        }
    };
    let stringFormatter = (0, $d32NX$useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($a0360622247416a9$exports))), "@react-spectrum/inlinealert");
    let Icon = null;
    let iconAlt;
    if (variant in $abf1e6290496dda5$var$ICONS) {
        Icon = $abf1e6290496dda5$var$ICONS[variant];
        iconAlt = stringFormatter.format(variant);
    }
    let autoFocusRef = (0, $d32NX$useRef)(props.autoFocus);
    (0, $d32NX$useEffect)(()=>{
        if (autoFocusRef.current && domRef.current) domRef.current.focus();
        autoFocusRef.current = false;
    }, [
        domRef
    ]);
    return /*#__PURE__*/ (0, $d32NX$react).createElement((0, $d32NX$FocusRing), {
        focusRingClass: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["focus-ring"]
    }, /*#__PURE__*/ (0, $d32NX$react).createElement("div", {
        ...(0, $d32NX$filterDOMProps)(props),
        ...styleProps,
        ref: domRef,
        tabIndex: autoFocus ? -1 : undefined,
        autoFocus: autoFocus,
        className: (0, $d32NX$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports))), "spectrum-InLineAlert", `spectrum-InLineAlert--${variant}`, styleProps.className),
        role: "alert"
    }, /*#__PURE__*/ (0, $d32NX$react).createElement((0, $d32NX$Grid), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["spectrum-InLineAlert-grid"]
    }, /*#__PURE__*/ (0, $d32NX$react).createElement((0, $d32NX$SlotProvider), {
        slots: slots
    }, Icon && /*#__PURE__*/ (0, $d32NX$react).createElement(Icon, {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($48fe1f4c438508c0$exports)))["spectrum-InLineAlert-icon"],
        "aria-label": iconAlt
    }), children))));
}
/**
 * Inline alerts display a non-modal message associated with objects in a view.
 * These are often used in form validation, providing a place to aggregate feedback related to multiple fields.
 */ const $abf1e6290496dda5$export$a3b2c96db9b0eb71 = /*#__PURE__*/ (0, $d32NX$react).forwardRef($abf1e6290496dda5$var$InlineAlert);




export {$abf1e6290496dda5$export$a3b2c96db9b0eb71 as InlineAlert};
