// @react-spectrum/inlinealert/dist/main.js
"use client";

var $70305e81a8d2187b$exports = require("./ar-AE.main.js");
var $0c660aaf1ba719ea$exports = require("./bg-BG.main.js");
var $09ed32d40da22228$exports = require("./cs-CZ.main.js");
var $fb9891466d18e5a4$exports = require("./da-DK.main.js");
var $5ccd3660efbdf1ba$exports = require("./de-DE.main.js");
var $d99fc87568761416$exports = require("./el-GR.main.js");
var $fc814fd646f97b4a$exports = require("./en-US.main.js");
var $cd2deb9e4ace11a8$exports = require("./es-ES.main.js");
var $f0d5c61765f22ed3$exports = require("./et-EE.main.js");
var $a2bb977ca4ce5ca1$exports = require("./fi-FI.main.js");
var $116c45d32f2f45ca$exports = require("./fr-FR.main.js");
var $1071566f78693587$exports = require("./he-IL.main.js");
var $c636cbe8751599df$exports = require("./hr-HR.main.js");
var $2f9ee8544d50270a$exports = require("./hu-HU.main.js");
var $c3a100d558fcd0c3$exports = require("./it-IT.main.js");
var $9b6747fa80c2005c$exports = require("./ja-JP.main.js");
var $7aa436d0bc11d8df$exports = require("./ko-KR.main.js");
var $2306a8fe8a217aae$exports = require("./lt-LT.main.js");
var $7ee88da038778f20$exports = require("./lv-LV.main.js");
var $4ba767217b9690c2$exports = require("./nb-NO.main.js");
var $3c4e9f1e46e8db66$exports = require("./nl-NL.main.js");
var $15ba96b5246db651$exports = require("./pl-PL.main.js");
var $969881ee58ec3452$exports = require("./pt-BR.main.js");
var $ded9cae395371ad3$exports = require("./pt-PT.main.js");
var $9966420945e42ed8$exports = require("./ro-RO.main.js");
var $9c9f39649e3598f2$exports = require("./ru-RU.main.js");
var $d9dce3fa927805b2$exports = require("./sk-SK.main.js");
var $bdb67023e40cc961$exports = require("./sl-SI.main.js");
var $21eca3566d4fbc9e$exports = require("./sr-SP.main.js");
var $77c4ed2ec49f90d9$exports = require("./sv-SE.main.js");
var $aea4f814e594ee44$exports = require("./tr-TR.main.js");
var $ed1255c500b98f56$exports = require("./uk-UA.main.js");
var $2adae4170a5c1992$exports = require("./zh-CN.main.js");
var $7cf811ce2d1c9d87$exports = require("./zh-TW.main.js");
require("./main.css");
var $RX49b$spectrumiconsuiAlertMedium = require("@spectrum-icons/ui/AlertMedium");
var $RX49b$reactspectrumutils = require("@react-spectrum/utils");
var $RX49b$reactariautils = require("@react-aria/utils");
var $RX49b$reactariafocus = require("@react-aria/focus");
var $RX49b$reactspectrumlayout = require("@react-spectrum/layout");
var $RX49b$spectrumiconsuiInfoMedium = require("@spectrum-icons/ui/InfoMedium");
var $RX49b$react = require("react");
var $RX49b$spectrumiconsuiSuccessMedium = require("@spectrum-icons/ui/SuccessMedium");
var $RX49b$reactariai18n = require("@react-aria/i18n");
var $RX49b$reactspectrumprovider = require("@react-spectrum/provider");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "InlineAlert", () => $5de8c6729f05c665$export$a3b2c96db9b0eb71);
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2023 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 





var $cf1d087b199fc7c5$exports = {};


































$cf1d087b199fc7c5$exports = {
    "ar-AE": $70305e81a8d2187b$exports,
    "bg-BG": $0c660aaf1ba719ea$exports,
    "cs-CZ": $09ed32d40da22228$exports,
    "da-DK": $fb9891466d18e5a4$exports,
    "de-DE": $5ccd3660efbdf1ba$exports,
    "el-GR": $d99fc87568761416$exports,
    "en-US": $fc814fd646f97b4a$exports,
    "es-ES": $cd2deb9e4ace11a8$exports,
    "et-EE": $f0d5c61765f22ed3$exports,
    "fi-FI": $a2bb977ca4ce5ca1$exports,
    "fr-FR": $116c45d32f2f45ca$exports,
    "he-IL": $1071566f78693587$exports,
    "hr-HR": $c636cbe8751599df$exports,
    "hu-HU": $2f9ee8544d50270a$exports,
    "it-IT": $c3a100d558fcd0c3$exports,
    "ja-JP": $9b6747fa80c2005c$exports,
    "ko-KR": $7aa436d0bc11d8df$exports,
    "lt-LT": $2306a8fe8a217aae$exports,
    "lv-LV": $7ee88da038778f20$exports,
    "nb-NO": $4ba767217b9690c2$exports,
    "nl-NL": $3c4e9f1e46e8db66$exports,
    "pl-PL": $15ba96b5246db651$exports,
    "pt-BR": $969881ee58ec3452$exports,
    "pt-PT": $ded9cae395371ad3$exports,
    "ro-RO": $9966420945e42ed8$exports,
    "ru-RU": $9c9f39649e3598f2$exports,
    "sk-SK": $d9dce3fa927805b2$exports,
    "sl-SI": $bdb67023e40cc961$exports,
    "sr-SP": $21eca3566d4fbc9e$exports,
    "sv-SE": $77c4ed2ec49f90d9$exports,
    "tr-TR": $aea4f814e594ee44$exports,
    "uk-UA": $ed1255c500b98f56$exports,
    "zh-CN": $2adae4170a5c1992$exports,
    "zh-TW": $7cf811ce2d1c9d87$exports
};



var $8d976ce4ebad141b$exports = {};

$parcel$export($8d976ce4ebad141b$exports, "content", () => $8d976ce4ebad141b$export$a7db06668cad9adb, (v) => $8d976ce4ebad141b$export$a7db06668cad9adb = v);
$parcel$export($8d976ce4ebad141b$exports, "focus-ring", () => $8d976ce4ebad141b$export$f39a09f249340e2a, (v) => $8d976ce4ebad141b$export$f39a09f249340e2a = v);
$parcel$export($8d976ce4ebad141b$exports, "heading", () => $8d976ce4ebad141b$export$46e4a324ac90507f, (v) => $8d976ce4ebad141b$export$46e4a324ac90507f = v);
$parcel$export($8d976ce4ebad141b$exports, "i18nFontFamily", () => $8d976ce4ebad141b$export$8c4ee2c50c22c514, (v) => $8d976ce4ebad141b$export$8c4ee2c50c22c514 = v);
$parcel$export($8d976ce4ebad141b$exports, "icon", () => $8d976ce4ebad141b$export$1ca1ec8b29a4ce27, (v) => $8d976ce4ebad141b$export$1ca1ec8b29a4ce27 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-FocusRing-ring", () => $8d976ce4ebad141b$export$4109102f950813a6, (v) => $8d976ce4ebad141b$export$4109102f950813a6 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-FocusRing", () => $8d976ce4ebad141b$export$24c7f46a6e3605dd, (v) => $8d976ce4ebad141b$export$24c7f46a6e3605dd = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-FocusRing--quiet", () => $8d976ce4ebad141b$export$2927016961429360, (v) => $8d976ce4ebad141b$export$2927016961429360 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert", () => $8d976ce4ebad141b$export$73769bb5d26d9ff7, (v) => $8d976ce4ebad141b$export$73769bb5d26d9ff7 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--info", () => $8d976ce4ebad141b$export$71c1df0099076ffd, (v) => $8d976ce4ebad141b$export$71c1df0099076ffd = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--negative", () => $8d976ce4ebad141b$export$22162c465dd1f294, (v) => $8d976ce4ebad141b$export$22162c465dd1f294 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--neutral", () => $8d976ce4ebad141b$export$1d6dd31283ff6c5c, (v) => $8d976ce4ebad141b$export$1d6dd31283ff6c5c = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--notice", () => $8d976ce4ebad141b$export$29b25aea79cc7183, (v) => $8d976ce4ebad141b$export$29b25aea79cc7183 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert--positive", () => $8d976ce4ebad141b$export$fe577b621f09240b, (v) => $8d976ce4ebad141b$export$fe577b621f09240b = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert-content", () => $8d976ce4ebad141b$export$b9bc26a202328584, (v) => $8d976ce4ebad141b$export$b9bc26a202328584 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert-grid", () => $8d976ce4ebad141b$export$c31ea092819cec48, (v) => $8d976ce4ebad141b$export$c31ea092819cec48 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert-heading", () => $8d976ce4ebad141b$export$5f11e05135494154, (v) => $8d976ce4ebad141b$export$5f11e05135494154 = v);
$parcel$export($8d976ce4ebad141b$exports, "spectrum-InLineAlert-icon", () => $8d976ce4ebad141b$export$3be02ea40f92bdfc, (v) => $8d976ce4ebad141b$export$3be02ea40f92bdfc = v);
var $8d976ce4ebad141b$export$a7db06668cad9adb;
var $8d976ce4ebad141b$export$f39a09f249340e2a;
var $8d976ce4ebad141b$export$46e4a324ac90507f;
var $8d976ce4ebad141b$export$8c4ee2c50c22c514;
var $8d976ce4ebad141b$export$1ca1ec8b29a4ce27;
var $8d976ce4ebad141b$export$4109102f950813a6;
var $8d976ce4ebad141b$export$24c7f46a6e3605dd;
var $8d976ce4ebad141b$export$2927016961429360;
var $8d976ce4ebad141b$export$73769bb5d26d9ff7;
var $8d976ce4ebad141b$export$71c1df0099076ffd;
var $8d976ce4ebad141b$export$22162c465dd1f294;
var $8d976ce4ebad141b$export$1d6dd31283ff6c5c;
var $8d976ce4ebad141b$export$29b25aea79cc7183;
var $8d976ce4ebad141b$export$fe577b621f09240b;
var $8d976ce4ebad141b$export$b9bc26a202328584;
var $8d976ce4ebad141b$export$c31ea092819cec48;
var $8d976ce4ebad141b$export$5f11e05135494154;
var $8d976ce4ebad141b$export$3be02ea40f92bdfc;
$8d976ce4ebad141b$export$a7db06668cad9adb = `G_feca_content`;
$8d976ce4ebad141b$export$f39a09f249340e2a = `G_feca_focus-ring`;
$8d976ce4ebad141b$export$46e4a324ac90507f = `G_feca_heading`;
$8d976ce4ebad141b$export$8c4ee2c50c22c514 = `G_feca_i18nFontFamily`;
$8d976ce4ebad141b$export$1ca1ec8b29a4ce27 = `G_feca_icon`;
$8d976ce4ebad141b$export$4109102f950813a6 = `G_feca_spectrum-FocusRing-ring`;
$8d976ce4ebad141b$export$24c7f46a6e3605dd = `G_feca_spectrum-FocusRing ${$8d976ce4ebad141b$export$4109102f950813a6}`;
$8d976ce4ebad141b$export$2927016961429360 = `G_feca_spectrum-FocusRing--quiet`;
$8d976ce4ebad141b$export$73769bb5d26d9ff7 = `G_feca_spectrum-InLineAlert ${$8d976ce4ebad141b$export$24c7f46a6e3605dd}`;
$8d976ce4ebad141b$export$71c1df0099076ffd = `G_feca_spectrum-InLineAlert--info`;
$8d976ce4ebad141b$export$22162c465dd1f294 = `G_feca_spectrum-InLineAlert--negative`;
$8d976ce4ebad141b$export$1d6dd31283ff6c5c = `G_feca_spectrum-InLineAlert--neutral`;
$8d976ce4ebad141b$export$29b25aea79cc7183 = `G_feca_spectrum-InLineAlert--notice`;
$8d976ce4ebad141b$export$fe577b621f09240b = `G_feca_spectrum-InLineAlert--positive`;
$8d976ce4ebad141b$export$b9bc26a202328584 = `G_feca_spectrum-InLineAlert-content`;
$8d976ce4ebad141b$export$c31ea092819cec48 = `G_feca_spectrum-InLineAlert-grid`;
$8d976ce4ebad141b$export$5f11e05135494154 = `G_feca_spectrum-InLineAlert-heading`;
$8d976ce4ebad141b$export$3be02ea40f92bdfc = `G_feca_spectrum-InLineAlert-icon`;





let $5de8c6729f05c665$var$ICONS = {
    info: (0, ($parcel$interopDefault($RX49b$spectrumiconsuiInfoMedium))),
    positive: (0, ($parcel$interopDefault($RX49b$spectrumiconsuiSuccessMedium))),
    notice: (0, ($parcel$interopDefault($RX49b$spectrumiconsuiAlertMedium))),
    negative: (0, ($parcel$interopDefault($RX49b$spectrumiconsuiAlertMedium)))
};
function $5de8c6729f05c665$var$InlineAlert(props, ref) {
    props = (0, $RX49b$reactspectrumprovider.useProviderProps)(props);
    let { children: children, variant: variant = "neutral", autoFocus: autoFocus, ...otherProps } = props;
    let { styleProps: styleProps } = (0, $RX49b$reactspectrumutils.useStyleProps)(otherProps);
    let domRef = (0, $RX49b$reactspectrumutils.useDOMRef)(ref);
    let slots = {
        heading: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["spectrum-InLineAlert-heading"]
        },
        content: {
            UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["spectrum-InLineAlert-content"]
        }
    };
    let stringFormatter = (0, $RX49b$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($cf1d087b199fc7c5$exports))), "@react-spectrum/inlinealert");
    let Icon = null;
    let iconAlt;
    if (variant in $5de8c6729f05c665$var$ICONS) {
        Icon = $5de8c6729f05c665$var$ICONS[variant];
        iconAlt = stringFormatter.format(variant);
    }
    let autoFocusRef = (0, $RX49b$react.useRef)(props.autoFocus);
    (0, $RX49b$react.useEffect)(()=>{
        if (autoFocusRef.current && domRef.current) domRef.current.focus();
        autoFocusRef.current = false;
    }, [
        domRef
    ]);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement((0, $RX49b$reactariafocus.FocusRing), {
        focusRingClass: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["focus-ring"]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement("div", {
        ...(0, $RX49b$reactariautils.filterDOMProps)(props),
        ...styleProps,
        ref: domRef,
        tabIndex: autoFocus ? -1 : undefined,
        autoFocus: autoFocus,
        className: (0, $RX49b$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports))), "spectrum-InLineAlert", `spectrum-InLineAlert--${variant}`, styleProps.className),
        role: "alert"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement((0, $RX49b$reactspectrumlayout.Grid), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["spectrum-InLineAlert-grid"]
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement((0, $RX49b$reactspectrumutils.SlotProvider), {
        slots: slots
    }, Icon && /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).createElement(Icon, {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($8d976ce4ebad141b$exports)))["spectrum-InLineAlert-icon"],
        "aria-label": iconAlt
    }), children))));
}
/**
 * Inline alerts display a non-modal message associated with objects in a view.
 * These are often used in form validation, providing a place to aggregate feedback related to multiple fields.
 */ const $5de8c6729f05c665$export$a3b2c96db9b0eb71 = /*#__PURE__*/ (0, ($parcel$interopDefault($RX49b$react))).forwardRef($5de8c6729f05c665$var$InlineAlert);
