// @react-spectrum/button/dist/main.js
"use client";

var $8dfbe9f11d8f0c18$exports = require("./ar-AE.main.js");
var $f1d09da51b7863de$exports = require("./bg-BG.main.js");
var $a938a0b253d9f247$exports = require("./cs-CZ.main.js");
var $cf0c54eea40f6fcb$exports = require("./da-DK.main.js");
var $4748cd40594ea272$exports = require("./de-DE.main.js");
var $650c1cff0501e881$exports = require("./el-GR.main.js");
var $e400898e815c69b7$exports = require("./en-US.main.js");
var $6d85accad7f1b1b0$exports = require("./es-ES.main.js");
var $8bd503882fd613bd$exports = require("./et-EE.main.js");
var $23d05f8457cd40f2$exports = require("./fi-FI.main.js");
var $e63063aea75f1c21$exports = require("./fr-FR.main.js");
var $9906c3303b98f896$exports = require("./he-IL.main.js");
var $8e94047ff7a4f28e$exports = require("./hr-HR.main.js");
var $29cc0bceb80e8c91$exports = require("./hu-HU.main.js");
var $986fbd55bef3d71d$exports = require("./it-IT.main.js");
var $7f1ef5a9cae227cd$exports = require("./ja-JP.main.js");
var $9d9c1e7e8c3604b6$exports = require("./ko-KR.main.js");
var $bb0c9e2a421c29e6$exports = require("./lt-LT.main.js");
var $d4bcb4a7735702a6$exports = require("./lv-LV.main.js");
var $bbac6f483322e4cb$exports = require("./nb-NO.main.js");
var $2e38f272eed1ee93$exports = require("./nl-NL.main.js");
var $efdabc3b4ba6ae3a$exports = require("./pl-PL.main.js");
var $32c199c1c0b84ea6$exports = require("./pt-BR.main.js");
var $4525387e786c4e45$exports = require("./pt-PT.main.js");
var $946eea38cb062cf0$exports = require("./ro-RO.main.js");
var $fe41a7461b3fc080$exports = require("./ru-RU.main.js");
var $2d1b4cc9945c3226$exports = require("./sk-SK.main.js");
var $5d18d41ac4266fa7$exports = require("./sl-SI.main.js");
var $b2209169f376772d$exports = require("./sr-SP.main.js");
var $d3ab0389431476ca$exports = require("./sv-SE.main.js");
var $33bce89e4fc9865a$exports = require("./tr-TR.main.js");
var $a803ef1dfa0d114e$exports = require("./uk-UA.main.js");
var $732189c02a0cbea0$exports = require("./zh-CN.main.js");
var $feaa03429c1f5a33$exports = require("./zh-TW.main.js");
require("./main.css");
var $IPaYO$reactspectrumutils = require("@react-spectrum/utils");
var $IPaYO$reactariafocus = require("@react-aria/focus");
var $IPaYO$reactariautils = require("@react-aria/utils");
var $IPaYO$reactspectrumprogress = require("@react-spectrum/progress");
var $IPaYO$react = require("react");
var $IPaYO$reactspectrumtext = require("@react-spectrum/text");
var $IPaYO$reactariabutton = require("@react-aria/button");
var $IPaYO$reactariainteractions = require("@react-aria/interactions");
var $IPaYO$reactariai18n = require("@react-aria/i18n");
var $IPaYO$reactspectrumprovider = require("@react-spectrum/provider");
var $IPaYO$spectrumiconsuiCornerTriangle = require("@spectrum-icons/ui/CornerTriangle");
var $IPaYO$spectrumiconsuiCrossSmall = require("@spectrum-icons/ui/CrossSmall");
var $IPaYO$reactstatelytoggle = require("@react-stately/toggle");


function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Button", () => $842a195ed1671b15$export$353f5b6fc5456de1);
$parcel$export(module.exports, "ActionButton", () => $65dbde0e8e7aba71$export$cfc7921d29ef7b80);
$parcel$export(module.exports, "FieldButton", () => $5931ec1ea1ec1b0d$export$47dc48f595b075da);
$parcel$export(module.exports, "LogicButton", () => $3bd294eda342b0c4$export$9b0b80fed00ba8b1);
$parcel$export(module.exports, "ClearButton", () => $9c84a434f1c9da63$export$13ec83e50bf04290);
$parcel$export(module.exports, "ToggleButton", () => $b5c927da7d8f1be6$export$d2b052e7b4be1756);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

var $b2da92ebd9a2bc00$exports = {};


































$b2da92ebd9a2bc00$exports = {
    "ar-AE": $8dfbe9f11d8f0c18$exports,
    "bg-BG": $f1d09da51b7863de$exports,
    "cs-CZ": $a938a0b253d9f247$exports,
    "da-DK": $cf0c54eea40f6fcb$exports,
    "de-DE": $4748cd40594ea272$exports,
    "el-GR": $650c1cff0501e881$exports,
    "en-US": $e400898e815c69b7$exports,
    "es-ES": $6d85accad7f1b1b0$exports,
    "et-EE": $8bd503882fd613bd$exports,
    "fi-FI": $23d05f8457cd40f2$exports,
    "fr-FR": $e63063aea75f1c21$exports,
    "he-IL": $9906c3303b98f896$exports,
    "hr-HR": $8e94047ff7a4f28e$exports,
    "hu-HU": $29cc0bceb80e8c91$exports,
    "it-IT": $986fbd55bef3d71d$exports,
    "ja-JP": $7f1ef5a9cae227cd$exports,
    "ko-KR": $9d9c1e7e8c3604b6$exports,
    "lt-LT": $bb0c9e2a421c29e6$exports,
    "lv-LV": $d4bcb4a7735702a6$exports,
    "nb-NO": $bbac6f483322e4cb$exports,
    "nl-NL": $2e38f272eed1ee93$exports,
    "pl-PL": $efdabc3b4ba6ae3a$exports,
    "pt-BR": $32c199c1c0b84ea6$exports,
    "pt-PT": $4525387e786c4e45$exports,
    "ro-RO": $946eea38cb062cf0$exports,
    "ru-RU": $fe41a7461b3fc080$exports,
    "sk-SK": $2d1b4cc9945c3226$exports,
    "sl-SI": $5d18d41ac4266fa7$exports,
    "sr-SP": $b2209169f376772d$exports,
    "sv-SE": $d3ab0389431476ca$exports,
    "tr-TR": $33bce89e4fc9865a$exports,
    "uk-UA": $a803ef1dfa0d114e$exports,
    "zh-CN": $732189c02a0cbea0$exports,
    "zh-TW": $feaa03429c1f5a33$exports
};





var $afc5a4514aca2340$exports = {};

$parcel$export($afc5a4514aca2340$exports, "focus-ring", () => $afc5a4514aca2340$export$f39a09f249340e2a, (v) => $afc5a4514aca2340$export$f39a09f249340e2a = v);
$parcel$export($afc5a4514aca2340$exports, "i18nFontFamily", () => $afc5a4514aca2340$export$8c4ee2c50c22c514, (v) => $afc5a4514aca2340$export$8c4ee2c50c22c514 = v);
$parcel$export($afc5a4514aca2340$exports, "is-active", () => $afc5a4514aca2340$export$20fd0f7cd4e6112f, (v) => $afc5a4514aca2340$export$20fd0f7cd4e6112f = v);
$parcel$export($afc5a4514aca2340$exports, "is-disabled", () => $afc5a4514aca2340$export$d35bc1e505d1ebbf, (v) => $afc5a4514aca2340$export$d35bc1e505d1ebbf = v);
$parcel$export($afc5a4514aca2340$exports, "is-focused", () => $afc5a4514aca2340$export$e7dc768d35940237, (v) => $afc5a4514aca2340$export$e7dc768d35940237 = v);
$parcel$export($afc5a4514aca2340$exports, "is-hovered", () => $afc5a4514aca2340$export$b8813cd5d7824ce7, (v) => $afc5a4514aca2340$export$b8813cd5d7824ce7 = v);
$parcel$export($afc5a4514aca2340$exports, "is-open", () => $afc5a4514aca2340$export$a9781837241c946d, (v) => $afc5a4514aca2340$export$a9781837241c946d = v);
$parcel$export($afc5a4514aca2340$exports, "is-placeholder", () => $afc5a4514aca2340$export$e5b2f5233e4e5194, (v) => $afc5a4514aca2340$export$e5b2f5233e4e5194 = v);
$parcel$export($afc5a4514aca2340$exports, "is-selected", () => $afc5a4514aca2340$export$1e0fb04f31d3c22a, (v) => $afc5a4514aca2340$export$1e0fb04f31d3c22a = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-BaseButton", () => $afc5a4514aca2340$export$f7b415378ab4f839, (v) => $afc5a4514aca2340$export$f7b415378ab4f839 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-FocusRing-ring", () => $afc5a4514aca2340$export$4109102f950813a6, (v) => $afc5a4514aca2340$export$4109102f950813a6 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-FocusRing", () => $afc5a4514aca2340$export$24c7f46a6e3605dd, (v) => $afc5a4514aca2340$export$24c7f46a6e3605dd = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ActionButton", () => $afc5a4514aca2340$export$53da69f51b770d3, (v) => $afc5a4514aca2340$export$53da69f51b770d3 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ActionButton--emphasized", () => $afc5a4514aca2340$export$5aabb61a966e266f, (v) => $afc5a4514aca2340$export$5aabb61a966e266f = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ActionButton--quiet", () => $afc5a4514aca2340$export$40e1f009544bacf3, (v) => $afc5a4514aca2340$export$40e1f009544bacf3 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ActionButton--staticBlack", () => $afc5a4514aca2340$export$8f7c9db66978d20f, (v) => $afc5a4514aca2340$export$8f7c9db66978d20f = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ActionButton--staticColor", () => $afc5a4514aca2340$export$29b7b5d6a5a497c5, (v) => $afc5a4514aca2340$export$29b7b5d6a5a497c5 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ActionButton--staticWhite", () => $afc5a4514aca2340$export$ccc536e1adbdc059, (v) => $afc5a4514aca2340$export$ccc536e1adbdc059 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ActionButton-hold", () => $afc5a4514aca2340$export$4a4f01c443eff992, (v) => $afc5a4514aca2340$export$4a4f01c443eff992 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ActionButton-label", () => $afc5a4514aca2340$export$ebb4635c66bacdb9, (v) => $afc5a4514aca2340$export$ebb4635c66bacdb9 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-Button", () => $afc5a4514aca2340$export$1db4cca5b4ade39a, (v) => $afc5a4514aca2340$export$1db4cca5b4ade39a = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-Button--iconOnly", () => $afc5a4514aca2340$export$c1d306ec0fae3838, (v) => $afc5a4514aca2340$export$c1d306ec0fae3838 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-Button--overBackground", () => $afc5a4514aca2340$export$5cda1128f9112d2e, (v) => $afc5a4514aca2340$export$5cda1128f9112d2e = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-Button--pending", () => $afc5a4514aca2340$export$54592d0984f245d5, (v) => $afc5a4514aca2340$export$54592d0984f245d5 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-Button-circleLoader", () => $afc5a4514aca2340$export$76d4531f0342bbe0, (v) => $afc5a4514aca2340$export$76d4531f0342bbe0 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-Button-label", () => $afc5a4514aca2340$export$c081ba562e204942, (v) => $afc5a4514aca2340$export$c081ba562e204942 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ClearButton", () => $afc5a4514aca2340$export$b752ce409e5660c, (v) => $afc5a4514aca2340$export$b752ce409e5660c = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ClearButton--overBackground", () => $afc5a4514aca2340$export$116f2fccf941d34d, (v) => $afc5a4514aca2340$export$116f2fccf941d34d = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-ClearButton--small", () => $afc5a4514aca2340$export$49280ffa5236ba8b, (v) => $afc5a4514aca2340$export$49280ffa5236ba8b = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-FieldButton", () => $afc5a4514aca2340$export$a024c99ecf6b5741, (v) => $afc5a4514aca2340$export$a024c99ecf6b5741 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-FieldButton--invalid", () => $afc5a4514aca2340$export$c9f503f672e8a3c1, (v) => $afc5a4514aca2340$export$c9f503f672e8a3c1 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-FocusRing--quiet", () => $afc5a4514aca2340$export$2927016961429360, (v) => $afc5a4514aca2340$export$2927016961429360 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-FieldButton--quiet", () => $afc5a4514aca2340$export$5ca7b274e4fff9fc, (v) => $afc5a4514aca2340$export$5ca7b274e4fff9fc = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-Icon", () => $afc5a4514aca2340$export$d374b04f30360026, (v) => $afc5a4514aca2340$export$d374b04f30360026 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-LogicButton", () => $afc5a4514aca2340$export$36ae67a131730482, (v) => $afc5a4514aca2340$export$36ae67a131730482 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-LogicButton--and", () => $afc5a4514aca2340$export$e52daab43a62c528, (v) => $afc5a4514aca2340$export$e52daab43a62c528 = v);
$parcel$export($afc5a4514aca2340$exports, "spectrum-LogicButton--or", () => $afc5a4514aca2340$export$da76be6126aaa633, (v) => $afc5a4514aca2340$export$da76be6126aaa633 = v);
var $afc5a4514aca2340$export$f39a09f249340e2a;
var $afc5a4514aca2340$export$8c4ee2c50c22c514;
var $afc5a4514aca2340$export$20fd0f7cd4e6112f;
var $afc5a4514aca2340$export$d35bc1e505d1ebbf;
var $afc5a4514aca2340$export$e7dc768d35940237;
var $afc5a4514aca2340$export$b8813cd5d7824ce7;
var $afc5a4514aca2340$export$a9781837241c946d;
var $afc5a4514aca2340$export$e5b2f5233e4e5194;
var $afc5a4514aca2340$export$1e0fb04f31d3c22a;
var $afc5a4514aca2340$export$f7b415378ab4f839;
var $afc5a4514aca2340$export$4109102f950813a6;
var $afc5a4514aca2340$export$24c7f46a6e3605dd;
var $afc5a4514aca2340$export$53da69f51b770d3;
var $afc5a4514aca2340$export$5aabb61a966e266f;
var $afc5a4514aca2340$export$40e1f009544bacf3;
var $afc5a4514aca2340$export$8f7c9db66978d20f;
var $afc5a4514aca2340$export$29b7b5d6a5a497c5;
var $afc5a4514aca2340$export$ccc536e1adbdc059;
var $afc5a4514aca2340$export$4a4f01c443eff992;
var $afc5a4514aca2340$export$ebb4635c66bacdb9;
var $afc5a4514aca2340$export$1db4cca5b4ade39a;
var $afc5a4514aca2340$export$c1d306ec0fae3838;
var $afc5a4514aca2340$export$5cda1128f9112d2e;
var $afc5a4514aca2340$export$54592d0984f245d5;
var $afc5a4514aca2340$export$76d4531f0342bbe0;
var $afc5a4514aca2340$export$c081ba562e204942;
var $afc5a4514aca2340$export$b752ce409e5660c;
var $afc5a4514aca2340$export$116f2fccf941d34d;
var $afc5a4514aca2340$export$49280ffa5236ba8b;
var $afc5a4514aca2340$export$a024c99ecf6b5741;
var $afc5a4514aca2340$export$c9f503f672e8a3c1;
var $afc5a4514aca2340$export$2927016961429360;
var $afc5a4514aca2340$export$5ca7b274e4fff9fc;
var $afc5a4514aca2340$export$d374b04f30360026;
var $afc5a4514aca2340$export$36ae67a131730482;
var $afc5a4514aca2340$export$e52daab43a62c528;
var $afc5a4514aca2340$export$da76be6126aaa633;
$afc5a4514aca2340$export$f39a09f249340e2a = `Dniwja_focus-ring`;
$afc5a4514aca2340$export$8c4ee2c50c22c514 = `Dniwja_i18nFontFamily`;
$afc5a4514aca2340$export$20fd0f7cd4e6112f = `Dniwja_is-active`;
$afc5a4514aca2340$export$d35bc1e505d1ebbf = `Dniwja_is-disabled`;
$afc5a4514aca2340$export$e7dc768d35940237 = `Dniwja_is-focused`;
$afc5a4514aca2340$export$b8813cd5d7824ce7 = `Dniwja_is-hovered`;
$afc5a4514aca2340$export$a9781837241c946d = `Dniwja_is-open`;
$afc5a4514aca2340$export$e5b2f5233e4e5194 = `Dniwja_is-placeholder`;
$afc5a4514aca2340$export$1e0fb04f31d3c22a = `Dniwja_is-selected`;
$afc5a4514aca2340$export$f7b415378ab4f839 = `Dniwja_spectrum-BaseButton ${$afc5a4514aca2340$export$8c4ee2c50c22c514}`;
$afc5a4514aca2340$export$4109102f950813a6 = `Dniwja_spectrum-FocusRing-ring`;
$afc5a4514aca2340$export$24c7f46a6e3605dd = `Dniwja_spectrum-FocusRing ${$afc5a4514aca2340$export$4109102f950813a6}`;
$afc5a4514aca2340$export$53da69f51b770d3 = `Dniwja_spectrum-ActionButton ${$afc5a4514aca2340$export$f7b415378ab4f839} ${$afc5a4514aca2340$export$24c7f46a6e3605dd}`;
$afc5a4514aca2340$export$5aabb61a966e266f = `Dniwja_spectrum-ActionButton--emphasized`;
$afc5a4514aca2340$export$40e1f009544bacf3 = `Dniwja_spectrum-ActionButton--quiet`;
$afc5a4514aca2340$export$8f7c9db66978d20f = `Dniwja_spectrum-ActionButton--staticBlack`;
$afc5a4514aca2340$export$29b7b5d6a5a497c5 = `Dniwja_spectrum-ActionButton--staticColor`;
$afc5a4514aca2340$export$ccc536e1adbdc059 = `Dniwja_spectrum-ActionButton--staticWhite`;
$afc5a4514aca2340$export$4a4f01c443eff992 = `Dniwja_spectrum-ActionButton-hold`;
$afc5a4514aca2340$export$ebb4635c66bacdb9 = `Dniwja_spectrum-ActionButton-label`;
$afc5a4514aca2340$export$1db4cca5b4ade39a = `Dniwja_spectrum-Button ${$afc5a4514aca2340$export$f7b415378ab4f839} ${$afc5a4514aca2340$export$24c7f46a6e3605dd}`;
$afc5a4514aca2340$export$c1d306ec0fae3838 = `Dniwja_spectrum-Button--iconOnly`;
$afc5a4514aca2340$export$5cda1128f9112d2e = `Dniwja_spectrum-Button--overBackground`;
$afc5a4514aca2340$export$54592d0984f245d5 = `Dniwja_spectrum-Button--pending`;
$afc5a4514aca2340$export$76d4531f0342bbe0 = `Dniwja_spectrum-Button-circleLoader`;
$afc5a4514aca2340$export$c081ba562e204942 = `Dniwja_spectrum-Button-label`;
$afc5a4514aca2340$export$b752ce409e5660c = `Dniwja_spectrum-ClearButton ${$afc5a4514aca2340$export$f7b415378ab4f839} ${$afc5a4514aca2340$export$24c7f46a6e3605dd}`;
$afc5a4514aca2340$export$116f2fccf941d34d = `Dniwja_spectrum-ClearButton--overBackground`;
$afc5a4514aca2340$export$49280ffa5236ba8b = `Dniwja_spectrum-ClearButton--small`;
$afc5a4514aca2340$export$a024c99ecf6b5741 = `Dniwja_spectrum-FieldButton ${$afc5a4514aca2340$export$f7b415378ab4f839} ${$afc5a4514aca2340$export$24c7f46a6e3605dd}`;
$afc5a4514aca2340$export$c9f503f672e8a3c1 = `Dniwja_spectrum-FieldButton--invalid`;
$afc5a4514aca2340$export$2927016961429360 = `Dniwja_spectrum-FocusRing--quiet`;
$afc5a4514aca2340$export$5ca7b274e4fff9fc = `Dniwja_spectrum-FieldButton--quiet ${$afc5a4514aca2340$export$2927016961429360}`;
$afc5a4514aca2340$export$d374b04f30360026 = `Dniwja_spectrum-Icon`;
$afc5a4514aca2340$export$36ae67a131730482 = `Dniwja_spectrum-LogicButton ${$afc5a4514aca2340$export$f7b415378ab4f839} ${$afc5a4514aca2340$export$24c7f46a6e3605dd}`;
$afc5a4514aca2340$export$e52daab43a62c528 = `Dniwja_spectrum-LogicButton--and`;
$afc5a4514aca2340$export$da76be6126aaa633 = `Dniwja_spectrum-LogicButton--or`;







function $842a195ed1671b15$var$disablePendingProps(props) {
    // Don't allow interaction while isPending is true
    if (props.isPending) {
        props.onPress = undefined;
        props.onPressStart = undefined;
        props.onPressEnd = undefined;
        props.onPressChange = undefined;
        props.onPressUp = undefined;
        props.onKeyDown = undefined;
        props.onKeyUp = undefined;
        props.onClick = undefined;
        props.href = undefined;
    }
    return props;
}
function $842a195ed1671b15$var$Button(props, ref) {
    var _buttonProps_arialabelledby;
    props = (0, $IPaYO$reactspectrumprovider.useProviderProps)(props);
    props = (0, $IPaYO$reactspectrumutils.useSlotProps)(props, "button");
    props = $842a195ed1671b15$var$disablePendingProps(props);
    let { elementType: Element = "button", children: children, variant: variant, style: style = variant === "accent" || variant === "cta" ? "fill" : "outline", staticColor: staticColor, isDisabled: isDisabled, isPending: isPending, autoFocus: autoFocus, ...otherProps } = props;
    let domRef = (0, $IPaYO$reactspectrumutils.useFocusableRef)(ref);
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $IPaYO$reactariabutton.useButton)(props, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $IPaYO$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let [isFocused, onFocusChange] = (0, $IPaYO$react.useState)(false);
    let { focusProps: focusProps } = (0, $IPaYO$reactariainteractions.useFocus)({
        onFocusChange: onFocusChange,
        isDisabled: isDisabled
    });
    let stringFormatter = (0, $IPaYO$reactariai18n.useLocalizedStringFormatter)((0, (/*@__PURE__*/$parcel$interopDefault($b2da92ebd9a2bc00$exports))), "@react-spectrum/button");
    let { styleProps: styleProps } = (0, $IPaYO$reactspectrumutils.useStyleProps)(otherProps);
    let hasLabel = (0, $IPaYO$reactspectrumutils.useHasChild)(`.${(0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports)))["spectrum-Button-label"]}`, domRef);
    let hasIcon = (0, $IPaYO$reactspectrumutils.useHasChild)(`.${(0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports)))["spectrum-Icon"]}`, domRef);
    // an aria label will block children and their labels from being read, this is undesirable for pending state
    let hasAriaLabel = !!buttonProps["aria-label"] || !!buttonProps["aria-labelledby"];
    let [isProgressVisible, setIsProgressVisible] = (0, $IPaYO$react.useState)(false);
    let backupButtonId = (0, $IPaYO$reactariautils.useId)();
    let buttonId = buttonProps.id || backupButtonId;
    let iconId = (0, $IPaYO$reactariautils.useId)();
    let textId = (0, $IPaYO$reactariautils.useId)();
    let spinnerId = (0, $IPaYO$reactariautils.useId)();
    (0, $IPaYO$react.useEffect)(()=>{
        let timeout;
        if (isPending) // Start timer when isPending is set to true.
        timeout = setTimeout(()=>{
            setIsProgressVisible(true);
        }, 1000);
        else // Exit loading state when isPending is set to false. */
        setIsProgressVisible(false);
        return ()=>{
            // Clean up on unmount or when user removes isPending prop before entering loading state.
            clearTimeout(timeout);
        };
    }, [
        isPending
    ]);
    if (variant === "cta") variant = "accent";
    else if (variant === "overBackground") {
        variant = "primary";
        staticColor = "white";
    }
    const isPendingAriaLiveLabel = `${hasAriaLabel ? buttonProps["aria-label"] : ""} ${stringFormatter.format("pending")}`.trim();
    var _buttonProps_arialabelledby_replace;
    const isPendingAriaLiveLabelledby = hasAriaLabel ? (_buttonProps_arialabelledby_replace = (_buttonProps_arialabelledby = buttonProps["aria-labelledby"]) === null || _buttonProps_arialabelledby === void 0 ? void 0 : _buttonProps_arialabelledby.replace(buttonId, spinnerId)) !== null && _buttonProps_arialabelledby_replace !== void 0 ? _buttonProps_arialabelledby_replace : spinnerId : `${hasIcon ? iconId : ""} ${hasLabel ? textId : ""} ${spinnerId}`.trim();
    let ariaLive = "polite";
    if ((0, $IPaYO$reactariautils.isAppleDevice)() && (!hasAriaLabel || (0, $IPaYO$reactariautils.isFirefox)())) ariaLive = "off";
    return /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactariafocus.FocusRing), {
        focusRingClass: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement(Element, {
        ...styleProps,
        ...(0, $IPaYO$reactariautils.mergeProps)(buttonProps, hoverProps, focusProps),
        id: buttonId,
        ref: domRef,
        "data-variant": variant,
        "data-style": style,
        "data-static-color": staticColor || undefined,
        "aria-disabled": isPending ? "true" : undefined,
        "aria-label": isPending ? isPendingAriaLiveLabel : buttonProps["aria-label"],
        "aria-labelledby": isPending ? isPendingAriaLiveLabelledby : buttonProps["aria-labelledby"],
        className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-Button", {
            "spectrum-Button--iconOnly": hasIcon && !hasLabel,
            "is-disabled": isDisabled || isProgressVisible,
            "is-active": isPressed,
            "is-hovered": isHovered,
            "spectrum-Button--pending": isProgressVisible
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactspectrumutils.SlotProvider), {
        slots: {
            icon: {
                id: iconId,
                size: "S",
                UNSAFE_className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-Icon")
            },
            text: {
                id: textId,
                UNSAFE_className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-Button-label")
            }
        }
    }, typeof children === "string" ? /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactspectrumtext.Text), null, children) : children, isPending && /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("div", {
        "aria-hidden": "true",
        style: {
            visibility: isProgressVisible ? "visible" : "hidden"
        },
        className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-Button-circleLoader")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactspectrumprogress.ProgressCircle), {
        "aria-label": isPendingAriaLiveLabel,
        isIndeterminate: true,
        size: "S",
        staticColor: staticColor
    })), isPending && /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, ($parcel$interopDefault($IPaYO$react))).Fragment, null, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("div", {
        "aria-live": isFocused ? ariaLive : "off"
    }, isProgressVisible && /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("div", {
        role: "img",
        "aria-labelledby": isPendingAriaLiveLabelledby
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("div", {
        id: spinnerId,
        role: "img",
        "aria-label": isPendingAriaLiveLabel
    })))));
}
/**
 * Buttons allow users to perform an action or to navigate to another page.
 * They have multiple styles for various needs, and are ideal for calling attention to
 * where a user needs to do something in order to move forward in a flow.
 */ let $842a195ed1671b15$export$353f5b6fc5456de1 = /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).forwardRef($842a195ed1671b15$var$Button);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 









function $65dbde0e8e7aba71$var$ActionButton(props, ref) {
    props = (0, $IPaYO$reactspectrumprovider.useProviderProps)(props);
    props = (0, $IPaYO$reactspectrumutils.useSlotProps)(props, "actionButton");
    let { isQuiet: isQuiet, isDisabled: isDisabled, staticColor: staticColor, children: children, autoFocus: autoFocus, holdAffordance: // @ts-ignore (private)
    holdAffordance, ...otherProps } = props;
    let domRef = (0, $IPaYO$reactspectrumutils.useFocusableRef)(ref);
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $IPaYO$reactariabutton.useButton)(props, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $IPaYO$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps } = (0, $IPaYO$reactspectrumutils.useStyleProps)(otherProps);
    let isTextOnly = (0, ($parcel$interopDefault($IPaYO$react))).Children.toArray(props.children).every((c)=>!/*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).isValidElement(c));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactariafocus.FocusRing), {
        focusRingClass: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("button", {
        ...styleProps,
        ...(0, $IPaYO$reactariautils.mergeProps)(buttonProps, hoverProps),
        ref: domRef,
        className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-ActionButton", {
            "spectrum-ActionButton--quiet": isQuiet,
            "spectrum-ActionButton--staticColor": !!staticColor,
            "spectrum-ActionButton--staticWhite": staticColor === "white",
            "spectrum-ActionButton--staticBlack": staticColor === "black",
            "is-active": isPressed,
            "is-disabled": isDisabled,
            "is-hovered": isHovered
        }, styleProps.className)
    }, holdAffordance && /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, ($parcel$interopDefault($IPaYO$spectrumiconsuiCornerTriangle))), {
        UNSAFE_className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-ActionButton-hold")
    }), /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactspectrumutils.SlotProvider), {
        slots: {
            icon: {
                size: "S",
                UNSAFE_className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-Icon")
            },
            text: {
                UNSAFE_className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-ActionButton-label")
            }
        }
    }, typeof children === "string" || isTextOnly ? /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactspectrumtext.Text), null, children) : children)));
}
/**
 * ActionButtons allow users to perform an action.
 * They’re used for similar, task-based options within a workflow, and are ideal for interfaces where buttons aren’t meant to draw a lot of attention.
 */ let $65dbde0e8e7aba71$export$cfc7921d29ef7b80 = /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).forwardRef($65dbde0e8e7aba71$var$ActionButton);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






// @private
function $5931ec1ea1ec1b0d$var$FieldButton(props, ref) {
    props = (0, $IPaYO$reactspectrumutils.useSlotProps)(props, "button");
    let { isQuiet: isQuiet, isDisabled: isDisabled, validationState: validationState, isInvalid: isInvalid, children: children, autoFocus: autoFocus, isActive: isActive, focusRingClass: focusRingClass, ...otherProps } = props;
    let domRef = (0, $IPaYO$reactspectrumutils.useFocusableRef)(ref);
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $IPaYO$reactariabutton.useButton)(props, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $IPaYO$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps } = (0, $IPaYO$reactspectrumutils.useStyleProps)(otherProps);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactariafocus.FocusRing), {
        focusRingClass: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "focus-ring", focusRingClass),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("button", {
        ...(0, $IPaYO$reactariautils.mergeProps)(buttonProps, hoverProps),
        ref: domRef,
        className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-FieldButton", {
            "spectrum-FieldButton--quiet": isQuiet,
            "is-active": isActive || isPressed,
            "is-disabled": isDisabled,
            "spectrum-FieldButton--invalid": isInvalid || validationState === "invalid",
            "is-hovered": isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactspectrumutils.SlotProvider), {
        slots: {
            icon: {
                size: "S",
                UNSAFE_className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-Icon")
            }
        }
    }, children)));
}
let $5931ec1ea1ec1b0d$export$47dc48f595b075da = /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).forwardRef($5931ec1ea1ec1b0d$var$FieldButton);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







function $3bd294eda342b0c4$var$LogicButton(props, ref) {
    props = (0, $IPaYO$reactspectrumprovider.useProviderProps)(props);
    let { variant: variant, children: children, isDisabled: isDisabled, autoFocus: autoFocus, ...otherProps } = props;
    let domRef = (0, $IPaYO$reactspectrumutils.useFocusableRef)(ref);
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $IPaYO$reactariabutton.useButton)(props, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $IPaYO$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps } = (0, $IPaYO$reactspectrumutils.useStyleProps)(otherProps);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactariafocus.FocusRing), {
        focusRingClass: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("button", {
        ...styleProps,
        ...(0, $IPaYO$reactariautils.mergeProps)(buttonProps, hoverProps),
        ref: domRef,
        className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-LogicButton", {
            [`spectrum-LogicButton--${variant}`]: variant,
            "is-disabled": isDisabled,
            "is-active": isPressed,
            "is-hovered": isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("span", {
        className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-Button-label")
    }, children)));
}
/**
 * A LogicButton displays an operator within a boolean logic sequence.
 */ let $3bd294eda342b0c4$export$9b0b80fed00ba8b1 = /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).forwardRef($3bd294eda342b0c4$var$LogicButton);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 







function $9c84a434f1c9da63$var$ClearButton(props, ref) {
    let { children: children = /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, ($parcel$interopDefault($IPaYO$spectrumiconsuiCrossSmall))), {
        UNSAFE_className: (0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports)))["spectrum-Icon"]
    }), focusClassName: focusClassName, variant: variant, autoFocus: autoFocus, isDisabled: isDisabled, preventFocus: preventFocus, elementType: elementType = preventFocus ? "div" : "button", ...otherProps } = props;
    let domRef = (0, $IPaYO$reactspectrumutils.useFocusableRef)(ref);
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $IPaYO$reactariabutton.useButton)({
        ...props,
        elementType: elementType
    }, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $IPaYO$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps } = (0, $IPaYO$reactspectrumutils.useStyleProps)(otherProps);
    // For cases like the clear button in a search field, remove the tabIndex so
    // iOS 14 with VoiceOver doesn't focus the button and hide the keyboard when
    // moving the cursor over the clear button.
    if (preventFocus) delete buttonProps.tabIndex;
    let ElementType = elementType;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactariafocus.FocusRing), {
        focusRingClass: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "focus-ring", focusClassName),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement(ElementType, {
        ...styleProps,
        ...(0, $IPaYO$reactariautils.mergeProps)(buttonProps, hoverProps),
        ref: domRef,
        className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-ClearButton", {
            [`spectrum-ClearButton--${variant}`]: variant,
            "is-disabled": isDisabled,
            "is-active": isPressed,
            "is-hovered": isHovered
        }, styleProps.className)
    }, children));
}
let $9c84a434f1c9da63$export$13ec83e50bf04290 = /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).forwardRef($9c84a434f1c9da63$var$ClearButton);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 









function $b5c927da7d8f1be6$var$ToggleButton(props, ref) {
    props = (0, $IPaYO$reactspectrumprovider.useProviderProps)(props);
    let { isQuiet: isQuiet, isDisabled: isDisabled, isEmphasized: isEmphasized, staticColor: staticColor, children: children, autoFocus: autoFocus, ...otherProps } = props;
    let domRef = (0, $IPaYO$reactspectrumutils.useFocusableRef)(ref);
    let state = (0, $IPaYO$reactstatelytoggle.useToggleState)(props);
    let { buttonProps: buttonProps, isPressed: isPressed } = (0, $IPaYO$reactariabutton.useToggleButton)(props, state, domRef);
    let { hoverProps: hoverProps, isHovered: isHovered } = (0, $IPaYO$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let { styleProps: styleProps } = (0, $IPaYO$reactspectrumutils.useStyleProps)(otherProps);
    let isTextOnly = (0, ($parcel$interopDefault($IPaYO$react))).Children.toArray(props.children).every((c)=>!/*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).isValidElement(c));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactariafocus.FocusRing), {
        focusRingClass: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement("button", {
        ...styleProps,
        ...(0, $IPaYO$reactariautils.mergeProps)(buttonProps, hoverProps),
        ref: domRef,
        className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-ActionButton", {
            "spectrum-ActionButton--quiet": isQuiet,
            "spectrum-ActionButton--emphasized": isEmphasized,
            "spectrum-ActionButton--staticColor": !!staticColor,
            "spectrum-ActionButton--staticWhite": staticColor === "white",
            "spectrum-ActionButton--staticBlack": staticColor === "black",
            "is-active": isPressed,
            "is-disabled": isDisabled,
            "is-hovered": isHovered,
            "is-selected": state.isSelected
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactspectrumutils.SlotProvider), {
        slots: {
            icon: {
                size: "S",
                UNSAFE_className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-Icon")
            },
            text: {
                UNSAFE_className: (0, $IPaYO$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($afc5a4514aca2340$exports))), "spectrum-ActionButton-label")
            }
        }
    }, typeof children === "string" || isTextOnly ? /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).createElement((0, $IPaYO$reactspectrumtext.Text), null, children) : children)));
}
/**
 * ToggleButtons allow users to toggle a selection on or off, for example
 * switching between two states or modes.
 */ let $b5c927da7d8f1be6$export$d2b052e7b4be1756 = /*#__PURE__*/ (0, ($parcel$interopDefault($IPaYO$react))).forwardRef($b5c927da7d8f1be6$var$ToggleButton);
